# Introduction 


## Microbiome

> Microbiota are the range of microorganisms that may be commensal, symbiotic, or pathogenic found in and on all multicellular organisms, including plants. Microbiota include bacteria, archaea, protists, fungi, and viruses, and have been found to be crucial for immunologic, hormonal, and metabolic homeostasis of their host.

> The term microbiome describes either the collective genomes of the microbes that reside in an ecological niche or within the microbes themselves.

## Sequencing technology 

The **next generation sequencing (NGS)** technology have flourished the microbial research. More and more uncultured microbiota have been sequenced and identified, and their roles on host also have been investigated. According to the sequenced DNA parts, there are several methods to obtain the bacteria.

### Amplicon sequencing

It also named 16s sequencing, uses the V3/V4 of bacteria 16s rRNA to sequence the bacteria. The in-house 16s sequencing pipeline uses QIIME2 platform wrapped DADA2 algorithm to acquire the amplicon sequence variants which is the same meaning of OTUs (operational taxonomic units).

### Whole genome shotgun sequencing

It is also short for metagenomic sequencing, could detect the whole DNA of the microbial community. The in-house metagenomic sequencing pipeline uses the metaphlan2 or metaphlan3 algorithm which is based on marker genes of microbiome to obtain the microbial profiles.


**CAUTIONS: Both of two workflow have advantages and disadvantages, one of the disadvantages is that we can only detect or identify the annotated taxonomy by the reference database.**

## The gut microbiome: Relationships with disease and opportunities for therapy

Here, we cited this article [@durack2019gut] to give users an abstract.

> Over the past decade, our view of human-associated microbes has expanded beyond that of a few species toward an appreciation of the diverse and niche-specialized microbial communities that develop in the human host with chronological age. The largest reservoir of microbes exists in the distal gastrointestinal tract, both in the lumen, where microbes facilitate primary and secondary metabolism, and on mucosal surfaces, where they interact with host immune cell populations. While local microbial-driven immunomodulation in the gut is well described, more recent studies have demonstrated a role for the gut microbiome in influencing remote organs and mucosal and hematopoietic immune function. Unsurprisingly, therefore, perturbation to the composition and function of the gut microbiota has been associated with chronic diseases ranging from gastrointestinal inflammatory and metabolic conditions to neurological, cardiovascular, and respiratory illnesses. Considerable effort is currently focused on understanding the natural history of microbiome development in humans in the context of health outcomes, in parallel with improving our knowledge of microbiomeâ€“host molecular interactions. These efforts ultimately aim to develop effective approaches to rehabilitate perturbed human microbial ecosystems as a means to restore health or prevent disease. This review details the role of the gut microbiome in modulating host health with a focus on immunomodulation and discusses strategies for manipulating the gut microbiome for the management or prevention of chronic inflammatory conditions.


## Data Analysis apply for discovering the potential microbial biomarkers 


In this tutorial, we focus on differential analysis with the step-by-step procedures by using the R programming language. We also briefly introduce the concepts and principals of the statistical methods before the applications and give conclusions on the results.

The following table shows more than 10 differential analysis methods:


```{r, warning=FALSE, echo=FALSE}
DA_tools <- data.frame(`Tool(version)`=c("ALDEx2  (1.26.0)", 
                                         "limma  voom (3.50.1)",
                                         "mbzinb  (0.2)",
                                         "omnibus  (0.2)",
                                         "RAIDA  (1.0)",
                                         "Wilcox(rare/CLR)",
                                         "LEfSe",
                                         "t-test  (rare)",
                                         "metagenomeSeq  (1.36.0)",
                                         "DESeq2  (1.34.0)",
                                         "edgeR  (3.36.0)",
                                         "ANCOM-II  (2.1)",
                                         "Corncob  (0.2.0)",
                                         "MaAslin2  (1.8.0)"),
                       Input=c("Counts",
                               "Counts/Relative",
                               "Counts",
                               "Counts",
                               "Counts",
                               "Counts/Relative",
                               "Rarefied  Counts/Relative",
                               "Counts/Relative",
                               "Counts",
                               "Counts",
                               "Counts",
                               "Counts/Relative",
                               "Counts",
                               "Counts/Relative"),
                       Normalization=c("None",
                                       "None/TMM",
                                       "RLE",
                                       "GMPR(Geometric  Mean of Pairwise Ratios)",
                                       "None",
                                       "None",
                                       "TSS",
                                       "None",
                                       "CSS",
                                       "RLE",
                                       "RLE/TMM",
                                       "None",
                                       "None",
                                       "None/TSS"),
                       Transformation=c("CLR",
                                        "Log; Precision weighting",
                                        "None",
                                        "None",
                                        "zero-inflated Log",
                                        "None/CLR",
                                        "None",
                                        "None",
                                        "Log",
                                        "None",
                                        "None",
                                        "ALR",
                                        "None",
                                        "AST"),
                       Distribution=c("Dirichlet-multinormial",
                                      "Normal",
                                      "Zero-inflated negative binomial",
                                      "Zero-inflated negative binomial",
                                      "Modified t-test",
                                      "Non-parametric",
                                      "Non-parametric",
                                      "Normal",
                                      "Zero-inflated (log) Normal",
                                      "Negative binomial",
                                      "Negative binomial",
                                      "Non-parametric",
                                      "Beta-binomial",
                                      "Normal"),
                       MicrobialData=c("16s",
                                       "16s/MGS",
                                       "16s",
                                       "16s",
                                       "16s",
                                       "16s/MGS",
                                       "16s/MGS",
                                       "16s/MGS",
                                       "16s",
                                       "16s",
                                       "16s",
                                       "16s/MGS",
                                       "16s",
                                       "16s/MGS"))

knitr::kable(DA_tools, longtable = TRUE, booktabs = TRUE,
  caption = 'Differential analysis tools in XMAS.',
  format = "html") %>% 
  kableExtra::kable_styling() %>%
  kableExtra::scroll_box(width = "100%", box_css = "border: 0px;") 
```

We suggest you using the reasonable method to fit your own microbiota data.


## Book

This tutorial was organized according to the **XMAS 2.0** functions. 

* In the beginning three chapters, we specially provided the basic requirements of bioinformatics and overview on microbiota, and the installation of XMAS. 

* In **Chapter 4**, we convert the results from in-house pipeline into phyloseq-class object for downstream data analysis. 

* In **Chapter 5** provided the functions to evaluate the data quality.

* In **Chapter 6**, we introduced the preprocessing methods on microbiota data before differential analysis.

* In **Chapter 7** provided the alpha diversity analysis.

* In **Chapter 8** provided the beta diversity analysis.

* In **Chapter 9** provided the microbiota composition analysis.

* In **Chapter 10** provided the core microbiota analysis.

* In **Chapter 11** provided the differential analysis.

* In **Chapter 12** provided some functions for visualization.

* In **Chapter 13** provided the association analysis.

* In **Chapter 14** showed the principals of the differential analysis methods.

* In **Chapter 15** provided the examples by using XMAS 2.0 on microbiota data.


## Citation 

Kindly cite by using `citation("XMAS")` if you think **XAMS** helps you. Alternative way is Tong B, Zou H (2022). _XMAS: A set of tools for statistical analysis in metagenomics_. R package version 2.1.6, <URL:https://gitlab.xbiome.com/Analytics/xmas/>.

## Change log

### XMAS 2.1.6

+ add *Gut microbiota Data from Zeybel et al. - 2022*. (2022-07-06)
+ add `run_cor`. (2022-07-06)
+ fix bugs `run_lefse`. (2022-07-07)
+ add `run_partial_cor`. (2022-07-08)
+ fix bugs `summarize_LowAbundance_taxa`. (2022-07-08)
+ fix bugs `summarize_taxa`. (2022-07-11)
+ add parameter *group_number* in `plot_barplot`, `plot_boxplot` and `plot_dotplot`. (2022-07-11)
+ change parameter `run_RefCheck`. (2022-07-11)  
+ add `plot_correlation_heatmap`. (2022-07-12) 

### XMAS 2.1.5

+ add *paired* parameter in `run_ttest` and `run_wilcox`. (2022-06-27)
+ add *Data from Early Childhood Antibiotics and the Microbiome (ECAM) study*. (2022-06-27)
+ add `run_filter2`. (2022-06-29)
+ fix bugs in `determine_tables`. (2022-06-29)
+ fix bugs in `run_RefCheck`. (2022-06-30)
+ fix bugs in `run_RefCheck`. (2022-07-03)
+ modify the scripts' annotations. (2022-07-05)
+ fix bugs in `run_multiple_da`. (2022-07-05)
+ fix bugs in `run_edger`. (2022-07-05)

### XMAS 2.1.4

+ fix bugs in `plot_Dada2Track` and `plot_RarefCurve`. (2022-06-06)
+ fix bugs in `determine_tables`. (2022-06-07)
+ format the codes in elegant style. (2022-06-08)
+ fix bugs in `import_dada2_taxa`. (2022-06-13)
+ fix bugs in `run_permanova`, `run_ANOSIM` and `run_MRPP`. (2022-06-14) 
+ fix bugs in `get_GroupPhyloseq`. (2022-06-14)
+ fix bugs in `run_RefCheck`. (2022-06-15)
+ fix bugs in scripts, for instance, replacing "SampleID" by "TempRowNames". (2022-06-15)
+ fix bugs in `run_lefse`. (2022-06-16)
+ fix bugs in `plot_RankAbundance`, `transform_abundances` and `run_trim`. (2022-06-17)
+ update **SchematicFigure** in vignettes and README. (2022-06-20)
+ add `plot_double_barplot`. (2022-06-20)
+ fix bugs in `aggregate_LOD_taxa`. (2022-06-22)
+ fix bugs in `import_metaphlan_taxa`. (2022-6-24)


### XMAS 2.1.3

+ add `aggregate_LOD_taxa` and rename scripts. (2022-05-31)


### XMAS 2.1.2

+ reassign the packages to *imports* or *suggests*. (2022-05-26)


### XMAS 2.1.1

+ fix bugs in `run_permanova`, `plot_multiple_DA`, `summarize_LowAbundance_taxa` and `plot_boxplot`. (2022-05-23)


### XMAS 2.1.0

+ add `run_mulitple_da` and `plot_multiple_DA` for multiple DA methods' results. (2022-05-18)


### XMAS 2.0.13

+ fix bugs. (2022-05-17)


### XMAS 2.0.12

+ Add `run_MantelTest`. (2022-05-11)
+ Add `plot_2DA_venn`. (2022-05-11)
+ Modify `plot_taxa_heatmap` and `plot_dotplot`. (2022-05-12)
+ Add topN significant taxa boxplot `plot_topN_boxplot` (2022-05-12)
+ Add `run_MRPP`. (2022-05-13)
+ Add `run_rank_abundance` and `plot_RankAbundance`. (2022-05-13)
+ Add `run_impute` for imputation. (2022-05-13)


### XMAS 2.0.7 (2022-04-12)

+ Add global view modules and other parts.


### XMAS 2.0.2 (2022-03-17)

+ Recode the differential analysis scripts `run_aldex` etc.
+ Modify the import scripts.


### XMAS 2.0.1 (2022-03-14)

+ Rearrange the package's structure.


### XMAS 1.0.0 (2021-12-13)

+ The final version 1.0.0.


### Initial repository 0.0.1 (2021-07-25)

+ Submitted to gitlab.


