# Quality Evaluation

Quality control of DADA2 results will help us have more rational determinations on the further data analysis. Firstly, the reads' track of DADA2 could show us the Changed Ratio of reads through the in-house standard amplicon sequencing data upstream pipeline. Then, the Evaluation of the spike-in samples from the Reference Matrix will reflect the quality of sequencing data. Finally, we recommend you that utilizing the rarefaction curves to assess the sequence depth per sample and choose the rational cutoff of OTU Number to do rarefy. 

**Loading packages**
```R
library(XMAS)
library(dplyr)
library(tibble)
library(phyloseq)
library(ggplot2)
```

## Reads' track

* this procedure only perform in 16s data

```{r, warning=FALSE, fig.align="center", fig.cap="Reads' track of DADA2"}
plot_Dada2Track(
  data=dada2_res$reads_track,
  mode="PE",
  relative_abundance=TRUE,
  legend_position="top"
  )
```


## Spike-in (BRS) sample assessment

The spike-in sample is use to evaluate the consistent quality on bacteria when we have multiple sequence batches data. We devised an evaluation system containing the *Correlation Coefficient*, *Bray Curtis Distance* and *Impurity Level* to assess the sequencing data quality. 

**Please use the default Reference and Save directory to obtain and save the spike-in sample matrix when you run `run_RefCheck`.**

### 16s data 

The taxonomic levels of spike-in sample's bacteria is genus. Firstly, using the `summarize_taxa` to get the genus level phyloseq object and then do `run_RefCheck` under the optimal parameters.

* *group*: the column contains the spike-in sample;

* *group_names*: the group names of the spike-in sample;

* *Reference*: the directory of the latest spike-in sample matrix (default: /share/projects/Analytics/analytics/XMAS/RefCheck/);

* *Save*: the directory to save the latest spike-in sample matrix (default: /share/projects/Analytics/analytics/XMAS/RefCheck/).

To see more details to use `?run_RefCheck`.

```{r, warning=FALSE, collapse=TRUE}
dada2_ps_genus <- summarize_taxa(dada2_ps, taxa_level = "Genus")

dada2_ps_genus@sam_data

RefCheck_16s <- run_RefCheck(
    ps=dada2_ps_genus,
    BRS_ID="S8005",
    Reference=NULL,
    Ref_type="16s",
    Save=NULL
    )
```


We could see that the messages are comprised of four parts.

1. the 1st part showed the type of *reference matrix* and whether the spike-in sample had been added to reference matrix;

2. the 2nd part revealed that *what* and *how many* the _matched bacterica_ of the spike-in sample are;

3. the 3nd part showed that the *value* of evaluation system in the spike-in sample;

4. the 4nd part showed that whether the spike-in sample *passes* the cutoff of evaluation system.


### Metagenomic data 

The taxonomic levels of spike-in sample's bacteria is species. Firstly, using the `summarize_taxa` to get the genus level phyloseq object and then do `run_RefCheck` under the optimal parameters.

<details>
<summary>**Quality Control by spike-in sample in metagenomic**</summary>
```{r, warning=FALSE, message=FALSE, collapse=TRUE}
metaphlan2_ps_species <- summarize_taxa(metaphlan2_ps, taxa_level = "Species")
metaphlan2_ps_species@sam_data

RefCheck_MGS <- run_RefCheck(
    ps=metaphlan2_ps_species,
    BRS_ID="refE",
    Reference=NULL,
    Ref_type="MGS",
    Save=NULL
    )
```
</details>


## Rarefaction Curves

>  Rarefaction curves are often used when calculating alpha diversity indices, because increasing numbers of sequenced taxa allow increasingly accurate estimates of total population diversity. Rarefaction curves can therefore be used to estimate the full sample richness, as compared to the observed sample richness.

```{r, warning=FALSE, message=FALSE, fig.align="center", fig.cap="Rarefaction Curves"}
plot_RarefCure(dada2_ps,
               taxa_level = "OTU",
               step = 100,
               label = "Group",
               color = "Group",
               parallel = TRUE,
               se = TRUE
               )
```


The result showed that all the samples had different sequencing depth but had the full sample richness.


## Summarize phyloseq-class object

Summarizing the phyloseq-class object by using `summarize_phyloseq`. It displayed that briefly introduction of the object.

```{r, warning=FALSE, collapse=TRUE}
summarize_phyloseq(dada2_ps)
```

The minus account of the OTU counts is *51181* in the phyloseq object, and we can use it as the threshold to rarefy. 

Notice the **Sparsity (0.865)**, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.



## Systematic Information
```{r}
sessionInfo()
```

