# Pre-processing 

Performing the pre-processing steps on the phyloseq-class object. 

**Loading packages**

```R
library(XMAS)
library(dplyr)
library(tibble)
library(phyloseq)
library(ggplot2)
```

* amplicon sequencing (16s)
```{r, warning=FALSE}
data("dada2_ps_genus")
dada2_ps_genus
```

<details>
<summary>**metagenomic sequencing (metaphlan2/3)**</summary>
```{r, warning=FALSE}
data("metaphlan2_ps")
metaphlan2_ps
```
</details>


## Removing BRS

Using `get_GroupPhyloseq` to remove some samples. For instance, removing the spike-in sample QC in Group column.

```{r, warning=FALSE, message=FALSE}
dada2_ps_remove_BRS <- get_GroupPhyloseq(
                     ps = dada2_ps,
                     group = "Group",
                     group_names = "QC",
                     discard = TRUE
                     )
dada2_ps_remove_BRS

metaphlan2_ps_remove_BRS <- get_GroupPhyloseq(
                     ps = metaphlan2_ps,
                     group = "Group",
                     group_names = "QC",
                     discard = TRUE
                     )
metaphlan2_ps_remove_BRS
```



## Transformation

Transforming the taxa abundances in `otu_table` by sample, which means the counts of each sample will be transformed individually. The options include:

* "identity", return the original data without any transformation;

* "log10", the transformation is `log10(object)`, and if the data contains zeros the transformation is `log10(1 + object)`;

* "log10p", the transformation is `log10(1 + object)`.

```{r, warning=FALSE}
dada2_ps_genus <- summarize_taxa(dada2_ps_remove_BRS, taxa_level = "Genus")

ps_transform <- transform_abundances(dada2_ps_genus, transform = "log10")
head(ps_transform@otu_table@.Data, 3)
```

<details>
<summary>**transforming metagenomic sequencing**</summary>
```{r, warning=FALSE}
metaphlan2_ps_species <- summarize_taxa(metaphlan2_ps_remove_BRS, taxa_level = "Species")

ps_transform_mgs <- transform_abundances(metaphlan2_ps, transform = "log10")
head(ps_transform_mgs@otu_table@.Data, 3)
```
</details>

## Normalization

Normalizing the OTU_table in phyloseq-class object. It is critical to normalize the feature table to eliminate any bias due to differences in the sampling sequencing depth. This function implements 7 widely-used normalization methods for microbial compositional data.

* **rarefy**: random subsampling counts to the smallest library size in the data set. **Caution: the default library size is 25000 according to our own results(Rarefaction Curves)**.
```{r, warning=FALSE, message=FALSE}
normalize(dada2_ps_genus, method = "rarefy")
```


* **TSS**: total sum scaling, also referred to as "relative abundance", the abundances were normalized by dividing the corresponding sample library size
```{r, warning=FALSE, message=FALSE}
normalize(dada2_ps_genus, method = "TSS")
```


* **TMM**: trimmed mean of m-values. First, a sample is chosen as reference. The scaling factor is then derived using a weighted trimmed mean over the differences of the log-transformed gene-count fold-change between the sample and the reference.
```{r, warning=FALSE, message=FALSE}
normalize(dada2_ps_genus, method = "TMM")
```


* **RLE**: relative log expression, RLE uses a pseudo-reference calculated using the geometric mean of the gene-specific abundances over all samples. The scaling factors are then calculated as the median of the gene counts ratios between the samples and the reference.
```{r, warning=FALSE, message=FALSE}
normalize(dada2_ps_genus, method = "RLE")
```


* **CSS**: cumulative sum scaling, calculates scaling factors as the cumulative sum of gene abundances up to a data-derived threshold. While standard relative abundance (fraction/percentage) normalization re-scales all samples to the same total sum (100%), CSS keeps a variation in total counts between samples. CSS re-scales the samples based on a subset (quartile) of lower abundant taxa (relatively constant and independent), thereby excluding the impact of (study dominating) high abundant taxa.
```{r, warning=FALSE, message=FALSE}
normalize(dada2_ps_genus, method = "CSS")
```

* **CLR**: centered log-ratio normalization.
```{r, warning=FALSE, message=FALSE}
normalize(dada2_ps_genus, method = "CLR")
```

* **CPM**: pre-sample normalization of the sum of the values to 1e+06.
```{r, warning=FALSE, message=FALSE}
normalize(dada2_ps_genus, method = "CPM")
```



## Filtering

Whether to filter the low relative abundance or unclassified taxa by the threshold.
```{r,, warning=FALSE, message=FALSE}
ps_genus_rb <- summarize_taxa(dada2_ps_remove_BRS, taxa_level = "Genus", absolute = FALSE)
ps_genus_rb
ps_genus_rb_filter <- run_filter(ps_genus_rb, cutoff = 1e-04, unclass = TRUE)
ps_genus_rb_filter
```

**39** genus's relative abundance or attributes were below *1e-04* or unclassified and they were removed by the cutoff.


## Trimming

The previous function (`run_filter`) only focuses on the low relative abundance and unclassified taxa. Microbial data always have so many zeros. Trimming samples or taxa in `otu_table` by occurrences or prevalence before downstream analysis is also crucial.

* trimming by TaxaID
```{r, warning=FALSE, message=FALSE}
run_trim(dada2_ps_genus, cutoff = 0.4, trim = "feature")
```

Dropping the taxa whose prevalence or occurrences are less than 0.4. 


* trimming by SampleID
```{r, warning=FALSE, message=FALSE}
run_trim(dada2_ps_genus, cutoff = 0.4, trim = "sample")
```

Dropping the samples whose prevalence  or occurrences are less than 0.4. 


* trimming by TaxaID & SampleID
```{r, warning=FALSE, message=FALSE}
run_trim(dada2_ps_genus, cutoff = 0.4, trim = "both")
```

Dropping the taxa and samples whose prevalence are less than 0.4. 

<details>
<summary>**filtering metagenomic sequencing**</summary>
```{r, warning=FALSE, message=FALSE}
ps_trim_mgs <- run_trim(metaphlan2_ps_remove_BRS, cutoff = 0.4, trim = "feature")
ps_trim_mgs
```
</details>


## Imputation

The missing values in otu table maybe affect the statistical results, imputing the NAs or Zero values should taken into account.

* limit of detection
```{r, warning=FALSE, message=FALSE, collapse=TRUE}
min(dada2_ps_genus@otu_table)

LOD_imputed_ps <- run_impute(dada2_ps_genus, impute = "LOD", LOD = 2)

min(LOD_imputed_ps@otu_table)
```

## Extracting specific taxa phyloseq-class object

The taxonomic level are *Kingdom*, *Phylum*, *Class*, *Order*, *Family*, *Genu*s, *Species* and choosing the specific taxa to regenerate the phyloseq-class object. 

* amplicon sequencing: Phylum
```{r, warning=FALSE, message=FALSE}
data("dada2_ps")
ps_phylum <- summarize_taxa(dada2_ps_remove_BRS, taxa_level = "Phylum")
ps_phylum
```

* amplicon sequencing: Order
```{r, warning=FALSE, message=FALSE}
ps_order <- summarize_taxa(dada2_ps_remove_BRS, taxa_level = "Order")
ps_order
```


* amplicon sequencing: Family
```{r, warning=FALSE, message=FALSE}
ps_family <- summarize_taxa(dada2_ps_remove_BRS, taxa_level = "Family")
ps_family
```

* amplicon sequencing: Genus
```{r, warning=FALSE, message=FALSE}
ps_genus <- summarize_taxa(dada2_ps_remove_BRS, taxa_level = "Genus")
ps_genus
```

<details>
<summary>**extracting metagenomic sequencing**</summary>
```{r, warning=FALSE, message=FALSE}
ps_genus_mgs <- summarize_taxa(metaphlan2_ps_remove_BRS, taxa_level = "Genus")
ps_genus_mgs

ps_species_mgs <- summarize_taxa(metaphlan2_ps_remove_BRS, taxa_level = "Species")
ps_species_mgs
```
</details>


## Aggregating low relative abundance or unclassified taxa into others

1. Taxa with relative abundance less than **0.0001** will be summarized into `Others_LowAbundance`;

2. Unclassified taxa will be summarized into `Others_Unclassified`.

* amplicon sequencing
```{r, warning=FALSE, message=FALSE}
# relative abundance
ps_genus_rb <- summarize_taxa(dada2_ps_remove_BRS, taxa_level = "Genus", absolute = FALSE)
ps_genus_LRA <- summarize_LowAbundance_taxa(ps_genus_rb, cutoff = 1e-04, unclass = TRUE)
ps_genus_LRA 
tail(phyloseq::taxa_names(ps_genus_LRA))

# absolute abundance
ps_genus_counts <- summarize_LowAbundance_taxa(dada2_ps_genus, cutoff = 10)
ps_genus_counts
```

<details>
<summary>**aggregating metagenomic sequencing**</summary>
```{r, warning=FALSE, message=FALSE}
ps_genus_LRA_mgs <- summarize_LowAbundance_taxa(ps_genus_mgs, cutoff = 1e-04, unclass = TRUE)
ps_genus_LRA_mgs
tail(phyloseq::taxa_names(ps_genus_LRA_mgs))
```
</details>



## Systematic Information
```{r}
sessionInfo()
```
