# Aplha diversity

Alpha diversity is a measure of microbiome diversity applicable to a single sample.  The commonly used metrics/indices are Shannon, Chao1, Observed, Simpson, Gini and Inverse Simpson. The reference of this part is [@Sudarshan2020Microbial].

> **It is important to note that, alpha diversity indices are sensitive to noise that is inherent to application of polymerase chain reaction and the sequencing errors.** 

**Cautions**: *Shannon, Simpson and InvSimpson* measures are both for relative or absolute abundance.

As defined by Shannon, the index of Shannon index is given as follow:

$$H^{'} = \sum_{i=1}^{S}p_{i}\,\ln\,p_{i}$$
where $p_{i}$ is the proportion of individuals (or relative abundance) of species $i$ in the community and $S$ is the total number of species present so that $\sum_{i=1}^{S}p_{i} = 1$.

**Loading packages**

```{r, echo=TRUE, results="hide", warning=FALSE, message=FALSE}
library(XMAS2)
library(dplyr)
library(tibble)
library(phyloseq)
library(ggplot2)
library(ggpubr)
library(SummarizedExperiment)
```

## Equal sample sums

* Remove BRS
```{r, warning=FALSE, message=FALSE}
data("dada2_ps")
dada2_ps_remove_BRS <- get_GroupPhyloseq(
                     ps = dada2_ps,
                     group = "Group",
                     group_names = "QC",
                     discard = TRUE)
dada2_ps_remove_BRS
```

* Summary counts 

The previous result from *Rarefaction Curves* had demonstrated that the sequence depth of each sample has a larger significance and normalizing the samples' depth into equal sampling depth is necessary.


```{r, warning=FALSE}
summary(sample_sums(dada2_ps_remove_BRS))
```

As is evident there is a large difference in the number of reads. Minimum is _51181_ and maximum is _71667_!! There is a **~1.4X** difference.

```{r, warning=FALSE, message=FALSE}
dada2_ps_rarefy <- norm_rarefy(dada2_ps_remove_BRS, size = 51181, rng_seed = 123)
summary(sample_sums(dada2_ps_rarefy))
```

Quickly check taxa prevalence
```{r, warning=FALSE, fig.width=7, fig.height=7, fig.align="center", fig.cap="Taxa prevalence after rarefying"}
plot_taxa_prevalence(dada2_ps_rarefy, "Phylum")
```


## Diversities 

```{r, warning=FALSE}
data("amplicon_ps")
summary(sample_sums(amplicon_ps))
```

As is evident there is a large difference in the number of reads. Minimum is _1114_ and maximum is _11449_!! There is a **~10X** difference!

```{r, warning=FALSE, message=FALSE}
amplicon_ps_rarefy <- norm_rarefy(amplicon_ps, size = 1114, rng_seed = 123)
summary(sample_sums(amplicon_ps))
```

### Non-phylogenetic diversities

#### XMAS 

1. Calculation
```{r, warning=FALSE, message=FALSE}
amplicon_ps_rarefy_alpha <- run_alpha_diversity(ps=amplicon_ps_rarefy, measures = c("Shannon", "Chao1", "Observed")) 
head(amplicon_ps_rarefy_alpha)
```


* **Visualization**

1. single measure
```{r, warning=FALSE, fig.width=6, fig.height=5, fig.align="center", fig.cap="Alpha diversity from XMAS(one measure)"}
plot_boxplot(data = amplicon_ps_rarefy_alpha,
             y_index = "Shannon",
             group = "SampleType",
             group_names = NULL,
             group_color = NULL,
             do_test = TRUE,
             ref_group = NULL,
             method = "wilcox.test",
             outlier = TRUE)
```

2. multiple measures
```{r, warning=FALSE, fig.width=9, fig.height=5, fig.align="center", fig.cap="Alpha diversity from XMAS(multiple measures)"}
plot_boxplot(data = amplicon_ps_rarefy_alpha,
             y_index = c("Shannon", "Chao1", "Observed"),
             group = "SampleType",
             group_names = c("gut", "right palm", "tongue"),
             group_color = c("red", "green", "blue"),
             do_test = TRUE,
             ref_group = "gut",
             method = "wilcox.test",
             outlier = TRUE)
```



#### microbiome

* **Calculation**
```{r, warning=FALSE, message=FALSE}
amplicon_ps_rarefy_alpha_v2 <- microbiome::alpha(amplicon_ps_rarefy, index = "all")
DT::datatable(amplicon_ps_rarefy_alpha_v2)
```

* **Visualization**
```{r, warning=FALSE, fig.width=8, fig.height=6, fig.align="center", fig.cap="Alpha diversity from microbiome"}
amplicon_ps_rarefy_metadata <- phyloseq::sample_data(amplicon_ps_rarefy) %>%
  data.frame()
amplicon_ps_rarefy_alpha_v2$SampleID <- rownames(amplicon_ps_rarefy_metadata)
amplicon_ps_rarefy_metadata$SampleID <- rownames(amplicon_ps_rarefy_metadata)
dat_diversity <- dplyr::inner_join(amplicon_ps_rarefy_metadata, amplicon_ps_rarefy_alpha_v2, by = "SampleID")
dat_diversity_v2 <- dat_diversity[, c("SampleType", "diversity_inverse_simpson", 
                                 "diversity_gini_simpson", "diversity_shannon", 
                                 "diversity_fisher", "diversity_coverage")]

colnames(dat_diversity_v2) <- c("SampleType", "Inverse Simpson", "Gini-Simpson", "Shannon", "Fisher", "Coverage")

plotdata <- dat_diversity_v2 %>% 
  tidyr::gather(key = "Variable", value = "Value", -SampleType)
groups <- unique(dat_diversity_v2$SampleType) 
cmp_list <- combn(seq_along(groups), 2, simplify = FALSE, FUN = function(x) {groups[x]})
pval_sign <- list(
  cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 0.1, 1),
  symbols = c("****", "***", "**", "*", "n.s")
)

ggboxplot(plotdata, x = "SampleType", y = "Value",
          fill = "SampleType", 
          palette = "jco", 
          legend= "right",
          facet.by = "Variable", 
          scales = "free")+
  rotate_x_text()+
  rremove("x.text")+
  stat_compare_means(
  comparisons = cmp_list,
  label = "p.signif",
  symnum.args = pval_sign
  )
```


###  Phylogenetic diversity

Phylogenetic diversity is calculated using the picante package.

* **Calculation**
```{r, warning=FALSE, message=FALSE}
library(picante)
amplicon_ps_rarefy_tab <- as.data.frame(amplicon_ps_rarefy@otu_table)
amplicon_ps_rarefy_tree <- amplicon_ps_rarefy@phy_tree
dat_pd <- pd(t(amplicon_ps_rarefy_tab), 
             amplicon_ps_rarefy_tree, 
             include.root = FALSE)

DT::datatable(dat_pd)
```

* **Visualization**
```{r, warning=FALSE, fig.align="center", fig.cap="Alpha diversity from picante(Phylogenetic diversity)"}
amplicon_ps_rarefy_metadata$Phylogenetic_Diversity <- dat_pd$PD

ggboxplot(amplicon_ps_rarefy_metadata,
    x = "SampleType",
    y = "Phylogenetic_Diversity",
    fill = "SampleType",
    palette = "jco",
    ylab = "Phylogenetic Diversity",
    xlab = "Sample Type",
    legend = "right")+ 
  rotate_x_text()+
  stat_compare_means(
    comparisons = cmp_list,
    label = "p.signif",
    symnum.args = pval_sign)
```


**Cautions**:

> There are arguments both for and against the use of rarefying to equal library size.
The application of normalization method will depend on the type of research question. It is always good to check if there is a correlation between increasing library sizes and richness. Observed ASVs and Phylogenetic diversity can be affected by library sizes. It is always good to check for this before making a choice.

* Calculation
```{r, warning=FALSE, message=FALSE}
lib.div <- microbiome::alpha(amplicon_ps, index = "all")
lib.div2 <- microbiome::richness(amplicon_ps)
lib.div$ReadsPerSample <- sample_sums(amplicon_ps)
lib.div$Observed <- lib.div2$observed

colnames(lib.div)
```

* scatterplot
```{r, warning=FALSE, fig.width=8, fig.height=6, fig.align="center", fig.cap="Correlation between increasing library sizes and richness"}
p1 <- ggscatter(lib.div, "diversity_shannon", "ReadsPerSample")+
  stat_cor(method = "pearson")

p2 <- ggscatter(lib.div, "diversity_inverse_simpson", "ReadsPerSample",dd = "loess")+
  stat_cor(method = "pearson")

p3 <- ggscatter(lib.div, "Observed", "ReadsPerSample", add = "loess") +
  stat_cor(
    method = "pearson",
    label.x = 100,
    label.y = 50000
  )

ggarrange(p1, p2, p3, ncol = 2, nrow = 2)
```


## Rank Abundance

A rank abundance curve is a chart used by ecologists to display relative species abundance, a component of biodiversity. It can also be used to visualize species richness and species evenness. It overcomes the shortcomings of biodiversity indices that cannot display the relative role different variables played in their calculation.
The curve is a 2D chart with relative abundance on the Y-axis and the abundance rank on the X-axis.

* X-axis: The abundance rank. The most abundant species is given rank 1, the second most abundant is 2 and so on.
* Y-axis: The relative abundance. Usually measured on a log scale, this is a measure of a species abundance (e.g., the number of individuals) relative to the abundance of other species.

```{r, warning=FALSE, fig.width=5, fig.height=4, fig.align="center", fig.cap="Rank Abundance"}
plot_RankAbundance(
        ps = dada2_ps_remove_BRS,
        taxa_level = "Family",
        group = "Group",
        group_names = c("AA", "BB"))
```

**Results**:

1. From the horizontal level (Family rank), the higher degree of width means the higher degree of **Family richness**; 

2. From the vertical level (Relative abundance), the slope of the line reflects the **Family evenness**.


## Systematic Information
```{r}
sessionInfo()
```
