# Principals of Differential Analysis methods

[@nearing2022microbiome] has showed that the multiple DA methods have different performances on microbiota (16s). Here, we integrated the its methods' part and our own understandings to give a draft whole picture for users who using DA of XMAS2. 


Microbiota data have the following characteristics:

1. **Compositional**: Mathematically, a data is defined as compositional, if it contains D multiple parts of nonnegative numbers whose sum is 1 or any constant-sum constraint. It can be formally stated as:

$$S^{D} = \left \{ X = [x_{1}, x_{2}, ..., x_{D}] | x_{i} > 0, i = 1, 2, ..., D; \sum^{D}_{i=1} x_{i} = K \right \}$$


2. **Over-Dispersed**: The variance are generally greater than the mean. There are several possible causes of overdispersion: correlated taxa counts, clustering of subjects, and within group heterogeneity. Negative Binomial Model addresses overdispersion by by explicitly modeling the correlated and sparse events via a latent variable.


3. **Sparsity**: Microbiome data typically is overdispersed and sparse with many zeros. When the number of zeros is excess than the standard distributions (e.g., normal, Poisson, binomial, NB, beta or gamma) can be readily fit, the data set is considered as ‘zero inflate.



## ALDEx2

ALDEx2 is from [@fernandes2014unifying]

### Vignette

[ANOVA-Like Differential Expression tool for high throughput sequencing data: ALDEx2](https://www.bioconductor.org/packages/devel/bioc/vignettes/ALDEx2/inst/doc/ALDEx2_vignette.html)

### Input Data Type

All the input data type should be **Raw Counts**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **reads**: Raw counts matrix (Rows->Features; columns->Samples).

2. **conditions**: A vector of group labels for testing.

3. **mc.samples**: The number of Monte Carlo samples, default is 128.

4. **denom**: The methods for Geometric Mean calculation, default is "all".

5. **test**: The test to perform, default is "t.test".


### Procedures

1. Monte Carlo samples of Dirichlet distributions for each sample, using a uniform prior; (Sparsity)

2. Performing CLR (log-ratio) transformation  of each realization; (Compositional & Overdispersion)

3. Performing Wilcoxon tests on the transformed realizations. 

Finally, the function returned the expected Benjamini-Hochberg (BH) FDR-corrected p-value for each feature based on the results the different across Monte Carlo samples.

### Key output

1. **TaxaID**: taxa ID.

2. **EffectSize**: Effect Size generated by `Aldex2`, which equals to Log2FoldChange.

3. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

4. **Median CLR (All/Groups)**: median value of CLR abundance normalized by `Aldex2`.

5. **Occurrence (All/Groups)**: prevalence of taxa.



### Original Syntax

```R
aldex(
  reads,
  conditions,
  mc.samples = 128,
  test = "t",
  effect = TRUE,
  include.sample.summary = FALSE,
  verbose = FALSE,
  denom = "all",
  iterate = FALSE,
  ...
)
```


### XMAS2 Syntax

```R
da_res <- ALDEx2::aldex(reads = otu_tab,
                        conditions = sam_tab$Compvar,
                        mc.samples = 128,
                        test = test,
                        effect = TRUE,
                        include.sample.summary = FALSE,
                        denom = "all",
                        verbose = TRUE)
```

### Notes

1. Effect Size could be regarded as Log2FoldChange; 


## ANCOM-II

ANCOM-II is from [ANCOM-II (version 2.1)](https://github.com/FrederickHuangLin/Microbiome-Review-Code-Archive/tree/master/scripts)  and it derived from [@mandal2015analysis]. 


### Input Data Type

All the input data type should be **Raw Counts**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **feature_table**: Raw counts matrix (Rows->Features; columns->Samples).

2. **meta_data**: data.frame with group information.

3. **struc_zero**: data.frame with structure zero taxa.

4. **group_var**: group variable for comparison.

5. **alpha**: the significant level cutoff for testing.


### Procedures

1. `feature_table_pre_process` firstly identify outlier zeros and structural zeros; 

  * Outlier zeros, identified by finding outliers in the distribution of taxon counts within each sample grouping, were ignored during differential abundance analysis, and replaced with NA.
  
  * Structural zeros, taxa that were absent in one grouping but present in the other, were ignored during data analysis and automatically called as differential abundant.

2. additive log-ratios for transformation with a pseudo count of 1; 

3. Wilcoxon rank-sum tests for significance, and p-values were FDR-corrected using the BH method;

4. Detection threshold for Ajusted-pvalue per taxa, if the number (ratio) of Ajusted-pvalue reaches nominal significance would be called as DA taxa.

### Key output

1. **TaxaID**: taxa ID.

2. **EffectSize**: Effect Size to evaluate the quality of _(W)q-values < alpha_.

3. **W_ratio**: the ratio of significant taxa compared to totoal number of taxa.

4. **detected**: the taxa is significant passed the W_ratio cutoff.


### Original Syntax

```R
preprocess_ANCOM <- function(
          feature_table,
          meta_data,
          sample_var = NULL,
          group_var = NULL,
          out_cut = 0.05,
          zero_cut = 0.90,
          lib_cut = 1000,
          neg_lb = FALSE)

main_ANCOM <- function(
                  feature_table,
                  meta_data,
                  struc_zero = NULL,
                  group_var = NULL,
                  p_adj_method = "BH",
                  alpha = 0.05,
                  adj_formula = NULL,
                  rand_formula = NULL,
                  w_cutoff = 0.7,
                  ...)
```


### XMAS2 Syntax

```R
prepro_res <- preprocess_ANCOM(
      feature_table = otu_tab,
      meta_data = sam_tab %>% tibble::rownames_to_column("TempRowNames"),
      sample_var = "TempRowNames",
      group_var = "Compvar",
      out_cut = 0.05,
      zero_cut = 0.90,
      lib_cut = 1000,
      neg_lb = FALSE)

da_res <- main_ANCOM(
      feature_table = prepro_res$feature_table,
      meta_data = prepro_res$meta_data,
      struc_zero = prepro_res$structure_zeros,
      group_var = "Compvar",
      p_adj_method = p_adjust,
      alpha = pvalue_cutoff,
      rand_formula = NULL,
      w_cutoff = 0.7)
```

### Notes

1. the column **detected_** with TRUE shows the significant taxa by `run_ancom``; 



## corncob


corncob is from [@martin2020modeling]

### Vignette

[Count Regression for Correlated Observations with the Beta-Binomial: corncob](https://cran.r-project.org/web/packages/corncob/vignettes/corncob-intro.pdf)

### Input Data Type

All the input data type should be **Raw Counts**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **data**: a data frame containing the OTU table, or 
phyloseq object containing the variables in the models (counts matrix).

2. **formula**: an object of class formula without the response.

3. **test**: test method, options include: "Wald" and "LRT" for two groups comparison (default: "Wald").

4. **fdr_cutoff**: cutoff of FDR.

### Procedures

1. Taxon count abundance fit to a **beta-binomial model**, using 
logit link functions for both the mean and overdispersion.

2. Wald tests for significance testing, and BH for FDR-corrected p-values.

### Key output

1. **TaxaID**: taxa ID.

2. **EffectSize**: Effect Size to evaluate the power of pvalue.

3. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

4. **Occurrence (All/Groups)**: prevalence of taxa.


### Original Syntax

```R
differentialTest(
  formula,
  phi.formula,
  formula_null,
  phi.formula_null,
  data,
  link = "logit",
  phi.link = "logit",
  test,
  boot = FALSE,
  B = 1000,
  sample_data = NULL,
  taxa_are_rows = TRUE,
  filter_discriminant = TRUE,
  fdr_cutoff = 0.05,
  fdr = "fdr",
  full_output = FALSE,
  inits = NULL,
  inits_null = NULL,
  try_only = NULL,
  ...
)
```


### XMAS2 Syntax

```R
da_res <- corncob::differentialTest(
                    data = ps_normed_new,
                    formula = my_formula,
                    phi.formula = my_formula,
                    phi.formula_null = my_formula,
                    formula_null = ~ 1,
                    test = method,
                    boot = F,
                    fdr_cutoff = pvalue_cutoff,
                    fdr = p_adjust)
```

### Notes

1. Columns with Log2FoldChange are based on **raw counts**; 




## DESeq2

DESeq2 is from [@love2014moderated]

### Vignette

[DESeq2](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html)

### Input Data Type

All the input data type should be **un-normalized raw counts**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **countData**: Raw counts matrix without normalization (Rows->Features; columns->Samples).

2. **colData**: a DataFrame or data.frame with at least a single column..

3. **design**: a formula or matrix. the formula expresses how the counts for each gene depend on the variables in colData.


### Procedures

1. estimation of size factors, which are used to normalize library sizes in a model-based fashion;

2. estimation of dispersions from the negative binomial likelihood for each feature, and subsequent shrinkage of each dispersion estimate towards the parametric (default) trendline by empirical Bayes;

3. fitting each feature to the specified class groupings with negative binomial generalized linear models and performing hypothesis testing, for which we chose the default Wald test;

4. Finally, using the `results` function to obtain the resulting BH FDR-corrected p-values.

### Key output

1. **TaxaID**: taxa ID.

2. **logFC**: Log2FoldChange by `DESeq2::DESeq`.

3. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

4. **Occurrence (All/Groups)**: prevalence of taxa.


### Original Syntax

```R
DESeqDataSetFromMatrix(
  countData,
  colData,
  design,
  tidy = FALSE,
  ignoreRank = FALSE,
  ...
)
```


### XMAS2 Syntax

```R
## build Matrix
ddsm <- DESeq2::DESeqDataSetFromMatrix(
                    countData = otu_tab,
                    colData =  sam_tab,
                    design =~ Compvar)
dds <- DESeq2::DESeq(ddsm)

# extract the results
DESeq_res <- DESeq2::results(dds, contrast = c("Compvar", rev(group_names)))
```


## edgeR

edgeR is from [@robinson2010scaling]

### Vignette

[edgeR](https://bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf)

### Input Data Type

All the input data type should be **Raw Counts**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **counts**: Raw counts matrix (Rows->Features; columns->Samples) for DEGlist.

2. **design**: A vector of group labels for testing.

3. **contrast**: A string for comparison groups.


### Procedures

1. Add a pseudocount of 1 to the non-rarefied feature table and used the 
function `calcNormFactors` from the edgeR package to compute relative 
log expression normalization factors;

2. Negative binomial dispersion parameters were then estimated using the 
functions `estimateCommonDisp` followed by `estimateTagwiseDisp` to shrink 
feature-wise dispersion estimates through an empirical Bayes approach;

3. `exactTest` for negative binomial data to identify features that differ 
between the specified groups;

4. The resulting p-values were then corrected for multiple testing with 
the BH method with the function `topTags`.

### Key output

1. **TaxaID**: taxa ID.

2. **logFC**: Log2FoldChange by `edgeR::topTags`.

3. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

4. **Occurrence (All/Groups)**: prevalence of taxa.


### Original Syntax

```R
# DGEList object & normalized factor

# GLM estimates of dispersion

# GLM testing for differential expression
```


### XMAS2 Syntax

```R
# Filter data
keep <- rowSums(edgeR::cpm(otu_tab) > 100) >= 2
otu_tab <- otu_tab[keep, ]

# DGEList object: otu_tab -> colnames:Samples; rownames:Taxa
dge_obj <- edgeR::DGEList(counts = otu_tab)
dge_obj_factors <- edgeR::calcNormFactors(dge_obj)
limma::plotMDS(dge_obj_factors, method = "bcv", col = as.numeric(sam_tab$Compvar))
graphics::legend("bottomleft", group_names, col = 1:2, pch = 20)

# GLM estimates of dispersion
dge <- edgeR::estimateGLMCommonDisp(dge_obj_factors, design)
dge <- edgeR::estimateGLMTrendedDisp(dge, design, method = "power")

# You can change method to "auto", "bin.spline", "power", "spline", "bin.loess".
# The default is "auto" which chooses "bin.spline" when > 200 tags and "power" otherwise.
dge <- edgeR::estimateGLMTagwiseDisp(dge, design)

# GLM testing for differential expression:
fit <- edgeR::glmFit(dge, design)
fit2 <- edgeR::glmLRT(fit, contrast = contrast) # Normal:-1; Tumor:1
edgeR_res <- edgeR::topTags(fit2, n = nrow(otu_tab)) %>% data.frame()
```

### Notes

1. Effect Size could be regarded as Log2FoldChange; 


## LEfSe

LEfSe is from [@segata2011metagenomic]

### Vignette

[Liner discriminant analysis (LDA) effect size: LEfSe](https://bioconductor.org/packages/devel/bioc/vignettes/lefser/inst/doc/lefser.html)

### Input Data Type

All the input data type could be **Raw Counts** or **Relative abundance**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA.

### Input Key Arguments

1. **assays**: data.frame or matrix of features (Rows->Features; columns->Samples).

2. **colData**: data.frame or matrix with group information.

3. **kruskal.threshold **: cutoff of KW test pvalue.

4. **wilcox.threshold**: cutoff of wilcox test pvalue.

5. **lda.threshold**: cutoff of LDA scores.


### Procedures

1. Normalizing the matrix data using *CPM: pre-sample normalization of the sum of the values to 1e+06.*;

2. Performing a Kruskal-Wallis (which in our two-group case reduces to 
the Wilcoxon rank-sum) hypothesis test to identify potential differentially abundant features;

3. Using differentially abundant features to perform linear discriminant analysis (LDA) of 
class labels on abundances to estimate the effect sizes for sig- nificant features.

4. Only features with scaled LDA analysis scores above the threshold (default: 2.0) 
were called as DA.

### Key output

1. **TaxaID**: taxa ID.

2. **LDA score**: LDA score to determine the significant taxa.


### Original Syntax

```R
lefser(expr,
       kruskal.threshold = 0.05,
       wilcox.threshold = 0.05,
       lda.threshold = 2.0,
       groupCol = "GROUP",
       blockCol = NULL,
       assay = 1L,
       trim.names = FALSE)
```


### XMAS2 Syntax

```R
se_object <- SummarizedExperiment::SummarizedExperiment(
                        assays = list(counts = otu_tab),
                        colData = sam_tab,
                        metadata = "Profile")

lefse_res <- lefser(se_object,
                    kruskal.threshold = 0.05,
                    wilcox.threshold  = wl.p,
                    lda.threshold     = Lda,
                    groupCol = "Compvar",
                    blockCol = NULL,
                    assay    = 1L,
                    trim.names = TRUE)
```

### Notes

1. Effect Size could be regarded as Log2FoldChange; 


## limma voom

limma-voom is from [@law2014voom]

### Vignette

[limma](https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf)

### Input Data Type

All the input data type should be **Raw Counts**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **counts**: a numeric matrix containing raw counts (Rows->Features; columns->Samples).

2. **design**: design matrix with rows corresponding to samples and columns to coefficients to be estimated.

3. **lib.size**: numeric vector containing total library sizes for each sample.

4. **span**: width of the smoothing window used for the lowess mean-variance trend.


### Procedures

1. Normalizing table by the trimmed mean of M-values (TMM) or TMM with 
singleton pairing (TMMwsp) option via `calcNormFactors function` 
(highly sparse data); 

2. After normalization, using the limma R package function `voom` to 
convert normalized counts to log2-counts-per-million and assign precision 
weights to each observation based on the mean-variance trend;

3. Using the functions `lmFit`, `eBayes`, and `topTable` in the limma R package to fit weighted linear regression models, perform tests based on an empirical Bayes moderated t-statistic76 and obtain BH FDR-corrected p-values.

### Key output

1. **TaxaID**: taxa ID.

2. **EffectSize** and **logFC**: both results equals to Log2FoldChange.

3. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

4. **Occurrence (All/Groups)**: prevalence of taxa.

### Original Syntax

### XMAS2 Syntax

```R
# voom fitting
voom_out <- limma::voom(
    otu_tab,
    design = design,
    lib.size = lib_size,
    span = voom_span)

# linear model
fit_out <- limma::lmFit(voom_out, design = design)
test_out <- limma::eBayes(fit_out)
test_df <- limma::topTable(
    test_out,
    number = nrow(otu_tab),
    adjust.method = p_adjust)
```

### Notes

1. Effect Size could be regarded as Log2FoldChange; 


## MaAsLin2

MaAsLin2 is from [@mallick2021multivariable]

### Vignette

[Multivariable association discovery in population-scale meta-omics studies: MaAsLin2](https://rdrr.io/bioc/Maaslin2/f/vignettes/maaslin2.Rmd)

### Input Data Type

All the input data type could be **Raw Counts** or **Relative abundance**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **input_data**: The tab-delimited input file of features (Rows->Features; columns->Samples).

2. **input_metadata**: The tab-delimited input file of metadata.

3. **normalization**: The normalization method to apply.

4. **transform**: The transform to apply.

5. **analysis_method**: The analysis method for linear regression to apply.

6. **random_effects**: The random effects for the model.

7. **fixed_effects**: The fixed effects for the model, such as the groups to compare.


### Procedures

1. Using *arcsine square-root transformation (AST)* for transformation 
and *total sum scaling normalization (TSS)* for normalization; 

2. The function fit a linear model (without random effects chosen) to 
each feature’s transformed abundance on the specified sample grouping, 
tested significance using a Wald test, and output BH FDR-corrected p-values;

### Key output

1. **TaxaID**: taxa ID.

2. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

3. **Occurrence (All/Groups)**: prevalence of taxa.


### Original Syntax

```R
Maaslin2(
    input_data,
    input_metadata,
    output,
    min_abundance = 0.0,
    min_prevalence = 0.1,
    min_variance = 0.0,
    normalization = "TSS",
    transform = "LOG",
    analysis_method = "LM",
    max_significance = 0.25,
    random_effects = NULL,
    fixed_effects = NULL,
    correction = "BH",
    standardize = TRUE,
    cores = 1,
    plot_heatmap = TRUE,
    plot_scatter = TRUE,
    heatmap_first_n = 50,
    reference = NULL
)
```


### XMAS2 Syntax

```R
da_res <- Maaslin2::Maaslin2(
              input_data = otu_tab %>% t() %>% data.frame(), # row->smapleID; col->Taxa
              input_metadata = sam_tab,
              output = outdir,
              normalization = norm,
              transform = transform,
              analysis_method = method,
              max_significance = pvalue_cutoff,
              fixed_effects = "Compvar",
              correction = p_adjust,
              plot_heatmap = FALSE,
              plot_scatter = FALSE,
              cores = 2)
```


## metagenomeSeq

metagenomeSeq is from [@paulson2013differential]

### Vignette

[metagenomeSeq](https://bioconductor.org/packages/release/bioc/vignettes/metagenomeSeq/inst/doc/metagenomeSeq.pdf)

### Input Data Type

All the input data type should be **Raw Counts**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **reads**: Raw counts matrix (Rows->Features; columns->Samples).

2. **conditions**: A vector of group labels for testing.

3. **mc.samples**: The number of Monte Carlo samples, default is 128.

4. **denom**: The methods for Geometric Mean calculation, default is "all".

5. **test**: The test to perform, default is "t.test".


### Procedures

1. Converting the counts and sample information into newMRexperiment 
object by the function `newMRexperiment` from the metagenomeSeq R package;

2. Using `cumNormStat` and `cumNorm` to apply **cumulative sum-scaling normalization (CSS)**, 
which attempts to normalize sequence counts based on the lower-quartile abundance of features;

3. Using `fitFeatureModel` to fit normalized feature counts with 
zero-inflated log-normal models (with pseudo-counts of 1 added 
prior to log2 transformation) and perform empirical Bayes moderated t-tests, 
and `MRfulltable` to obtain BH FDR-corrected p-values.

### Key output

1. **TaxaID**: taxa ID.

2. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.


### XMAS2 Syntax

```R
# run DA
if (method == "ZILN") {
  fit <- metagenomeSeq::fitFeatureModel(mgs_summarized, mod)
} else {
  fit <- metagenomeSeq::fitZig(mgs_summarized, mod)
}

# metagenomeSeq vignette: We recommend the user remove features based on
# the number of estimated effective samples, please see
# calculateEffectiveSamples. We recommend removing features with less
# than the average number of effective samples in all features. In
# essence, setting eff = .5 when using MRcoefs, MRfulltable, or MRtable.
res <- metagenomeSeq::MRcoefs(
    fit,
    number = ntaxa(ps_normed_new),
    adjustMethod = p_adjust,
    group = 3,
    eff = 0.5)
  res <- dplyr::rename(
    res,
    Pvalue = .data$pvalues,
    AdjustedPvalue = .data$adjPvalues)
```


## t-test


### Input Data Type

All the input data type could be **Raw Counts** or **relative abundance**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **x**: Raw counts matrix (Rows->Features; columns->Samples).

2. **y**: A vector of group labels for testing.

### Procedures

1. Applying *total sum scaling normalization (TSS)* or not to the feature table; 

2. Performing an unpaired Welch’s t-test for each feature to compare the specified groups;

3. Correcting the resulting p-values with the BH method.

### Key output

1. **TaxaID**: taxa ID.

2. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

3. **Occurrence (All/Groups)**: prevalence of taxa.

### Original Syntax

```R
t.test(x, y = NULL,
       alternative = c("two.sided", "less", "greater"),
       mu = 0, paired = FALSE, var.equal = FALSE,
       conf.level = 0.95, ...)
```


### XMAS2 Syntax

```R
# run wilcox rank sum test
  if (paired) {
    if (!is.null(paired_column)) {
      res <- paired_t(x = otu_tab,
                      y = sam_tab,
                      group = "Compvar",
                      gnames = group_names,
                      paired = paired,
                      PID = paired_column,
                      p_ad = p_adjust)
    } else {
      stop("Please provide the paired column for paired test")
    }

  } else {
    res <- apply(otu_tab, 1, function(x, y) {
      dat <- data.frame(value = as.numeric(x), group = y)
      rest <- t.test(data = dat, value ~ group)
      res <- c(rest$statistic, rest$p.value)
      return(res)
    }, sam_tab$Compvar) %>%
      t() %>% data.frame() %>%
      stats::setNames(c("Statistic", "Pvalue")) %>%
      tibble::rownames_to_column("TaxaID") %>%
      dplyr::mutate(AdjustedPvalue = p.adjust(as.numeric(Pvalue), method = p_adjust))
  }
```

## Wilcoxon test

### Input Data Type

All the input data type could be **Raw Counts** or **relative abundance**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **x**: Raw counts matrix (Rows->Features; columns->Samples).

2. **y**: A vector of group labels for testing.


### Procedures

1. CLR (after applying a pseudocount of 1) transforms or not to abundances; 

2. Performing Wilcoxon rank-sum tests for each feature to compare the specified sample groupings;

3. Correcting the resulting p-values with the BH method.

### Key output

1. **TaxaID**: taxa ID.

2. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

3. **Occurrence (All/Groups)**: prevalence of taxa.


### Original Syntax

```R
wilcox.test(x, y = NULL,
            alternative = c("two.sided", "less", "greater"),
            mu = 0, paired = FALSE, exact = NULL, correct = TRUE,
            conf.int = FALSE, conf.level = 0.95,
            tol.root = 1e-4, digits.rank = Inf, ...)
```


### XMAS2 Syntax

```R
# run wilcox rank sum test
  if (paired) {
    if (!is.null(paired_column)) {
      res <- paired_wilcox(x = otu_tab,
                      y = sam_tab,
                      group = "Compvar",
                      gnames = group_names,
                      paired = paired,
                      PID = paired_column,
                      p_ad = p_adjust)
    } else {
      stop("Please provide the paired column for paired test")
    }

  } else {
  res <- apply(otu_tab, 1, function(x, y) {
      dat <- data.frame(value = as.numeric(x), group = y)
      rest <- wilcox.test(data = dat, value ~ group)
      res <- c(rest$statistic, rest$p.value)
      return(res)
    }, sam_tab$Compvar) %>%
    t() %>% data.frame() %>%
    stats::setNames(c("Statistic", "Pvalue")) %>%
    tibble::rownames_to_column("TaxaID") %>%
    dplyr::mutate(AdjustedPvalue = p.adjust(as.numeric(Pvalue), method = p_adjust))
  }
```


## RAIDA

RAIDA is from [@sohn2015robust]

### Vignette

[A robust approach for identifying differentially abundant features in metagenomic samples: RAIDA](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495302/)

### Input Data Type

All the input data type should be **Raw Counts**:

1. 16s rRNA sequencing (amplicon sequencing); 

2. Metagenomic sequencing (absolute abundance);

3. RNA-seq.

### Input Key Arguments

1. **c.data**: a data frame containing counts for both conditions.

2. **n.lib**: a vector containing the numbers of samples for both conditions.

### Procedures

**Approach for Identifying Differential Abundance (RAIDA)** - is a robust approach for identifying differentially abundant features in metagenomic samples across different conditions. It utilizes the ratio between features in a modified zero-inflated log-normal model.

1. Utilizing the ratios between the counts of features in each sample, eliminating possible problems associated with counts on different scales within and between conditions;

2. To fit ratios with zeros by EM algorithm, using a modified ZIL (zero-inflated log normal: sparsity);

3. Constructing a moderated t-statistics for the log ratio of each feature $y_{ij}$ using the estimated mean $\mu$ and variance $r^2$ and obtain $P$ values for the null hypotheses for all features;

4. Adjust $P$ values using a multiple testing correction method

### Key output

1. **TaxaID**: taxa ID.

2. **AdjustedPvalue**: Adjusted Pvalue by `stats::p.adjust`.

3. **Occurrence (All/Groups)**: prevalence of taxa.


### Original Syntax

```R
raida(c.data, 
      n.lib, 
      show.ref.features = FALSE, 
      show.all.features = FALSE, 
      mtcm = "BH", 
      zsc = 0)
```


### XMAS2 Syntax

```R
raida_res <- RAIDA::raida(c.data = otu_tab_reorder,
                            n.lib = n.samples)
```


## Summary

1. The most commonly assumed data distribution was the negative binomial distribution (DESeq2, edgeR, omnibus, mbzinb). 

> No data transformations were performed for negative binomial methods, or metagenomeSeq methods, to try and bring the data to normality as non-normality of data is taken into account in their statistical models. 


2. The remaining parametric methods (ALDEx2 t-test, t-test, limma-voom) all used statistical tests that assumed a Gaussian distribution of the data, therefore, transformations were needed before analysis, which here included a log transform of some kind.


3. Three methods (ALDEx2 Wilcoxon, ANCOM-II, LEfSe) were considered non-parametric (assumes no underlying distribution of data) as they used statistical tests that transformed data to ranks.


4. Two of the four negative binomial methods (DESeq2, edgeR) calculated scaling factors for each sample to account for uneven sequence count. 

5. Cumulative sum scaling (CSS) was used for both metagenomeSeq and MaAsLin2. Total sum scaling (TSS; also referred to as relative abundance) was performed for LEfSe, and for method that did not have a built-in normalization function (t-test) as this strategy is commonly used in the literature. Log-ratio based transformations were used for ALDEx2 methods, ANCOM, and, in addition to TSS, were applied to methods without a built-in normalization function.


## Advantages and disadvantages

1. limma-voom, edgeR, Wilcoxon (CLR), and LEfSe output a high number of significant ASVs on average, while ALDEx2 and ANCOM-II tended to identify only a relatively small number of ASVs as significant;

2. limma-voom, edgeR, Wilcoxon (CLR), and LEfSe methods have been previously found to exhibit a high FDR, in contrast, ANCOM appropriately controls the FDR;

3. We can clearly recommend that users avoid using edgeR (a tool primarily intended for RNA-seq data) as well as LEfSe (without p-value correction) for conducting DA testing with 16S rRNA gene data.

4. Users should also be aware that limma voom and the Wilcoxon (CLR) approaches may perform poorly on unfiltered data that is highly sparse.

5. More generally, we recommend that users employ several methods and focus on significant features identified by most tools. 


