# Examples

Here, we give users two examples to practice the data analysis workflow by **XMAS 2.0**. By the way, we also recommend users handling your own microbiota data in a reasonable manner when you utilize this package or workflow. Pay attention to whether your data fit the methods this package provided.


**Loading packages**

```{r, echo=TRUE, results="hide", warning=FALSE, message=FALSE}
library(XMAS2)
library(dplyr)
library(tibble)
library(phyloseq)
library(ggplot2)
library(ggpubr)
library(SummarizedExperiment)
```

## Workflow description

The standard data analysis of 16S and MGS data by **XMAS 2.0**.

* 16S 
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.height=8, fig.width=8, fig.align="center", fig.cap="Functions of XMAS 2.0 in 16s"}
library(Gmisc, quietly = TRUE)
library(glue)
library(htmlTable)
library(grid)
library(magrittr)

ReadsTrack <- boxGrob("Step1: Reads' Track of DADA2\nplot_Dada2Track()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
ImportData <- boxGrob("Step2: Importing data\nimport_dada2_taxa()/get_dada2_phyloseq()/summarize_phyloseq()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
RefCheck <- boxGrob("Step3: BRS checking\nsummarize_taxa()/run_RefCheck()/get_GroupPhyloseq()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
Rarefaction <- boxGrob("Step4: Rarefaction curves\nplot_RarefCurve()/summarize_phyloseq()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
Normalization <- boxGrob("Step5: Normalization (Rarefy or not)",
                      txt_gp = gpar(col = "darkblue", cex = .8))
Rarefy <- boxGrob("Rarefy OTU table [Optional]\nnorm_rarefy()",
                      txt_gp = gpar(col = "darkgrey", cex = .8))

Extract <- boxGrob("Step6: Extract specific taxonomic level\nsummarize_taxa()",
                      txt_gp = gpar(col = "darkblue", cex = .8))

GlobalView <- boxGrob("Step7: GlobalView",
                      txt_gp = gpar(col = "darkblue", cex = .8)) 
GlobalView_a <- boxGrob("Step7.1: Alpha diveristy\nrun_alpha_diversity()/plot_boxplot()",
                      txt_gp = gpar(col = "darkblue", cex = .6)) 
GlobalView_b <- boxGrob("Step7.2: Beta diveristy\nrun_beta_diversity()/run_permanova()/run_ordination()/plot_Ordination()",
                      txt_gp = gpar(col = "darkblue", cex = .6))  
GlobalView_c <- boxGrob("Step7.3: Composition\nplot_stacked_bar_XIVZ()/plot_core_taxa()/core_members()",
                      txt_gp = gpar(col = "darkblue", cex = .6))  

Differential <- boxGrob("Step8: DifferentialAnalysis\nrun_filter()/run_trim()/run_lefse()/plot_lefse()/run_wilcox()/plot_volcano()",
                      txt_gp = gpar(col = "darkblue", cex = .8))  


grid.newpage()
vert <- spreadVertical(ReadsTrack,
                       ImportData = ImportData,
                       RefCheck = RefCheck,
                       Rarefaction = Rarefaction,
                       Normalization = Normalization,
                       #Rarefy = Rarefy,
                       Extract = Extract,
                       GlobalView = GlobalView,
                       grps = GlobalView_a,
                       Differential = Differential)
grps <- alignVertical(reference = vert$grps,
                      GlobalView_a, GlobalView_b, GlobalView_c) %>%
  spreadHorizontal()

vert$grps <- NULL

Rarefy <- moveBox(Rarefy,
                  x = .8,
                  y = coords(vert$Extract)$top + 
                      Gmisc::distance(vert$Normalization, vert$Extract, half = TRUE, center = FALSE))

for (i in 1:(length(vert) - 1)) {
  connectGrob(vert[[i]], vert[[i + 1]], type = "vert") %>%
    print
}

connectGrob(vert$GlobalView, grps[[1]], type = "N")
connectGrob(vert$GlobalView, grps[[2]], type = "N")
connectGrob(vert$GlobalView, grps[[3]], type = "N")

connectGrob(vert$Extract, Rarefy, type = "L")

vert
grps
Rarefy
```


* MGS 
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.height=6, fig.width=8, fig.align="center", fig.cap="Functions of XMAS 2.0 in MGS"}
ImportData <- boxGrob("Step1: Importing data\nimport_metaphlan_taxa()/get_metaphlan_phyloseq()",
                      txt_gp = gpar(col = "darkblue", cex = .8))

TransformLOD <- boxGrob("Transform LOD into Zeros\naggregate_LOD_taxa()",
                        txt_gp = gpar(col = "grey", cex = .8))

RefCheck <- boxGrob("Step2: BRS checking\nsummarize_taxa()/run_RefCheck()/get_GroupPhyloseq()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
Extract <- boxGrob("Step3: Extract specific taxonomic level\nsummarize_taxa()",
                      txt_gp = gpar(col = "darkblue", cex = .8))
GlobalView <- boxGrob("Step4: GlobalView",
                      txt_gp = gpar(col = "darkblue", cex = .8)) 
GlobalView_a <- boxGrob("Step4.1: Alpha diveristy\nrun_alpha_diversity()/plot_boxplot()",
                      txt_gp = gpar(col = "darkblue", cex = .6)) 
GlobalView_b <- boxGrob("Step4.2: Beta diveristy\nrun_beta_diversity()/run_permanova()/run_ordination()/plot_Ordination()",
                      txt_gp = gpar(col = "darkblue", cex = .6))  
GlobalView_c <- boxGrob("Step4.3: Composition\nplot_stacked_bar_XIVZ()/plot_core_taxa()/core_members()",
                      txt_gp = gpar(col = "darkblue", cex = .6))  

Differential <- boxGrob("Step5: DifferentialAnalysis\nrun_filter()/run_trim()/run_lefse()/plot_lefse()/run_wilcox()/plot_volcano()",
                      txt_gp = gpar(col = "darkblue", cex = .8))  


grid.newpage()
vert <- spreadVertical(ImportData = ImportData,
                       RefCheck = RefCheck,
                       Extract = Extract,
                       GlobalView = GlobalView,
                       grps = GlobalView_a,
                       Differential = Differential)
grps <- alignVertical(reference = vert$grps,
                      GlobalView_a, GlobalView_b, GlobalView_c) %>%
  spreadHorizontal()

vert$grps <- NULL

TransformLOD <- moveBox(TransformLOD,
                  x = .8,
                  y = coords(vert$RefCheck)$top + 
                      Gmisc::distance(vert$RefCheck, vert$Extract, half = TRUE, center = FALSE))

for (i in 1:(length(vert) - 1)) {
  connectGrob(vert[[i]], vert[[i + 1]], type = "vert") %>%
    print
}

connectGrob(vert$GlobalView, grps[[1]], type = "N")
connectGrob(vert$GlobalView, grps[[2]], type = "N")
connectGrob(vert$GlobalView, grps[[3]], type = "N")

connectGrob(vert$ImportData, TransformLOD, type = "L")

vert
grps
TransformLOD
```


## Amplicon sequencing (16s)

The upstream process is performed by in-house pipeline. and this example just shows how to perform downstream data analysis. In briefly, the example comprises the following steps:

1. Converting inputs into phyloseq object;

2. Quality Evaluation;

3. Pre-Processing Data;

4. Diversity analysis;

5. Ordination analysis;

6. Composition analysis;

7. Differential analysis.


### Converting inputs into phyloseq-class object

**dada2** result from standardized_analytics_workflow_R_function.

1. /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/xdada2/dada2_res.rds

2. /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/fasta2tree/tree.nwk

3. /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/metadata.txt

```{r, warning=FALSE, message=FALSE}
# dada2 results from in-house 16s pipeline
dada2_res <- readRDS(
  system.file(
    "extdata", "dada2_res.rds",
    package = "XMAS2"    
    )
)

# the metadata matches to dada2 result
sam_tab <- read.table(
    system.file(
        "extdata", "dada2_metadata.tsv",
        package = "XMAS2"
    ),
    sep = "\t",
    header = TRUE,
    stringsAsFactors = FALSE
)

# tree file from dada2 reference data silva
tree <- phyloseq::read_tree(
  system.file(
    "extdata", "tree.nwk",
    package = "XMAS2"    
    )
)

tax_tab <- import_dada2_taxa(dada2_res$tax_tab)
otu_tab <- dada2_res$seq_tab
sam_tab <- sam_tab %>% tibble::column_to_rownames("seqID")

# Shouldn't use the Total Number as SampleID (wrong: 123456; right: X123456)
rownames(otu_tab) <- paste0("S", rownames(otu_tab))
rownames(sam_tab) <- paste0("S", rownames(sam_tab))

dada2_ps <- get_dada2_phyloseq(seq_tab = otu_tab, 
                         tax_tab = tax_tab, 
                         sam_tab = sam_tab, 
                         phy_tree = tree)
dada2_ps
```

Here, the phyloseq object comprises five components (*OTU Table*, *Sample Data*, *Taxonomy Table*, *Phylogenetic Tree* and *DNAStringSet*).

####  Summarize phyloseq-class object
```{r, warning=FALSE, collapse=TRUE}
summarize_phyloseq(dada2_ps)
```

The minus account of the OTU counts is *51181* in the phyloseq object, and we can use it as the threshold to rarefy. 

Notice the **Sparsity (0.865)**, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.


### Quality Control

Quality control of DADA2 results will help us have more rational determinations on the further data analysis. 

#### Reads' track by DADA2
```{r, warning=FALSE, message=FALSE, fig.align="center", fig.width=6, fig.height=4, fig.cap="Reads' track by DADA2 (16s example)"}
plot_Dada2Track(data=dada2_res$reads_track)
```


The percentage of the final remained read counts approximate **70%**, indicating that we should consider the sequence depth for analysis when we build the sequence library.

#### Spike-in sample (BRS) assessment

* Extract the genus level phyloseq and getting the BRS_ID
```{r, warning=FALSE, message=FALSE}
dada2_ps_genus <- summarize_taxa(ps = dada2_ps, taxa_level = "Genus")
sample_data(dada2_ps_genus)
```

**The BRS_ID is *S8005* **.

* Run `run_RefCheck`
```{r, warning=FALSE, collapse=TRUE}
run_RefCheck(
    ps=dada2_ps_genus,
    BRS_ID="S8005",
    Reference=NULL,
    Ref_type="16s",
    Save=NULL)
```


#### Spike-in sample's remove
```{r, warning=FALSE, message=FALSE}
dada2_ps_remove_BRS <- get_GroupPhyloseq(
                     ps = dada2_ps,
                     group = "Group",
                     group_names = "QC",
                     discard = TRUE)
dada2_ps_remove_BRS
```


#### Rarefaction curves
```{r, warning=FALSE, message=FALSE, fig.align="center", fig.width=6, fig.height=4, fig.cap="Rarefaction curves (16s example)"}
plot_RarefCurve(ps = dada2_ps_remove_BRS,
               taxa_level = "OTU",
               step = 400,
               label = "Group",
               color = "Group")
```

The result showed that all the samples had **different sequence depth** but had the full sample richness.


### Data processing

This part has too may procedures and we only choose some of them. Please go to **Chapter 6** to see more approaches and details for being familiar with this part.

#### Rarefy otu counts

From previous results of quality evaluation, the sequence depth of samples are different which have effects on the downstream analysis. Here, choosing the **rarefy** (Normaliztion method: random subsampling counts to the smallest library size) to get the equal sample sums.

```{r, warning=FALSE, message=FALSE}
dada2_ps_rarefy <- norm_rarefy(dada2_ps_remove_BRS, size = 51181, rng_seed = 123)
dada2_ps_rarefy
```

In addition, we could also perform some other normalization methods on the rarefied phyloseq object. By the way, we didn't transform the data by using log algorithm because the count matrix is required by the following data analysis methods.

#### Extracting specific taxonomic level
```{r, warning=FALSE, message=FALSE}
dada2_ps_rare_genus <- summarize_taxa(dada2_ps_rarefy, taxa_level = "Genus")
dada2_ps_rare_genus
```

#### Filtering the low relative abundance or unclassified taxa by the threshold (total counts < 10)
```{r, warning=FALSE, message=FALSE}
dada2_ps_rare_genus_filter <- run_filter(dada2_ps_rare_genus, cutoff = 10, unclass = TRUE)
dada2_ps_rare_genus_filter 
```

#### Trimming the taxa with low occurrence less than threshold
```{r, warning=FALSE, message=FALSE}
dada2_ps_rare_genus_filter_trim <- run_trim(dada2_ps_rare_genus_filter, cutoff = 0.2, trim = "feature")
dada2_ps_rare_genus_filter_trim
```

Finally, we obtained the final phyloseq-class object _dada2\_ps\_rare\_genus\_filter\_trim_ and changed its name.

### Diversity analysis

#### Alpha diveristy

* Calculate the alpha diversity 

**Notes:** the otu table must be counts matrix (*rarefied but not trim counts matrix*) when you choose *Observed* etc measures. 
```{r, warning=FALSE, message=FALSE}
dada_ps_rare_genus_alpha <- run_alpha_diversity(ps=dada2_ps_rare_genus, measures = c("Shannon", "Chao1", "Observed"))
print(dada_ps_rare_genus_alpha)
```

* visualization
```{r, warning=FALSE, message=FALSE, fig.width=6, fig.height=4, fig.align="center", fig.cap="Alpha diversity (16s example)"}
plot_boxplot(data=dada_ps_rare_genus_alpha,
             y_index = c("Shannon", "Chao1", "Observed"),
             group = "Group",
             group_names = c("AA", "BB"),
             group_color = c("red", "blue"),
             method = "wilcox.test")
```


#### Beta diversity

* beta dipersion
```{r, warning=FALSE, message=FALSE, fig.width=5, fig.height=4, fig.align="center", fig.cap="Beta diversity (16s example)"}
dada2_ps_beta <- run_beta_diversity(ps=dada2_ps_rare_genus_filter_trim, method = "bray", group = "Group")
dada2_ps_beta$BetaDispersion
```


### PERMANOVA + Ordination

#### PERMANOVA
```{r, warning=FALSE, message=FALSE}
dada2_ps_per <- run_permanova(ps=dada2_ps_rare_genus_filter_trim, method = "bray", columns = "Group")
print(dada2_ps_per)
```

The PERMANOVA result of the Group (`Pr(>F)` > 0.05) revealed that the two groups had not the distinct patterns of microbial community.


#### Ordination

We performed ordination by using Principal Coordinate Analysis (PCoA). If you want to try other methods please go to see **Chapter 8** for more details.
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=8, fig.align="center", fig.cap="PCoA (16s example)"}
dada2_ps_ordination <- run_ordination(
                       ps = dada2_ps_rare_genus_filter_trim,
                       group = "Group",
                       method = "PCoA")

plot_Ordination(ResultList = dada2_ps_ordination, 
                group = "Group", 
                group_names = c("AA", "BB"),
                group_color = c("blue", "red"))
```


### Microbial composition

A whole picture of the microbial composition.

#### Stacked barplot 

* **XMAS** package
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=7, fig.align="center", fig.cap="Microbial composition (16s example)"}
plot_StackBarPlot(
        ps=dada2_ps_rarefy,
        taxa_level="Phylum",
        group = "Group",
        cluster = TRUE)
```

* **XVIZ** package
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=9, fig.align="center", fig.cap="Microbial composition (16s example) XVIZ"}
plot_stacked_bar_XIVZ(
        phyloseq=dada2_ps_rarefy,
        level="Phylum",
        feature = "Group")
```

#### Core microbiota

* convert absolute abundance into relative abundance 
```{r}
dada2_ps_rare_genus_filter_trim_rb <- XMAS2::normalize(dada2_ps_rare_genus_filter_trim, method = "TSS")
dada2_ps_rare_genus_filter_trim_rb
```

* visualization
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=7, fig.align="center", fig.cap="Core taxa (16s example)"}
prevalences <- seq(0.05, 1, 0.05)
detections <- 10^seq(log10(1e-3), log10(0.2), length = 10)

pl_core <- plot_core_taxa(dada2_ps_rare_genus_filter_trim_rb, 
                    plot.type = "heatmap", 
                    colours = gray(seq(0, 1, length=5)),
                    prevalences = prevalences, 
                    detections = detections, 
                    min.prevalence = 0.5)+
    xlab("Detection Threshold (Relative Abundance (%))")

pl_core
```

The degree of color indicates the size of abundance and prevalence.

* Use `core_members` to obtain the core taxa. detection for abundance and prevalence for occurrence.
```{r, warning=FALSE, message=FALSE}
core_taxa_name <- core_members(dada2_ps_rare_genus_filter_trim_rb, detection = 0.01, prevalence = 0.8)
print(core_taxa_name)
```

**Result**:

Only 2 genera (g__Bifidobacterium and g__Blautia) passed the threshold of detection and prevalence which we choose.

### Differential Analysis

There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to **Chapter 10** to see more detials.

#### Liner discriminant analysis (LDA) effect size (LEfSe)

* Calculation
```{r, warning=FALSE, message=FALSE}
dada2_ps_lefse <- run_lefse(
                    dada2_ps_rare_genus_filter_trim,
                    group = "Group",
                    group_names = c("AA", "BB"),
                    Lda = 2)
head(dada2_ps_lefse)
```

* Visualization
```{r, warning=FALSE, message=FALSE, fig.width=6, fig.height=4, fig.align="center", fig.cap="Lefse analysis (16s example)"}
# # don't run this code when you do lefse in reality
# dada2_ps_lefse$LDA_Score <- dada2_ps_lefse$LDA_Score * 1000

plot_lefse(
  da_res = dada2_ps_lefse,
  x_index = "LDA_Score",
  x_index_cutoff = 1,
  group_color = c("green", "red"))
```


#### Wilcoxon Rank-Sum test

* Calculation
```{r, warning=FALSE, message=FALSE}
dada2_ps_wilcox <- run_wilcox(
                    dada2_ps_rare_genus_filter_trim,
                    group = "Group",
                    group_names = c("AA", "BB"))

head(dada2_ps_wilcox)
```


* Volcano
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=6, fig.align="center", fig.cap="Wilcoxon Rank-Sum test (16s example)"}
plot_volcano(
    dada2_ps_wilcox,
    group_names = c("AA", "BB"),
    x_index = "Log2FoldChange (Rank)\nAA_vs_BB",
    x_index_cutoff = 0.5,
    y_index = "Pvalue",
    y_index_cutoff = 0.05,
    group_color = c("red", "grey", "blue"),
    topN = 5)
```


## Metagenomics (MGS)


The metagenomic data analysis pipeline is just the same as 16s. In briefly, the example comprises the following steps:

1. Converting inputs into phyloseq object;

2. Quality Evaluation;

3. Pre-Processing Data;

4. Diversity analysis;

5. Ordination analysis;

6. Composition analysis;

### Converting inputs into phyloseq-class object

The result of the in-house **[Metaphlan2/3](https://huttenhower.sph.harvard.edu/metaphlan2)** pipeline:

1. /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metaphlan2_merged.tsv

2. /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metadata.txt

```{r, warning=FALSE, message=FALSE}
metaphlan2_res <- read.table(
    system.file(
        "extdata", "metaphlan2_merged.tsv",
        package = "XMAS2"
    ),
    header = TRUE, 
    stringsAsFactors = FALSE
)
metaphlan2_sam <- read.table(
    system.file(
        "extdata", "metaphlan2_metadata.tsv",
        package = "XMAS2"
    ),
    sep = "\t",
    header = TRUE,
    stringsAsFactors = FALSE
)

metaphlan2_res_list <- import_metaphlan_taxa(metaphlan2_res, taxa_level = "Species")
otu_tab <- metaphlan2_res_list$abu_tab
tax_tab <- metaphlan2_res_list$tax_tab
sam_tab <- metaphlan2_sam %>% tibble::column_to_rownames("SampleID")

metaphlan2_ps <- get_metaphlan_phyloseq(otu_tab = otu_tab, 
                                   sam_tab = sam_tab,
                                   tax_tab = tax_tab)
metaphlan2_ps
```

Here, the phyloseq object comprises three components (*OTU Table*, *Sample Data* and *Taxonomy Table*).

####  Summarize phyloseq-class object
```{r, warning=FALSE}
summarize_phyloseq(metaphlan2_ps)
```

Notice the **Sparsity (0.713)**, indicating the data has many zeros and pay attention to the downstream data analysis.


### Quality Control

#### Spike-in sample (BRS) assessment

* Extract the species level phyloseq and obtain the BRS_ID
```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_species <- summarize_taxa(metaphlan2_ps, taxa_level = "Species")
metaphlan2_ps_species@sam_data
```

* Run `run_RefCheck`
```{r, warning=FALSE, collapse=TRUE}
run_RefCheck(
    ps=metaphlan2_ps_species,
    BRS_ID="refE",
    Reference=NULL,
    Ref_type="MGS",
    Save=NULL)
```

The spike-in samples didn't pass the cutoff and failed to add the the Reference Matrix.

#### Spike-in sample's remove
```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_remove_BRS <- get_GroupPhyloseq(
                     ps = metaphlan2_ps,
                     group = "Group",
                     group_names = "QC",
                     discard = TRUE)
metaphlan2_ps_remove_BRS
```

### Data processing

This part has too may procedures and we only choose some of them. Please go to **Chapter 6** to see more approaches and details for being familiar with this part.

#### Extracting specific taxonomic level
```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_species <- summarize_taxa(metaphlan2_ps_remove_BRS, taxa_level = "Species")
metaphlan2_ps_species
```

#### Filtering the low relative abundance or unclassified taxa by the threshold (total counts < 1e-4)

The condition to filter low relative abundance is according to this article [@thingholm2019obese].

> Species from taxonomic profiles were retained for further analysis if their mean relative abundance exceeded 0.005
(0.5%) across the dataset with a minimum abundance of 0.05 (5%) in at least one sample and non-zero abundance in at least
60% of samples. 

There are three conditions

1. Mean relative abundance: 0.005; 

2. Minimum relative abundance: 0.05;

3. Occurrence: 60%. 

Here, we use 0.01 (the 1e-4 regarded as 0.01 compared to the Referece because Metaphlan2 data had been divided 100)

```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_species_filter <- run_filter(metaphlan2_ps_species, cutoff = 1e-4, unclass = TRUE)
metaphlan2_ps_species_filter 
```

#### Trimming the taxa with low occurrence less than threshold
```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_species_filter_trim <- run_trim(metaphlan2_ps_species_filter, cutoff = 0.1, trim = "feature")
metaphlan2_ps_species_filter_trim
```

Finally, we obtained the final phyloseq-class object _metaphlan2\_ps\_species\_filter\_trim_ and changed its name.

### Diversity analysis

#### Alpha diveristy

* Calculate the alpha diversity 

**Notes:** choosing the measures (*Shannon, Simpson and InvSimpson*) only for relative abundance. 
```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_genus_alpha <- run_alpha_diversity(ps=metaphlan2_ps_remove_BRS, measures = c("Shannon", "Simpson", "InvSimpson"))
head(metaphlan2_ps_genus_alpha)
```

* visualization
```{r, warning=FALSE, message=FALSE, fig.width=6, fig.height=4, fig.align="center", fig.cap="Alpha diversity (MGS example)"}
plot_boxplot(data=metaphlan2_ps_genus_alpha,
             y_index = c("Shannon", "Simpson", "InvSimpson"),
             group = "Group",
             group_names = c("AA", "BB"),
             group_color = c("red", "blue"),
             method = "wilcox.test")
```


#### Beta diversity

* beta dipersion
```{r, warning=FALSE, message=FALSE, fig.width=5, fig.height=4, fig.align="center", fig.cap="Beta diversity (MGS example)"}
metaphlan2_ps_beta <- run_beta_diversity(ps=metaphlan2_ps_species_filter_trim, method = "bray", group = "Group")
metaphlan2_ps_beta$BetaDispersion
```


### PERMANOVA + Ordination

#### PERMANOVA
```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_per <- run_permanova(ps=metaphlan2_ps_species_filter_trim, method = "bray", columns = "Group")
head(metaphlan2_ps_per)
```

The PERMANOVA result of the Group (AdjustedPvalue < 0.05) revealed that the two groups had the distinct patterns of microbial community.


#### Ordination

We performed ordination by using Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP). If you want to try other methods please go to see **Chapter 8** for more details.
```{r, warning=FALSE, message=FALSE, fig.width=5, fig.height=4, fig.align="center", fig.cap="PCoA (MGS example)"}
metaphlan2_ps_ordination <- run_ordination(
                       ps = metaphlan2_ps_species_filter_trim,
                       group = "Group",
                       method = "UMAP")

plot_Ordination(ResultList = metaphlan2_ps_ordination, 
                group = "Group", 
                group_names = c("AA", "BB"),
                group_color = c("blue", "red"),
                sample = TRUE,
                sidelinechart = FALSE,
                circle_type = "ellipse_line",
                sideboxplot = TRUE)
```


### Microbial composition

#### Stacked barplot 

* **XVIZ** package
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=9, fig.align="center", fig.cap="Microbial composition (MGS) XVIZ"}
plot_stacked_bar_XIVZ(
        phyloseq = metaphlan2_ps_species,
        level = "Phylum",
        feature = "Group")
```

* **XMAS** package
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=7, fig.align="center", fig.cap="Microbial composition (MGS example)"}
plot_StackBarPlot(
        ps=metaphlan2_ps_species,
        taxa_level="Phylum",
        group = "Group",
        cluster = TRUE)
```

#### Core microbiota

* visualization
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=7, fig.align="center", fig.cap="Core taxa (MGS example)"}
library(RColorBrewer)

prevalences <- seq(0.05, 1, 0.05)
detections <- 10^seq(log10(1e-3), log10(.2), length = 10)

plot_core_taxa(metaphlan2_ps_species_filter_trim, 
               plot.type = "heatmap", 
               colours = rev(brewer.pal(5, "Spectral")),
               prevalences = prevalences, 
               detections = detections, 
               min.prevalence = 0.5)+
    xlab("Detection Threshold (Relative Abundance (%))") +
    theme(axis.text.y = element_text(face="italic"))
```

The degree of color indicates the size of abundance and prevalence.

* Use `core_members` to obtain the core taxa. detection for abundance and prevalence for occurrence.
```{r, warning=FALSE, message=FALSE}
core_taxa_name <- core_members(metaphlan2_ps_species_filter_trim, detection = 0.001, prevalence = 0.5)
print(core_taxa_name)
```

**Result**:

17 species passed the threshold of detection and prevalence which we choose.

### Differential Analysis

There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to **Chapter 10** to see more detials.

#### Liner discriminant analysis (LDA) effect size (LEfSe)

* Calculation
```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_lefse <- run_lefse(
                    metaphlan2_ps_species_filter_trim,
                    group = "Group",
                    group_names = c("AA", "BB"),
                    Lda = 2)
head(metaphlan2_ps_lefse)
```

* Visualization
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=6, fig.align="center", fig.cap="Lefse analysis (MGS example)"}
# # don't run this code when you do lefse in reality
# metaphlan2_ps_lefse$LDA_Score <- metaphlan2_ps_lefse$LDA_Score * 100

plot_lefse(
  da_res = metaphlan2_ps_lefse,
  x_index = "LDA_Score",
  x_index_cutoff = 2,
  group_color = c("green", "red"))
```


#### Wilcoxon Rank-Sum test

* Calculation
```{r, warning=FALSE, message=FALSE}
metaphlan2_ps_wilcox <- run_wilcox(
                    metaphlan2_ps_species_filter_trim,
                    group = "Group",
                    group_names = c("AA", "BB"))

head(metaphlan2_ps_wilcox)
```


* Volcano
```{r, warning=FALSE, message=FALSE, fig.width=8, fig.height=6, fig.align="center", fig.cap="Wilcoxon Rank-Sum test (MGS example)"}
plot_volcano(
    metaphlan2_ps_wilcox,
    group_names = c("AA", "BB"),
    x_index = "Log2FoldChange (Rank)\nAA_vs_BB",
    x_index_cutoff = 0.5,
    y_index = "Pvalue",
    y_index_cutoff = 0.05,
    group_color = c("red", "grey", "blue"),
    topN = 5)
```

## Systematic Information
```{r}
devtools::session_info()
```
