<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Aplha diversity | Data analysis based on XMAS 2.0 package in microbiota data</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Aplha diversity | Data analysis based on XMAS 2.0 package in microbiota data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Aplha diversity | Data analysis based on XMAS 2.0 package in microbiota data" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-06-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="pre-processing.html"/>
<link rel="next" href="beta-diversity.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation<span></span></a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette<span></span></a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors<span></span></a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2<span></span></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>4.2.1</b> Importing results from dada2 pipeline<span></span></a></li>
<li class="chapter" data-level="4.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>4.2.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>4.2.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.2.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>4.2.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.2.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>4.2.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>4.3.1</b> Importing results<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table-1"><i class="fa fa-check"></i><b>4.3.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table-1"><i class="fa fa-check"></i><b>4.3.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.3.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table-1"><i class="fa fa-check"></i><b>4.3.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.3.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object-1"><i class="fa fa-check"></i><b>4.3.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Reads’ track<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>5.6</b> Variability<span></span></a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>5.6.1</b> Coefficient of variation<span></span></a></li>
<li class="chapter" data-level="5.6.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>5.6.2</b> Distribution of number of sequences<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.7</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>6.8</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)<span></span></a></li>
<li class="chapter" data-level="6.9" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.9</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities<span></span></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>7.2.1</b> Non-phylogenetic diversities<span></span></a></li>
<li class="chapter" data-level="7.2.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2.2</b> Phylogenetic diversity<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics<span></span></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>8.6</b> Mantel Test<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.7</b> Analysis of Similarity (ANOSIM)<span></span></a></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>8.8</b> Multi-response permutation procedures (MRPP)<span></span></a></li>
<li class="chapter" data-level="8.9" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.9</b> Ordination<span></span></a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.9.1</b> Principal Component Analysis (PCA)<span></span></a></li>
<li class="chapter" data-level="8.9.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.9.2</b> Principal Coordinate Analysis (PCoA)<span></span></a></li>
<li class="chapter" data-level="8.9.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.9.3</b> t-distributed stochastic neighbor embedding (t-SNE)<span></span></a></li>
<li class="chapter" data-level="8.9.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.9.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.10</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="composition.html"><a href="composition.html"><i class="fa fa-check"></i><b>9</b> Composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="composition.html"><a href="composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="composition.html"><a href="composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.2</b> XVIZ <code>plot_stacked_bar</code><span></span></a></li>
<li class="chapter" data-level="9.3" data-path="composition.html"><a href="composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.3</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="composition.html"><a href="composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="composition.html"><a href="composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus<span></span></a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq<span></span></a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR<span></span></a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM<span></span></a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob<span></span></a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)<span></span></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom-1"><i class="fa fa-check"></i><b>11.2.1</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.2</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.3</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.4</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.2.5" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.5</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot<span></span></a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot<span></span></a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot<span></span></a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density<span></span></a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination<span></span></a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot<span></span></a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot<span></span></a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn<span></span></a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot"><i class="fa fa-check"></i><b>12.10</b> plot the DA results from the significant taxa by double barplot<span></span></a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.11</b> plot_stacked_bar_XIVZ<span></span></a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.12</b> plot_StackBarPlot<span></span></a></li>
<li class="chapter" data-level="12.13" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.13</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>13</b> Principals of Differential Analysis methods<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>13.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>13.2</b> ANCOM-II<span></span></a></li>
<li class="chapter" data-level="13.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>13.3</b> corncob<span></span></a></li>
<li class="chapter" data-level="13.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>13.4</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="13.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>13.5</b> edgeR<span></span></a></li>
<li class="chapter" data-level="13.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>13.6</b> LEfSe<span></span></a></li>
<li class="chapter" data-level="13.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>13.7</b> limma voom<span></span></a></li>
<li class="chapter" data-level="13.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>13.8</b> MaAsLin2<span></span></a></li>
<li class="chapter" data-level="13.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>13.9</b> metagenomeSeq<span></span></a></li>
<li class="chapter" data-level="13.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>13.10</b> t-test<span></span></a></li>
<li class="chapter" data-level="13.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>13.11</b> Wilcoxon test<span></span></a></li>
<li class="chapter" data-level="13.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>13.12</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="13.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>13.13</b> Summary<span></span></a></li>
<li class="chapter" data-level="13.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>13.14</b> Advantages and disadvantages<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>14</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>14.1</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="14.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>14.2</b> Amplicon sequencing (16s)<span></span></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>14.2.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="14.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>14.2.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="14.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>14.2.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="14.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>14.2.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="14.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>14.2.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="14.2.6" data-path="examples.html"><a href="examples.html#microbial-composition"><i class="fa fa-check"></i><b>14.2.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="14.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>14.2.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="examples.html"><a href="examples.html#metagenomics-mgs"><i class="fa fa-check"></i><b>14.3</b> Metagenomics (MGS)<span></span></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>14.3.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="14.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>14.3.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="14.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>14.3.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="14.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>14.3.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="14.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>14.3.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="14.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>14.3.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="14.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>14.3.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>14.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data analysis based on XMAS 2.0 package in microbiota data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="aplha-diversity" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Aplha diversity<a href="aplha-diversity.html#aplha-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Alpha diversity is a measure of microbiome diversity applicable to a single sample. The commonly used metrics/indices are Shannon, Chao1, Observed, Simpson, Gini and Inverse Simpson. The reference of this part is <span class="citation">(<a href="#ref-Sudarshan2020Microbial" role="doc-biblioref">A et al. 2020</a>)</span>.</p>
<blockquote>
<p><strong>It is important to note that, alpha diversity indices are sensitive to noise that is inherent to application of polymerase chain reaction and the sequencing errors.</strong></p>
</blockquote>
<p><strong>Cautions</strong>: <em>Shannon, Simpson and InvSimpson</em> measures are both for relative or absolute abundance.</p>
<p>As defined by Shannon, the index of Shannon index is given as follow:</p>
<p><span class="math display">\[H^{&#39;} = \sum_{i=1}^{S}p_{i}\,\ln\,p_{i}\]</span>
where <span class="math inline">\(p_{i}\)</span> is the proportion of individuals (or relative abundance) of species <span class="math inline">\(i\)</span> in the community and <span class="math inline">\(S\)</span> is the total number of species present so that <span class="math inline">\(\sum_{i=1}^{S}p_{i} = 1\)</span>.</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="aplha-diversity.html#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb122-2"><a href="aplha-diversity.html#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb122-3"><a href="aplha-diversity.html#cb122-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb122-4"><a href="aplha-diversity.html#cb122-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb122-5"><a href="aplha-diversity.html#cb122-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb122-6"><a href="aplha-diversity.html#cb122-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb122-7"><a href="aplha-diversity.html#cb122-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
<div id="equal-sample-sums" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Equal sample sums<a href="aplha-diversity.html#equal-sample-sums" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Remove BRS</li>
</ul>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="aplha-diversity.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;dada2_ps&quot;</span>)</span>
<span id="cb123-2"><a href="aplha-diversity.html#cb123-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb123-3"><a href="aplha-diversity.html#cb123-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb123-4"><a href="aplha-diversity.html#cb123-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb123-5"><a href="aplha-diversity.html#cb123-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb123-6"><a href="aplha-diversity.html#cb123-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb123-7"><a href="aplha-diversity.html#cb123-7" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
<ul>
<li>Summary counts</li>
</ul>
<p>The previous result from <em>Rarefaction Curves</em> had demonstrated that the sequence depth of each sample has a larger significance and normalizing the samples’ depth into equal sampling depth is necessary.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="aplha-diversity.html#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sample_sums</span>(dada2_ps_remove_BRS))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   51181   58608   61435   61257   64283   71667</code></pre>
<p>As is evident there is a large difference in the number of reads. Minimum is <em>51181</em> and maximum is <em>71667</em>!! There is a <strong>~1.4X</strong> difference.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="aplha-diversity.html#cb127-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(dada2_ps_remove_BRS, <span class="at">size =</span> <span class="dv">51181</span>, <span class="at">rng_seed =</span> <span class="dv">123</span>)</span>
<span id="cb127-2"><a href="aplha-diversity.html#cb127-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sample_sums</span>(dada2_ps_rarefy))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   51181   51181   51181   51181   51181   51181</code></pre>
<p>Quickly check taxa prevalence</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="aplha-diversity.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_taxa_prevalence</span>(dada2_ps_rarefy, <span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-63"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-63-1.png" alt="Taxa prevalence after rarefying" width="672" />
<p class="caption">
Figure 7.1: Taxa prevalence after rarefying
</p>
</div>
</div>
<div id="diversities" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Diversities<a href="aplha-diversity.html#diversities" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="aplha-diversity.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;amplicon_ps&quot;</span>)</span>
<span id="cb130-2"><a href="aplha-diversity.html#cb130-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sample_sums</span>(amplicon_ps))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    1114    2139    4965    5357    8438   11449</code></pre>
<p>As is evident there is a large difference in the number of reads. Minimum is <em>1114</em> and maximum is <em>11449</em>!! There is a <strong>~10X</strong> difference!</p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="aplha-diversity.html#cb132-1" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(amplicon_ps, <span class="at">size =</span> <span class="dv">1114</span>, <span class="at">rng_seed =</span> <span class="dv">123</span>)</span>
<span id="cb132-2"><a href="aplha-diversity.html#cb132-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">sample_sums</span>(amplicon_ps))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    1114    2139    4965    5357    8438   11449</code></pre>
<div id="non-phylogenetic-diversities" class="section level3 hasAnchor" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Non-phylogenetic diversities<a href="aplha-diversity.html#non-phylogenetic-diversities" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="xmas" class="section level4 hasAnchor" number="7.2.1.1">
<h4><span class="header-section-number">7.2.1.1</span> XMAS<a href="aplha-diversity.html#xmas" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ol style="list-style-type: decimal">
<li>Calculation</li>
</ol>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="aplha-diversity.html#cb134-1" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps=</span>amplicon_ps_rarefy, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>)) </span>
<span id="cb134-2"><a href="aplha-diversity.html#cb134-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(amplicon_ps_rarefy_alpha)</span></code></pre></div>
<pre><code>##   TempRowNames SampleType Year Month Day Subject ReportedAntibioticUsage DaysSinceExperimentStart         Description
## 1       L1S140        gut 2008    10  28       2                     Yes                        0   2_Fece_10_28_2008
## 2       L1S208        gut 2009     1  20       2                      No                       84    2_Fece_1_20_2009
## 3         L1S8        gut 2008    10  28       1                     Yes                        0   1_Fece_10_28_2008
## 4       L1S281        gut 2009     4  14       2                      No                      168    2_Fece_4_14_2009
## 5       L3S242 right palm 2008    10  28       1                     Yes                        0 1_R_Palm_10_28_2008
## 6       L2S309  left palm 2009     1  20       2                      No                       84  2_L_Palm_1_20_2009
##   Observed    Chao1  se.chao1  Shannon
## 1      161 332.7308  50.17139 3.149765
## 2      183 365.0000  50.23137 3.617229
## 3      112 303.1538  69.80534 2.817767
## 4      212 587.9130 102.79585 3.892763
## 5      120 278.8824  54.21486 2.911609
## 6      296 560.0526  55.47859 4.494509</code></pre>
<ul>
<li><strong>Visualization</strong></li>
</ul>
<ol style="list-style-type: decimal">
<li>single measure</li>
</ol>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="aplha-diversity.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> amplicon_ps_rarefy_alpha,</span>
<span id="cb136-2"><a href="aplha-diversity.html#cb136-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb136-3"><a href="aplha-diversity.html#cb136-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb136-4"><a href="aplha-diversity.html#cb136-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="cn">NULL</span>,</span>
<span id="cb136-5"><a href="aplha-diversity.html#cb136-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="cn">NULL</span>,</span>
<span id="cb136-6"><a href="aplha-diversity.html#cb136-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb136-7"><a href="aplha-diversity.html#cb136-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">ref_group =</span> <span class="cn">NULL</span>,</span>
<span id="cb136-8"><a href="aplha-diversity.html#cb136-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb136-9"><a href="aplha-diversity.html#cb136-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">outlier =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-67"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-67-1.png" alt="Alpha diversity from XMAS(one measure)" width="576" />
<p class="caption">
Figure 7.2: Alpha diversity from XMAS(one measure)
</p>
</div>
<ol start="2" style="list-style-type: decimal">
<li>multiple measures</li>
</ol>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="aplha-diversity.html#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> amplicon_ps_rarefy_alpha,</span>
<span id="cb137-2"><a href="aplha-diversity.html#cb137-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb137-3"><a href="aplha-diversity.html#cb137-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb137-4"><a href="aplha-diversity.html#cb137-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb137-5"><a href="aplha-diversity.html#cb137-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb137-6"><a href="aplha-diversity.html#cb137-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb137-7"><a href="aplha-diversity.html#cb137-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">ref_group =</span> <span class="st">&quot;gut&quot;</span>,</span>
<span id="cb137-8"><a href="aplha-diversity.html#cb137-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb137-9"><a href="aplha-diversity.html#cb137-9" aria-hidden="true" tabindex="-1"></a>             <span class="at">outlier =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-68"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-68-1.png" alt="Alpha diversity from XMAS(multiple measures)" width="864" />
<p class="caption">
Figure 7.3: Alpha diversity from XMAS(multiple measures)
</p>
</div>
</div>
<div id="microbiome-1" class="section level4 hasAnchor" number="7.2.1.2">
<h4><span class="header-section-number">7.2.1.2</span> microbiome<a href="aplha-diversity.html#microbiome-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Calculation</strong></li>
</ul>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="aplha-diversity.html#cb138-1" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy_alpha_v2 <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">alpha</span>(amplicon_ps_rarefy, <span class="at">index =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb138-2"><a href="aplha-diversity.html#cb138-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(amplicon_ps_rarefy_alpha_v2)</span></code></pre></div>
<div id="htmlwidget-5944fc29754bf6fa5db1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5944fc29754bf6fa5db1">{"x":{"filter":"none","vertical":false,"data":[["L1S140","L1S208","L1S8","L1S281","L3S242","L2S309","L2S357","L4S112","L2S155","L2S382","L4S63","L2S222","L3S341","L3S360","L2S204","L4S137","L3S294","L2S175","L5S203","L6S93","L6S20","L6S68","L1S257","L3S378","L1S105","L5S174","L5S155","L5S222","L5S104","L1S57","L1S76","L5S240","L3S313","L2S240"],[161,183,112,212,120,296,256,287,252,253,337,442,321,408,296,245,210,286,124,132,113,104,200,174,154,96,100,102,101,175,151,103,253,133],[332.730769230769,365,303.153846153846,587.913043478261,278.882352941176,560.052631578947,574.025641025641,816.371428571429,620.15625,378.234042553192,831.042553191489,1000.22388059701,723.061224489796,845.635135135135,843.25,425.622222222222,392.083333333333,749.323529411765,425.875,309.5,206.157894736842,176.058823529412,418.903225806452,400.384615384615,281.730769230769,166.714285714286,166.352941176471,190.4,219.461538461538,553.066666666667,347.894736842105,248.090909090909,510.4,244.714285714286],[8.60500076273419,13.1679045880905,8.47107810345534,15.2460257008772,8.34103587799599,25.6076100862531,31.5742926928557,25.2347797795762,27.0569921074434,57.5174267704857,45.5511672294817,110.212788632327,30.8874508437453,86.2162011949423,12.4268605303212,29.6463449593884,22.146405881933,22.3643178951162,18.872741650952,23.2152798563304,9.29334411694225,13.0086165328414,16.5492612151278,9.57499537065613,15.7331068231953,9.41832366959109,11.2893764896385,11.6696381554201,14.1118489879463,16.7146512943458,11.4474577521954,11.3569441393953,21.1362877678237,17.0893717811011],[0.883788505361822,0.924057772950114,0.881951271398135,0.934409135887626,0.880110814216968,0.960949108619206,0.968328665040016,0.960372152690258,0.963040976763825,0.982613964912055,0.978046665742678,0.990926642793369,0.967624392020603,0.988401251897669,0.919529152390499,0.966269029070198,0.954845946320536,0.955285915506577,0.947013527843764,0.956924921595235,0.892396107642571,0.92312787470709,0.939574341899571,0.895561307208081,0.936439762900122,0.893823992986279,0.911421148819174,0.914307540072651,0.929137563698835,0.940172248742139,0.912644359852892,0.911948144877179,0.952688002217574,0.941484098256562],[3.1497652623412,3.61722919652132,2.81776688827325,3.89276275697651,2.9116085568317,4.49450900228555,4.35503742499801,4.39380762780162,4.29719612911591,4.78447479601937,4.8428133115822,5.49747548077037,4.59153700621914,5.28574222003789,4.10143172536166,4.40962896994681,4.04956612762706,4.37844645899076,3.53403735993864,3.6727384673317,3.02776247130175,3.18021244540142,3.78989812743537,3.25756506054248,3.52377476976675,3.01969954220326,3.1351278178536,3.11615865090018,3.26889188559163,3.69938738514812,3.37546124443187,3.11185664439608,4.21265101941164,3.47782921450368],[51.6641364375989,62.2759711533285,31.0438978006773,77.6204783065867,34.1334308281282,131.758983710791,104.064855708661,125.20208990477,101.493950100822,102.133431440571,164.235085615154,270.888501715264,151.034006100605,232.116094264133,131.758983710791,97.0775428181732,76.5110101892887,124.485576685834,35.7183145821745,38.9686580260582,31.4242492029377,28.0610038057045,71.0789169191763,57.830151725378,48.467067220361,25.1847595164987,26.6095580089867,27.3319496352217,26.9699208707475,58.3169157893264,47.1229886717681,27.6956440160304,102.133431440571,39.3825312887874],[3,5,3,8,3,16,12,14,12,27,22,58,14,40,8,14,9,12,7,8,3,5,7,4,6,4,5,4,6,6,4,4,9,6],[0.952551936393947,0.896940086921545,0.965608168761221,0.773330849225975,0.968798623578695,0.928957009073706,0.943958987881508,0.957424980764298,0.962422700977459,0.967946579998723,0.892812811319649,0.877519354655272,0.958131288556296,0.915713996550146,0.96358083846863,0.979936247389441,0.998221766264854,0.950986177198026,0.992891063879076,0.992036613894783,0.995551389396419,0.991144178980804,0.824052962298025,0.955245671598671,0.93191144582527,0.995614527229204,0.992181328545781,0.993830745942901,0.999111221714631,0.905391125929726,0.934428763360957,0.993777343954263,0.946466459931451,0.990253911259297],[0.619861171459995,0.694354316413408,0.597174433041636,0.726724551302941,0.608169938093667,0.789846237019386,0.785373861991309,0.776362122945884,0.777150056743003,0.864655344751383,0.83208665057441,0.902511214697096,0.795561612474532,0.87930582133239,0.720768478385844,0.801567350809765,0.757337701619719,0.774125317970729,0.73315994342647,0.752178467519381,0.640472621966195,0.684742631372686,0.715302211106422,0.631426660814907,0.699584659200131,0.6615839580004,0.680784355677601,0.673767993176018,0.708300438452899,0.716271187969058,0.672767187443682,0.671421490296186,0.761314747135286,0.711161743940417],[0.0534472097064236,0.0719557627764508,0.0756346259237084,0.0719152155701753,0.0695086323166333,0.0865121962373417,0.123337080831468,0.087926061949743,0.107369016299379,0.227341607788481,0.135166668336741,0.249350200525626,0.0962225882982721,0.211314218615055,0.0419826369267609,0.121005489630157,0.105459075628253,0.0781969157171896,0.152199529443161,0.175873332244928,0.0822419833357721,0.125082851277321,0.0827463060756388,0.0550287090267594,0.10216303131945,0.0981075382249072,0.112893764896385,0.114408217210001,0.139721277108379,0.0955122931105476,0.0758109784913603,0.110261593586362,0.0835426393985126,0.128491517151136],[0.467956575322355,0.458331371853065,0.358904645431084,0.472745796864006,0.370421900833387,0.573945722186412,0.506209399503341,0.544313016597417,0.510348327122651,0.520802952690293,0.583475782013867,0.669069958072932,0.603775521199812,0.641431274574156,0.606069062867714,0.509334508724203,0.471522551496088,0.56569338812813,0.324730444572644,0.335293124634427,0.351406398382064,0.322233242592469,0.473754250053323,0.478440450078449,0.409030890515106,0.317692947290534,0.311761305601707,0.316203489419807,0.286965316741691,0.420487402389718,0.408322039696004,0.311672675394312,0.539360091926896,0.36406260481044],[0.305335502692998,0.359752007418076,0.259368884144468,0.383435295719282,0.275545765882655,0.466250798770654,0.420861055373676,0.431080589069817,0.430686589369631,0.535878145393406,0.493139266478584,0.594303789738517,0.484868716337522,0.554951720122277,0.436691720171622,0.467124819730995,0.407815278362382,0.464934328640272,0.343565266891448,0.355969136733044,0.305751474737112,0.322375416151017,0.362648069792409,0.314587851931798,0.330669670621091,0.334423131437211,0.323794497941715,0.316991663259683,0.333276481149013,0.361589178480778,0.341394374625972,0.312202978139191,0.444041919580519,0.333761764865894],[0.269299820466786,0.217235188509874,0.219928186714542,0.216337522441652,0.235188509874327,0.155296229802513,0.102333931777379,0.149012567324955,0.112208258527828,0.0754039497307002,0.102333931777379,0.0583482944344704,0.107719928186715,0.0583482944344704,0.250448833034111,0.135547576301616,0.140035906642729,0.159784560143627,0.119389587073609,0.0879712746858169,0.236086175942549,0.146319569120287,0.171454219030521,0.241472172351885,0.134649910233393,0.255834829443447,0.209156193895871,0.193895870736086,0.187612208258528,0.138240574506284,0.203770197486535,0.214542190305206,0.154398563734291,0.129263913824057],[0.422800718132855,0.323159784560144,0.382405745062837,0.300718132854578,0.430879712746858,0.228904847396768,0.175044883303411,0.240574506283662,0.206463195691203,0.128366247755835,0.152603231597846,0.085278276481149,0.206463195691203,0.0960502692998205,0.359066427289048,0.196588868940754,0.233393177737882,0.254937163375224,0.236983842010772,0.168761220825853,0.399461400359066,0.285457809694794,0.295332136445242,0.393177737881508,0.265709156193896,0.383303411131059,0.366247755834829,0.317773788150808,0.300718132854578,0.272890484739677,0.356373429084381,0.340215439856373,0.257630161579892,0.226211849192101],[300,242,245,241,262,173,114,166,125,84,114,65,120,65,279,151,156,178,133,98,263,163,191,269,150,285,233,216,209,154,227,239,172,144],[0.269299820466786,0.217235188509874,0.219928186714542,0.216337522441652,0.235188509874327,0.155296229802513,0.102333931777379,0.149012567324955,0.112208258527828,0.0754039497307002,0.102333931777379,0.0583482944344704,0.107719928186715,0.0583482944344704,0.250448833034111,0.135547576301616,0.140035906642729,0.159784560143627,0.119389587073609,0.0879712746858169,0.236086175942549,0.146319569120287,0.171454219030521,0.241472172351885,0.134649910233393,0.255834829443447,0.209156193895871,0.193895870736086,0.187612208258528,0.138240574506284,0.203770197486535,0.214542190305206,0.154398563734291,0.129263913824057],[0.116211494638178,0.0759422270498857,0.118048728601865,0.0655908641123742,0.119889185783032,0.0390508913807941,0.0316713349599838,0.0396278473097415,0.0369590232361748,0.0173860350879455,0.0219533342573223,0.0090733572066308,0.0323756079793972,0.0115987481023307,0.0804708476095008,0.0337309709298015,0.0451540536794639,0.044714084493423,0.0529864721562358,0.0430750784047652,0.107603892357429,0.0768721252929099,0.060425658100429,0.104438692791919,0.0635602370998778,0.106176007013721,0.0885788511808257,0.0856924599273487,0.0708624363011646,0.0598277512578606,0.0873556401471077,0.0880518551228207,0.0473119977824264,0.0585159017434383],[0,0.00269299820466786,0.00359066427289048,0,0.00628366247755835,0.173249551166966,0.192100538599641,0.202872531418312,0.0718132854578097,0.16786355475763,0.0628366247755835,0.0430879712746858,0.0879712746858169,0.0457809694793537,0.0287253141831239,0.272890484739677,0.0448833034111311,0.192998204667864,0.438958707360862,0.287253141831239,0.574506283662478,0.492818671454219,0,0,0.00269299820466786,0.223518850987433,0.600538599640934,0.363554757630162,0.343806104129264,0,0,0.528725314183124,0.152603231597846,0.433572710951526],[0.985580558139823,0.980602945340383,0.99185799762904,0.974948582558831,0.990830426679196,0.954647638842651,0.964828447768902,0.958982728095854,0.965538646313341,0.955521968704588,0.943191117306441,0.903996403112743,0.948819810261085,0.918795447261729,0.958221189897607,0.964052129973332,0.973733544085203,0.957729573818639,0.987249096888815,0.985673902964944,0.990618067202051,0.990593953122229,0.977532678467529,0.983849011633879,0.9846354417855,0.991544514591352,0.990951774951849,0.991052120638852,0.990297583302477,0.981322478367336,0.985505882279728,0.991091792189527,0.965136485691794,0.986878051208972],[2.06006567355224,2.0588866634065,2.05322339185869,2.05613020259501,2.05496623312259,2.0549124671045,2.05642865398474,2.05879451936693,2.05757583881007,2.0417525909029,2.05824196308129,2.05736952110065,2.06007479056845,2.05728410237043,2.05950693994291,2.0526089139523,2.0560868916665,2.05554926651677,2.04978203206812,2.05006767231133,2.05142781631264,2.04475291698497,2.05918809494236,2.05687680678857,2.05821287544001,2.03984007433414,2.05178133632348,2.04903588615044,2.04356674215334,2.05233702555193,2.05502923371167,2.04790258517659,2.05890707009378,2.05471143548178],[0.13016157989228,0.146319569120287,0.085278276481149,0.157989228007181,0.0951526032315979,0.25673249551167,0.210053859964093,0.234290843806104,0.193895870736086,0.180430879712747,0.276481149012567,0.364452423698384,0.264811490125673,0.359964093357271,0.241472172351885,0.193895870736086,0.166068222621185,0.21903052064632,0.0754039497307002,0.0870736086175943,0.0861759425493716,0.0736086175942549,0.158886894075404,0.142728904847397,0.118491921005386,0.0637342908438061,0.0718132854578097,0.0718132854578097,0.0718132854578097,0.121184919210054,0.110412926391382,0.0691202872531418,0.199281867145422,0.0978456014362657],[0.290843806104129,0.447037701974865,0.298922800718133,0.57181328545781,0.26032315978456,0.340215439856373,0.317773788150808,0.318671454219031,0.57181328545781,0.60951526032316,0.459605026929982,0.714542190305206,0.434470377019749,0.716337522441652,0.440754039497307,0.472172351885099,0.536804308797127,0.331238779174147,0.128366247755835,0.316876122082585,0.111310592459605,0.108617594254937,0.564631956912029,0.321364452423698,0.41113105924596,0.138240574506284,0.111310592459605,0.100538599640934,0.0879712746858169,0.468581687612208,0.381508078994614,0.0852782764811491,0.324955116696589,0.156193895870736]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>observed<\/th>\n      <th>chao1<\/th>\n      <th>diversity_inverse_simpson<\/th>\n      <th>diversity_gini_simpson<\/th>\n      <th>diversity_shannon<\/th>\n      <th>diversity_fisher<\/th>\n      <th>diversity_coverage<\/th>\n      <th>evenness_camargo<\/th>\n      <th>evenness_pielou<\/th>\n      <th>evenness_simpson<\/th>\n      <th>evenness_evar<\/th>\n      <th>evenness_bulla<\/th>\n      <th>dominance_dbp<\/th>\n      <th>dominance_dmn<\/th>\n      <th>dominance_absolute<\/th>\n      <th>dominance_relative<\/th>\n      <th>dominance_simpson<\/th>\n      <th>dominance_core_abundance<\/th>\n      <th>dominance_gini<\/th>\n      <th>rarity_log_modulo_skewness<\/th>\n      <th>rarity_low_abundance<\/th>\n      <th>rarity_rare_abundance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li><strong>Visualization</strong></li>
</ul>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="aplha-diversity.html#cb139-1" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy_metadata <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(amplicon_ps_rarefy) <span class="sc">%&gt;%</span></span>
<span id="cb139-2"><a href="aplha-diversity.html#cb139-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb139-3"><a href="aplha-diversity.html#cb139-3" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy_alpha_v2<span class="sc">$</span>SampleID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(amplicon_ps_rarefy_metadata)</span>
<span id="cb139-4"><a href="aplha-diversity.html#cb139-4" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy_metadata<span class="sc">$</span>SampleID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(amplicon_ps_rarefy_metadata)</span>
<span id="cb139-5"><a href="aplha-diversity.html#cb139-5" aria-hidden="true" tabindex="-1"></a>dat_diversity <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(amplicon_ps_rarefy_metadata, amplicon_ps_rarefy_alpha_v2, <span class="at">by =</span> <span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb139-6"><a href="aplha-diversity.html#cb139-6" aria-hidden="true" tabindex="-1"></a>dat_diversity_v2 <span class="ot">&lt;-</span> dat_diversity[, <span class="fu">c</span>(<span class="st">&quot;SampleType&quot;</span>, <span class="st">&quot;diversity_inverse_simpson&quot;</span>, </span>
<span id="cb139-7"><a href="aplha-diversity.html#cb139-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">&quot;diversity_gini_simpson&quot;</span>, <span class="st">&quot;diversity_shannon&quot;</span>, </span>
<span id="cb139-8"><a href="aplha-diversity.html#cb139-8" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">&quot;diversity_fisher&quot;</span>, <span class="st">&quot;diversity_coverage&quot;</span>)]</span>
<span id="cb139-9"><a href="aplha-diversity.html#cb139-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-10"><a href="aplha-diversity.html#cb139-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(dat_diversity_v2) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SampleType&quot;</span>, <span class="st">&quot;Inverse Simpson&quot;</span>, <span class="st">&quot;Gini-Simpson&quot;</span>, <span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Fisher&quot;</span>, <span class="st">&quot;Coverage&quot;</span>)</span>
<span id="cb139-11"><a href="aplha-diversity.html#cb139-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-12"><a href="aplha-diversity.html#cb139-12" aria-hidden="true" tabindex="-1"></a>plotdata <span class="ot">&lt;-</span> dat_diversity_v2 <span class="sc">%&gt;%</span> </span>
<span id="cb139-13"><a href="aplha-diversity.html#cb139-13" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">gather</span>(<span class="at">key =</span> <span class="st">&quot;Variable&quot;</span>, <span class="at">value =</span> <span class="st">&quot;Value&quot;</span>, <span class="sc">-</span>SampleType)</span>
<span id="cb139-14"><a href="aplha-diversity.html#cb139-14" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">&lt;-</span> <span class="fu">unique</span>(dat_diversity_v2<span class="sc">$</span>SampleType) </span>
<span id="cb139-15"><a href="aplha-diversity.html#cb139-15" aria-hidden="true" tabindex="-1"></a>cmp_list <span class="ot">&lt;-</span> <span class="fu">combn</span>(<span class="fu">seq_along</span>(groups), <span class="dv">2</span>, <span class="at">simplify =</span> <span class="cn">FALSE</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) {groups[x]})</span>
<span id="cb139-16"><a href="aplha-diversity.html#cb139-16" aria-hidden="true" tabindex="-1"></a>pval_sign <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb139-17"><a href="aplha-diversity.html#cb139-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">cutpoints =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.0001</span>, <span class="fl">0.001</span>, <span class="fl">0.01</span>, <span class="fl">0.05</span>, <span class="fl">0.1</span>, <span class="dv">1</span>),</span>
<span id="cb139-18"><a href="aplha-diversity.html#cb139-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">symbols =</span> <span class="fu">c</span>(<span class="st">&quot;****&quot;</span>, <span class="st">&quot;***&quot;</span>, <span class="st">&quot;**&quot;</span>, <span class="st">&quot;*&quot;</span>, <span class="st">&quot;n.s&quot;</span>)</span>
<span id="cb139-19"><a href="aplha-diversity.html#cb139-19" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb139-20"><a href="aplha-diversity.html#cb139-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb139-21"><a href="aplha-diversity.html#cb139-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(plotdata, <span class="at">x =</span> <span class="st">&quot;SampleType&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Value&quot;</span>,</span>
<span id="cb139-22"><a href="aplha-diversity.html#cb139-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">fill =</span> <span class="st">&quot;SampleType&quot;</span>, </span>
<span id="cb139-23"><a href="aplha-diversity.html#cb139-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">palette =</span> <span class="st">&quot;jco&quot;</span>, </span>
<span id="cb139-24"><a href="aplha-diversity.html#cb139-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend=</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb139-25"><a href="aplha-diversity.html#cb139-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">facet.by =</span> <span class="st">&quot;Variable&quot;</span>, </span>
<span id="cb139-26"><a href="aplha-diversity.html#cb139-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)<span class="sc">+</span></span>
<span id="cb139-27"><a href="aplha-diversity.html#cb139-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>()<span class="sc">+</span></span>
<span id="cb139-28"><a href="aplha-diversity.html#cb139-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rremove</span>(<span class="st">&quot;x.text&quot;</span>)<span class="sc">+</span></span>
<span id="cb139-29"><a href="aplha-diversity.html#cb139-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb139-30"><a href="aplha-diversity.html#cb139-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">comparisons =</span> cmp_list,</span>
<span id="cb139-31"><a href="aplha-diversity.html#cb139-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> <span class="st">&quot;p.signif&quot;</span>,</span>
<span id="cb139-32"><a href="aplha-diversity.html#cb139-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">symnum.args =</span> pval_sign</span>
<span id="cb139-33"><a href="aplha-diversity.html#cb139-33" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-70"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-70-1.png" alt="Alpha diversity from microbiome" width="768" />
<p class="caption">
Figure 7.4: Alpha diversity from microbiome
</p>
</div>
</div>
</div>
<div id="phylogenetic-diversity" class="section level3 hasAnchor" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Phylogenetic diversity<a href="aplha-diversity.html#phylogenetic-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Phylogenetic diversity is calculated using the picante package.</p>
<ul>
<li><strong>Calculation</strong></li>
</ul>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="aplha-diversity.html#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(picante)</span>
<span id="cb140-2"><a href="aplha-diversity.html#cb140-2" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy_tab <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(amplicon_ps_rarefy<span class="sc">@</span>otu_table)</span>
<span id="cb140-3"><a href="aplha-diversity.html#cb140-3" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy_tree <span class="ot">&lt;-</span> amplicon_ps_rarefy<span class="sc">@</span>phy_tree</span>
<span id="cb140-4"><a href="aplha-diversity.html#cb140-4" aria-hidden="true" tabindex="-1"></a>dat_pd <span class="ot">&lt;-</span> <span class="fu">pd</span>(<span class="fu">t</span>(amplicon_ps_rarefy_tab), </span>
<span id="cb140-5"><a href="aplha-diversity.html#cb140-5" aria-hidden="true" tabindex="-1"></a>             amplicon_ps_rarefy_tree, </span>
<span id="cb140-6"><a href="aplha-diversity.html#cb140-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">include.root =</span> <span class="cn">FALSE</span>)</span>
<span id="cb140-7"><a href="aplha-diversity.html#cb140-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-8"><a href="aplha-diversity.html#cb140-8" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(dat_pd)</span></code></pre></div>
<div id="htmlwidget-6c581cc3f76815f9c8cc" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6c581cc3f76815f9c8cc">{"x":{"filter":"none","vertical":false,"data":[["L1S140","L1S208","L1S8","L1S281","L3S242","L2S309","L2S357","L4S112","L2S155","L2S382","L4S63","L2S222","L3S341","L3S360","L2S204","L4S137","L3S294","L2S175","L5S203","L6S93","L6S20","L6S68","L1S257","L3S378","L1S105","L5S174","L5S155","L5S222","L5S104","L1S57","L1S76","L5S240","L3S313","L2S240"],[10.95687,13.15985,7.28409,13.47078,10.87712,24.68686,21.77435,26.13078,22.78599,21.97112,30.03251,34.46333,29.28208,32.05889,26.26564,20.60712,19.75832,23.56375,10.90434,12.57366,12.06673,10.49032,13.09457,12.70959,11.72443,10.04502,9.14102,10.576,11.73492,10.77197,11.04506,11.16986,22.24153,11.84624],[161,183,112,212,120,296,256,287,252,253,337,442,321,408,296,245,210,286,124,132,113,104,200,174,154,96,100,102,101,175,151,103,253,133]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>PD<\/th>\n      <th>SR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li><strong>Visualization</strong></li>
</ul>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="aplha-diversity.html#cb141-1" aria-hidden="true" tabindex="-1"></a>amplicon_ps_rarefy_metadata<span class="sc">$</span>Phylogenetic_Diversity <span class="ot">&lt;-</span> dat_pd<span class="sc">$</span>PD</span>
<span id="cb141-2"><a href="aplha-diversity.html#cb141-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-3"><a href="aplha-diversity.html#cb141-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggboxplot</span>(amplicon_ps_rarefy_metadata,</span>
<span id="cb141-4"><a href="aplha-diversity.html#cb141-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb141-5"><a href="aplha-diversity.html#cb141-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">&quot;Phylogenetic_Diversity&quot;</span>,</span>
<span id="cb141-6"><a href="aplha-diversity.html#cb141-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb141-7"><a href="aplha-diversity.html#cb141-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">palette =</span> <span class="st">&quot;jco&quot;</span>,</span>
<span id="cb141-8"><a href="aplha-diversity.html#cb141-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;Phylogenetic Diversity&quot;</span>,</span>
<span id="cb141-9"><a href="aplha-diversity.html#cb141-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Sample Type&quot;</span>,</span>
<span id="cb141-10"><a href="aplha-diversity.html#cb141-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend =</span> <span class="st">&quot;right&quot;</span>)<span class="sc">+</span> </span>
<span id="cb141-11"><a href="aplha-diversity.html#cb141-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rotate_x_text</span>()<span class="sc">+</span></span>
<span id="cb141-12"><a href="aplha-diversity.html#cb141-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(</span>
<span id="cb141-13"><a href="aplha-diversity.html#cb141-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparisons =</span> cmp_list,</span>
<span id="cb141-14"><a href="aplha-diversity.html#cb141-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">&quot;p.signif&quot;</span>,</span>
<span id="cb141-15"><a href="aplha-diversity.html#cb141-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">symnum.args =</span> pval_sign)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-72"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-72-1.png" alt="Alpha diversity from picante(Phylogenetic diversity)" width="672" />
<p class="caption">
Figure 7.5: Alpha diversity from picante(Phylogenetic diversity)
</p>
</div>
<p><strong>Cautions</strong>:</p>
<blockquote>
<p>There are arguments both for and against the use of rarefying to equal library size.
The application of normalization method will depend on the type of research question. It is always good to check if there is a correlation between increasing library sizes and richness. Observed ASVs and Phylogenetic diversity can be affected by library sizes. It is always good to check for this before making a choice.</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="aplha-diversity.html#cb142-1" aria-hidden="true" tabindex="-1"></a>lib.div <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">alpha</span>(amplicon_ps, <span class="at">index =</span> <span class="st">&quot;all&quot;</span>)</span>
<span id="cb142-2"><a href="aplha-diversity.html#cb142-2" aria-hidden="true" tabindex="-1"></a>lib.div2 <span class="ot">&lt;-</span> microbiome<span class="sc">::</span><span class="fu">richness</span>(amplicon_ps)</span>
<span id="cb142-3"><a href="aplha-diversity.html#cb142-3" aria-hidden="true" tabindex="-1"></a>lib.div<span class="sc">$</span>ReadsPerSample <span class="ot">&lt;-</span> <span class="fu">sample_sums</span>(amplicon_ps)</span>
<span id="cb142-4"><a href="aplha-diversity.html#cb142-4" aria-hidden="true" tabindex="-1"></a>lib.div<span class="sc">$</span>Observed <span class="ot">&lt;-</span> lib.div2<span class="sc">$</span>observed</span>
<span id="cb142-5"><a href="aplha-diversity.html#cb142-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-6"><a href="aplha-diversity.html#cb142-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(lib.div)</span></code></pre></div>
<pre><code>##  [1] &quot;observed&quot;                   &quot;chao1&quot;                      &quot;diversity_inverse_simpson&quot;  &quot;diversity_gini_simpson&quot;    
##  [5] &quot;diversity_shannon&quot;          &quot;diversity_fisher&quot;           &quot;diversity_coverage&quot;         &quot;evenness_camargo&quot;          
##  [9] &quot;evenness_pielou&quot;            &quot;evenness_simpson&quot;           &quot;evenness_evar&quot;              &quot;evenness_bulla&quot;            
## [13] &quot;dominance_dbp&quot;              &quot;dominance_dmn&quot;              &quot;dominance_absolute&quot;         &quot;dominance_relative&quot;        
## [17] &quot;dominance_simpson&quot;          &quot;dominance_core_abundance&quot;   &quot;dominance_gini&quot;             &quot;rarity_log_modulo_skewness&quot;
## [21] &quot;rarity_low_abundance&quot;       &quot;rarity_rare_abundance&quot;      &quot;ReadsPerSample&quot;             &quot;Observed&quot;</code></pre>
<ul>
<li>scatterplot</li>
</ul>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="aplha-diversity.html#cb144-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggscatter</span>(lib.div, <span class="st">&quot;diversity_shannon&quot;</span>, <span class="st">&quot;ReadsPerSample&quot;</span>)<span class="sc">+</span></span>
<span id="cb144-2"><a href="aplha-diversity.html#cb144-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb144-3"><a href="aplha-diversity.html#cb144-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-4"><a href="aplha-diversity.html#cb144-4" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggscatter</span>(lib.div, <span class="st">&quot;diversity_inverse_simpson&quot;</span>, <span class="st">&quot;ReadsPerSample&quot;</span>,<span class="at">dd =</span> <span class="st">&quot;loess&quot;</span>)<span class="sc">+</span></span>
<span id="cb144-5"><a href="aplha-diversity.html#cb144-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb144-6"><a href="aplha-diversity.html#cb144-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-7"><a href="aplha-diversity.html#cb144-7" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggscatter</span>(lib.div, <span class="st">&quot;Observed&quot;</span>, <span class="st">&quot;ReadsPerSample&quot;</span>, <span class="at">add =</span> <span class="st">&quot;loess&quot;</span>) <span class="sc">+</span></span>
<span id="cb144-8"><a href="aplha-diversity.html#cb144-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(</span>
<span id="cb144-9"><a href="aplha-diversity.html#cb144-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb144-10"><a href="aplha-diversity.html#cb144-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.x =</span> <span class="dv">100</span>,</span>
<span id="cb144-11"><a href="aplha-diversity.html#cb144-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.y =</span> <span class="dv">50000</span></span>
<span id="cb144-12"><a href="aplha-diversity.html#cb144-12" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb144-13"><a href="aplha-diversity.html#cb144-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-14"><a href="aplha-diversity.html#cb144-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(p1, p2, p3, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-74"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-74-1.png" alt="Correlation between increasing library sizes and richness" width="768" />
<p class="caption">
Figure 7.6: Correlation between increasing library sizes and richness
</p>
</div>
</div>
</div>
<div id="rank-abundance" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Rank Abundance<a href="aplha-diversity.html#rank-abundance" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A rank abundance curve is a chart used by ecologists to display relative species abundance, a component of biodiversity. It can also be used to visualize species richness and species evenness. It overcomes the shortcomings of biodiversity indices that cannot display the relative role different variables played in their calculation.
The curve is a 2D chart with relative abundance on the Y-axis and the abundance rank on the X-axis.</p>
<ul>
<li>X-axis: The abundance rank. The most abundant species is given rank 1, the second most abundant is 2 and so on.</li>
<li>Y-axis: The relative abundance. Usually measured on a log scale, this is a measure of a species abundance (e.g., the number of individuals) relative to the abundance of other species.</li>
</ul>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="aplha-diversity.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_RankAbundance</span>(</span>
<span id="cb146-2"><a href="aplha-diversity.html#cb146-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">ps =</span> dada2_ps_remove_BRS,</span>
<span id="cb146-3"><a href="aplha-diversity.html#cb146-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">taxa_level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb146-4"><a href="aplha-diversity.html#cb146-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb146-5"><a href="aplha-diversity.html#cb146-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-75"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-75-1.png" alt="Rank Abundance" width="480" />
<p class="caption">
Figure 7.7: Rank Abundance
</p>
</div>
<p><strong>Results</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>From the horizontal level (Family rank), the higher degree of width means the higher degree of <strong>Family richness</strong>;</p></li>
<li><p>From the vertical level (Relative abundance), the slope of the line reflects the <strong>Family evenness</strong>.</p></li>
</ol>
</div>
<div id="systematic-information-4" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Systematic Information<a href="aplha-diversity.html#systematic-information-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="aplha-diversity.html#cb147-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-06-23
##  rstudio  2022.02.0+443 Prairie Trillium (desktop)
##  pandoc   2.17.1.1 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version  date (UTC) lib source
##  abind                  1.4-5    2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18   2021-09-16 [1] CRAN (R 4.1.0)
##  annotate               1.72.0   2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2   2021-11-09 [1] Bioconductor
##  ape                  * 5.6-2    2022-03-02 [1] CRAN (R 4.1.2)
##  assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1    2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3    2015-07-28 [1] CRAN (R 4.1.0)
##  bayesm                 3.1-4    2019-10-15 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0   2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0   2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3   2021-12-09 [1] Bioconductor
##  biomformat             1.22.0   2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0   2021-10-26 [1] Bioconductor
##  bit                    4.0.4    2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5    2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7    2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2    2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.24     2021-09-02 [1] CRAN (R 4.1.0)
##  brio                   1.1.3    2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12   2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1    2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6    2021-08-19 [1] CRAN (R 4.1.0)
##  callr                  3.7.0    2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12   2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5    2022-01-06 [1] CRAN (R 4.1.2)
##  caTools                1.18.2   2021-03-28 [1] CRAN (R 4.1.0)
##  cellranger             1.1.0    2016-07-27 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0    2020-02-06 [1] CRAN (R 4.1.0)
##  cli                    3.2.0    2022-02-14 [1] CRAN (R 4.1.2)
##  cluster                2.1.2    2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18   2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2    2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3    2022-02-21 [1] CRAN (R 4.1.2)
##  compositions           2.0-4    2022-01-05 [1] CRAN (R 4.1.2)
##  conflicted           * 1.1.0    2021-11-26 [1] CRAN (R 4.1.0)
##  corrplot               0.92     2021-11-18 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1    2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0    2022-02-14 [1] CRAN (R 4.1.2)
##  crosstalk              1.2.0    2021-11-04 [1] CRAN (R 4.1.0)
##  data.table             1.14.2   2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2    2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0   2021-10-26 [1] Bioconductor
##  DEoptimR               1.0-10   2022-01-03 [1] CRAN (R 4.1.2)
##  desc                   1.4.1    2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0   2021-10-26 [1] Bioconductor
##  devtools               2.4.3    2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29   2021-12-01 [1] CRAN (R 4.1.0)
##  dplyr                * 1.0.8    2022-02-08 [1] CRAN (R 4.1.2)
##  DT                     0.21     2022-02-26 [1] CRAN (R 4.1.2)
##  edgeR                  3.36.0   2021-10-26 [1] Bioconductor
##  ellipsis               0.3.2    2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15     2022-02-18 [1] CRAN (R 4.1.2)
##  fansi                  1.0.2    2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0    2021-02-28 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0    2021-01-25 [1] CRAN (R 4.1.0)
##  foreach                1.5.2    2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82   2022-01-13 [1] CRAN (R 4.1.2)
##  forestplot             2.0.1    2021-09-03 [1] CRAN (R 4.1.0)
##  Formula                1.2-4    2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2    2021-12-08 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0   2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0   2021-10-26 [1] Bioconductor
##  generics               0.1.2    2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1   2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7    2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1   2021-11-18 [1] Bioconductor
##  ggplot2              * 3.3.5    2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr               * 0.4.0    2020-06-27 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1    2021-01-15 [1] CRAN (R 4.1.0)
##  ggsci                  2.9      2018-05-14 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3    2021-09-09 [1] CRAN (R 4.1.0)
##  glmnet                 4.1-3    2021-11-02 [1] CRAN (R 4.1.0)
##  glue                 * 1.6.2    2022-02-24 [1] CRAN (R 4.1.2)
##  Gmisc                * 3.0.0    2022-01-03 [1] CRAN (R 4.1.2)
##  gplots                 3.1.1    2020-11-28 [1] CRAN (R 4.1.0)
##  gridExtra              2.3      2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2    2021-06-06 [1] CRAN (R 4.1.0)
##  highr                  0.9      2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0    2021-10-07 [1] CRAN (R 4.1.0)
##  htmlTable            * 2.4.0    2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2    2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4    2021-09-08 [1] CRAN (R 4.1.0)
##  httr                   1.4.2    2020-07-20 [1] CRAN (R 4.1.0)
##  igraph                 1.2.11   2022-01-04 [1] CRAN (R 4.1.2)
##  IRanges              * 2.28.0   2021-10-26 [1] Bioconductor
##  iterators              1.0.14   2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9    2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4    2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0    2022-02-22 [1] CRAN (R 4.1.2)
##  kableExtra             1.3.4    2021-02-20 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0   2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20  2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.37     2021-12-16 [1] CRAN (R 4.1.0)
##  labeling               0.4.2    2020-10-20 [1] CRAN (R 4.1.0)
##  lattice              * 0.20-45  2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.1.0)
##  libcoin                1.0-9    2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1    2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1   2022-02-17 [1] Bioconductor
##  locfit                 1.5-9.5  2022-03-03 [1] CRAN (R 4.1.2)
##  lubridate              1.8.0    2021-10-07 [1] CRAN (R 4.1.0)
##  magrittr             * 2.0.2    2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55   2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix                 1.4-0    2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0    2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0   2021-09-17 [1] CRAN (R 4.1.0)
##  memoise                2.0.1    2021-11-26 [1] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0   2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39   2022-02-24 [1] CRAN (R 4.1.2)
##  microbiome             1.16.0   2021-10-26 [1] Bioconductor
##  modeltools             0.2-23   2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18   2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0   2021-10-26 [1] Bioconductor
##  munsell                0.5.0    2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3    2021-10-08 [1] CRAN (R 4.1.0)
##  nlme                 * 3.1-155  2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17   2022-01-13 [1] CRAN (R 4.1.2)
##  permute              * 0.9-7    2022-01-27 [1] CRAN (R 4.1.2)
##  pheatmap               1.0.12   2019-01-04 [1] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0   2021-10-26 [1] Bioconductor
##  picante              * 1.8.2    2020-06-10 [1] CRAN (R 4.1.0)
##  pillar                 1.7.0    2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1    2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4    2021-11-30 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7    2013-12-03 [1] CRAN (R 4.1.0)
##  prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.1.0)
##  processx               3.5.2    2021-04-30 [1] CRAN (R 4.1.0)
##  ps                     1.6.0    2021-02-28 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4    2020-04-17 [1] CRAN (R 4.1.0)
##  R6                     2.5.1    2021-08-19 [1] CRAN (R 4.1.0)
##  RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                 * 1.0.8.2  2022-03-11 [1] CRAN (R 4.1.2)
##  RCurl                  1.98-1.6 2022-02-08 [1] CRAN (R 4.1.2)
##  readxl               * 1.4.0    2022-03-28 [1] CRAN (R 4.1.2)
##  remotes                2.4.2    2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.1.0)
##  rhdf5                  2.38.1   2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0    2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0   2021-10-26 [1] Bioconductor
##  rlang                  1.0.2    2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.13     2022-03-10 [1] CRAN (R 4.1.2)
##  robustbase             0.93-9   2021-09-27 [1] CRAN (R 4.1.0)
##  rpart                  4.1.16   2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2    2020-11-15 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10   2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0    2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13     2020-11-12 [1] CRAN (R 4.1.0)
##  Rtsne                  0.15     2018-11-10 [1] CRAN (R 4.1.0)
##  rvest                  1.0.2    2021-10-16 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3   2021-11-21 [1] Bioconductor
##  sandwich               3.0-1    2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0    2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1    2020-05-11 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2    2021-12-06 [1] CRAN (R 4.1.0)
##  shape                  1.4.6    2021-05-19 [1] CRAN (R 4.1.0)
##  stringi                1.7.6    2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0    2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0   2021-10-26 [1] Bioconductor
##  survival               3.3-1    2022-03-03 [1] CRAN (R 4.1.2)
##  svglite                2.1.0    2022-02-03 [1] CRAN (R 4.1.2)
##  systemfonts            1.0.4    2022-02-11 [1] CRAN (R 4.1.2)
##  tensorA                0.36.2   2020-11-19 [1] CRAN (R 4.1.0)
##  testthat               3.1.2    2022-01-20 [1] CRAN (R 4.1.2)
##  TH.data                1.1-0    2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6    2021-11-07 [1] CRAN (R 4.1.0)
##  tidyr                  1.2.0    2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2    2022-02-21 [1] CRAN (R 4.1.2)
##  usethis                2.1.5    2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2    2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8    2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                * 2.5-7    2020-11-28 [1] CRAN (R 4.1.0)
##  viridisLite            0.4.0    2021-04-13 [1] CRAN (R 4.1.0)
##  webshot                0.5.3    2022-04-14 [1] CRAN (R 4.1.2)
##  withr                  2.5.0    2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0   2021-10-26 [1] Bioconductor
##  xfun                   0.30     2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                * 2.1.4    2022-06-22 [1] local
##  XML                    3.99-0.9 2022-02-24 [1] CRAN (R 4.1.2)
##  xml2                   1.3.3    2021-11-30 [1] CRAN (R 4.1.0)
##  xtable                 1.8-4    2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0   2021-10-26 [1] Bioconductor
##  yaml                   2.3.5    2022-02-21 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0   2021-10-26 [1] Bioconductor
##  zoo                    1.8-9    2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Sudarshan2020Microbial" class="csl-entry">
A, Shetty Sudarshan, Lahti Leo, Hermes Gerben DA, and Hauke Smidt. 2020. <em>Microbial Bioinformatics Introductory Course Material 2018 (Version V3.0)</em>. Vol. 100. Zenodo.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="pre-processing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="beta-diversity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/06-alpha-diversity.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
