<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Beta diversity | Data analysis based on XMAS 2.0 package in microbiota data</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Beta diversity | Data analysis based on XMAS 2.0 package in microbiota data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Beta diversity | Data analysis based on XMAS 2.0 package in microbiota data" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-06-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="aplha-diversity.html"/>
<link rel="next" href="composition.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation<span></span></a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette<span></span></a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors<span></span></a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2<span></span></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>4.2.1</b> Importing results from dada2 pipeline<span></span></a></li>
<li class="chapter" data-level="4.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>4.2.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>4.2.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.2.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>4.2.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.2.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>4.2.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>4.3.1</b> Importing results<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table-1"><i class="fa fa-check"></i><b>4.3.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table-1"><i class="fa fa-check"></i><b>4.3.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.3.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table-1"><i class="fa fa-check"></i><b>4.3.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.3.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object-1"><i class="fa fa-check"></i><b>4.3.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Readsâ€™ track<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>5.6</b> Variability<span></span></a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>5.6.1</b> Coefficient of variation<span></span></a></li>
<li class="chapter" data-level="5.6.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>5.6.2</b> Distribution of number of sequences<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.7</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>6.8</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)<span></span></a></li>
<li class="chapter" data-level="6.9" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.9</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities<span></span></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>7.2.1</b> Non-phylogenetic diversities<span></span></a></li>
<li class="chapter" data-level="7.2.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2.2</b> Phylogenetic diversity<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics<span></span></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>8.6</b> Mantel Test<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.7</b> Analysis of Similarity (ANOSIM)<span></span></a></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>8.8</b> Multi-response permutation procedures (MRPP)<span></span></a></li>
<li class="chapter" data-level="8.9" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.9</b> Ordination<span></span></a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.9.1</b> Principal Component Analysis (PCA)<span></span></a></li>
<li class="chapter" data-level="8.9.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.9.2</b> Principal Coordinate Analysis (PCoA)<span></span></a></li>
<li class="chapter" data-level="8.9.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.9.3</b> t-distributed stochastic neighbor embedding (t-SNE)<span></span></a></li>
<li class="chapter" data-level="8.9.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.9.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.10</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="composition.html"><a href="composition.html"><i class="fa fa-check"></i><b>9</b> Composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="composition.html"><a href="composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="composition.html"><a href="composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.2</b> XVIZ <code>plot_stacked_bar</code><span></span></a></li>
<li class="chapter" data-level="9.3" data-path="composition.html"><a href="composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.3</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="composition.html"><a href="composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="composition.html"><a href="composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus<span></span></a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq<span></span></a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR<span></span></a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM<span></span></a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob<span></span></a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)<span></span></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom-1"><i class="fa fa-check"></i><b>11.2.1</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.2</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.3</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.4</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.2.5" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.5</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot<span></span></a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot<span></span></a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot<span></span></a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density<span></span></a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination<span></span></a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot<span></span></a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot<span></span></a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn<span></span></a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot"><i class="fa fa-check"></i><b>12.10</b> plot the DA results from the significant taxa by double barplot<span></span></a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.11</b> plot_stacked_bar_XIVZ<span></span></a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.12</b> plot_StackBarPlot<span></span></a></li>
<li class="chapter" data-level="12.13" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.13</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>13</b> Principals of Differential Analysis methods<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>13.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>13.2</b> ANCOM-II<span></span></a></li>
<li class="chapter" data-level="13.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>13.3</b> corncob<span></span></a></li>
<li class="chapter" data-level="13.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>13.4</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="13.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>13.5</b> edgeR<span></span></a></li>
<li class="chapter" data-level="13.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>13.6</b> LEfSe<span></span></a></li>
<li class="chapter" data-level="13.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>13.7</b> limma voom<span></span></a></li>
<li class="chapter" data-level="13.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>13.8</b> MaAsLin2<span></span></a></li>
<li class="chapter" data-level="13.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>13.9</b> metagenomeSeq<span></span></a></li>
<li class="chapter" data-level="13.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>13.10</b> t-test<span></span></a></li>
<li class="chapter" data-level="13.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>13.11</b> Wilcoxon test<span></span></a></li>
<li class="chapter" data-level="13.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>13.12</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="13.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>13.13</b> Summary<span></span></a></li>
<li class="chapter" data-level="13.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>13.14</b> Advantages and disadvantages<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>14</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>14.1</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="14.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>14.2</b> Amplicon sequencing (16s)<span></span></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>14.2.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="14.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>14.2.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="14.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>14.2.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="14.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>14.2.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="14.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>14.2.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="14.2.6" data-path="examples.html"><a href="examples.html#microbial-composition"><i class="fa fa-check"></i><b>14.2.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="14.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>14.2.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="examples.html"><a href="examples.html#metagenomics-mgs"><i class="fa fa-check"></i><b>14.3</b> Metagenomics (MGS)<span></span></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>14.3.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="14.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>14.3.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="14.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>14.3.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="14.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>14.3.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="14.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>14.3.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="14.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>14.3.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="14.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>14.3.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>14.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data analysis based on XMAS 2.0 package in microbiota data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="beta-diversity" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Beta diversity<a href="beta-diversity.html#beta-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Beta-diversity: Measures for differences between samples from different groups to identify if there are differences in the overall community composition and structure.</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="beta-diversity.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb149-2"><a href="beta-diversity.html#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb149-3"><a href="beta-diversity.html#cb149-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb149-4"><a href="beta-diversity.html#cb149-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb149-5"><a href="beta-diversity.html#cb149-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb149-6"><a href="beta-diversity.html#cb149-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb149-7"><a href="beta-diversity.html#cb149-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
<div id="loading-data-and-removing-brs" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Loading data and Removing BRS<a href="beta-diversity.html#loading-data-and-removing-brs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>Using <code>get_GroupPhyloseq</code> to remove some samples. For instance, removing the spike-in sample QC in Group column.</p></li>
<li><p>Rarefying counts</p></li>
</ul>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="beta-diversity.html#cb150-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb150-2"><a href="beta-diversity.html#cb150-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb150-3"><a href="beta-diversity.html#cb150-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb150-4"><a href="beta-diversity.html#cb150-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb150-5"><a href="beta-diversity.html#cb150-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb150-6"><a href="beta-diversity.html#cb150-6" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(dada2_ps_remove_BRS, <span class="at">size =</span> <span class="dv">51181</span>, <span class="at">rng_seed =</span> <span class="dv">123</span>)</span>
<span id="cb150-7"><a href="beta-diversity.html#cb150-7" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 891 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 891 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 891 tips and 888 internal nodes ]
## refseq()      DNAStringSet:      [ 891 reference sequences ]</code></pre>
</div>
<div id="phylogenetic-beta-diversity-metrics" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Phylogenetic beta-diversity metrics<a href="beta-diversity.html#phylogenetic-beta-diversity-metrics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="unweighted-unifrac" class="section level3 hasAnchor" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Unweighted Unifrac<a href="beta-diversity.html#unweighted-unifrac" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Unweighted Unifrac is based on presence/absence of different taxa and abundance is not important.</p>
<blockquote>
<p>The UniFrac distance, also known as unweighted UniFrac distance, was introduced by Lozupone et al.Â The goal of the UniFrac distance metric was to enable objective comparison between microbiome samples from different conditions.</p>
</blockquote>
<p><span class="math display">\[d^{u} = \sum^{n}_{i=1}\frac{b_{i}|I(p_{i}^{A} &gt; 0) - I(p_{i}^{B} &gt; 0)|}{\sum_{i=1}^{n}b_{i}}\]</span></p>
<p>where,</p>
<ul>
<li><p><span class="math inline">\(d^{u}\)</span> = unweighted UniFrac distance;</p></li>
<li><p><span class="math inline">\(A\)</span>, <span class="math inline">\(B\)</span> = microbiome community A and B, respectively;</p></li>
<li><p><span class="math inline">\(n\)</span> = rooted phylogenetic treeâ€™s branches;</p></li>
<li><p><span class="math inline">\(b_{i}\)</span> =length of the branch <span class="math inline">\(i\)</span>;</p></li>
<li><p><span class="math inline">\(p_{i}^{A}\)</span> and <span class="math inline">\(p_{i}^{A}\)</span> = taxa proportions descending from the branch <span class="math inline">\(i\)</span> for community A and B, respectively.</p></li>
<li><p>Calculation</p></li>
</ul>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="beta-diversity.html#cb152-1" aria-hidden="true" tabindex="-1"></a>dada2_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>dada2_ps_rarefy, <span class="at">method =</span> <span class="st">&quot;unifrac&quot;</span>)</span></code></pre></div>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="beta-diversity.html#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_distance_corrplot</span>(dada2_beta<span class="sc">$</span>BetaDistance)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-80"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-80-1.png" alt="Unweighted Unifrac Distance" width="576" />
<p class="caption">
Figure 8.1: Unweighted Unifrac Distance
</p>
</div>
<p>Try repeating the above ordination using filtered phyloseq object (<strong>discarding singletons/OTUs with very low reads</strong>).</p>
</div>
<div id="weighted-unifrac" class="section level3 hasAnchor" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> Weighted Unifrac<a href="beta-diversity.html#weighted-unifrac" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Weighted Unifrac will consider the abundances of different taxa.</p>
<blockquote>
<p>In 2007, Lozupone et al.Â added a proportional weighting to the original unweighted method, hence called this new UniFrac measure as weighted UniFrac.</p>
</blockquote>
<p>The weighted UniFrac distance is defined as:</p>
<p><span class="math display">\[d^{W} = \frac{\sum_{i=1}^{n}b_{i}|p_{i}^{A}-p_{i}^{B}|}{\sum_{i=1}^{n}b_{i}(p_{i}^{A}+p_{i}^{B})}\]</span></p>
<p>where,</p>
<ul>
<li><p><span class="math inline">\(d^{W}\)</span> = (normalized) weighted UniFrac distance;</p></li>
<li><p><span class="math inline">\(A\)</span>, <span class="math inline">\(B\)</span> = microbiome community A and B, respectively;</p></li>
<li><p><span class="math inline">\(n\)</span> = rooted phylogenetic treeâ€™s branches;</p></li>
<li><p><span class="math inline">\(b_{i}\)</span> = length of the branch <span class="math inline">\(i\)</span>.</p></li>
</ul>
<blockquote>
<p>By adding a proportional weighting to UniFrac distance, weighted UniFrac distance reduces the problem of low abundance taxa being represented as a 0 or by a low count depending on sampling depth.</p>
</blockquote>
<ul>
<li><strong>Calculation</strong></li>
</ul>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="beta-diversity.html#cb154-1" aria-hidden="true" tabindex="-1"></a>dada2_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>dada2_ps_rarefy, <span class="at">method =</span> <span class="st">&quot;wunifrac&quot;</span>)</span></code></pre></div>
<p>The result of dispersion test (<code>Pr(&gt;F) &gt; 0.05</code>) showed that the <strong>homogeneity condition</strong> of two groups were not significant.</p>
<ul>
<li><strong>Visualization</strong></li>
</ul>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="beta-diversity.html#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_distance_corrplot</span>(dada2_beta<span class="sc">$</span>BetaDistance)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-82"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-82-1.png" alt="Weighted Unifrac Distance" width="576" />
<p class="caption">
Figure 8.2: Weighted Unifrac Distance
</p>
</div>
</div>
<div id="generalized-unifrac-distance-metrics" class="section level3 hasAnchor" number="8.2.3">
<h3><span class="header-section-number">8.2.3</span> Generalized UniFrac Distance Metrics<a href="beta-diversity.html#generalized-unifrac-distance-metrics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>However, either unweighted or weighted UniFrac distances may not be very powerful in detecting change in moderately abundant lineages <span class="citation">(<a href="#ref-chen2012associating" role="doc-biblioref">J. Chen et al. 2012</a>)</span> because they assign too much weight either to rare lineages or to most abundant lineages. Thus, Chen et al.Â proposed the following generalized UniFrac distances to unify weighted UniFrac and unweighted UniFrac distances.</p>
<p>The Generalized UniFrac Distance Metrics is defined as:</p>
<p><span class="math display">\[d^{\alpha} = \frac{\sum_{i=1}^{n}b_{i}(p_{i}^{A}+p_{i}^{B})^{\alpha}|\frac{p_{i}^{A}-p_{i}^{B}}{p_{i}^{A}+p_{i}^{B}}|}{\sum_{i=1}^{n}b_{i}(p_{i}^{A}+p_{i}^{B})}\]</span></p>
<p>where,</p>
<ul>
<li><p><span class="math inline">\(d^{\alpha}\)</span> = generalized UniFrac distances;</p></li>
<li><p><span class="math inline">\(\alpha\in [0, 1]\)</span> is used to controls the contribution from high-abundance branches;</p></li>
<li><p><strong>Calculation</strong></p></li>
</ul>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="beta-diversity.html#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="co"># approach1</span></span>
<span id="cb156-2"><a href="beta-diversity.html#cb156-2" aria-hidden="true" tabindex="-1"></a>dada2_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>dada2_ps_rarefy, <span class="at">method =</span> <span class="st">&quot;GUniFrac&quot;</span>, <span class="at">GUniFrac_alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb156-3"><a href="beta-diversity.html#cb156-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-4"><a href="beta-diversity.html#cb156-4" aria-hidden="true" tabindex="-1"></a><span class="co"># approach2</span></span>
<span id="cb156-5"><a href="beta-diversity.html#cb156-5" aria-hidden="true" tabindex="-1"></a>dada2_otu_tab <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_rarefy)</span>
<span id="cb156-6"><a href="beta-diversity.html#cb156-6" aria-hidden="true" tabindex="-1"></a>dada2_tree <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">phy_tree</span>(dada2_ps_rarefy)</span>
<span id="cb156-7"><a href="beta-diversity.html#cb156-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-8"><a href="beta-diversity.html#cb156-8" aria-hidden="true" tabindex="-1"></a>dada2_unifracs <span class="ot">&lt;-</span> <span class="fu">run_GUniFrac</span>(dada2_otu_tab, dada2_tree, <span class="at">alpha=</span><span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>, <span class="dv">1</span>))<span class="sc">$</span>unifracs</span>
<span id="cb156-9"><a href="beta-diversity.html#cb156-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-10"><a href="beta-diversity.html#cb156-10" aria-hidden="true" tabindex="-1"></a>dada2_dw <span class="ot">&lt;-</span> dada2_unifracs[, , <span class="st">&quot;d_1&quot;</span>]    <span class="co"># Weighted UniFrac</span></span>
<span id="cb156-11"><a href="beta-diversity.html#cb156-11" aria-hidden="true" tabindex="-1"></a>dada2_du <span class="ot">&lt;-</span> dada2_unifracs[, , <span class="st">&quot;d_UW&quot;</span>]   <span class="co"># Unweighted UniFrac</span></span>
<span id="cb156-12"><a href="beta-diversity.html#cb156-12" aria-hidden="true" tabindex="-1"></a>dada2_dv <span class="ot">&lt;-</span> dada2_unifracs[, , <span class="st">&quot;d_VAW&quot;</span>]  <span class="co"># Variance adjusted weighted UniFrac</span></span>
<span id="cb156-13"><a href="beta-diversity.html#cb156-13" aria-hidden="true" tabindex="-1"></a>dada2_d0 <span class="ot">&lt;-</span> dada2_unifracs[, , <span class="st">&quot;d_0&quot;</span>]    <span class="co"># GUniFrac with alpha 0</span></span>
<span id="cb156-14"><a href="beta-diversity.html#cb156-14" aria-hidden="true" tabindex="-1"></a>dada2_d5 <span class="ot">&lt;-</span> dada2_unifracs[, , <span class="st">&quot;d_0.5&quot;</span>]  <span class="co"># GUniFrac with alpha 0.5</span></span></code></pre></div>
<ul>
<li><strong>Visualization</strong>: GUniFrac with alpha 0.5</li>
</ul>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="beta-diversity.html#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_distance_corrplot</span>(dada2_beta<span class="sc">$</span>BetaDistance)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-84"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-84-1.png" alt="Generalized UniFrac Distance (alpha=0.5)" width="576" />
<p class="caption">
Figure 8.3: Generalized UniFrac Distance (alpha=0.5)
</p>
</div>
</div>
</div>
<div id="distance-dissimilarity-coefficients-bray-curtis-index" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For microbiome abundance data, the measures of distance coefficients are not really distances. They actually measure â€œdissimilarityâ€.</p>
<p>Measures of dissimilarity include Euclidian distance, Manhattan, and Bray-Curtis measures. Here, we take Bray-Curtis distance as an example.</p>
<p>As defined by Bray and Curtis, the index of dissimilarity is given as follow:</p>
<p><span class="math display">\[BC = \frac{\sum_{i=1}^{n}|X_{ij}-X_{ik}|}{\sum_{i=1}^{n}|X_{ij}+X_{ik}|}\]</span>
where,</p>
<ul>
<li><p><span class="math inline">\(BC\)</span> Bray-Curtis measure of dissimilarity;</p></li>
<li><p><span class="math inline">\(X_{ij}\)</span>, <span class="math inline">\(X_{ik}\)</span> Number of individuals in species;</p></li>
<li><p><span class="math inline">\(i\)</span> in each <span class="math inline">\(sample (j, k)\)</span>;</p></li>
<li><p><span class="math inline">\(n\)</span> Total number of species in samples.</p></li>
</ul>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="beta-diversity.html#cb158-1" aria-hidden="true" tabindex="-1"></a>dada2_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>dada2_ps_rarefy, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb158-2"><a href="beta-diversity.html#cb158-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_distance_corrplot</span>(dada2_beta<span class="sc">$</span>BetaDistance)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-85"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-85-1.png" alt="Bray Curtis Distance" width="576" />
<p class="caption">
Figure 8.4: Bray Curtis Distance
</p>
</div>
</div>
<div id="checking-the-homogeneity-condition" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Checking the homogeneity condition<a href="beta-diversity.html#checking-the-homogeneity-condition" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="beta-diversity.html#cb159-1" aria-hidden="true" tabindex="-1"></a>dada2_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>dada2_ps_rarefy, <span class="at">method =</span> <span class="st">&quot;jsd&quot;</span>, <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
## Groups     1 0.000486 0.0004856 0.1207    999  0.725
## Residuals 21 0.084520 0.0040247                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         AA    BB
## AA         0.735
## BB 0.73178</code></pre>
<p>The result of dispersion test (<code>Pr(&gt;F) &lt; 0.05</code>) showed that the <strong>homogeneity condition</strong> of two groups were significant.</p>
</div>
<div id="permutational-multivariate-analysis-of-variance-permanova" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">8.5</span> Permutational Multivariate Analysis of Variance (PERMANOVA)<a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Permutational Multivariate Analysis of Variance (PERMANOVA)</strong> test <span class="citation">(<a href="#ref-anderson2014permutational" role="doc-biblioref">Anderson 2014</a>)</span> is to investigate the associations between the environmental factors including discrete or continuous variables (treatments or populations representatives, age, gender etc) and the whole microbial community.</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="beta-diversity.html#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_permanova</span>(dada2_ps_rarefy, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">columns =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>##       SumsOfSample Df SumsOfSqs   MeanSqs  F.Model         R2 Pr(&gt;F) AdjustedPvalue
## Group           23  1 0.3126266 0.3126266 1.004313 0.04564163  0.444          0.444</code></pre>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="beta-diversity.html#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_permanova</span>(amplicon_ps, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span></code></pre></div>
<pre><code>##                          SumsOfSample Df  SumsOfSqs   MeanSqs   F.Model         R2 Pr(&gt;F) AdjustedPvalue
## TempRowNames                       34 33 12.4906568 0.3785048 0.0000000 1.00000000  1.000      1.0000000
## SampleType                         34  3  5.5551059 1.8517020 8.0096101 0.44474090  0.001      0.0090000
## Year                               34  1  0.5879699 0.5879699 1.5807386 0.04707278  0.136      0.3465000
## Month                              34  4  1.3032780 0.3258195 0.8445915 0.10434023  0.731      0.9398571
## Day                                34  3  1.0649549 0.3549850 0.9320696 0.08526012  0.556      0.9398571
## Subject                            34  1  0.5598389 0.5598389 1.5015606 0.04482062  0.154      0.3465000
## ReportedAntibioticUsage            34  1  0.5879699 0.5879699 1.5807386 0.04707278  0.136      0.3465000
## DaysSinceExperimentStart           34  4  1.3032780 0.3258195 0.8445915 0.10434023  0.731      0.9398571
## Description                        34 33 12.4906568 0.3785048 0.0000000 1.00000000  1.000      1.0000000</code></pre>
<p><strong>Results:</strong></p>
<ol style="list-style-type: decimal">
<li><p>The PERMANOVA result of the <em>SampleType</em> (AdjustedPvalue &lt; 0.05) revealed that different groups of SampleType had the distinct patterns of microbial community. From the previous metadata, <em>SampleType</em> had four groups (<em>gut</em>, <em>right palm</em>, <em>left palm</em>, <em>tongue</em>) which are different body sites.</p></li>
<li><p>The other continuous variables such as Year, Day and Month didnâ€™t show any significant association with microbial community.</p></li>
<li><p>We suggest you performing the PERMANOVA test before you do correlation analysis between individual taxa and environmental factors. If the whole microbial community had related to one environmental factor, we could find more associations between individual taxa and environmental factors.</p></li>
<li><p>Ordination analysis is usually utilized for dimensionality reduction and then we decipher their results by using scatterplot. In fact, we should combine the statistical results, for example PERMANOVA, ANOSIM or others and dimension reduction results to provide stronger evidences to display the association and difference.</p></li>
</ol>
</div>
<div id="mantel-test" class="section level2 hasAnchor" number="8.6">
<h2><span class="header-section-number">8.6</span> Mantel Test<a href="beta-diversity.html#mantel-test" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p><strong>Mantel tests</strong> <span class="citation">(<a href="#ref-mantel1967detection" role="doc-biblioref">Mantel 1967</a>)</span> are correlation tests that determine the correlation between two matrices (rather than two variables). When using the test for microbial ecology, the matrices are often distance/dissimilarity matrices with corresponding positions (i.e.Â samples in the same order in both matrices).</p>
</blockquote>
<blockquote>
<p>A significant Mantel test will tell you that the distances between samples in one matrix are correlated with the distances between samples in the other matrix. Therefore, as the distance between samples increases with respect to one matrix, the distances between the same samples also increases in the other matrix.</p>
</blockquote>
<p>There are also three distance matrix for mantel test:</p>
<ul>
<li><strong>Species abundance dissimilarity matrix</strong>: created using a distance measure, i.e.Â Bray-curtis dissimilarity. This is the same type of dissimilarity matrix used when conducting an ANOSIM test or when making an NMDS plot;</li>
<li><strong>Environmental factors distance matrix</strong>: generally created using Euclidean Distance;</li>
<li><strong>Geographic distance matrix</strong>: the physical distance between sites for z_distance (i.e.Â Haversine distance).</li>
</ul>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="beta-diversity.html#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;amplicon_ps&quot;</span>)</span>
<span id="cb165-2"><a href="beta-diversity.html#cb165-2" aria-hidden="true" tabindex="-1"></a><span class="fu">run_MantelTest</span>(<span class="at">ps =</span> amplicon_ps,</span>
<span id="cb165-3"><a href="beta-diversity.html#cb165-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y_variables =</span> <span class="fu">c</span>(<span class="st">&quot;SampleType&quot;</span>, <span class="st">&quot;Subject&quot;</span>, <span class="st">&quot;ReportedAntibioticUsage&quot;</span>, <span class="st">&quot;DaysSinceExperimentStart&quot;</span>, <span class="st">&quot;Description&quot;</span>),</span>
<span id="cb165-4"><a href="beta-diversity.html#cb165-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">z_variables =</span> <span class="fu">c</span>(<span class="st">&quot;Year&quot;</span>, <span class="st">&quot;Month&quot;</span>, <span class="st">&quot;Day&quot;</span>),</span>
<span id="cb165-5"><a href="beta-diversity.html#cb165-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_mantel =</span> <span class="st">&quot;mantel.partial&quot;</span>,</span>
<span id="cb165-6"><a href="beta-diversity.html#cb165-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_cor =</span> <span class="st">&quot;spearman&quot;</span>,</span>
<span id="cb165-7"><a href="beta-diversity.html#cb165-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_dist =</span> <span class="fu">c</span>(<span class="st">&quot;bray&quot;</span>, <span class="st">&quot;euclidean&quot;</span>, <span class="st">&quot;jaccard&quot;</span>)</span>
<span id="cb165-8"><a href="beta-diversity.html#cb165-8" aria-hidden="true" tabindex="-1"></a>          )</span></code></pre></div>
<pre><code>## 
## Partial Mantel statistic based on Spearman&#39;s rank correlation rho 
## 
## Call:
## vegan::mantel.partial(xdis = x_dis, ydis = y_dis, zdis = z_dis,      method = method_cor, permutations = 999) 
## 
## Mantel statistic r: 0.3318 
##       Significance: 0.001 
## 
## Upper quantiles of permutations (null model):
##    90%    95%  97.5%    99% 
## 0.0524 0.0822 0.1095 0.1362 
## Permutation: free
## Number of permutations: 999</code></pre>
<p><strong>Results</strong>:</p>
<p>From the results, we could see that the <strong>y_dis (environmental factors) has a strong relationship with the OTU Bray-Curtis dissimiliarity matrix</strong> (Mantel statistic R: 0.3318, p value = 0.001) after adjusting the z_dis matrix. In other words, as samples become more dissimilar in terms of environmental factors, they also become more dissimilar in terms of microbial community composition.</p>
<details>
<summary>
<strong>Options for Mantel test</strong>
</summary>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="beta-diversity.html#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_MantelTest</span>(<span class="at">ps =</span> amplicon_ps,</span>
<span id="cb167-2"><a href="beta-diversity.html#cb167-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">y_variables =</span> <span class="fu">c</span>(<span class="st">&quot;SampleType&quot;</span>, <span class="st">&quot;Subject&quot;</span>, <span class="st">&quot;ReportedAntibioticUsage&quot;</span>, <span class="st">&quot;DaysSinceExperimentStart&quot;</span>, <span class="st">&quot;Description&quot;</span>),</span>
<span id="cb167-3"><a href="beta-diversity.html#cb167-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_mantel =</span> <span class="st">&quot;mantel&quot;</span>,</span>
<span id="cb167-4"><a href="beta-diversity.html#cb167-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_cor =</span> <span class="st">&quot;spearman&quot;</span>,</span>
<span id="cb167-5"><a href="beta-diversity.html#cb167-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_dist =</span> <span class="fu">c</span>(<span class="st">&quot;bray&quot;</span>, <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb167-6"><a href="beta-diversity.html#cb167-6" aria-hidden="true" tabindex="-1"></a>          )</span></code></pre></div>
<pre><code>## 
## Mantel statistic based on Spearman&#39;s rank correlation rho 
## 
## Call:
## vegan::mantel(xdis = x_dis, ydis = y_dis, method = method_cor,      permutations = 999) 
## 
## Mantel statistic r: 0.2935 
##       Significance: 0.001 
## 
## Upper quantiles of permutations (null model):
##    90%    95%  97.5%    99% 
## 0.0693 0.0992 0.1306 0.1592 
## Permutation: free
## Number of permutations: 999</code></pre>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="beta-diversity.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_MantelTest</span>(<span class="at">ps =</span> amplicon_ps,</span>
<span id="cb169-2"><a href="beta-diversity.html#cb169-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">y_variables =</span> <span class="fu">c</span>(<span class="st">&quot;SampleType&quot;</span>, <span class="st">&quot;Subject&quot;</span>, <span class="st">&quot;ReportedAntibioticUsage&quot;</span>, <span class="st">&quot;DaysSinceExperimentStart&quot;</span>, <span class="st">&quot;Description&quot;</span>),</span>
<span id="cb169-3"><a href="beta-diversity.html#cb169-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_mantel =</span> <span class="st">&quot;mantel.randtest&quot;</span>,</span>
<span id="cb169-4"><a href="beta-diversity.html#cb169-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_cor =</span> <span class="st">&quot;spearman&quot;</span>,</span>
<span id="cb169-5"><a href="beta-diversity.html#cb169-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">method_dist =</span> <span class="fu">c</span>(<span class="st">&quot;bray&quot;</span>, <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb169-6"><a href="beta-diversity.html#cb169-6" aria-hidden="true" tabindex="-1"></a>          )</span></code></pre></div>
<pre><code>## Monte-Carlo test
## Call: ade4::mantel.randtest(m1 = x_dis, m2 = y_dis, nrepet = 999)
## 
## Observation: 0.2843383 
## 
## Based on 999 replicates
## Simulated p-value: 0.001 
## Alternative hypothesis: greater 
## 
##     Std.Obs Expectation    Variance 
## 6.223278099 0.002464050 0.002051504</code></pre>
</details>
</div>
<div id="analysis-of-similarity-anosim" class="section level2 hasAnchor" number="8.7">
<h2><span class="header-section-number">8.7</span> Analysis of Similarity (ANOSIM)<a href="beta-diversity.html#analysis-of-similarity-anosim" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Analysis of Similarity (ANOSIM)</strong> is simply a modified version of the Mantel Test based on a standardized <em>rank</em> correlation between two distance matrices.</p>
<blockquote>
<p>It is a nonparametric procedure for testing the hypothesis of no difference between two or more groups of samples based on permutation test of among-and within-group similarities.</p>
</blockquote>
<p>The <em>ANOSIM test statistic(R)</em> is based on the difference of mean ranks between groups and within groups. It is given below:</p>
<p><span class="math display">\[R = \frac{\bar{r}_{B} - \bar{r}_{W}}{M/2}\]</span>
where,</p>
<ul>
<li><p><span class="math inline">\(R\)</span> test statistic, is an index of relative within-group dissimilarity;</p></li>
<li><p><span class="math inline">\(M = N(N âˆ’ 1)/2\)</span> number of sample pairs;</p></li>
<li><p><span class="math inline">\(N\)</span> is the total number of samples (subjects);</p></li>
<li><p><span class="math inline">\(r_{B}\)</span> is the mean of the ranked similarity between groups;</p></li>
<li><p><span class="math inline">\(r_{W}\)</span> is the mean of the ranked similarity within groups.</p></li>
</ul>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="beta-diversity.html#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_ANOSIM</span>(<span class="at">ps =</span> dada2_ps_rarefy, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Call:
## vegan::anosim(x = dis, grouping = datphe, permutations = 999) 
## Dissimilarity: bray 
## 
## ANOSIM statistic R: -0.03337 
##       Significance: 0.63 
## 
## Permutation: free
## Number of permutations: 999</code></pre>
<p>The p-value of 0.62 is more than 0.05, which indicates that within-group similarity is not greater or less than between-group similarity at 0.05 significant level. We can conclude that there is no strong evidence that the within-group samples are more different than would be expected by random chance.</p>
</div>
<div id="multi-response-permutation-procedures-mrpp" class="section level2 hasAnchor" number="8.8">
<h2><span class="header-section-number">8.8</span> Multi-response permutation procedures (MRPP)<a href="beta-diversity.html#multi-response-permutation-procedures-mrpp" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Multi-response permutation procedures (MRPP)</strong> <span class="citation">(<a href="#ref-mielke1991application" role="doc-biblioref">Mielke Jr 1991</a>)</span> is a nonparametric procedure for testing the hypothesis of no difference between two or more groups of samples based on permutation test of among-and within-group dissimilarities.</p>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="beta-diversity.html#cb173-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_MRPP</span>(<span class="at">ps =</span> dada2_ps_rarefy, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Call:
## vegan::mrpp(dat = dis, grouping = datphe, permutations = 999) 
## 
## Dissimilarity index: bray 
## Weights for groups:  n 
## 
## Class means and counts:
## 
##       AA     BB   
## delta 0.7736 0.785
## n     9      14   
## 
## Chance corrected within-group agreement A: 0.0006646 
## Based on observed delta 0.7805 and expected delta 0.781 
## 
## Significance of delta: 0.428 
## Permutation: free
## Number of permutations: 999</code></pre>
<p>The observed and expected deltas are 0.6469 and 0.851, respectively. The significance of delta is 0.001 with the chance corrected within-group agreement A of 0.2399. <em>We conclude that there is statistically significant difference of the four SampleType at the 0.05 of significance level</em>.</p>
</div>
<div id="ordination" class="section level2 hasAnchor" number="8.9">
<h2><span class="header-section-number">8.9</span> Ordination<a href="beta-diversity.html#ordination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Ordination is one of the main multivariate methods to reduce the dimensions of taxaâ€“for instance, the whole microbial community often contains too much taxa which makes the microbial profile have too large dimensions. Utilizing ordination to convert the data into two or three dimensions could have more interpretable visualization. Compared to clustering methods, ordination focuses on the dimension reduction and has explanation loss (variation).</p>
<p>We also use a distance method to calculate the distance matrix among samples and then do ordination analysis. Here, we give five universal ordination methods. There is no <em>one-fit-all</em> method for all microbiota data, so please pay attention to your own ordination analysis.</p>
<div id="principal-component-analysis-pca" class="section level3 hasAnchor" number="8.9.1">
<h3><span class="header-section-number">8.9.1</span> Principal Component Analysis (PCA)<a href="beta-diversity.html#principal-component-analysis-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Principal Component Analysis (PCA)</strong> uses a linear combination algorithm to obtain the principal components (PC) (the number of PCs according to the samples). The results have been effected by the normalization methods because different count numbers would affect the PCs. We recommend you using filtering or normalization before performing PCA.</p>
<div id="preprocessing-phyloseq-object" class="section level4 hasAnchor" number="8.9.1.1">
<h4><span class="header-section-number">8.9.1.1</span> Preprocessing phyloseq object<a href="beta-diversity.html#preprocessing-phyloseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="beta-diversity.html#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;dada2_ps&quot;</span>)</span>
<span id="cb175-2"><a href="beta-diversity.html#cb175-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-3"><a href="beta-diversity.html#cb175-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb175-4"><a href="beta-diversity.html#cb175-4" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb175-5"><a href="beta-diversity.html#cb175-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb175-6"><a href="beta-diversity.html#cb175-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb175-7"><a href="beta-diversity.html#cb175-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb175-8"><a href="beta-diversity.html#cb175-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-9"><a href="beta-diversity.html#cb175-9" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Rarefying counts in phyloseq-class object</span></span>
<span id="cb175-10"><a href="beta-diversity.html#cb175-10" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(dada2_ps_remove_BRS, <span class="at">size =</span> <span class="dv">51181</span>, <span class="at">rng_seed =</span> <span class="dv">123</span>)</span>
<span id="cb175-11"><a href="beta-diversity.html#cb175-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-12"><a href="beta-diversity.html#cb175-12" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb175-13"><a href="beta-diversity.html#cb175-13" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(dada2_ps_rarefy, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, <span class="at">absolute =</span> <span class="cn">TRUE</span>)</span>
<span id="cb175-14"><a href="beta-diversity.html#cb175-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-15"><a href="beta-diversity.html#cb175-15" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb175-16"><a href="beta-diversity.html#cb175-16" aria-hidden="true" tabindex="-1"></a><span class="co">#dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(dada2_ps_rare_genus, cutoff = 10, unclass = TRUE)</span></span>
<span id="cb175-17"><a href="beta-diversity.html#cb175-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-18"><a href="beta-diversity.html#cb175-18" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb175-19"><a href="beta-diversity.html#cb175-19" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(dada2_ps_rare_genus, <span class="at">cutoff =</span> <span class="dv">10</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb175-20"><a href="beta-diversity.html#cb175-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-21"><a href="beta-diversity.html#cb175-21" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb175-22"><a href="beta-diversity.html#cb175-22" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(dada2_ps_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb175-23"><a href="beta-diversity.html#cb175-23" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 100 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 100 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="running-pca" class="section level4 hasAnchor" number="8.9.1.2">
<h4><span class="header-section-number">8.9.1.2</span> Running PCA<a href="beta-diversity.html#running-pca" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="beta-diversity.html#cb177-1" aria-hidden="true" tabindex="-1"></a>ordination_PCA <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb177-2"><a href="beta-diversity.html#cb177-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb177-3"><a href="beta-diversity.html#cb177-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb177-4"><a href="beta-diversity.html#cb177-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb177-5"><a href="beta-diversity.html#cb177-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb177-6"><a href="beta-diversity.html#cb177-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ordination_PCA)</span></code></pre></div>
<pre><code>## [1] &quot;fit&quot;           &quot;dat&quot;           &quot;explains&quot;      &quot;eigvalue&quot;      &quot;PERMANOVA&quot;     &quot;axis_taxa_cor&quot;</code></pre>
<p>The object of <code>run_ordination</code> is comprising of several results which could be used for visualization.</p>
<ul>
<li><p><em>fit</em>: the result of PCA functions from <code>stats::prcomp</code>;</p></li>
<li><p><em>dat</em>: the combination of PCs score and metadata group information;</p></li>
<li><p><em>explains</em>: the 1st and 2nd PCsâ€™ explains;</p></li>
<li><p><em>eigvalue</em>: the eigvalues of all the PCs;</p></li>
<li><p><em>PERMANOVA</em>: the result of PERMANOVA between the whole microbial community and group;</p></li>
<li><p><em>axis_taxa_cor</em>: inherit from the <code>XVIZ</code> package for visualization (only for PCoA analysis).</p></li>
</ul>
</div>
<div id="visualization" class="section level4 hasAnchor" number="8.9.1.3">
<h4><span class="header-section-number">8.9.1.3</span> Visualization<a href="beta-diversity.html#visualization" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><code>plot_Ordination</code> provides too many parameters for users to display the ordination results by using ggplot2 format. Here is the ordinary pattern.</p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="beta-diversity.html#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb179-2"><a href="beta-diversity.html#cb179-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-96"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-96-1.png" alt="Principal Component Analysis (PCA): Ordinary pattern" width="768" />
<p class="caption">
Figure 8.5: Principal Component Analysis (PCA): Ordinary pattern
</p>
</div>
<p><strong>Results:</strong></p>
<ol style="list-style-type: decimal">
<li><p>Scatterplot showed the distribution of samples in the two groups;</p></li>
<li><p>PERMANOVA results revealed that there was no association between the groups and the genus microbial community;</p></li>
<li><p>Sidelinechart showed the top 3 taxa related to PCA1 and PCA2.</p></li>
</ol>
<ul>
<li>plot with SampleID and setting group colors</li>
</ul>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="beta-diversity.html#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb180-2"><a href="beta-diversity.html#cb180-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb180-3"><a href="beta-diversity.html#cb180-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb180-4"><a href="beta-diversity.html#cb180-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb180-5"><a href="beta-diversity.html#cb180-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-97"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-97-1.png" alt="Principal Component Analysis (PCA): Ordinary pattern with SampleID" width="768" />
<p class="caption">
Figure 8.6: Principal Component Analysis (PCA): Ordinary pattern with SampleID
</p>
</div>
<ul>
<li>biplot with topN dominant taxa</li>
</ul>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="beta-diversity.html#cb181-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ggbiplot</span>(<span class="at">ResultList =</span> ordination_PCA,</span>
<span id="cb181-2"><a href="beta-diversity.html#cb181-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb181-3"><a href="beta-diversity.html#cb181-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb181-4"><a href="beta-diversity.html#cb181-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">topN =</span> <span class="dv">5</span>,</span>
<span id="cb181-5"><a href="beta-diversity.html#cb181-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">ellipse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb181-6"><a href="beta-diversity.html#cb181-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">labels =</span> <span class="st">&quot;SampleID&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-98"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-98-1.png" alt="Principal Component Analysis (PCA): biplot" width="576" />
<p class="caption">
Figure 8.7: Principal Component Analysis (PCA): biplot
</p>
</div>
<p><strong>biplot</strong> not only shows the distribution of samples, but also displays correlation among the dominant taxa. We chose top 5 dominant taxa.</p>
<blockquote>
<p>The length of vector approximates standard deviation of variables (bacteria); the angles between variables (bacteria in this case) reflect their correlations: the cosine of angle approximates correlation between variables (bacteria).</p>
</blockquote>
</div>
</div>
<div id="principal-coordinate-analysis-pcoa" class="section level3 hasAnchor" number="8.9.2">
<h3><span class="header-section-number">8.9.2</span> Principal Coordinate Analysis (PCoA)<a href="beta-diversity.html#principal-coordinate-analysis-pcoa" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Principal Coordinate Analysis (PCoA)</strong> could use different distance measures (e.g., Jaccard, Bray-Curtis, Euclidean, etc.) as input for ordination, but pay attention to the data matrix with negative values (not suitable for Bray-Curtis distance).</p>
<blockquote>
<p>As PCA, PCoA uses eigenvalues to measure the importance of a set of returned orthogonal axes. The dimensionality of matrix is reduced by determining each eigenvector and eigenvalue. The principal coordinates are obtained by scaling each eigenvector.</p>
</blockquote>
<div id="running-pcoa" class="section level4 hasAnchor" number="8.9.2.1">
<h4><span class="header-section-number">8.9.2.1</span> Running PCoA<a href="beta-diversity.html#running-pcoa" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Different distance measures could be affected by different factors (eg, low abundance taxa for Bary Curtis distance), so we recommend users taking care for whether performing the preprocess before calculating the distance. Here, we coincided to the tactics of the PCA analysis.</p>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="beta-diversity.html#cb182-1" aria-hidden="true" tabindex="-1"></a>ordination_PCoA <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb182-2"><a href="beta-diversity.html#cb182-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb182-3"><a href="beta-diversity.html#cb182-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb182-4"><a href="beta-diversity.html#cb182-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb182-5"><a href="beta-diversity.html#cb182-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ordination_PCoA)</span></code></pre></div>
<pre><code>## [1] &quot;fit&quot;           &quot;dat&quot;           &quot;explains&quot;      &quot;eigvalue&quot;      &quot;PERMANOVA&quot;     &quot;axis_taxa_cor&quot;</code></pre>
</div>
<div id="visualization-1" class="section level4 hasAnchor" number="8.9.2.2">
<h4><span class="header-section-number">8.9.2.2</span> Visualization<a href="beta-diversity.html#visualization-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="beta-diversity.html#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(ordination_PCoA, </span>
<span id="cb184-2"><a href="beta-diversity.html#cb184-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-100"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-100-1.png" alt="Principal Coordinate Analysis (PCoA)" width="768" />
<p class="caption">
Figure 8.8: Principal Coordinate Analysis (PCoA)
</p>
</div>
</div>
</div>
<div id="t-distributed-stochastic-neighbor-embedding-t-sne" class="section level3 hasAnchor" number="8.9.3">
<h3><span class="header-section-number">8.9.3</span> t-distributed stochastic neighbor embedding (t-SNE)<a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>t-distributed stochastic neighbor embedding (t-SNE) is a statistical method for visualizing high-dimensional data by giving each datapoint a location in a two or three-dimensional map. It is based on Stochastic Neighbor Embedding originally developed by Sam Roweis and Geoffrey Hinton, where Laurens van der Maaten proposed the t-distributed variant. It is a nonlinear dimensionality reduction technique well-suited for embedding high-dimensional data for visualization in a low-dimensional space of two or three dimensions. Specifically, it models each high-dimensional object by a two- or three-dimensional point in such a way that similar objects are modeled by nearby points and dissimilar objects are modeled by distant points with high probability.</p>
</blockquote>
<div id="running-tsne" class="section level4 hasAnchor" number="8.9.3.1">
<h4><span class="header-section-number">8.9.3.1</span> Running tSNE<a href="beta-diversity.html#running-tsne" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><em>Perplexity</em> parameter is for how many neighbors to be chosen.</p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="beta-diversity.html#cb185-1" aria-hidden="true" tabindex="-1"></a>ordination_tSNE <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb185-2"><a href="beta-diversity.html#cb185-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb185-3"><a href="beta-diversity.html#cb185-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb185-4"><a href="beta-diversity.html#cb185-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;tSNE&quot;</span>,</span>
<span id="cb185-5"><a href="beta-diversity.html#cb185-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">para =</span><span class="fu">list</span>(<span class="at">Perplexity=</span><span class="dv">2</span>))</span>
<span id="cb185-6"><a href="beta-diversity.html#cb185-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb185-7"><a href="beta-diversity.html#cb185-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ordination_tSNE)</span></code></pre></div>
<pre><code>## [1] &quot;fit&quot;           &quot;dat&quot;           &quot;explains&quot;      &quot;eigvalue&quot;      &quot;PERMANOVA&quot;     &quot;axis_taxa_cor&quot;</code></pre>
</div>
<div id="visualization-2" class="section level4 hasAnchor" number="8.9.3.2">
<h4><span class="header-section-number">8.9.3.2</span> Visualization<a href="beta-diversity.html#visualization-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><code>plot_Ordination</code> provides too many parameters for users to display the ordination results by using ggplot2 format. Here is the ordinary pattern.</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="beta-diversity.html#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_tSNE, </span>
<span id="cb187-2"><a href="beta-diversity.html#cb187-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-102"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-102-1.png" alt="t-distributed stochastic neighbor embedding (t-SNE): Ordinary pattern" width="768" />
<p class="caption">
Figure 8.9: t-distributed stochastic neighbor embedding (t-SNE): Ordinary pattern
</p>
</div>
</div>
</div>
<div id="uniform-manifold-approximation-and-projection-for-dimension-reduction-umap" class="section level3 hasAnchor" number="8.9.4">
<h3><span class="header-section-number">8.9.4</span> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Uniform Manifold Approximation and Projection (UMAP) <span class="citation">(<a href="#ref-mcinnes2022uniform" role="doc-biblioref">McInnes et al. 2022</a>)</span> is a dimension reduction technique that can be used for visualisation similarly to t-SNE, but also for general non-linear dimension reduction. The algorithm is founded on three assumptions about the data.</p>
</blockquote>
<div id="running-umap" class="section level4 hasAnchor" number="8.9.4.1">
<h4><span class="header-section-number">8.9.4.1</span> Running UMAP<a href="beta-diversity.html#running-umap" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="beta-diversity.html#cb188-1" aria-hidden="true" tabindex="-1"></a>ordination_UMAP <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb188-2"><a href="beta-diversity.html#cb188-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb188-3"><a href="beta-diversity.html#cb188-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb188-4"><a href="beta-diversity.html#cb188-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;UMAP&quot;</span>)</span>
<span id="cb188-5"><a href="beta-diversity.html#cb188-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-6"><a href="beta-diversity.html#cb188-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ordination_UMAP)</span></code></pre></div>
<pre><code>## [1] &quot;fit&quot;           &quot;dat&quot;           &quot;explains&quot;      &quot;eigvalue&quot;      &quot;PERMANOVA&quot;     &quot;axis_taxa_cor&quot;</code></pre>
</div>
<div id="visualization-3" class="section level4 hasAnchor" number="8.9.4.2">
<h4><span class="header-section-number">8.9.4.2</span> Visualization<a href="beta-diversity.html#visualization-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><code>plot_Ordination</code> provides too many parameters for users to display the ordination results by using ggplot2 format. Here is the ordinary pattern.</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="beta-diversity.html#cb190-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_UMAP, </span>
<span id="cb190-2"><a href="beta-diversity.html#cb190-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-104"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-104-1.png" alt="Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP): Ordinary pattern" width="768" />
<p class="caption">
Figure 8.10: Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP): Ordinary pattern
</p>
</div>
</div>
</div>
</div>
<div id="systematic-information-5" class="section level2 hasAnchor" number="8.10">
<h2><span class="header-section-number">8.10</span> Systematic Information<a href="beta-diversity.html#systematic-information-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="beta-diversity.html#cb191-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-06-23
##  rstudio  2022.02.0+443 Prairie Trillium (desktop)
##  pandoc   2.17.1.1 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/ (via rmarkdown)
## 
## â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
##  package              * version  date (UTC) lib source
##  abind                  1.4-5    2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18   2021-09-16 [1] CRAN (R 4.1.0)
##  ALDEx2                 1.26.0   2021-10-26 [1] Bioconductor
##  annotate               1.72.0   2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2   2021-11-09 [1] Bioconductor
##  ape                  * 5.6-2    2022-03-02 [1] CRAN (R 4.1.2)
##  askpass                1.1      2019-01-13 [1] CRAN (R 4.1.0)
##  assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1    2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3    2015-07-28 [1] CRAN (R 4.1.0)
##  bayesm                 3.1-4    2019-10-15 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0   2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0   2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3   2021-12-09 [1] Bioconductor
##  biomformat             1.22.0   2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0   2021-10-26 [1] Bioconductor
##  bit                    4.0.4    2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5    2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7    2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2    2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.24     2021-09-02 [1] CRAN (R 4.1.0)
##  brio                   1.1.3    2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12   2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1    2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6    2021-08-19 [1] CRAN (R 4.1.0)
##  callr                  3.7.0    2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12   2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5    2022-01-06 [1] CRAN (R 4.1.2)
##  caTools                1.18.2   2021-03-28 [1] CRAN (R 4.1.0)
##  cellranger             1.1.0    2016-07-27 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0    2020-02-06 [1] CRAN (R 4.1.0)
##  class                  7.3-20   2022-01-13 [1] CRAN (R 4.1.2)
##  classInt               0.4-3    2020-04-07 [1] CRAN (R 4.1.0)
##  cli                    3.2.0    2022-02-14 [1] CRAN (R 4.1.2)
##  cluster                2.1.2    2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18   2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2    2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3    2022-02-21 [1] CRAN (R 4.1.2)
##  compositions           2.0-4    2022-01-05 [1] CRAN (R 4.1.2)
##  conflicted           * 1.1.0    2021-11-26 [1] CRAN (R 4.1.0)
##  corrplot               0.92     2021-11-18 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1    2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0    2022-02-14 [1] CRAN (R 4.1.2)
##  crosstalk              1.2.0    2021-11-04 [1] CRAN (R 4.1.0)
##  data.table             1.14.2   2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2    2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0   2021-10-26 [1] Bioconductor
##  DEoptimR               1.0-10   2022-01-03 [1] CRAN (R 4.1.2)
##  desc                   1.4.1    2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0   2021-10-26 [1] Bioconductor
##  devtools               2.4.3    2021-11-30 [1] CRAN (R 4.1.0)
##  DiagrammeR             1.0.9    2022-03-05 [1] CRAN (R 4.1.2)
##  digest                 0.6.29   2021-12-01 [1] CRAN (R 4.1.0)
##  dplyr                * 1.0.8    2022-02-08 [1] CRAN (R 4.1.2)
##  DT                     0.21     2022-02-26 [1] CRAN (R 4.1.2)
##  e1071                  1.7-9    2021-09-16 [1] CRAN (R 4.1.0)
##  edgeR                  3.36.0   2021-10-26 [1] Bioconductor
##  ellipsis               0.3.2    2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15     2022-02-18 [1] CRAN (R 4.1.2)
##  fansi                  1.0.2    2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0    2021-02-28 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0    2021-01-25 [1] CRAN (R 4.1.0)
##  foreach                1.5.2    2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82   2022-01-13 [1] CRAN (R 4.1.2)
##  forestplot             2.0.1    2021-09-03 [1] CRAN (R 4.1.0)
##  Formula                1.2-4    2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2    2021-12-08 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0   2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0   2021-10-26 [1] Bioconductor
##  generics               0.1.2    2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1   2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7    2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1   2021-11-18 [1] Bioconductor
##  ggplot2              * 3.3.5    2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr               * 0.4.0    2020-06-27 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1    2021-01-15 [1] CRAN (R 4.1.0)
##  ggsci                  2.9      2018-05-14 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3    2021-09-09 [1] CRAN (R 4.1.0)
##  ggVennDiagram          1.2.1    2022-04-13 [1] Github (gaospecial/ggVennDiagram@db6742d)
##  glmnet                 4.1-3    2021-11-02 [1] CRAN (R 4.1.0)
##  glue                 * 1.6.2    2022-02-24 [1] CRAN (R 4.1.2)
##  Gmisc                * 3.0.0    2022-01-03 [1] CRAN (R 4.1.2)
##  gplots                 3.1.1    2020-11-28 [1] CRAN (R 4.1.0)
##  gridExtra              2.3      2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2    2021-06-06 [1] CRAN (R 4.1.0)
##  highr                  0.9      2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0    2021-10-07 [1] CRAN (R 4.1.0)
##  htmlTable            * 2.4.0    2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2    2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4    2021-09-08 [1] CRAN (R 4.1.0)
##  httr                   1.4.2    2020-07-20 [1] CRAN (R 4.1.0)
##  igraph                 1.2.11   2022-01-04 [1] CRAN (R 4.1.2)
##  IRanges              * 2.28.0   2021-10-26 [1] Bioconductor
##  iterators              1.0.14   2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9    2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4    2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0    2022-02-22 [1] CRAN (R 4.1.2)
##  kableExtra             1.3.4    2021-02-20 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0   2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20  2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.37     2021-12-16 [1] CRAN (R 4.1.0)
##  labeling               0.4.2    2020-10-20 [1] CRAN (R 4.1.0)
##  lattice              * 0.20-45  2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.1.0)
##  libcoin                1.0-9    2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1    2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1   2022-02-17 [1] Bioconductor
##  locfit                 1.5-9.5  2022-03-03 [1] CRAN (R 4.1.2)
##  lubridate              1.8.0    2021-10-07 [1] CRAN (R 4.1.0)
##  magrittr             * 2.0.2    2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55   2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix                 1.4-0    2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0    2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0   2021-09-17 [1] CRAN (R 4.1.0)
##  mbzinb                 0.2      2022-03-16 [1] local
##  memoise                2.0.1    2021-11-26 [1] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0   2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39   2022-02-24 [1] CRAN (R 4.1.2)
##  microbiome             1.16.0   2021-10-26 [1] Bioconductor
##  modeltools             0.2-23   2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18   2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0   2021-10-26 [1] Bioconductor
##  munsell                0.5.0    2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3    2021-10-08 [1] CRAN (R 4.1.0)
##  NADA                   1.6-1.1  2020-03-22 [1] CRAN (R 4.1.0)
##  nlme                 * 3.1-155  2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17   2022-01-13 [1] CRAN (R 4.1.2)
##  openssl                2.0.0    2022-03-02 [1] CRAN (R 4.1.2)
##  permute              * 0.9-7    2022-01-27 [1] CRAN (R 4.1.2)
##  pheatmap               1.0.12   2019-01-04 [1] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0   2021-10-26 [1] Bioconductor
##  picante              * 1.8.2    2020-06-10 [1] CRAN (R 4.1.0)
##  pillar                 1.7.0    2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1    2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4    2021-11-30 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7    2013-12-03 [1] CRAN (R 4.1.0)
##  prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.1.0)
##  processx               3.5.2    2021-04-30 [1] CRAN (R 4.1.0)
##  protoclust             1.6.3    2019-01-31 [1] CRAN (R 4.1.0)
##  proxy                  0.4-26   2021-06-07 [1] CRAN (R 4.1.0)
##  ps                     1.6.0    2021-02-28 [1] CRAN (R 4.1.0)
##  pscl                   1.5.5    2020-03-07 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4    2020-04-17 [1] CRAN (R 4.1.0)
##  qvalue                 2.26.0   2021-10-26 [1] Bioconductor
##  R6                     2.5.1    2021-08-19 [1] CRAN (R 4.1.0)
##  RAIDA                  1.0      2022-03-14 [1] local
##  RColorBrewer         * 1.1-2    2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                 * 1.0.8.2  2022-03-11 [1] CRAN (R 4.1.2)
##  RcppZiggurat           0.1.6    2020-10-20 [1] CRAN (R 4.1.0)
##  RCurl                  1.98-1.6 2022-02-08 [1] CRAN (R 4.1.2)
##  readxl               * 1.4.0    2022-03-28 [1] CRAN (R 4.1.2)
##  remotes                2.4.2    2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.1.0)
##  reticulate             1.24     2022-01-26 [1] CRAN (R 4.1.2)
##  Rfast                  2.0.6    2022-02-16 [1] CRAN (R 4.1.2)
##  rhdf5                  2.38.1   2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0    2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0   2021-10-26 [1] Bioconductor
##  rlang                  1.0.2    2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.13     2022-03-10 [1] CRAN (R 4.1.2)
##  robustbase             0.93-9   2021-09-27 [1] CRAN (R 4.1.0)
##  rpart                  4.1.16   2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2    2020-11-15 [1] CRAN (R 4.1.0)
##  RSpectra               0.16-0   2019-12-01 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10   2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0    2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13     2020-11-12 [1] CRAN (R 4.1.0)
##  Rtsne                  0.15     2018-11-10 [1] CRAN (R 4.1.0)
##  RVenn                  1.1.0    2019-07-18 [1] CRAN (R 4.1.0)
##  rvest                  1.0.2    2021-10-16 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3   2021-11-21 [1] Bioconductor
##  sandwich               3.0-1    2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0    2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1    2020-05-11 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2    2021-12-06 [1] CRAN (R 4.1.0)
##  sf                     1.0-7    2022-03-07 [1] CRAN (R 4.1.2)
##  shape                  1.4.6    2021-05-19 [1] CRAN (R 4.1.0)
##  stringi                1.7.6    2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0    2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0   2021-10-26 [1] Bioconductor
##  survival               3.3-1    2022-03-03 [1] CRAN (R 4.1.2)
##  svglite                2.1.0    2022-02-03 [1] CRAN (R 4.1.2)
##  systemfonts            1.0.4    2022-02-11 [1] CRAN (R 4.1.2)
##  tensorA                0.36.2   2020-11-19 [1] CRAN (R 4.1.0)
##  testthat               3.1.2    2022-01-20 [1] CRAN (R 4.1.2)
##  TH.data                1.1-0    2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6    2021-11-07 [1] CRAN (R 4.1.0)
##  tidyr                  1.2.0    2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2    2022-02-21 [1] CRAN (R 4.1.2)
##  truncnorm              1.0-8    2018-02-27 [1] CRAN (R 4.1.0)
##  umap                   0.2.8.0  2022-03-23 [1] CRAN (R 4.1.2)
##  units                  0.8-0    2022-02-05 [1] CRAN (R 4.1.2)
##  usethis                2.1.5    2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2    2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8    2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                * 2.5-7    2020-11-28 [1] CRAN (R 4.1.0)
##  viridis              * 0.6.2    2021-10-13 [1] CRAN (R 4.1.0)
##  viridisLite          * 0.4.0    2021-04-13 [1] CRAN (R 4.1.0)
##  visNetwork             2.1.0    2021-09-29 [1] CRAN (R 4.1.0)
##  webshot                0.5.3    2022-04-14 [1] CRAN (R 4.1.2)
##  withr                  2.5.0    2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0   2021-10-26 [1] Bioconductor
##  xfun                   0.30     2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                * 2.1.4    2022-06-22 [1] local
##  XML                    3.99-0.9 2022-02-24 [1] CRAN (R 4.1.2)
##  xml2                   1.3.3    2021-11-30 [1] CRAN (R 4.1.0)
##  xtable                 1.8-4    2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0   2021-10-26 [1] Bioconductor
##  yaml                   2.3.5    2022-02-21 [1] CRAN (R 4.1.2)
##  zCompositions          1.4.0    2022-01-13 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0   2021-10-26 [1] Bioconductor
##  zoo                    1.8-9    2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-anderson2014permutational" class="csl-entry">
Anderson, Marti J. 2014. <span>â€œPermutational Multivariate Analysis of Variance (PERMANOVA).â€</span> <em>Wiley Statsref: Statistics Reference Online</em>, 1â€“15.
</div>
<div id="ref-chen2012associating" class="csl-entry">
Chen, Jun, Kyle Bittinger, Emily S Charlson, Christian Hoffmann, James Lewis, Gary D Wu, Ronald G Collman, Frederic D Bushman, and Hongzhe Li. 2012. <span>â€œAssociating Microbiome Composition with Environmental Covariates Using Generalized UniFrac Distances.â€</span> <em>Bioinformatics</em> 28 (16): 2106â€“13.
</div>
<div id="ref-mantel1967detection" class="csl-entry">
Mantel, Nathan. 1967. <span>â€œThe Detection of Disease Clustering and a Generalized Regression Approach.â€</span> <em>Cancer Research</em> 27 (2 Part 1): 209â€“20.
</div>
<div id="ref-mcinnes2022uniform" class="csl-entry">
McInnes, L et al. 2022. <span>â€œUniform Manifold Approximation and Projection for Dimension Reduction.â€</span> <em>arXiv Preprint arXiv:1802.03426</em>.
</div>
<div id="ref-mielke1991application" class="csl-entry">
Mielke Jr, Paul W. 1991. <span>â€œThe Application of Multivariate Permutation Methods Based on Distance Functions in the Earth Sciences.â€</span> <em>Earth-Science Reviews</em> 31 (1): 55â€“71.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="aplha-diversity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="composition.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/07-beta-diversity.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
