<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-06-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="core-microbiota.html"/>
<link rel="next" href="visualization-5.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation<span></span></a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette<span></span></a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors<span></span></a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2<span></span></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>4.2.1</b> Importing results from dada2 pipeline<span></span></a></li>
<li class="chapter" data-level="4.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>4.2.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>4.2.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.2.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>4.2.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.2.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>4.2.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>4.3.1</b> Importing results<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table-1"><i class="fa fa-check"></i><b>4.3.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table-1"><i class="fa fa-check"></i><b>4.3.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.3.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table-1"><i class="fa fa-check"></i><b>4.3.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.3.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object-1"><i class="fa fa-check"></i><b>4.3.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Readsâ€™ track<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>5.6</b> Variability<span></span></a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>5.6.1</b> Coefficient of variation<span></span></a></li>
<li class="chapter" data-level="5.6.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>5.6.2</b> Distribution of number of sequences<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.7</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>6.8</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)<span></span></a></li>
<li class="chapter" data-level="6.9" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.9</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities<span></span></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>7.2.1</b> Non-phylogenetic diversities<span></span></a></li>
<li class="chapter" data-level="7.2.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2.2</b> Phylogenetic diversity<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics<span></span></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>8.6</b> Mantel Test<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.7</b> Analysis of Similarity (ANOSIM)<span></span></a></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>8.8</b> Multi-response permutation procedures (MRPP)<span></span></a></li>
<li class="chapter" data-level="8.9" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.9</b> Ordination<span></span></a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.9.1</b> Principal Component Analysis (PCA)<span></span></a></li>
<li class="chapter" data-level="8.9.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.9.2</b> Principal Coordinate Analysis (PCoA)<span></span></a></li>
<li class="chapter" data-level="8.9.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.9.3</b> t-distributed stochastic neighbor embedding (t-SNE)<span></span></a></li>
<li class="chapter" data-level="8.9.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.9.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.10</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="composition.html"><a href="composition.html"><i class="fa fa-check"></i><b>9</b> Composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="composition.html"><a href="composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="composition.html"><a href="composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.2</b> XVIZ <code>plot_stacked_bar</code><span></span></a></li>
<li class="chapter" data-level="9.3" data-path="composition.html"><a href="composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.3</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="composition.html"><a href="composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="composition.html"><a href="composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus<span></span></a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq<span></span></a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR<span></span></a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM<span></span></a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob<span></span></a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)<span></span></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom-1"><i class="fa fa-check"></i><b>11.2.1</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.2</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.3</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.4</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.2.5" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.5</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot<span></span></a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot<span></span></a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot<span></span></a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density<span></span></a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination<span></span></a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot<span></span></a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot<span></span></a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn<span></span></a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.10</b> plot_stacked_bar_XIVZ<span></span></a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.11</b> plot_StackBarPlot<span></span></a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.12</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>13</b> Principals of Differential Analysis methods<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>13.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>13.2</b> ANCOM-II<span></span></a></li>
<li class="chapter" data-level="13.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>13.3</b> corncob<span></span></a></li>
<li class="chapter" data-level="13.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>13.4</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="13.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>13.5</b> edgeR<span></span></a></li>
<li class="chapter" data-level="13.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>13.6</b> LEfSe<span></span></a></li>
<li class="chapter" data-level="13.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>13.7</b> limma voom<span></span></a></li>
<li class="chapter" data-level="13.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>13.8</b> MaAsLin2<span></span></a></li>
<li class="chapter" data-level="13.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>13.9</b> metagenomeSeq<span></span></a></li>
<li class="chapter" data-level="13.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>13.10</b> t-test<span></span></a></li>
<li class="chapter" data-level="13.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>13.11</b> Wilcoxon test<span></span></a></li>
<li class="chapter" data-level="13.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>13.12</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="13.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>13.13</b> Summary<span></span></a></li>
<li class="chapter" data-level="13.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>13.14</b> Advantages and disadvantages<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>14</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>14.1</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="14.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>14.2</b> Amplicon sequencing (16s)<span></span></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>14.2.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="14.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>14.2.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="14.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>14.2.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="14.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>14.2.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="14.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>14.2.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="14.2.6" data-path="examples.html"><a href="examples.html#microbial-composition"><i class="fa fa-check"></i><b>14.2.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="14.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>14.2.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="examples.html"><a href="examples.html#metagenomics"><i class="fa fa-check"></i><b>14.3</b> Metagenomics<span></span></a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>14.3.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="14.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>14.3.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="14.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>14.3.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="14.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>14.3.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="14.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>14.3.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="14.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>14.3.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="14.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>14.3.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>14.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data analysis based on XMAS 2.0 package in microbiota data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-analysis" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Differential analysis<a href="differential-analysis.html#differential-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Identifying the significant taxa between case and control group is necessary. There are too many differential analysis (DA) approaches to choose <span class="citation">(<a href="#ref-nearing2022microbiome" role="doc-biblioref">Nearing et al. 2022</a>)</span>, we provide some of them which focusing on microbial data, including:</p>
<table style="width:100%;">
<colgroup>
<col width="14%" />
<col width="15%" />
<col width="24%" />
<col width="15%" />
<col width="19%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Tool(version)</strong></th>
<th><strong>Input</strong></th>
<th><strong>Normalization</strong></th>
<th><strong>Transformation</strong></th>
<th><strong>Distribution</strong></th>
<th><strong>MicrobialData</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ALDEx2 (1.26.0)</td>
<td>Counts</td>
<td>None</td>
<td>CLR</td>
<td>Dirichlet-multinormial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>limma voom (3.50.1)</td>
<td>Counts/Relative</td>
<td>None/TMM</td>
<td>Log; Precision weighting</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>mbzinb (0.2)</td>
<td>Counts</td>
<td>RLE</td>
<td>None</td>
<td>zero-inflated negative binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>omnibus (0.2)</td>
<td>Counts</td>
<td>GMPR(Geometric Mean of Pairwise Ratios)</td>
<td>None</td>
<td>zero-inflated negative binomial</td>
<td>16s</td>
</tr>
<tr class="odd">
<td>RAIDA (1.0)</td>
<td>Counts</td>
<td>None</td>
<td>zero-inflated log normal</td>
<td>modified t-test</td>
<td>16s</td>
</tr>
<tr class="even">
<td>Wilcox(rare/CLR)</td>
<td>Counts/Relative</td>
<td>None</td>
<td>None/CLR</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>LEfSe</td>
<td>Rarefied Counts//Relative</td>
<td>TSS</td>
<td>None</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="even">
<td>t-test (rare)</td>
<td>Counts//Relative</td>
<td>None</td>
<td>None</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>metagenomeSeq (1.36.0)</td>
<td>Counts</td>
<td>CSS</td>
<td>Log</td>
<td>Zero-inflated (log) Normal</td>
<td>16s</td>
</tr>
<tr class="even">
<td>DESeq2 (1.34.0)</td>
<td>Counts</td>
<td>RLE</td>
<td>None</td>
<td>Negative binomial</td>
<td>16s</td>
</tr>
<tr class="odd">
<td>edgeR (3.36.0)</td>
<td>Counts</td>
<td>RLE/TMM</td>
<td>None</td>
<td>Negative binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>ANCOM-II (2.1)</td>
<td>Counts/Relative</td>
<td>None</td>
<td>ALR</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>Corncob (0.2.0)</td>
<td>Counts</td>
<td>None</td>
<td>None</td>
<td>Beta-binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>MaAslin2 (1.8.0)</td>
<td>Counts/Relative abundance</td>
<td>None/TSS</td>
<td>AST</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
</tbody>
</table>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="differential-analysis.html#cb218-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb218-2"><a href="differential-analysis.html#cb218-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb218-3"><a href="differential-analysis.html#cb218-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb218-4"><a href="differential-analysis.html#cb218-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb218-5"><a href="differential-analysis.html#cb218-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb218-6"><a href="differential-analysis.html#cb218-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb218-7"><a href="differential-analysis.html#cb218-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
<div id="amplicon-sequencing-microbial-data-16s" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Amplicon sequencing microbial data (16s)<a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same strategy to filter phyloseq object.</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="differential-analysis.html#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;dada2_ps&quot;</span>)</span>
<span id="cb219-2"><a href="differential-analysis.html#cb219-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-3"><a href="differential-analysis.html#cb219-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb219-4"><a href="differential-analysis.html#cb219-4" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb219-5"><a href="differential-analysis.html#cb219-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb219-6"><a href="differential-analysis.html#cb219-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb219-7"><a href="differential-analysis.html#cb219-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb219-8"><a href="differential-analysis.html#cb219-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-9"><a href="differential-analysis.html#cb219-9" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Rarefying counts in phyloseq-class object</span></span>
<span id="cb219-10"><a href="differential-analysis.html#cb219-10" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(dada2_ps_remove_BRS, <span class="at">size =</span> <span class="dv">51181</span>, <span class="at">rng_seed =</span> <span class="dv">123</span>)</span>
<span id="cb219-11"><a href="differential-analysis.html#cb219-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-12"><a href="differential-analysis.html#cb219-12" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb219-13"><a href="differential-analysis.html#cb219-13" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(dada2_ps_rarefy, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, <span class="at">absolute =</span> <span class="cn">TRUE</span>)</span>
<span id="cb219-14"><a href="differential-analysis.html#cb219-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-15"><a href="differential-analysis.html#cb219-15" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb219-16"><a href="differential-analysis.html#cb219-16" aria-hidden="true" tabindex="-1"></a><span class="co">#dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(dada2_ps_rare_genus, cutoff = 10, unclass = TRUE)</span></span>
<span id="cb219-17"><a href="differential-analysis.html#cb219-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-18"><a href="differential-analysis.html#cb219-18" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb219-19"><a href="differential-analysis.html#cb219-19" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(dada2_ps_rare_genus, <span class="at">cutoff =</span> <span class="dv">10</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb219-20"><a href="differential-analysis.html#cb219-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb219-21"><a href="differential-analysis.html#cb219-21" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb219-22"><a href="differential-analysis.html#cb219-22" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(dada2_ps_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb219-23"><a href="differential-analysis.html#cb219-23" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 100 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 100 taxa by 6 taxonomic ranks ]</code></pre>
<div id="aldex2" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> ALDEx2<a href="differential-analysis.html#aldex2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ALDEx2</strong> package is from <em>Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis</em> <span class="citation">(<a href="#ref-fernandes2014unifying" role="doc-biblioref">Fernandes et al. 2014</a>)</span>, and its principle is using log-ratio transformation and statistical testing to find the significant Taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<p><code>run_aldex</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_aldex)</code>.</p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="differential-analysis.html#cb221-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb221-2"><a href="differential-analysis.html#cb221-2" aria-hidden="true" tabindex="-1"></a>                     dada2_ps_genus_filter_trim,</span>
<span id="cb221-3"><a href="differential-analysis.html#cb221-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb221-4"><a href="differential-analysis.html#cb221-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb221-5"><a href="differential-analysis.html#cb221-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="differential-analysis.html#cb223-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="differential-analysis.html#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment  EffectSize    Pvalue AdjustedPvalue Median CLR \n(All) Median CLR\nAA
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif  0.07239664 0.6829697      0.9074981         -3.6962601      -3.933943
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif  0.13567685 0.6264621      0.9051654          2.5968337       1.897346
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif  0.33922392 0.3086642      0.7773584         -0.7605107      -2.145597
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif -0.08944437 0.6774884      0.9159834          5.3290908       5.767424
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif -0.08293411 0.5757626      0.8766907         -3.2358773      -3.032950
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif -0.06025205 0.6819865      0.9219725          3.5138246       4.455583
##   Median CLR\nBB Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1    -3.53755567                                NA                       0                    0                  0.0
## 2     3.08696471                        -0.9577718                      30                   26                 50.5
## 3     0.06294616                                NA                       0                    0                  9.0
## 4     5.23163019                         0.1524904                     316                  329                296.0
## 5    -3.37837145                                NA                       0                    0                  0.0
## 6     3.34614921                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       1.7051442              58.82609         101.777778           31.21429                    21.74
## 2                      -1.3107676              50.91304          26.777778           66.42857                    82.61
## 3                      -2.4683647              19.21739           5.111111           28.28571                    43.48
## 4                       1.7488297             996.26087        1740.444444          517.85714                    65.22
## 5                      -2.1676332              93.78261          30.000000          134.78571                    21.74
## 6                       0.2000385             163.86957         177.888889          154.85714                    73.91
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                 88.89                 78.57       4.1 (6.8;1.3)
## 3                 33.33                 50.00        8.6 (13;4.3)
## 4                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                 22.22                 21.43      1.5 (2.3;0.71)
## 6                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Median CLR (All)/(group AA)/(group BB)</strong>: Median CLR (normalization by <strong>ALDEx2</strong>) in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>ALDEx2</strong>.</p>
<details>
<summary>
<strong>run_da() pattern</strong>
</summary>
<p>We also provide another function <code>run_da</code> to run ALDEx2 differential analysis.</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="differential-analysis.html#cb227-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb227-2"><a href="differential-analysis.html#cb227-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb227-3"><a href="differential-analysis.html#cb227-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb227-4"><a href="differential-analysis.html#cb227-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb227-5"><a href="differential-analysis.html#cb227-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;aldex&quot;</span>,</span>
<span id="cb227-6"><a href="differential-analysis.html#cb227-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="differential-analysis.html#cb229-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>otu_table or sample_table as inputdata</strong>
</summary>
<p>We also provide <em>data_otu</em> and <em>data_sam</em> as input data to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="differential-analysis.html#cb231-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb231-2"><a href="differential-analysis.html#cb231-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb231-3"><a href="differential-analysis.html#cb231-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb231-4"><a href="differential-analysis.html#cb231-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb231-5"><a href="differential-analysis.html#cb231-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb231-6"><a href="differential-analysis.html#cb231-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="differential-analysis.html#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>taxa_level option</strong>
</summary>
<p>We also provide <em>taxa_level</em> for choosing the specific taxonomic level to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="differential-analysis.html#cb235-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb235-2"><a href="differential-analysis.html#cb235-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb235-3"><a href="differential-analysis.html#cb235-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb235-4"><a href="differential-analysis.html#cb235-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb235-5"><a href="differential-analysis.html#cb235-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb235-6"><a href="differential-analysis.html#cb235-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="differential-analysis.html#cb237-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
</div>
<div id="limma_voom" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> limma_voom<a href="differential-analysis.html#limma_voom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>limma</strong> package is from <em>voom: Precision weights unlock linear model analysis tools for RNA-seq read counts</em> <span class="citation">(<a href="#ref-law2014voom" role="doc-biblioref">Law et al. 2014</a>)</span>. Firstly, transforming count data to log2-counts per million (logCPM), estimate the mean-variance relationship and use this to compute appropriate observation-level weights. Secondly, fitting multiple linear models by weighted or generalized least squares. Finally, performing empirical bayes statistics for differential expression.</p>
<p><code>run_limma_voom</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_limma_voom)</code>.</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="differential-analysis.html#cb239-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb239-2"><a href="differential-analysis.html#cb239-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb239-3"><a href="differential-analysis.html#cb239-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb239-4"><a href="differential-analysis.html#cb239-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb239-5"><a href="differential-analysis.html#cb239-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-6"><a href="differential-analysis.html#cb239-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="differential-analysis.html#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment EffectSize     logFC       Pvalue AdjustedPvalue
## 1 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif   5.026490  5.026490 0.0009204053     0.07370632
## 2             g__Intestinibacter 9_AA vs 14_BB  Nonsignif   4.689815  4.689815 0.0015682197     0.07370632
## 3                  g__Romboutsia 9_AA vs 14_BB  Nonsignif   4.756742  4.756742 0.0034087111     0.10680628
## 4     g__Ruminococcaceae_UCG_004 9_AA vs 14_BB  Nonsignif  -3.933508 -3.933508 0.0188321702     0.40047179
## 5               g__Lactobacillus 9_AA vs 14_BB  Nonsignif   3.886899  3.886899 0.0221288616     0.40047179
## 6                 g__Veillonella 9_AA vs 14_BB  Nonsignif   3.430915  3.430915 0.0259684765     0.40047179
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                      11                    0                 39.5
## 2                                NA                      14                    0                 58.0
## 3                         -7.036174                     107                    2                262.5
## 4                                NA                      11                  104                  0.0
## 5                         -3.033423                      56                   24                196.5
## 6                                NA                      13                    0                 40.5
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       -4.884638             257.21739           14.00000          413.57143                    60.87
## 2                       -2.157952              83.56522           26.88889          120.00000                    60.87
## 3                       -3.131606             296.47826           51.77778          453.78571                    78.26
## 4                        2.275424              65.26087          126.22222           26.07143                    52.17
## 5                       -5.090571             791.26087           37.44444         1275.85714                    86.96
## 6                       -0.635927              65.52174           49.00000           76.14286                    56.52
##   Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                 22.22                 85.71       11000 (11000;11000)
## 2                 22.22                 85.71               4.2 (7;1.4)
## 3                 55.56                 92.86                37 (44;30)
## 4                 77.78                 35.71           0.17 (-3.3;3.6)
## 5                 77.78                 92.86 1.8e+08 (1.8e+08;1.8e+08)
## 6                 22.22                 78.57            1.3 (1.8;0.79)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>limma</strong>;</p></li>
<li><p><strong>logFC</strong>: LogFC from groupsâ€™ coefficient by <strong>limma</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>limma</strong>;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>limma</strong>.</p>
<details>
<summary>
<strong>other options for limma-voom</strong>
</summary>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="differential-analysis.html#cb243-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb243-2"><a href="differential-analysis.html#cb243-2" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb243-3"><a href="differential-analysis.html#cb243-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb243-4"><a href="differential-analysis.html#cb243-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb243-5"><a href="differential-analysis.html#cb243-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb243-6"><a href="differential-analysis.html#cb243-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb243-7"><a href="differential-analysis.html#cb243-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb243-8"><a href="differential-analysis.html#cb243-8" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb243-9"><a href="differential-analysis.html#cb243-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb243-10"><a href="differential-analysis.html#cb243-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb243-11"><a href="differential-analysis.html#cb243-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb243-12"><a href="differential-analysis.html#cb243-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb243-13"><a href="differential-analysis.html#cb243-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb243-14"><a href="differential-analysis.html#cb243-14" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb243-15"><a href="differential-analysis.html#cb243-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb243-16"><a href="differential-analysis.html#cb243-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb243-17"><a href="differential-analysis.html#cb243-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb243-18"><a href="differential-analysis.html#cb243-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb243-19"><a href="differential-analysis.html#cb243-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span>
<span id="cb243-20"><a href="differential-analysis.html#cb243-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="mbzinb" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> mbzinb<a href="differential-analysis.html#mbzinb" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>mbzinb</strong> package is from <em>An omnibus test for differential distribution analysis of microbiome sequencing data</em> <span class="citation">(<a href="#ref-chen2018omnibus" role="doc-biblioref">J. Chen et al. 2018</a>)</span>. It uses zeroinflated negative binomial model to investigate the significant taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="differential-analysis.html#cb244-1" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb244-2"><a href="differential-analysis.html#cb244-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb244-3"><a href="differential-analysis.html#cb244-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb244-4"><a href="differential-analysis.html#cb244-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb244-5"><a href="differential-analysis.html#cb244-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;base.mean&quot;                         &quot;mean.LFC&quot;                          &quot;base.abund&quot;                       
## [10] &quot;abund.LFC&quot;                         &quot;base.prev&quot;                         &quot;prev.change&quot;                      
## [13] &quot;base.disp&quot;                         &quot;disp.LFC&quot;                          &quot;statistic&quot;                        
## [16] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [19] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [22] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [25] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="differential-analysis.html#cb246-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment      Pvalue AdjustedPvalue EffectSize base.mean  mean.LFC
## 1               g__Lactobacillus 9_AA vs 14_BB  Nonsignif 0.003080337      0.1719279 1238.41270  33.04066 4.5637942
## 2                 g__Veillonella 9_AA vs 14_BB  Nonsignif 0.005686254      0.1719279   27.14286  40.19925 0.6436683
## 3 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif 0.006927761      0.1719279  399.57143   9.95499 4.6283681
## 4         g__Ruminiclostridium_5 9_AA vs 14_BB  Nonsignif 0.007316082      0.1719279   25.62698  29.88426 0.8485199
## 5             g__Intestinibacter 9_AA vs 14_BB  Nonsignif 0.014271033      0.2063066   93.11111  19.19355 2.2016604
## 6                  g__Romboutsia 9_AA vs 14_BB  Nonsignif 0.015319618      0.2063066  402.00794  38.54948 3.3294991
##   base.abund  abund.LFC base.prev prev.change  base.disp   disp.LFC statistic Log2FoldChange (Median)\nAA_vs_BB
## 1   41.89243  4.2857354 0.7887024   0.1676483 0.87559689  1.2857331  13.87500                        -3.0334230
## 2  180.85020 -1.1919001 0.2222793   0.5710621 0.05956404  3.9688755  12.56199                                NA
## 3   43.43914  2.5028760 0.2291710   0.7708210 1.13937611  1.5408637  12.13717                                NA
## 4   29.88426  1.0706021 1.0000000  -0.1426729 0.11934577  1.7731358  12.01967                        -0.3289485
## 5   75.51123  0.4356221 0.2541814   0.6103370 2.58097390 -1.4780782  10.57342                                NA
## 6   65.20531  2.6641701 0.5912015   0.3464026 1.66131464 -0.4193493  10.41913                        -7.0361736
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                      56                   24                196.5                      -5.0905713             791.26087
## 2                      13                    0                 40.5                      -0.6359270              65.52174
## 3                      11                    0                 39.5                      -4.8846378             257.21739
## 4                      45                   41                 51.5                      -0.7849381              51.04348
## 5                      14                    0                 58.0                      -2.1579524              83.56522
## 6                     107                    2                262.5                      -3.1316062             296.47826
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1           37.44444         1275.85714                    86.96                 77.78                 92.86
## 2           49.00000           76.14286                    56.52                 22.22                 78.57
## 3           14.00000          413.57143                    60.87                 22.22                 85.71
## 4           35.44444           61.07143                    91.30                100.00                 85.71
## 5           26.88889          120.00000                    60.87                 22.22                 85.71
## 6           51.77778          453.78571                    78.26                 55.56                 92.86
##         Odds Ratio (95% CI)
## 1 1.8e+08 (1.8e+08;1.8e+08)
## 2            1.3 (1.8;0.79)
## 3       11000 (11000;11000)
## 4            2.4 (4.2;0.69)
## 5               4.2 (7;1.4)
## 6                37 (44;30)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>mbzinb</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>mbzinb</strong>;</p></li>
<li><p><strong>base.mean</strong>: fitted mean abundance parameter times the fitted prevalence in baseline group;</p></li>
<li><p><strong>mean.LFC</strong>: log2-fold change in fitted mean between other group and baseline;</p></li>
<li><p><strong>base.abund</strong>: fitted mean abundance parameter in baseline group;</p></li>
<li><p><strong>abund.LFC</strong>: log2-fold change in fitted mean abundance parameter between other group and baseline;</p></li>
<li><p><strong>base.prev</strong>: fitted prevalence in baseline group;</p></li>
<li><p><strong>prev.change</strong>: (linear) difference in prevalence between baseline group and other group (other-baseline);</p></li>
<li><p><strong>base.disp</strong>: fitted dispersion parameter in baseline group;</p></li>
<li><p><strong>disp.LFC</strong>: log2-fold change in fitted dispersion parameter between other group and baseline;</p></li>
<li><p><strong>statistic</strong>: value of likelihood ratio test statistic;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>mbzinb</strong>.</p>
<details>
<summary>
<strong>other options for mbzinb</strong>
</summary>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="differential-analysis.html#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb248-2"><a href="differential-analysis.html#cb248-2" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb248-3"><a href="differential-analysis.html#cb248-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb248-4"><a href="differential-analysis.html#cb248-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb248-5"><a href="differential-analysis.html#cb248-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb248-6"><a href="differential-analysis.html#cb248-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb248-7"><a href="differential-analysis.html#cb248-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-8"><a href="differential-analysis.html#cb248-8" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb248-9"><a href="differential-analysis.html#cb248-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb248-10"><a href="differential-analysis.html#cb248-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb248-11"><a href="differential-analysis.html#cb248-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb248-12"><a href="differential-analysis.html#cb248-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb248-13"><a href="differential-analysis.html#cb248-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-14"><a href="differential-analysis.html#cb248-14" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb248-15"><a href="differential-analysis.html#cb248-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb248-16"><a href="differential-analysis.html#cb248-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb248-17"><a href="differential-analysis.html#cb248-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb248-18"><a href="differential-analysis.html#cb248-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-19"><a href="differential-analysis.html#cb248-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span>
<span id="cb248-20"><a href="differential-analysis.html#cb248-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="omnibus" class="section level3 hasAnchor" number="11.1.4">
<h3><span class="header-section-number">11.1.4</span> omnibus<a href="differential-analysis.html#omnibus" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This approach is also from <strong>mbzinb</strong> <span class="citation">(<a href="#ref-chen2018omnibus" role="doc-biblioref">J. Chen et al. 2018</a>)</span> package. it uses GMPR (Geometric Mean of Pairwise Ratios) <span class="citation">(<a href="#ref-chen2018gmpr" role="doc-biblioref">L. Chen et al. 2018</a>)</span> to get the size factors.</p>
<blockquote>
<p>where we specify models for count (abundance), zero (prevalence) and dispersion part. We also provide likelihood ratio test (zinb.lrt) for different models.</p>
</blockquote>
<p>(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="differential-analysis.html#cb249-1" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb249-2"><a href="differential-analysis.html#cb249-2" aria-hidden="true" tabindex="-1"></a>                     dada2_ps_genus_filter_trim,</span>
<span id="cb249-3"><a href="differential-analysis.html#cb249-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb249-4"><a href="differential-analysis.html#cb249-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb249-5"><a href="differential-analysis.html#cb249-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span></code></pre></div>
<pre><code>## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  87  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="differential-analysis.html#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;chi.stat&quot;                          &quot;df&quot;                                &quot;abund.baseline&quot;                   
## [10] &quot;prev.baseline&quot;                     &quot;dispersion.baseline&quot;               &quot;abund.LFC.CompvarBB.est&quot;          
## [13] &quot;abund.LFC.CompvarBB.se&quot;            &quot;prev.LOD.CompvarBB.est&quot;            &quot;prev.LOD.CompvarBB.se&quot;            
## [16] &quot;dispersion.LFC.CompvarBB.est&quot;      &quot;dispersion.LFC.CompvarBB.se&quot;       &quot;method&quot;                           
## [19] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [22] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [25] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [28] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="differential-analysis.html#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment     Pvalue AdjustedPvalue EffectSize  chi.stat df abund.baseline
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.26993990      0.6960000   70.56349 3.9226381  3   0.0051278458
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.14553831      0.5275764   39.65079 5.3872772  3   0.0006995458
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.06379234      0.4269179   23.17460 7.2692892  3   0.0003689644
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.12015581      0.5220000 1222.58730 5.8304723  3   0.0560610913
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.86180157      0.9186186  104.78571 0.7482458  3   0.0031679714
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.97964464      0.9800328   23.03175 0.1870987  3   0.0049913684
##   prev.baseline dispersion.baseline abund.LFC.CompvarBB.est abund.LFC.CompvarBB.se prev.LOD.CompvarBB.est
## 1     0.2912821           0.2037930             -0.37106147              1.4052050             -0.4101193
## 2     0.8894855           3.1449736              0.72928600              0.3066857             -0.7729345
## 3     0.3333678          11.4842094              1.28359134              0.3623742              0.6963880
## 4     0.5555645           1.6248619             -1.15961032              0.4774192              0.6971297
## 5     0.2351930           0.5224326             -0.85062762              1.7380445              2.3859635
## 6     0.7814974           1.0034163              0.01952004              0.4762646             -0.3538909
##   prev.LOD.CompvarBB.se dispersion.LFC.CompvarBB.est dispersion.LFC.CompvarBB.se  method Log2FoldChange (Median)\nAA_vs_BB
## 1             0.9810350                    2.8157726                   1.1350785 omnibus                                NA
## 2             1.2481748                   -0.5258551                   0.6760915 omnibus                        -0.9577718
## 3             0.8863818                   -1.8335068                   1.5022561 omnibus                                NA
## 4             0.8947782                   -0.5315454                   0.7007912 omnibus                         0.1524904
## 5             1.0103235                   -2.4754249                   1.0588367 omnibus                                NA
## 6             1.0006899                    0.1699865                   0.6267560 omnibus                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                       0                    0                  0.0                       1.7051442              58.82609
## 2                      30                   26                 50.5                      -1.3107676              50.91304
## 3                       0                    0                  9.0                      -2.4683647              19.21739
## 4                     316                  329                296.0                       1.7488297             996.26087
## 5                       0                    0                  0.0                      -2.1676332              93.78261
## 6                      64                   64                 69.0                       0.2000385             163.86957
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1         101.777778           31.21429                    21.74                 22.22                 21.43
## 2          26.777778           66.42857                    82.61                 88.89                 78.57
## 3           5.111111           28.28571                    43.48                 33.33                 50.00
## 4        1740.444444          517.85714                    65.22                 55.56                 71.43
## 5          30.000000          134.78571                    21.74                 22.22                 21.43
## 6         177.888889          154.85714                    73.91                 77.78                 71.43
##   Odds Ratio (95% CI)
## 1    0.67 (-0.11;1.5)
## 2       4.1 (6.8;1.3)
## 3        8.6 (13;4.3)
## 4     0.39 (-1.4;2.2)
## 5      1.5 (2.3;0.71)
## 6     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>omnibus</strong>;</p></li>
<li><p><strong>chi.stat</strong>: chisquare statistics;</p></li>
<li><p><strong>df</strong>: degree freedom;</p></li>
<li><p><strong>abund.baseline</strong>: mean abundance in baseline group;</p></li>
<li><p><strong>prev.baseline</strong>: prevalence in baseline group;</p></li>
<li><p><strong>dispersion.baseline</strong>: dispersion in baseline group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.est</strong>: log2-fold change in abundance between group BB and other group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.se</strong>: log2-fold change of standard errors in abundance between group BB and other group;</p></li>
<li><p><strong>prev.LOD.CompvarBB.est</strong>: prevalence of low of detect value in group BB;</p></li>
<li><p><strong>prev.LOD.CompvarBB.se</strong>: prevalenceâ€™s standard errors of low of detect value in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.est</strong>: log2-fold change in dispersion in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.se</strong>: log2-fold change in dispersionâ€™s standard errors in group BB;</p></li>
<li><p><strong>method</strong>: methods of test;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>omnibus</strong>.</p>
<details>
<summary>
<strong>other options for omnibus</strong>
</summary>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="differential-analysis.html#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb255-2"><a href="differential-analysis.html#cb255-2" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb255-3"><a href="differential-analysis.html#cb255-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb255-4"><a href="differential-analysis.html#cb255-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb255-5"><a href="differential-analysis.html#cb255-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb255-6"><a href="differential-analysis.html#cb255-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb255-7"><a href="differential-analysis.html#cb255-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb255-8"><a href="differential-analysis.html#cb255-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-9"><a href="differential-analysis.html#cb255-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-10"><a href="differential-analysis.html#cb255-10" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb255-11"><a href="differential-analysis.html#cb255-11" aria-hidden="true" tabindex="-1"></a>                     dada2_ps,</span>
<span id="cb255-12"><a href="differential-analysis.html#cb255-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb255-13"><a href="differential-analysis.html#cb255-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb255-14"><a href="differential-analysis.html#cb255-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb255-15"><a href="differential-analysis.html#cb255-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb255-16"><a href="differential-analysis.html#cb255-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-17"><a href="differential-analysis.html#cb255-17" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb255-18"><a href="differential-analysis.html#cb255-18" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb255-19"><a href="differential-analysis.html#cb255-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb255-20"><a href="differential-analysis.html#cb255-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb255-21"><a href="differential-analysis.html#cb255-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;omnibus&quot;</span>,</span>
<span id="cb255-22"><a href="differential-analysis.html#cb255-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb255-23"><a href="differential-analysis.html#cb255-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb255-24"><a href="differential-analysis.html#cb255-24" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span>
<span id="cb255-25"><a href="differential-analysis.html#cb255-25" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="raida" class="section level3 hasAnchor" number="11.1.5">
<h3><span class="header-section-number">11.1.5</span> RAIDA<a href="differential-analysis.html#raida" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>RAIDA</strong> package is from <em>A robust approach for identifying differentially abundant features in metagenomic samples</em> <span class="citation">(<a href="#ref-sohn2015robust" role="doc-biblioref">Sohn, Du, and An 2015</a>)</span>. It uses Ratio Approach for Identifying Differential Abundance (RAIDA). (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="differential-analysis.html#cb256-1" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb256-2"><a href="differential-analysis.html#cb256-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb256-3"><a href="differential-analysis.html#cb256-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb256-4"><a href="differential-analysis.html#cb256-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb256-5"><a href="differential-analysis.html#cb256-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb256-6"><a href="differential-analysis.html#cb256-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;eta_AA&quot;                            &quot;mean_AA&quot;                           &quot;sd_AA&quot;                            
## [10] &quot;eta_BB&quot;                            &quot;mean_BB&quot;                           &quot;sd_BB&quot;                            
## [13] &quot;mod.pool.var&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [16] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [19] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [22] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [25] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="differential-analysis.html#cb258-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##                             TaxaID         Block Enrichment EffectSize      Pvalue AdjustedPvalue    eta_AA   mean_AA
## 1                   g__Megasphaera 9_AA vs 14_BB  Nonsignif  409.19048 0.004581601      0.3848544 0.6666667  1.550194
## 2                    g__Romboutsia 9_AA vs 14_BB  Nonsignif  402.00794 0.025287679      0.7391453 0.4444444 -3.748832
## 3                 g__Lactobacillus 9_AA vs 14_BB  Nonsignif 1238.41270 0.033492279      0.7391453 0.2222222 -3.438359
## 4                    g__Tyzzerella 9_AA vs 14_BB  Nonsignif   75.78571 0.059752064      0.7391453 0.4444444 -2.455253
## 5 g__Christensenellaceae_R_7_group 9_AA vs 14_BB  Nonsignif   44.21429 0.068999409      0.7391453 0.6666667 -2.344046
## 6              g__Faecalibacterium 9_AA vs 14_BB  Nonsignif 2742.60317 0.077398760      0.7391453 0.3333333  1.247464
##       sd_AA     eta_BB    mean_BB    sd_BB mod.pool.var Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1 0.4452225 0.64285714 -1.2212162 1.074673     1.185998                                NA                       0
## 2 1.5814829 0.07142857 -1.4673045 2.085972     3.260003                         -7.036174                     107
## 3 1.7766726 0.07142857 -1.2509083 2.484244     4.276535                         -3.033423                      56
## 4 1.3348729 0.78571429 -5.1233546 3.191661     3.081402                                NA                       0
## 5 0.5621146 0.64285714 -4.4200897 1.890465     2.021487                                NA                       0
## 6 0.6483325 0.28571429  0.1846047 1.206605     1.221271                          2.400169                    1373
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1                    0                  0.0                        1.738259             335.26087          584.33333
## 2                    2                262.5                       -3.131606             296.47826           51.77778
## 3                   24                196.5                       -5.090571             791.26087           37.44444
## 4                    4                  0.0                        2.433624              46.86957           93.00000
## 5                    0                  0.0                        1.801549              35.08696           62.00000
## 6                 4537                859.5                        1.436918            2679.47826         4348.88889
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1          175.14286                    34.78                 33.33                 35.71          0.54 (-0.65;1.7)
## 2          453.78571                    78.26                 55.56                 92.86                37 (44;30)
## 3         1275.85714                    86.96                 77.78                 92.86 1.8e+08 (1.8e+08;1.8e+08)
## 4           17.21429                    34.78                 55.56                 21.43            0.4 (-1.4;2.2)
## 5           17.78571                    34.78                 33.33                 35.71          0.54 (-0.65;1.7)
## 6         1606.28571                    69.57                 66.67                 71.43           0.37 (-1.6;2.3)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>RAIDA</strong>;</p></li>
<li><p><strong>eta_AA</strong>: vector containing estimated probabilities of the false zero state for group AA;</p></li>
<li><p><strong>mean_AA</strong>: vector containing estimated means of log ratios for group AA;</p></li>
<li><p><strong>sd_AA</strong>: vector containing estimated standard deviations of log ratios for group AA;</p></li>
<li><p><strong>eta_BB</strong>: vector containing estimated probabilities of the false zero state for group BB;</p></li>
<li><p><strong>mean_BB</strong>: vector containing estimated means of log ratios for group BB;</p></li>
<li><p><strong>sd_BB</strong>: vector containing estimated standard deviations of log ratios for group BB;</p></li>
<li><p><strong>mod.pool.var</strong>: vector containing estimated posterior variances of log ratios;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>RAIDA</strong>.</p>
<details>
<summary>
<strong>other options for RAIDA</strong>
</summary>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="differential-analysis.html#cb260-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb260-2"><a href="differential-analysis.html#cb260-2" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb260-3"><a href="differential-analysis.html#cb260-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb260-4"><a href="differential-analysis.html#cb260-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb260-5"><a href="differential-analysis.html#cb260-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb260-6"><a href="differential-analysis.html#cb260-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb260-7"><a href="differential-analysis.html#cb260-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-8"><a href="differential-analysis.html#cb260-8" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb260-9"><a href="differential-analysis.html#cb260-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb260-10"><a href="differential-analysis.html#cb260-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb260-11"><a href="differential-analysis.html#cb260-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb260-12"><a href="differential-analysis.html#cb260-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb260-13"><a href="differential-analysis.html#cb260-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-14"><a href="differential-analysis.html#cb260-14" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb260-15"><a href="differential-analysis.html#cb260-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb260-16"><a href="differential-analysis.html#cb260-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb260-17"><a href="differential-analysis.html#cb260-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb260-18"><a href="differential-analysis.html#cb260-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;raida&quot;</span>)</span>
<span id="cb260-19"><a href="differential-analysis.html#cb260-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-20"><a href="differential-analysis.html#cb260-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span>
<span id="cb260-21"><a href="differential-analysis.html#cb260-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="wilcoxon-rank-sum-and-signed-rank-tests" class="section level3 hasAnchor" number="11.1.6">
<h3><span class="header-section-number">11.1.6</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparametric test methods, use the rank of taxa abundance to find the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="differential-analysis.html#cb261-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb261-2"><a href="differential-analysis.html#cb261-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb261-3"><a href="differential-analysis.html#cb261-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb261-4"><a href="differential-analysis.html#cb261-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb261-5"><a href="differential-analysis.html#cb261-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb261-6"><a href="differential-analysis.html#cb261-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="differential-analysis.html#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif   70.56349      63.5 1.0000000      1.0000000
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif   39.65079      44.5 0.2556616      0.7346565
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif   23.17460      44.0 0.1980169      0.7346565
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 1222.58730      71.0 0.6293971      0.9192573
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif  104.78571      64.5 0.9304707      0.9718249
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif   23.03175      67.0 0.8239942      0.9445787
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0                    0                  0.0
## 2                        -0.9577718                      30                   26                 50.5
## 3                                NA                       0                    0                  9.0
## 4                         0.1524904                     316                  329                296.0
## 5                                NA                       0                    0                  0.0
## 6                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA Mean Rank Abundance\nBB Occurrence (100%)\n(All)
## 1                      0.01201252                   12.06                   11.96                    21.74
## 2                     -0.42227633                    9.94                   13.32                    82.61
## 3                     -0.43387758                    9.89                   13.36                    43.48
## 4                      0.17342686                   12.89                   11.43                    65.22
## 5                      0.03358045                   12.17                   11.89                    21.74
## 6                      0.08724541                   12.44                   11.71                    73.91
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                 88.89                 78.57       4.1 (6.8;1.3)
## 3                 33.33                 50.00        8.6 (13;4.3)
## 4                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                 22.22                 21.43      1.5 (2.3;0.71)
## 6                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean Rank)_vs_BB**: Log2FoldChange (Mean Rank Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Rank Abundance (All)/(group AA)/(group BB)</strong>: Mean Rank Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for wilcox</strong>
</summary>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="differential-analysis.html#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb265-2"><a href="differential-analysis.html#cb265-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb265-3"><a href="differential-analysis.html#cb265-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb265-4"><a href="differential-analysis.html#cb265-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb265-5"><a href="differential-analysis.html#cb265-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb265-6"><a href="differential-analysis.html#cb265-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb265-7"><a href="differential-analysis.html#cb265-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-8"><a href="differential-analysis.html#cb265-8" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb265-9"><a href="differential-analysis.html#cb265-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb265-10"><a href="differential-analysis.html#cb265-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb265-11"><a href="differential-analysis.html#cb265-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb265-12"><a href="differential-analysis.html#cb265-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb265-13"><a href="differential-analysis.html#cb265-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-14"><a href="differential-analysis.html#cb265-14" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb265-15"><a href="differential-analysis.html#cb265-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb265-16"><a href="differential-analysis.html#cb265-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb265-17"><a href="differential-analysis.html#cb265-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb265-18"><a href="differential-analysis.html#cb265-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb265-19"><a href="differential-analysis.html#cb265-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-20"><a href="differential-analysis.html#cb265-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span>
<span id="cb265-21"><a href="differential-analysis.html#cb265-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="differential-analysis.html#cb266-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summary</span></span>
<span id="cb266-2"><a href="differential-analysis.html#cb266-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(dada2_ps_genus_filter_trim)</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 32359&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 49328&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 1008016&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 43826.7826086957&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 45596&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.501304347826087&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 0&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 1&quot;
## 
## [[11]]
## [1] &quot;Group&quot;</code></pre>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="differential-analysis.html#cb268-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run</span></span>
<span id="cb268-2"><a href="differential-analysis.html#cb268-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb268-3"><a href="differential-analysis.html#cb268-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb268-4"><a href="differential-analysis.html#cb268-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb268-5"><a href="differential-analysis.html#cb268-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb268-6"><a href="differential-analysis.html#cb268-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb268-7"><a href="differential-analysis.html#cb268-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_rarefy</strong>
</summary>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="differential-analysis.html#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb270-2"><a href="differential-analysis.html#cb270-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb270-3"><a href="differential-analysis.html#cb270-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb270-4"><a href="differential-analysis.html#cb270-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb270-5"><a href="differential-analysis.html#cb270-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb270-6"><a href="differential-analysis.html#cb270-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb270-7"><a href="differential-analysis.html#cb270-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb270-8"><a href="differential-analysis.html#cb270-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-9"><a href="differential-analysis.html#cb270-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_rarefy)</span>
<span id="cb270-10"><a href="differential-analysis.html#cb270-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_CLR</strong>: centered log-ratio normalization.</li>
</ul>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="differential-analysis.html#cb271-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb271-2"><a href="differential-analysis.html#cb271-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb271-3"><a href="differential-analysis.html#cb271-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb271-4"><a href="differential-analysis.html#cb271-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb271-5"><a href="differential-analysis.html#cb271-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb271-6"><a href="differential-analysis.html#cb271-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-7"><a href="differential-analysis.html#cb271-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_CLR)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_CLR</strong>
</summary>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="differential-analysis.html#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb273-2"><a href="differential-analysis.html#cb273-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb273-3"><a href="differential-analysis.html#cb273-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb273-4"><a href="differential-analysis.html#cb273-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb273-5"><a href="differential-analysis.html#cb273-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb273-6"><a href="differential-analysis.html#cb273-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb273-7"><a href="differential-analysis.html#cb273-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb273-8"><a href="differential-analysis.html#cb273-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-9"><a href="differential-analysis.html#cb273-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_CLR)</span>
<span id="cb273-10"><a href="differential-analysis.html#cb273-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse" class="section level3 hasAnchor" number="11.1.7">
<h3><span class="header-section-number">11.1.7</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="differential-analysis.html#cb274-1" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb274-2"><a href="differential-analysis.html#cb274-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb274-3"><a href="differential-analysis.html#cb274-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb274-4"><a href="differential-analysis.html#cb274-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb274-5"><a href="differential-analysis.html#cb274-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb274-6"><a href="differential-analysis.html#cb274-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;LDA_Score&quot;                         &quot;EffectSize&quot;                        &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
##  [7] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [10] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [13] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [16] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="differential-analysis.html#cb276-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB
## 1              g__Parasutterella 9_AA vs 14_BB         AA -3.704363   2.109795                          4.402050
## 2     g__Ruminococcaceae_UCG_004 9_AA vs 14_BB         AA -2.924042   2.030471                                NA
## 3                 g__Odoribacter 9_AA vs 14_BB         BB  1.951080   1.780728                                NA
## 4             g__Intestinibacter 9_AA vs 14_BB         BB  3.133350   2.358750                                NA
## 5 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB         BB  3.668859   2.661728                                NA
## 6                  g__Romboutsia 9_AA vs 14_BB         BB  3.705302   2.988350                         -6.856424
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1               172.77125           1080.35695             51.09966                        4.481148             3901.0977
## 2               241.24923           2451.90494              0.00000                        2.342928             1443.2092
## 3                21.93175            604.99989              0.00000                        2.099583              578.0758
## 4               383.50911              0.00000           1398.79740                       -1.964554             1883.3386
## 5               225.11921              0.00000           1173.88222                       -4.746617             5861.9051
## 6              2189.79596             50.35627           5835.02623                       -3.334490             7094.1306
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1          9320.3061           417.3209                    65.22                 88.89                 50.00
## 2          2822.7139           556.3847                    52.17                 77.78                 35.71
## 3          1083.9017           252.9020                    52.17                 77.78                 35.71
## 4           680.6440          2656.4994                    60.87                 22.22                 85.71
## 5           350.3379          9405.0554                    60.87                 22.22                 85.71
## 6          1086.1419         10956.4090                    78.26                 55.56                 92.86
##   Odds Ratio (95% CI)
## 1     0.0025 (-12;12)
## 2       0.14 (-3.7;4)
## 3       0.31 (-2;2.6)
## 4         3.4 (5.8;1)
## 5    4900 (5000;4900)
## 6          76 (85;68)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>lefse</em>;</p></li>
<li><p><strong>LDA_Score</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for lefse</strong>
</summary>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="differential-analysis.html#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb278-2"><a href="differential-analysis.html#cb278-2" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb278-3"><a href="differential-analysis.html#cb278-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb278-4"><a href="differential-analysis.html#cb278-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb278-5"><a href="differential-analysis.html#cb278-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb278-6"><a href="differential-analysis.html#cb278-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb278-7"><a href="differential-analysis.html#cb278-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb278-8"><a href="differential-analysis.html#cb278-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-9"><a href="differential-analysis.html#cb278-9" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb278-10"><a href="differential-analysis.html#cb278-10" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb278-11"><a href="differential-analysis.html#cb278-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb278-12"><a href="differential-analysis.html#cb278-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb278-13"><a href="differential-analysis.html#cb278-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb278-14"><a href="differential-analysis.html#cb278-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb278-15"><a href="differential-analysis.html#cb278-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-16"><a href="differential-analysis.html#cb278-16" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb278-17"><a href="differential-analysis.html#cb278-17" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb278-18"><a href="differential-analysis.html#cb278-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb278-19"><a href="differential-analysis.html#cb278-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb278-20"><a href="differential-analysis.html#cb278-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb278-21"><a href="differential-analysis.html#cb278-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb278-22"><a href="differential-analysis.html#cb278-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-23"><a href="differential-analysis.html#cb278-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span>
<span id="cb278-24"><a href="differential-analysis.html#cb278-24" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test" class="section level3 hasAnchor" number="11.1.8">
<h3><span class="header-section-number">11.1.8</span> t-test<a href="differential-analysis.html#t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="differential-analysis.html#cb279-1" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb279-2"><a href="differential-analysis.html#cb279-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb279-3"><a href="differential-analysis.html#cb279-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb279-4"><a href="differential-analysis.html#cb279-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb279-5"><a href="differential-analysis.html#cb279-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb279-6"><a href="differential-analysis.html#cb279-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                   
##  [3] &quot;Enrichment&quot;                               &quot;EffectSize&quot;                              
##  [5] &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;       
##  [9] &quot;Median Abundance\n(All)&quot;                  &quot;Median Abundance\nAA&quot;                    
## [11] &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;             
## [15] &quot;GeometricMean Abundance\nBB&quot;              &quot;Occurrence (100%)\n(All)&quot;                
## [17] &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="differential-analysis.html#cb281-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif   70.56349  0.6845254 0.51170983      0.7287988
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif   39.65079 -1.9079866 0.07478702      0.6684286
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif   23.17460 -2.1601919 0.04743155      0.6684286
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 1222.58730  1.4034718 0.19417159      0.6684286
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif  104.78571 -0.7632086 0.45788140      0.7184909
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif   23.03175  0.2691871 0.79125424      0.9131904
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0                    0                  0.0
## 2                        -0.9577718                      30                   26                 50.5
## 3                                NA                       0                    0                  9.0
## 4                         0.1524904                     316                  329                296.0
## 5                                NA                       0                    0                  0.0
## 6                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (geometricmean)\nAA_vs_BB GeometricMean Abundance\n(All) GeometricMean Abundance\nAA
## 1                                 2.674648                      1.2740132                   4.3847026
## 2                                -2.270362                      0.6530643                   0.1647934
## 3                                -3.268970                      0.9776732                   0.1479603
## 4                                 1.413058                      0.7929084                   1.0990412
## 5                                -3.452848                      1.3734381                   0.4150530
## 6                                -2.043760                      0.4449523                   0.2025430
##   GeometricMean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   0.6867376                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                   0.7950361                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3                   1.4262753                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4                   0.4127063                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                   4.5447977                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6                   0.8351231                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (GeometricMean)_vs_BB**: Log2FoldChange (GeometricMean Abundance) between group AA and group BB;</p></li>
<li><p><strong>GeometricMean Abundance (All)/(group AA)/(group BB)</strong>: GeometricMean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for t-test</strong>
</summary>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="differential-analysis.html#cb283-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb283-2"><a href="differential-analysis.html#cb283-2" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb283-3"><a href="differential-analysis.html#cb283-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb283-4"><a href="differential-analysis.html#cb283-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb283-5"><a href="differential-analysis.html#cb283-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-6"><a href="differential-analysis.html#cb283-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb283-7"><a href="differential-analysis.html#cb283-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-8"><a href="differential-analysis.html#cb283-8" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb283-9"><a href="differential-analysis.html#cb283-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb283-10"><a href="differential-analysis.html#cb283-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb283-11"><a href="differential-analysis.html#cb283-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-12"><a href="differential-analysis.html#cb283-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb283-13"><a href="differential-analysis.html#cb283-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-14"><a href="differential-analysis.html#cb283-14" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb283-15"><a href="differential-analysis.html#cb283-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb283-16"><a href="differential-analysis.html#cb283-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-17"><a href="differential-analysis.html#cb283-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb283-18"><a href="differential-analysis.html#cb283-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb283-19"><a href="differential-analysis.html#cb283-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-20"><a href="differential-analysis.html#cb283-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span>
<span id="cb283-21"><a href="differential-analysis.html#cb283-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>t-test_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="differential-analysis.html#cb284-1" aria-hidden="true" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb284-2"><a href="differential-analysis.html#cb284-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb284-3"><a href="differential-analysis.html#cb284-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb284-4"><a href="differential-analysis.html#cb284-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb284-5"><a href="differential-analysis.html#cb284-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb284-6"><a href="differential-analysis.html#cb284-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                   
##  [3] &quot;Enrichment&quot;                               &quot;EffectSize&quot;                              
##  [5] &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;       
##  [9] &quot;Median Abundance\n(All)&quot;                  &quot;Median Abundance\nAA&quot;                    
## [11] &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;             
## [15] &quot;GeometricMean Abundance\nBB&quot;              &quot;Occurrence (100%)\n(All)&quot;                
## [17] &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for ttest_rarefy</strong>
</summary>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="differential-analysis.html#cb286-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb286-2"><a href="differential-analysis.html#cb286-2" aria-hidden="true" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb286-3"><a href="differential-analysis.html#cb286-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb286-4"><a href="differential-analysis.html#cb286-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb286-5"><a href="differential-analysis.html#cb286-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb286-6"><a href="differential-analysis.html#cb286-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>,</span>
<span id="cb286-7"><a href="differential-analysis.html#cb286-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb286-8"><a href="differential-analysis.html#cb286-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-9"><a href="differential-analysis.html#cb286-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_rarefy)</span>
<span id="cb286-10"><a href="differential-analysis.html#cb286-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="metagenomeseq" class="section level3 hasAnchor" number="11.1.9">
<h3><span class="header-section-number">11.1.9</span> MetagenomeSeq<a href="differential-analysis.html#metagenomeseq" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>MetagenomeSeq</strong> package is from <em>Differential abundance analysis for microbial marker-gene surveys</em> <span class="citation">(<a href="#ref-paulson2013differential" role="doc-biblioref">Paulson et al. 2013</a>)</span>. It uses zero-inflated Log-Normal mixture model or Zero-inflated Gaussian mixture model to identify the significant taxa between groups. (Caution: the <strong>otu_table</strong> should be integers).</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="differential-analysis.html#cb287-1" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb287-2"><a href="differential-analysis.html#cb287-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb287-3"><a href="differential-analysis.html#cb287-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb287-4"><a href="differential-analysis.html#cb287-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb287-5"><a href="differential-analysis.html#cb287-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb287-6"><a href="differential-analysis.html#cb287-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb287-7"><a href="differential-analysis.html#cb287-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-8"><a href="differential-analysis.html#cb287-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;logFC&quot;                             &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;se&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="differential-analysis.html#cb289-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##                             TaxaID         Block Enrichment     logFC      Pvalue AdjustedPvalue        se
## 1                    g__Romboutsia 9_AA vs 14_BB  Nonsignif  1.593566 0.001706697      0.1604295 0.5079860
## 2                   g__Megasphaera 9_AA vs 14_BB  Nonsignif -1.732388 0.009444791      0.2982659 0.6674533
## 3               g__Intestinibacter 9_AA vs 14_BB  Nonsignif  1.205580 0.009519124      0.2982659 0.4649679
## 4   g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif  1.322139 0.026099811      0.6133456 0.5942938
## 5                   g__Haemophilus 9_AA vs 14_BB  Nonsignif -1.249150 0.078021778      0.8128851 0.7088254
## 6 g__Christensenellaceae_R_7_group 9_AA vs 14_BB  Nonsignif -1.319493 0.079359285      0.8128851 0.7520983
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                         -5.777348                98.07516             5.494505            301.35861
## 2                                NA                 0.00000             0.000000              0.00000
## 3                                NA                22.00489             0.000000             61.41378
## 4                                NA                10.08249             0.000000             42.61795
## 5                                NA                 0.00000             0.000000              0.00000
## 6                                NA                 0.00000             0.000000              0.00000
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       -4.038753             426.27423          41.005304          673.94712                    78.26
## 2                        3.202951             923.65546        2019.331511          219.29229                    34.78
## 3                       -3.378525             104.22978          15.506231          161.26635                    60.87
## 4                       -6.115136             357.04757           8.384505          581.18812                    60.87
## 5                        1.744289              89.30805         155.863624           46.52233                    34.78
## 6                        1.022398              27.97375          40.485129           19.93072                    34.78
##   Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                 55.56                 92.86          2100 (2200;2100)
## 2                 33.33                 35.71           0.29 (-2.1;2.7)
## 3                 22.22                 85.71                49 (57;42)
## 4                 22.22                 85.71 1.3e+11 (1.3e+11;1.3e+11)
## 5                 22.22                 42.86          0.63 (-0.28;1.5)
## 6                 33.33                 35.71         0.7 (-0.0096;1.4)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>se</strong>: standard error;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for metagenomeseq</strong>
</summary>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="differential-analysis.html#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb291-2"><a href="differential-analysis.html#cb291-2" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb291-3"><a href="differential-analysis.html#cb291-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb291-4"><a href="differential-analysis.html#cb291-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb291-5"><a href="differential-analysis.html#cb291-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb291-6"><a href="differential-analysis.html#cb291-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb291-7"><a href="differential-analysis.html#cb291-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb291-8"><a href="differential-analysis.html#cb291-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb291-9"><a href="differential-analysis.html#cb291-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb291-10"><a href="differential-analysis.html#cb291-10" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb291-11"><a href="differential-analysis.html#cb291-11" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb291-12"><a href="differential-analysis.html#cb291-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb291-13"><a href="differential-analysis.html#cb291-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb291-14"><a href="differential-analysis.html#cb291-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb291-15"><a href="differential-analysis.html#cb291-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb291-16"><a href="differential-analysis.html#cb291-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb291-17"><a href="differential-analysis.html#cb291-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb291-18"><a href="differential-analysis.html#cb291-18" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq<span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb291-19"><a href="differential-analysis.html#cb291-19" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb291-20"><a href="differential-analysis.html#cb291-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb291-21"><a href="differential-analysis.html#cb291-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb291-22"><a href="differential-analysis.html#cb291-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;metagenomeseq&quot;</span>,</span>
<span id="cb291-23"><a href="differential-analysis.html#cb291-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb291-24"><a href="differential-analysis.html#cb291-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb291-25"><a href="differential-analysis.html#cb291-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb291-26"><a href="differential-analysis.html#cb291-26" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span>
<span id="cb291-27"><a href="differential-analysis.html#cb291-27" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="deseq2" class="section level3 hasAnchor" number="11.1.10">
<h3><span class="header-section-number">11.1.10</span> DESeq2<a href="differential-analysis.html#deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>DESeq2</strong> package is from <em>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</em> <span class="citation">(<a href="#ref-love2014moderated" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution.(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="differential-analysis.html#cb292-1" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb292-2"><a href="differential-analysis.html#cb292-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb292-3"><a href="differential-analysis.html#cb292-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb292-4"><a href="differential-analysis.html#cb292-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb292-5"><a href="differential-analysis.html#cb292-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-6"><a href="differential-analysis.html#cb292-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;logFC&quot;                            
##  [7] &quot;Statistic&quot;                         &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="differential-analysis.html#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment      Pvalue AdjustedPvalue      logFC  Statistic
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.002388944     0.01759132  3.2744732  3.0370644
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.100346237     0.30103871  1.4768105  1.6431774
## 3   g__Adlercreutzia 9_AA vs 14_BB         BB 0.007237408     0.04509462  2.4566955  2.6857183
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.439215961     0.71505329 -0.8626086 -0.7735179
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.451454925             NA -0.4992262 -0.7529917
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.653088320     0.82656490 -0.4170110 -0.4494760
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0                    0                  0.0
## 2                        -0.9577718                      30                   26                 50.5
## 3                                NA                       0                    0                  9.0
## 4                         0.1524904                     316                  329                296.0
## 5                                NA                       0                    0                  0.0
## 6                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       1.7051442              58.82609         101.777778           31.21429                    21.74
## 2                      -1.3107676              50.91304          26.777778           66.42857                    82.61
## 3                      -2.4683647              19.21739           5.111111           28.28571                    43.48
## 4                       1.7488297             996.26087        1740.444444          517.85714                    65.22
## 5                      -2.1676332              93.78261          30.000000          134.78571                    21.74
## 6                       0.2000385             163.86957         177.888889          154.85714                    73.91
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                 88.89                 78.57       4.1 (6.8;1.3)
## 3                 33.33                 50.00        8.6 (13;4.3)
## 4                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                 22.22                 21.43      1.5 (2.3;0.71)
## 6                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>Statistic</strong>: test statistic (negative binomial model);</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for deseq2</strong>
</summary>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="differential-analysis.html#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb296-2"><a href="differential-analysis.html#cb296-2" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb296-3"><a href="differential-analysis.html#cb296-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb296-4"><a href="differential-analysis.html#cb296-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb296-5"><a href="differential-analysis.html#cb296-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb296-6"><a href="differential-analysis.html#cb296-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb296-7"><a href="differential-analysis.html#cb296-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-8"><a href="differential-analysis.html#cb296-8" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb296-9"><a href="differential-analysis.html#cb296-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb296-10"><a href="differential-analysis.html#cb296-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb296-11"><a href="differential-analysis.html#cb296-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb296-12"><a href="differential-analysis.html#cb296-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb296-13"><a href="differential-analysis.html#cb296-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-14"><a href="differential-analysis.html#cb296-14" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb296-15"><a href="differential-analysis.html#cb296-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb296-16"><a href="differential-analysis.html#cb296-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb296-17"><a href="differential-analysis.html#cb296-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb296-18"><a href="differential-analysis.html#cb296-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;deseq2&quot;</span>)</span>
<span id="cb296-19"><a href="differential-analysis.html#cb296-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-20"><a href="differential-analysis.html#cb296-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span>
<span id="cb296-21"><a href="differential-analysis.html#cb296-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="edger" class="section level3 hasAnchor" number="11.1.11">
<h3><span class="header-section-number">11.1.11</span> EdgeR<a href="differential-analysis.html#edger" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>EdgeR</strong> package is from <em>A scaling normalization method for differential expression analysis of RNA-seq data</em> <span class="citation">(<a href="#ref-robinson2010scaling" role="doc-biblioref">Robinson and Oshlack 2010</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="differential-analysis.html#cb297-1" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb297-2"><a href="differential-analysis.html#cb297-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb297-3"><a href="differential-analysis.html#cb297-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb297-4"><a href="differential-analysis.html#cb297-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-155"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-155-1.png" alt="EdgeR (BVC distance)" width="480" />
<p class="caption">
Figure 11.1: EdgeR (BVC distance)
</p>
</div>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="differential-analysis.html#cb298-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_edger)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjPval&quot;                           &quot;logFC&quot;                            
##  [7] &quot;logCPM&quot;                            &quot;LR&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
## [10] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [13] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [16] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [19] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="differential-analysis.html#cb300-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment       Pvalue     AdjPval     logFC    logCPM        LR
## 1 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB         BB 2.581898e-05 0.001842448  6.673441 13.327173 17.703216
## 2               g__Lactobacillus 9_AA vs 14_BB         BB 3.920102e-05 0.001842448  7.068223 16.214640 16.909687
## 3                  g__Romboutsia 9_AA vs 14_BB         BB 2.045466e-04 0.006148673  4.951091 13.656544 13.788850
## 4                 g__Eubacterium 9_AA vs 14_BB  Nonsignif 2.616456e-04 0.006148673 -3.816812  9.210384 13.326757
## 5              g__Parasutterella 9_AA vs 14_BB         AA 1.666532e-03 0.031076999 -3.225024 11.056077  9.884761
## 6             g__Intestinibacter 9_AA vs 14_BB         BB 1.983638e-03 0.031076999  3.628517 11.041294  9.564615
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                      11                    0                 39.5
## 2                         -3.033423                      56                   24                196.5
## 3                         -7.036174                     107                    2                262.5
## 4                                NA                       0                    0                  0.0
## 5                          4.321928                       7                   50                  2.5
## 6                                NA                      14                    0                 58.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       -4.884638            257.217391           14.00000         413.571429                    60.87
## 2                       -5.090571            791.260870           37.44444        1275.857143                    86.96
## 3                       -3.131606            296.478261           51.77778         453.785714                    78.26
## 4                        2.440393              9.173913           18.22222           3.357143                    30.43
## 5                        4.373725            167.565217          398.33333          19.214286                    65.22
## 6                       -2.157952             83.565217           26.88889         120.000000                    60.87
##   Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                 22.22                 85.71       11000 (11000;11000)
## 2                 77.78                 92.86 1.8e+08 (1.8e+08;1.8e+08)
## 3                 55.56                 92.86                37 (44;30)
## 4                 44.44                 21.43           0.34 (-1.8;2.5)
## 5                 88.89                 50.00         0.0066 (-9.8;9.9)
## 6                 22.22                 85.71               4.2 (7;1.4)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>logCPM</strong>: is the average expression of all samples for that particular gene across all samples on the log-scale expressed in counts per million (cpm, as calculated by edgeR after normalization);</p></li>
<li><p><strong>LR</strong>: the signed likelihood ratio test statistic;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for EdgeR</strong>
</summary>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="differential-analysis.html#cb302-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb302-2"><a href="differential-analysis.html#cb302-2" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb302-3"><a href="differential-analysis.html#cb302-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb302-4"><a href="differential-analysis.html#cb302-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb302-5"><a href="differential-analysis.html#cb302-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb302-6"><a href="differential-analysis.html#cb302-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb302-7"><a href="differential-analysis.html#cb302-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-8"><a href="differential-analysis.html#cb302-8" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb302-9"><a href="differential-analysis.html#cb302-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb302-10"><a href="differential-analysis.html#cb302-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb302-11"><a href="differential-analysis.html#cb302-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb302-12"><a href="differential-analysis.html#cb302-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb302-13"><a href="differential-analysis.html#cb302-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-14"><a href="differential-analysis.html#cb302-14" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb302-15"><a href="differential-analysis.html#cb302-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb302-16"><a href="differential-analysis.html#cb302-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb302-17"><a href="differential-analysis.html#cb302-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb302-18"><a href="differential-analysis.html#cb302-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;edger&quot;</span>)</span>
<span id="cb302-19"><a href="differential-analysis.html#cb302-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-20"><a href="differential-analysis.html#cb302-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span>
<span id="cb302-21"><a href="differential-analysis.html#cb302-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="ancom" class="section level3 hasAnchor" number="11.1.12">
<h3><span class="header-section-number">11.1.12</span> ANCOM<a href="differential-analysis.html#ancom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ANCOM</strong> (Analysis of composition of microbiomes) is from <em>Analysis of composition of microbiomes: a novel method for studying microbial compositionâ€, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-mandal2015analysis" role="doc-biblioref">Mandal et al. 2015</a>)</span>. ANCOM makes no distributional assumptions and can be implemented in a linear model framework. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="differential-analysis.html#cb303-1" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb303-2"><a href="differential-analysis.html#cb303-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb303-3"><a href="differential-analysis.html#cb303-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb303-4"><a href="differential-analysis.html#cb303-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-157"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-157-1.png" alt="ANCOM (Structure Zero)" width="480" />
<p class="caption">
Figure 11.2: ANCOM (Structure Zero)
</p>
</div>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="differential-analysis.html#cb304-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;(W)q-values &lt; alpha&quot;               &quot;W_ratio&quot;                          
##  [7] &quot;detected_0.7&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="differential-analysis.html#cb306-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment  EffectSize (W)q-values &lt; alpha W_ratio detected_0.7
## 1 g__Acidaminococcus 9_AA vs 14_BB         BB  0.01994016                   0       0        FALSE
## 2     g__Actinomyces 9_AA vs 14_BB         BB  0.68528127                   0       0        FALSE
## 3   g__Adlercreutzia 9_AA vs 14_BB         BB  0.27451133                   0       0        FALSE
## 4    g__Agathobacter 9_AA vs 14_BB         AA -0.52328657                   0       0        FALSE
## 5     g__Akkermansia 9_AA vs 14_BB         AA -0.08062438                   0       0        FALSE
## 6       g__Alistipes 9_AA vs 14_BB         AA -0.03107782                   0       0        FALSE
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0                    0                  0.0
## 2                        -0.9577718                      30                   26                 50.5
## 3                                NA                       0                    0                  9.0
## 4                         0.1524904                     316                  329                296.0
## 5                                NA                       0                    0                  0.0
## 6                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       1.7051442              58.82609         101.777778           31.21429                    21.74
## 2                      -1.3107676              50.91304          26.777778           66.42857                    82.61
## 3                      -2.4683647              19.21739           5.111111           28.28571                    43.48
## 4                       1.7488297             996.26087        1740.444444          517.85714                    65.22
## 5                      -2.1676332              93.78261          30.000000          134.78571                    21.74
## 6                       0.2000385             163.86957         177.888889          154.85714                    73.91
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                 88.89                 78.57       4.1 (6.8;1.3)
## 3                 33.33                 50.00        8.6 (13;4.3)
## 4                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                 22.22                 21.43      1.5 (2.3;0.71)
## 6                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groupsâ€™ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size by <strong>ANCOM</strong>;</p></li>
<li><p><strong>(W)q-values &lt; alpha</strong>: q-values less than alpha;</p></li>
<li><p><strong>W_ratio</strong>: the ratio of W values;</p></li>
<li><p><strong>detected_0.7</strong>: W_ratio more than 0.7;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for ANCOM</strong>
</summary>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="differential-analysis.html#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb308-2"><a href="differential-analysis.html#cb308-2" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb308-3"><a href="differential-analysis.html#cb308-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb308-4"><a href="differential-analysis.html#cb308-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb308-5"><a href="differential-analysis.html#cb308-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb308-6"><a href="differential-analysis.html#cb308-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb308-7"><a href="differential-analysis.html#cb308-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb308-8"><a href="differential-analysis.html#cb308-8" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb308-9"><a href="differential-analysis.html#cb308-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb308-10"><a href="differential-analysis.html#cb308-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb308-11"><a href="differential-analysis.html#cb308-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb308-12"><a href="differential-analysis.html#cb308-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb308-13"><a href="differential-analysis.html#cb308-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb308-14"><a href="differential-analysis.html#cb308-14" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb308-15"><a href="differential-analysis.html#cb308-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb308-16"><a href="differential-analysis.html#cb308-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb308-17"><a href="differential-analysis.html#cb308-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb308-18"><a href="differential-analysis.html#cb308-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ancom&quot;</span>)</span>
<span id="cb308-19"><a href="differential-analysis.html#cb308-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb308-20"><a href="differential-analysis.html#cb308-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb308-21"><a href="differential-analysis.html#cb308-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="corncob" class="section level3 hasAnchor" number="11.1.13">
<h3><span class="header-section-number">11.1.13</span> Corncob<a href="differential-analysis.html#corncob" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Corncob</strong> package is from <em>Modeling microbial abundances and dysbiosis with beta-binomial regressionâ€, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-martin2020modeling" role="doc-biblioref">Martin, Witten, and Willis 2020</a>)</span>. Corncob is based on beta-binomial regression. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="differential-analysis.html#cb309-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb309-2"><a href="differential-analysis.html#cb309-2" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb309-3"><a href="differential-analysis.html#cb309-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb309-4"><a href="differential-analysis.html#cb309-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb309-5"><a href="differential-analysis.html#cb309-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb309-6"><a href="differential-analysis.html#cb309-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb309-7"><a href="differential-analysis.html#cb309-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-8"><a href="differential-analysis.html#cb309-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span>
<span id="cb309-9"><a href="differential-analysis.html#cb309-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb309-10"><a href="differential-analysis.html#cb309-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>other options for Corncob</strong>
</summary>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="differential-analysis.html#cb310-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb310-2"><a href="differential-analysis.html#cb310-2" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb310-3"><a href="differential-analysis.html#cb310-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb310-4"><a href="differential-analysis.html#cb310-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb310-5"><a href="differential-analysis.html#cb310-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb310-6"><a href="differential-analysis.html#cb310-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb310-7"><a href="differential-analysis.html#cb310-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb310-8"><a href="differential-analysis.html#cb310-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-9"><a href="differential-analysis.html#cb310-9" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb310-10"><a href="differential-analysis.html#cb310-10" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb310-11"><a href="differential-analysis.html#cb310-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb310-12"><a href="differential-analysis.html#cb310-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb310-13"><a href="differential-analysis.html#cb310-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb310-14"><a href="differential-analysis.html#cb310-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb310-15"><a href="differential-analysis.html#cb310-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-16"><a href="differential-analysis.html#cb310-16" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb310-17"><a href="differential-analysis.html#cb310-17" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb310-18"><a href="differential-analysis.html#cb310-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb310-19"><a href="differential-analysis.html#cb310-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb310-20"><a href="differential-analysis.html#cb310-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;corncob&quot;</span>,</span>
<span id="cb310-21"><a href="differential-analysis.html#cb310-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb310-22"><a href="differential-analysis.html#cb310-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-23"><a href="differential-analysis.html#cb310-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_corncob)</span>
<span id="cb310-24"><a href="differential-analysis.html#cb310-24" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models" class="section level3 hasAnchor" number="11.1.14">
<h3><span class="header-section-number">11.1.14</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="differential-analysis.html#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb311-2"><a href="differential-analysis.html#cb311-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb311-3"><a href="differential-analysis.html#cb311-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb311-4"><a href="differential-analysis.html#cb311-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb311-5"><a href="differential-analysis.html#cb311-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb311-6"><a href="differential-analysis.html#cb311-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb311-7"><a href="differential-analysis.html#cb311-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb311-8"><a href="differential-analysis.html#cb311-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb311-9"><a href="differential-analysis.html#cb311-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb311-10"><a href="differential-analysis.html#cb311-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-11"><a href="differential-analysis.html#cb311-11" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb311-12"><a href="differential-analysis.html#cb311-12" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb311-13"><a href="differential-analysis.html#cb311-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb311-14"><a href="differential-analysis.html#cb311-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb311-15"><a href="differential-analysis.html#cb311-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb311-16"><a href="differential-analysis.html#cb311-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb311-17"><a href="differential-analysis.html#cb311-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb311-18"><a href="differential-analysis.html#cb311-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb311-19"><a href="differential-analysis.html#cb311-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb311-20"><a href="differential-analysis.html#cb311-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-21"><a href="differential-analysis.html#cb311-21" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb311-22"><a href="differential-analysis.html#cb311-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb311-23"><a href="differential-analysis.html#cb311-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb311-24"><a href="differential-analysis.html#cb311-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb311-25"><a href="differential-analysis.html#cb311-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb311-26"><a href="differential-analysis.html#cb311-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb311-27"><a href="differential-analysis.html#cb311-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb311-28"><a href="differential-analysis.html#cb311-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb311-29"><a href="differential-analysis.html#cb311-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb311-30"><a href="differential-analysis.html#cb311-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-31"><a href="differential-analysis.html#cb311-31" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb311-32"><a href="differential-analysis.html#cb311-32" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb311-33"><a href="differential-analysis.html#cb311-33" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb311-34"><a href="differential-analysis.html#cb311-34" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb311-35"><a href="differential-analysis.html#cb311-35" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb311-36"><a href="differential-analysis.html#cb311-36" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb311-37"><a href="differential-analysis.html#cb311-37" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb311-38"><a href="differential-analysis.html#cb311-38" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb311-39"><a href="differential-analysis.html#cb311-39" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb311-40"><a href="differential-analysis.html#cb311-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-41"><a href="differential-analysis.html#cb311-41" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2)</span>
<span id="cb311-42"><a href="differential-analysis.html#cb311-42" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
<div id="metagenomic-sequencing-microbial-data-metaphlan23" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Metagenomic sequencing microbial data (metaphlan2/3)<a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same strategy to filter phyloseq object.</p>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="differential-analysis.html#cb312-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;metaphlan2_ps&quot;</span>)</span>
<span id="cb312-2"><a href="differential-analysis.html#cb312-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-3"><a href="differential-analysis.html#cb312-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb312-4"><a href="differential-analysis.html#cb312-4" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb312-5"><a href="differential-analysis.html#cb312-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb312-6"><a href="differential-analysis.html#cb312-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb312-7"><a href="differential-analysis.html#cb312-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb312-8"><a href="differential-analysis.html#cb312-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-9"><a href="differential-analysis.html#cb312-9" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb312-10"><a href="differential-analysis.html#cb312-10" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps_remove_BRS, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb312-11"><a href="differential-analysis.html#cb312-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-12"><a href="differential-analysis.html#cb312-12" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb312-13"><a href="differential-analysis.html#cb312-13" aria-hidden="true" tabindex="-1"></a><span class="co">#dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(dada2_ps_genus, cutoff = 10, unclass = TRUE)</span></span>
<span id="cb312-14"><a href="differential-analysis.html#cb312-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-15"><a href="differential-analysis.html#cb312-15" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb312-16"><a href="differential-analysis.html#cb312-16" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(metaphlan2_ps_genus, <span class="at">cutoff =</span> <span class="fl">1e-4</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb312-17"><a href="differential-analysis.html#cb312-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-18"><a href="differential-analysis.html#cb312-18" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb312-19"><a href="differential-analysis.html#cb312-19" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(metaphlan2_ps_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb312-20"><a href="differential-analysis.html#cb312-20" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 54 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 54 taxa by 6 taxonomic ranks ]</code></pre>
<div id="limma_voom-1" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> limma_voom<a href="differential-analysis.html#limma_voom-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>limma</strong> package is from <em>voom: Precision weights unlock linear model analysis tools for RNA-seq read counts</em> <span class="citation">(<a href="#ref-law2014voom" role="doc-biblioref">Law et al. 2014</a>)</span>. Firstly, transforming count data to log2-counts per million (logCPM), estimate the mean-variance relationship and use this to compute appropriate observation-level weights. Secondly, fitting multiple linear models by weighted or generalized least squares. Finally, performing empirical bayes statistics for differential expression.</p>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="differential-analysis.html#cb314-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_mgs <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb314-2"><a href="differential-analysis.html#cb314-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb314-3"><a href="differential-analysis.html#cb314-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb314-4"><a href="differential-analysis.html#cb314-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb314-5"><a href="differential-analysis.html#cb314-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="differential-analysis.html#cb316-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom_mgs)</span></code></pre></div>
<pre><code>##                TaxaID         Block Enrichment EffectSize     logFC       Pvalue AdjustedPvalue
## 1     g__Ruminococcus 7_AA vs 15_BB         BB  0.2233696 0.2233696 1.548665e-07   3.894463e-06
## 2 g__Faecalibacterium 7_AA vs 15_BB         BB  0.3063850 0.3063850 1.854506e-07   3.894463e-06
## 3          g__Blautia 7_AA vs 15_BB         BB  0.1619875 0.1619875 1.187496e-05   1.662495e-04
## 4    g__Lactobacillus 7_AA vs 15_BB         BB  0.1399117 0.1399117 5.287895e-05   5.552289e-04
## 5  g__Bifidobacterium 7_AA vs 15_BB         BB  0.3779862 0.3779862 9.644339e-05   8.101244e-04
## 6        g__Roseburia 7_AA vs 15_BB         BB  0.1275366 0.1275366 1.728128e-04   1.209690e-03
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                         -2.906486              0.02141570            0.0044236            0.0331677
## 2                         -4.011562              0.05346880            0.0047826            0.0771373
## 3                         -3.182422              0.02000425            0.0036769            0.0333800
## 4                                NA              0.00155690            0.0000000            0.0024249
## 5                         -6.549698              0.03731980            0.0009090            0.0851568
## 6                         -4.442582              0.00752330            0.0008393            0.0182502
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       -2.892349            0.03578485       0.0066508000         0.04938074                    81.82
## 2                       -3.992131            0.05750352       0.0051489857         0.08193563                    77.27
## 3                       -2.265551            0.02431008       0.0067591429         0.03250052                   100.00
## 4                       -5.399318            0.02118822       0.0007282714         0.03073619                    72.73
## 5                       -3.394708            0.10730063       0.0143274857         0.15068810                   100.00
## 6                       -2.451452            0.02013462       0.0049745857         0.02720931                    81.82
##   Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                 57.14                 93.33             270 (280;260)
## 2                 57.14                 86.67                75 (83;66)
## 3                100.00                100.00                90 (99;81)
## 4                 42.86                 86.67 5.3e+10 (5.3e+10;5.3e+10)
## 5                100.00                100.00             130 (140;120)
## 6                 57.14                 93.33               12 (17;7.2)</code></pre>
<details>
<summary>
<strong>limma_voom Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="differential-analysis.html#cb318-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb318-2"><a href="differential-analysis.html#cb318-2" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb318-3"><a href="differential-analysis.html#cb318-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb318-4"><a href="differential-analysis.html#cb318-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb318-5"><a href="differential-analysis.html#cb318-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb318-6"><a href="differential-analysis.html#cb318-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;limma_voom&quot;</span>)</span>
<span id="cb318-7"><a href="differential-analysis.html#cb318-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="wilcoxon-rank-sum-and-signed-rank-tests-1" class="section level3 hasAnchor" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparameter test methods, use the rank of taxa abundance to find the significant taxa.</p>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="differential-analysis.html#cb319-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb319-2"><a href="differential-analysis.html#cb319-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb319-3"><a href="differential-analysis.html#cb319-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb319-4"><a href="differential-analysis.html#cb319-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb319-5"><a href="differential-analysis.html#cb319-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="differential-analysis.html#cb321-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic      Pvalue AdjustedPvalue
## 1       g__Alistipes 7_AA vs 15_BB  Nonsignif  1.7499838      71.0 0.201934650     0.53007846
## 2    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.2384009      31.0 0.138345605     0.38736769
## 3   g__Anaerotruncus 7_AA vs 15_BB  Nonsignif  0.9439864      47.5 0.712887079     0.83170159
## 4     g__Bacteroides 7_AA vs 15_BB  Nonsignif  1.1135266      85.0 0.021284830     0.07947442
## 5 g__Bifidobacterium 7_AA vs 15_BB         BB  4.2421190      11.0 0.002122619     0.04028636
## 6       g__Bilophila 7_AA vs 15_BB  Nonsignif  0.2514869      60.0 0.616070968     0.77874024
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                          2.061604              0.00711780            0.0236665            0.0056693
## 2                         -4.964322              0.00031435            0.0000229            0.0007149
## 3                                NA              0.00000000            0.0000000            0.0000000
## 4                          1.229057              0.24894650            0.4568864            0.1949060
## 5                         -6.549698              0.03731980            0.0009090            0.0851568
## 6                          1.177523              0.00104585            0.0020513            0.0009069
##   Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA Mean Rank Abundance\nBB Occurrence (100%)\n(All)
## 1                       0.4613459                   14.14                   10.27                    77.27
## 2                      -0.6171177                    8.43                   12.93                    86.36
## 3                      -0.1327552                   10.79                   11.83                    36.36
## 4                       0.7906916                   16.14                    9.33                   100.00
## 5                      -1.3572361                    5.57                   14.27                   100.00
## 6                       0.1924811                   12.57                   11.00                    68.18
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 71.43                 80.00     0.22 (-2.7;3.2)
## 2                 71.43                 93.33         6 (9.5;2.5)
## 3                 28.57                 40.00      1.3 (1.8;0.78)
## 4                100.00                100.00       0.31 (-2;2.6)
## 5                100.00                100.00       130 (140;120)
## 6                 71.43                 66.67      0.4 (-1.4;2.2)</code></pre>
<details>
<summary>
<strong>wilcox Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="differential-analysis.html#cb323-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb323-2"><a href="differential-analysis.html#cb323-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb323-3"><a href="differential-analysis.html#cb323-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb323-4"><a href="differential-analysis.html#cb323-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb323-5"><a href="differential-analysis.html#cb323-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb323-6"><a href="differential-analysis.html#cb323-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb323-7"><a href="differential-analysis.html#cb323-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-8"><a href="differential-analysis.html#cb323-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-1" class="section level3 hasAnchor" number="11.2.3">
<h3><span class="header-section-number">11.2.3</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="differential-analysis.html#cb324-1" aria-hidden="true" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb324-2"><a href="differential-analysis.html#cb324-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb324-3"><a href="differential-analysis.html#cb324-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb324-4"><a href="differential-analysis.html#cb324-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb324-5"><a href="differential-analysis.html#cb324-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)  </span>
<span id="cb324-6"><a href="differential-analysis.html#cb324-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb324-7"><a href="differential-analysis.html#cb324-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;LDA_Score&quot;                         &quot;EffectSize&quot;                        &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
##  [7] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [10] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [13] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [16] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="differential-analysis.html#cb326-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##              TaxaID         Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1    g__Bacteroides 7_AA vs 15_BB         AA -5.020331  2.4235602                          1.190750            254167.50879
## 2  g__Fusobacterium 7_AA vs 15_BB         AA -4.268159  1.9028644                                NA                 0.00000
## 3 g__Granulicatella 7_AA vs 15_BB         BB  2.826238  0.4271128                                NA                25.66255
## 4    g__Eggerthella 7_AA vs 15_BB         BB  3.221472  1.2650206                                NA               591.34240
## 5    g__Collinsella 7_AA vs 15_BB         BB  3.914103  2.4170225                                NA              7735.89039
## 6          g__Dorea 7_AA vs 15_BB         BB  3.917984  2.0751421                         -5.725323              8554.29279
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1          464614.8499         203536.16836                        1.141368           291673.9273       464989.41139
## 2             123.8325              0.00000                        6.941519             9891.9128        30556.16386
## 3               0.0000             52.35042                       -3.071984               64.1451           10.59936
## 4               0.0000            771.47121                       -3.764280             2231.9872          232.91482
## 5               0.0000          11641.72307                       -2.928116            12495.9911         2268.85954
## 6             270.5150          14311.50707                       -3.454344            11339.4984         1455.31524
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1       210793.36812                   100.00                100.00                100.00     0.32 (-1.9;2.5)
## 2          248.59562                    40.91                 85.71                 20.00     0.0032 (-11;11)
## 3           89.13311                    72.73                 42.86                 86.67       220 (230;210)
## 4         3164.88766                    63.64                 42.86                 73.33       430 (450;420)
## 5        17268.65252                    68.18                 42.86                 80.00          20 (25;14)
## 6        15952.11728                    90.91                 71.43                100.00          28 (34;21)</code></pre>
<details>
<summary>
<strong>lefse Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="differential-analysis.html#cb328-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb328-2"><a href="differential-analysis.html#cb328-2" aria-hidden="true" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb328-3"><a href="differential-analysis.html#cb328-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb328-4"><a href="differential-analysis.html#cb328-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb328-5"><a href="differential-analysis.html#cb328-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb328-6"><a href="differential-analysis.html#cb328-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb328-7"><a href="differential-analysis.html#cb328-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb328-8"><a href="differential-analysis.html#cb328-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test-1" class="section level3 hasAnchor" number="11.2.4">
<h3><span class="header-section-number">11.2.4</span> t-test<a href="differential-analysis.html#t-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="differential-analysis.html#cb329-1" aria-hidden="true" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb329-2"><a href="differential-analysis.html#cb329-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb329-3"><a href="differential-analysis.html#cb329-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb329-4"><a href="differential-analysis.html#cb329-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb329-5"><a href="differential-analysis.html#cb329-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                   
##  [3] &quot;Enrichment&quot;                               &quot;EffectSize&quot;                              
##  [5] &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;       
##  [9] &quot;Median Abundance\n(All)&quot;                  &quot;Median Abundance\nAA&quot;                    
## [11] &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;             
## [15] &quot;GeometricMean Abundance\nBB&quot;              &quot;Occurrence (100%)\n(All)&quot;                
## [17] &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="differential-analysis.html#cb331-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue
## 1       g__Alistipes 7_AA vs 15_BB  Nonsignif  1.7499838  1.5963681 0.15951558      0.4674316
## 2    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.2384009 -1.6970563 0.10937430      0.3828101
## 3   g__Anaerotruncus 7_AA vs 15_BB  Nonsignif  0.9439864 -0.5871152 0.56369794      0.6398733
## 4     g__Bacteroides 7_AA vs 15_BB  Nonsignif  1.1135266  2.2882319 0.04545820      0.1909244
## 5 g__Bifidobacterium 7_AA vs 15_BB  Nonsignif  4.2421190 -2.7252006 0.01523564      0.1089140
## 6       g__Bilophila 7_AA vs 15_BB  Nonsignif  0.2514869  1.2030139 0.27277490      0.5565431
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                          2.061604              0.00711780            0.0236665            0.0056693
## 2                         -4.964322              0.00031435            0.0000229            0.0007149
## 3                                NA              0.00000000            0.0000000            0.0000000
## 4                          1.229057              0.24894650            0.4568864            0.1949060
## 5                         -6.549698              0.03731980            0.0009090            0.0851568
## 6                          1.177523              0.00104585            0.0020513            0.0009069
##   Log2FoldChange (geometricmean)\nAA_vs_BB GeometricMean Abundance\n(All) GeometricMean Abundance\nAA
## 1                               2.05528762                      0.3989446                   0.7346341
## 2                                       NA                      1.8882218                         NaN
## 3                              -0.05926392                      1.0212861                   0.9903011
## 4                               1.60721398                      0.3871167                   0.6044631
## 5                                       NA                      0.1936930                         NaN
## 6                               3.17465522                      0.6988408                   2.6168456
##   GeometricMean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   0.1767534                    77.27                 71.43                 80.00     0.22 (-2.7;3.2)
## 2                   1.8882218                    86.36                 71.43                 93.33         6 (9.5;2.5)
## 3                   1.0318284                    36.36                 28.57                 40.00      1.3 (1.8;0.78)
## 4                   0.1984039                   100.00                100.00                100.00       0.31 (-2;2.6)
## 5                   0.1936930                   100.00                100.00                100.00       130 (140;120)
## 6                   0.2898089                    68.18                 71.43                 66.67      0.4 (-1.4;2.2)</code></pre>
<details>
<summary>
<strong>ttest Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="differential-analysis.html#cb333-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb333-2"><a href="differential-analysis.html#cb333-2" aria-hidden="true" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb333-3"><a href="differential-analysis.html#cb333-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb333-4"><a href="differential-analysis.html#cb333-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb333-5"><a href="differential-analysis.html#cb333-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb333-6"><a href="differential-analysis.html#cb333-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb333-7"><a href="differential-analysis.html#cb333-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models-1" class="section level3 hasAnchor" number="11.2.5">
<h3><span class="header-section-number">11.2.5</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="differential-analysis.html#cb334-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb334-2"><a href="differential-analysis.html#cb334-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb334-3"><a href="differential-analysis.html#cb334-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb334-4"><a href="differential-analysis.html#cb334-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb334-5"><a href="differential-analysis.html#cb334-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb334-6"><a href="differential-analysis.html#cb334-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb334-7"><a href="differential-analysis.html#cb334-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb334-8"><a href="differential-analysis.html#cb334-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb334-9"><a href="differential-analysis.html#cb334-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb334-10"><a href="differential-analysis.html#cb334-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb334-11"><a href="differential-analysis.html#cb334-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2_mgs)</span>
<span id="cb334-12"><a href="differential-analysis.html#cb334-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>maaslin2 Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="differential-analysis.html#cb335-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb335-2"><a href="differential-analysis.html#cb335-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb335-3"><a href="differential-analysis.html#cb335-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb335-4"><a href="differential-analysis.html#cb335-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb335-5"><a href="differential-analysis.html#cb335-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb335-6"><a href="differential-analysis.html#cb335-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb335-7"><a href="differential-analysis.html#cb335-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb335-8"><a href="differential-analysis.html#cb335-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb335-9"><a href="differential-analysis.html#cb335-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb335-10"><a href="differential-analysis.html#cb335-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb335-11"><a href="differential-analysis.html#cb335-11" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
</div>
<div id="visualization-4" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Visualization<a href="differential-analysis.html#visualization-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Volcano plot is used to display differential analysis. <code>plot_volcano</code> provides multiple parameters for plotting volcano. More details to see <code>help(plot_volcano)</code>.</p>
<p>The barplot is used to display the lefse results.</p>
<div id="volcano-plot" class="section level3 hasAnchor" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Volcano plot<a href="differential-analysis.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The X and Y coordinate axis are flexible to choose for Volcano. Here, we choose <em>logFC</em> and <em>AdjustedPvalue</em> to visualize the results</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="differential-analysis.html#cb336-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb336-2"><a href="differential-analysis.html#cb336-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb336-3"><a href="differential-analysis.html#cb336-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb336-4"><a href="differential-analysis.html#cb336-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb336-5"><a href="differential-analysis.html#cb336-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb336-6"><a href="differential-analysis.html#cb336-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="differential-analysis.html#cb338-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_volcano <span class="ot">&lt;-</span> <span class="fu">plot_volcano</span>(</span>
<span id="cb338-2"><a href="differential-analysis.html#cb338-2" aria-hidden="true" tabindex="-1"></a>    DA_limma_voom,</span>
<span id="cb338-3"><a href="differential-analysis.html#cb338-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb338-4"><a href="differential-analysis.html#cb338-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;logFC&quot;</span>,</span>
<span id="cb338-5"><a href="differential-analysis.html#cb338-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb338-6"><a href="differential-analysis.html#cb338-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb338-7"><a href="differential-analysis.html#cb338-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb338-8"><a href="differential-analysis.html#cb338-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb338-9"><a href="differential-analysis.html#cb338-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span>
<span id="cb338-10"><a href="differential-analysis.html#cb338-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb338-11"><a href="differential-analysis.html#cb338-11" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_volcano</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-173"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-173-1.png" alt="Volcano (limma-voom: logFC and AdjustedPvalue)" width="672" />
<p class="caption">
Figure 11.3: Volcano (limma-voom: logFC and AdjustedPvalue)
</p>
</div>
<details>
<summary>
<strong>DA_limma_voom_volcano effectsize</strong>
</summary>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="differential-analysis.html#cb339-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb339-2"><a href="differential-analysis.html#cb339-2" aria-hidden="true" tabindex="-1"></a>    DA_limma_voom,</span>
<span id="cb339-3"><a href="differential-analysis.html#cb339-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb339-4"><a href="differential-analysis.html#cb339-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;EffectSize&quot;</span>,</span>
<span id="cb339-5"><a href="differential-analysis.html#cb339-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb339-6"><a href="differential-analysis.html#cb339-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb339-7"><a href="differential-analysis.html#cb339-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb339-8"><a href="differential-analysis.html#cb339-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb339-9"><a href="differential-analysis.html#cb339-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">8</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-174"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-174-1.png" alt="Volcano (limma-voom: EffectSize and Pvalue)" width="672" />
<p class="caption">
Figure 11.4: Volcano (limma-voom: EffectSize and Pvalue)
</p>
</div>
<p>The logFC and EffectSize are the same values in limma-voom.</p>
</details>
</div>
<div id="barplot-in-lefse" class="section level3 hasAnchor" number="11.3.2">
<h3><span class="header-section-number">11.3.2</span> barplot in lefse<a href="differential-analysis.html#barplot-in-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="differential-analysis.html#cb340-1" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb340-2"><a href="differential-analysis.html#cb340-2" aria-hidden="true" tabindex="-1"></a>    dada2_ps_genus_filter_trim,</span>
<span id="cb340-3"><a href="differential-analysis.html#cb340-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb340-4"><a href="differential-analysis.html#cb340-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb340-5"><a href="differential-analysis.html#cb340-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb340-6"><a href="differential-analysis.html#cb340-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb340-7"><a href="differential-analysis.html#cb340-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb340-8"><a href="differential-analysis.html#cb340-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">da_res =</span> DA_lefse,</span>
<span id="cb340-9"><a href="differential-analysis.html#cb340-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb340-10"><a href="differential-analysis.html#cb340-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb340-11"><a href="differential-analysis.html#cb340-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>)) </span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-175"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-175-1.png" alt="Barplot (Lefse)" width="576" />
<p class="caption">
Figure 11.5: Barplot (Lefse)
</p>
</div>
</div>
</div>
<div id="dominant-taxa" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Dominant taxa<a href="differential-analysis.html#dominant-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Display the significant taxa with selection using boxplot.</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="differential-analysis.html#cb341-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb341-2"><a href="differential-analysis.html#cb341-2" aria-hidden="true" tabindex="-1"></a>   metaphlan2_ps_genus_filter_trim,</span>
<span id="cb341-3"><a href="differential-analysis.html#cb341-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb341-4"><a href="differential-analysis.html#cb341-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb341-5"><a href="differential-analysis.html#cb341-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb341-6"><a href="differential-analysis.html#cb341-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_topN_boxplot</span>(</span>
<span id="cb341-7"><a href="differential-analysis.html#cb341-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb341-8"><a href="differential-analysis.html#cb341-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> DA_wilcox,</span>
<span id="cb341-9"><a href="differential-analysis.html#cb341-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Median)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb341-10"><a href="differential-analysis.html#cb341-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.2</span>,</span>
<span id="cb341-11"><a href="differential-analysis.html#cb341-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb341-12"><a href="differential-analysis.html#cb341-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.3</span>,</span>
<span id="cb341-13"><a href="differential-analysis.html#cb341-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">3</span>,</span>
<span id="cb341-14"><a href="differential-analysis.html#cb341-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">taxa_name =</span> <span class="st">&quot;s__Ruminococcus_torques&quot;</span>,</span>
<span id="cb341-15"><a href="differential-analysis.html#cb341-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-176"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-176-1.png" alt="Dominant Taxa" width="768" />
<p class="caption">
Figure 11.6: Dominant Taxa
</p>
</div>
</div>
<div id="multiple-differential-analysis-by-one-function" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> Multiple differential analysis by one function<a href="differential-analysis.html#multiple-differential-analysis-by-one-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>here, we provide the <code>run_multiple_da</code> for obtaining the results list from multiple differential analysis methods.</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="differential-analysis.html#cb342-1" aria-hidden="true" tabindex="-1"></a>multiple_res <span class="ot">&lt;-</span> <span class="fu">run_multiple_da</span>(</span>
<span id="cb342-2"><a href="differential-analysis.html#cb342-2" aria-hidden="true" tabindex="-1"></a>       dada2_ps_genus_filter_trim,</span>
<span id="cb342-3"><a href="differential-analysis.html#cb342-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb342-4"><a href="differential-analysis.html#cb342-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb342-5"><a href="differential-analysis.html#cb342-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">da_method =</span> <span class="fu">c</span>(<span class="st">&quot;aldex&quot;</span>, <span class="st">&quot;limma_voom&quot;</span>, <span class="st">&quot;mbzinb&quot;</span>, <span class="st">&quot;omnibus&quot;</span>),</span>
<span id="cb342-6"><a href="differential-analysis.html#cb342-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">p_adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|
## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  87  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="differential-analysis.html#cb344-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(multiple_res)</span></code></pre></div>
<pre><code>## [1] &quot;aldex&quot;      &quot;limma_voom&quot; &quot;mbzinb&quot;     &quot;omnibus&quot;</code></pre>
<ul>
<li>plot results</li>
</ul>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="differential-analysis.html#cb346-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_multiple_DA</span>(</span>
<span id="cb346-2"><a href="differential-analysis.html#cb346-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Multip_DA_res =</span> multiple_res,</span>
<span id="cb346-3"><a href="differential-analysis.html#cb346-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_list =</span> <span class="fu">c</span>(<span class="st">&quot;EffectSize&quot;</span>, <span class="st">&quot;logFC&quot;</span>, <span class="st">&quot;mean.LFC&quot;</span>, <span class="st">&quot;abund.LFC.CompvarBB.est&quot;</span>),</span>
<span id="cb346-4"><a href="differential-analysis.html#cb346-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb346-5"><a href="differential-analysis.html#cb346-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_index =</span> <span class="st">&quot;AdjustedPvalue&quot;</span>,</span>
<span id="cb346-6"><a href="differential-analysis.html#cb346-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb346-7"><a href="differential-analysis.html#cb346-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellwidth =</span> <span class="dv">50</span>,</span>
<span id="cb346-8"><a href="differential-analysis.html#cb346-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellheight =</span> <span class="dv">10</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-178"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-178-1.png" alt="Multiple DA results" width="768" />
<p class="caption">
Figure 11.7: Multiple DA results
</p>
</div>
</div>
<div id="systematic-information-8" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> Systematic Information<a href="differential-analysis.html#systematic-information-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="differential-analysis.html#cb347-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] RColorBrewer_1.1-2          viridis_0.6.2               viridisLite_0.4.0           picante_1.8.2              
##  [5] nlme_3.1-155                vegan_2.5-7                 lattice_0.20-45             permute_0.9-7              
##  [9] ape_5.6-2                   conflicted_1.1.0            SummarizedExperiment_1.24.0 Biobase_2.54.0             
## [13] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1         IRanges_2.28.0              S4Vectors_0.32.3           
## [17] BiocGenerics_0.40.0         MatrixGenerics_1.6.0        matrixStats_0.61.0          ggpubr_0.4.0               
## [21] ggplot2_3.3.5               phyloseq_1.38.0             tibble_3.1.6                dplyr_1.0.8                
## [25] XMAS2_2.1.4                
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2             reticulate_1.24        tidyselect_1.1.2       RSQLite_2.2.10         AnnotationDbi_1.56.2  
##   [6] htmlwidgets_1.5.4      grid_4.1.2             BiocParallel_1.28.3    Rtsne_0.15             munsell_0.5.0         
##  [11] units_0.8-0            codetools_0.2-18       DT_0.21                umap_0.2.8.0           withr_2.5.0           
##  [16] colorspace_2.0-3       highr_0.9              knitr_1.37             rstudioapi_0.13        pscl_1.5.5            
##  [21] robustbase_0.93-9      bayesm_3.1-4           ggsignif_0.6.3         labeling_0.4.2         mbzinb_0.2            
##  [26] GenomeInfoDbData_1.2.7 pheatmap_1.0.12        bit64_4.0.5            farver_2.1.0           rhdf5_2.38.1          
##  [31] vctrs_0.3.8            generics_0.1.2         TH.data_1.1-0          xfun_0.30              R6_2.5.1              
##  [36] RVenn_1.1.0            locfit_1.5-9.5         RcppZiggurat_0.1.6     bitops_1.0-7           rhdf5filters_1.6.0    
##  [41] microbiome_1.16.0      cachem_1.0.6           DelayedArray_0.20.0    assertthat_0.2.1       scales_1.1.1          
##  [46] multcomp_1.4-18        nnet_7.3-17            gtable_0.3.0           sandwich_3.0-1         rlang_1.0.2           
##  [51] genefilter_1.76.0      splines_4.1.2          protoclust_1.6.3       rstatix_0.7.0          broom_0.7.12          
##  [56] checkmate_2.0.0        yaml_2.3.5             reshape2_1.4.4         abind_1.4-5            crosstalk_1.2.0       
##  [61] backports_1.4.1        Rfast_2.0.6            qvalue_2.26.0          Hmisc_4.6-0            DiagrammeR_1.0.9      
##  [66] tensorA_0.36.2         tools_4.1.2            bookdown_0.24          zCompositions_1.4.0    ellipsis_0.3.2        
##  [71] gplots_3.1.1           jquerylib_0.1.4        biomformat_1.22.0      proxy_0.4-26           Rcpp_1.0.8.2          
##  [76] plyr_1.8.6             visNetwork_2.1.0       base64enc_0.1-3        zlibbioc_1.40.0        classInt_0.4-3        
##  [81] purrr_0.3.4            RCurl_1.98-1.6         rpart_4.1.16           openssl_2.0.0          Wrench_1.12.0         
##  [86] cowplot_1.1.1          zoo_1.8-9              ggrepel_0.9.1          cluster_2.1.2          magrittr_2.0.2        
##  [91] data.table_1.14.2      RSpectra_0.16-0        truncnorm_1.0-8        mvtnorm_1.1-3          ALDEx2_1.26.0         
##  [96] RAIDA_1.0              evaluate_0.15          xtable_1.8-4           XML_3.99-0.9           jpeg_0.1-9            
## [101] gridExtra_2.3          shape_1.4.6            compiler_4.1.2         KernSmooth_2.23-20     crayon_1.5.0          
## [106] htmltools_0.5.2        mgcv_1.8-39            ggVennDiagram_1.2.1    Formula_1.2-4          tidyr_1.2.0           
## [111] geneplotter_1.72.0     libcoin_1.0-9          DBI_1.1.2              corrplot_0.92          MASS_7.3-55           
## [116] sf_1.0-7               compositions_2.0-4     Matrix_1.4-0           ade4_1.7-18            car_3.0-12            
## [121] cli_3.2.0              parallel_4.1.2         igraph_1.2.11          pkgconfig_2.0.3        coin_1.4-2            
## [126] foreign_0.8-82         foreach_1.5.2          annotate_1.72.0        bslib_0.3.1            multtest_2.50.0       
## [131] XVector_0.34.0         NADA_1.6-1.1           stringr_1.4.0          digest_0.6.29          Biostrings_2.62.0     
## [136] rmarkdown_2.13         htmlTable_2.4.0        edgeR_3.36.0           gtools_3.9.2           modeltools_0.2-23     
## [141] lifecycle_1.0.1        jsonlite_1.8.0         Rhdf5lib_1.16.0        carData_3.0-5          askpass_1.1           
## [146] limma_3.50.1           fansi_1.0.2            pillar_1.7.0           ggsci_2.9              DEoptimR_1.0-10       
## [151] KEGGREST_1.34.0        fastmap_1.1.0          httr_1.4.2             survival_3.3-1         glue_1.6.2            
## [156] png_0.1-7              iterators_1.0.14       glmnet_4.1-3           bit_4.0.4              class_7.3-20          
## [161] stringi_1.7.6          sass_0.4.0             metagenomeSeq_1.36.0   blob_1.2.2             DESeq2_1.34.0         
## [166] latticeExtra_0.6-29    caTools_1.18.2         memoise_2.0.1          e1071_1.7-9</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chen2018omnibus" class="csl-entry">
Chen, Jun, Emily King, Rebecca Deek, Zhi Wei, Yue Yu, Diane Grill, and Karla Ballman. 2018. <span>â€œAn Omnibus Test for Differential Distribution Analysis of Microbiome Sequencing Data.â€</span> <em>Bioinformatics</em> 34 (4): 643â€“51.
</div>
<div id="ref-chen2018gmpr" class="csl-entry">
Chen, Li, James Reeve, Lujun Zhang, Shengbing Huang, Xuefeng Wang, and Jun Chen. 2018. <span>â€œGMPR: A Robust Normalization Method for Zero-Inflated Count Data with Application to Microbiome Sequencing Data.â€</span> <em>PeerJ</em> 6: e4600.
</div>
<div id="ref-fernandes2014unifying" class="csl-entry">
Fernandes, Andrew D, Jennifer Ns Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>â€œUnifying the Analysis of High-Throughput Sequencing Datasets: Characterizing RNA-Seq, 16s rRNA Gene Sequencing and Selective Growth Experiments by Compositional Data Analysis.â€</span> <em>Microbiome</em> 2 (1): 1â€“13.
</div>
<div id="ref-law2014voom" class="csl-entry">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. <span>â€œVoom: Precision Weights Unlock Linear Model Analysis Tools for RNA-Seq Read Counts.â€</span> <em>Genome Biology</em> 15 (2): 1â€“17.
</div>
<div id="ref-love2014moderated" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>â€œModerated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.â€</span> <em>Genome Biology</em> 15 (12): 1â€“21.
</div>
<div id="ref-mallick2021multivariable" class="csl-entry">
Mallick, Himel, Ali Rahnavard, Lauren J McIver, Siyuan Ma, Yancong Zhang, Long H Nguyen, Timothy L Tickle, et al. 2021. <span>â€œMultivariable Association Discovery in Population-Scale Meta-Omics Studies.â€</span> <em>PLoS Computational Biology</em> 17 (11): e1009442.
</div>
<div id="ref-mandal2015analysis" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A White, Merete EggesbÃ¸, Rob Knight, and Shyamal D Peddada. 2015. <span>â€œAnalysis of Composition of Microbiomes: A Novel Method for Studying Microbial Composition.â€</span> <em>Microbial Ecology in Health and Disease</em> 26 (1): 27663.
</div>
<div id="ref-martin2020modeling" class="csl-entry">
Martin, Bryan D, Daniela Witten, and Amy D Willis. 2020. <span>â€œModeling Microbial Abundances and Dysbiosis with Beta-Binomial Regression.â€</span> <em>The Annals of Applied Statistics</em> 14 (1): 94.
</div>
<div id="ref-nearing2022microbiome" class="csl-entry">
Nearing, Jacob T, Gavin M Douglas, Molly G Hayes, Jocelyn MacDonald, Dhwani K Desai, Nicole Allward, Casey Jones, et al. 2022. <span>â€œMicrobiome Differential Abundance Methods Produce Different Results Across 38 Datasets.â€</span> <em>Nature Communications</em> 13 (1): 1â€“16.
</div>
<div id="ref-paulson2013differential" class="csl-entry">
Paulson, Joseph N, O Colin Stine, HÃ©ctor Corrada Bravo, and Mihai Pop. 2013. <span>â€œDifferential Abundance Analysis for Microbial Marker-Gene Surveys.â€</span> <em>Nature Methods</em> 10 (12): 1200â€“1202.
</div>
<div id="ref-robinson2010scaling" class="csl-entry">
Robinson, Mark D, and Alicia Oshlack. 2010. <span>â€œA Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data.â€</span> <em>Genome Biology</em> 11 (3): 1â€“9.
</div>
<div id="ref-segata2011metagenomic" class="csl-entry">
Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. <span>â€œMetagenomic Biomarker Discovery and Explanation.â€</span> <em>Genome Biology</em> 12 (6): 1â€“18.
</div>
<div id="ref-sohn2015robust" class="csl-entry">
Sohn, Michael B, Ruofei Du, and Lingling An. 2015. <span>â€œA Robust Approach for Identifying Differentially Abundant Features in Metagenomic Samples.â€</span> <em>Bioinformatics</em> 31 (14): 2269â€“75.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="core-microbiota.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualization-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/10-differentail-analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
