<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Differential Analysis | XMAS 2.0 Tutorial</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Differential Analysis | XMAS 2.0 Tutorial" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Differential Analysis | XMAS 2.0 Tutorial" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-08-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="core-microbiota.html"/>
<link rel="next" href="visualization-5.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing</a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers</a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book</a></li>
<li class="chapter" data-level="2.6" data-path="introduction.html"><a href="introduction.html#input-data"><i class="fa fa-check"></i><b>2.6</b> Input data</a></li>
<li class="chapter" data-level="2.7" data-path="introduction.html"><a href="introduction.html#key-publications"><i class="fa fa-check"></i><b>2.7</b> Key publications</a></li>
<li class="chapter" data-level="2.8" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>2.8</b> Citation</a></li>
<li class="chapter" data-level="2.9" data-path="introduction.html"><a href="introduction.html#change-log"><i class="fa fa-check"></i><b>2.9</b> Change log</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="introduction.html"><a href="introduction.html#xmas-2.1.7"><i class="fa fa-check"></i><b>2.9.1</b> XMAS 2.1.7</a></li>
<li class="chapter" data-level="2.9.2" data-path="introduction.html"><a href="introduction.html#xmas-2.1.6"><i class="fa fa-check"></i><b>2.9.2</b> XMAS 2.1.6</a></li>
<li class="chapter" data-level="2.9.3" data-path="introduction.html"><a href="introduction.html#xmas-2.1.5"><i class="fa fa-check"></i><b>2.9.3</b> XMAS 2.1.5</a></li>
<li class="chapter" data-level="2.9.4" data-path="introduction.html"><a href="introduction.html#xmas-2.1.4"><i class="fa fa-check"></i><b>2.9.4</b> XMAS 2.1.4</a></li>
<li class="chapter" data-level="2.9.5" data-path="introduction.html"><a href="introduction.html#xmas-2.1.3"><i class="fa fa-check"></i><b>2.9.5</b> XMAS 2.1.3</a></li>
<li class="chapter" data-level="2.9.6" data-path="introduction.html"><a href="introduction.html#xmas-2.1.2"><i class="fa fa-check"></i><b>2.9.6</b> XMAS 2.1.2</a></li>
<li class="chapter" data-level="2.9.7" data-path="introduction.html"><a href="introduction.html#xmas-2.1.1"><i class="fa fa-check"></i><b>2.9.7</b> XMAS 2.1.1</a></li>
<li class="chapter" data-level="2.9.8" data-path="introduction.html"><a href="introduction.html#xmas-2.1.0"><i class="fa fa-check"></i><b>2.9.8</b> XMAS 2.1.0</a></li>
<li class="chapter" data-level="2.9.9" data-path="introduction.html"><a href="introduction.html#xmas-2.0.13"><i class="fa fa-check"></i><b>2.9.9</b> XMAS 2.0.13</a></li>
<li class="chapter" data-level="2.9.10" data-path="introduction.html"><a href="introduction.html#xmas-2.0.12"><i class="fa fa-check"></i><b>2.9.10</b> XMAS 2.0.12</a></li>
<li class="chapter" data-level="2.9.11" data-path="introduction.html"><a href="introduction.html#xmas-2.0.7-2022-04-12"><i class="fa fa-check"></i><b>2.9.11</b> XMAS 2.0.7 (2022-04-12)</a></li>
<li class="chapter" data-level="2.9.12" data-path="introduction.html"><a href="introduction.html#xmas-2.0.2-2022-03-17"><i class="fa fa-check"></i><b>2.9.12</b> XMAS 2.0.2 (2022-03-17)</a></li>
<li class="chapter" data-level="2.9.13" data-path="introduction.html"><a href="introduction.html#xmas-2.0.1-2022-03-14"><i class="fa fa-check"></i><b>2.9.13</b> XMAS 2.0.1 (2022-03-14)</a></li>
<li class="chapter" data-level="2.9.14" data-path="introduction.html"><a href="introduction.html#xmas-1.0.0-2021-12-13"><i class="fa fa-check"></i><b>2.9.14</b> XMAS 1.0.0 (2021-12-13)</a></li>
<li class="chapter" data-level="2.9.15" data-path="introduction.html"><a href="introduction.html#initial-repository-0.0.1-2021-07-25"><i class="fa fa-check"></i><b>2.9.15</b> Initial repository 0.0.1 (2021-07-25)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial</a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2</a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description</a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages</a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation</a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette</a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors</a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan</a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>4.2.1</b> Importing results from dada2 pipeline</a></li>
<li class="chapter" data-level="4.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>4.2.2</b> taxa table</a></li>
<li class="chapter" data-level="4.2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>4.2.3</b> otu table</a></li>
<li class="chapter" data-level="4.2.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>4.2.4</b> metadata table</a></li>
<li class="chapter" data-level="4.2.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>4.2.5</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>4.3.1</b> Importing results</a></li>
<li class="chapter" data-level="4.3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table-1"><i class="fa fa-check"></i><b>4.3.2</b> taxa table</a></li>
<li class="chapter" data-level="4.3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table-1"><i class="fa fa-check"></i><b>4.3.3</b> otu table</a></li>
<li class="chapter" data-level="4.3.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table-1"><i class="fa fa-check"></i><b>4.3.4</b> metadata table</a></li>
<li class="chapter" data-level="4.3.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object-1"><i class="fa fa-check"></i><b>4.3.5</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Reads’ track</a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS</a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves</a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object</a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>5.6</b> Variability</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>5.6.1</b> Coefficient of variation</a></li>
<li class="chapter" data-level="5.6.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>5.6.2</b> Distribution of number of sequences</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation</a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization</a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering</a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming</a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation</a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object</a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others</a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>6.8</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)</a></li>
<li class="chapter" data-level="6.9" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>7.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="7.2.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance</a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#evenness"><i class="fa fa-check"></i><b>7.4</b> Evenness</a></li>
<li class="chapter" data-level="7.5" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS</a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac</a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition</a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)</a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>8.6</b> Mantel Test</a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.7</b> Analysis of Similarity (ANOSIM)</a></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>8.8</b> Multi-response permutation procedures (MRPP)</a></li>
<li class="chapter" data-level="8.9" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.9</b> Ordination</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.9.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="8.9.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.9.2</b> Principal Coordinate Analysis (PCoA)</a></li>
<li class="chapter" data-level="8.9.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.9.3</b> t-distributed stochastic neighbor embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.9.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.9.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>9</b> Microbial Composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbial-composition.html"><a href="microbial-composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object</a></li>
<li class="chapter" data-level="9.2" data-path="microbial-composition.html"><a href="microbial-composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.2</b> XVIZ <code>plot_stacked_bar</code></a></li>
<li class="chapter" data-level="9.3" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.3</b> Stacked barplot</a></li>
<li class="chapter" data-level="9.4" data-path="microbial-composition.html"><a href="microbial-composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps</a></li>
<li class="chapter" data-level="9.5" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota</a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap</a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test</a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq</a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2</a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR</a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM</a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob</a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
<li class="chapter" data-level="11.1.15" data-path="differential-analysis.html"><a href="differential-analysis.html#locom"><i class="fa fa-check"></i><b>11.1.15</b> LOCOM</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.1</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.3</b> t-test</a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.4</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot</a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa</a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#comparing-outputs-from-xmas2-lefse-conda-and-lefse-galaxy-using-the-same-in-house-datasets-amplicon_pszeybel_gut"><i class="fa fa-check"></i><b>11.6</b> Comparing outputs from XMAS2, lefse-conda, and lefse-galaxy using the same in-house datasets (amplicon_ps|Zeybel_Gut)</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="differential-analysis.html"><a href="differential-analysis.html#introduction-1"><i class="fa fa-check"></i><b>11.6.1</b> Introduction</a></li>
<li class="chapter" data-level="11.6.2" data-path="differential-analysis.html"><a href="differential-analysis.html#dataset"><i class="fa fa-check"></i><b>11.6.2</b> Dataset</a></li>
<li class="chapter" data-level="11.6.3" data-path="differential-analysis.html"><a href="differential-analysis.html#run-lefse-independently-with-the-three-applications-r-conda-galaxy"><i class="fa fa-check"></i><b>11.6.3</b> Run lefse independently with the three applications (R, conda, galaxy)</a></li>
<li class="chapter" data-level="11.6.4" data-path="differential-analysis.html"><a href="differential-analysis.html#comparison-of-lefse-conda-with-xmas2"><i class="fa fa-check"></i><b>11.6.4</b> Comparison of lefse-conda with XMAS2</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization</a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot</a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot</a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot</a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot</a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density</a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination</a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot</a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot</a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn</a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot"><i class="fa fa-check"></i><b>12.10</b> plot the DA results from the significant taxa by double barplot</a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.11</b> plot_stacked_bar_XIVZ</a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.12</b> plot_StackBarPlot</a></li>
<li class="chapter" data-level="12.13" data-path="visualization-5.html"><a href="visualization-5.html#color-palettes"><i class="fa fa-check"></i><b>12.13</b> Color Palettes</a>
<ul>
<li class="chapter" data-level="12.13.1" data-path="visualization-5.html"><a href="visualization-5.html#wes-anderson-palettes"><i class="fa fa-check"></i><b>12.13.1</b> Wes Anderson Palettes</a></li>
</ul></li>
<li class="chapter" data-level="12.14" data-path="visualization-5.html"><a href="visualization-5.html#ordination-plots-with-ggplot2"><i class="fa fa-check"></i><b>12.14</b> Ordination plots with ggplot2</a>
<ul>
<li class="chapter" data-level="12.14.1" data-path="visualization-5.html"><a href="visualization-5.html#principal-components-analysis-with-the-iris-data-set"><i class="fa fa-check"></i><b>12.14.1</b> principal components analysis with the iris data set</a></li>
<li class="chapter" data-level="12.14.2" data-path="visualization-5.html"><a href="visualization-5.html#multiple-correspondence-analysis-with-the-tea-dataset"><i class="fa fa-check"></i><b>12.14.2</b> multiple correspondence analysis with the tea dataset</a></li>
<li class="chapter" data-level="12.14.3" data-path="visualization-5.html"><a href="visualization-5.html#linear-discriminant-analysis"><i class="fa fa-check"></i><b>12.14.3</b> linear discriminant analysis</a></li>
</ul></li>
<li class="chapter" data-level="12.15" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.15</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>13</b> Association Analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="association-analysis.html"><a href="association-analysis.html#loading-and-showing-phyloseq-object"><i class="fa fa-check"></i><b>13.1</b> Loading and showing phyloseq object</a></li>
<li class="chapter" data-level="13.2" data-path="association-analysis.html"><a href="association-analysis.html#spearman-correlation-analysis"><i class="fa fa-check"></i><b>13.2</b> Spearman Correlation Analysis</a></li>
<li class="chapter" data-level="13.3" data-path="association-analysis.html"><a href="association-analysis.html#partial-correlation-analysis"><i class="fa fa-check"></i><b>13.3</b> Partial Correlation Analysis</a></li>
<li class="chapter" data-level="13.4" data-path="association-analysis.html"><a href="association-analysis.html#systematic-information-10"><i class="fa fa-check"></i><b>13.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>14</b> Principals of Differential Analysis methods</a>
<ul>
<li class="chapter" data-level="14.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>14.1</b> ALDEx2</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-1"><i class="fa fa-check"></i><b>14.1.1</b> Vignette</a></li>
<li class="chapter" data-level="14.1.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type"><i class="fa fa-check"></i><b>14.1.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.1.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments"><i class="fa fa-check"></i><b>14.1.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.1.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures"><i class="fa fa-check"></i><b>14.1.4</b> Procedures</a></li>
<li class="chapter" data-level="14.1.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output"><i class="fa fa-check"></i><b>14.1.5</b> Key output</a></li>
<li class="chapter" data-level="14.1.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax"><i class="fa fa-check"></i><b>14.1.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.1.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax"><i class="fa fa-check"></i><b>14.1.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.1.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes"><i class="fa fa-check"></i><b>14.1.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>14.2</b> ANCOM-II</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-1"><i class="fa fa-check"></i><b>14.2.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.2.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-1"><i class="fa fa-check"></i><b>14.2.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.2.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-1"><i class="fa fa-check"></i><b>14.2.3</b> Procedures</a></li>
<li class="chapter" data-level="14.2.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-1"><i class="fa fa-check"></i><b>14.2.4</b> Key output</a></li>
<li class="chapter" data-level="14.2.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-1"><i class="fa fa-check"></i><b>14.2.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.2.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-1"><i class="fa fa-check"></i><b>14.2.6</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.2.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-1"><i class="fa fa-check"></i><b>14.2.7</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>14.3</b> corncob</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-2"><i class="fa fa-check"></i><b>14.3.1</b> Vignette</a></li>
<li class="chapter" data-level="14.3.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-2"><i class="fa fa-check"></i><b>14.3.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.3.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-2"><i class="fa fa-check"></i><b>14.3.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.3.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-2"><i class="fa fa-check"></i><b>14.3.4</b> Procedures</a></li>
<li class="chapter" data-level="14.3.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-2"><i class="fa fa-check"></i><b>14.3.5</b> Key output</a></li>
<li class="chapter" data-level="14.3.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-2"><i class="fa fa-check"></i><b>14.3.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.3.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-2"><i class="fa fa-check"></i><b>14.3.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.3.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-2"><i class="fa fa-check"></i><b>14.3.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>14.4</b> DESeq2</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-3"><i class="fa fa-check"></i><b>14.4.1</b> Vignette</a></li>
<li class="chapter" data-level="14.4.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-3"><i class="fa fa-check"></i><b>14.4.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.4.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-3"><i class="fa fa-check"></i><b>14.4.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.4.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-3"><i class="fa fa-check"></i><b>14.4.4</b> Procedures</a></li>
<li class="chapter" data-level="14.4.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-3"><i class="fa fa-check"></i><b>14.4.5</b> Key output</a></li>
<li class="chapter" data-level="14.4.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-3"><i class="fa fa-check"></i><b>14.4.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.4.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-3"><i class="fa fa-check"></i><b>14.4.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>14.5</b> edgeR</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-4"><i class="fa fa-check"></i><b>14.5.1</b> Vignette</a></li>
<li class="chapter" data-level="14.5.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-4"><i class="fa fa-check"></i><b>14.5.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.5.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-4"><i class="fa fa-check"></i><b>14.5.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.5.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-4"><i class="fa fa-check"></i><b>14.5.4</b> Procedures</a></li>
<li class="chapter" data-level="14.5.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-4"><i class="fa fa-check"></i><b>14.5.5</b> Key output</a></li>
<li class="chapter" data-level="14.5.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-4"><i class="fa fa-check"></i><b>14.5.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.5.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-4"><i class="fa fa-check"></i><b>14.5.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.5.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-3"><i class="fa fa-check"></i><b>14.5.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>14.6</b> LEfSe</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-5"><i class="fa fa-check"></i><b>14.6.1</b> Vignette</a></li>
<li class="chapter" data-level="14.6.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-5"><i class="fa fa-check"></i><b>14.6.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.6.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-5"><i class="fa fa-check"></i><b>14.6.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.6.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-5"><i class="fa fa-check"></i><b>14.6.4</b> Procedures</a></li>
<li class="chapter" data-level="14.6.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-5"><i class="fa fa-check"></i><b>14.6.5</b> Key output</a></li>
<li class="chapter" data-level="14.6.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-5"><i class="fa fa-check"></i><b>14.6.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.6.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-5"><i class="fa fa-check"></i><b>14.6.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.6.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-4"><i class="fa fa-check"></i><b>14.6.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>14.7</b> limma voom</a>
<ul>
<li class="chapter" data-level="14.7.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-6"><i class="fa fa-check"></i><b>14.7.1</b> Vignette</a></li>
<li class="chapter" data-level="14.7.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-6"><i class="fa fa-check"></i><b>14.7.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.7.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-6"><i class="fa fa-check"></i><b>14.7.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.7.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-6"><i class="fa fa-check"></i><b>14.7.4</b> Procedures</a></li>
<li class="chapter" data-level="14.7.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-6"><i class="fa fa-check"></i><b>14.7.5</b> Key output</a></li>
<li class="chapter" data-level="14.7.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-6"><i class="fa fa-check"></i><b>14.7.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.7.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-6"><i class="fa fa-check"></i><b>14.7.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.7.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-5"><i class="fa fa-check"></i><b>14.7.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>14.8</b> MaAsLin2</a>
<ul>
<li class="chapter" data-level="14.8.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-7"><i class="fa fa-check"></i><b>14.8.1</b> Vignette</a></li>
<li class="chapter" data-level="14.8.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-7"><i class="fa fa-check"></i><b>14.8.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.8.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-7"><i class="fa fa-check"></i><b>14.8.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.8.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-7"><i class="fa fa-check"></i><b>14.8.4</b> Procedures</a></li>
<li class="chapter" data-level="14.8.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-7"><i class="fa fa-check"></i><b>14.8.5</b> Key output</a></li>
<li class="chapter" data-level="14.8.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-7"><i class="fa fa-check"></i><b>14.8.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.8.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-7"><i class="fa fa-check"></i><b>14.8.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>14.9</b> metagenomeSeq</a>
<ul>
<li class="chapter" data-level="14.9.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-8"><i class="fa fa-check"></i><b>14.9.1</b> Vignette</a></li>
<li class="chapter" data-level="14.9.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-8"><i class="fa fa-check"></i><b>14.9.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.9.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-8"><i class="fa fa-check"></i><b>14.9.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.9.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-8"><i class="fa fa-check"></i><b>14.9.4</b> Procedures</a></li>
<li class="chapter" data-level="14.9.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-8"><i class="fa fa-check"></i><b>14.9.5</b> Key output</a></li>
<li class="chapter" data-level="14.9.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-8"><i class="fa fa-check"></i><b>14.9.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>14.10</b> t-test</a>
<ul>
<li class="chapter" data-level="14.10.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-9"><i class="fa fa-check"></i><b>14.10.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.10.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-9"><i class="fa fa-check"></i><b>14.10.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.10.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-9"><i class="fa fa-check"></i><b>14.10.3</b> Procedures</a></li>
<li class="chapter" data-level="14.10.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-9"><i class="fa fa-check"></i><b>14.10.4</b> Key output</a></li>
<li class="chapter" data-level="14.10.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-8"><i class="fa fa-check"></i><b>14.10.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.10.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-9"><i class="fa fa-check"></i><b>14.10.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>14.11</b> Wilcoxon test</a>
<ul>
<li class="chapter" data-level="14.11.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-10"><i class="fa fa-check"></i><b>14.11.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.11.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-10"><i class="fa fa-check"></i><b>14.11.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.11.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-10"><i class="fa fa-check"></i><b>14.11.3</b> Procedures</a></li>
<li class="chapter" data-level="14.11.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-10"><i class="fa fa-check"></i><b>14.11.4</b> Key output</a></li>
<li class="chapter" data-level="14.11.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-9"><i class="fa fa-check"></i><b>14.11.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.11.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-10"><i class="fa fa-check"></i><b>14.11.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>14.12</b> RAIDA</a>
<ul>
<li class="chapter" data-level="14.12.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-9"><i class="fa fa-check"></i><b>14.12.1</b> Vignette</a></li>
<li class="chapter" data-level="14.12.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-11"><i class="fa fa-check"></i><b>14.12.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.12.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-11"><i class="fa fa-check"></i><b>14.12.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.12.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-11"><i class="fa fa-check"></i><b>14.12.4</b> Procedures</a></li>
<li class="chapter" data-level="14.12.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-11"><i class="fa fa-check"></i><b>14.12.5</b> Key output</a></li>
<li class="chapter" data-level="14.12.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-10"><i class="fa fa-check"></i><b>14.12.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.12.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-11"><i class="fa fa-check"></i><b>14.12.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>14.13</b> Summary</a></li>
<li class="chapter" data-level="14.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>14.14</b> Advantages and disadvantages</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>15</b> Examples</a>
<ul>
<li class="chapter" data-level="15.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>15.1</b> Workflow description</a></li>
<li class="chapter" data-level="15.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>15.2</b> Amplicon sequencing (16s)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>15.2.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="15.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>15.2.2</b> Quality Control</a></li>
<li class="chapter" data-level="15.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>15.2.3</b> Data processing</a></li>
<li class="chapter" data-level="15.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>15.2.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="15.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>15.2.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="15.2.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>15.2.6</b> Microbial composition</a></li>
<li class="chapter" data-level="15.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>15.2.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="examples.html"><a href="examples.html#metagenomics-mgs"><i class="fa fa-check"></i><b>15.3</b> Metagenomics (MGS)</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>15.3.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="15.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>15.3.2</b> Quality Control</a></li>
<li class="chapter" data-level="15.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>15.3.3</b> Data processing</a></li>
<li class="chapter" data-level="15.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>15.3.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="15.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>15.3.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="15.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-2"><i class="fa fa-check"></i><b>15.3.6</b> Microbial composition</a></li>
<li class="chapter" data-level="15.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>15.3.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="examples.html"><a href="examples.html#systematic-information-11"><i class="fa fa-check"></i><b>15.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">XMAS 2.0 Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-analysis" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Differential Analysis<a href="differential-analysis.html#differential-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Identifying the significant taxa between case and control group is necessary. There are too many differential analysis (DA) approaches to choose <span class="citation">(<a href="#ref-nearing2022microbiome" role="doc-biblioref">Nearing et al. 2022</a>)</span>, we provide some of them which focusing on microbial data, including:</p>
<div style="border: 0px;overflow-x: scroll; width:100%; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-139">Table 11.1: </span>Differential analysis tools in XMAS.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Tool.version.
</th>
<th style="text-align:left;">
Input
</th>
<th style="text-align:left;">
Normalization
</th>
<th style="text-align:left;">
Transformation
</th>
<th style="text-align:left;">
Distribution
</th>
<th style="text-align:left;">
MicrobialData
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ALDEx2 (1.26.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
CLR
</td>
<td style="text-align:left;">
Dirichlet-multinormial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
limma voom (3.50.1)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None/TMM
</td>
<td style="text-align:left;">
Log; Precision weighting
</td>
<td style="text-align:left;">
Normal
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
mbzinb (0.2)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
RLE
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Zero-inflated negative binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
omnibus (0.2)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
GMPR(Geometric Mean of Pairwise Ratios)
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Zero-inflated negative binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
RAIDA (1.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
zero-inflated Log
</td>
<td style="text-align:left;">
Modified t-test
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
Wilcox(rare/CLR)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None/CLR
</td>
<td style="text-align:left;">
Non-parametric
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
LEfSe
</td>
<td style="text-align:left;">
Rarefied Counts/Relative
</td>
<td style="text-align:left;">
TSS
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Non-parametric
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
t-test (rare)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Normal
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
metagenomeSeq (1.36.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
CSS
</td>
<td style="text-align:left;">
Log
</td>
<td style="text-align:left;">
Zero-inflated (log) Normal
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
DESeq2 (1.34.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
RLE
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Negative binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
edgeR (3.36.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
RLE/TMM
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Negative binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
ANCOM-II (2.1)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
ALR
</td>
<td style="text-align:left;">
Non-parametric
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
Corncob (0.2.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Beta-binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
MaAslin2 (1.8.0)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None/TSS
</td>
<td style="text-align:left;">
AST
</td>
<td style="text-align:left;">
Normal
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
</tbody>
</table>
</div>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="differential-analysis.html#cb223-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb223-2"><a href="differential-analysis.html#cb223-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb223-3"><a href="differential-analysis.html#cb223-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb223-4"><a href="differential-analysis.html#cb223-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb223-5"><a href="differential-analysis.html#cb223-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb223-6"><a href="differential-analysis.html#cb223-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
<div id="amplicon-sequencing-microbial-data-16s" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Amplicon sequencing microbial data (16s)<a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same strategy to filter phyloseq object.</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="differential-analysis.html#cb224-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;dada2_ps&quot;</span>)</span>
<span id="cb224-2"><a href="differential-analysis.html#cb224-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb224-3"><a href="differential-analysis.html#cb224-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb224-4"><a href="differential-analysis.html#cb224-4" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb224-5"><a href="differential-analysis.html#cb224-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb224-6"><a href="differential-analysis.html#cb224-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb224-7"><a href="differential-analysis.html#cb224-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb224-8"><a href="differential-analysis.html#cb224-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb224-9"><a href="differential-analysis.html#cb224-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb224-10"><a href="differential-analysis.html#cb224-10" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Rarefying counts in phyloseq-class object</span></span>
<span id="cb224-11"><a href="differential-analysis.html#cb224-11" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(<span class="at">object =</span> dada2_ps_remove_BRS, </span>
<span id="cb224-12"><a href="differential-analysis.html#cb224-12" aria-hidden="true" tabindex="-1"></a>                               <span class="at">size =</span> <span class="dv">51181</span>)</span>
<span id="cb224-13"><a href="differential-analysis.html#cb224-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb224-14"><a href="differential-analysis.html#cb224-14" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb224-15"><a href="differential-analysis.html#cb224-15" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps_rarefy, </span>
<span id="cb224-16"><a href="differential-analysis.html#cb224-16" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, </span>
<span id="cb224-17"><a href="differential-analysis.html#cb224-17" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">absolute =</span> <span class="cn">TRUE</span>)</span>
<span id="cb224-18"><a href="differential-analysis.html#cb224-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb224-19"><a href="differential-analysis.html#cb224-19" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb224-20"><a href="differential-analysis.html#cb224-20" aria-hidden="true" tabindex="-1"></a><span class="co"># dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(ps = dada2_ps_rare_genus, </span></span>
<span id="cb224-21"><a href="differential-analysis.html#cb224-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                                                   cutoff = 10, </span></span>
<span id="cb224-22"><a href="differential-analysis.html#cb224-22" aria-hidden="true" tabindex="-1"></a><span class="co">#                                                   unclass = TRUE)</span></span>
<span id="cb224-23"><a href="differential-analysis.html#cb224-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb224-24"><a href="differential-analysis.html#cb224-24" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb224-25"><a href="differential-analysis.html#cb224-25" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb224-26"><a href="differential-analysis.html#cb224-26" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">cutoff =</span> <span class="dv">10</span>, </span>
<span id="cb224-27"><a href="differential-analysis.html#cb224-27" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb224-28"><a href="differential-analysis.html#cb224-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb224-29"><a href="differential-analysis.html#cb224-29" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb224-30"><a href="differential-analysis.html#cb224-30" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> dada2_ps_genus_filter, </span>
<span id="cb224-31"><a href="differential-analysis.html#cb224-31" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">cutoff =</span> <span class="fl">0.2</span>, </span>
<span id="cb224-32"><a href="differential-analysis.html#cb224-32" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb224-33"><a href="differential-analysis.html#cb224-33" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 100 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 100 taxa by 6 taxonomic ranks ]</code></pre>
<div id="aldex2" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> ALDEx2<a href="differential-analysis.html#aldex2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ALDEx2</strong> package is from <em>Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis</em> <span class="citation">(<a href="#ref-fernandes2014unifying" role="doc-biblioref">Fernandes et al. 2014</a>)</span>, and its principle is using log-ratio transformation and statistical testing to find the significant Taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<p><code>run_aldex</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_aldex)</code>.</p>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="differential-analysis.html#cb226-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb226-2"><a href="differential-analysis.html#cb226-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb226-3"><a href="differential-analysis.html#cb226-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb226-4"><a href="differential-analysis.html#cb226-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb226-5"><a href="differential-analysis.html#cb226-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="differential-analysis.html#cb228-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="differential-analysis.html#cb230-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment  EffectSize    Pvalue AdjustedPvalue Median CLR \n(All) Median CLR\nAA
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif  0.03216861 0.6548749      0.9018534         -3.4244631      -3.555905
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif  0.09330854 0.6658567      0.9138848          2.7645196       2.167638
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif  0.30832148 0.3184500      0.7716171         -0.7734251      -2.223805
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif -0.08089118 0.6408318      0.9042170          5.4463483       6.081374
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif -0.15579201 0.5972170      0.8723074         -2.9580004      -2.541838
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif -0.09579649 0.6356202      0.9052481          3.7226931       4.768802
##   Median CLR\nBB Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1     -3.3102684                                NA                       0                    0                  0.0
## 2      3.1218379                        -0.9577718                      30                   26                 50.5
## 3      0.3041529                                NA                       0                    0                  9.0
## 4      5.3302183                         0.1524904                     316                  329                296.0
## 5     -3.2004909                                NA                       0                    0                  0.0
## 6      3.4586946                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       1.7051442              58.82609         101.777778           31.21429                    21.74
## 2                      -1.3107676              50.91304          26.777778           66.42857                    82.61
## 3                      -2.4683647              19.21739           5.111111           28.28571                    43.48
## 4                       1.7488297             996.26087        1740.444444          517.85714                    65.22
## 5                      -2.1676332              93.78261          30.000000          134.78571                    21.74
## 6                       0.2000385             163.86957         177.888889          154.85714                    73.91
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                 88.89                 78.57       4.1 (6.8;1.3)
## 3                 33.33                 50.00        8.6 (13;4.3)
## 4                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                 22.22                 21.43      1.5 (2.3;0.71)
## 6                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Median CLR (All)/(group AA)/(group BB)</strong>: Median CLR (normalization by <strong>ALDEx2</strong>) in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>ALDEx2</strong>.</p>
<details>
<summary>
<strong>run_da() pattern</strong>
</summary>
<p>We also provide another function <code>run_da</code> to run ALDEx2 differential analysis.</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="differential-analysis.html#cb232-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb232-2"><a href="differential-analysis.html#cb232-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb232-3"><a href="differential-analysis.html#cb232-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb232-4"><a href="differential-analysis.html#cb232-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb232-5"><a href="differential-analysis.html#cb232-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;aldex&quot;</span>,</span>
<span id="cb232-6"><a href="differential-analysis.html#cb232-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="differential-analysis.html#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>otu_table or sample_table as inputdata</strong>
</summary>
<p>We also provide <em>data_otu</em> and <em>data_sam</em> as input data to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="differential-analysis.html#cb236-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb236-2"><a href="differential-analysis.html#cb236-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb236-3"><a href="differential-analysis.html#cb236-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb236-4"><a href="differential-analysis.html#cb236-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb236-5"><a href="differential-analysis.html#cb236-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb236-6"><a href="differential-analysis.html#cb236-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="differential-analysis.html#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>taxa_level option</strong>
</summary>
<p>We also provide <em>taxa_level</em> for choosing the specific taxonomic level to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="differential-analysis.html#cb240-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb240-2"><a href="differential-analysis.html#cb240-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb240-3"><a href="differential-analysis.html#cb240-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb240-4"><a href="differential-analysis.html#cb240-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb240-5"><a href="differential-analysis.html#cb240-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb240-6"><a href="differential-analysis.html#cb240-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="differential-analysis.html#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
</div>
<div id="limma_voom" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> limma_voom<a href="differential-analysis.html#limma_voom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>limma</strong> package is from <em>voom: Precision weights unlock linear model analysis tools for RNA-seq read counts</em> <span class="citation">(<a href="#ref-law2014voom" role="doc-biblioref">Law et al. 2014</a>)</span>. Firstly, transforming count data to log2-counts per million (logCPM), estimate the mean-variance relationship and use this to compute appropriate observation-level weights. Secondly, fitting multiple linear models by weighted or generalized least squares. Finally, performing empirical bayes statistics for differential expression.</p>
<p><code>run_limma_voom</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_limma_voom)</code>.</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="differential-analysis.html#cb244-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb244-2"><a href="differential-analysis.html#cb244-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb244-3"><a href="differential-analysis.html#cb244-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb244-4"><a href="differential-analysis.html#cb244-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb244-5"><a href="differential-analysis.html#cb244-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb244-6"><a href="differential-analysis.html#cb244-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="differential-analysis.html#cb246-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment EffectSize     logFC       Pvalue AdjustedPvalue
## 1 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif   5.021733  5.021733 0.0008906114      0.0766607
## 2             g__Intestinibacter 9_AA vs 14_BB  Nonsignif   4.686583  4.686583 0.0015332140      0.0766607
## 3                  g__Romboutsia 9_AA vs 14_BB  Nonsignif   4.756063  4.756063 0.0034873620      0.1162454
## 4     g__Ruminococcaceae_UCG_004 9_AA vs 14_BB  Nonsignif  -3.940866 -3.940866 0.0181806330      0.3696540
## 5               g__Lactobacillus 9_AA vs 14_BB  Nonsignif   3.888865  3.888865 0.0221002911      0.3696540
## 6                 g__Veillonella 9_AA vs 14_BB  Nonsignif   3.431177  3.431177 0.0258757799      0.3696540
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                      11                    0                 39.5
## 2                                NA                      14                    0                 58.0
## 3                         -7.036174                     107                    2                262.5
## 4                                NA                      11                  104                  0.0
## 5                         -3.033423                      56                   24                196.5
## 6                                NA                      13                    0                 40.5
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       -4.884638             257.21739           14.00000          413.57143                    60.87
## 2                       -2.157952              83.56522           26.88889          120.00000                    60.87
## 3                       -3.131606             296.47826           51.77778          453.78571                    78.26
## 4                        2.275424              65.26087          126.22222           26.07143                    52.17
## 5                       -5.090571             791.26087           37.44444         1275.85714                    86.96
## 6                       -0.635927              65.52174           49.00000           76.14286                    56.52
##   Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                 22.22                 85.71       11000 (11000;11000)
## 2                 22.22                 85.71               4.2 (7;1.4)
## 3                 55.56                 92.86                37 (44;30)
## 4                 77.78                 35.71           0.17 (-3.3;3.6)
## 5                 77.78                 92.86 1.8e+08 (1.8e+08;1.8e+08)
## 6                 22.22                 78.57            1.3 (1.8;0.79)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>limma</strong>;</p></li>
<li><p><strong>logFC</strong>: LogFC from groups’ coefficient by <strong>limma</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>limma</strong>;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>limma</strong>.</p>
<details>
<summary>
<strong>other options for limma-voom</strong>
</summary>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="differential-analysis.html#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb248-2"><a href="differential-analysis.html#cb248-2" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb248-3"><a href="differential-analysis.html#cb248-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb248-4"><a href="differential-analysis.html#cb248-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb248-5"><a href="differential-analysis.html#cb248-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb248-6"><a href="differential-analysis.html#cb248-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb248-7"><a href="differential-analysis.html#cb248-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-8"><a href="differential-analysis.html#cb248-8" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb248-9"><a href="differential-analysis.html#cb248-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps,</span>
<span id="cb248-10"><a href="differential-analysis.html#cb248-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb248-11"><a href="differential-analysis.html#cb248-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb248-12"><a href="differential-analysis.html#cb248-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb248-13"><a href="differential-analysis.html#cb248-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-14"><a href="differential-analysis.html#cb248-14" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb248-15"><a href="differential-analysis.html#cb248-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb248-16"><a href="differential-analysis.html#cb248-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb248-17"><a href="differential-analysis.html#cb248-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb248-18"><a href="differential-analysis.html#cb248-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-19"><a href="differential-analysis.html#cb248-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span>
<span id="cb248-20"><a href="differential-analysis.html#cb248-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="mbzinb" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> mbzinb<a href="differential-analysis.html#mbzinb" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>mbzinb</strong> package is from <em>An omnibus test for differential distribution analysis of microbiome sequencing data</em> <span class="citation">(<a href="#ref-chen2018omnibus" role="doc-biblioref">J. Chen et al. 2018</a>)</span>. It uses zeroinflated negative binomial model to investigate the significant taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="differential-analysis.html#cb249-1" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb249-2"><a href="differential-analysis.html#cb249-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb249-3"><a href="differential-analysis.html#cb249-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb249-4"><a href="differential-analysis.html#cb249-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb249-5"><a href="differential-analysis.html#cb249-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;base.mean&quot;                         &quot;mean.LFC&quot;                          &quot;base.abund&quot;                       
## [10] &quot;abund.LFC&quot;                         &quot;base.prev&quot;                         &quot;prev.change&quot;                      
## [13] &quot;base.disp&quot;                         &quot;disp.LFC&quot;                          &quot;statistic&quot;                        
## [16] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [19] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [22] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [25] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="differential-analysis.html#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment      Pvalue AdjustedPvalue EffectSize base.mean  mean.LFC base.abund
## 1               g__Lactobacillus 9_AA vs 14_BB  Nonsignif 0.003445014      0.1564333 1238.41270  33.59784 4.5605786   42.55933
## 2                 g__Veillonella 9_AA vs 14_BB  Nonsignif 0.005578505      0.1564333   27.14286  40.19132 0.6303784  180.85352
## 3 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif 0.007406534      0.1564333  399.57143  10.61857 4.5552518   46.18153
## 4         g__Ruminiclostridium_5 9_AA vs 14_BB  Nonsignif 0.007582511      0.1564333   25.62698  30.04447 0.8479487   30.04447
## 5              g__Ruminococcus_1 9_AA vs 14_BB  Nonsignif 0.008839059      0.1564333   88.71429  39.70707 1.7153262  178.70020
## 6                g__Tyzzerella_4 9_AA vs 14_BB  Nonsignif 0.010053476      0.1564333  169.41270  51.73606 1.7830194  465.54115
##    abund.LFC base.prev prev.change    base.disp  disp.LFC statistic Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1  4.2823480 0.7894354   0.1679182 9.014570e-01  1.275777  13.63594                        -3.0334230                      56
## 2 -1.2059334 0.2222314   0.5713478 5.952821e-02  3.984325  12.60309                                NA                      13
## 3  2.4345380 0.2299311   0.7700602 1.205364e+00  1.475574  11.99319                                NA                      11
## 4  1.0696202 1.0000000  -0.1424287 1.214928e-01  1.749897  11.94257                        -0.3289485                      45
## 5  1.0269047 0.2221994   0.1358792 2.507315e-09 28.621384  11.61166                                NA                       0
## 6 -0.4966242 0.1111310   0.4284745 1.370351e-09 30.800186  11.33333                                NA                       0
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1                   24                196.5                      -5.0905713             791.26087           37.44444
## 2                    0                 40.5                      -0.6359270              65.52174           49.00000
## 3                    0                 39.5                      -4.8846378             257.21739           14.00000
## 4                   41                 51.5                      -0.7849381              51.04348           35.44444
## 5                    0                  0.0                      -1.4189209             107.00000           53.00000
## 6                    0                  4.5                      -2.0807691             155.56522           52.44444
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1         1275.85714                    86.96                 77.78                 92.86 1.8e+08 (1.8e+08;1.8e+08)
## 2           76.14286                    56.52                 22.22                 78.57            1.3 (1.8;0.79)
## 3          413.57143                    60.87                 22.22                 85.71       11000 (11000;11000)
## 4           61.07143                    91.30                100.00                 85.71            2.4 (4.2;0.69)
## 5          141.71429                    30.43                 22.22                 35.71            1.6 (2.6;0.68)
## 6          221.85714                    34.78                 11.11                 50.00            2.1 (3.5;0.64)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>mbzinb</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>mbzinb</strong>;</p></li>
<li><p><strong>base.mean</strong>: fitted mean abundance parameter times the fitted prevalence in baseline group;</p></li>
<li><p><strong>mean.LFC</strong>: log2-fold change in fitted mean between other group and baseline;</p></li>
<li><p><strong>base.abund</strong>: fitted mean abundance parameter in baseline group;</p></li>
<li><p><strong>abund.LFC</strong>: log2-fold change in fitted mean abundance parameter between other group and baseline;</p></li>
<li><p><strong>base.prev</strong>: fitted prevalence in baseline group;</p></li>
<li><p><strong>prev.change</strong>: (linear) difference in prevalence between baseline group and other group (other-baseline);</p></li>
<li><p><strong>base.disp</strong>: fitted dispersion parameter in baseline group;</p></li>
<li><p><strong>disp.LFC</strong>: log2-fold change in fitted dispersion parameter between other group and baseline;</p></li>
<li><p><strong>statistic</strong>: value of likelihood ratio test statistic;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>mbzinb</strong>.</p>
<details>
<summary>
<strong>other options for mbzinb</strong>
</summary>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="differential-analysis.html#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb253-2"><a href="differential-analysis.html#cb253-2" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb253-3"><a href="differential-analysis.html#cb253-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb253-4"><a href="differential-analysis.html#cb253-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb253-5"><a href="differential-analysis.html#cb253-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb253-6"><a href="differential-analysis.html#cb253-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb253-7"><a href="differential-analysis.html#cb253-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-8"><a href="differential-analysis.html#cb253-8" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb253-9"><a href="differential-analysis.html#cb253-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb253-10"><a href="differential-analysis.html#cb253-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb253-11"><a href="differential-analysis.html#cb253-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb253-12"><a href="differential-analysis.html#cb253-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb253-13"><a href="differential-analysis.html#cb253-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-14"><a href="differential-analysis.html#cb253-14" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb253-15"><a href="differential-analysis.html#cb253-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb253-16"><a href="differential-analysis.html#cb253-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb253-17"><a href="differential-analysis.html#cb253-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb253-18"><a href="differential-analysis.html#cb253-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-19"><a href="differential-analysis.html#cb253-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span>
<span id="cb253-20"><a href="differential-analysis.html#cb253-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="omnibus" class="section level3 hasAnchor" number="11.1.4">
<h3><span class="header-section-number">11.1.4</span> omnibus<a href="differential-analysis.html#omnibus" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This approach is also from <strong>mbzinb</strong> <span class="citation">(<a href="#ref-chen2018omnibus" role="doc-biblioref">J. Chen et al. 2018</a>)</span> package. it uses GMPR (Geometric Mean of Pairwise Ratios) <span class="citation">(<a href="#ref-chen2018gmpr" role="doc-biblioref">L. Chen et al. 2018</a>)</span> to get the size factors.</p>
<blockquote>
<p>where we specify models for count (abundance), zero (prevalence) and dispersion part. We also provide likelihood ratio test (zinb.lrt) for different models.</p>
</blockquote>
<p>(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="differential-analysis.html#cb254-1" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb254-2"><a href="differential-analysis.html#cb254-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb254-3"><a href="differential-analysis.html#cb254-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb254-4"><a href="differential-analysis.html#cb254-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb254-5"><a href="differential-analysis.html#cb254-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span></code></pre></div>
<pre><code>## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  92  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="differential-analysis.html#cb256-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;chi.stat&quot;                          &quot;df&quot;                                &quot;abund.baseline&quot;                   
## [10] &quot;prev.baseline&quot;                     &quot;dispersion.baseline&quot;               &quot;abund.LFC.CompvarBB.est&quot;          
## [13] &quot;abund.LFC.CompvarBB.se&quot;            &quot;prev.LOD.CompvarBB.est&quot;            &quot;prev.LOD.CompvarBB.se&quot;            
## [16] &quot;dispersion.LFC.CompvarBB.est&quot;      &quot;dispersion.LFC.CompvarBB.se&quot;       &quot;method&quot;                           
## [19] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [22] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [25] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [28] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="differential-analysis.html#cb258-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment     Pvalue AdjustedPvalue EffectSize  chi.stat df abund.baseline prev.baseline
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.26476211      0.6245670   70.56349 3.9696315  3   0.0050842054     0.2910300
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.13905545      0.4437791   39.65079 5.4930407  3   0.0006912502     0.8894593
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.06293962      0.3795000   23.17460 7.2995223  3   0.0003647798     0.3333734
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.12024562      0.4437791 1222.58730 5.8287526  3   0.0553800205     0.5555645
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.86344681      0.9142217  104.78571 0.7413110  3   0.0031212219     0.2357130
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.97967175      0.9796718   23.03175 0.1869261  3   0.0049314397     0.7815505
##   dispersion.baseline abund.LFC.CompvarBB.est abund.LFC.CompvarBB.se prev.LOD.CompvarBB.est prev.LOD.CompvarBB.se
## 1           0.2041100             -0.36978708              1.4040484             -0.4088983             0.9811747
## 2           3.1647486              0.73754098              0.3062639             -0.7727912             1.2480678
## 3          11.0696374              1.29262130              0.3614490              0.6959681             0.8863794
## 4           1.6324698             -1.15572656              0.4756819              0.6968830             0.8947568
## 5           0.5149302             -0.83042868              1.7432216              2.3976744             1.0114415
## 6           1.0012623              0.03215791              0.4770876             -0.3541131             1.0007538
##   dispersion.LFC.CompvarBB.est dispersion.LFC.CompvarBB.se  method Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1                    2.8343543                   1.1361534 omnibus                                NA                       0
## 2                   -0.5318657                   0.6766040 omnibus                        -0.9577718                      30
## 3                   -1.7738398                   1.4812754 omnibus                                NA                       0
## 4                   -0.5257607                   0.7011677 omnibus                         0.1524904                     316
## 5                   -2.4675480                   1.0580187 omnibus                                NA                       0
## 6                    0.1662740                   0.6265328 omnibus                        -0.1085245                      64
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1                    0                  0.0                       1.7051442              58.82609         101.777778
## 2                   26                 50.5                      -1.3107676              50.91304          26.777778
## 3                    0                  9.0                      -2.4683647              19.21739           5.111111
## 4                  329                296.0                       1.7488297             996.26087        1740.444444
## 5                    0                  0.0                      -2.1676332              93.78261          30.000000
## 6                   64                 69.0                       0.2000385             163.86957         177.888889
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1           31.21429                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2           66.42857                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3           28.28571                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4          517.85714                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5          134.78571                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6          154.85714                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>omnibus</strong>;</p></li>
<li><p><strong>chi.stat</strong>: chisquare statistics;</p></li>
<li><p><strong>df</strong>: degree freedom;</p></li>
<li><p><strong>abund.baseline</strong>: mean abundance in baseline group;</p></li>
<li><p><strong>prev.baseline</strong>: prevalence in baseline group;</p></li>
<li><p><strong>dispersion.baseline</strong>: dispersion in baseline group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.est</strong>: log2-fold change in abundance between group BB and other group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.se</strong>: log2-fold change of standard errors in abundance between group BB and other group;</p></li>
<li><p><strong>prev.LOD.CompvarBB.est</strong>: prevalence of low of detect value in group BB;</p></li>
<li><p><strong>prev.LOD.CompvarBB.se</strong>: prevalence’s standard errors of low of detect value in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.est</strong>: log2-fold change in dispersion in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.se</strong>: log2-fold change in dispersion’s standard errors in group BB;</p></li>
<li><p><strong>method</strong>: methods of test;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>omnibus</strong>.</p>
<details>
<summary>
<strong>other options for omnibus</strong>
</summary>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="differential-analysis.html#cb260-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb260-2"><a href="differential-analysis.html#cb260-2" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb260-3"><a href="differential-analysis.html#cb260-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb260-4"><a href="differential-analysis.html#cb260-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb260-5"><a href="differential-analysis.html#cb260-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb260-6"><a href="differential-analysis.html#cb260-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb260-7"><a href="differential-analysis.html#cb260-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb260-8"><a href="differential-analysis.html#cb260-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-9"><a href="differential-analysis.html#cb260-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-10"><a href="differential-analysis.html#cb260-10" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb260-11"><a href="differential-analysis.html#cb260-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps,</span>
<span id="cb260-12"><a href="differential-analysis.html#cb260-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb260-13"><a href="differential-analysis.html#cb260-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb260-14"><a href="differential-analysis.html#cb260-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb260-15"><a href="differential-analysis.html#cb260-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb260-16"><a href="differential-analysis.html#cb260-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-17"><a href="differential-analysis.html#cb260-17" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb260-18"><a href="differential-analysis.html#cb260-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb260-19"><a href="differential-analysis.html#cb260-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb260-20"><a href="differential-analysis.html#cb260-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb260-21"><a href="differential-analysis.html#cb260-21" aria-hidden="true" tabindex="-1"></a>                  <span class="at">da_method =</span> <span class="st">&quot;omnibus&quot;</span>,</span>
<span id="cb260-22"><a href="differential-analysis.html#cb260-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb260-23"><a href="differential-analysis.html#cb260-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-24"><a href="differential-analysis.html#cb260-24" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span>
<span id="cb260-25"><a href="differential-analysis.html#cb260-25" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="raida" class="section level3 hasAnchor" number="11.1.5">
<h3><span class="header-section-number">11.1.5</span> RAIDA<a href="differential-analysis.html#raida" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>RAIDA</strong> package is from <em>A robust approach for identifying differentially abundant features in metagenomic samples</em> <span class="citation">(<a href="#ref-sohn2015robust" role="doc-biblioref">Sohn, Du, and An 2015</a>)</span>. It uses Ratio Approach for Identifying Differential Abundance (RAIDA). (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="differential-analysis.html#cb261-1" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb261-2"><a href="differential-analysis.html#cb261-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb261-3"><a href="differential-analysis.html#cb261-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb261-4"><a href="differential-analysis.html#cb261-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb261-5"><a href="differential-analysis.html#cb261-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb261-6"><a href="differential-analysis.html#cb261-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;eta_AA&quot;                            &quot;mean_AA&quot;                           &quot;sd_AA&quot;                            
## [10] &quot;eta_BB&quot;                            &quot;mean_BB&quot;                           &quot;sd_BB&quot;                            
## [13] &quot;mod.pool.var&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [16] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [19] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [22] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [25] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="differential-analysis.html#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##                             TaxaID         Block Enrichment EffectSize      Pvalue AdjustedPvalue    eta_AA   mean_AA
## 1                 g__Mogibacterium 9_AA vs 14_BB  Nonsignif   32.53175 0.004340020      0.2061720 1.0000000 -6.665684
## 2                   g__Megasphaera 9_AA vs 14_BB  Nonsignif  409.19048 0.004581601      0.2061720 0.6666667  1.550194
## 3                    g__Romboutsia 9_AA vs 14_BB  Nonsignif  402.00794 0.025287679      0.7353741 0.4444444 -3.748832
## 4                 g__Lactobacillus 9_AA vs 14_BB  Nonsignif 1238.41270 0.033492279      0.7353741 0.2222222 -3.438359
## 5                    g__Tyzzerella 9_AA vs 14_BB  Nonsignif   75.78571 0.059752064      0.7353741 0.4444444 -2.455253
## 6 g__Christensenellaceae_R_7_group 9_AA vs 14_BB  Nonsignif   44.21429 0.068999409      0.7353741 0.6666667 -2.344046
##       sd_AA     eta_BB   mean_BB    sd_BB mod.pool.var Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1 0.0000010 0.57142857 -3.409936 1.909244     2.765864                                NA                       0
## 2 0.4452225 0.64285714 -1.221216 1.074673     1.185998                                NA                       0
## 3 1.5814829 0.07142857 -1.467304 2.085972     3.260003                         -7.036174                     107
## 4 1.7766726 0.07142857 -1.250908 2.484244     4.276535                         -3.033423                      56
## 5 1.3348729 0.78571429 -5.123355 3.191661     3.081402                                NA                       0
## 6 0.5621146 0.64285714 -4.420090 1.890465     2.021487                                NA                       0
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1                    0                  0.0                       -8.198620              19.91304          0.1111111
## 2                    0                  0.0                        1.738259             335.26087        584.3333333
## 3                    2                262.5                       -3.131606             296.47826         51.7777778
## 4                   24                196.5                       -5.090571             791.26087         37.4444444
## 5                    4                  0.0                        2.433624              46.86957         93.0000000
## 6                    0                  0.0                        1.801549              35.08696         62.0000000
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1           32.64286                    30.43                 11.11                 42.86    2e+283 (2e+283;2e+283)
## 2          175.14286                    34.78                 33.33                 35.71          0.54 (-0.65;1.7)
## 3          453.78571                    78.26                 55.56                 92.86                37 (44;30)
## 4         1275.85714                    86.96                 77.78                 92.86 1.8e+08 (1.8e+08;1.8e+08)
## 5           17.21429                    34.78                 55.56                 21.43            0.4 (-1.4;2.2)
## 6           17.78571                    34.78                 33.33                 35.71          0.54 (-0.65;1.7)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>RAIDA</strong>;</p></li>
<li><p><strong>eta_AA</strong>: vector containing estimated probabilities of the false zero state for group AA;</p></li>
<li><p><strong>mean_AA</strong>: vector containing estimated means of log ratios for group AA;</p></li>
<li><p><strong>sd_AA</strong>: vector containing estimated standard deviations of log ratios for group AA;</p></li>
<li><p><strong>eta_BB</strong>: vector containing estimated probabilities of the false zero state for group BB;</p></li>
<li><p><strong>mean_BB</strong>: vector containing estimated means of log ratios for group BB;</p></li>
<li><p><strong>sd_BB</strong>: vector containing estimated standard deviations of log ratios for group BB;</p></li>
<li><p><strong>mod.pool.var</strong>: vector containing estimated posterior variances of log ratios;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>RAIDA</strong>.</p>
<details>
<summary>
<strong>other options for RAIDA</strong>
</summary>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="differential-analysis.html#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb265-2"><a href="differential-analysis.html#cb265-2" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb265-3"><a href="differential-analysis.html#cb265-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb265-4"><a href="differential-analysis.html#cb265-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb265-5"><a href="differential-analysis.html#cb265-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb265-6"><a href="differential-analysis.html#cb265-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb265-7"><a href="differential-analysis.html#cb265-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-8"><a href="differential-analysis.html#cb265-8" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb265-9"><a href="differential-analysis.html#cb265-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb265-10"><a href="differential-analysis.html#cb265-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb265-11"><a href="differential-analysis.html#cb265-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb265-12"><a href="differential-analysis.html#cb265-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb265-13"><a href="differential-analysis.html#cb265-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-14"><a href="differential-analysis.html#cb265-14" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb265-15"><a href="differential-analysis.html#cb265-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb265-16"><a href="differential-analysis.html#cb265-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb265-17"><a href="differential-analysis.html#cb265-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb265-18"><a href="differential-analysis.html#cb265-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;raida&quot;</span>)</span>
<span id="cb265-19"><a href="differential-analysis.html#cb265-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-20"><a href="differential-analysis.html#cb265-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span>
<span id="cb265-21"><a href="differential-analysis.html#cb265-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="wilcoxon-rank-sum-and-signed-rank-tests" class="section level3 hasAnchor" number="11.1.6">
<h3><span class="header-section-number">11.1.6</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparametric test methods, use the rank of taxa abundance to find the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="differential-analysis.html#cb266-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb266-2"><a href="differential-analysis.html#cb266-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb266-3"><a href="differential-analysis.html#cb266-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb266-4"><a href="differential-analysis.html#cb266-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb266-5"><a href="differential-analysis.html#cb266-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-6"><a href="differential-analysis.html#cb266-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="differential-analysis.html#cb268-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif   70.56349      63.5 1.0000000      1.0000000                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif   39.65079      44.5 0.2556616      0.7014746                        -0.9577718
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif   23.17460      44.0 0.1980169      0.7014746                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 1222.58730      71.0 0.6293971      0.8964389                         0.1524904
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif  104.78571      64.5 0.9304707      0.9692403                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif   23.03175      67.0 0.8239942      0.9363570                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1                       0                    0                  0.0                      0.01201252                   12.06
## 2                      30                   26                 50.5                     -0.42227633                    9.94
## 3                       0                    0                  9.0                     -0.43387758                    9.89
## 4                     316                  329                296.0                      0.17342686                   12.89
## 5                       0                    0                  0.0                      0.03358045                   12.17
## 6                      64                   64                 69.0                      0.08724541                   12.44
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   11.96                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                   13.32                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3                   13.36                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4                   11.43                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                   11.89                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6                   11.71                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean Rank)_vs_BB**: Log2FoldChange (Mean Rank Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Rank Abundance (All)/(group AA)/(group BB)</strong>: Mean Rank Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for wilcox</strong>
</summary>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="differential-analysis.html#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb270-2"><a href="differential-analysis.html#cb270-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb270-3"><a href="differential-analysis.html#cb270-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb270-4"><a href="differential-analysis.html#cb270-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb270-5"><a href="differential-analysis.html#cb270-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb270-6"><a href="differential-analysis.html#cb270-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb270-7"><a href="differential-analysis.html#cb270-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-8"><a href="differential-analysis.html#cb270-8" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb270-9"><a href="differential-analysis.html#cb270-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb270-10"><a href="differential-analysis.html#cb270-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb270-11"><a href="differential-analysis.html#cb270-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb270-12"><a href="differential-analysis.html#cb270-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb270-13"><a href="differential-analysis.html#cb270-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-14"><a href="differential-analysis.html#cb270-14" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb270-15"><a href="differential-analysis.html#cb270-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb270-16"><a href="differential-analysis.html#cb270-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb270-17"><a href="differential-analysis.html#cb270-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb270-18"><a href="differential-analysis.html#cb270-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb270-19"><a href="differential-analysis.html#cb270-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-20"><a href="differential-analysis.html#cb270-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span>
<span id="cb270-21"><a href="differential-analysis.html#cb270-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="differential-analysis.html#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summary</span></span>
<span id="cb271-2"><a href="differential-analysis.html#cb271-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(<span class="at">ps =</span> dada2_ps_genus_filter_trim)</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 32359&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 49328&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 1008016&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 43826.7826086957&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 45596&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.501304347826087&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 0&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 1&quot;
## 
## [[11]]
## [1] &quot;Group&quot;</code></pre>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="differential-analysis.html#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run</span></span>
<span id="cb273-2"><a href="differential-analysis.html#cb273-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb273-3"><a href="differential-analysis.html#cb273-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb273-4"><a href="differential-analysis.html#cb273-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb273-5"><a href="differential-analysis.html#cb273-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb273-6"><a href="differential-analysis.html#cb273-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb273-7"><a href="differential-analysis.html#cb273-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_rarefy</strong>
</summary>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="differential-analysis.html#cb275-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb275-2"><a href="differential-analysis.html#cb275-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb275-3"><a href="differential-analysis.html#cb275-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb275-4"><a href="differential-analysis.html#cb275-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb275-5"><a href="differential-analysis.html#cb275-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb275-6"><a href="differential-analysis.html#cb275-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb275-7"><a href="differential-analysis.html#cb275-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb275-8"><a href="differential-analysis.html#cb275-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb275-9"><a href="differential-analysis.html#cb275-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_rarefy)</span>
<span id="cb275-10"><a href="differential-analysis.html#cb275-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_CLR</strong>: centered log-ratio normalization.</li>
</ul>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="differential-analysis.html#cb276-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb276-2"><a href="differential-analysis.html#cb276-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb276-3"><a href="differential-analysis.html#cb276-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb276-4"><a href="differential-analysis.html#cb276-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb276-5"><a href="differential-analysis.html#cb276-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb276-6"><a href="differential-analysis.html#cb276-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb276-7"><a href="differential-analysis.html#cb276-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_CLR)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_CLR</strong>
</summary>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="differential-analysis.html#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb278-2"><a href="differential-analysis.html#cb278-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb278-3"><a href="differential-analysis.html#cb278-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb278-4"><a href="differential-analysis.html#cb278-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb278-5"><a href="differential-analysis.html#cb278-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb278-6"><a href="differential-analysis.html#cb278-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb278-7"><a href="differential-analysis.html#cb278-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb278-8"><a href="differential-analysis.html#cb278-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-9"><a href="differential-analysis.html#cb278-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_CLR)</span>
<span id="cb278-10"><a href="differential-analysis.html#cb278-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse" class="section level3 hasAnchor" number="11.1.7">
<h3><span class="header-section-number">11.1.7</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="differential-analysis.html#cb279-1" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb279-2"><a href="differential-analysis.html#cb279-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb279-3"><a href="differential-analysis.html#cb279-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb279-4"><a href="differential-analysis.html#cb279-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb279-5"><a href="differential-analysis.html#cb279-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb279-6"><a href="differential-analysis.html#cb279-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">2</span>)</span>
<span id="cb279-7"><a href="differential-analysis.html#cb279-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;LDA_Score&quot;                         &quot;EffectSize&quot;                        &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
##  [7] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [10] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [13] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [16] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="differential-analysis.html#cb281-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB
## 1              g__Parasutterella 9_AA vs 14_BB         AA -3.604579   2.234915                          4.402050
## 2     g__Ruminococcaceae_UCG_004 9_AA vs 14_BB         AA -2.823442   2.137082                                NA
## 3                 g__Odoribacter 9_AA vs 14_BB         BB  2.340309   1.876320                                NA
## 4             g__Intestinibacter 9_AA vs 14_BB         BB  3.146337   2.384308                                NA
## 5 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB         BB  3.671138   2.679298                                NA
## 6                  g__Romboutsia 9_AA vs 14_BB         BB  3.705317   2.929140                         -6.856424
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1               172.77125           1080.35695             51.09966                        4.481148             3901.0977
## 2               241.24923           2451.90494              0.00000                        2.342928             1443.2092
## 3                21.93175            604.99989              0.00000                        2.099583              578.0758
## 4               383.50911              0.00000           1398.79740                       -1.964554             1883.3386
## 5               225.11921              0.00000           1173.88222                       -4.746617             5861.9051
## 6              2189.79596             50.35627           5835.02623                       -3.334490             7094.1306
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1          9320.3061           417.3209                    65.22                 88.89                 50.00
## 2          2822.7139           556.3847                    52.17                 77.78                 35.71
## 3          1083.9017           252.9020                    52.17                 77.78                 35.71
## 4           680.6440          2656.4994                    60.87                 22.22                 85.71
## 5           350.3379          9405.0554                    60.87                 22.22                 85.71
## 6          1086.1419         10956.4090                    78.26                 55.56                 92.86
##   Odds Ratio (95% CI)
## 1     0.0025 (-12;12)
## 2       0.14 (-3.7;4)
## 3       0.31 (-2;2.6)
## 4         3.4 (5.8;1)
## 5    4900 (5000;4900)
## 6          76 (85;68)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>lefse</em>;</p></li>
<li><p><strong>LDA_Score</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for lefse</strong>
</summary>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="differential-analysis.html#cb283-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb283-2"><a href="differential-analysis.html#cb283-2" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb283-3"><a href="differential-analysis.html#cb283-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb283-4"><a href="differential-analysis.html#cb283-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb283-5"><a href="differential-analysis.html#cb283-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-6"><a href="differential-analysis.html#cb283-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb283-7"><a href="differential-analysis.html#cb283-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb283-8"><a href="differential-analysis.html#cb283-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb283-9"><a href="differential-analysis.html#cb283-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-10"><a href="differential-analysis.html#cb283-10" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb283-11"><a href="differential-analysis.html#cb283-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb283-12"><a href="differential-analysis.html#cb283-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb283-13"><a href="differential-analysis.html#cb283-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-14"><a href="differential-analysis.html#cb283-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb283-15"><a href="differential-analysis.html#cb283-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb283-16"><a href="differential-analysis.html#cb283-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb283-17"><a href="differential-analysis.html#cb283-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-18"><a href="differential-analysis.html#cb283-18" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb283-19"><a href="differential-analysis.html#cb283-19" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb283-20"><a href="differential-analysis.html#cb283-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-21"><a href="differential-analysis.html#cb283-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb283-22"><a href="differential-analysis.html#cb283-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb283-23"><a href="differential-analysis.html#cb283-23" aria-hidden="true" tabindex="-1"></a>                <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb283-24"><a href="differential-analysis.html#cb283-24" aria-hidden="true" tabindex="-1"></a>                <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb283-25"><a href="differential-analysis.html#cb283-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-26"><a href="differential-analysis.html#cb283-26" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span>
<span id="cb283-27"><a href="differential-analysis.html#cb283-27" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test" class="section level3 hasAnchor" number="11.1.8">
<h3><span class="header-section-number">11.1.8</span> t-test<a href="differential-analysis.html#t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="differential-analysis.html#cb284-1" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb284-2"><a href="differential-analysis.html#cb284-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb284-3"><a href="differential-analysis.html#cb284-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb284-4"><a href="differential-analysis.html#cb284-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb284-5"><a href="differential-analysis.html#cb284-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-6"><a href="differential-analysis.html#cb284-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                   
##  [3] &quot;Enrichment&quot;                               &quot;EffectSize&quot;                              
##  [5] &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;       
##  [9] &quot;Median Abundance\n(All)&quot;                  &quot;Median Abundance\nAA&quot;                    
## [11] &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;             
## [15] &quot;GeometricMean Abundance\nBB&quot;              &quot;Occurrence (100%)\n(All)&quot;                
## [17] &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="differential-analysis.html#cb286-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif   70.56349  0.6845254 0.51170983      0.7107081
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif   39.65079 -1.9079866 0.07478702      0.6397026
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif   23.17460 -2.1601919 0.04743155      0.6397026
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 1222.58730  1.4034718 0.19417159      0.6397026
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif  104.78571 -0.7632086 0.45788140      0.6983360
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif   23.03175  0.2691871 0.79125424      0.9059862
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0                    0                  0.0
## 2                        -0.9577718                      30                   26                 50.5
## 3                                NA                       0                    0                  9.0
## 4                         0.1524904                     316                  329                296.0
## 5                                NA                       0                    0                  0.0
## 6                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (geometricmean)\nAA_vs_BB GeometricMean Abundance\n(All) GeometricMean Abundance\nAA
## 1                                 2.674648                      1.2740132                   4.3847026
## 2                                -2.270362                      0.6530643                   0.1647934
## 3                                -3.268970                      0.9776732                   0.1479603
## 4                                 1.413058                      0.7929084                   1.0990412
## 5                                -3.452848                      1.3734381                   0.4150530
## 6                                -2.043760                      0.4449523                   0.2025430
##   GeometricMean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   0.6867376                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                   0.7950361                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3                   1.4262753                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4                   0.4127063                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                   4.5447977                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6                   0.8351231                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (GeometricMean)_vs_BB**: Log2FoldChange (GeometricMean Abundance) between group AA and group BB;</p></li>
<li><p><strong>GeometricMean Abundance (All)/(group AA)/(group BB)</strong>: GeometricMean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for t-test</strong>
</summary>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="differential-analysis.html#cb288-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb288-2"><a href="differential-analysis.html#cb288-2" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb288-3"><a href="differential-analysis.html#cb288-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb288-4"><a href="differential-analysis.html#cb288-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb288-5"><a href="differential-analysis.html#cb288-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb288-6"><a href="differential-analysis.html#cb288-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb288-7"><a href="differential-analysis.html#cb288-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-8"><a href="differential-analysis.html#cb288-8" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb288-9"><a href="differential-analysis.html#cb288-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb288-10"><a href="differential-analysis.html#cb288-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb288-11"><a href="differential-analysis.html#cb288-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb288-12"><a href="differential-analysis.html#cb288-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb288-13"><a href="differential-analysis.html#cb288-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-14"><a href="differential-analysis.html#cb288-14" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb288-15"><a href="differential-analysis.html#cb288-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb288-16"><a href="differential-analysis.html#cb288-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb288-17"><a href="differential-analysis.html#cb288-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb288-18"><a href="differential-analysis.html#cb288-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb288-19"><a href="differential-analysis.html#cb288-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-20"><a href="differential-analysis.html#cb288-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span>
<span id="cb288-21"><a href="differential-analysis.html#cb288-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>t-test_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="differential-analysis.html#cb289-1" aria-hidden="true" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb289-2"><a href="differential-analysis.html#cb289-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb289-3"><a href="differential-analysis.html#cb289-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb289-4"><a href="differential-analysis.html#cb289-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb289-5"><a href="differential-analysis.html#cb289-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb289-6"><a href="differential-analysis.html#cb289-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                   
##  [3] &quot;Enrichment&quot;                               &quot;EffectSize&quot;                              
##  [5] &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;       
##  [9] &quot;Median Abundance\n(All)&quot;                  &quot;Median Abundance\nAA&quot;                    
## [11] &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;             
## [15] &quot;GeometricMean Abundance\nBB&quot;              &quot;Occurrence (100%)\n(All)&quot;                
## [17] &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for ttest_rarefy</strong>
</summary>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="differential-analysis.html#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb291-2"><a href="differential-analysis.html#cb291-2" aria-hidden="true" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb291-3"><a href="differential-analysis.html#cb291-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb291-4"><a href="differential-analysis.html#cb291-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb291-5"><a href="differential-analysis.html#cb291-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb291-6"><a href="differential-analysis.html#cb291-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>,</span>
<span id="cb291-7"><a href="differential-analysis.html#cb291-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb291-8"><a href="differential-analysis.html#cb291-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb291-9"><a href="differential-analysis.html#cb291-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_rarefy)</span>
<span id="cb291-10"><a href="differential-analysis.html#cb291-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="metagenomeseq" class="section level3 hasAnchor" number="11.1.9">
<h3><span class="header-section-number">11.1.9</span> MetagenomeSeq<a href="differential-analysis.html#metagenomeseq" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>MetagenomeSeq</strong> package is from <em>Differential abundance analysis for microbial marker-gene surveys</em> <span class="citation">(<a href="#ref-paulson2013differential" role="doc-biblioref">Paulson et al. 2013</a>)</span>. It uses zero-inflated Log-Normal mixture model or Zero-inflated Gaussian mixture model to identify the significant taxa between groups. (Caution: the <strong>otu_table</strong> should be integers).</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="differential-analysis.html#cb292-1" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb292-2"><a href="differential-analysis.html#cb292-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb292-3"><a href="differential-analysis.html#cb292-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb292-4"><a href="differential-analysis.html#cb292-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb292-5"><a href="differential-analysis.html#cb292-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb292-6"><a href="differential-analysis.html#cb292-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb292-7"><a href="differential-analysis.html#cb292-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-8"><a href="differential-analysis.html#cb292-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;logFC&quot;                             &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;se&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="differential-analysis.html#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##                       TaxaID         Block Enrichment     logFC       Pvalue AdjustedPvalue        se
## 1        g__Terrisporobacter 9_AA vs 14_BB         AA 2.1231996 4.332167e-05   0.0007364684 0.5192453
## 2            g__Tyzzerella_4 9_AA vs 14_BB         AA 2.5901577 2.752647e-04   0.0023397503 0.7120649
## 3           g__Mogibacterium 9_AA vs 14_BB  Nonsignif 1.4203861 8.431066e-04   0.0039964426 0.4254894
## 4                   g__F0332 9_AA vs 14_BB  Nonsignif 1.2608227 9.403394e-04   0.0039964426 0.3811672
## 5 g__Lachnospiraceae_UCG_001 9_AA vs 14_BB  Nonsignif 1.0116177 3.025566e-02   0.1028692297 0.4668874
## 6      g__Peptostreptococcus 9_AA vs 14_BB  Nonsignif 0.6034439 6.819273e-02   0.1932127227 0.3308843
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0                    0             6.784969
## 2                                NA                       0                    0             4.361618
## 3                                NA                       0                    0             0.000000
## 4                                NA                       0                    0             0.000000
## 5                                NA                       0                    0             0.000000
## 6                                NA                       0                    0             0.000000
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                      -3.7339932             35.933695          4.2322026           56.31323                    34.78
## 2                      -2.0976784            308.151607        102.8322440          440.14263                    34.78
## 3                      -7.4275944             22.915793          0.2178649           37.50732                    30.43
## 4                      -3.8972729              6.573925          0.6948392           10.35334                    30.43
## 5                       1.0638692             17.276654         25.3148039           12.10927                    21.74
## 6                       0.5877487              2.786766          3.4995626            2.32854                    21.74
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 11.11                 50.00          25 (31;19)
## 2                 11.11                 50.00      2.1 (3.5;0.64)
## 3                 11.11                 42.86       Inf (Inf;Inf)
## 4                 11.11                 42.86 19000 (19000;19000)
## 5                 11.11                 28.57    0.67 (-0.11;1.5)
## 6                 11.11                 28.57     0.77 (0.27;1.3)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>se</strong>: standard error;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for metagenomeseq</strong>
</summary>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="differential-analysis.html#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb296-2"><a href="differential-analysis.html#cb296-2" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb296-3"><a href="differential-analysis.html#cb296-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb296-4"><a href="differential-analysis.html#cb296-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb296-5"><a href="differential-analysis.html#cb296-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb296-6"><a href="differential-analysis.html#cb296-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb296-7"><a href="differential-analysis.html#cb296-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb296-8"><a href="differential-analysis.html#cb296-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb296-9"><a href="differential-analysis.html#cb296-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-10"><a href="differential-analysis.html#cb296-10" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb296-11"><a href="differential-analysis.html#cb296-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">ps =</span> dada2_ps,</span>
<span id="cb296-12"><a href="differential-analysis.html#cb296-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb296-13"><a href="differential-analysis.html#cb296-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb296-14"><a href="differential-analysis.html#cb296-14" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb296-15"><a href="differential-analysis.html#cb296-15" aria-hidden="true" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb296-16"><a href="differential-analysis.html#cb296-16" aria-hidden="true" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb296-17"><a href="differential-analysis.html#cb296-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-18"><a href="differential-analysis.html#cb296-18" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq<span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb296-19"><a href="differential-analysis.html#cb296-19" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb296-20"><a href="differential-analysis.html#cb296-20" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb296-21"><a href="differential-analysis.html#cb296-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb296-22"><a href="differential-analysis.html#cb296-22" aria-hidden="true" tabindex="-1"></a>                      <span class="at">da_method =</span> <span class="st">&quot;metagenomeseq&quot;</span>,</span>
<span id="cb296-23"><a href="differential-analysis.html#cb296-23" aria-hidden="true" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb296-24"><a href="differential-analysis.html#cb296-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb296-25"><a href="differential-analysis.html#cb296-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-26"><a href="differential-analysis.html#cb296-26" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span>
<span id="cb296-27"><a href="differential-analysis.html#cb296-27" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="deseq2" class="section level3 hasAnchor" number="11.1.10">
<h3><span class="header-section-number">11.1.10</span> DESeq2<a href="differential-analysis.html#deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>DESeq2</strong> package is from <em>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</em> <span class="citation">(<a href="#ref-love2014moderated" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution.(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="differential-analysis.html#cb297-1" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb297-2"><a href="differential-analysis.html#cb297-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb297-3"><a href="differential-analysis.html#cb297-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb297-4"><a href="differential-analysis.html#cb297-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb297-5"><a href="differential-analysis.html#cb297-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb297-6"><a href="differential-analysis.html#cb297-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;logFC&quot;                            
##  [7] &quot;Statistic&quot;                         &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="differential-analysis.html#cb299-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment      Pvalue AdjustedPvalue      logFC  Statistic
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.003314859     0.02209906  3.1129168  2.9369235
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.122064731     0.31069563  1.3609640  1.5461650
## 3   g__Adlercreutzia 9_AA vs 14_BB         BB 0.008736333     0.04408614  2.4112622  2.6222032
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.301287313     0.51946088 -1.1696067 -1.0336767
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.395896745     0.60907192 -0.5625950 -0.8489722
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.535255838     0.68622543 -0.5817917 -0.6200030
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0                    0                  0.0
## 2                        -0.9577718                      30                   26                 50.5
## 3                                NA                       0                    0                  9.0
## 4                         0.1524904                     316                  329                296.0
## 5                                NA                       0                    0                  0.0
## 6                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       1.7051442              58.82609         101.777778           31.21429                    21.74
## 2                      -1.3107676              50.91304          26.777778           66.42857                    82.61
## 3                      -2.4683647              19.21739           5.111111           28.28571                    43.48
## 4                       1.7488297             996.26087        1740.444444          517.85714                    65.22
## 5                      -2.1676332              93.78261          30.000000          134.78571                    21.74
## 6                       0.2000385             163.86957         177.888889          154.85714                    73.91
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                 88.89                 78.57       4.1 (6.8;1.3)
## 3                 33.33                 50.00        8.6 (13;4.3)
## 4                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                 22.22                 21.43      1.5 (2.3;0.71)
## 6                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>Statistic</strong>: test statistic (negative binomial model);</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for deseq2</strong>
</summary>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="differential-analysis.html#cb301-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb301-2"><a href="differential-analysis.html#cb301-2" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb301-3"><a href="differential-analysis.html#cb301-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb301-4"><a href="differential-analysis.html#cb301-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb301-5"><a href="differential-analysis.html#cb301-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb301-6"><a href="differential-analysis.html#cb301-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb301-7"><a href="differential-analysis.html#cb301-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb301-8"><a href="differential-analysis.html#cb301-8" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb301-9"><a href="differential-analysis.html#cb301-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb301-10"><a href="differential-analysis.html#cb301-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb301-11"><a href="differential-analysis.html#cb301-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb301-12"><a href="differential-analysis.html#cb301-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb301-13"><a href="differential-analysis.html#cb301-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb301-14"><a href="differential-analysis.html#cb301-14" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb301-15"><a href="differential-analysis.html#cb301-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb301-16"><a href="differential-analysis.html#cb301-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb301-17"><a href="differential-analysis.html#cb301-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb301-18"><a href="differential-analysis.html#cb301-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;deseq2&quot;</span>)</span>
<span id="cb301-19"><a href="differential-analysis.html#cb301-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb301-20"><a href="differential-analysis.html#cb301-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span>
<span id="cb301-21"><a href="differential-analysis.html#cb301-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="edger" class="section level3 hasAnchor" number="11.1.11">
<h3><span class="header-section-number">11.1.11</span> EdgeR<a href="differential-analysis.html#edger" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>EdgeR</strong> package is from <em>A scaling normalization method for differential expression analysis of RNA-seq data</em> <span class="citation">(<a href="#ref-robinson2010scaling" role="doc-biblioref">Robinson and Oshlack 2010</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="differential-analysis.html#cb302-1" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb302-2"><a href="differential-analysis.html#cb302-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb302-3"><a href="differential-analysis.html#cb302-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb302-4"><a href="differential-analysis.html#cb302-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-170"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-170-1.png" alt="EdgeR (BVC distance)" width="480" />
<p class="caption">
Figure 11.1: EdgeR (BVC distance)
</p>
</div>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="differential-analysis.html#cb303-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_edger)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;logFC&quot;                            
##  [7] &quot;logCPM&quot;                            &quot;LR&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
## [10] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [13] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [16] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [19] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="differential-analysis.html#cb305-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment       Pvalue AdjustedPvalue     logFC    logCPM        LR
## 1               g__Lactobacillus 9_AA vs 14_BB         BB 3.058026e-05    0.002540141  6.631114 15.517455 17.381408
## 2 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB         BB 5.080282e-05    0.002540141  6.104389 12.947492 16.417915
## 3               g__Mogibacterium 9_AA vs 14_BB  Nonsignif 1.378148e-04    0.004593825  4.555227  8.826604 14.531783
## 4                  g__Romboutsia 9_AA vs 14_BB         BB 4.056005e-04    0.010140012  4.573970 13.481432 12.506215
## 5                 g__Eubacterium 9_AA vs 14_BB  Nonsignif 6.697887e-04    0.013395774 -3.297185  8.720115 11.571264
## 6              g__Parasutterella 9_AA vs 14_BB         AA 2.038985e-03    0.033983079 -3.214725 11.156853  9.514105
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                         -3.033423                      56                   24                196.5
## 2                                NA                      11                    0                 39.5
## 3                                NA                       0                    0                  0.0
## 4                         -7.036174                     107                    2                262.5
## 5                                NA                       0                    0                  0.0
## 6                          4.321928                       7                   50                  2.5
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       -5.090571            791.260870         37.4444444        1275.857143                    86.96
## 2                       -4.884638            257.217391         14.0000000         413.571429                    60.87
## 3                       -8.198620             19.913043          0.1111111          32.642857                    30.43
## 4                       -3.131606            296.478261         51.7777778         453.785714                    78.26
## 5                        2.440393              9.173913         18.2222222           3.357143                    30.43
## 6                        4.373725            167.565217        398.3333333          19.214286                    65.22
##   Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                 77.78                 92.86 1.8e+08 (1.8e+08;1.8e+08)
## 2                 22.22                 85.71       11000 (11000;11000)
## 3                 11.11                 42.86    2e+283 (2e+283;2e+283)
## 4                 55.56                 92.86                37 (44;30)
## 5                 44.44                 21.43           0.34 (-1.8;2.5)
## 6                 88.89                 50.00         0.0066 (-9.8;9.9)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>logCPM</strong>: is the average expression of all samples for that particular gene across all samples on the log-scale expressed in counts per million (cpm, as calculated by edgeR after normalization);</p></li>
<li><p><strong>LR</strong>: the signed likelihood ratio test statistic;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for EdgeR</strong>
</summary>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="differential-analysis.html#cb307-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb307-2"><a href="differential-analysis.html#cb307-2" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb307-3"><a href="differential-analysis.html#cb307-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb307-4"><a href="differential-analysis.html#cb307-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb307-5"><a href="differential-analysis.html#cb307-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb307-6"><a href="differential-analysis.html#cb307-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb307-7"><a href="differential-analysis.html#cb307-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-8"><a href="differential-analysis.html#cb307-8" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb307-9"><a href="differential-analysis.html#cb307-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb307-10"><a href="differential-analysis.html#cb307-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb307-11"><a href="differential-analysis.html#cb307-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb307-12"><a href="differential-analysis.html#cb307-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb307-13"><a href="differential-analysis.html#cb307-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-14"><a href="differential-analysis.html#cb307-14" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb307-15"><a href="differential-analysis.html#cb307-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb307-16"><a href="differential-analysis.html#cb307-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb307-17"><a href="differential-analysis.html#cb307-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb307-18"><a href="differential-analysis.html#cb307-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;edger&quot;</span>)</span>
<span id="cb307-19"><a href="differential-analysis.html#cb307-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-20"><a href="differential-analysis.html#cb307-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span>
<span id="cb307-21"><a href="differential-analysis.html#cb307-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="ancom" class="section level3 hasAnchor" number="11.1.12">
<h3><span class="header-section-number">11.1.12</span> ANCOM<a href="differential-analysis.html#ancom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ANCOM</strong> (Analysis of composition of microbiomes) is from <em>Analysis of composition of microbiomes: a novel method for studying microbial composition”, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-mandal2015analysis" role="doc-biblioref">Mandal et al. 2015</a>)</span>. ANCOM makes no distributional assumptions and can be implemented in a linear model framework. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="differential-analysis.html#cb308-1" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb308-2"><a href="differential-analysis.html#cb308-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb308-3"><a href="differential-analysis.html#cb308-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb308-4"><a href="differential-analysis.html#cb308-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-172"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-172-1.png" alt="ANCOM (Structure Zero)" width="480" />
<p class="caption">
Figure 11.2: ANCOM (Structure Zero)
</p>
</div>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="differential-analysis.html#cb309-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;(W)q-values &lt; alpha&quot;               &quot;W_ratio&quot;                          
##  [7] &quot;detected_0.7&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="differential-analysis.html#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment  EffectSize (W)q-values &lt; alpha W_ratio detected_0.7
## 1 g__Acidaminococcus 9_AA vs 14_BB         BB  0.02320439                   0       0        FALSE
## 2     g__Actinomyces 9_AA vs 14_BB         BB  0.73520463                   0       0        FALSE
## 3   g__Adlercreutzia 9_AA vs 14_BB         BB  0.30465472                   0       0        FALSE
## 4    g__Agathobacter 9_AA vs 14_BB         AA -0.48457752                   0       0        FALSE
## 5     g__Akkermansia 9_AA vs 14_BB         AA -0.07760459                   0       0        FALSE
## 6       g__Alistipes 9_AA vs 14_BB         AA -0.01131933                   0       0        FALSE
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA                       0                    0                  0.0
## 2                        -0.9577718                      30                   26                 50.5
## 3                                NA                       0                    0                  9.0
## 4                         0.1524904                     316                  329                296.0
## 5                                NA                       0                    0                  0.0
## 6                        -0.1085245                      64                   64                 69.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       1.7051442              58.82609         101.777778           31.21429                    21.74
## 2                      -1.3107676              50.91304          26.777778           66.42857                    82.61
## 3                      -2.4683647              19.21739           5.111111           28.28571                    43.48
## 4                       1.7488297             996.26087        1740.444444          517.85714                    65.22
## 5                      -2.1676332              93.78261          30.000000          134.78571                    21.74
## 6                       0.2000385             163.86957         177.888889          154.85714                    73.91
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                 88.89                 78.57       4.1 (6.8;1.3)
## 3                 33.33                 50.00        8.6 (13;4.3)
## 4                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                 22.22                 21.43      1.5 (2.3;0.71)
## 6                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size by <strong>ANCOM</strong>;</p></li>
<li><p><strong>(W)q-values &lt; alpha</strong>: q-values less than alpha;</p></li>
<li><p><strong>W_ratio</strong>: the ratio of W values;</p></li>
<li><p><strong>detected_0.7</strong>: W_ratio more than 0.7;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for ANCOM</strong>
</summary>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="differential-analysis.html#cb313-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb313-2"><a href="differential-analysis.html#cb313-2" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb313-3"><a href="differential-analysis.html#cb313-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb313-4"><a href="differential-analysis.html#cb313-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb313-5"><a href="differential-analysis.html#cb313-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb313-6"><a href="differential-analysis.html#cb313-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb313-7"><a href="differential-analysis.html#cb313-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb313-8"><a href="differential-analysis.html#cb313-8" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb313-9"><a href="differential-analysis.html#cb313-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb313-10"><a href="differential-analysis.html#cb313-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb313-11"><a href="differential-analysis.html#cb313-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb313-12"><a href="differential-analysis.html#cb313-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb313-13"><a href="differential-analysis.html#cb313-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb313-14"><a href="differential-analysis.html#cb313-14" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb313-15"><a href="differential-analysis.html#cb313-15" aria-hidden="true" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb313-16"><a href="differential-analysis.html#cb313-16" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb313-17"><a href="differential-analysis.html#cb313-17" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb313-18"><a href="differential-analysis.html#cb313-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;ancom&quot;</span>)</span>
<span id="cb313-19"><a href="differential-analysis.html#cb313-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb313-20"><a href="differential-analysis.html#cb313-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb313-21"><a href="differential-analysis.html#cb313-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="corncob" class="section level3 hasAnchor" number="11.1.13">
<h3><span class="header-section-number">11.1.13</span> Corncob<a href="differential-analysis.html#corncob" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Corncob</strong> package is from <em>Modeling microbial abundances and dysbiosis with beta-binomial regression”, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-martin2020modeling" role="doc-biblioref">Martin, Witten, and Willis 2020</a>)</span>. Corncob is based on beta-binomial regression. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="differential-analysis.html#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb314-2"><a href="differential-analysis.html#cb314-2" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb314-3"><a href="differential-analysis.html#cb314-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb314-4"><a href="differential-analysis.html#cb314-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb314-5"><a href="differential-analysis.html#cb314-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb314-6"><a href="differential-analysis.html#cb314-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb314-7"><a href="differential-analysis.html#cb314-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-8"><a href="differential-analysis.html#cb314-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span>
<span id="cb314-9"><a href="differential-analysis.html#cb314-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb314-10"><a href="differential-analysis.html#cb314-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>other options for Corncob</strong>
</summary>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="differential-analysis.html#cb315-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb315-2"><a href="differential-analysis.html#cb315-2" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb315-3"><a href="differential-analysis.html#cb315-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb315-4"><a href="differential-analysis.html#cb315-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb315-5"><a href="differential-analysis.html#cb315-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb315-6"><a href="differential-analysis.html#cb315-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb315-7"><a href="differential-analysis.html#cb315-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb315-8"><a href="differential-analysis.html#cb315-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb315-9"><a href="differential-analysis.html#cb315-9" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb315-10"><a href="differential-analysis.html#cb315-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps,</span>
<span id="cb315-11"><a href="differential-analysis.html#cb315-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb315-12"><a href="differential-analysis.html#cb315-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb315-13"><a href="differential-analysis.html#cb315-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb315-14"><a href="differential-analysis.html#cb315-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb315-15"><a href="differential-analysis.html#cb315-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb315-16"><a href="differential-analysis.html#cb315-16" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb315-17"><a href="differential-analysis.html#cb315-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb315-18"><a href="differential-analysis.html#cb315-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb315-19"><a href="differential-analysis.html#cb315-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb315-20"><a href="differential-analysis.html#cb315-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">da_method =</span> <span class="st">&quot;corncob&quot;</span>,</span>
<span id="cb315-21"><a href="differential-analysis.html#cb315-21" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb315-22"><a href="differential-analysis.html#cb315-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb315-23"><a href="differential-analysis.html#cb315-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_corncob)</span>
<span id="cb315-24"><a href="differential-analysis.html#cb315-24" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models" class="section level3 hasAnchor" number="11.1.14">
<h3><span class="header-section-number">11.1.14</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="differential-analysis.html#cb316-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb316-2"><a href="differential-analysis.html#cb316-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb316-3"><a href="differential-analysis.html#cb316-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb316-4"><a href="differential-analysis.html#cb316-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb316-5"><a href="differential-analysis.html#cb316-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb316-6"><a href="differential-analysis.html#cb316-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb316-7"><a href="differential-analysis.html#cb316-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb316-8"><a href="differential-analysis.html#cb316-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb316-9"><a href="differential-analysis.html#cb316-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb316-10"><a href="differential-analysis.html#cb316-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-11"><a href="differential-analysis.html#cb316-11" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb316-12"><a href="differential-analysis.html#cb316-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps,</span>
<span id="cb316-13"><a href="differential-analysis.html#cb316-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb316-14"><a href="differential-analysis.html#cb316-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb316-15"><a href="differential-analysis.html#cb316-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb316-16"><a href="differential-analysis.html#cb316-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb316-17"><a href="differential-analysis.html#cb316-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb316-18"><a href="differential-analysis.html#cb316-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb316-19"><a href="differential-analysis.html#cb316-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb316-20"><a href="differential-analysis.html#cb316-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-21"><a href="differential-analysis.html#cb316-21" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb316-22"><a href="differential-analysis.html#cb316-22" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb316-23"><a href="differential-analysis.html#cb316-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb316-24"><a href="differential-analysis.html#cb316-24" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb316-25"><a href="differential-analysis.html#cb316-25" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb316-26"><a href="differential-analysis.html#cb316-26" aria-hidden="true" tabindex="-1"></a>                  <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb316-27"><a href="differential-analysis.html#cb316-27" aria-hidden="true" tabindex="-1"></a>                  <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb316-28"><a href="differential-analysis.html#cb316-28" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb316-29"><a href="differential-analysis.html#cb316-29" aria-hidden="true" tabindex="-1"></a>                  <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb316-30"><a href="differential-analysis.html#cb316-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-31"><a href="differential-analysis.html#cb316-31" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb316-32"><a href="differential-analysis.html#cb316-32" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb316-33"><a href="differential-analysis.html#cb316-33" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb316-34"><a href="differential-analysis.html#cb316-34" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb316-35"><a href="differential-analysis.html#cb316-35" aria-hidden="true" tabindex="-1"></a>                  <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb316-36"><a href="differential-analysis.html#cb316-36" aria-hidden="true" tabindex="-1"></a>                  <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb316-37"><a href="differential-analysis.html#cb316-37" aria-hidden="true" tabindex="-1"></a>                  <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb316-38"><a href="differential-analysis.html#cb316-38" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb316-39"><a href="differential-analysis.html#cb316-39" aria-hidden="true" tabindex="-1"></a>                  <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb316-40"><a href="differential-analysis.html#cb316-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-41"><a href="differential-analysis.html#cb316-41" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2)</span>
<span id="cb316-42"><a href="differential-analysis.html#cb316-42" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="locom" class="section level3 hasAnchor" number="11.1.15">
<h3><span class="header-section-number">11.1.15</span> LOCOM<a href="differential-analysis.html#locom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>LOCOM: A logistic regression model for testing differential abundance in compositional microbiome data
with false discovery rate control <span class="citation">(<a href="#ref-hu2022locom" role="doc-biblioref">Hu, Satten, and Hu 2022</a>)</span>.</p>
<p><strong>Cautions</strong>: It reports an error information which the matrix is <strong>singularError</strong> when user uses low <em>prev.cut</em>.</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="differential-analysis.html#cb317-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(permute)</span>
<span id="cb317-2"><a href="differential-analysis.html#cb317-2" aria-hidden="true" tabindex="-1"></a>DA_locom <span class="ot">&lt;-</span> <span class="fu">run_locom</span>(<span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb317-3"><a href="differential-analysis.html#cb317-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb317-4"><a href="differential-analysis.html#cb317-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb317-5"><a href="differential-analysis.html#cb317-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fdr.nominal =</span> <span class="fl">0.2</span>,</span>
<span id="cb317-6"><a href="differential-analysis.html#cb317-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prev.cut =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<pre><code>## Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred

## Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred

## Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
<pre><code>## 88 OTU(s) with fewer than 18.4 in all samples are removed 
## permutations: 1</code></pre>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="differential-analysis.html#cb320-1" aria-hidden="true" tabindex="-1"></a>DA_locom</span></code></pre></div>
<pre><code>##                 TaxaID         Block EffectSize Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1     g__Lactobacillus 9_AA vs 14_BB      3.180  0.022          0.114                        -3.0334230
## 2       g__Bacteroides 9_AA vs 14_BB     -1.030  0.024          0.114                         1.5367266
## 3 g__Lachnoclostridium 9_AA vs 14_BB     -0.672  0.032          0.114                         0.7874146
## 4           g__Blautia 9_AA vs 14_BB     -0.570  0.038          0.114                         0.2641697
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                      56                   24                196.5                      -5.0905713              791.2609
## 2                    3807                 9103               3137.5                       0.9841180             5745.5652
## 3                     422                  485                281.0                       0.4658394              417.9565
## 4                    8199                 8645               7198.5                       0.3178904             9145.9565
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1           37.44444          1275.8571                    86.96                 77.78                 92.86
## 2         8219.44444          4155.2143                   100.00                100.00                100.00
## 3          502.33333           363.7143                    95.65                100.00                 92.86
## 4        10397.55556          8341.3571                   100.00                100.00                100.00
##         Odds Ratio (95% CI)
## 1 1.8e+08 (1.8e+08;1.8e+08)
## 2             0.45 (-1.1;2)
## 3          0.63 (-0.27;1.5)
## 4           0.62 (-0.3;1.5)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>EffectSize</strong>: effect size by <strong>ANCOM</strong>;</p></li>
<li><p><strong>Pvalue</strong>: p-values for OTU-specific tests;</p></li>
<li><p><strong>AdjustedPvalue </strong>: q-values (adjusted p-values by the BH procedure) for OTU-specific tests;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
</div>
</div>
<div id="metagenomic-sequencing-microbial-data-metaphlan23" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Metagenomic sequencing microbial data (metaphlan2/3)<a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same strategy to filter phyloseq object.</p>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="differential-analysis.html#cb322-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;metaphlan2_ps&quot;</span>)</span>
<span id="cb322-2"><a href="differential-analysis.html#cb322-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb322-3"><a href="differential-analysis.html#cb322-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb322-4"><a href="differential-analysis.html#cb322-4" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb322-5"><a href="differential-analysis.html#cb322-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb322-6"><a href="differential-analysis.html#cb322-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb322-7"><a href="differential-analysis.html#cb322-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb322-8"><a href="differential-analysis.html#cb322-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb322-9"><a href="differential-analysis.html#cb322-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb322-10"><a href="differential-analysis.html#cb322-10" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb322-11"><a href="differential-analysis.html#cb322-11" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> metaphlan2_ps_remove_BRS, </span>
<span id="cb322-12"><a href="differential-analysis.html#cb322-12" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb322-13"><a href="differential-analysis.html#cb322-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb322-14"><a href="differential-analysis.html#cb322-14" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb322-15"><a href="differential-analysis.html#cb322-15" aria-hidden="true" tabindex="-1"></a><span class="co"># dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(ps = dada2_ps_genus, </span></span>
<span id="cb322-16"><a href="differential-analysis.html#cb322-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                                                   cutoff = 10, </span></span>
<span id="cb322-17"><a href="differential-analysis.html#cb322-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                                                   unclass = TRUE)</span></span>
<span id="cb322-18"><a href="differential-analysis.html#cb322-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb322-19"><a href="differential-analysis.html#cb322-19" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb322-20"><a href="differential-analysis.html#cb322-20" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> metaphlan2_ps_genus, </span>
<span id="cb322-21"><a href="differential-analysis.html#cb322-21" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">cutoff =</span> <span class="fl">1e-4</span>, </span>
<span id="cb322-22"><a href="differential-analysis.html#cb322-22" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb322-23"><a href="differential-analysis.html#cb322-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb322-24"><a href="differential-analysis.html#cb322-24" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb322-25"><a href="differential-analysis.html#cb322-25" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> metaphlan2_ps_genus_filter, </span>
<span id="cb322-26"><a href="differential-analysis.html#cb322-26" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">cutoff =</span> <span class="fl">0.2</span>, </span>
<span id="cb322-27"><a href="differential-analysis.html#cb322-27" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb322-28"><a href="differential-analysis.html#cb322-28" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 54 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 54 taxa by 6 taxonomic ranks ]</code></pre>
<div id="wilcoxon-rank-sum-and-signed-rank-tests-1" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparameter test methods, use the rank of taxa abundance to find the significant taxa.</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="differential-analysis.html#cb324-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb324-2"><a href="differential-analysis.html#cb324-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb324-3"><a href="differential-analysis.html#cb324-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb324-4"><a href="differential-analysis.html#cb324-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb324-5"><a href="differential-analysis.html#cb324-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="differential-analysis.html#cb326-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic     Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 7_AA vs 15_BB  Nonsignif  0.9253506      36.0 0.19074499      0.4830467                                NA
## 2     g__Actinomyces 7_AA vs 15_BB  Nonsignif  4.3383273      25.0 0.04564844      0.1896166                                NA
## 3   g__Adlercreutzia 7_AA vs 15_BB  Nonsignif  0.4045684      28.0 0.05769072      0.2076866                                NA
## 4       g__Alistipes 7_AA vs 15_BB  Nonsignif  1.3860231      71.0 0.20193465      0.4830467                          2.061604
## 5    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.5796404      31.0 0.13834560      0.3931928                         -4.964322
## 6   g__Anaerotruncus 7_AA vs 15_BB  Nonsignif  0.8015454      47.5 0.71288708      0.8019980                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1              0.00000000            0.0000000            0.0000000                      -0.4631577                    9.14
## 2              0.00000280            0.0000000            0.0000061                      -0.8163116                    7.57
## 3              0.00000000            0.0000000            0.0000558                      -0.7147950                    8.00
## 4              0.00711780            0.0236665            0.0056693                       0.4613459                   14.14
## 5              0.00031435            0.0000229            0.0007149                      -0.6171177                    8.43
## 6              0.00000000            0.0000000            0.0000000                      -0.1327552                   10.79
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   12.60                    36.36                 14.29                 46.67    8700 (8700;8700)
## 2                   13.33                    54.55                 14.29                 73.33      1.7 (2.7;0.66)
## 3                   13.13                    40.91                 14.29                 53.33       Inf (Inf;Inf)
## 4                   10.27                    77.27                 71.43                 80.00     0.22 (-2.7;3.2)
## 5                   12.93                    86.36                 71.43                 93.33         6 (9.5;2.5)
## 6                   11.83                    36.36                 28.57                 40.00      1.3 (1.8;0.78)</code></pre>
<details>
<summary>
<strong>wilcox Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="differential-analysis.html#cb328-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb328-2"><a href="differential-analysis.html#cb328-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb328-3"><a href="differential-analysis.html#cb328-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb328-4"><a href="differential-analysis.html#cb328-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb328-5"><a href="differential-analysis.html#cb328-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb328-6"><a href="differential-analysis.html#cb328-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb328-7"><a href="differential-analysis.html#cb328-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb328-8"><a href="differential-analysis.html#cb328-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-1" class="section level3 hasAnchor" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="differential-analysis.html#cb329-1" aria-hidden="true" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb329-2"><a href="differential-analysis.html#cb329-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb329-3"><a href="differential-analysis.html#cb329-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb329-4"><a href="differential-analysis.html#cb329-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb329-5"><a href="differential-analysis.html#cb329-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb329-6"><a href="differential-analysis.html#cb329-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">2</span>)  </span>
<span id="cb329-7"><a href="differential-analysis.html#cb329-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb329-8"><a href="differential-analysis.html#cb329-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;LDA_Score&quot;                         &quot;EffectSize&quot;                        &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
##  [7] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [10] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [13] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [16] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="differential-analysis.html#cb331-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##              TaxaID         Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1    g__Bacteroides 7_AA vs 15_BB         AA -5.081345  2.5463041                          1.190750            2.541675e+05
## 2  g__Fusobacterium 7_AA vs 15_BB         AA -4.125358  2.5168454                                NA            0.000000e+00
## 3    g__Actinomyces 7_AA vs 15_BB         BB  2.233078  1.2263822                                NA            2.866622e+00
## 4 g__Granulicatella 7_AA vs 15_BB         BB  2.343080  0.1976324                                NA            2.566255e+01
## 5    g__Eggerthella 7_AA vs 15_BB         BB  3.145393  1.3914213                                NA            5.913424e+02
## 6          g__Dorea 7_AA vs 15_BB         BB  3.830972  2.1498789                         -5.725323            8.554293e+03
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1          464614.8499         2.035362e+05                        1.141368          291673.92734       464989.41139
## 2             123.8325         0.000000e+00                        6.941519            9891.91278        30556.16386
## 3               0.0000         7.245027e+00                       -1.246758              29.78669           15.38419
## 4               0.0000         5.235042e+01                       -3.071984              64.14510           10.59936
## 5               0.0000         7.714712e+02                       -3.764280            2231.98721          232.91482
## 6             270.5150         1.431151e+04                       -3.454344           11339.49845         1455.31524
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1       210793.36812                   100.00                100.00                100.00     0.32 (-1.9;2.5)
## 2          248.59562                    40.91                 85.71                 20.00     0.0032 (-11;11)
## 3           36.50785                    54.55                 14.29                 73.33      1.7 (2.8;0.66)
## 4           89.13311                    72.73                 42.86                 86.67       220 (230;210)
## 5         3164.88766                    63.64                 42.86                 73.33       430 (450;420)
## 6        15952.11728                    90.91                 71.43                100.00          28 (34;21)</code></pre>
<details>
<summary>
<strong>lefse Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="differential-analysis.html#cb333-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb333-2"><a href="differential-analysis.html#cb333-2" aria-hidden="true" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb333-3"><a href="differential-analysis.html#cb333-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb333-4"><a href="differential-analysis.html#cb333-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb333-5"><a href="differential-analysis.html#cb333-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb333-6"><a href="differential-analysis.html#cb333-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb333-7"><a href="differential-analysis.html#cb333-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb333-8"><a href="differential-analysis.html#cb333-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb333-9"><a href="differential-analysis.html#cb333-9" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test-1" class="section level3 hasAnchor" number="11.2.3">
<h3><span class="header-section-number">11.2.3</span> t-test<a href="differential-analysis.html#t-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="differential-analysis.html#cb334-1" aria-hidden="true" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb334-2"><a href="differential-analysis.html#cb334-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb334-3"><a href="differential-analysis.html#cb334-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb334-4"><a href="differential-analysis.html#cb334-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb334-5"><a href="differential-analysis.html#cb334-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                   
##  [3] &quot;Enrichment&quot;                               &quot;EffectSize&quot;                              
##  [5] &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;       
##  [9] &quot;Median Abundance\n(All)&quot;                  &quot;Median Abundance\nAA&quot;                    
## [11] &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;             
## [15] &quot;GeometricMean Abundance\nBB&quot;              &quot;Occurrence (100%)\n(All)&quot;                
## [17] &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="differential-analysis.html#cb336-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue
## 1 g__Acidaminococcus 7_AA vs 15_BB  Nonsignif  0.9253506 -1.0515590 0.31074638      0.5630606
## 2     g__Actinomyces 7_AA vs 15_BB  Nonsignif  4.3383273 -1.0131776 0.32823254      0.5717599
## 3   g__Adlercreutzia 7_AA vs 15_BB  Nonsignif  0.4045684 -2.1929042 0.04570343      0.2243623
## 4       g__Alistipes 7_AA vs 15_BB  Nonsignif  1.3860231  1.5963681 0.15951558      0.5060914
## 5    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.5796404 -1.6970563 0.10937430      0.4218723
## 6   g__Anaerotruncus 7_AA vs 15_BB  Nonsignif  0.8015454 -0.5871152 0.56369794      0.6617324
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA              0.00000000            0.0000000            0.0000000
## 2                                NA              0.00000280            0.0000000            0.0000061
## 3                                NA              0.00000000            0.0000000            0.0000558
## 4                          2.061604              0.00711780            0.0236665            0.0056693
## 5                         -4.964322              0.00031435            0.0000229            0.0007149
## 6                                NA              0.00000000            0.0000000            0.0000000
##   Log2FoldChange (geometricmean)\nAA_vs_BB GeometricMean Abundance\n(All) GeometricMean Abundance\nAA
## 1                                       NA                      4.4722826                         NaN
## 2                               1.54124914                      0.6816828                   1.7031944
## 3                                       NA                      1.5989554                         NaN
## 4                               2.05528762                      0.3989446                   0.7346341
## 5                                       NA                      1.8882218                         NaN
## 6                              -0.05926392                      1.0212861                   0.9903011
##   GeometricMean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   4.4722826                    36.36                 14.29                 46.67    8700 (8700;8700)
## 2                   0.5851969                    54.55                 14.29                 73.33      1.7 (2.7;0.66)
## 3                   1.5989554                    40.91                 14.29                 53.33       Inf (Inf;Inf)
## 4                   0.1767534                    77.27                 71.43                 80.00     0.22 (-2.7;3.2)
## 5                   1.8882218                    86.36                 71.43                 93.33         6 (9.5;2.5)
## 6                   1.0318284                    36.36                 28.57                 40.00      1.3 (1.8;0.78)</code></pre>
<details>
<summary>
<strong>ttest Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="differential-analysis.html#cb338-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb338-2"><a href="differential-analysis.html#cb338-2" aria-hidden="true" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb338-3"><a href="differential-analysis.html#cb338-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb338-4"><a href="differential-analysis.html#cb338-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb338-5"><a href="differential-analysis.html#cb338-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb338-6"><a href="differential-analysis.html#cb338-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb338-7"><a href="differential-analysis.html#cb338-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models-1" class="section level3 hasAnchor" number="11.2.4">
<h3><span class="header-section-number">11.2.4</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="differential-analysis.html#cb339-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb339-2"><a href="differential-analysis.html#cb339-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb339-3"><a href="differential-analysis.html#cb339-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb339-4"><a href="differential-analysis.html#cb339-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb339-5"><a href="differential-analysis.html#cb339-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb339-6"><a href="differential-analysis.html#cb339-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb339-7"><a href="differential-analysis.html#cb339-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb339-8"><a href="differential-analysis.html#cb339-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb339-9"><a href="differential-analysis.html#cb339-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb339-10"><a href="differential-analysis.html#cb339-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb339-11"><a href="differential-analysis.html#cb339-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2_mgs)</span>
<span id="cb339-12"><a href="differential-analysis.html#cb339-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>maaslin2 Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="differential-analysis.html#cb340-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb340-2"><a href="differential-analysis.html#cb340-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb340-3"><a href="differential-analysis.html#cb340-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb340-4"><a href="differential-analysis.html#cb340-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb340-5"><a href="differential-analysis.html#cb340-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb340-6"><a href="differential-analysis.html#cb340-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb340-7"><a href="differential-analysis.html#cb340-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb340-8"><a href="differential-analysis.html#cb340-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb340-9"><a href="differential-analysis.html#cb340-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb340-10"><a href="differential-analysis.html#cb340-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb340-11"><a href="differential-analysis.html#cb340-11" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
</div>
<div id="visualization-4" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Visualization<a href="differential-analysis.html#visualization-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Volcano plot is used to display differential analysis. <code>plot_volcano</code> provides multiple parameters for plotting volcano. More details to see <code>help(plot_volcano)</code>.</p>
<p>The barplot is used to display the lefse results.</p>
<div id="volcano-plot" class="section level3 hasAnchor" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Volcano plot<a href="differential-analysis.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The X and Y coordinate axis are flexible to choose for Volcano. Here, we choose <em>logFC</em> and <em>AdjustedPvalue</em> to visualize the results</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="differential-analysis.html#cb341-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb341-2"><a href="differential-analysis.html#cb341-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb341-3"><a href="differential-analysis.html#cb341-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb341-4"><a href="differential-analysis.html#cb341-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb341-5"><a href="differential-analysis.html#cb341-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb341-6"><a href="differential-analysis.html#cb341-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="differential-analysis.html#cb343-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_volcano <span class="ot">&lt;-</span> <span class="fu">plot_volcano</span>(</span>
<span id="cb343-2"><a href="differential-analysis.html#cb343-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">da_res =</span> DA_limma_voom,</span>
<span id="cb343-3"><a href="differential-analysis.html#cb343-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb343-4"><a href="differential-analysis.html#cb343-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">x_index =</span> <span class="st">&quot;logFC&quot;</span>,</span>
<span id="cb343-5"><a href="differential-analysis.html#cb343-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb343-6"><a href="differential-analysis.html#cb343-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb343-7"><a href="differential-analysis.html#cb343-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb343-8"><a href="differential-analysis.html#cb343-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb343-9"><a href="differential-analysis.html#cb343-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">topN =</span> <span class="dv">5</span>)</span>
<span id="cb343-10"><a href="differential-analysis.html#cb343-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb343-11"><a href="differential-analysis.html#cb343-11" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_volcano</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-187"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-187-1.png" alt="Volcano (limma-voom: logFC and AdjustedPvalue)" width="672" />
<p class="caption">
Figure 11.3: Volcano (limma-voom: logFC and AdjustedPvalue)
</p>
</div>
<details>
<summary>
<strong>DA_limma_voom_volcano effectsize</strong>
</summary>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="differential-analysis.html#cb344-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb344-2"><a href="differential-analysis.html#cb344-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> DA_limma_voom,</span>
<span id="cb344-3"><a href="differential-analysis.html#cb344-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb344-4"><a href="differential-analysis.html#cb344-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;EffectSize&quot;</span>,</span>
<span id="cb344-5"><a href="differential-analysis.html#cb344-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb344-6"><a href="differential-analysis.html#cb344-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb344-7"><a href="differential-analysis.html#cb344-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb344-8"><a href="differential-analysis.html#cb344-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb344-9"><a href="differential-analysis.html#cb344-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">8</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-188"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-188-1.png" alt="Volcano (limma-voom: EffectSize and Pvalue)" width="672" />
<p class="caption">
Figure 11.4: Volcano (limma-voom: EffectSize and Pvalue)
</p>
</div>
<p>The logFC and EffectSize are the same values in limma-voom.</p>
</details>
</div>
<div id="barplot-in-lefse" class="section level3 hasAnchor" number="11.3.2">
<h3><span class="header-section-number">11.3.2</span> barplot in lefse<a href="differential-analysis.html#barplot-in-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="differential-analysis.html#cb345-1" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb345-2"><a href="differential-analysis.html#cb345-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb345-3"><a href="differential-analysis.html#cb345-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb345-4"><a href="differential-analysis.html#cb345-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb345-5"><a href="differential-analysis.html#cb345-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb345-6"><a href="differential-analysis.html#cb345-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">Lda =</span> <span class="dv">2</span>)</span>
<span id="cb345-7"><a href="differential-analysis.html#cb345-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-8"><a href="differential-analysis.html#cb345-8" aria-hidden="true" tabindex="-1"></a><span class="co"># # don&#39;t run this code when you do lefse in reality</span></span>
<span id="cb345-9"><a href="differential-analysis.html#cb345-9" aria-hidden="true" tabindex="-1"></a><span class="co"># DA_lefse$LDA_Score &lt;- DA_lefse$LDA_Score * 1000</span></span>
<span id="cb345-10"><a href="differential-analysis.html#cb345-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-11"><a href="differential-analysis.html#cb345-11" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb345-12"><a href="differential-analysis.html#cb345-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">da_res =</span> DA_lefse,</span>
<span id="cb345-13"><a href="differential-analysis.html#cb345-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb345-14"><a href="differential-analysis.html#cb345-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb345-15"><a href="differential-analysis.html#cb345-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>)) </span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-189"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-189-1.png" alt="Barplot (Lefse)" width="576" />
<p class="caption">
Figure 11.5: Barplot (Lefse)
</p>
</div>
</div>
</div>
<div id="dominant-taxa" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Dominant taxa<a href="differential-analysis.html#dominant-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Display the significant taxa with selection using boxplot.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="differential-analysis.html#cb346-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb346-2"><a href="differential-analysis.html#cb346-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb346-3"><a href="differential-analysis.html#cb346-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb346-4"><a href="differential-analysis.html#cb346-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb346-5"><a href="differential-analysis.html#cb346-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb346-6"><a href="differential-analysis.html#cb346-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_topN_boxplot</span>(</span>
<span id="cb346-7"><a href="differential-analysis.html#cb346-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb346-8"><a href="differential-analysis.html#cb346-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> DA_wilcox,</span>
<span id="cb346-9"><a href="differential-analysis.html#cb346-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Median)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb346-10"><a href="differential-analysis.html#cb346-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.2</span>,</span>
<span id="cb346-11"><a href="differential-analysis.html#cb346-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb346-12"><a href="differential-analysis.html#cb346-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.3</span>,</span>
<span id="cb346-13"><a href="differential-analysis.html#cb346-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">3</span>,</span>
<span id="cb346-14"><a href="differential-analysis.html#cb346-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">taxa_name =</span> <span class="st">&quot;s__Ruminococcus_torques&quot;</span>,</span>
<span id="cb346-15"><a href="differential-analysis.html#cb346-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-190"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-190-1.png" alt="Dominant Taxa" width="768" />
<p class="caption">
Figure 11.6: Dominant Taxa
</p>
</div>
</div>
<div id="multiple-differential-analysis-by-one-function" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> Multiple differential analysis by one function<a href="differential-analysis.html#multiple-differential-analysis-by-one-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>here, we provide the <code>run_multiple_da</code> for obtaining the results list from multiple differential analysis methods.</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="differential-analysis.html#cb347-1" aria-hidden="true" tabindex="-1"></a>multiple_res <span class="ot">&lt;-</span> <span class="fu">run_multiple_da</span>(</span>
<span id="cb347-2"><a href="differential-analysis.html#cb347-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb347-3"><a href="differential-analysis.html#cb347-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb347-4"><a href="differential-analysis.html#cb347-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb347-5"><a href="differential-analysis.html#cb347-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">da_method =</span> <span class="fu">c</span>(<span class="st">&quot;aldex&quot;</span>, <span class="st">&quot;limma_voom&quot;</span>, <span class="st">&quot;mbzinb&quot;</span>, <span class="st">&quot;omnibus&quot;</span>),</span>
<span id="cb347-6"><a href="differential-analysis.html#cb347-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">p_adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|
## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  92  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="differential-analysis.html#cb349-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(multiple_res)</span></code></pre></div>
<pre><code>## [1] &quot;aldex&quot;      &quot;limma_voom&quot; &quot;mbzinb&quot;     &quot;omnibus&quot;</code></pre>
<ul>
<li>plot results</li>
</ul>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="differential-analysis.html#cb351-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_multiple_DA</span>(</span>
<span id="cb351-2"><a href="differential-analysis.html#cb351-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">Multip_DA_res =</span> multiple_res,</span>
<span id="cb351-3"><a href="differential-analysis.html#cb351-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_list =</span> <span class="fu">c</span>(<span class="st">&quot;EffectSize&quot;</span>, <span class="st">&quot;logFC&quot;</span>, <span class="st">&quot;mean.LFC&quot;</span>, <span class="st">&quot;abund.LFC.CompvarBB.est&quot;</span>),</span>
<span id="cb351-4"><a href="differential-analysis.html#cb351-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb351-5"><a href="differential-analysis.html#cb351-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;AdjustedPvalue&quot;</span>,</span>
<span id="cb351-6"><a href="differential-analysis.html#cb351-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb351-7"><a href="differential-analysis.html#cb351-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">cellwidth =</span> <span class="dv">50</span>,</span>
<span id="cb351-8"><a href="differential-analysis.html#cb351-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">cellheight =</span> <span class="dv">10</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-192"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-192-1.png" alt="Multiple DA results" width="768" />
<p class="caption">
Figure 11.7: Multiple DA results
</p>
</div>
</div>
<div id="comparing-outputs-from-xmas2-lefse-conda-and-lefse-galaxy-using-the-same-in-house-datasets-amplicon_pszeybel_gut" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> Comparing outputs from XMAS2, lefse-conda, and lefse-galaxy using the same in-house datasets (amplicon_ps|Zeybel_Gut)<a href="differential-analysis.html#comparing-outputs-from-xmas2-lefse-conda-and-lefse-galaxy-using-the-same-in-house-datasets-amplicon_pszeybel_gut" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="introduction-1" class="section level3 hasAnchor" number="11.6.1">
<h3><span class="header-section-number">11.6.1</span> Introduction<a href="differential-analysis.html#introduction-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this document, Comparing the output from lefse through different applications:</p>
<ul>
<li>XMAS2 (R package)</li>
<li>lefse-conda (command line)</li>
<li>lefse-galaxy (from the galaxy platfrom)</li>
</ul>
<p>In all cases, using the same dataset, amplicon_ps and Zeybel_Gut, which are included in the XMAS package.</p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="differential-analysis.html#cb352-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb352-2"><a href="differential-analysis.html#cb352-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb352-3"><a href="differential-analysis.html#cb352-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb352-4"><a href="differential-analysis.html#cb352-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb352-5"><a href="differential-analysis.html#cb352-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb352-6"><a href="differential-analysis.html#cb352-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb352-7"><a href="differential-analysis.html#cb352-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb352-8"><a href="differential-analysis.html#cb352-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb352-9"><a href="differential-analysis.html#cb352-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(VennDiagram)</span>
<span id="cb352-10"><a href="differential-analysis.html#cb352-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb352-11"><a href="differential-analysis.html#cb352-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb352-12"><a href="differential-analysis.html#cb352-12" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb352-13"><a href="differential-analysis.html#cb352-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb352-14"><a href="differential-analysis.html#cb352-14" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="dataset" class="section level3 hasAnchor" number="11.6.2">
<h3><span class="header-section-number">11.6.2</span> Dataset<a href="differential-analysis.html#dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="s-genus" class="section level4 hasAnchor" number="11.6.2.1">
<h4><span class="header-section-number">11.6.2.1</span> 16s genus<a href="differential-analysis.html#s-genus" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="differential-analysis.html#cb353-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;amplicon_ps&quot;</span>)</span>
<span id="cb353-2"><a href="differential-analysis.html#cb353-2" aria-hidden="true" tabindex="-1"></a>amplicon_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(amplicon_ps, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb353-3"><a href="differential-analysis.html#cb353-3" aria-hidden="true" tabindex="-1"></a>amplicon_ps_genus</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 406 taxa and 34 samples ]
## sample_data() Sample Data:       [ 34 samples by 8 sample variables ]
## tax_table()   Taxonomy Table:    [ 406 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="metagenomics-species" class="section level4 hasAnchor" number="11.6.2.2">
<h4><span class="header-section-number">11.6.2.2</span> metagenomics species<a href="differential-analysis.html#metagenomics-species" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="differential-analysis.html#cb355-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;Zeybel_Gut&quot;</span>)</span>
<span id="cb355-2"><a href="differential-analysis.html#cb355-2" aria-hidden="true" tabindex="-1"></a>Zeybel_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(Zeybel_Gut, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb355-3"><a href="differential-analysis.html#cb355-3" aria-hidden="true" tabindex="-1"></a>Zeybel_ps_species</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 372 taxa and 42 samples ]
## sample_data() Sample Data:       [ 42 samples by 51 sample variables ]
## tax_table()   Taxonomy Table:    [ 372 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="preparing-for-lefse-galaxy-and-conda" class="section level4 hasAnchor" number="11.6.2.3">
<h4><span class="header-section-number">11.6.2.3</span> Preparing for lefse galaxy and conda<a href="differential-analysis.html#preparing-for-lefse-galaxy-and-conda" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><p>1st row: class (required)</p></li>
<li><p>2nd row: subclass (optional)</p></li>
<li><p>3rd row: sampleID (required)</p></li>
<li><p>rownames: taxon</p></li>
<li><p>data format: splitted by “</p></li>
</ul>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="differential-analysis.html#cb357-1" aria-hidden="true" tabindex="-1"></a>prepare_lefse <span class="ot">&lt;-</span> <span class="cf">function</span>(ps,</span>
<span id="cb357-2"><a href="differential-analysis.html#cb357-2" aria-hidden="true" tabindex="-1"></a>                          Class,</span>
<span id="cb357-3"><a href="differential-analysis.html#cb357-3" aria-hidden="true" tabindex="-1"></a>                          Class_names,</span>
<span id="cb357-4"><a href="differential-analysis.html#cb357-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Subclass =</span> <span class="cn">NULL</span>,</span>
<span id="cb357-5"><a href="differential-analysis.html#cb357-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">cutoff =</span> <span class="dv">10</span>) {</span>
<span id="cb357-6"><a href="differential-analysis.html#cb357-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-7"><a href="differential-analysis.html#cb357-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ps = amplicon_ps_genus</span></span>
<span id="cb357-8"><a href="differential-analysis.html#cb357-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Class = &quot;SampleType&quot;</span></span>
<span id="cb357-9"><a href="differential-analysis.html#cb357-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Class_names = c(&quot;gut&quot;, &quot;tongue&quot;)</span></span>
<span id="cb357-10"><a href="differential-analysis.html#cb357-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Subclass = NULL</span></span>
<span id="cb357-11"><a href="differential-analysis.html#cb357-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cutoff = 10</span></span>
<span id="cb357-12"><a href="differential-analysis.html#cb357-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-13"><a href="differential-analysis.html#cb357-13" aria-hidden="true" tabindex="-1"></a>    sam_tab <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps) <span class="sc">%&gt;%</span></span>
<span id="cb357-14"><a href="differential-analysis.html#cb357-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>()</span>
<span id="cb357-15"><a href="differential-analysis.html#cb357-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(sam_tab)[<span class="fu">which</span>(<span class="fu">colnames</span>(sam_tab) <span class="sc">==</span> Class)] <span class="ot">&lt;-</span> <span class="st">&quot;CompClass&quot;</span></span>
<span id="cb357-16"><a href="differential-analysis.html#cb357-16" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-17"><a href="differential-analysis.html#cb357-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(Subclass)) {</span>
<span id="cb357-18"><a href="differential-analysis.html#cb357-18" aria-hidden="true" tabindex="-1"></a>        sam_tab_final <span class="ot">&lt;-</span> sam_tab <span class="sc">%&gt;%</span></span>
<span id="cb357-19"><a href="differential-analysis.html#cb357-19" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(CompClass) <span class="sc">%&gt;%</span></span>
<span id="cb357-20"><a href="differential-analysis.html#cb357-20" aria-hidden="true" tabindex="-1"></a>            tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;TempRowNames&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb357-21"><a href="differential-analysis.html#cb357-21" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(CompClass <span class="sc">%in%</span> Class_names) <span class="sc">%&gt;%</span></span>
<span id="cb357-22"><a href="differential-analysis.html#cb357-22" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;TempRowNames&quot;</span>, <span class="st">&quot;CompClass&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb357-23"><a href="differential-analysis.html#cb357-23" aria-hidden="true" tabindex="-1"></a>            tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;TempRowNames&quot;</span>)</span>
<span id="cb357-24"><a href="differential-analysis.html#cb357-24" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb357-25"><a href="differential-analysis.html#cb357-25" aria-hidden="true" tabindex="-1"></a>        sam_tab_final <span class="ot">&lt;-</span> sam_tab <span class="sc">%&gt;%</span></span>
<span id="cb357-26"><a href="differential-analysis.html#cb357-26" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;CompClass&quot;</span>, Subclass))) <span class="sc">%&gt;%</span></span>
<span id="cb357-27"><a href="differential-analysis.html#cb357-27" aria-hidden="true" tabindex="-1"></a>            tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;TempRowNames&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb357-28"><a href="differential-analysis.html#cb357-28" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">filter</span>(CompClass <span class="sc">%in%</span> Class_names) <span class="sc">%&gt;%</span></span>
<span id="cb357-29"><a href="differential-analysis.html#cb357-29" aria-hidden="true" tabindex="-1"></a>            dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;TempRowNames&quot;</span>, <span class="st">&quot;CompClass&quot;</span>, Subclass))) <span class="sc">%&gt;%</span></span>
<span id="cb357-30"><a href="differential-analysis.html#cb357-30" aria-hidden="true" tabindex="-1"></a>            tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;TempRowNames&quot;</span>)</span>
<span id="cb357-31"><a href="differential-analysis.html#cb357-31" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb357-32"><a href="differential-analysis.html#cb357-32" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-33"><a href="differential-analysis.html#cb357-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(sam_tab_final)[<span class="fu">which</span>(<span class="fu">colnames</span>(sam_tab_final) <span class="sc">==</span> <span class="st">&quot;CompClass&quot;</span>)] <span class="ot">&lt;-</span> Class</span>
<span id="cb357-34"><a href="differential-analysis.html#cb357-34" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-35"><a href="differential-analysis.html#cb357-35" aria-hidden="true" tabindex="-1"></a>    phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(ps) <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(sam_tab_final)</span>
<span id="cb357-36"><a href="differential-analysis.html#cb357-36" aria-hidden="true" tabindex="-1"></a>    otu_tab <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps) <span class="sc">%&gt;%</span></span>
<span id="cb357-37"><a href="differential-analysis.html#cb357-37" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>()</span>
<span id="cb357-38"><a href="differential-analysis.html#cb357-38" aria-hidden="true" tabindex="-1"></a>    otu_tab_final <span class="ot">&lt;-</span> otu_tab[<span class="fu">rowSums</span>(otu_tab) <span class="sc">&gt;</span> cutoff, <span class="fu">colSums</span>(otu_tab) <span class="sc">&gt;</span> cutoff, F]</span>
<span id="cb357-39"><a href="differential-analysis.html#cb357-39" aria-hidden="true" tabindex="-1"></a>    phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(ps) <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(<span class="fu">as.matrix</span>(otu_tab_final), <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb357-40"><a href="differential-analysis.html#cb357-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-41"><a href="differential-analysis.html#cb357-41" aria-hidden="true" tabindex="-1"></a>    lefse_data <span class="ot">&lt;-</span> sam_tab_final <span class="sc">%&gt;%</span> </span>
<span id="cb357-42"><a href="differential-analysis.html#cb357-42" aria-hidden="true" tabindex="-1"></a>        tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb357-43"><a href="differential-analysis.html#cb357-43" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">inner_join</span>(otu_tab_final <span class="sc">%&gt;%</span> </span>
<span id="cb357-44"><a href="differential-analysis.html#cb357-44" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb357-45"><a href="differential-analysis.html#cb357-45" aria-hidden="true" tabindex="-1"></a>                              tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample&quot;</span>),</span>
<span id="cb357-46"><a href="differential-analysis.html#cb357-46" aria-hidden="true" tabindex="-1"></a>                          <span class="at">by =</span> <span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb357-47"><a href="differential-analysis.html#cb357-47" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(Class), Sample, <span class="fu">all_of</span>(Subclass), <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb357-48"><a href="differential-analysis.html#cb357-48" aria-hidden="true" tabindex="-1"></a>        <span class="co">#stats::setNames(c(Class, &quot;Sample&quot;, Subclass, rownames(otu_tab_final))) %&gt;%</span></span>
<span id="cb357-49"><a href="differential-analysis.html#cb357-49" aria-hidden="true" tabindex="-1"></a>        <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>()</span>
<span id="cb357-50"><a href="differential-analysis.html#cb357-50" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-51"><a href="differential-analysis.html#cb357-51" aria-hidden="true" tabindex="-1"></a>    lefse_data_nosub <span class="ot">&lt;-</span> sam_tab_final <span class="sc">%&gt;%</span> </span>
<span id="cb357-52"><a href="differential-analysis.html#cb357-52" aria-hidden="true" tabindex="-1"></a>        tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb357-53"><a href="differential-analysis.html#cb357-53" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">inner_join</span>(otu_tab_final <span class="sc">%&gt;%</span> </span>
<span id="cb357-54"><a href="differential-analysis.html#cb357-54" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb357-55"><a href="differential-analysis.html#cb357-55" aria-hidden="true" tabindex="-1"></a>                              tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample&quot;</span>),</span>
<span id="cb357-56"><a href="differential-analysis.html#cb357-56" aria-hidden="true" tabindex="-1"></a>                          <span class="at">by =</span> <span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb357-57"><a href="differential-analysis.html#cb357-57" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Sample) <span class="sc">%&gt;%</span></span>
<span id="cb357-58"><a href="differential-analysis.html#cb357-58" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(Class), <span class="fu">all_of</span>(Subclass), <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb357-59"><a href="differential-analysis.html#cb357-59" aria-hidden="true" tabindex="-1"></a>        <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>()</span>
<span id="cb357-60"><a href="differential-analysis.html#cb357-60" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-61"><a href="differential-analysis.html#cb357-61" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">ps=</span>ps,</span>
<span id="cb357-62"><a href="differential-analysis.html#cb357-62" aria-hidden="true" tabindex="-1"></a>                <span class="at">lefse=</span>lefse_data,</span>
<span id="cb357-63"><a href="differential-analysis.html#cb357-63" aria-hidden="true" tabindex="-1"></a>                <span class="at">lefse_nosub=</span>lefse_data_nosub)</span>
<span id="cb357-64"><a href="differential-analysis.html#cb357-64" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb357-65"><a href="differential-analysis.html#cb357-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb357-66"><a href="differential-analysis.html#cb357-66" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb357-67"><a href="differential-analysis.html#cb357-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb357-68"><a href="differential-analysis.html#cb357-68" aria-hidden="true" tabindex="-1"></a>amplicon_ps_genus_lefse <span class="ot">&lt;-</span> <span class="fu">prepare_lefse</span>(</span>
<span id="cb357-69"><a href="differential-analysis.html#cb357-69" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ps =</span> amplicon_ps_genus,</span>
<span id="cb357-70"><a href="differential-analysis.html#cb357-70" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Class =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb357-71"><a href="differential-analysis.html#cb357-71" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Class_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb357-72"><a href="differential-analysis.html#cb357-72" aria-hidden="true" tabindex="-1"></a>                          <span class="at">cutoff =</span> <span class="dv">10</span>)</span>
<span id="cb357-73"><a href="differential-analysis.html#cb357-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb357-74"><a href="differential-analysis.html#cb357-74" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(amplicon_ps_genus_lefse<span class="sc">$</span>lefse, <span class="st">&quot;amplicon_ps_genus_lefse.tsv&quot;</span>, <span class="at">quote =</span> F, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col.names =</span> F)</span>
<span id="cb357-75"><a href="differential-analysis.html#cb357-75" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(amplicon_ps_genus_lefse<span class="sc">$</span>lefse_nosub, <span class="st">&quot;amplicon_ps_genus_lefse_nosub.tsv&quot;</span>, <span class="at">quote =</span> F, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col.names =</span> F)</span>
<span id="cb357-76"><a href="differential-analysis.html#cb357-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb357-77"><a href="differential-analysis.html#cb357-77" aria-hidden="true" tabindex="-1"></a>Zeybel_ps_species_lefse <span class="ot">&lt;-</span> <span class="fu">prepare_lefse</span>(</span>
<span id="cb357-78"><a href="differential-analysis.html#cb357-78" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ps =</span> Zeybel_ps_species,</span>
<span id="cb357-79"><a href="differential-analysis.html#cb357-79" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Class =</span> <span class="st">&quot;LiverFatClass&quot;</span>,</span>
<span id="cb357-80"><a href="differential-analysis.html#cb357-80" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Class_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>),</span>
<span id="cb357-81"><a href="differential-analysis.html#cb357-81" aria-hidden="true" tabindex="-1"></a>                          <span class="at">cutoff =</span> <span class="fl">1e-4</span>)</span>
<span id="cb357-82"><a href="differential-analysis.html#cb357-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb357-83"><a href="differential-analysis.html#cb357-83" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Zeybel_ps_species_lefse<span class="sc">$</span>lefse, <span class="st">&quot;Zeybel_ps_species_lefse.tsv&quot;</span>, <span class="at">quote =</span> F, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col.names =</span> F)</span>
<span id="cb357-84"><a href="differential-analysis.html#cb357-84" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(Zeybel_ps_species_lefse<span class="sc">$</span>lefse_nosub, <span class="st">&quot;Zeybel_ps_species_lefse_nosub.tsv&quot;</span>, <span class="at">quote =</span> F, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">col.names =</span> F)</span></code></pre></div>
</div>
</div>
<div id="run-lefse-independently-with-the-three-applications-r-conda-galaxy" class="section level3 hasAnchor" number="11.6.3">
<h3><span class="header-section-number">11.6.3</span> Run lefse independently with the three applications (R, conda, galaxy)<a href="differential-analysis.html#run-lefse-independently-with-the-three-applications-r-conda-galaxy" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="running-lefse-in-r-xmas2" class="section level4 hasAnchor" number="11.6.3.1">
<h4><span class="header-section-number">11.6.3.1</span> Running lefse in R (XMAS2)<a href="differential-analysis.html#running-lefse-in-r-xmas2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Perform the analysis with the <code>run_lefse2</code> function:</p>
<ul>
<li>amplicon_ps_genus</li>
</ul>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="differential-analysis.html#cb358-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run_lefse</span></span>
<span id="cb358-2"><a href="differential-analysis.html#cb358-2" aria-hidden="true" tabindex="-1"></a>amplicon_xmas2_output <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb358-3"><a href="differential-analysis.html#cb358-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ps =</span> amplicon_ps_genus_lefse<span class="sc">$</span>ps,</span>
<span id="cb358-4"><a href="differential-analysis.html#cb358-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb358-5"><a href="differential-analysis.html#cb358-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb358-6"><a href="differential-analysis.html#cb358-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb358-7"><a href="differential-analysis.html#cb358-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">app_name =</span> <span class="st">&quot;xmas_lefse&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb358-8"><a href="differential-analysis.html#cb358-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(LDA_Score)</span>
<span id="cb358-9"><a href="differential-analysis.html#cb358-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(amplicon_xmas2_output)</span></code></pre></div>
<pre><code>##                            TaxaID             Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\ngut_vs_tongue
## 1                  g__Bacteroides 8_gut vs 9_tongue        gut -5.475334   5.734982                               8.706188
## 2 g__Lachnospiraceae_unclassified 8_gut vs 9_tongue        gut -4.445346   2.296951                               3.048798
## 3 g__Ruminococcaceae_unclassified 8_gut vs 9_tongue        gut -4.430034   5.962548                               8.833464
## 4                  g__Lachnospira 8_gut vs 9_tongue        gut -4.327495   6.503718                                     NA
## 5        g__Phascolarctobacterium 8_gut vs 9_tongue        gut -4.214235   6.206589                                     NA
## 6             g__Faecalibacterium 8_gut vs 9_tongue        gut -4.136991   5.492180                                     NA
##   Median Abundance\n(All) Median Abundance\ngut Median Abundance\ntongue Log2FoldChange (Mean)\ngut_vs_tongue
## 1               4552.3520             604210.78                 1446.655                             7.995845
## 2              11431.9387              64739.42                 7823.286                             3.186478
## 3                945.6265              53763.30                  117.855                             7.701566
## 4                  0.0000              50368.43                    0.000                                   NA
## 5                  0.0000              18238.84                    0.000                                   NA
## 6                  0.0000              19194.92                    0.000                                   NA
##   Mean Abundance\n(All) Mean Abundance\ngut Mean Abundance\ntongue Occurrence (100%)\n(All) Occurrence (100%)\ngut
## 1             277146.67           586352.50              2297.0454                   100.00                    100
## 2              34604.22            65446.48              7188.8734                   100.00                    100
## 3              26586.43            56192.47               269.9452                    76.47                    100
## 4              22767.22            48380.35                 0.0000                    47.06                    100
## 5              13444.82            28570.25                 0.0000                    47.06                    100
## 6              13991.79            29732.55                 0.0000                    47.06                    100
##   Occurrence (100%)\ntongue Odds Ratio (95% CI)   app_name
## 1                    100.00    4.7e-14 (-60;60) xmas_lefse
## 2                    100.00  4.4e-24 (-110;110) xmas_lefse
## 3                     55.56  7.2e-41 (-180;180) xmas_lefse
## 4                      0.00                &lt;NA&gt; xmas_lefse
## 5                      0.00                &lt;NA&gt; xmas_lefse
## 6                      0.00                &lt;NA&gt; xmas_lefse</code></pre>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="differential-analysis.html#cb360-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run_lefse2</span></span>
<span id="cb360-2"><a href="differential-analysis.html#cb360-2" aria-hidden="true" tabindex="-1"></a>amplicon_xmas2_output2 <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb360-3"><a href="differential-analysis.html#cb360-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ps =</span> amplicon_ps_genus_lefse<span class="sc">$</span>ps,</span>
<span id="cb360-4"><a href="differential-analysis.html#cb360-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb360-5"><a href="differential-analysis.html#cb360-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb360-6"><a href="differential-analysis.html#cb360-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb360-7"><a href="differential-analysis.html#cb360-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">app_name =</span> <span class="st">&quot;xmas_lefse2&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb360-8"><a href="differential-analysis.html#cb360-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(LDA_Score)</span>
<span id="cb360-9"><a href="differential-analysis.html#cb360-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(amplicon_xmas2_output2)</span></code></pre></div>
<pre><code>##                            TaxaID             Block LDA_Score Enrichment EffectSize       Pvalue
## 1                  g__Bacteroides 8_gut vs 9_tongue -5.763040        gut   5.734982 0.0005320055
## 2 g__Lachnospiraceae_unclassified 8_gut vs 9_tongue -4.774847        gut   2.296951 0.0005320055
## 3 g__Ruminococcaceae_unclassified 8_gut vs 9_tongue -4.756686        gut   5.962548 0.0004911726
## 4                  g__Lachnospira 8_gut vs 9_tongue -4.684106        gut   6.503718 0.0001762277
## 5             g__Faecalibacterium 8_gut vs 9_tongue -4.447011        gut   5.492180 0.0001762277
## 6        g__Phascolarctobacterium 8_gut vs 9_tongue -4.441230        gut   6.206589 0.0001762277
##   Log2FoldChange (Median)\ngut_vs_tongue Median Abundance\n(All) Median Abundance\ngut Median Abundance\ntongue
## 1                               8.706188               4552.3520             604210.78                 1446.655
## 2                               3.048798              11431.9387              64739.42                 7823.286
## 3                               8.833464                945.6265              53763.30                  117.855
## 4                                     NA                  0.0000              50368.43                    0.000
## 5                                     NA                  0.0000              19194.92                    0.000
## 6                                     NA                  0.0000              18238.84                    0.000
##   Log2FoldChange (Mean)\ngut_vs_tongue Mean Abundance\n(All) Mean Abundance\ngut Mean Abundance\ntongue
## 1                             7.995845             277146.67           586352.50              2297.0454
## 2                             3.186478              34604.22            65446.48              7188.8734
## 3                             7.701566              26586.43            56192.47               269.9452
## 4                                   NA              22767.22            48380.35                 0.0000
## 5                                   NA              13991.79            29732.55                 0.0000
## 6                                   NA              13444.82            28570.25                 0.0000
##   Occurrence (100%)\n(All) Occurrence (100%)\ngut Occurrence (100%)\ntongue Odds Ratio (95% CI)    app_name
## 1                   100.00                    100                    100.00    4.7e-14 (-60;60) xmas_lefse2
## 2                   100.00                    100                    100.00  4.4e-24 (-110;110) xmas_lefse2
## 3                    76.47                    100                     55.56  7.2e-41 (-180;180) xmas_lefse2
## 4                    47.06                    100                      0.00                &lt;NA&gt; xmas_lefse2
## 5                    47.06                    100                      0.00                &lt;NA&gt; xmas_lefse2
## 6                    47.06                    100                      0.00                &lt;NA&gt; xmas_lefse2</code></pre>
<ul>
<li>Zeybel_ps_species</li>
</ul>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb362-1"><a href="differential-analysis.html#cb362-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run_lefse</span></span>
<span id="cb362-2"><a href="differential-analysis.html#cb362-2" aria-hidden="true" tabindex="-1"></a>MGS_xmas2_output <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb362-3"><a href="differential-analysis.html#cb362-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ps =</span> Zeybel_ps_species_lefse<span class="sc">$</span>ps,</span>
<span id="cb362-4"><a href="differential-analysis.html#cb362-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;LiverFatClass&quot;</span>,</span>
<span id="cb362-5"><a href="differential-analysis.html#cb362-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>),</span>
<span id="cb362-6"><a href="differential-analysis.html#cb362-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb362-7"><a href="differential-analysis.html#cb362-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">app_name =</span> <span class="st">&quot;xmas_lefse&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb362-8"><a href="differential-analysis.html#cb362-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(LDA_Score)</span>
<span id="cb362-9"><a href="differential-analysis.html#cb362-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(MGS_xmas2_output)</span></code></pre></div>
<pre><code>##                            TaxaID                  Block Enrichment LDA_Score EffectSize
## 1         s__Butyricimonas_virosa 12_Mild vs 12_Moderate   Moderate  2.445966   1.668501
## 2       s__Bacteroides_salyersiae 12_Mild vs 12_Moderate   Moderate  2.659689   1.573816
## 3           s__Bacteroides_clarus 12_Mild vs 12_Moderate   Moderate  3.191513   2.331714
## 4 s__Bacteroides_thetaiotaomicron 12_Mild vs 12_Moderate   Moderate  3.210748   2.775490
## 5        s__Bacteroides_coprocola 12_Mild vs 12_Moderate   Moderate  3.314238   1.367262
## 6 s__Bacteroides_cellulosilyticus 12_Mild vs 12_Moderate   Moderate  3.408064   3.145457
##   Log2FoldChange (Median)\nMild_vs_Moderate Median Abundance\n(All) Median Abundance\nMild Median Abundance\nModerate
## 1                                        NA               0.8999998                      0                   262.9238
## 2                                        NA               0.0000000                      0                     0.0000
## 3                                        NA               0.0000000                      0                     0.0000
## 4                                        NA             223.8648101                      0                  1123.0082
## 5                                        NA               0.0000000                      0                     0.0000
## 6                                        NA              82.9748375                      0                  2707.7169
##   Log2FoldChange (Mean)\nMild_vs_Moderate Mean Abundance\n(All) Mean Abundance\nMild Mean Abundance\nModerate
## 1                               -2.329681              353.2093             117.2101                 589.2084
## 2                                      NA              262.0187               0.0000                 524.0375
## 3                                      NA             1283.6658               0.0000                2567.3315
## 4                               -2.241224             2413.3440             842.6510                3984.0371
## 5                                      NA             1704.3673               0.0000                3408.7346
## 6                               -3.325019             3170.2327             575.2844                5765.1810
##   Occurrence (100%)\n(All) Occurrence (100%)\nMild Occurrence (100%)\nModerate Odds Ratio (95% CI)   app_name
## 1                    50.00                   25.00                       75.00       3.9 (6.6;1.2) xmas_lefse
## 2                    16.67                    0.00                       33.33                &lt;NA&gt; xmas_lefse
## 3                    20.83                    0.00                       41.67                &lt;NA&gt; xmas_lefse
## 4                    66.67                   41.67                       91.67      3.1 (5.3;0.87) xmas_lefse
## 5                    16.67                    0.00                       33.33                &lt;NA&gt; xmas_lefse
## 6                    62.50                   41.67                       83.33          17 (23;11) xmas_lefse</code></pre>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="differential-analysis.html#cb364-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run_lefse2</span></span>
<span id="cb364-2"><a href="differential-analysis.html#cb364-2" aria-hidden="true" tabindex="-1"></a>MGS_xmas2_output2 <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb364-3"><a href="differential-analysis.html#cb364-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">ps =</span> Zeybel_ps_species_lefse<span class="sc">$</span>ps,</span>
<span id="cb364-4"><a href="differential-analysis.html#cb364-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;LiverFatClass&quot;</span>,</span>
<span id="cb364-5"><a href="differential-analysis.html#cb364-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>),</span>
<span id="cb364-6"><a href="differential-analysis.html#cb364-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb364-7"><a href="differential-analysis.html#cb364-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">app_name =</span> <span class="st">&quot;xmas_lefse2&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb364-8"><a href="differential-analysis.html#cb364-8" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(LDA_Score)</span>
<span id="cb364-9"><a href="differential-analysis.html#cb364-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(MGS_xmas2_output2)</span></code></pre></div>
<pre><code>##                            TaxaID                  Block LDA_Score Enrichment EffectSize     Pvalue
## 1         s__Butyricimonas_virosa 12_Mild vs 12_Moderate  2.709350   Moderate   1.668501 0.02432292
## 2       s__Bacteroides_salyersiae 12_Mild vs 12_Moderate  2.763901   Moderate   1.573816 0.03286923
## 3           s__Bacteroides_clarus 12_Mild vs 12_Moderate  3.354092   Moderate   2.331714 0.01473169
## 4 s__Bacteroides_thetaiotaomicron 12_Mild vs 12_Moderate  3.502262   Moderate   2.775490 0.01050852
## 5     s__Bacteroides_intestinalis 12_Mild vs 12_Moderate  3.524676   Moderate   1.184353 0.03286923
## 6        s__Bacteroides_coprocola 12_Mild vs 12_Moderate  3.538227   Moderate   1.367262 0.03286923
##   Log2FoldChange (Median)\nMild_vs_Moderate Median Abundance\n(All) Median Abundance\nMild Median Abundance\nModerate
## 1                                        NA               0.8999998                      0                   262.9238
## 2                                        NA               0.0000000                      0                     0.0000
## 3                                        NA               0.0000000                      0                     0.0000
## 4                                        NA             223.8648101                      0                  1123.0082
## 5                                        NA               0.0000000                      0                     0.0000
## 6                                        NA               0.0000000                      0                     0.0000
##   Log2FoldChange (Mean)\nMild_vs_Moderate Mean Abundance\n(All) Mean Abundance\nMild Mean Abundance\nModerate
## 1                               -2.329681              353.2093             117.2101                 589.2084
## 2                                      NA              262.0187               0.0000                 524.0375
## 3                                      NA             1283.6658               0.0000                2567.3315
## 4                               -2.241224             2413.3440             842.6510                3984.0371
## 5                                      NA             1745.9150               0.0000                3491.8300
## 6                                      NA             1704.3673               0.0000                3408.7346
##   Occurrence (100%)\n(All) Occurrence (100%)\nMild Occurrence (100%)\nModerate Odds Ratio (95% CI)    app_name
## 1                    50.00                   25.00                       75.00       3.9 (6.6;1.2) xmas_lefse2
## 2                    16.67                    0.00                       33.33                &lt;NA&gt; xmas_lefse2
## 3                    20.83                    0.00                       41.67                &lt;NA&gt; xmas_lefse2
## 4                    66.67                   41.67                       91.67      3.1 (5.3;0.87) xmas_lefse2
## 5                    16.67                    0.00                       33.33                &lt;NA&gt; xmas_lefse2
## 6                    16.67                    0.00                       33.33                &lt;NA&gt; xmas_lefse2</code></pre>
</div>
<div id="running-lefse-conda-command-line" class="section level4 hasAnchor" number="11.6.3.2">
<h4><span class="header-section-number">11.6.3.2</span> Running lefse-conda (command line)<a href="differential-analysis.html#running-lefse-conda-command-line" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="lefse-conda-installation-and-version" class="section level6 hasAnchor" number="11.6.3.2.0.1">
<h6><span class="header-section-number">11.6.3.2.0.1</span> lefse-conda installation and version<a href="differential-analysis.html#lefse-conda-installation-and-version" class="anchor-section" aria-label="Anchor link to header"></a></h6>
<p>Note: I installed lefse following the instructions from <a href="https://github.com/biobakery/conda-biobakery">this site</a>
after installing conda.</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb366-1"><a href="differential-analysis.html#cb366-1" aria-hidden="true" tabindex="-1"></a><span class="co">## Add channels</span></span>
<span id="cb366-2"><a href="differential-analysis.html#cb366-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels defaults</span>
<span id="cb366-3"><a href="differential-analysis.html#cb366-3" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels bioconda</span>
<span id="cb366-4"><a href="differential-analysis.html#cb366-4" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels conda-forge</span>
<span id="cb366-5"><a href="differential-analysis.html#cb366-5" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> config <span class="at">--add</span> channels biobakery</span>
<span id="cb366-6"><a href="differential-analysis.html#cb366-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb366-7"><a href="differential-analysis.html#cb366-7" aria-hidden="true" tabindex="-1"></a><span class="co">## Install lefse</span></span>
<span id="cb366-8"><a href="differential-analysis.html#cb366-8" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> lefse <span class="at">-c</span> biobakery lefse <span class="at">-y</span></span></code></pre></div>
<p>Conda and lefse versions:</p>
<div class="sourceCode" id="cb367"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb367-1"><a href="differential-analysis.html#cb367-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> <span class="at">--version</span></span>
<span id="cb367-2"><a href="differential-analysis.html#cb367-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; conda 4.12.0</span></span>
<span id="cb367-3"><a href="differential-analysis.html#cb367-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb367-4"><a href="differential-analysis.html#cb367-4" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> list <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-e</span> <span class="st">&quot;lefse&quot;</span></span>
<span id="cb367-5"><a href="differential-analysis.html#cb367-5" aria-hidden="true" tabindex="-1"></a><span class="co"># packages in environment at /home/samuel/miniconda3/envs/lefse:</span></span>
<span id="cb367-6"><a href="differential-analysis.html#cb367-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; lefse                     1.1.2              pyhdfd78af_0    bioconda</span></span></code></pre></div>
</div>
<div id="run-lefse-conda" class="section level5 hasAnchor" number="11.6.3.2.1">
<h5><span class="header-section-number">11.6.3.2.1</span> Run lefse-conda<a href="differential-analysis.html#run-lefse-conda" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p>Generate a tabular dataset (amplicon_ps_genus_lefse or Zeybel_ps_species_lefse) compatible with lefse-conda and
lefse-galaxy using the <code>get_dataset.R</code> script.</p></li>
<li><p>Run the script <code>run_lefse.sh</code> (linux) with the following parameters:</p></li>
</ol>
<div class="sourceCode" id="cb368"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb368-1"><a href="differential-analysis.html#cb368-1" aria-hidden="true" tabindex="-1"></a><span class="co"># In general</span></span>
<span id="cb368-2"><a href="differential-analysis.html#cb368-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ./run_lefse.sh &lt;path/to/conda/activate&gt; &lt;env_name&gt; &lt;path/to/Rscript&gt; &lt;filename prefix&gt;</span></span>
<span id="cb368-3"><a href="differential-analysis.html#cb368-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb368-4"><a href="differential-analysis.html#cb368-4" aria-hidden="true" tabindex="-1"></a><span class="co"># in my case (Hua Zou)</span></span>
<span id="cb368-5"><a href="differential-analysis.html#cb368-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb368-6"><a href="differential-analysis.html#cb368-6" aria-hidden="true" tabindex="-1"></a><span class="co">## amplicon_ps_genus_lefse </span></span>
<span id="cb368-7"><a href="differential-analysis.html#cb368-7" aria-hidden="true" tabindex="-1"></a><span class="ex">./run_lefse.sh</span> /Users/zouhua/opt/anaconda3/bin/activate lefse /usr/local/bin/R amplicon_ps_genus_lefse</span>
<span id="cb368-8"><a href="differential-analysis.html#cb368-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb368-9"><a href="differential-analysis.html#cb368-9" aria-hidden="true" tabindex="-1"></a><span class="co">## Zeybel_ps_species_lefse </span></span>
<span id="cb368-10"><a href="differential-analysis.html#cb368-10" aria-hidden="true" tabindex="-1"></a><span class="ex">./run_lefse.sh</span> /Users/zouhua/opt/anaconda3/bin/activate lefse /usr/local/bin/R Zeybel_ps_species_lefse</span></code></pre></div>
<p>Note: All script files, <code>get_dataset.R</code> and <code>run_lefse.sh</code>, and this rmarkdown
document must be in the same directory.</p>
</div>
<div id="import-output-from-lefse-conda-into-r" class="section level5 hasAnchor" number="11.6.3.2.2">
<h5><span class="header-section-number">11.6.3.2.2</span> Import output from lefse-conda into R<a href="differential-analysis.html#import-output-from-lefse-conda-into-r" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="differential-analysis.html#cb369-1" aria-hidden="true" tabindex="-1"></a>get_lefse_python <span class="ot">&lt;-</span> <span class="cf">function</span>(datres, </span>
<span id="cb369-2"><a href="differential-analysis.html#cb369-2" aria-hidden="true" tabindex="-1"></a>                            Class_names,</span>
<span id="cb369-3"><a href="differential-analysis.html#cb369-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">name =</span> <span class="st">&quot;lefse_conda&quot;</span>,</span>
<span id="cb369-4"><a href="differential-analysis.html#cb369-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">LDA_names =</span> <span class="st">&quot;lefse_conda_LDA&quot;</span>,</span>
<span id="cb369-5"><a href="differential-analysis.html#cb369-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">LDA_cutoff =</span> <span class="dv">2</span>) {</span>
<span id="cb369-6"><a href="differential-analysis.html#cb369-6" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb369-7"><a href="differential-analysis.html#cb369-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># datres = &quot;amplicon_ps_genus_lefse.res&quot;</span></span>
<span id="cb369-8"><a href="differential-analysis.html#cb369-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Class_names = c(&quot;gut&quot;, &quot;tongue&quot;)</span></span>
<span id="cb369-9"><a href="differential-analysis.html#cb369-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># LDA_cutoff = 2</span></span>
<span id="cb369-10"><a href="differential-analysis.html#cb369-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb369-11"><a href="differential-analysis.html#cb369-11" aria-hidden="true" tabindex="-1"></a>    col_names <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb369-12"><a href="differential-analysis.html#cb369-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;TaxaID&quot;</span>, <span class="st">&quot;log_hi_class_avg&quot;</span>, <span class="st">&quot;Enrichment&quot;</span>, <span class="st">&quot;lefse_conda_LDA&quot;</span>, <span class="st">&quot;pval&quot;</span>)</span>
<span id="cb369-13"><a href="differential-analysis.html#cb369-13" aria-hidden="true" tabindex="-1"></a>    lefse_conda <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(datres, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb369-14"><a href="differential-analysis.html#cb369-14" aria-hidden="true" tabindex="-1"></a>        magrittr<span class="sc">::</span><span class="fu">set_colnames</span>(col_names) <span class="sc">%&gt;%</span> </span>
<span id="cb369-15"><a href="differential-analysis.html#cb369-15" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(lefse_conda_LDA)) <span class="sc">%&gt;%</span></span>
<span id="cb369-16"><a href="differential-analysis.html#cb369-16" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb369-17"><a href="differential-analysis.html#cb369-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">lefse_conda_LDA =</span> <span class="fu">ifelse</span>(</span>
<span id="cb369-18"><a href="differential-analysis.html#cb369-18" aria-hidden="true" tabindex="-1"></a>                Enrichment <span class="sc">==</span> Class_names[<span class="dv">1</span>], <span class="sc">-</span>lefse_conda_LDA, lefse_conda_LDA),</span>
<span id="cb369-19"><a href="differential-analysis.html#cb369-19" aria-hidden="true" tabindex="-1"></a>            <span class="at">app_name =</span> name) <span class="sc">%&gt;%</span> </span>
<span id="cb369-20"><a href="differential-analysis.html#cb369-20" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(lefse_conda_LDA) <span class="sc">&gt;=</span> LDA_cutoff) <span class="sc">%&gt;%</span></span>
<span id="cb369-21"><a href="differential-analysis.html#cb369-21" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">arrange</span>(lefse_conda_LDA)</span>
<span id="cb369-22"><a href="differential-analysis.html#cb369-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb369-23"><a href="differential-analysis.html#cb369-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(lefse_conda)[<span class="fu">which</span>(<span class="fu">colnames</span>(lefse_conda) <span class="sc">==</span> <span class="st">&quot;lefse_conda_LDA&quot;</span>)] <span class="ot">&lt;-</span> LDA_names</span>
<span id="cb369-24"><a href="differential-analysis.html#cb369-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb369-25"><a href="differential-analysis.html#cb369-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(lefse_conda)</span>
<span id="cb369-26"><a href="differential-analysis.html#cb369-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb369-27"><a href="differential-analysis.html#cb369-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb369-28"><a href="differential-analysis.html#cb369-28" aria-hidden="true" tabindex="-1"></a>amplicon_ps_genus_lefse_conda <span class="ot">&lt;-</span> <span class="fu">get_lefse_python</span>(</span>
<span id="cb369-29"><a href="differential-analysis.html#cb369-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">datres =</span> <span class="st">&quot;amplicon_ps_genus_lefse.res&quot;</span>, </span>
<span id="cb369-30"><a href="differential-analysis.html#cb369-30" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Class_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb369-31"><a href="differential-analysis.html#cb369-31" aria-hidden="true" tabindex="-1"></a>                    <span class="at">LDA_names =</span> <span class="st">&quot;lefse_conda_LDA&quot;</span>)</span>
<span id="cb369-32"><a href="differential-analysis.html#cb369-32" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(amplicon_ps_genus_lefse_conda)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##   TaxaID                          log_hi_class_avg Enrichment lefse_conda_LDA pval                  app_name   
##   &lt;chr&gt;                                      &lt;dbl&gt; &lt;chr&gt;                &lt;dbl&gt; &lt;chr&gt;                 &lt;chr&gt;      
## 1 g__Bacteroides                              5.77 gut                  -5.47 0.0005320055051392497 lefse_conda
## 2 g__Lachnospiraceae_unclassified             4.82 gut                  -4.47 0.0005320055051392497 lefse_conda
## 3 g__Ruminococcaceae_unclassified             4.75 gut                  -4.46 0.0004911726274481772 lefse_conda
## 4 g__Lachnospira                              4.68 gut                  -4.36 0.0001762276980982556 lefse_conda
## 5 g__Faecalibacterium                         4.47 gut                  -4.20 0.0001762276980982556 lefse_conda
## 6 g__Phascolarctobacterium                    4.46 gut                  -4.15 0.0001762276980982556 lefse_conda</code></pre>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="differential-analysis.html#cb371-1" aria-hidden="true" tabindex="-1"></a>Zeybel_ps_species_lefse_conda <span class="ot">&lt;-</span> <span class="fu">get_lefse_python</span>(</span>
<span id="cb371-2"><a href="differential-analysis.html#cb371-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">datres =</span> <span class="st">&quot;Zeybel_ps_species_lefse.res&quot;</span>, </span>
<span id="cb371-3"><a href="differential-analysis.html#cb371-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Class_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>),</span>
<span id="cb371-4"><a href="differential-analysis.html#cb371-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">LDA_names =</span> <span class="st">&quot;lefse_conda_LDA&quot;</span>)</span>
<span id="cb371-5"><a href="differential-analysis.html#cb371-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Zeybel_ps_species_lefse_conda)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##   TaxaID                          log_hi_class_avg Enrichment lefse_conda_LDA pval                 app_name   
##   &lt;chr&gt;                                      &lt;dbl&gt; &lt;chr&gt;                &lt;dbl&gt; &lt;chr&gt;                &lt;chr&gt;      
## 1 s__Butyricimonas_virosa                     2.77 Moderate              2.46 0.02432291566171718  lefse_conda
## 2 s__Bacteroides_salyersiae                   2.72 Moderate              2.47 0.032869234300658745 lefse_conda
## 3 s__Bacteroides_clarus                       3.41 Moderate              3.07 0.014731687897477879 lefse_conda
## 4 s__Bacteroides_thetaiotaomicron             3.60 Moderate              3.26 0.01050851971947486  lefse_conda
## 5 s__Bacteroides_coprocola                    3.53 Moderate              3.28 0.032869234300658745 lefse_conda
## 6 s__Bacteroides_intestinalis                 3.54 Moderate              3.32 0.032869234300658745 lefse_conda</code></pre>
</div>
</div>
<div id="running-lefse-from-galaxy" class="section level4 hasAnchor" number="11.6.3.3">
<h4><span class="header-section-number">11.6.3.3</span> Running lefse from galaxy<a href="differential-analysis.html#running-lefse-from-galaxy" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Using the <code>amplicon_ps_genus_lefse_nosub.txt</code> or <code>Zeybel_ps_species_lefse_nosub.txt</code> file (no subjects included) as input for lefse from the galaxy platform of the Huttenhower lab at <a href="https://huttenhower.sph.harvard.edu/galaxy/">galaxy</a>.</p>
<p>The conditions as follow:</p>
<ul>
<li><p>alpha were 0.05 for both KW and Wilcox,</p></li>
<li><p>2.0 for LDA.</p></li>
<li><p>TSS normalization was applied as well.</p></li>
</ul>
<p>converting the output into compared format:</p>
<ul>
<li><p>amplicon_ps_genus_lefse_nosub.res</p></li>
<li><p>Zeybel_ps_species_lefse_nosub.res</p></li>
</ul>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="differential-analysis.html#cb373-1" aria-hidden="true" tabindex="-1"></a>amplicon_ps_genus_lefse_galaxy <span class="ot">&lt;-</span> <span class="fu">get_lefse_python</span>(</span>
<span id="cb373-2"><a href="differential-analysis.html#cb373-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">datres =</span> <span class="st">&quot;amplicon_ps_genus_lefse_nosub.res&quot;</span>,</span>
<span id="cb373-3"><a href="differential-analysis.html#cb373-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;lefse_galaxy&quot;</span>,</span>
<span id="cb373-4"><a href="differential-analysis.html#cb373-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Class_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb373-5"><a href="differential-analysis.html#cb373-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">LDA_names =</span> <span class="st">&quot;lefse_galaxy_LDA&quot;</span>)</span>
<span id="cb373-6"><a href="differential-analysis.html#cb373-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(amplicon_ps_genus_lefse_galaxy)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##   TaxaID                          log_hi_class_avg Enrichment lefse_galaxy_LDA pval              app_name    
##   &lt;chr&gt;                                      &lt;dbl&gt; &lt;chr&gt;                 &lt;dbl&gt; &lt;chr&gt;             &lt;chr&gt;       
## 1 g__Bacteroides                              5.77 gut                   -5.46 0.000532005505139 lefse_galaxy
## 2 g__Lachnospiraceae_unclassified             4.82 gut                   -4.45 0.000532005505139 lefse_galaxy
## 3 g__Ruminococcaceae_unclassified             4.75 gut                   -4.41 0.000491172627448 lefse_galaxy
## 4 g__Lachnospira                              4.68 gut                   -4.38 0.000176227698098 lefse_galaxy
## 5 g__Phascolarctobacterium                    4.46 gut                   -4.18 0.000176227698098 lefse_galaxy
## 6 g__Faecalibacterium                         4.47 gut                   -4.18 0.000176227698098 lefse_galaxy</code></pre>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="differential-analysis.html#cb375-1" aria-hidden="true" tabindex="-1"></a>Zeybel_ps_species_lefse_galaxy <span class="ot">&lt;-</span> <span class="fu">get_lefse_python</span>(</span>
<span id="cb375-2"><a href="differential-analysis.html#cb375-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">datres =</span> <span class="st">&quot;Zeybel_ps_species_lefse_nosub.res&quot;</span>,</span>
<span id="cb375-3"><a href="differential-analysis.html#cb375-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;lefse_galaxy&quot;</span>,</span>
<span id="cb375-4"><a href="differential-analysis.html#cb375-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Class_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>),</span>
<span id="cb375-5"><a href="differential-analysis.html#cb375-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">LDA_names =</span> <span class="st">&quot;lefse_galaxy_LDA&quot;</span>)</span>
<span id="cb375-6"><a href="differential-analysis.html#cb375-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(amplicon_ps_genus_lefse_galaxy)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##   TaxaID                          log_hi_class_avg Enrichment lefse_galaxy_LDA pval              app_name    
##   &lt;chr&gt;                                      &lt;dbl&gt; &lt;chr&gt;                 &lt;dbl&gt; &lt;chr&gt;             &lt;chr&gt;       
## 1 g__Bacteroides                              5.77 gut                   -5.46 0.000532005505139 lefse_galaxy
## 2 g__Lachnospiraceae_unclassified             4.82 gut                   -4.45 0.000532005505139 lefse_galaxy
## 3 g__Ruminococcaceae_unclassified             4.75 gut                   -4.41 0.000491172627448 lefse_galaxy
## 4 g__Lachnospira                              4.68 gut                   -4.38 0.000176227698098 lefse_galaxy
## 5 g__Phascolarctobacterium                    4.46 gut                   -4.18 0.000176227698098 lefse_galaxy
## 6 g__Faecalibacterium                         4.47 gut                   -4.18 0.000176227698098 lefse_galaxy</code></pre>
</div>
<div id="extracting-results-from-xmas2-results" class="section level4 hasAnchor" number="11.6.3.4">
<h4><span class="header-section-number">11.6.3.4</span> Extracting results from XMAS2 results<a href="differential-analysis.html#extracting-results-from-xmas2-results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><p>run_lefse (lefser R package)</p></li>
<li><p>run_lefse2 (microbiomeMarker R package)</p></li>
</ul>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="differential-analysis.html#cb377-1" aria-hidden="true" tabindex="-1"></a>get_lefse_R <span class="ot">&lt;-</span> <span class="cf">function</span>(datres,</span>
<span id="cb377-2"><a href="differential-analysis.html#cb377-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;Rrun_lefse&quot;</span>,</span>
<span id="cb377-3"><a href="differential-analysis.html#cb377-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">LDA_names =</span> <span class="st">&quot;lefse_R_LDA&quot;</span>,</span>
<span id="cb377-4"><a href="differential-analysis.html#cb377-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">LDA_cutoff =</span> <span class="dv">2</span>) {</span>
<span id="cb377-5"><a href="differential-analysis.html#cb377-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb377-6"><a href="differential-analysis.html#cb377-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># datres = amplicon_xmas2_output</span></span>
<span id="cb377-7"><a href="differential-analysis.html#cb377-7" aria-hidden="true" tabindex="-1"></a>    <span class="co"># name = &quot;Rrun_lefse&quot;</span></span>
<span id="cb377-8"><a href="differential-analysis.html#cb377-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># LDA_cutoff = 2</span></span>
<span id="cb377-9"><a href="differential-analysis.html#cb377-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb377-10"><a href="differential-analysis.html#cb377-10" aria-hidden="true" tabindex="-1"></a>    col_names <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb377-11"><a href="differential-analysis.html#cb377-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;TaxaID&quot;</span>, <span class="st">&quot;Block&quot;</span>, <span class="st">&quot;Enrichment&quot;</span>, <span class="st">&quot;LDA_Score&quot;</span>, <span class="st">&quot;EffectSize&quot;</span>)</span>
<span id="cb377-12"><a href="differential-analysis.html#cb377-12" aria-hidden="true" tabindex="-1"></a>    lefse_R <span class="ot">&lt;-</span> datres <span class="sc">%&gt;%</span> </span>
<span id="cb377-13"><a href="differential-analysis.html#cb377-13" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(col_names)) <span class="sc">%&gt;%</span> </span>
<span id="cb377-14"><a href="differential-analysis.html#cb377-14" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">app_name =</span> name) <span class="sc">%&gt;%</span> </span>
<span id="cb377-15"><a href="differential-analysis.html#cb377-15" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(LDA_Score) <span class="sc">&gt;=</span> LDA_cutoff) <span class="sc">%&gt;%</span></span>
<span id="cb377-16"><a href="differential-analysis.html#cb377-16" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">arrange</span>(LDA_Score)</span>
<span id="cb377-17"><a href="differential-analysis.html#cb377-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb377-18"><a href="differential-analysis.html#cb377-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colnames</span>(lefse_R)[<span class="fu">which</span>(<span class="fu">colnames</span>(lefse_R) <span class="sc">==</span> <span class="st">&quot;LDA_Score&quot;</span>)] <span class="ot">&lt;-</span> LDA_names</span>
<span id="cb377-19"><a href="differential-analysis.html#cb377-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb377-20"><a href="differential-analysis.html#cb377-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(lefse_R)</span>
<span id="cb377-21"><a href="differential-analysis.html#cb377-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb377-22"><a href="differential-analysis.html#cb377-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb377-23"><a href="differential-analysis.html#cb377-23" aria-hidden="true" tabindex="-1"></a>amplicon_ps_genus_lefse_R <span class="ot">&lt;-</span> <span class="fu">get_lefse_R</span>(</span>
<span id="cb377-24"><a href="differential-analysis.html#cb377-24" aria-hidden="true" tabindex="-1"></a>                        <span class="at">datres =</span> amplicon_xmas2_output, </span>
<span id="cb377-25"><a href="differential-analysis.html#cb377-25" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;Rrun_lefse&quot;</span>,</span>
<span id="cb377-26"><a href="differential-analysis.html#cb377-26" aria-hidden="true" tabindex="-1"></a>                        <span class="at">LDA_names =</span> <span class="st">&quot;lefse_R_LDA&quot;</span>)</span>
<span id="cb377-27"><a href="differential-analysis.html#cb377-27" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(amplicon_ps_genus_lefse_R)</span></code></pre></div>
<pre><code>##                            TaxaID             Block Enrichment lefse_R_LDA EffectSize   app_name
## 1                  g__Bacteroides 8_gut vs 9_tongue        gut   -5.475334   5.734982 Rrun_lefse
## 2 g__Lachnospiraceae_unclassified 8_gut vs 9_tongue        gut   -4.445346   2.296951 Rrun_lefse
## 3 g__Ruminococcaceae_unclassified 8_gut vs 9_tongue        gut   -4.430034   5.962548 Rrun_lefse
## 4                  g__Lachnospira 8_gut vs 9_tongue        gut   -4.327495   6.503718 Rrun_lefse
## 5        g__Phascolarctobacterium 8_gut vs 9_tongue        gut   -4.214235   6.206589 Rrun_lefse
## 6             g__Faecalibacterium 8_gut vs 9_tongue        gut   -4.136991   5.492180 Rrun_lefse</code></pre>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="differential-analysis.html#cb379-1" aria-hidden="true" tabindex="-1"></a>amplicon_ps_genus_lefse_R2 <span class="ot">&lt;-</span> <span class="fu">get_lefse_R</span>(</span>
<span id="cb379-2"><a href="differential-analysis.html#cb379-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">datres =</span> amplicon_xmas2_output2, </span>
<span id="cb379-3"><a href="differential-analysis.html#cb379-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;Rrun_lefse2&quot;</span>,</span>
<span id="cb379-4"><a href="differential-analysis.html#cb379-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">LDA_names =</span> <span class="st">&quot;lefse_R2_LDA&quot;</span>)</span>
<span id="cb379-5"><a href="differential-analysis.html#cb379-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(amplicon_ps_genus_lefse_R2)</span></code></pre></div>
<pre><code>##                            TaxaID             Block Enrichment lefse_R2_LDA EffectSize    app_name
## 1                  g__Bacteroides 8_gut vs 9_tongue        gut    -5.763040   5.734982 Rrun_lefse2
## 2 g__Lachnospiraceae_unclassified 8_gut vs 9_tongue        gut    -4.774847   2.296951 Rrun_lefse2
## 3 g__Ruminococcaceae_unclassified 8_gut vs 9_tongue        gut    -4.756686   5.962548 Rrun_lefse2
## 4                  g__Lachnospira 8_gut vs 9_tongue        gut    -4.684106   6.503718 Rrun_lefse2
## 5             g__Faecalibacterium 8_gut vs 9_tongue        gut    -4.447011   5.492180 Rrun_lefse2
## 6        g__Phascolarctobacterium 8_gut vs 9_tongue        gut    -4.441230   6.206589 Rrun_lefse2</code></pre>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="differential-analysis.html#cb381-1" aria-hidden="true" tabindex="-1"></a>Zeybel_ps_species_lefse_R <span class="ot">&lt;-</span> <span class="fu">get_lefse_R</span>(</span>
<span id="cb381-2"><a href="differential-analysis.html#cb381-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">datres =</span> MGS_xmas2_output, </span>
<span id="cb381-3"><a href="differential-analysis.html#cb381-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;Rrun_lefse&quot;</span>,</span>
<span id="cb381-4"><a href="differential-analysis.html#cb381-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">LDA_names =</span> <span class="st">&quot;lefse_R_LDA&quot;</span>)</span>
<span id="cb381-5"><a href="differential-analysis.html#cb381-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Zeybel_ps_species_lefse_R)</span></code></pre></div>
<pre><code>##                            TaxaID                  Block Enrichment lefse_R_LDA EffectSize   app_name
## 1         s__Butyricimonas_virosa 12_Mild vs 12_Moderate   Moderate    2.445966   1.668501 Rrun_lefse
## 2       s__Bacteroides_salyersiae 12_Mild vs 12_Moderate   Moderate    2.659689   1.573816 Rrun_lefse
## 3           s__Bacteroides_clarus 12_Mild vs 12_Moderate   Moderate    3.191513   2.331714 Rrun_lefse
## 4 s__Bacteroides_thetaiotaomicron 12_Mild vs 12_Moderate   Moderate    3.210748   2.775490 Rrun_lefse
## 5        s__Bacteroides_coprocola 12_Mild vs 12_Moderate   Moderate    3.314238   1.367262 Rrun_lefse
## 6 s__Bacteroides_cellulosilyticus 12_Mild vs 12_Moderate   Moderate    3.408064   3.145457 Rrun_lefse</code></pre>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="differential-analysis.html#cb383-1" aria-hidden="true" tabindex="-1"></a>Zeybel_ps_species_lefse_R2 <span class="ot">&lt;-</span> <span class="fu">get_lefse_R</span>(</span>
<span id="cb383-2"><a href="differential-analysis.html#cb383-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">datres =</span> MGS_xmas2_output2, </span>
<span id="cb383-3"><a href="differential-analysis.html#cb383-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">&quot;Rrun_lefse2&quot;</span>,</span>
<span id="cb383-4"><a href="differential-analysis.html#cb383-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">LDA_names =</span> <span class="st">&quot;lefse_R2_LDA&quot;</span>)</span>
<span id="cb383-5"><a href="differential-analysis.html#cb383-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Zeybel_ps_species_lefse_R2)</span></code></pre></div>
<pre><code>##                            TaxaID                  Block Enrichment lefse_R2_LDA EffectSize    app_name
## 1         s__Butyricimonas_virosa 12_Mild vs 12_Moderate   Moderate     2.709350   1.668501 Rrun_lefse2
## 2       s__Bacteroides_salyersiae 12_Mild vs 12_Moderate   Moderate     2.763901   1.573816 Rrun_lefse2
## 3           s__Bacteroides_clarus 12_Mild vs 12_Moderate   Moderate     3.354092   2.331714 Rrun_lefse2
## 4 s__Bacteroides_thetaiotaomicron 12_Mild vs 12_Moderate   Moderate     3.502262   2.775490 Rrun_lefse2
## 5     s__Bacteroides_intestinalis 12_Mild vs 12_Moderate   Moderate     3.524676   1.184353 Rrun_lefse2
## 6        s__Bacteroides_coprocola 12_Mild vs 12_Moderate   Moderate     3.538227   1.367262 Rrun_lefse2</code></pre>
</div>
</div>
<div id="comparison-of-lefse-conda-with-xmas2" class="section level3 hasAnchor" number="11.6.4">
<h3><span class="header-section-number">11.6.4</span> Comparison of lefse-conda with XMAS2<a href="differential-analysis.html#comparison-of-lefse-conda-with-xmas2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="number-of-features-reported-as-significant" class="section level4 hasAnchor" number="11.6.4.1">
<h4><span class="header-section-number">11.6.4.1</span> Number of features reported as significant<a href="differential-analysis.html#number-of-features-reported-as-significant" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>amplicon_ps_genus_lefse</li>
</ul>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="differential-analysis.html#cb385-1" aria-hidden="true" tabindex="-1"></a>plot_signif_taxa_num <span class="ot">&lt;-</span> <span class="cf">function</span>(dat1, dat2, dat3, dat4) {</span>
<span id="cb385-2"><a href="differential-analysis.html#cb385-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb385-3"><a href="differential-analysis.html#cb385-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dat1 = amplicon_ps_genus_lefse_conda</span></span>
<span id="cb385-4"><a href="differential-analysis.html#cb385-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dat2 = amplicon_ps_genus_lefse_galaxy</span></span>
<span id="cb385-5"><a href="differential-analysis.html#cb385-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dat3 = amplicon_ps_genus_lefse_R</span></span>
<span id="cb385-6"><a href="differential-analysis.html#cb385-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dat4 = amplicon_ps_genus_lefse_R2</span></span>
<span id="cb385-7"><a href="differential-analysis.html#cb385-7" aria-hidden="true" tabindex="-1"></a>                     </span>
<span id="cb385-8"><a href="differential-analysis.html#cb385-8" aria-hidden="true" tabindex="-1"></a>    combined_outputs <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(dat1, dat2, dat3, dat4) <span class="sc">%&gt;%</span> </span>
<span id="cb385-9"><a href="differential-analysis.html#cb385-9" aria-hidden="true" tabindex="-1"></a>       dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">LDA =</span> <span class="fu">coalesce</span>(lefse_conda_LDA, </span>
<span id="cb385-10"><a href="differential-analysis.html#cb385-10" aria-hidden="true" tabindex="-1"></a>                                    lefse_galaxy_LDA, </span>
<span id="cb385-11"><a href="differential-analysis.html#cb385-11" aria-hidden="true" tabindex="-1"></a>                                    lefse_R_LDA, </span>
<span id="cb385-12"><a href="differential-analysis.html#cb385-12" aria-hidden="true" tabindex="-1"></a>                                    lefse_R2_LDA))</span>
<span id="cb385-13"><a href="differential-analysis.html#cb385-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb385-14"><a href="differential-analysis.html#cb385-14" aria-hidden="true" tabindex="-1"></a>    pl <span class="ot">&lt;-</span> combined_outputs <span class="sc">%&gt;%</span> </span>
<span id="cb385-15"><a href="differential-analysis.html#cb385-15" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">count</span>(app_name) <span class="sc">%&gt;%</span> </span>
<span id="cb385-16"><a href="differential-analysis.html#cb385-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggplot</span>(<span class="fu">aes</span>(app_name, n)) <span class="sc">+</span></span>
<span id="cb385-17"><a href="differential-analysis.html#cb385-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb385-18"><a href="differential-analysis.html#cb385-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_label</span>(<span class="fu">aes</span>(<span class="at">label =</span> n)) <span class="sc">+</span></span>
<span id="cb385-19"><a href="differential-analysis.html#cb385-19" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">&#39;Number of significiant features identified by the different applications using lefse&#39;</span>) </span>
<span id="cb385-20"><a href="differential-analysis.html#cb385-20" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb385-21"><a href="differential-analysis.html#cb385-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(pl)</span>
<span id="cb385-22"><a href="differential-analysis.html#cb385-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb385-23"><a href="differential-analysis.html#cb385-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb385-24"><a href="differential-analysis.html#cb385-24" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_signif_taxa_num</span>(<span class="at">dat1 =</span> amplicon_ps_genus_lefse_conda, </span>
<span id="cb385-25"><a href="differential-analysis.html#cb385-25" aria-hidden="true" tabindex="-1"></a>                     <span class="at">dat2 =</span> amplicon_ps_genus_lefse_galaxy, </span>
<span id="cb385-26"><a href="differential-analysis.html#cb385-26" aria-hidden="true" tabindex="-1"></a>                     <span class="at">dat3 =</span> amplicon_ps_genus_lefse_R, </span>
<span id="cb385-27"><a href="differential-analysis.html#cb385-27" aria-hidden="true" tabindex="-1"></a>                     <span class="at">dat4 =</span> amplicon_ps_genus_lefse_R2)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-202"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-202-1.png" alt="Number of significiant features identified by the different applications using lefse (16s)" width="768" />
<p class="caption">
Figure 11.8: Number of significiant features identified by the different applications using lefse (16s)
</p>
</div>
<ul>
<li>Zeybel_ps_species_lefse</li>
</ul>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb386-1"><a href="differential-analysis.html#cb386-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_signif_taxa_num</span>(<span class="at">dat1 =</span> Zeybel_ps_species_lefse_conda, </span>
<span id="cb386-2"><a href="differential-analysis.html#cb386-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">dat2 =</span> Zeybel_ps_species_lefse_galaxy, </span>
<span id="cb386-3"><a href="differential-analysis.html#cb386-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">dat3 =</span> Zeybel_ps_species_lefse_R, </span>
<span id="cb386-4"><a href="differential-analysis.html#cb386-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">dat4 =</span> Zeybel_ps_species_lefse_R2)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-203"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-203-1.png" alt="Number of significiant features identified by the different applications using lefse (MGS)" width="768" />
<p class="caption">
Figure 11.9: Number of significiant features identified by the different applications using lefse (MGS)
</p>
</div>
</div>
<div id="overlap-of-features-reported-as-significant" class="section level4 hasAnchor" number="11.6.4.2">
<h4><span class="header-section-number">11.6.4.2</span> Overlap of features reported as significant<a href="differential-analysis.html#overlap-of-features-reported-as-significant" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>amplicon_ps_genus_lefse</li>
</ul>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="differential-analysis.html#cb387-1" aria-hidden="true" tabindex="-1"></a>plot_signif_taxa_venn <span class="ot">&lt;-</span> <span class="cf">function</span>(dat1, dat2, dat3, dat4) {</span>
<span id="cb387-2"><a href="differential-analysis.html#cb387-2" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb387-3"><a href="differential-analysis.html#cb387-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dat1 = amplicon_ps_genus_lefse_conda</span></span>
<span id="cb387-4"><a href="differential-analysis.html#cb387-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dat2 = amplicon_ps_genus_lefse_galaxy</span></span>
<span id="cb387-5"><a href="differential-analysis.html#cb387-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dat3 = amplicon_ps_genus_lefse_R</span></span>
<span id="cb387-6"><a href="differential-analysis.html#cb387-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># dat4 = amplicon_ps_genus_lefse_R2</span></span>
<span id="cb387-7"><a href="differential-analysis.html#cb387-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb387-8"><a href="differential-analysis.html#cb387-8" aria-hidden="true" tabindex="-1"></a>    set1 <span class="ot">=</span> dat1<span class="sc">$</span>TaxaID</span>
<span id="cb387-9"><a href="differential-analysis.html#cb387-9" aria-hidden="true" tabindex="-1"></a>    set2 <span class="ot">=</span> dat2<span class="sc">$</span>TaxaID</span>
<span id="cb387-10"><a href="differential-analysis.html#cb387-10" aria-hidden="true" tabindex="-1"></a>    set3 <span class="ot">=</span> dat3<span class="sc">$</span>TaxaID</span>
<span id="cb387-11"><a href="differential-analysis.html#cb387-11" aria-hidden="true" tabindex="-1"></a>    set4 <span class="ot">=</span> dat4<span class="sc">$</span>TaxaID</span>
<span id="cb387-12"><a href="differential-analysis.html#cb387-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb387-13"><a href="differential-analysis.html#cb387-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grid.newpage</span>()</span>
<span id="cb387-14"><a href="differential-analysis.html#cb387-14" aria-hidden="true" tabindex="-1"></a>    venn_object <span class="ot">&lt;-</span> <span class="fu">venn.diagram</span>(</span>
<span id="cb387-15"><a href="differential-analysis.html#cb387-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">x =</span> <span class="fu">list</span>(set1, set2, set3, set4),</span>
<span id="cb387-16"><a href="differential-analysis.html#cb387-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">category.names =</span> <span class="fu">c</span>(<span class="st">&quot;lefse-conda&quot;</span>, <span class="st">&quot;lefse-galaxy&quot;</span>, </span>
<span id="cb387-17"><a href="differential-analysis.html#cb387-17" aria-hidden="true" tabindex="-1"></a>                           <span class="st">&quot;run_lefse(lefser)&quot;</span>, <span class="st">&quot;run_lefse2(microbiomeMarker)&quot;</span>),</span>
<span id="cb387-18"><a href="differential-analysis.html#cb387-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">filename =</span> <span class="cn">NULL</span></span>
<span id="cb387-19"><a href="differential-analysis.html#cb387-19" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb387-20"><a href="differential-analysis.html#cb387-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grid.draw</span>(venn_object)    </span>
<span id="cb387-21"><a href="differential-analysis.html#cb387-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb387-22"><a href="differential-analysis.html#cb387-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb387-23"><a href="differential-analysis.html#cb387-23" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_signif_taxa_venn</span>(<span class="at">dat1 =</span> amplicon_ps_genus_lefse_conda, </span>
<span id="cb387-24"><a href="differential-analysis.html#cb387-24" aria-hidden="true" tabindex="-1"></a>                      <span class="at">dat2 =</span> amplicon_ps_genus_lefse_galaxy, </span>
<span id="cb387-25"><a href="differential-analysis.html#cb387-25" aria-hidden="true" tabindex="-1"></a>                      <span class="at">dat3 =</span> amplicon_ps_genus_lefse_R, </span>
<span id="cb387-26"><a href="differential-analysis.html#cb387-26" aria-hidden="true" tabindex="-1"></a>                      <span class="at">dat4 =</span> amplicon_ps_genus_lefse_R2)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-204"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-204-1.png" alt="Venn (16s)" width="768" />
<p class="caption">
Figure 11.10: Venn (16s)
</p>
</div>
<ul>
<li>Zeybel_ps_species_lefse</li>
</ul>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="differential-analysis.html#cb388-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_signif_taxa_venn</span>(<span class="at">dat1 =</span> Zeybel_ps_species_lefse_conda, </span>
<span id="cb388-2"><a href="differential-analysis.html#cb388-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">dat2 =</span> Zeybel_ps_species_lefse_galaxy, </span>
<span id="cb388-3"><a href="differential-analysis.html#cb388-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">dat3 =</span> Zeybel_ps_species_lefse_R, </span>
<span id="cb388-4"><a href="differential-analysis.html#cb388-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">dat4 =</span> Zeybel_ps_species_lefse_R2)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-205"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-205-1.png" alt="Venn (MGS)" width="768" />
<p class="caption">
Figure 11.11: Venn (MGS)
</p>
</div>
</div>
<div id="lda-scores-comparison" class="section level4 hasAnchor" number="11.6.4.3">
<h4><span class="header-section-number">11.6.4.3</span> LDA scores’ comparison<a href="differential-analysis.html#lda-scores-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>LDA scores of the 14 overlapping features are similar.</p>
<ul>
<li>amplicon_ps_genus_lefse</li>
</ul>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="differential-analysis.html#cb389-1" aria-hidden="true" tabindex="-1"></a>amplicon_joint_output <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">reduce</span>(</span>
<span id="cb389-2"><a href="differential-analysis.html#cb389-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">.x =</span> <span class="fu">list</span>(amplicon_ps_genus_lefse_conda, amplicon_ps_genus_lefse_galaxy, </span>
<span id="cb389-3"><a href="differential-analysis.html#cb389-3" aria-hidden="true" tabindex="-1"></a>              amplicon_ps_genus_lefse_R, amplicon_ps_genus_lefse_R2),</span>
<span id="cb389-4"><a href="differential-analysis.html#cb389-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">.f =</span> <span class="sc">~</span> <span class="fu">inner_join</span>(.x, .y, <span class="at">by =</span> <span class="st">&quot;TaxaID&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb389-5"><a href="differential-analysis.html#cb389-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(TaxaID, lefse_conda_LDA, lefse_galaxy_LDA, </span>
<span id="cb389-6"><a href="differential-analysis.html#cb389-6" aria-hidden="true" tabindex="-1"></a>                  lefse_R_LDA, lefse_R2_LDA)</span>
<span id="cb389-7"><a href="differential-analysis.html#cb389-7" aria-hidden="true" tabindex="-1"></a>amplicon_joint_output</span></code></pre></div>
<pre><code>## # A tibble: 57 × 5
##    TaxaID                          lefse_conda_LDA lefse_galaxy_LDA lefse_R_LDA lefse_R2_LDA
##    &lt;chr&gt;                                     &lt;dbl&gt;            &lt;dbl&gt;       &lt;dbl&gt;        &lt;dbl&gt;
##  1 g__Bacteroides                            -5.47            -5.46       -5.48        -5.76
##  2 g__Lachnospiraceae_unclassified           -4.47            -4.45       -4.45        -4.77
##  3 g__Ruminococcaceae_unclassified           -4.46            -4.41       -4.43        -4.76
##  4 g__Lachnospira                            -4.36            -4.38       -4.33        -4.68
##  5 g__Faecalibacterium                       -4.20            -4.18       -4.14        -4.45
##  6 g__Phascolarctobacterium                  -4.15            -4.18       -4.21        -4.44
##  7 g__Clostridiales_unclassified             -4.03            -4.06       -4.02        -4.37
##  8 g__Akkermansia                            -3.91            -4.04       -4.08        -4.22
##  9 g__Oscillospira                           -3.87            -3.89       -3.87        -4.20
## 10 g__Rikenellaceae_unclassified             -3.87            -3.91       -3.85        -4.27
## # … with 47 more rows</code></pre>
<ul>
<li>Zeybel_ps_species_lefse</li>
</ul>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="differential-analysis.html#cb391-1" aria-hidden="true" tabindex="-1"></a>MGS_joint_output <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">reduce</span>(</span>
<span id="cb391-2"><a href="differential-analysis.html#cb391-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">.x =</span> <span class="fu">list</span>(Zeybel_ps_species_lefse_conda, Zeybel_ps_species_lefse_galaxy, </span>
<span id="cb391-3"><a href="differential-analysis.html#cb391-3" aria-hidden="true" tabindex="-1"></a>              Zeybel_ps_species_lefse_R, Zeybel_ps_species_lefse_R2),</span>
<span id="cb391-4"><a href="differential-analysis.html#cb391-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">.f =</span> <span class="sc">~</span> <span class="fu">inner_join</span>(.x, .y, <span class="at">by =</span> <span class="st">&quot;TaxaID&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb391-5"><a href="differential-analysis.html#cb391-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(TaxaID, lefse_conda_LDA, lefse_galaxy_LDA, </span>
<span id="cb391-6"><a href="differential-analysis.html#cb391-6" aria-hidden="true" tabindex="-1"></a>                  lefse_R_LDA, lefse_R2_LDA)</span>
<span id="cb391-7"><a href="differential-analysis.html#cb391-7" aria-hidden="true" tabindex="-1"></a>MGS_joint_output</span></code></pre></div>
<pre><code>## # A tibble: 11 × 5
##    TaxaID                          lefse_conda_LDA lefse_galaxy_LDA lefse_R_LDA lefse_R2_LDA
##    &lt;chr&gt;                                     &lt;dbl&gt;            &lt;dbl&gt;       &lt;dbl&gt;        &lt;dbl&gt;
##  1 s__Butyricimonas_virosa                    2.46             2.82        2.45         2.71
##  2 s__Bacteroides_salyersiae                  2.47             3.16        2.66         2.76
##  3 s__Bacteroides_clarus                      3.07             3.29        3.19         3.35
##  4 s__Bacteroides_thetaiotaomicron            3.26             3.28        3.21         3.50
##  5 s__Bacteroides_coprocola                   3.28             3.24        3.31         3.54
##  6 s__Bacteroides_intestinalis                3.32             3.31        3.41         3.52
##  7 s__Bacteroides_cellulosilyticus            3.41             3.36        3.41         3.71
##  8 s__Bacteroides_eggerthii                   3.52             3.59        3.47         3.80
##  9 s__Parabacteroides_merdae                  3.59             3.68        3.54         3.93
## 10 s__Barnesiella_intestinihominis            3.68             3.72        3.67         3.95
## 11 s__Prevotella_sp_CAG_520                   4.04             4.26        4.38         4.63</code></pre>
<div id="xmas2-lda-scores-vs-lefse-conda-lda-scores" class="section level5 hasAnchor" number="11.6.4.3.1">
<h5><span class="header-section-number">11.6.4.3.1</span> XMAS2 LDA scores vs lefse-conda LDA scores<a href="differential-analysis.html#xmas2-lda-scores-vs-lefse-conda-lda-scores" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div id="amplicon_ps_genus_lefse" class="section level6 hasAnchor" number="11.6.4.3.1.1">
<h6><span class="header-section-number">11.6.4.3.1.1</span> amplicon_ps_genus_lefse<a href="differential-analysis.html#amplicon_ps_genus_lefse" class="anchor-section" aria-label="Anchor link to header"></a></h6>
<ul>
<li>run_lefse (lefser R package) vs lefse-conda</li>
</ul>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="differential-analysis.html#cb393-1" aria-hidden="true" tabindex="-1"></a>amplicon_joint_output <span class="sc">%&gt;%</span> </span>
<span id="cb393-2"><a href="differential-analysis.html#cb393-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(lefse_conda_LDA, lefse_R_LDA)) <span class="sc">+</span> </span>
<span id="cb393-3"><a href="differential-analysis.html#cb393-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb393-4"><a href="differential-analysis.html#cb393-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb393-5"><a href="differential-analysis.html#cb393-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb393-6"><a href="differential-analysis.html#cb393-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Comparison of LDA scores of features reported as significant </span></span>
<span id="cb393-7"><a href="differential-analysis.html#cb393-7" aria-hidden="true" tabindex="-1"></a><span class="st">    by both lefse-conda and run_lefse&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-208"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-208-1.png" alt="Comparison of LDA scores of features reported as significant by both lefse-conda and run_lefse (16s)" width="768" />
<p class="caption">
Figure 11.12: Comparison of LDA scores of features reported as significant by both lefse-conda and run_lefse (16s)
</p>
</div>
<ul>
<li>run_lefse2 (microbiomeMarker R package) vs lefse-conda</li>
</ul>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="differential-analysis.html#cb394-1" aria-hidden="true" tabindex="-1"></a>amplicon_joint_output <span class="sc">%&gt;%</span> </span>
<span id="cb394-2"><a href="differential-analysis.html#cb394-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(lefse_conda_LDA, lefse_R2_LDA)) <span class="sc">+</span> </span>
<span id="cb394-3"><a href="differential-analysis.html#cb394-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb394-4"><a href="differential-analysis.html#cb394-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb394-5"><a href="differential-analysis.html#cb394-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb394-6"><a href="differential-analysis.html#cb394-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Comparison of LDA scores of features reported as significant </span></span>
<span id="cb394-7"><a href="differential-analysis.html#cb394-7" aria-hidden="true" tabindex="-1"></a><span class="st">    by both lefse-conda and run_lefse2&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-209"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-209-1.png" alt="Comparison of LDA scores of features reported as significant by both lefse-conda and run_lefse2 (16s)" width="768" />
<p class="caption">
Figure 11.13: Comparison of LDA scores of features reported as significant by both lefse-conda and run_lefse2 (16s)
</p>
</div>
</div>
<div id="zeybel_ps_species_lefse" class="section level6 hasAnchor" number="11.6.4.3.1.2">
<h6><span class="header-section-number">11.6.4.3.1.2</span> Zeybel_ps_species_lefse<a href="differential-analysis.html#zeybel_ps_species_lefse" class="anchor-section" aria-label="Anchor link to header"></a></h6>
<ul>
<li>run_lefse (lefser R package) vs lefse-conda</li>
</ul>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="differential-analysis.html#cb395-1" aria-hidden="true" tabindex="-1"></a>MGS_joint_output <span class="sc">%&gt;%</span> </span>
<span id="cb395-2"><a href="differential-analysis.html#cb395-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(lefse_conda_LDA, lefse_R_LDA)) <span class="sc">+</span> </span>
<span id="cb395-3"><a href="differential-analysis.html#cb395-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb395-4"><a href="differential-analysis.html#cb395-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb395-5"><a href="differential-analysis.html#cb395-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb395-6"><a href="differential-analysis.html#cb395-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Comparison of LDA scores of features reported as significant </span></span>
<span id="cb395-7"><a href="differential-analysis.html#cb395-7" aria-hidden="true" tabindex="-1"></a><span class="st">    by both lefse-conda and run_lefse&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-210"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-210-1.png" alt="Comparison of LDA scores of features reported as significant by both lefse-conda and run_lefse (MGS)" width="768" />
<p class="caption">
Figure 11.14: Comparison of LDA scores of features reported as significant by both lefse-conda and run_lefse (MGS)
</p>
</div>
<ul>
<li>run_lefse2 (microbiomeMarker R package) vs lefse-conda</li>
</ul>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="differential-analysis.html#cb396-1" aria-hidden="true" tabindex="-1"></a>MGS_joint_output <span class="sc">%&gt;%</span> </span>
<span id="cb396-2"><a href="differential-analysis.html#cb396-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(lefse_conda_LDA, lefse_R2_LDA)) <span class="sc">+</span> </span>
<span id="cb396-3"><a href="differential-analysis.html#cb396-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb396-4"><a href="differential-analysis.html#cb396-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb396-5"><a href="differential-analysis.html#cb396-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb396-6"><a href="differential-analysis.html#cb396-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Comparison of LDA scores of features reported as significant </span></span>
<span id="cb396-7"><a href="differential-analysis.html#cb396-7" aria-hidden="true" tabindex="-1"></a><span class="st">    by both lefse-conda and run_lefse2&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-211"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-211-1.png" alt="Comparison of LDA scores of features reported as significant by both lefse-conda and run_lefse2 (MGS)" width="768" />
<p class="caption">
Figure 11.15: Comparison of LDA scores of features reported as significant by both lefse-conda and run_lefse2 (MGS)
</p>
</div>
</div>
</div>
<div id="xmas2-lda-scores-vs-lefse-galaxy-lda-scores" class="section level5 hasAnchor" number="11.6.4.3.2">
<h5><span class="header-section-number">11.6.4.3.2</span> XMAS2 LDA scores vs lefse-galaxy LDA scores<a href="differential-analysis.html#xmas2-lda-scores-vs-lefse-galaxy-lda-scores" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div id="amplicon_ps_genus_lefse-1" class="section level6 hasAnchor" number="11.6.4.3.2.1">
<h6><span class="header-section-number">11.6.4.3.2.1</span> amplicon_ps_genus_lefse<a href="differential-analysis.html#amplicon_ps_genus_lefse-1" class="anchor-section" aria-label="Anchor link to header"></a></h6>
<ul>
<li>run_lefse (lefser R package) vs lefse-galaxy</li>
</ul>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="differential-analysis.html#cb397-1" aria-hidden="true" tabindex="-1"></a>amplicon_joint_output <span class="sc">%&gt;%</span> </span>
<span id="cb397-2"><a href="differential-analysis.html#cb397-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(lefse_galaxy_LDA, lefse_R_LDA)) <span class="sc">+</span> </span>
<span id="cb397-3"><a href="differential-analysis.html#cb397-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb397-4"><a href="differential-analysis.html#cb397-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb397-5"><a href="differential-analysis.html#cb397-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb397-6"><a href="differential-analysis.html#cb397-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Comparison of LDA scores of features reported as significant </span></span>
<span id="cb397-7"><a href="differential-analysis.html#cb397-7" aria-hidden="true" tabindex="-1"></a><span class="st">    by both lefse-galaxy and run_lefse&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-212"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-212-1.png" alt="Comparison of LDA scores of features reported as significant by both lefse-galaxy and run_lefse (16s)" width="768" />
<p class="caption">
Figure 11.16: Comparison of LDA scores of features reported as significant by both lefse-galaxy and run_lefse (16s)
</p>
</div>
<ul>
<li>run_lefse2 (microbiomeMarker R package) vs lefse-galaxy</li>
</ul>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="differential-analysis.html#cb398-1" aria-hidden="true" tabindex="-1"></a>amplicon_joint_output <span class="sc">%&gt;%</span> </span>
<span id="cb398-2"><a href="differential-analysis.html#cb398-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(lefse_galaxy_LDA, lefse_R2_LDA)) <span class="sc">+</span> </span>
<span id="cb398-3"><a href="differential-analysis.html#cb398-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb398-4"><a href="differential-analysis.html#cb398-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb398-5"><a href="differential-analysis.html#cb398-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb398-6"><a href="differential-analysis.html#cb398-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Comparison of LDA scores of features reported as significant </span></span>
<span id="cb398-7"><a href="differential-analysis.html#cb398-7" aria-hidden="true" tabindex="-1"></a><span class="st">    by both lefse-galaxy and run_lefse2&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-213"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-213-1.png" alt="Comparison of LDA scores of features reported as significant by both lefse-galaxy and run_lefse2 (16s)" width="768" />
<p class="caption">
Figure 11.17: Comparison of LDA scores of features reported as significant by both lefse-galaxy and run_lefse2 (16s)
</p>
</div>
</div>
<div id="zeybel_ps_species_lefse-1" class="section level6 hasAnchor" number="11.6.4.3.2.2">
<h6><span class="header-section-number">11.6.4.3.2.2</span> Zeybel_ps_species_lefse<a href="differential-analysis.html#zeybel_ps_species_lefse-1" class="anchor-section" aria-label="Anchor link to header"></a></h6>
<ul>
<li>run_lefse (lefser R package) vs lefse-galaxy</li>
</ul>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="differential-analysis.html#cb399-1" aria-hidden="true" tabindex="-1"></a>MGS_joint_output <span class="sc">%&gt;%</span> </span>
<span id="cb399-2"><a href="differential-analysis.html#cb399-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(lefse_galaxy_LDA, lefse_R_LDA)) <span class="sc">+</span> </span>
<span id="cb399-3"><a href="differential-analysis.html#cb399-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb399-4"><a href="differential-analysis.html#cb399-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb399-5"><a href="differential-analysis.html#cb399-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb399-6"><a href="differential-analysis.html#cb399-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Comparison of LDA scores of features reported as significant </span></span>
<span id="cb399-7"><a href="differential-analysis.html#cb399-7" aria-hidden="true" tabindex="-1"></a><span class="st">    by both lefse-galaxy and run_lefse&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-214"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-214-1.png" alt="Comparison of LDA scores of features reported as significant by both lefse-galaxy and run_lefse (MGS)" width="768" />
<p class="caption">
Figure 11.18: Comparison of LDA scores of features reported as significant by both lefse-galaxy and run_lefse (MGS)
</p>
</div>
<ul>
<li>run_lefse2 (microbiomeMarker R package) vs lefse-galaxy</li>
</ul>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="differential-analysis.html#cb400-1" aria-hidden="true" tabindex="-1"></a>MGS_joint_output <span class="sc">%&gt;%</span> </span>
<span id="cb400-2"><a href="differential-analysis.html#cb400-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(lefse_galaxy_LDA, lefse_R2_LDA)) <span class="sc">+</span> </span>
<span id="cb400-3"><a href="differential-analysis.html#cb400-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb400-4"><a href="differential-analysis.html#cb400-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb400-5"><a href="differential-analysis.html#cb400-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>) <span class="sc">+</span></span>
<span id="cb400-6"><a href="differential-analysis.html#cb400-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Comparison of LDA scores of features reported as significant </span></span>
<span id="cb400-7"><a href="differential-analysis.html#cb400-7" aria-hidden="true" tabindex="-1"></a><span class="st">    by both lefse-galaxy and run_lefse2&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-215"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-215-1.png" alt="Comparison of LDA scores of features reported as significant by both lefse-galaxy and run_lefse2 (MGS)" width="768" />
<p class="caption">
Figure 11.19: Comparison of LDA scores of features reported as significant by both lefse-galaxy and run_lefse2 (MGS)
</p>
</div>
<p><strong>Results:</strong></p>
<ul>
<li>The overlap between <code>run_lefse2</code> (microbiomeMarker R package) and lefse-conda or lefse-galaxy have the similar LDA scores. However, the overlap between <code>run_lefse</code> (lefser R package) and lefse-conda or lefse-galaxy seem have slightly different LDA scores.</li>
</ul>
</div>
</div>
<div id="differences-bewteen-xmas2-lda-scores-and-lefse-conda" class="section level5 hasAnchor" number="11.6.4.3.3">
<h5><span class="header-section-number">11.6.4.3.3</span> Differences bewteen XMAS2 LDA scores and lefse-conda<a href="differential-analysis.html#differences-bewteen-xmas2-lda-scores-and-lefse-conda" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb401-1"><a href="differential-analysis.html#cb401-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(amplicon_ps_genus_lefse_conda<span class="sc">$</span>TaxaID, amplicon_ps_genus_lefse_R<span class="sc">$</span>TaxaID)</span></code></pre></div>
<pre><code>## [1] &quot;g__Holdemania&quot;                  &quot;g__Alcaligenaceae_unclassified&quot;</code></pre>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="differential-analysis.html#cb403-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(amplicon_ps_genus_lefse_conda<span class="sc">$</span>TaxaID, amplicon_ps_genus_lefse_R2<span class="sc">$</span>TaxaID)</span></code></pre></div>
<pre><code>## character(0)</code></pre>
</div>
</div>
</div>
</div>
<div id="systematic-information-8" class="section level2 hasAnchor" number="11.7">
<h2><span class="header-section-number">11.7</span> Systematic Information<a href="differential-analysis.html#systematic-information-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="differential-analysis.html#cb405-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ──────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-08-09
##  rstudio  2022.07.1+554 Spotted Wakerobin (desktop)
##  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version  date (UTC) lib source
##  abind                  1.4-5    2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18   2021-09-16 [1] CRAN (R 4.1.0)
##  ALDEx2                 1.26.0   2021-10-26 [1] Bioconductor
##  annotate               1.72.0   2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2   2021-11-09 [1] Bioconductor
##  ape                  * 5.6-2    2022-03-02 [1] CRAN (R 4.1.2)
##  askpass                1.1      2019-01-13 [1] CRAN (R 4.1.0)
##  assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1    2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3    2015-07-28 [1] CRAN (R 4.1.0)
##  bayesm                 3.1-4    2019-10-15 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0   2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0   2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3   2021-12-09 [1] Bioconductor
##  biomformat             1.22.0   2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0   2021-10-26 [1] Bioconductor
##  bit                    4.0.4    2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5    2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7    2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2    2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.27     2022-06-14 [1] CRAN (R 4.1.2)
##  brio                   1.1.3    2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12   2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1    2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6    2021-08-19 [1] CRAN (R 4.1.0)
##  callr                  3.7.0    2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12   2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5    2022-01-06 [1] CRAN (R 4.1.2)
##  caTools                1.18.2   2021-03-28 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0    2020-02-06 [1] CRAN (R 4.1.0)
##  cli                    3.3.0    2022-04-25 [1] CRAN (R 4.1.2)
##  cluster                2.1.2    2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18   2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2    2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3    2022-02-21 [1] CRAN (R 4.1.2)
##  compositions           2.0-4    2022-01-05 [1] CRAN (R 4.1.2)
##  conflicted           * 1.1.0    2021-11-26 [1] CRAN (R 4.1.0)
##  corrplot               0.92     2021-11-18 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1    2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0    2022-02-14 [1] CRAN (R 4.1.2)
##  crosstalk              1.2.0    2021-11-04 [1] CRAN (R 4.1.0)
##  data.table             1.14.2   2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2    2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0   2021-10-26 [1] Bioconductor
##  DEoptimR               1.0-10   2022-01-03 [1] CRAN (R 4.1.2)
##  desc                   1.4.1    2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0   2021-10-26 [1] Bioconductor
##  devtools             * 2.4.3    2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29   2021-12-01 [1] CRAN (R 4.1.0)
##  dplyr                * 1.0.8    2022-02-08 [1] CRAN (R 4.1.2)
##  DT                     0.21     2022-02-26 [1] CRAN (R 4.1.2)
##  edgeR                  3.36.0   2021-10-26 [1] Bioconductor
##  ellipsis               0.3.2    2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15     2022-02-18 [1] CRAN (R 4.1.2)
##  FactoMineR             2.4      2020-12-11 [1] CRAN (R 4.1.0)
##  fansi                  1.0.2    2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0    2021-02-28 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0    2021-01-25 [1] CRAN (R 4.1.0)
##  flashClust             1.01-2   2012-08-21 [1] CRAN (R 4.1.0)
##  foreach                1.5.2    2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82   2022-01-13 [1] CRAN (R 4.1.2)
##  formatR                1.11     2021-06-01 [1] CRAN (R 4.1.0)
##  Formula                1.2-4    2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2    2021-12-08 [1] CRAN (R 4.1.0)
##  futile.logger        * 1.4.3    2016-07-10 [1] CRAN (R 4.1.0)
##  futile.options         1.0.1    2018-04-20 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0   2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0   2021-10-26 [1] Bioconductor
##  generics               0.1.2    2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1   2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7    2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1   2021-11-18 [1] Bioconductor
##  ggplot2              * 3.3.5    2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr               * 0.4.0    2020-06-27 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1    2021-01-15 [1] CRAN (R 4.1.0)
##  ggsci                  2.9      2018-05-14 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3    2021-09-09 [1] CRAN (R 4.1.0)
##  glmnet                 4.1-3    2021-11-02 [1] CRAN (R 4.1.0)
##  glue                   1.6.2    2022-02-24 [1] CRAN (R 4.1.2)
##  gplots                 3.1.1    2020-11-28 [1] CRAN (R 4.1.0)
##  gridExtra              2.3      2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2    2021-06-06 [1] CRAN (R 4.1.0)
##  highr                  0.9      2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0    2021-10-07 [1] CRAN (R 4.1.0)
##  hms                    1.1.1    2021-09-26 [1] CRAN (R 4.1.0)
##  htmlTable              2.4.0    2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2    2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4    2021-09-08 [1] CRAN (R 4.1.0)
##  httr                   1.4.2    2020-07-20 [1] CRAN (R 4.1.0)
##  igraph                 1.2.11   2022-01-04 [1] CRAN (R 4.1.2)
##  IRanges              * 2.28.0   2021-10-26 [1] Bioconductor
##  iterators              1.0.14   2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9    2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4    2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0    2022-02-22 [1] CRAN (R 4.1.2)
##  kableExtra             1.3.4    2021-02-20 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0   2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20  2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.39     2022-04-26 [1] CRAN (R 4.1.2)
##  labeling               0.4.2    2020-10-20 [1] CRAN (R 4.1.0)
##  lambda.r               1.2.4    2019-09-18 [1] CRAN (R 4.1.0)
##  lattice              * 0.20-45  2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.1.0)
##  leaps                  3.1      2020-01-16 [1] CRAN (R 4.1.0)
##  libcoin                1.0-9    2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1    2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1   2022-02-17 [1] Bioconductor
##  locfit                 1.5-9.5  2022-03-03 [1] CRAN (R 4.1.2)
##  LOCOM                  1.1      2022-08-05 [1] Github (yijuanhu/LOCOM@c181e0f)
##  magrittr             * 2.0.2    2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55   2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix                 1.4-0    2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0    2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0   2021-09-17 [1] CRAN (R 4.1.0)
##  mbzinb                 0.2      2022-03-16 [1] local
##  memoise                2.0.1    2021-11-26 [1] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0   2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39   2022-02-24 [1] CRAN (R 4.1.2)
##  microbiome             1.16.0   2021-10-26 [1] Bioconductor
##  modeltools             0.2-23   2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18   2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0   2021-10-26 [1] Bioconductor
##  munsell                0.5.0    2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3    2021-10-08 [1] CRAN (R 4.1.0)
##  NADA                   1.6-1.1  2020-03-22 [1] CRAN (R 4.1.0)
##  nlme                 * 3.1-155  2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17   2022-01-13 [1] CRAN (R 4.1.2)
##  openssl                2.0.0    2022-03-02 [1] CRAN (R 4.1.2)
##  permute              * 0.9-7    2022-01-27 [1] CRAN (R 4.1.2)
##  pheatmap               1.0.12   2019-01-04 [1] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0   2021-10-26 [1] Bioconductor
##  picante              * 1.8.2    2020-06-10 [1] CRAN (R 4.1.0)
##  pillar                 1.7.0    2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1    2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4    2021-11-30 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7    2013-12-03 [1] CRAN (R 4.1.0)
##  prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.1.0)
##  processx               3.5.2    2021-04-30 [1] CRAN (R 4.1.0)
##  protoclust             1.6.3    2019-01-31 [1] CRAN (R 4.1.0)
##  ps                     1.6.0    2021-02-28 [1] CRAN (R 4.1.0)
##  pscl                   1.5.5    2020-03-07 [1] CRAN (R 4.1.0)
##  purrr                * 0.3.4    2020-04-17 [1] CRAN (R 4.1.0)
##  qvalue                 2.26.0   2021-10-26 [1] Bioconductor
##  R6                     2.5.1    2021-08-19 [1] CRAN (R 4.1.0)
##  RAIDA                  1.0      2022-03-14 [1] local
##  RColorBrewer         * 1.1-2    2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                   1.0.8.2  2022-03-11 [1] CRAN (R 4.1.2)
##  RcppZiggurat           0.1.6    2020-10-20 [1] CRAN (R 4.1.0)
##  RCurl                  1.98-1.6 2022-02-08 [1] CRAN (R 4.1.2)
##  readr                * 2.1.2    2022-01-30 [1] CRAN (R 4.1.2)
##  remotes                2.4.2    2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.1.0)
##  reticulate             1.24     2022-01-26 [1] CRAN (R 4.1.2)
##  Rfast                  2.0.6    2022-02-16 [1] CRAN (R 4.1.2)
##  rhdf5                  2.38.1   2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0    2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0   2021-10-26 [1] Bioconductor
##  rlang                  1.0.2    2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.14     2022-04-25 [1] CRAN (R 4.1.2)
##  robustbase             0.93-9   2021-09-27 [1] CRAN (R 4.1.0)
##  rpart                  4.1.16   2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2    2020-11-15 [1] CRAN (R 4.1.0)
##  RSpectra               0.16-0   2019-12-01 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10   2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0    2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13     2020-11-12 [1] CRAN (R 4.1.0)
##  Rtsne                  0.15     2018-11-10 [1] CRAN (R 4.1.0)
##  rvest                  1.0.2    2021-10-16 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3   2021-11-21 [1] Bioconductor
##  sandwich               3.0-1    2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0    2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1    2020-05-11 [1] CRAN (R 4.1.0)
##  scatterplot3d          0.3-41   2018-03-14 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2    2021-12-06 [1] CRAN (R 4.1.0)
##  shape                  1.4.6    2021-05-19 [1] CRAN (R 4.1.0)
##  stringi                1.7.6    2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0    2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0   2021-10-26 [1] Bioconductor
##  survival               3.3-1    2022-03-03 [1] CRAN (R 4.1.2)
##  svglite                2.1.0    2022-02-03 [1] CRAN (R 4.1.2)
##  systemfonts            1.0.4    2022-02-11 [1] CRAN (R 4.1.2)
##  tensorA                0.36.2   2020-11-19 [1] CRAN (R 4.1.0)
##  testthat               3.1.2    2022-01-20 [1] CRAN (R 4.1.2)
##  TH.data                1.1-0    2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6    2021-11-07 [1] CRAN (R 4.1.0)
##  tidyr                * 1.2.0    2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2    2022-02-21 [1] CRAN (R 4.1.2)
##  truncnorm              1.0-8    2018-02-27 [1] CRAN (R 4.1.0)
##  tzdb                   0.2.0    2021-10-27 [1] CRAN (R 4.1.0)
##  umap                   0.2.8.0  2022-03-23 [1] CRAN (R 4.1.2)
##  usethis              * 2.1.5    2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2    2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8    2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                * 2.5-7    2020-11-28 [1] CRAN (R 4.1.0)
##  VennDiagram          * 1.7.3    2022-04-12 [1] CRAN (R 4.1.2)
##  viridis              * 0.6.2    2021-10-13 [1] CRAN (R 4.1.0)
##  viridisLite          * 0.4.0    2021-04-13 [1] CRAN (R 4.1.0)
##  vroom                  1.5.7    2021-11-30 [1] CRAN (R 4.1.0)
##  webshot                0.5.3    2022-04-14 [1] CRAN (R 4.1.2)
##  withr                  2.5.0    2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0   2021-10-26 [1] Bioconductor
##  xfun                   0.30     2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                * 2.1.7.4  2022-08-09 [1] local
##  XML                    3.99-0.9 2022-02-24 [1] CRAN (R 4.1.2)
##  xml2                   1.3.3    2021-11-30 [1] CRAN (R 4.1.0)
##  xtable                 1.8-4    2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0   2021-10-26 [1] Bioconductor
##  yaml                   2.3.5    2022-02-21 [1] CRAN (R 4.1.2)
##  zCompositions          1.4.0    2022-01-13 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0   2021-10-26 [1] Bioconductor
##  zoo                    1.8-9    2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chen2018omnibus" class="csl-entry">
Chen, Jun, Emily King, Rebecca Deek, Zhi Wei, Yue Yu, Diane Grill, and Karla Ballman. 2018. <span>“An Omnibus Test for Differential Distribution Analysis of Microbiome Sequencing Data.”</span> <em>Bioinformatics</em> 34 (4): 643–51.
</div>
<div id="ref-chen2018gmpr" class="csl-entry">
Chen, Li, James Reeve, Lujun Zhang, Shengbing Huang, Xuefeng Wang, and Jun Chen. 2018. <span>“GMPR: A Robust Normalization Method for Zero-Inflated Count Data with Application to Microbiome Sequencing Data.”</span> <em>PeerJ</em> 6: e4600.
</div>
<div id="ref-fernandes2014unifying" class="csl-entry">
Fernandes, Andrew D, Jennifer Ns Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>“Unifying the Analysis of High-Throughput Sequencing Datasets: Characterizing RNA-Seq, 16s rRNA Gene Sequencing and Selective Growth Experiments by Compositional Data Analysis.”</span> <em>Microbiome</em> 2 (1): 1–13.
</div>
<div id="ref-hu2022locom" class="csl-entry">
Hu, Yingtian, Glen A Satten, and Yi-Juan Hu. 2022. <span>“LOCOM: A Logistic Regression Model for Testing Differential Abundance in Compositional Microbiome Data with False Discovery Rate Control.”</span> <em>Proceedings of the National Academy of Sciences</em> 119 (30): e2122788119.
</div>
<div id="ref-law2014voom" class="csl-entry">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. <span>“Voom: Precision Weights Unlock Linear Model Analysis Tools for RNA-Seq Read Counts.”</span> <em>Genome Biology</em> 15 (2): 1–17.
</div>
<div id="ref-love2014moderated" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.”</span> <em>Genome Biology</em> 15 (12): 1–21.
</div>
<div id="ref-mallick2021multivariable" class="csl-entry">
Mallick, Himel, Ali Rahnavard, Lauren J McIver, Siyuan Ma, Yancong Zhang, Long H Nguyen, Timothy L Tickle, et al. 2021. <span>“Multivariable Association Discovery in Population-Scale Meta-Omics Studies.”</span> <em>PLoS Computational Biology</em> 17 (11): e1009442.
</div>
<div id="ref-mandal2015analysis" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A White, Merete Eggesbø, Rob Knight, and Shyamal D Peddada. 2015. <span>“Analysis of Composition of Microbiomes: A Novel Method for Studying Microbial Composition.”</span> <em>Microbial Ecology in Health and Disease</em> 26 (1): 27663.
</div>
<div id="ref-martin2020modeling" class="csl-entry">
Martin, Bryan D, Daniela Witten, and Amy D Willis. 2020. <span>“Modeling Microbial Abundances and Dysbiosis with Beta-Binomial Regression.”</span> <em>The Annals of Applied Statistics</em> 14 (1): 94.
</div>
<div id="ref-nearing2022microbiome" class="csl-entry">
Nearing, Jacob T, Gavin M Douglas, Molly G Hayes, Jocelyn MacDonald, Dhwani K Desai, Nicole Allward, Casey Jones, et al. 2022. <span>“Microbiome Differential Abundance Methods Produce Different Results Across 38 Datasets.”</span> <em>Nature Communications</em> 13 (1): 1–16.
</div>
<div id="ref-paulson2013differential" class="csl-entry">
Paulson, Joseph N, O Colin Stine, Héctor Corrada Bravo, and Mihai Pop. 2013. <span>“Differential Abundance Analysis for Microbial Marker-Gene Surveys.”</span> <em>Nature Methods</em> 10 (12): 1200–1202.
</div>
<div id="ref-robinson2010scaling" class="csl-entry">
Robinson, Mark D, and Alicia Oshlack. 2010. <span>“A Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data.”</span> <em>Genome Biology</em> 11 (3): 1–9.
</div>
<div id="ref-segata2011metagenomic" class="csl-entry">
Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. <span>“Metagenomic Biomarker Discovery and Explanation.”</span> <em>Genome Biology</em> 12 (6): 1–18.
</div>
<div id="ref-sohn2015robust" class="csl-entry">
Sohn, Michael B, Ruofei Du, and Lingling An. 2015. <span>“A Robust Approach for Identifying Differentially Abundant Features in Metagenomic Samples.”</span> <em>Bioinformatics</em> 31 (14): 2269–75.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="core-microbiota.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualization-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/10-Differentail-Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
