<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-05-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="core-microbiota.html"/>
<link rel="next" href="visualization-5.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation<span></span></a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette<span></span></a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors<span></span></a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2<span></span></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>4.2.1</b> Importing results from dada2 pipeline<span></span></a></li>
<li class="chapter" data-level="4.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>4.2.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>4.2.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.2.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>4.2.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.2.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>4.2.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>4.3.1</b> Importing results<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table-1"><i class="fa fa-check"></i><b>4.3.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table-1"><i class="fa fa-check"></i><b>4.3.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.3.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table-1"><i class="fa fa-check"></i><b>4.3.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.3.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object-1"><i class="fa fa-check"></i><b>4.3.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Reads’ track<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics<span></span></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.6</b> Analysis of Similarity (ANOSIM)<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.7</b> Ordination<span></span></a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.7.1</b> Principal Component Analysis (PCA)<span></span></a></li>
<li class="chapter" data-level="8.7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.7.2</b> Principal Coordinate Analysis (PCoA)<span></span></a></li>
<li class="chapter" data-level="8.7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.7.3</b> t-distributed stochastic neighbor embedding (t-SNE)<span></span></a></li>
<li class="chapter" data-level="8.7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.7.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="composition.html"><a href="composition.html"><i class="fa fa-check"></i><b>9</b> Composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="composition.html"><a href="composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="composition.html"><a href="composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.2</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="composition.html"><a href="composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.3</b> XVIZ <code>plot_stacked_bar</code><span></span></a></li>
<li class="chapter" data-level="9.4" data-path="composition.html"><a href="composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="composition.html"><a href="composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus<span></span></a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq<span></span></a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR<span></span></a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM<span></span></a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob<span></span></a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)<span></span></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom-1"><i class="fa fa-check"></i><b>11.2.1</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.2</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.3</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.4</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.2.5" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.5</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot<span></span></a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot<span></span></a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot<span></span></a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density<span></span></a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination<span></span></a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot<span></span></a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot<span></span></a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn<span></span></a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.10</b> plot_stacked_bar_XIVZ<span></span></a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.11</b> plot_StackBarPlot<span></span></a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.12</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>13</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>13.1</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>13.2</b> Amplicon sequencing (16s)<span></span></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>13.2.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="13.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>13.2.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="13.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>13.2.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="13.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>13.2.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="13.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>13.2.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="13.2.6" data-path="examples.html"><a href="examples.html#microbial-composition"><i class="fa fa-check"></i><b>13.2.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="13.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>13.2.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="examples.html"><a href="examples.html#metagenomics"><i class="fa fa-check"></i><b>13.3</b> Metagenomics<span></span></a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>13.3.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="13.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>13.3.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="13.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>13.3.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="13.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>13.3.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="13.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>13.3.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="13.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>13.3.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="13.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>13.3.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>13.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data analysis based on XMAS 2.0 package in microbiota data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-analysis" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Differential analysis<a href="differential-analysis.html#differential-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Identifying the significant taxa between case and control group is necessary. There are too many differential analysis (DA) approaches to choose <span class="citation">(<a href="#ref-nearing2022microbiome" role="doc-biblioref">Nearing et al. 2022</a>)</span>, we provide some of them which focusing on microbial data, including:</p>
<table style="width:100%;">
<colgroup>
<col width="14%" />
<col width="15%" />
<col width="24%" />
<col width="15%" />
<col width="19%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Tool(version)</strong></th>
<th><strong>Input</strong></th>
<th><strong>Normalization</strong></th>
<th><strong>Transformation</strong></th>
<th><strong>Distribution</strong></th>
<th><strong>MicrobialData</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ALDEx2 (1.26.0)</td>
<td>Counts</td>
<td>None</td>
<td>CLR</td>
<td>Dirichlet-multinormial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>limma voom (3.50.1)</td>
<td>Counts/Relative</td>
<td>None/TMM</td>
<td>Log; Precision weighting</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>mbzinb (0.2)</td>
<td>Counts</td>
<td>RLE</td>
<td>None</td>
<td>zero-inflated negative binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>omnibus (0.2)</td>
<td>Counts</td>
<td>GMPR(Geometric Mean of Pairwise Ratios)</td>
<td>None</td>
<td>zero-inflated negative binomial</td>
<td>16s</td>
</tr>
<tr class="odd">
<td>RAIDA (1.0)</td>
<td>Counts</td>
<td>None</td>
<td>Log</td>
<td>modified t-test</td>
<td>16s</td>
</tr>
<tr class="even">
<td>Wilcox(rare/CLR)</td>
<td>Counts/Relative</td>
<td>None</td>
<td>None/CLR</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>LEfSe</td>
<td>Rarefied Counts//Relative</td>
<td>TSS</td>
<td>None</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="even">
<td>t-test (rare)</td>
<td>Counts//Relative</td>
<td>None</td>
<td>None</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>metagenomeSeq (1.36.0)</td>
<td>Counts</td>
<td>CSS</td>
<td>Log</td>
<td>Zero-inflated (log) Normal</td>
<td>16s</td>
</tr>
<tr class="even">
<td>DESeq2 (1.34.0)</td>
<td>Counts</td>
<td>RLE</td>
<td>None</td>
<td>Negative binomial</td>
<td>16s</td>
</tr>
<tr class="odd">
<td>edgeR (3.36.0)</td>
<td>Counts</td>
<td>RLE/TMM</td>
<td>None</td>
<td>Negative binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>ANCOM-II (2.1)</td>
<td>Counts/Relative</td>
<td>None</td>
<td>ALR</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>Corncob (0.2.0)</td>
<td>Counts</td>
<td>None</td>
<td>None</td>
<td>Beta-binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>MaAslin2 (1.8.0)</td>
<td>Counts/Relative abundance</td>
<td>None/TSS</td>
<td>AST</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
</tbody>
</table>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="differential-analysis.html#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS)</span>
<span id="cb187-2"><a href="differential-analysis.html#cb187-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb187-3"><a href="differential-analysis.html#cb187-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb187-4"><a href="differential-analysis.html#cb187-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb187-5"><a href="differential-analysis.html#cb187-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<div id="amplicon-sequencing-microbial-data-16s" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Amplicon sequencing microbial data (16s)<a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same strategy to filter phyloseq object.</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="differential-analysis.html#cb188-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;dada2_ps&quot;</span>)</span>
<span id="cb188-2"><a href="differential-analysis.html#cb188-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-3"><a href="differential-analysis.html#cb188-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb188-4"><a href="differential-analysis.html#cb188-4" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb188-5"><a href="differential-analysis.html#cb188-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb188-6"><a href="differential-analysis.html#cb188-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb188-7"><a href="differential-analysis.html#cb188-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb188-8"><a href="differential-analysis.html#cb188-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-9"><a href="differential-analysis.html#cb188-9" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Rarefying counts in phyloseq-class object</span></span>
<span id="cb188-10"><a href="differential-analysis.html#cb188-10" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(dada2_ps_remove_BRS, <span class="at">size =</span> <span class="dv">51181</span>, <span class="at">rng_seed =</span> <span class="dv">123</span>)</span>
<span id="cb188-11"><a href="differential-analysis.html#cb188-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-12"><a href="differential-analysis.html#cb188-12" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb188-13"><a href="differential-analysis.html#cb188-13" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(dada2_ps_rarefy, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, <span class="at">absolute =</span> <span class="cn">TRUE</span>)</span>
<span id="cb188-14"><a href="differential-analysis.html#cb188-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-15"><a href="differential-analysis.html#cb188-15" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb188-16"><a href="differential-analysis.html#cb188-16" aria-hidden="true" tabindex="-1"></a><span class="co">#dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(dada2_ps_rare_genus, cutoff = 10, unclass = TRUE)</span></span>
<span id="cb188-17"><a href="differential-analysis.html#cb188-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-18"><a href="differential-analysis.html#cb188-18" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb188-19"><a href="differential-analysis.html#cb188-19" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(dada2_ps_rare_genus, <span class="at">cutoff =</span> <span class="dv">10</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb188-20"><a href="differential-analysis.html#cb188-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb188-21"><a href="differential-analysis.html#cb188-21" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb188-22"><a href="differential-analysis.html#cb188-22" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(dada2_ps_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb188-23"><a href="differential-analysis.html#cb188-23" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 99 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 99 taxa by 6 taxonomic ranks ]</code></pre>
<div id="aldex2" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> ALDEx2<a href="differential-analysis.html#aldex2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ALDEx2</strong> package is from <em>Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis</em> <span class="citation">(<a href="#ref-fernandes2014unifying" role="doc-biblioref">Fernandes et al. 2014</a>)</span>, and its principle is using log-ratio transformation and statistical testing to find the significant Taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<p><code>run_aldex</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_aldex)</code>.</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="differential-analysis.html#cb190-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb190-2"><a href="differential-analysis.html#cb190-2" aria-hidden="true" tabindex="-1"></a>                     dada2_ps_genus_filter_trim,</span>
<span id="cb190-3"><a href="differential-analysis.html#cb190-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb190-4"><a href="differential-analysis.html#cb190-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb190-5"><a href="differential-analysis.html#cb190-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span>
<span id="cb190-6"><a href="differential-analysis.html#cb190-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="differential-analysis.html#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment  EffectSize    Pvalue AdjustedPvalue Median CLR \n(All) Median CLR\nAA Median CLR\nBB
## 1   g__Actinomyces 9_AA vs 14_BB  Nonsignif  0.20107672 0.5589168      0.8980570           1.857087      0.5668075      2.4851585
## 2 g__Adlercreutzia 9_AA vs 14_BB  Nonsignif  0.32991526 0.2422022      0.7763186          -1.632999     -2.9753218     -0.9749769
## 3  g__Agathobacter 9_AA vs 14_BB  Nonsignif -0.04590925 0.8337783      0.9604595           4.368404      4.7683067      4.2619500
## 4     g__Alistipes 9_AA vs 14_BB  Nonsignif -0.06269794 0.8102824      0.9564874           2.570901      2.8604190      2.4889260
## 5  g__Anaerostipes 9_AA vs 14_BB  Nonsignif  0.17863931 0.4600141      0.8789664           5.558388      5.0016632      5.8204768
## 6   g__Bacteroides 9_AA vs 14_BB  Nonsignif -0.28723651 0.3695268      0.8619353           8.448104      8.8781307      8.2735587
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB
## 1                        -1.1699250                      30                   22                 49.5                      -1.3859026
## 2                                NA                       0                    0                  9.5                      -2.5541902
## 3                         0.1772545                     311                  333                294.5                       1.7486195
## 4                        -0.4280937                      64                   55                 74.0                       0.2295656
## 5                         1.0204641                     515                  923                455.0                      -0.1010837
## 6                         1.5520348                    3806                 9118               3109.5                       0.9814940
##   Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1              50.65217          25.555556           66.78571                    82.61                 88.89                 78.57
## 2              19.39130           4.888889           28.71429                    43.48                 33.33                 50.00
## 3             997.26087        1742.111111          518.42857                    65.22                 55.56                 71.43
## 4             163.17391         179.222222          152.85714                    73.91                 77.78                 71.43
## 5             821.30435         786.555556          843.64286                    95.65                100.00                 92.86
## 6            5740.60870        8205.777778         4155.85714                   100.00                100.00                100.00
##   Odds Ratio (95% CI)
## 1       4.3 (7.1;1.4)
## 2           13 (18;8)
## 3     0.39 (-1.4;2.2)
## 4     0.87 (0.59;1.1)
## 5      1.1 (1.2;0.93)
## 6       0.45 (-1.1;2)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Median CLR (All)/(group AA)/(group BB)</strong>: Median CLR (normalization by <strong>ALDEx2</strong>) in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>ALDEx2</strong>.</p>
<details>
<summary>
<strong>run_da() pattern</strong>
</summary>
<p>We also provide another function <code>run_da</code> to run ALDEx2 differential analysis.</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="differential-analysis.html#cb194-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb194-2"><a href="differential-analysis.html#cb194-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb194-3"><a href="differential-analysis.html#cb194-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb194-4"><a href="differential-analysis.html#cb194-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb194-5"><a href="differential-analysis.html#cb194-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;aldex&quot;</span>,</span>
<span id="cb194-6"><a href="differential-analysis.html#cb194-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span>
<span id="cb194-7"><a href="differential-analysis.html#cb194-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>otu_table or sample_table as inputdata</strong>
</summary>
<p>We also provide <em>data_otu</em> and <em>data_sam</em> as input data to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="differential-analysis.html#cb196-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb196-2"><a href="differential-analysis.html#cb196-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb196-3"><a href="differential-analysis.html#cb196-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb196-4"><a href="differential-analysis.html#cb196-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb196-5"><a href="differential-analysis.html#cb196-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb196-6"><a href="differential-analysis.html#cb196-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span>
<span id="cb196-7"><a href="differential-analysis.html#cb196-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb196-8"><a href="differential-analysis.html#cb196-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>taxa_level option</strong>
</summary>
<p>We also provide <em>taxa_level</em> for choosing the specific taxonomic level to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="differential-analysis.html#cb198-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb198-2"><a href="differential-analysis.html#cb198-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb198-3"><a href="differential-analysis.html#cb198-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb198-4"><a href="differential-analysis.html#cb198-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb198-5"><a href="differential-analysis.html#cb198-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb198-6"><a href="differential-analysis.html#cb198-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span>
<span id="cb198-7"><a href="differential-analysis.html#cb198-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
</div>
<div id="limma_voom" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> limma_voom<a href="differential-analysis.html#limma_voom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>limma</strong> package is from <em>voom: Precision weights unlock linear model analysis tools for RNA-seq read counts</em> <span class="citation">(<a href="#ref-law2014voom" role="doc-biblioref">Law et al. 2014</a>)</span>. Firstly, transforming count data to log2-counts per million (logCPM), estimate the mean-variance relationship and use this to compute appropriate observation-level weights. Secondly, fitting multiple linear models by weighted or generalized least squares. Finally, performing empirical bayes statistics for differential expression.</p>
<p><code>run_limma_voom</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_limma_voom)</code>.</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="differential-analysis.html#cb200-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb200-2"><a href="differential-analysis.html#cb200-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb200-3"><a href="differential-analysis.html#cb200-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb200-4"><a href="differential-analysis.html#cb200-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb200-5"><a href="differential-analysis.html#cb200-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb200-6"><a href="differential-analysis.html#cb200-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="differential-analysis.html#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##                       TaxaID         Block Enrichment EffectSize     logFC      Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1              g__Romboutsia 9_AA vs 14_BB  Nonsignif   4.695463  4.695463 0.003755076      0.2591003                         -6.499846
## 2 g__Ruminococcaceae_UCG_004 9_AA vs 14_BB  Nonsignif  -3.917539 -3.917539 0.021019060      0.5453230                                NA
## 3           g__Lactobacillus 9_AA vs 14_BB  Nonsignif   3.819030  3.819030 0.025755771      0.5453230                         -2.870821
## 4          g__Parasutterella 9_AA vs 14_BB  Nonsignif  -3.581423 -3.581423 0.031612929      0.5453230                          5.029747
## 5             g__Odoribacter 9_AA vs 14_BB  Nonsignif  -3.085026 -3.085026 0.043219346      0.5964270                                NA
## 6               g__Roseburia 9_AA vs 14_BB  Nonsignif   3.470587  3.470587 0.065834317      0.6632463                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                     105                    3                271.5                       -3.197677             295.47826
## 2                      11                  107                  0.0                        2.211165              66.91304
## 3                      57                   27                197.5                       -5.067893             799.95652
## 4                       7                   49                  1.5                        4.244838             164.56522
## 5                       3                   30                  0.0                        2.050293              26.43478
## 6                     125                    0                252.5                       -3.184643             669.60870
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1           49.44444          453.64286                    78.26                 55.56                 92.86          42 (49;34)
## 2          128.00000           27.64286                    52.17                 77.78                 35.71     0.19 (-3.1;3.5)
## 3           38.44444         1289.50000                    86.96                 77.78                 92.86 4e+08 (4e+08;4e+08)
## 4          388.66667           20.50000                    65.22                 88.89                 50.00         0.01 (-9;9)
## 5           49.11111           11.85714                    52.17                 77.78                 35.71     0.33 (-1.9;2.5)
## 6          113.00000         1027.42857                    65.22                 44.44                 78.57          64 (72;56)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>limma</strong>;</p></li>
<li><p><strong>logFC</strong>: LogFC from groups’ coefficient by <strong>limma</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>limma</strong>;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>limma</strong>.</p>
<details>
<summary>
<strong>other options for limma-voom</strong>
</summary>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="differential-analysis.html#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb204-2"><a href="differential-analysis.html#cb204-2" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb204-3"><a href="differential-analysis.html#cb204-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb204-4"><a href="differential-analysis.html#cb204-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb204-5"><a href="differential-analysis.html#cb204-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb204-6"><a href="differential-analysis.html#cb204-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb204-7"><a href="differential-analysis.html#cb204-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb204-8"><a href="differential-analysis.html#cb204-8" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb204-9"><a href="differential-analysis.html#cb204-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb204-10"><a href="differential-analysis.html#cb204-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb204-11"><a href="differential-analysis.html#cb204-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb204-12"><a href="differential-analysis.html#cb204-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb204-13"><a href="differential-analysis.html#cb204-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb204-14"><a href="differential-analysis.html#cb204-14" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb204-15"><a href="differential-analysis.html#cb204-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb204-16"><a href="differential-analysis.html#cb204-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb204-17"><a href="differential-analysis.html#cb204-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb204-18"><a href="differential-analysis.html#cb204-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb204-19"><a href="differential-analysis.html#cb204-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span>
<span id="cb204-20"><a href="differential-analysis.html#cb204-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="mbzinb" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> mbzinb<a href="differential-analysis.html#mbzinb" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>mbzinb</strong> package is from <em>An omnibus test for differential distribution analysis of microbiome sequencing data</em> <span class="citation">(<a href="#ref-chen2018omnibus" role="doc-biblioref">J. Chen et al. 2018</a>)</span>. It uses zeroinflated negative binomial model to investigate the significant taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="differential-analysis.html#cb205-1" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb205-2"><a href="differential-analysis.html#cb205-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb205-3"><a href="differential-analysis.html#cb205-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb205-4"><a href="differential-analysis.html#cb205-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb205-5"><a href="differential-analysis.html#cb205-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;PValue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;base.mean&quot;                         &quot;mean.LFC&quot;                          &quot;base.abund&quot;                       
## [10] &quot;abund.LFC&quot;                         &quot;base.prev&quot;                         &quot;prev.change&quot;                      
## [13] &quot;base.disp&quot;                         &quot;disp.LFC&quot;                          &quot;statistic&quot;                        
## [16] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [19] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [22] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [25] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="differential-analysis.html#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##                       TaxaID         Block Enrichment      PValue AdjustedPvalue EffectSize  base.mean   mean.LFC base.abund   abund.LFC
## 1           g__Lactobacillus 9_AA vs 14_BB  Nonsignif 0.002086875      0.1439944  2.2952990  34.129282  4.4898620   43.61763  4.19415460
## 2     g__Ruminiclostridium_5 9_AA vs 14_BB  Nonsignif 0.008876265      0.3003035  0.3212103  30.315071  0.7865167   30.31507  1.00848113
## 3              g__Romboutsia 9_AA vs 14_BB  Nonsignif 0.016443976      0.3003035  2.2507136  36.557345  3.1571628   63.05112  2.46745862
## 4           g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.018666615      0.3003035  0.5762774   3.911818  2.6554273   11.73606  2.06902032
## 5 g__Lachnospiraceae_UCG_010 9_AA vs 14_BB  Nonsignif 0.022486902      0.3003035  0.4854298  25.770711 -2.0949222   46.38413 -1.24068800
## 6          g__Butyricicoccus 9_AA vs 14_BB  Nonsignif 0.029426570      0.3003035  0.5544952 231.635435 -0.1958844  297.86046  0.07898054
##   base.prev prev.change  base.disp  disp.LFC statistic Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA
## 1 0.7824654   0.1780005 0.70022332  1.652437 14.705041                        -2.8708213                      57                   27
## 2 1.0000000  -0.1426028 0.11584182  1.935466 11.602584                        -0.3504972                      46                   40
## 3 0.5798048   0.3553929 1.40372047 -0.312006 10.264842                        -6.4998459                     105                    3
## 4 0.3333161   0.1671589 0.02507192  4.101251  9.988211                                NA                       0                    0
## 5 0.5555933  -0.2482619 0.08910223  3.810059  9.580823                                NA                       0                   34
## 6 0.7776642  -0.1349032 0.04031021  3.417434  8.989818                         2.0596429                     292                  321
##   Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1                197.5                      -5.0678928             799.95652          38.444444         1289.50000
## 2                 51.0                      -0.7826083              51.78261          36.000000           61.92857
## 3                271.5                      -3.1976766             295.47826          49.444444          453.64286
## 4                  9.5                      -2.5541902              19.39130           4.888889           28.71429
## 5                  0.0                       2.2719126              18.73913          36.222222            7.50000
## 6                 77.0                       0.3575932             241.04348         278.222222          217.14286
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                    86.96                 77.78                 92.86 4e+08 (4e+08;4e+08)
## 2                    91.30                100.00                 85.71        2.3 (4;0.67)
## 3                    78.26                 55.56                 92.86          42 (49;34)
## 4                    43.48                 33.33                 50.00           13 (18;8)
## 5                    39.13                 55.56                 28.57     0.33 (-1.8;2.5)
## 6                    69.57                 77.78                 64.29     0.76 (0.21;1.3)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>mbzinb</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>mbzinb</strong>;</p></li>
<li><p><strong>base.mean</strong>: fitted mean abundance parameter times the fitted prevalence in baseline group;</p></li>
<li><p><strong>mean.LFC</strong>: log2-fold change in fitted mean between other group and baseline;</p></li>
<li><p><strong>base.abund</strong>: fitted mean abundance parameter in baseline group;</p></li>
<li><p><strong>abund.LFC</strong>: log2-fold change in fitted mean abundance parameter between other group and baseline;</p></li>
<li><p><strong>base.prev</strong>: fitted prevalence in baseline group;</p></li>
<li><p><strong>prev.change</strong>: (linear) difference in prevalence between baseline group and other group (other-baseline);</p></li>
<li><p><strong>base.disp</strong>: fitted dispersion parameter in baseline group;</p></li>
<li><p><strong>disp.LFC</strong>: log2-fold change in fitted dispersion parameter between other group and baseline;</p></li>
<li><p><strong>statistic</strong>: value of likelihood ratio test statistic;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>mbzinb</strong>.</p>
<details>
<summary>
<strong>other options for mbzinb</strong>
</summary>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="differential-analysis.html#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb209-2"><a href="differential-analysis.html#cb209-2" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb209-3"><a href="differential-analysis.html#cb209-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb209-4"><a href="differential-analysis.html#cb209-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb209-5"><a href="differential-analysis.html#cb209-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb209-6"><a href="differential-analysis.html#cb209-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb209-7"><a href="differential-analysis.html#cb209-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-8"><a href="differential-analysis.html#cb209-8" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb209-9"><a href="differential-analysis.html#cb209-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb209-10"><a href="differential-analysis.html#cb209-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb209-11"><a href="differential-analysis.html#cb209-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb209-12"><a href="differential-analysis.html#cb209-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb209-13"><a href="differential-analysis.html#cb209-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-14"><a href="differential-analysis.html#cb209-14" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb209-15"><a href="differential-analysis.html#cb209-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb209-16"><a href="differential-analysis.html#cb209-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb209-17"><a href="differential-analysis.html#cb209-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb209-18"><a href="differential-analysis.html#cb209-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-19"><a href="differential-analysis.html#cb209-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span>
<span id="cb209-20"><a href="differential-analysis.html#cb209-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="omnibus" class="section level3 hasAnchor" number="11.1.4">
<h3><span class="header-section-number">11.1.4</span> omnibus<a href="differential-analysis.html#omnibus" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This approach is also from <strong>mbzinb</strong> <span class="citation">(<a href="#ref-chen2018omnibus" role="doc-biblioref">J. Chen et al. 2018</a>)</span> package. it uses GMPR (Geometric Mean of Pairwise Ratios) <span class="citation">(<a href="#ref-chen2018gmpr" role="doc-biblioref">L. Chen et al. 2018</a>)</span> to get the size factors.</p>
<blockquote>
<p>where we specify models for count (abundance), zero (prevalence) and dispersion part. We also provide likelihood ratio test (zinb.lrt) for different models.</p>
</blockquote>
<p>(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="differential-analysis.html#cb210-1" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb210-2"><a href="differential-analysis.html#cb210-2" aria-hidden="true" tabindex="-1"></a>                     dada2_ps_genus_filter_trim,</span>
<span id="cb210-3"><a href="differential-analysis.html#cb210-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb210-4"><a href="differential-analysis.html#cb210-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb210-5"><a href="differential-analysis.html#cb210-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span></code></pre></div>
<pre><code>## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  69  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="differential-analysis.html#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;chi.stat&quot;                          &quot;df&quot;                                &quot;abund.baseline&quot;                   
## [10] &quot;prev.baseline&quot;                     &quot;dispersion.baseline&quot;               &quot;abund.LFC.CompvarBB.est&quot;          
## [13] &quot;abund.LFC.CompvarBB.se&quot;            &quot;prev.LOD.CompvarBB.est&quot;            &quot;prev.LOD.CompvarBB.se&quot;            
## [16] &quot;dispersion.LFC.CompvarBB.est&quot;      &quot;dispersion.LFC.CompvarBB.se&quot;       &quot;method&quot;                           
## [19] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [22] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [25] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [28] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="differential-analysis.html#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment     Pvalue AdjustedPvalue EffectSize  chi.stat df abund.baseline prev.baseline
## 1   g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.17169244      0.5280000 0.87195760 5.0014303  3   0.0007418605     0.8901709
## 2 g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.03616107      0.3388752 0.57627740 8.5347448  3   0.0003617986     0.3333489
## 3  g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.10406623      0.4876000 0.10803167 6.1603052  3   0.0581883369     0.5555657
## 4     g__Alistipes 9_AA vs 14_BB  Nonsignif 0.96368031      0.9778521 0.07097212 0.2803523  3   0.0051269736     0.7813914
## 5  g__Anaerostipes 9_AA vs 14_BB  Nonsignif 0.63526925      0.8276736 0.40991461 1.7074928  3   0.0191836324     0.9999998
## 6   g__Bacteroides 9_AA vs 14_BB  Nonsignif 0.08900000      0.4723846 0.93028412        NA NA             NA            NA
##   dispersion.baseline abund.LFC.CompvarBB.est abund.LFC.CompvarBB.se prev.LOD.CompvarBB.est prev.LOD.CompvarBB.se
## 1           2.5621526              0.76587774              0.3430326             -0.7589141             1.2527366
## 2          22.4213282              1.30993086              0.3239451              0.6948755             0.8863894
## 3           1.6994493             -1.17340512              0.4639835              0.6956651             0.8946536
## 4           1.0309783             -0.03887534              0.4626257             -0.3549679             1.0004486
## 5           0.6029724              0.05053905              0.5148806            -13.1146633                    NA
## 6                  NA                      NA                     NA                     NA                    NA
##   dispersion.LFC.CompvarBB.est dispersion.LFC.CompvarBB.se              method Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1                   -0.5841087                   0.6581748             omnibus                        -1.1699250                      30
## 2                   -2.3168718                   2.0799668             omnibus                                NA                       0
## 3                   -0.5041838                   0.7037559             omnibus                         0.1772545                     311
## 4                    0.2574561                   0.6308201             omnibus                        -0.4280937                      64
## 5                    0.4588537                   0.5267880             omnibus                         1.0204641                     515
## 6                           NA                          NA permutation(n=1000)                         1.5520348                    3806
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1                   22                 49.5                      -1.3859026              50.65217          25.555556           66.78571
## 2                    0                  9.5                      -2.5541902              19.39130           4.888889           28.71429
## 3                  333                294.5                       1.7486195             997.26087        1742.111111          518.42857
## 4                   55                 74.0                       0.2295656             163.17391         179.222222          152.85714
## 5                  923                455.0                      -0.1010837             821.30435         786.555556          843.64286
## 6                 9118               3109.5                       0.9814940            5740.60870        8205.777778         4155.85714
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 2                    43.48                 33.33                 50.00           13 (18;8)
## 3                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 4                    73.91                 77.78                 71.43     0.87 (0.59;1.1)
## 5                    95.65                100.00                 92.86      1.1 (1.2;0.93)
## 6                   100.00                100.00                100.00       0.45 (-1.1;2)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>omnibus</strong>;</p></li>
<li><p><strong>chi.stat</strong>: chisquare statistics;</p></li>
<li><p><strong>df</strong>: degree freedom;</p></li>
<li><p><strong>abund.baseline</strong>: mean abundance in baseline group;</p></li>
<li><p><strong>prev.baseline</strong>: prevalence in baseline group;</p></li>
<li><p><strong>dispersion.baseline</strong>: dispersion in baseline group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.est</strong>: log2-fold change in abundance between group BB and other group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.se</strong>: log2-fold change of standard errors in abundance between group BB and other group;</p></li>
<li><p><strong>prev.LOD.CompvarBB.est</strong>: prevalence of low of detect value in group BB;</p></li>
<li><p><strong>prev.LOD.CompvarBB.se</strong>: prevalence’s standard errors of low of detect value in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.est</strong>: log2-fold change in dispersion in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.se</strong>: log2-fold change in dispersion’s standard errors in group BB;</p></li>
<li><p><strong>method</strong>: methods of test;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>omnibus</strong>.</p>
<details>
<summary>
<strong>other options for omnibus</strong>
</summary>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="differential-analysis.html#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb216-2"><a href="differential-analysis.html#cb216-2" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb216-3"><a href="differential-analysis.html#cb216-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb216-4"><a href="differential-analysis.html#cb216-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb216-5"><a href="differential-analysis.html#cb216-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb216-6"><a href="differential-analysis.html#cb216-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb216-7"><a href="differential-analysis.html#cb216-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb216-8"><a href="differential-analysis.html#cb216-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-9"><a href="differential-analysis.html#cb216-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-10"><a href="differential-analysis.html#cb216-10" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb216-11"><a href="differential-analysis.html#cb216-11" aria-hidden="true" tabindex="-1"></a>                     dada2_ps,</span>
<span id="cb216-12"><a href="differential-analysis.html#cb216-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb216-13"><a href="differential-analysis.html#cb216-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb216-14"><a href="differential-analysis.html#cb216-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb216-15"><a href="differential-analysis.html#cb216-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb216-16"><a href="differential-analysis.html#cb216-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-17"><a href="differential-analysis.html#cb216-17" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb216-18"><a href="differential-analysis.html#cb216-18" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb216-19"><a href="differential-analysis.html#cb216-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb216-20"><a href="differential-analysis.html#cb216-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb216-21"><a href="differential-analysis.html#cb216-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;omnibus&quot;</span>,</span>
<span id="cb216-22"><a href="differential-analysis.html#cb216-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb216-23"><a href="differential-analysis.html#cb216-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-24"><a href="differential-analysis.html#cb216-24" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span>
<span id="cb216-25"><a href="differential-analysis.html#cb216-25" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="raida" class="section level3 hasAnchor" number="11.1.5">
<h3><span class="header-section-number">11.1.5</span> RAIDA<a href="differential-analysis.html#raida" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>RAIDA</strong> package is from <em>A robust approach for identifying differentially abundant features in metagenomic samples</em> <span class="citation">(<a href="#ref-sohn2015robust" role="doc-biblioref">Sohn, Du, and An 2015</a>)</span>. It uses Ratio Approach for Identifying Differential Abundance (RAIDA). (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="differential-analysis.html#cb217-1" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb217-2"><a href="differential-analysis.html#cb217-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb217-3"><a href="differential-analysis.html#cb217-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb217-4"><a href="differential-analysis.html#cb217-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb217-5"><a href="differential-analysis.html#cb217-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-6"><a href="differential-analysis.html#cb217-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;eta_AA&quot;                            &quot;mean_AA&quot;                           &quot;sd_AA&quot;                            
## [10] &quot;eta_BB&quot;                            &quot;mean_BB&quot;                           &quot;sd_BB&quot;                            
## [13] &quot;mod.pool.var&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [16] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [19] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [22] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [25] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="differential-analysis.html#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##                    TaxaID         Block Enrichment EffectSize      Pvalue AdjustedPvalue    eta_AA   mean_AA     sd_AA     eta_BB
## 1          g__Megasphaera 9_AA vs 14_BB  Nonsignif  0.8198271 0.005697322      0.3133527 0.6666667  1.052529 0.6673820 0.64285714
## 2     g__Faecalibacterium 9_AA vs 14_BB  Nonsignif  0.4701759 0.035043427      0.4002378 0.3333333  1.197269 0.4839668 0.28571429
## 3        g__Lactobacillus 9_AA vs 14_BB  Nonsignif  2.2952990 0.043419259      0.4002378 0.2222222 -3.497406 1.4602720 0.07142857
## 4          g__Bacteroides 9_AA vs 14_BB  Nonsignif  0.9302841 0.043697499      0.4002378 0.0000000  1.365432 1.1945235 0.00000000
## 5 g__Escherichia.Shigella 9_AA vs 14_BB  Nonsignif  0.9681020 0.045317520      0.4002378 0.1111111 -1.182989 1.8000680 0.14285714
## 6           g__Romboutsia 9_AA vs 14_BB  Nonsignif  2.2507136 0.055806331      0.4002378 0.4444444 -3.677903 1.4067837 0.07142857
##       mean_BB    sd_BB mod.pool.var Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1 -1.81131976 1.343843     1.389145                                NA                       0                    0                  0.0
## 2 -0.04932509 1.202811     1.144126                          2.373851                    1377                 4512                870.5
## 3 -1.62700629 2.310432     3.509818                         -2.870821                      57                   27                197.5
## 4 -0.01095745 1.823840     2.322156                          1.552035                    3806                 9118               3109.5
## 5 -2.81738839 1.877394     2.882621                          2.261867                      70                  247                 51.5
## 6 -1.88434753 1.982971     2.856684                         -6.499846                     105                    3                271.5
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       1.7193715              333.9130          579.55556           176.0000                    34.78
## 2                       1.4344092             2678.7826         4345.00000          1607.6429                    69.57
## 3                      -5.0678928              799.9565           38.44444          1289.5000                    86.96
## 4                       0.9814940             5740.6087         8205.77778          4155.8571                   100.00
## 5                       0.4374399              574.2609          683.00000           504.3571                    86.96
## 6                      -3.1976766              295.4783           49.44444           453.6429                    78.26
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 33.33                 35.71    0.55 (-0.64;1.7)
## 2                 66.67                 71.43     0.37 (-1.6;2.3)
## 3                 77.78                 92.86 4e+08 (4e+08;4e+08)
## 4                100.00                100.00       0.45 (-1.1;2)
## 5                 88.89                 85.71     0.85 (0.53;1.2)
## 6                 55.56                 92.86          42 (49;34)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>RAIDA</strong>;</p></li>
<li><p><strong>eta_AA</strong>: vector containing estimated probabilities of the false zero state for group AA;</p></li>
<li><p><strong>mean_AA</strong>: vector containing estimated means of log ratios for group AA;</p></li>
<li><p><strong>sd_AA</strong>: vector containing estimated standard deviations of log ratios for group AA;</p></li>
<li><p><strong>eta_BB</strong>: vector containing estimated probabilities of the false zero state for group BB;</p></li>
<li><p><strong>mean_BB</strong>: vector containing estimated means of log ratios for group BB;</p></li>
<li><p><strong>sd_BB</strong>: vector containing estimated standard deviations of log ratios for group BB;</p></li>
<li><p><strong>mod.pool.var</strong>: vector containing estimated posterior variances of log ratios;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>RAIDA</strong>.</p>
<details>
<summary>
<strong>other options for RAIDA</strong>
</summary>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="differential-analysis.html#cb221-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb221-2"><a href="differential-analysis.html#cb221-2" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb221-3"><a href="differential-analysis.html#cb221-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb221-4"><a href="differential-analysis.html#cb221-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb221-5"><a href="differential-analysis.html#cb221-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb221-6"><a href="differential-analysis.html#cb221-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb221-7"><a href="differential-analysis.html#cb221-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb221-8"><a href="differential-analysis.html#cb221-8" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb221-9"><a href="differential-analysis.html#cb221-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb221-10"><a href="differential-analysis.html#cb221-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb221-11"><a href="differential-analysis.html#cb221-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb221-12"><a href="differential-analysis.html#cb221-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb221-13"><a href="differential-analysis.html#cb221-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb221-14"><a href="differential-analysis.html#cb221-14" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb221-15"><a href="differential-analysis.html#cb221-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb221-16"><a href="differential-analysis.html#cb221-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb221-17"><a href="differential-analysis.html#cb221-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb221-18"><a href="differential-analysis.html#cb221-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;raida&quot;</span>)</span>
<span id="cb221-19"><a href="differential-analysis.html#cb221-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb221-20"><a href="differential-analysis.html#cb221-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span>
<span id="cb221-21"><a href="differential-analysis.html#cb221-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="wilcoxon-rank-sum-and-signed-rank-tests" class="section level3 hasAnchor" number="11.1.6">
<h3><span class="header-section-number">11.1.6</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparametric test methods, use the rank of taxa abundance to find the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="differential-analysis.html#cb222-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb222-2"><a href="differential-analysis.html#cb222-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb222-3"><a href="differential-analysis.html#cb222-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb222-4"><a href="differential-analysis.html#cb222-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb222-5"><a href="differential-analysis.html#cb222-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb222-6"><a href="differential-analysis.html#cb222-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="differential-analysis.html#cb224-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1   g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.87195760        45 0.2689844      0.6737725                        -1.1699250
## 2 g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.57627740        44 0.1981520      0.6737725                                NA
## 3  g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.10803167        71 0.6293971      0.9129068                         0.1772545
## 4     g__Alistipes 9_AA vs 14_BB  Nonsignif 0.07097212        67 0.8239507      0.9129068                        -0.4280937
## 5  g__Anaerostipes 9_AA vs 14_BB  Nonsignif 0.40991461        58 0.7767603      0.9129068                         1.0204641
## 6   g__Bacteroides 9_AA vs 14_BB  Nonsignif 0.93028412        89 0.1093014      0.6737725                         1.5520348
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1                      30                   22                 49.5                     -0.41034110                   10.00
## 2                       0                    0                  9.5                     -0.43387758                    9.89
## 3                     311                  333                294.5                      0.17342686                   12.89
## 4                      64                   55                 74.0                      0.08724541                   12.44
## 5                     515                  923                455.0                     -0.11159169                   11.44
## 6                    3806                 9118               3109.5                      0.55428610                   14.89
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   13.29                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 2                   13.36                    43.48                 33.33                 50.00           13 (18;8)
## 3                   11.43                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 4                   11.71                    73.91                 77.78                 71.43     0.87 (0.59;1.1)
## 5                   12.36                    95.65                100.00                 92.86      1.1 (1.2;0.93)
## 6                   10.14                   100.00                100.00                100.00       0.45 (-1.1;2)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean Rank)_vs_BB**: Log2FoldChange (Mean Rank Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Rank Abundance (All)/(group AA)/(group BB)</strong>: Mean Rank Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for wilcox</strong>
</summary>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="differential-analysis.html#cb226-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb226-2"><a href="differential-analysis.html#cb226-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb226-3"><a href="differential-analysis.html#cb226-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb226-4"><a href="differential-analysis.html#cb226-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb226-5"><a href="differential-analysis.html#cb226-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb226-6"><a href="differential-analysis.html#cb226-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb226-7"><a href="differential-analysis.html#cb226-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb226-8"><a href="differential-analysis.html#cb226-8" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb226-9"><a href="differential-analysis.html#cb226-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb226-10"><a href="differential-analysis.html#cb226-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb226-11"><a href="differential-analysis.html#cb226-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb226-12"><a href="differential-analysis.html#cb226-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb226-13"><a href="differential-analysis.html#cb226-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb226-14"><a href="differential-analysis.html#cb226-14" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb226-15"><a href="differential-analysis.html#cb226-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb226-16"><a href="differential-analysis.html#cb226-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb226-17"><a href="differential-analysis.html#cb226-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb226-18"><a href="differential-analysis.html#cb226-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb226-19"><a href="differential-analysis.html#cb226-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb226-20"><a href="differential-analysis.html#cb226-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span>
<span id="cb226-21"><a href="differential-analysis.html#cb226-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="differential-analysis.html#cb227-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summary</span></span>
<span id="cb227-2"><a href="differential-analysis.html#cb227-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(dada2_ps_genus_filter_trim)</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 32373&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 49344&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 1007994&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 43825.8260869565&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 45587&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.49802371541502&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 0&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 1&quot;
## 
## [[11]]
## [1] &quot;Group&quot;</code></pre>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="differential-analysis.html#cb229-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run</span></span>
<span id="cb229-2"><a href="differential-analysis.html#cb229-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb229-3"><a href="differential-analysis.html#cb229-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb229-4"><a href="differential-analysis.html#cb229-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb229-5"><a href="differential-analysis.html#cb229-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb229-6"><a href="differential-analysis.html#cb229-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb229-7"><a href="differential-analysis.html#cb229-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_rarefy</strong>
</summary>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="differential-analysis.html#cb231-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb231-2"><a href="differential-analysis.html#cb231-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb231-3"><a href="differential-analysis.html#cb231-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb231-4"><a href="differential-analysis.html#cb231-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb231-5"><a href="differential-analysis.html#cb231-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb231-6"><a href="differential-analysis.html#cb231-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb231-7"><a href="differential-analysis.html#cb231-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb231-8"><a href="differential-analysis.html#cb231-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb231-9"><a href="differential-analysis.html#cb231-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_rarefy)</span>
<span id="cb231-10"><a href="differential-analysis.html#cb231-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_CLR</strong>: centered log-ratio normalization.</li>
</ul>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="differential-analysis.html#cb232-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb232-2"><a href="differential-analysis.html#cb232-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb232-3"><a href="differential-analysis.html#cb232-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb232-4"><a href="differential-analysis.html#cb232-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb232-5"><a href="differential-analysis.html#cb232-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb232-6"><a href="differential-analysis.html#cb232-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb232-7"><a href="differential-analysis.html#cb232-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_CLR)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_CLR</strong>
</summary>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="differential-analysis.html#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb234-2"><a href="differential-analysis.html#cb234-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb234-3"><a href="differential-analysis.html#cb234-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb234-4"><a href="differential-analysis.html#cb234-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb234-5"><a href="differential-analysis.html#cb234-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb234-6"><a href="differential-analysis.html#cb234-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb234-7"><a href="differential-analysis.html#cb234-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb234-8"><a href="differential-analysis.html#cb234-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb234-9"><a href="differential-analysis.html#cb234-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_CLR)</span>
<span id="cb234-10"><a href="differential-analysis.html#cb234-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse" class="section level3 hasAnchor" number="11.1.7">
<h3><span class="header-section-number">11.1.7</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="differential-analysis.html#cb235-1" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb235-2"><a href="differential-analysis.html#cb235-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb235-3"><a href="differential-analysis.html#cb235-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb235-4"><a href="differential-analysis.html#cb235-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb235-5"><a href="differential-analysis.html#cb235-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb235-6"><a href="differential-analysis.html#cb235-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;LDA_Score&quot;                         &quot;EffectSize&quot;                        &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
##  [7] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [10] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [13] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [16] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="differential-analysis.html#cb237-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span></code></pre></div>
<pre><code>##                       TaxaID         Block Enrichment  LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1          g__Parasutterella 9_AA vs 14_BB         AA -2.3392917  1.1118869                          5.029747                       7
## 2 g__Ruminococcaceae_UCG_004 9_AA vs 14_BB         AA -0.7725001  1.0652610                                NA                      11
## 3             g__Odoribacter 9_AA vs 14_BB         BB  1.3955403  0.6595894                                NA                       3
## 4              g__Romboutsia 9_AA vs 14_BB         BB  2.2623313  2.2507136                         -6.499846                     105
## 5           g__Lactobacillus 9_AA vs 14_BB         BB  2.7737753  2.2952990                         -2.870821                      57
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1                   49                  1.5                        4.244838             164.56522          388.66667           20.50000
## 2                  107                  0.0                        2.211165              66.91304          128.00000           27.64286
## 3                   30                  0.0                        2.050293              26.43478           49.11111           11.85714
## 4                    3                271.5                       -3.197677             295.47826           49.44444          453.64286
## 5                   27                197.5                       -5.067893             799.95652           38.44444         1289.50000
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                    65.22                 88.89                 50.00         0.01 (-9;9)
## 2                    52.17                 77.78                 35.71     0.19 (-3.1;3.5)
## 3                    52.17                 77.78                 35.71     0.33 (-1.9;2.5)
## 4                    78.26                 55.56                 92.86          42 (49;34)
## 5                    86.96                 77.78                 92.86 4e+08 (4e+08;4e+08)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>lefse</em>;</p></li>
<li><p><strong>LDA_Score</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for lefse</strong>
</summary>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="differential-analysis.html#cb239-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb239-2"><a href="differential-analysis.html#cb239-2" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb239-3"><a href="differential-analysis.html#cb239-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb239-4"><a href="differential-analysis.html#cb239-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb239-5"><a href="differential-analysis.html#cb239-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb239-6"><a href="differential-analysis.html#cb239-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb239-7"><a href="differential-analysis.html#cb239-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb239-8"><a href="differential-analysis.html#cb239-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-9"><a href="differential-analysis.html#cb239-9" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb239-10"><a href="differential-analysis.html#cb239-10" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb239-11"><a href="differential-analysis.html#cb239-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb239-12"><a href="differential-analysis.html#cb239-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb239-13"><a href="differential-analysis.html#cb239-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb239-14"><a href="differential-analysis.html#cb239-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb239-15"><a href="differential-analysis.html#cb239-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-16"><a href="differential-analysis.html#cb239-16" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb239-17"><a href="differential-analysis.html#cb239-17" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb239-18"><a href="differential-analysis.html#cb239-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb239-19"><a href="differential-analysis.html#cb239-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb239-20"><a href="differential-analysis.html#cb239-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb239-21"><a href="differential-analysis.html#cb239-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb239-22"><a href="differential-analysis.html#cb239-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-23"><a href="differential-analysis.html#cb239-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span>
<span id="cb239-24"><a href="differential-analysis.html#cb239-24" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test" class="section level3 hasAnchor" number="11.1.8">
<h3><span class="header-section-number">11.1.8</span> t-test<a href="differential-analysis.html#t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="differential-analysis.html#cb240-1" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb240-2"><a href="differential-analysis.html#cb240-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb240-3"><a href="differential-analysis.html#cb240-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb240-4"><a href="differential-analysis.html#cb240-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb240-5"><a href="differential-analysis.html#cb240-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb240-6"><a href="differential-analysis.html#cb240-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                    &quot;Enrichment&quot;                              
##  [4] &quot;EffectSize&quot;                               &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;        &quot;Median Abundance\n(All)&quot;                 
## [10] &quot;Median Abundance\nAA&quot;                     &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;              &quot;GeometricMean Abundance\nBB&quot;             
## [16] &quot;Occurrence (100%)\n(All)&quot;                 &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="differential-analysis.html#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1   g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.87195760 -1.9509301 0.06966436      0.5805112                        -1.1699250
## 2 g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.57627740 -2.2520991 0.04004033      0.5805112                                NA
## 3  g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.10803167  1.4025255 0.19444323      0.5805112                         0.1772545
## 4     g__Alistipes 9_AA vs 14_BB  Nonsignif 0.07097212  0.3094918 0.76107100      0.9054120                        -0.4280937
## 5  g__Anaerostipes 9_AA vs 14_BB  Nonsignif 0.40991461 -0.1605561 0.87404155      0.9283954                         1.0204641
## 6   g__Bacteroides 9_AA vs 14_BB  Nonsignif 0.93028412  1.5658052 0.14459422      0.5805112                         1.5520348
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (geometricmean)\nAA_vs_BB
## 1                      30                   22                 49.5                               -5.2268338
## 2                       0                    0                  9.5                               -3.1311631
## 3                     311                  333                294.5                                1.4704215
## 4                      64                   55                 74.0                               -0.6655791
## 5                     515                  923                455.0                               -0.8553998
## 6                    3806                 9118               3109.5                                1.4548821
##   GeometricMean Abundance\n(All) GeometricMean Abundance\nAA GeometricMean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA
## 1                      0.5117863                  0.02149483                   0.8049481                    82.61                 88.89
## 2                      0.5218980                  0.08122039                   0.7116052                    43.48                 33.33
## 3                      0.7835288                  1.10053019                   0.3971559                    65.22                 55.56
## 4                      0.7002247                  0.54190943                   0.8595794                    73.91                 77.78
## 5                      0.6131952                  0.45587764                   0.8248012                    95.65                100.00
## 6                      0.6828714                  1.13062995                   0.4124367                   100.00                100.00
##   Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 78.57       4.3 (7.1;1.4)
## 2                 50.00           13 (18;8)
## 3                 71.43     0.39 (-1.4;2.2)
## 4                 71.43     0.87 (0.59;1.1)
## 5                 92.86      1.1 (1.2;0.93)
## 6                100.00       0.45 (-1.1;2)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (GeometricMean)_vs_BB**: Log2FoldChange (GeometricMean Abundance) between group AA and group BB;</p></li>
<li><p><strong>GeometricMean Abundance (All)/(group AA)/(group BB)</strong>: GeometricMean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for t-test</strong>
</summary>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="differential-analysis.html#cb244-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb244-2"><a href="differential-analysis.html#cb244-2" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb244-3"><a href="differential-analysis.html#cb244-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb244-4"><a href="differential-analysis.html#cb244-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb244-5"><a href="differential-analysis.html#cb244-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb244-6"><a href="differential-analysis.html#cb244-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb244-7"><a href="differential-analysis.html#cb244-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb244-8"><a href="differential-analysis.html#cb244-8" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb244-9"><a href="differential-analysis.html#cb244-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb244-10"><a href="differential-analysis.html#cb244-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb244-11"><a href="differential-analysis.html#cb244-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb244-12"><a href="differential-analysis.html#cb244-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb244-13"><a href="differential-analysis.html#cb244-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb244-14"><a href="differential-analysis.html#cb244-14" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb244-15"><a href="differential-analysis.html#cb244-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb244-16"><a href="differential-analysis.html#cb244-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb244-17"><a href="differential-analysis.html#cb244-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb244-18"><a href="differential-analysis.html#cb244-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb244-19"><a href="differential-analysis.html#cb244-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb244-20"><a href="differential-analysis.html#cb244-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span>
<span id="cb244-21"><a href="differential-analysis.html#cb244-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>t-test_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="differential-analysis.html#cb245-1" aria-hidden="true" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb245-2"><a href="differential-analysis.html#cb245-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb245-3"><a href="differential-analysis.html#cb245-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb245-4"><a href="differential-analysis.html#cb245-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb245-5"><a href="differential-analysis.html#cb245-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb245-6"><a href="differential-analysis.html#cb245-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                    &quot;Enrichment&quot;                              
##  [4] &quot;EffectSize&quot;                               &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;        &quot;Median Abundance\n(All)&quot;                 
## [10] &quot;Median Abundance\nAA&quot;                     &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;              &quot;GeometricMean Abundance\nBB&quot;             
## [16] &quot;Occurrence (100%)\n(All)&quot;                 &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for ttest_rarefy</strong>
</summary>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="differential-analysis.html#cb247-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb247-2"><a href="differential-analysis.html#cb247-2" aria-hidden="true" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb247-3"><a href="differential-analysis.html#cb247-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb247-4"><a href="differential-analysis.html#cb247-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb247-5"><a href="differential-analysis.html#cb247-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb247-6"><a href="differential-analysis.html#cb247-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>,</span>
<span id="cb247-7"><a href="differential-analysis.html#cb247-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb247-8"><a href="differential-analysis.html#cb247-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb247-9"><a href="differential-analysis.html#cb247-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_rarefy)</span>
<span id="cb247-10"><a href="differential-analysis.html#cb247-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="metagenomeseq" class="section level3 hasAnchor" number="11.1.9">
<h3><span class="header-section-number">11.1.9</span> MetagenomeSeq<a href="differential-analysis.html#metagenomeseq" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>MetagenomeSeq</strong> package is from <em>Differential abundance analysis for microbial marker-gene surveys</em> <span class="citation">(<a href="#ref-paulson2013differential" role="doc-biblioref">Paulson et al. 2013</a>)</span>. It uses zero-inflated Log-Normal mixture model or Zero-inflated Gaussian mixture model to identify the significant taxa between groups. (Caution: the <strong>otu_table</strong> should be integers).</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="differential-analysis.html#cb248-1" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb248-2"><a href="differential-analysis.html#cb248-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb248-3"><a href="differential-analysis.html#cb248-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb248-4"><a href="differential-analysis.html#cb248-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb248-5"><a href="differential-analysis.html#cb248-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb248-6"><a href="differential-analysis.html#cb248-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb248-7"><a href="differential-analysis.html#cb248-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-8"><a href="differential-analysis.html#cb248-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;logFC&quot;                             &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;se&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="differential-analysis.html#cb250-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##                       TaxaID         Block Enrichment      logFC      Pvalue AdjustedPvalue        se Log2FoldChange (Median)\nAA_vs_BB
## 1              g__Romboutsia 9_AA vs 14_BB  Nonsignif  1.5820439 0.001857198      0.1300039 0.5083417                         -5.213950
## 2             g__Megasphaera 9_AA vs 14_BB  Nonsignif -1.7778094 0.004391697      0.1537094 0.6241071                                NA
## 3 g__Ruminococcaceae_UCG_013 9_AA vs 14_BB  Nonsignif  0.7685692 0.057122207      0.8174723 0.4040052                         -2.792157
## 4 g__Lachnospiraceae_UCG_010 9_AA vs 14_BB  Nonsignif -1.1072541 0.083441340      0.8174723 0.6396399                                NA
## 5               g__Roseburia 9_AA vs 14_BB  Nonsignif  0.9583484 0.088348576      0.8174723 0.5623536                                NA
## 6              g__Tyzzerella 9_AA vs 14_BB  Nonsignif -1.6944399 0.088913558      0.8174723 0.9960516                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                96.15385             7.672634            284.77369                       -4.157644             423.30698
## 2                 0.00000             0.000000              0.00000                        3.234514             882.94339
## 3                45.19774            10.662605             73.85609                       -1.702861             121.28674
## 4                 0.00000            25.894897              0.00000                        2.002867              16.61215
## 5                94.30707             0.000000            237.61595                       -4.059516             868.99235
## 6                 0.00000             9.345794              0.00000                        2.773081              51.49608
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1           37.61073         671.254571                    78.26                 55.56                 92.86    5300 (5300;5300)
## 2         1936.38092         205.733552                    34.78                 33.33                 35.71     0.29 (-2.1;2.7)
## 3           51.11355         166.398077                    60.87                 55.56                 64.29         3.4 (5.9;1)
## 4           30.58335           7.630657                    39.13                 55.56                 28.57     0.38 (-1.5;2.3)
## 5           82.44231        1374.631667                    65.22                 44.44                 78.57    6500 (6600;6500)
## 6          107.20501          15.683192                    34.78                 55.56                 21.43     0.35 (-1.7;2.4)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>se</strong>: standard error;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for metagenomeseq</strong>
</summary>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="differential-analysis.html#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb252-2"><a href="differential-analysis.html#cb252-2" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb252-3"><a href="differential-analysis.html#cb252-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb252-4"><a href="differential-analysis.html#cb252-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb252-5"><a href="differential-analysis.html#cb252-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb252-6"><a href="differential-analysis.html#cb252-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb252-7"><a href="differential-analysis.html#cb252-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb252-8"><a href="differential-analysis.html#cb252-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb252-9"><a href="differential-analysis.html#cb252-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-10"><a href="differential-analysis.html#cb252-10" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb252-11"><a href="differential-analysis.html#cb252-11" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb252-12"><a href="differential-analysis.html#cb252-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb252-13"><a href="differential-analysis.html#cb252-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb252-14"><a href="differential-analysis.html#cb252-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb252-15"><a href="differential-analysis.html#cb252-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb252-16"><a href="differential-analysis.html#cb252-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb252-17"><a href="differential-analysis.html#cb252-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-18"><a href="differential-analysis.html#cb252-18" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq<span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb252-19"><a href="differential-analysis.html#cb252-19" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb252-20"><a href="differential-analysis.html#cb252-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb252-21"><a href="differential-analysis.html#cb252-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb252-22"><a href="differential-analysis.html#cb252-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;metagenomeseq&quot;</span>,</span>
<span id="cb252-23"><a href="differential-analysis.html#cb252-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb252-24"><a href="differential-analysis.html#cb252-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb252-25"><a href="differential-analysis.html#cb252-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb252-26"><a href="differential-analysis.html#cb252-26" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span>
<span id="cb252-27"><a href="differential-analysis.html#cb252-27" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="deseq2" class="section level3 hasAnchor" number="11.1.10">
<h3><span class="header-section-number">11.1.10</span> DESeq2<a href="differential-analysis.html#deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>DESeq2</strong> package is from <em>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</em> <span class="citation">(<a href="#ref-love2014moderated" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution.(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="differential-analysis.html#cb253-1" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb253-2"><a href="differential-analysis.html#cb253-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb253-3"><a href="differential-analysis.html#cb253-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb253-4"><a href="differential-analysis.html#cb253-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb253-5"><a href="differential-analysis.html#cb253-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-6"><a href="differential-analysis.html#cb253-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;logFC&quot;                            
##  [7] &quot;Statistic&quot;                         &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="differential-analysis.html#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment       Pvalue AdjustedPvalue      logFC  Statistic Log2FoldChange (Median)\nAA_vs_BB
## 1   g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.0551305049    0.237750302  1.9320873  1.9178463                        -1.1699250
## 2 g__Adlercreutzia 9_AA vs 14_BB         BB 0.0006165856    0.006077772  3.1571509  3.4242117                                NA
## 3  g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.6414696139    0.878400624 -0.4875701 -0.4656450                         0.1772545
## 4     g__Alistipes 9_AA vs 14_BB  Nonsignif 0.7735607924    0.953137405  0.2630218  0.2877204                        -0.4280937
## 5  g__Anaerostipes 9_AA vs 14_BB  Nonsignif 0.1814463414    0.395198703  1.0037124  1.3363150                         1.0204641
## 6   g__Bacteroides 9_AA vs 14_BB  Nonsignif 0.4391206065    0.688620951 -0.6885763 -0.7736791                         1.5520348
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                      30                   22                 49.5                      -1.3859026              50.65217
## 2                       0                    0                  9.5                      -2.5541902              19.39130
## 3                     311                  333                294.5                       1.7486195             997.26087
## 4                      64                   55                 74.0                       0.2295656             163.17391
## 5                     515                  923                455.0                      -0.1010837             821.30435
## 6                    3806                 9118               3109.5                       0.9814940            5740.60870
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1          25.555556           66.78571                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 2           4.888889           28.71429                    43.48                 33.33                 50.00           13 (18;8)
## 3        1742.111111          518.42857                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 4         179.222222          152.85714                    73.91                 77.78                 71.43     0.87 (0.59;1.1)
## 5         786.555556          843.64286                    95.65                100.00                 92.86      1.1 (1.2;0.93)
## 6        8205.777778         4155.85714                   100.00                100.00                100.00       0.45 (-1.1;2)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>Statistic</strong>: test statistic (negative binomial model);</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for deseq2</strong>
</summary>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="differential-analysis.html#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb257-2"><a href="differential-analysis.html#cb257-2" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb257-3"><a href="differential-analysis.html#cb257-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb257-4"><a href="differential-analysis.html#cb257-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb257-5"><a href="differential-analysis.html#cb257-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb257-6"><a href="differential-analysis.html#cb257-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb257-7"><a href="differential-analysis.html#cb257-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb257-8"><a href="differential-analysis.html#cb257-8" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb257-9"><a href="differential-analysis.html#cb257-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb257-10"><a href="differential-analysis.html#cb257-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb257-11"><a href="differential-analysis.html#cb257-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb257-12"><a href="differential-analysis.html#cb257-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb257-13"><a href="differential-analysis.html#cb257-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb257-14"><a href="differential-analysis.html#cb257-14" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb257-15"><a href="differential-analysis.html#cb257-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb257-16"><a href="differential-analysis.html#cb257-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb257-17"><a href="differential-analysis.html#cb257-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb257-18"><a href="differential-analysis.html#cb257-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;deseq2&quot;</span>)</span>
<span id="cb257-19"><a href="differential-analysis.html#cb257-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb257-20"><a href="differential-analysis.html#cb257-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span>
<span id="cb257-21"><a href="differential-analysis.html#cb257-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="edger" class="section level3 hasAnchor" number="11.1.11">
<h3><span class="header-section-number">11.1.11</span> EdgeR<a href="differential-analysis.html#edger" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>EdgeR</strong> package is from <em>A scaling normalization method for differential expression analysis of RNA-seq data</em> <span class="citation">(<a href="#ref-robinson2010scaling" role="doc-biblioref">Robinson and Oshlack 2010</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="differential-analysis.html#cb258-1" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb258-2"><a href="differential-analysis.html#cb258-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb258-3"><a href="differential-analysis.html#cb258-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb258-4"><a href="differential-analysis.html#cb258-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-142"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-142-1.png" alt="EdgeR (BVC distance)" width="480" />
<p class="caption">
Figure 11.1: EdgeR (BVC distance)
</p>
</div>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="differential-analysis.html#cb259-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_edger)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjPval&quot;                           &quot;logFC&quot;                            
##  [7] &quot;logCPM&quot;                            &quot;LR&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
## [10] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [13] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [16] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [19] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="differential-analysis.html#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment       Pvalue    AdjPval     logFC    logCPM        LR Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Lactobacillus 9_AA vs 14_BB         BB 0.0002688579 0.01293042  5.145965 14.920303 13.275774                         -2.870821
## 2    g__Romboutsia 9_AA vs 14_BB         BB 0.0003747947 0.01293042  4.322922 13.242686 12.653842                         -6.499846
## 3   g__Eubacterium 9_AA vs 14_BB  Nonsignif 0.0017789184 0.04064484 -2.889222  8.386044  9.764728                                NA
## 4   g__Odoribacter 9_AA vs 14_BB         AA 0.0023562226 0.04064484 -2.718081  9.376449  9.249014                                NA
## 5     g__Roseburia 9_AA vs 14_BB         BB 0.0029682519 0.04096188  4.327703 14.410912  8.826882                                NA
## 6 g__Adlercreutzia 9_AA vs 14_BB         BB 0.0036077438 0.04148905  2.879377  8.887302  8.471396                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                      57                   27                197.5                       -5.067893            799.956522
## 2                     105                    3                271.5                       -3.197677            295.478261
## 3                       0                    0                  0.0                        2.508697              8.304348
## 4                       3                   30                  0.0                        2.050293             26.434783
## 5                     125                    0                252.5                       -3.184643            669.608696
## 6                       0                    0                  9.5                       -2.554190             19.391304
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1          38.444444        1289.500000                    86.96                 77.78                 92.86 4e+08 (4e+08;4e+08)
## 2          49.444444         453.642857                    78.26                 55.56                 92.86          42 (49;34)
## 3          16.666667           2.928571                    30.43                 44.44                 21.43     0.33 (-1.8;2.5)
## 4          49.111111          11.857143                    52.17                 77.78                 35.71     0.33 (-1.9;2.5)
## 5         113.000000        1027.428571                    65.22                 44.44                 78.57          64 (72;56)
## 6           4.888889          28.714286                    43.48                 33.33                 50.00           13 (18;8)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>logCPM</strong>: is the average expression of all samples for that particular gene across all samples on the log-scale expressed in counts per million (cpm, as calculated by edgeR after normalization);</p></li>
<li><p><strong>LR</strong>: the signed likelihood ratio test statistic;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for EdgeR</strong>
</summary>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="differential-analysis.html#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb263-2"><a href="differential-analysis.html#cb263-2" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb263-3"><a href="differential-analysis.html#cb263-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb263-4"><a href="differential-analysis.html#cb263-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb263-5"><a href="differential-analysis.html#cb263-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb263-6"><a href="differential-analysis.html#cb263-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb263-7"><a href="differential-analysis.html#cb263-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb263-8"><a href="differential-analysis.html#cb263-8" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb263-9"><a href="differential-analysis.html#cb263-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb263-10"><a href="differential-analysis.html#cb263-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb263-11"><a href="differential-analysis.html#cb263-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb263-12"><a href="differential-analysis.html#cb263-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb263-13"><a href="differential-analysis.html#cb263-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb263-14"><a href="differential-analysis.html#cb263-14" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb263-15"><a href="differential-analysis.html#cb263-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb263-16"><a href="differential-analysis.html#cb263-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb263-17"><a href="differential-analysis.html#cb263-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb263-18"><a href="differential-analysis.html#cb263-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;edger&quot;</span>)</span>
<span id="cb263-19"><a href="differential-analysis.html#cb263-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb263-20"><a href="differential-analysis.html#cb263-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span>
<span id="cb263-21"><a href="differential-analysis.html#cb263-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="ancom" class="section level3 hasAnchor" number="11.1.12">
<h3><span class="header-section-number">11.1.12</span> ANCOM<a href="differential-analysis.html#ancom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ANCOM</strong> (Analysis of composition of microbiomes) is from <em>Analysis of composition of microbiomes: a novel method for studying microbial composition”, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-mandal2015analysis" role="doc-biblioref">Mandal et al. 2015</a>)</span>. ANCOM makes no distributional assumptions and can be implemented in a linear model framework. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="differential-analysis.html#cb264-1" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb264-2"><a href="differential-analysis.html#cb264-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb264-3"><a href="differential-analysis.html#cb264-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb264-4"><a href="differential-analysis.html#cb264-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-144"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-144-1.png" alt="ANCOM (Structure Zero)" width="480" />
<p class="caption">
Figure 11.2: ANCOM (Structure Zero)
</p>
</div>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="differential-analysis.html#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;(W)q-values &lt; alpha&quot;               &quot;W_ratio&quot;                          
##  [7] &quot;detected_0.7&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="differential-analysis.html#cb267-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment    EffectSize (W)q-values &lt; alpha W_ratio detected_0.7 Log2FoldChange (Median)\nAA_vs_BB
## 1   g__Actinomyces 9_AA vs 14_BB         BB  0.7005817485                   0       0        FALSE                        -1.1699250
## 2 g__Adlercreutzia 9_AA vs 14_BB         BB  0.2230005222                   0       0        FALSE                                NA
## 3  g__Agathobacter 9_AA vs 14_BB         AA -0.5666589896                   0       0        FALSE                         0.1772545
## 4     g__Alistipes 9_AA vs 14_BB         BB  0.0005271225                   0       0        FALSE                        -0.4280937
## 5  g__Anaerostipes 9_AA vs 14_BB         BB  0.3772522367                   0       0        FALSE                         1.0204641
## 6   g__Bacteroides 9_AA vs 14_BB         AA -1.2561760038                   0       0        FALSE                         1.5520348
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                      30                   22                 49.5                      -1.3859026              50.65217
## 2                       0                    0                  9.5                      -2.5541902              19.39130
## 3                     311                  333                294.5                       1.7486195             997.26087
## 4                      64                   55                 74.0                       0.2295656             163.17391
## 5                     515                  923                455.0                      -0.1010837             821.30435
## 6                    3806                 9118               3109.5                       0.9814940            5740.60870
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1          25.555556           66.78571                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 2           4.888889           28.71429                    43.48                 33.33                 50.00           13 (18;8)
## 3        1742.111111          518.42857                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 4         179.222222          152.85714                    73.91                 77.78                 71.43     0.87 (0.59;1.1)
## 5         786.555556          843.64286                    95.65                100.00                 92.86      1.1 (1.2;0.93)
## 6        8205.777778         4155.85714                   100.00                100.00                100.00       0.45 (-1.1;2)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size by <strong>ANCOM</strong>;</p></li>
<li><p><strong>(W)q-values &lt; alpha</strong>: q-values less than alpha;</p></li>
<li><p><strong>W_ratio</strong>: the ratio of W values;</p></li>
<li><p><strong>detected_0.7</strong>: W_ratio more than 0.7;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for ANCOM</strong>
</summary>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="differential-analysis.html#cb269-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb269-2"><a href="differential-analysis.html#cb269-2" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb269-3"><a href="differential-analysis.html#cb269-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb269-4"><a href="differential-analysis.html#cb269-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb269-5"><a href="differential-analysis.html#cb269-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb269-6"><a href="differential-analysis.html#cb269-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb269-7"><a href="differential-analysis.html#cb269-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-8"><a href="differential-analysis.html#cb269-8" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb269-9"><a href="differential-analysis.html#cb269-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb269-10"><a href="differential-analysis.html#cb269-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb269-11"><a href="differential-analysis.html#cb269-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb269-12"><a href="differential-analysis.html#cb269-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb269-13"><a href="differential-analysis.html#cb269-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-14"><a href="differential-analysis.html#cb269-14" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb269-15"><a href="differential-analysis.html#cb269-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb269-16"><a href="differential-analysis.html#cb269-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb269-17"><a href="differential-analysis.html#cb269-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb269-18"><a href="differential-analysis.html#cb269-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ancom&quot;</span>)</span>
<span id="cb269-19"><a href="differential-analysis.html#cb269-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-20"><a href="differential-analysis.html#cb269-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb269-21"><a href="differential-analysis.html#cb269-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="corncob" class="section level3 hasAnchor" number="11.1.13">
<h3><span class="header-section-number">11.1.13</span> Corncob<a href="differential-analysis.html#corncob" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Corncob</strong> package is from <em>Modeling microbial abundances and dysbiosis with beta-binomial regression”, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-martin2020modeling" role="doc-biblioref">Martin, Witten, and Willis 2020</a>)</span>. Corncob is based on beta-binomial regression. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="differential-analysis.html#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb270-2"><a href="differential-analysis.html#cb270-2" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb270-3"><a href="differential-analysis.html#cb270-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb270-4"><a href="differential-analysis.html#cb270-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb270-5"><a href="differential-analysis.html#cb270-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb270-6"><a href="differential-analysis.html#cb270-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb270-7"><a href="differential-analysis.html#cb270-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-8"><a href="differential-analysis.html#cb270-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span>
<span id="cb270-9"><a href="differential-analysis.html#cb270-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb270-10"><a href="differential-analysis.html#cb270-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>other options for Corncob</strong>
</summary>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="differential-analysis.html#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb271-2"><a href="differential-analysis.html#cb271-2" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb271-3"><a href="differential-analysis.html#cb271-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb271-4"><a href="differential-analysis.html#cb271-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb271-5"><a href="differential-analysis.html#cb271-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb271-6"><a href="differential-analysis.html#cb271-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb271-7"><a href="differential-analysis.html#cb271-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb271-8"><a href="differential-analysis.html#cb271-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-9"><a href="differential-analysis.html#cb271-9" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb271-10"><a href="differential-analysis.html#cb271-10" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb271-11"><a href="differential-analysis.html#cb271-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb271-12"><a href="differential-analysis.html#cb271-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb271-13"><a href="differential-analysis.html#cb271-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb271-14"><a href="differential-analysis.html#cb271-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb271-15"><a href="differential-analysis.html#cb271-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-16"><a href="differential-analysis.html#cb271-16" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb271-17"><a href="differential-analysis.html#cb271-17" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb271-18"><a href="differential-analysis.html#cb271-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb271-19"><a href="differential-analysis.html#cb271-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb271-20"><a href="differential-analysis.html#cb271-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;corncob&quot;</span>,</span>
<span id="cb271-21"><a href="differential-analysis.html#cb271-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb271-22"><a href="differential-analysis.html#cb271-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-23"><a href="differential-analysis.html#cb271-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_corncob)</span>
<span id="cb271-24"><a href="differential-analysis.html#cb271-24" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models" class="section level3 hasAnchor" number="11.1.14">
<h3><span class="header-section-number">11.1.14</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="differential-analysis.html#cb272-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb272-2"><a href="differential-analysis.html#cb272-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb272-3"><a href="differential-analysis.html#cb272-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb272-4"><a href="differential-analysis.html#cb272-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb272-5"><a href="differential-analysis.html#cb272-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb272-6"><a href="differential-analysis.html#cb272-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb272-7"><a href="differential-analysis.html#cb272-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb272-8"><a href="differential-analysis.html#cb272-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb272-9"><a href="differential-analysis.html#cb272-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb272-10"><a href="differential-analysis.html#cb272-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb272-11"><a href="differential-analysis.html#cb272-11" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb272-12"><a href="differential-analysis.html#cb272-12" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb272-13"><a href="differential-analysis.html#cb272-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb272-14"><a href="differential-analysis.html#cb272-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb272-15"><a href="differential-analysis.html#cb272-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb272-16"><a href="differential-analysis.html#cb272-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb272-17"><a href="differential-analysis.html#cb272-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb272-18"><a href="differential-analysis.html#cb272-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb272-19"><a href="differential-analysis.html#cb272-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb272-20"><a href="differential-analysis.html#cb272-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb272-21"><a href="differential-analysis.html#cb272-21" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb272-22"><a href="differential-analysis.html#cb272-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb272-23"><a href="differential-analysis.html#cb272-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb272-24"><a href="differential-analysis.html#cb272-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb272-25"><a href="differential-analysis.html#cb272-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb272-26"><a href="differential-analysis.html#cb272-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb272-27"><a href="differential-analysis.html#cb272-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb272-28"><a href="differential-analysis.html#cb272-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb272-29"><a href="differential-analysis.html#cb272-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb272-30"><a href="differential-analysis.html#cb272-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb272-31"><a href="differential-analysis.html#cb272-31" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb272-32"><a href="differential-analysis.html#cb272-32" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb272-33"><a href="differential-analysis.html#cb272-33" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb272-34"><a href="differential-analysis.html#cb272-34" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb272-35"><a href="differential-analysis.html#cb272-35" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb272-36"><a href="differential-analysis.html#cb272-36" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb272-37"><a href="differential-analysis.html#cb272-37" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb272-38"><a href="differential-analysis.html#cb272-38" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb272-39"><a href="differential-analysis.html#cb272-39" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb272-40"><a href="differential-analysis.html#cb272-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb272-41"><a href="differential-analysis.html#cb272-41" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2)</span>
<span id="cb272-42"><a href="differential-analysis.html#cb272-42" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
<div id="metagenomic-sequencing-microbial-data-metaphlan23" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Metagenomic sequencing microbial data (metaphlan2/3)<a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same strategy to filter phyloseq object.</p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="differential-analysis.html#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;metaphlan2_ps&quot;</span>)</span>
<span id="cb273-2"><a href="differential-analysis.html#cb273-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-3"><a href="differential-analysis.html#cb273-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb273-4"><a href="differential-analysis.html#cb273-4" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb273-5"><a href="differential-analysis.html#cb273-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb273-6"><a href="differential-analysis.html#cb273-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb273-7"><a href="differential-analysis.html#cb273-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb273-8"><a href="differential-analysis.html#cb273-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-9"><a href="differential-analysis.html#cb273-9" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb273-10"><a href="differential-analysis.html#cb273-10" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps_remove_BRS, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb273-11"><a href="differential-analysis.html#cb273-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-12"><a href="differential-analysis.html#cb273-12" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb273-13"><a href="differential-analysis.html#cb273-13" aria-hidden="true" tabindex="-1"></a><span class="co">#dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(dada2_ps_genus, cutoff = 10, unclass = TRUE)</span></span>
<span id="cb273-14"><a href="differential-analysis.html#cb273-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-15"><a href="differential-analysis.html#cb273-15" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb273-16"><a href="differential-analysis.html#cb273-16" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(metaphlan2_ps_genus, <span class="at">cutoff =</span> <span class="fl">1e-4</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb273-17"><a href="differential-analysis.html#cb273-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-18"><a href="differential-analysis.html#cb273-18" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb273-19"><a href="differential-analysis.html#cb273-19" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(metaphlan2_ps_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb273-20"><a href="differential-analysis.html#cb273-20" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 54 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 54 taxa by 6 taxonomic ranks ]</code></pre>
<div id="limma_voom-1" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> limma_voom<a href="differential-analysis.html#limma_voom-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>limma</strong> package is from <em>voom: Precision weights unlock linear model analysis tools for RNA-seq read counts</em> <span class="citation">(<a href="#ref-law2014voom" role="doc-biblioref">Law et al. 2014</a>)</span>. Firstly, transforming count data to log2-counts per million (logCPM), estimate the mean-variance relationship and use this to compute appropriate observation-level weights. Secondly, fitting multiple linear models by weighted or generalized least squares. Finally, performing empirical bayes statistics for differential expression.</p>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="differential-analysis.html#cb275-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_mgs <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb275-2"><a href="differential-analysis.html#cb275-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb275-3"><a href="differential-analysis.html#cb275-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb275-4"><a href="differential-analysis.html#cb275-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb275-5"><a href="differential-analysis.html#cb275-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="differential-analysis.html#cb277-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom_mgs)</span></code></pre></div>
<pre><code>##                TaxaID         Block Enrichment EffectSize     logFC       Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Faecalibacterium 7_AA vs 15_BB         BB  0.3073228 0.3073228 1.361399e-07   3.027405e-06                         -4.011562
## 2     g__Ruminococcus 7_AA vs 15_BB         BB  0.2253165 0.2253165 1.729946e-07   3.027405e-06                         -2.906486
## 3          g__Blautia 7_AA vs 15_BB         BB  0.1655853 0.1655853 7.359395e-06   8.585960e-05                         -3.182422
## 4    g__Lactobacillus 7_AA vs 15_BB         BB  0.1445633 0.1445633 5.294053e-05   4.632296e-04                                NA
## 5  g__Bifidobacterium 7_AA vs 15_BB         BB  0.3734565 0.3734565 2.016069e-04   1.079861e-03                         -6.549698
## 6        g__Roseburia 7_AA vs 15_BB         BB  0.1286164 0.1286164 1.261249e-04   8.828745e-04                         -4.442582
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1              0.05346880            0.0047826            0.0771373                       -3.992131            0.05750352
## 2              0.02141570            0.0044236            0.0331677                       -2.892349            0.03578485
## 3              0.02000425            0.0036769            0.0333800                       -2.265551            0.02431008
## 4              0.00155690            0.0000000            0.0024249                       -5.399318            0.02118822
## 5              0.03731980            0.0009090            0.0851568                       -3.394708            0.10730063
## 6              0.00752330            0.0008393            0.0182502                       -2.451452            0.02013462
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1       0.0051489857         0.08193563                    77.27                 57.14                 86.67                75 (83;66)
## 2       0.0066508000         0.04938074                    81.82                 57.14                 93.33             270 (280;260)
## 3       0.0067591429         0.03250052                   100.00                100.00                100.00                90 (99;81)
## 4       0.0007282714         0.03073619                    72.73                 42.86                 86.67 5.3e+10 (5.3e+10;5.3e+10)
## 5       0.0143274857         0.15068810                   100.00                100.00                100.00             130 (140;120)
## 6       0.0049745857         0.02720931                    81.82                 57.14                 93.33               12 (17;7.2)</code></pre>
<details>
<summary>
<strong>limma_voom Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="differential-analysis.html#cb279-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb279-2"><a href="differential-analysis.html#cb279-2" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb279-3"><a href="differential-analysis.html#cb279-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb279-4"><a href="differential-analysis.html#cb279-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb279-5"><a href="differential-analysis.html#cb279-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb279-6"><a href="differential-analysis.html#cb279-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;limma_voom&quot;</span>)</span>
<span id="cb279-7"><a href="differential-analysis.html#cb279-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="wilcoxon-rank-sum-and-signed-rank-tests-1" class="section level3 hasAnchor" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparameter test methods, use the rank of taxa abundance to find the significant taxa.</p>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="differential-analysis.html#cb280-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb280-2"><a href="differential-analysis.html#cb280-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb280-3"><a href="differential-analysis.html#cb280-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb280-4"><a href="differential-analysis.html#cb280-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb280-5"><a href="differential-analysis.html#cb280-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="differential-analysis.html#cb282-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic       Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1       g__Alistipes 7_AA vs 15_BB  Nonsignif  2.0074906        71 0.2019346498     0.44173205                          2.061604
## 2    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.0846993        31 0.1383456045     0.32280641                         -4.964322
## 3     g__Bacteroides 7_AA vs 15_BB  Nonsignif  1.3155909        85 0.0212848297     0.06622869                          1.229057
## 4 g__Bifidobacterium 7_AA vs 15_BB         BB  4.0400547        11 0.0021226194     0.03357196                         -6.549698
## 5       g__Bilophila 7_AA vs 15_BB  Nonsignif  0.5394300        60 0.6160709679     0.73547689                          1.177523
## 6         g__Blautia 7_AA vs 15_BB         BB  1.7897262         6 0.0003518154     0.01231354                         -3.182422
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1              0.00711780            0.0236665            0.0056693                       0.4613459                   14.14
## 2              0.00031435            0.0000229            0.0007149                      -0.6171177                    8.43
## 3              0.24894650            0.4568864            0.1949060                       0.7906916                   16.14
## 4              0.03731980            0.0009090            0.0851568                      -1.3572361                    5.57
## 5              0.00104585            0.0020513            0.0009069                       0.1924811                   12.57
## 6              0.02000425            0.0036769            0.0333800                      -1.5869402                    4.86
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   10.27                    77.27                 71.43                 80.00     0.22 (-2.7;3.2)
## 2                   12.93                    86.36                 71.43                 93.33         6 (9.5;2.5)
## 3                    9.33                   100.00                100.00                100.00       0.31 (-2;2.6)
## 4                   14.27                   100.00                100.00                100.00       130 (140;120)
## 5                   11.00                    68.18                 71.43                 66.67      0.4 (-1.4;2.2)
## 6                   14.60                   100.00                100.00                100.00          90 (99;81)</code></pre>
<details>
<summary>
<strong>wilcox Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="differential-analysis.html#cb284-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb284-2"><a href="differential-analysis.html#cb284-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb284-3"><a href="differential-analysis.html#cb284-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb284-4"><a href="differential-analysis.html#cb284-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb284-5"><a href="differential-analysis.html#cb284-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb284-6"><a href="differential-analysis.html#cb284-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb284-7"><a href="differential-analysis.html#cb284-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-8"><a href="differential-analysis.html#cb284-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-1" class="section level3 hasAnchor" number="11.2.3">
<h3><span class="header-section-number">11.2.3</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="differential-analysis.html#cb285-1" aria-hidden="true" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb285-2"><a href="differential-analysis.html#cb285-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb285-3"><a href="differential-analysis.html#cb285-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb285-4"><a href="differential-analysis.html#cb285-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb285-5"><a href="differential-analysis.html#cb285-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)  </span>
<span id="cb285-6"><a href="differential-analysis.html#cb285-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb285-7"><a href="differential-analysis.html#cb285-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;LDA_Score&quot;                         &quot;EffectSize&quot;                        &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
##  [7] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [10] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [13] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [16] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="differential-analysis.html#cb287-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##              TaxaID         Block Enrichment     LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1    g__Bacteroides 7_AA vs 15_BB         AA -0.0460731133  1.3155909                          1.229057              0.24894650
## 2  g__Fusobacterium 7_AA vs 15_BB         AA -0.0037987707  1.3865382                                NA              0.00000000
## 3 g__Granulicatella 7_AA vs 15_BB         AA -0.0001910907  2.3743196                                NA              0.00002535
## 4    g__Eggerthella 7_AA vs 15_BB         BB  0.0011304919  0.2383906                                NA              0.00058010
## 5    g__Collinsella 7_AA vs 15_BB         BB  0.0029335038  0.7257659                                NA              0.00738145
## 6          g__Dorea 7_AA vs 15_BB         BB  0.0040601735  0.8975049                         -5.583789              0.00821250
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1            0.4568864            0.1949060                        1.180768          2.848799e-01       4.602683e-01       0.2030320467
## 2            0.0001212            0.0000000                        6.949821          9.725173e-03       3.004406e-02       0.0002430267
## 3            0.0000000            0.0000512                       -2.980828          5.953182e-05       1.044286e-05       0.0000824400
## 4            0.0000000            0.0007616                       -3.708330          2.115218e-03       2.291571e-04       0.0029953800
## 5            0.0000000            0.0112480                       -2.904701          1.205134e-02       2.221829e-03       0.0166384400
## 6            0.0002690            0.0129015                       -3.390495          1.067307e-02       1.429129e-03       0.0149869067
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   100.00                100.00                100.00       0.31 (-2;2.6)
## 2                    40.91                 85.71                 20.00     0.0033 (-11;11)
## 3                    72.73                 42.86                 86.67       120 (130;110)
## 4                    63.64                 42.86                 73.33       450 (460;440)
## 5                    68.18                 42.86                 80.00          20 (25;14)
## 6                    90.91                 71.43                100.00          29 (36;23)</code></pre>
<details>
<summary>
<strong>lefse Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="differential-analysis.html#cb289-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb289-2"><a href="differential-analysis.html#cb289-2" aria-hidden="true" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb289-3"><a href="differential-analysis.html#cb289-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb289-4"><a href="differential-analysis.html#cb289-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb289-5"><a href="differential-analysis.html#cb289-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb289-6"><a href="differential-analysis.html#cb289-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb289-7"><a href="differential-analysis.html#cb289-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb289-8"><a href="differential-analysis.html#cb289-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test-1" class="section level3 hasAnchor" number="11.2.4">
<h3><span class="header-section-number">11.2.4</span> t-test<a href="differential-analysis.html#t-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="differential-analysis.html#cb290-1" aria-hidden="true" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb290-2"><a href="differential-analysis.html#cb290-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb290-3"><a href="differential-analysis.html#cb290-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb290-4"><a href="differential-analysis.html#cb290-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb290-5"><a href="differential-analysis.html#cb290-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                    &quot;Enrichment&quot;                              
##  [4] &quot;EffectSize&quot;                               &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;        &quot;Median Abundance\n(All)&quot;                 
## [10] &quot;Median Abundance\nAA&quot;                     &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;              &quot;GeometricMean Abundance\nBB&quot;             
## [16] &quot;Occurrence (100%)\n(All)&quot;                 &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="differential-analysis.html#cb292-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic       Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1       g__Alistipes 7_AA vs 15_BB  Nonsignif  2.0074906  1.596368 1.595156e-01    0.389526338                          2.061604
## 2    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.0846993 -1.697056 1.093743e-01    0.319008386                         -4.964322
## 3     g__Bacteroides 7_AA vs 15_BB  Nonsignif  1.3155909  2.288232 4.545820e-02    0.159103700                          1.229057
## 4 g__Bifidobacterium 7_AA vs 15_BB  Nonsignif  4.0400547 -2.725201 1.523564e-02    0.090761685                         -6.549698
## 5       g__Bilophila 7_AA vs 15_BB  Nonsignif  0.5394300  1.203014 2.727749e-01    0.500106397                          1.177523
## 6         g__Blautia 7_AA vs 15_BB         BB  1.7897262 -5.254033 4.588669e-05    0.001606034                         -3.182422
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (geometricmean)\nAA_vs_BB
## 1              0.00711780            0.0236665            0.0056693                                 2.055288
## 2              0.00031435            0.0000229            0.0007149                                       NA
## 3              0.24894650            0.4568864            0.1949060                                 1.607214
## 4              0.03731980            0.0009090            0.0851568                                       NA
## 5              0.00104585            0.0020513            0.0009069                                 3.174655
## 6              0.02000425            0.0036769            0.0333800                                       NA
##   GeometricMean Abundance\n(All) GeometricMean Abundance\nAA GeometricMean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA
## 1                      0.3989446                   0.7346341                   0.1767534                    77.27                 71.43
## 2                      1.8882218                         NaN                   1.8882218                    86.36                 71.43
## 3                      0.3871167                   0.6044631                   0.1984039                   100.00                100.00
## 4                      0.1936930                         NaN                   0.1936930                   100.00                100.00
## 5                      0.6988408                   2.6168456                   0.2898089                    68.18                 71.43
## 6                      0.9008965                         NaN                   0.9008965                   100.00                100.00
##   Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 80.00     0.22 (-2.7;3.2)
## 2                 93.33         6 (9.5;2.5)
## 3                100.00       0.31 (-2;2.6)
## 4                100.00       130 (140;120)
## 5                 66.67      0.4 (-1.4;2.2)
## 6                100.00          90 (99;81)</code></pre>
<details>
<summary>
<strong>ttest Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="differential-analysis.html#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb294-2"><a href="differential-analysis.html#cb294-2" aria-hidden="true" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb294-3"><a href="differential-analysis.html#cb294-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb294-4"><a href="differential-analysis.html#cb294-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb294-5"><a href="differential-analysis.html#cb294-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb294-6"><a href="differential-analysis.html#cb294-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb294-7"><a href="differential-analysis.html#cb294-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models-1" class="section level3 hasAnchor" number="11.2.5">
<h3><span class="header-section-number">11.2.5</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="differential-analysis.html#cb295-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb295-2"><a href="differential-analysis.html#cb295-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb295-3"><a href="differential-analysis.html#cb295-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb295-4"><a href="differential-analysis.html#cb295-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb295-5"><a href="differential-analysis.html#cb295-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb295-6"><a href="differential-analysis.html#cb295-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb295-7"><a href="differential-analysis.html#cb295-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb295-8"><a href="differential-analysis.html#cb295-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb295-9"><a href="differential-analysis.html#cb295-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb295-10"><a href="differential-analysis.html#cb295-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb295-11"><a href="differential-analysis.html#cb295-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2_mgs)</span>
<span id="cb295-12"><a href="differential-analysis.html#cb295-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>maaslin2 Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="differential-analysis.html#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb296-2"><a href="differential-analysis.html#cb296-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb296-3"><a href="differential-analysis.html#cb296-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb296-4"><a href="differential-analysis.html#cb296-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb296-5"><a href="differential-analysis.html#cb296-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb296-6"><a href="differential-analysis.html#cb296-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb296-7"><a href="differential-analysis.html#cb296-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb296-8"><a href="differential-analysis.html#cb296-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb296-9"><a href="differential-analysis.html#cb296-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb296-10"><a href="differential-analysis.html#cb296-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb296-11"><a href="differential-analysis.html#cb296-11" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
</div>
<div id="visualization-4" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Visualization<a href="differential-analysis.html#visualization-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Volcano plot is used to display differential analysis. <code>plot_volcano</code> provides multiple parameters for plotting volcano. More details to see <code>help(plot_volcano)</code>.</p>
<p>The barplot is used to display the lefse results.</p>
<div id="volcano-plot" class="section level3 hasAnchor" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Volcano plot<a href="differential-analysis.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The X and Y coordinate axis are flexible to choose for Volcano. Here, we choose <em>logFC</em> and <em>AdjustedPvalue</em> to visualize the results</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="differential-analysis.html#cb297-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb297-2"><a href="differential-analysis.html#cb297-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb297-3"><a href="differential-analysis.html#cb297-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb297-4"><a href="differential-analysis.html#cb297-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb297-5"><a href="differential-analysis.html#cb297-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb297-6"><a href="differential-analysis.html#cb297-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="differential-analysis.html#cb299-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_volcano <span class="ot">&lt;-</span> <span class="fu">plot_volcano</span>(</span>
<span id="cb299-2"><a href="differential-analysis.html#cb299-2" aria-hidden="true" tabindex="-1"></a>    DA_limma_voom,</span>
<span id="cb299-3"><a href="differential-analysis.html#cb299-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb299-4"><a href="differential-analysis.html#cb299-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;logFC&quot;</span>,</span>
<span id="cb299-5"><a href="differential-analysis.html#cb299-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb299-6"><a href="differential-analysis.html#cb299-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb299-7"><a href="differential-analysis.html#cb299-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb299-8"><a href="differential-analysis.html#cb299-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb299-9"><a href="differential-analysis.html#cb299-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span>
<span id="cb299-10"><a href="differential-analysis.html#cb299-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb299-11"><a href="differential-analysis.html#cb299-11" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_volcano</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-160"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-160-1.png" alt="Volcano (limma-voom: logFC and AdjustedPvalue)" width="672" />
<p class="caption">
Figure 11.3: Volcano (limma-voom: logFC and AdjustedPvalue)
</p>
</div>
<details>
<summary>
<strong>DA_limma_voom_volcano effectsize</strong>
</summary>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="differential-analysis.html#cb300-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb300-2"><a href="differential-analysis.html#cb300-2" aria-hidden="true" tabindex="-1"></a>    DA_limma_voom,</span>
<span id="cb300-3"><a href="differential-analysis.html#cb300-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb300-4"><a href="differential-analysis.html#cb300-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;EffectSize&quot;</span>,</span>
<span id="cb300-5"><a href="differential-analysis.html#cb300-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb300-6"><a href="differential-analysis.html#cb300-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb300-7"><a href="differential-analysis.html#cb300-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb300-8"><a href="differential-analysis.html#cb300-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb300-9"><a href="differential-analysis.html#cb300-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">8</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-161"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-161-1.png" alt="Volcano (limma-voom: EffectSize and Pvalue)" width="672" />
<p class="caption">
Figure 11.4: Volcano (limma-voom: EffectSize and Pvalue)
</p>
</div>
<p>The logFC and EffectSize are the same values in limma-voom.</p>
</details>
</div>
<div id="barplot-in-lefse" class="section level3 hasAnchor" number="11.3.2">
<h3><span class="header-section-number">11.3.2</span> barplot in lefse<a href="differential-analysis.html#barplot-in-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="differential-analysis.html#cb301-1" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb301-2"><a href="differential-analysis.html#cb301-2" aria-hidden="true" tabindex="-1"></a>    dada2_ps_genus_filter_trim,</span>
<span id="cb301-3"><a href="differential-analysis.html#cb301-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb301-4"><a href="differential-analysis.html#cb301-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb301-5"><a href="differential-analysis.html#cb301-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb301-6"><a href="differential-analysis.html#cb301-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb301-7"><a href="differential-analysis.html#cb301-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb301-8"><a href="differential-analysis.html#cb301-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">da_res =</span> DA_lefse,</span>
<span id="cb301-9"><a href="differential-analysis.html#cb301-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb301-10"><a href="differential-analysis.html#cb301-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb301-11"><a href="differential-analysis.html#cb301-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>)) </span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-162"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-162-1.png" alt="Barplot (Lefse)" width="576" />
<p class="caption">
Figure 11.5: Barplot (Lefse)
</p>
</div>
</div>
</div>
<div id="dominant-taxa" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Dominant taxa<a href="differential-analysis.html#dominant-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Display the significant taxa with selection using boxplot.</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="differential-analysis.html#cb302-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb302-2"><a href="differential-analysis.html#cb302-2" aria-hidden="true" tabindex="-1"></a>   metaphlan2_ps_genus_filter_trim,</span>
<span id="cb302-3"><a href="differential-analysis.html#cb302-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb302-4"><a href="differential-analysis.html#cb302-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb302-5"><a href="differential-analysis.html#cb302-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-6"><a href="differential-analysis.html#cb302-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_topN_boxplot</span>(</span>
<span id="cb302-7"><a href="differential-analysis.html#cb302-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb302-8"><a href="differential-analysis.html#cb302-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> DA_wilcox,</span>
<span id="cb302-9"><a href="differential-analysis.html#cb302-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Median)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb302-10"><a href="differential-analysis.html#cb302-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.2</span>,</span>
<span id="cb302-11"><a href="differential-analysis.html#cb302-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb302-12"><a href="differential-analysis.html#cb302-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.3</span>,</span>
<span id="cb302-13"><a href="differential-analysis.html#cb302-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">3</span>,</span>
<span id="cb302-14"><a href="differential-analysis.html#cb302-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">taxa_name =</span> <span class="st">&quot;s__Ruminococcus_torques&quot;</span>,</span>
<span id="cb302-15"><a href="differential-analysis.html#cb302-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-163"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-163-1.png" alt="Dominant Taxa" width="768" />
<p class="caption">
Figure 11.6: Dominant Taxa
</p>
</div>
</div>
<div id="multiple-differential-analysis-by-one-function" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> Multiple differential analysis by one function<a href="differential-analysis.html#multiple-differential-analysis-by-one-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>here, we provide the <code>run_multiple_da</code> for obtaining the results list from multiple differential analysis methods.</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="differential-analysis.html#cb303-1" aria-hidden="true" tabindex="-1"></a>multiple_res <span class="ot">&lt;-</span> <span class="fu">run_multiple_da</span>(</span>
<span id="cb303-2"><a href="differential-analysis.html#cb303-2" aria-hidden="true" tabindex="-1"></a>       dada2_ps_genus_filter_trim,</span>
<span id="cb303-3"><a href="differential-analysis.html#cb303-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb303-4"><a href="differential-analysis.html#cb303-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb303-5"><a href="differential-analysis.html#cb303-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">da_method =</span> <span class="fu">c</span>(<span class="st">&quot;aldex&quot;</span>, <span class="st">&quot;limma_voom&quot;</span>, <span class="st">&quot;mbzinb&quot;</span>, <span class="st">&quot;omnibus&quot;</span>),</span>
<span id="cb303-6"><a href="differential-analysis.html#cb303-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">p_adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  69  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="differential-analysis.html#cb305-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(multiple_res)</span></code></pre></div>
<pre><code>## [1] &quot;aldex&quot;      &quot;limma_voom&quot; &quot;mbzinb&quot;     &quot;omnibus&quot;</code></pre>
<ul>
<li>plot results</li>
</ul>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="differential-analysis.html#cb307-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_multiple_DA</span>(</span>
<span id="cb307-2"><a href="differential-analysis.html#cb307-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Multip_DA_res =</span> multiple_res,</span>
<span id="cb307-3"><a href="differential-analysis.html#cb307-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_list =</span> <span class="fu">c</span>(<span class="st">&quot;EffectSize&quot;</span>, <span class="st">&quot;logFC&quot;</span>, <span class="st">&quot;mean.LFC&quot;</span>, <span class="st">&quot;abund.LFC.CompvarBB.est&quot;</span>),</span>
<span id="cb307-4"><a href="differential-analysis.html#cb307-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb307-5"><a href="differential-analysis.html#cb307-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_index =</span> <span class="st">&quot;AdjustedPvalue&quot;</span>,</span>
<span id="cb307-6"><a href="differential-analysis.html#cb307-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb307-7"><a href="differential-analysis.html#cb307-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellwidth =</span> <span class="dv">50</span>,</span>
<span id="cb307-8"><a href="differential-analysis.html#cb307-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellheight =</span> <span class="dv">10</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-165"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-165-1.png" alt="Multiple DA results" width="768" />
<p class="caption">
Figure 11.7: Multiple DA results
</p>
</div>
</div>
<div id="systematic-information-8" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> Systematic Information<a href="differential-analysis.html#systematic-information-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="differential-analysis.html#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] RColorBrewer_1.1-2 viridis_0.6.2      viridisLite_0.4.0  ggpubr_0.4.0       ggplot2_3.3.5      phyloseq_1.38.0    tibble_3.1.6      
## [8] dplyr_1.0.8        XMAS_2.1.1        
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2                  reticulate_1.24             tidyselect_1.1.2            RSQLite_2.2.10             
##   [5] AnnotationDbi_1.56.2        htmlwidgets_1.5.4           grid_4.1.2                  BiocParallel_1.28.3        
##   [9] Rtsne_0.15                  munsell_0.5.0               units_0.8-0                 codetools_0.2-18           
##  [13] DT_0.21                     umap_0.2.8.0                withr_2.5.0                 colorspace_2.0-3           
##  [17] Biobase_2.54.0              highr_0.9                   knitr_1.37                  rstudioapi_0.13            
##  [21] stats4_4.1.2                pscl_1.5.5                  bayesm_3.1-4                robustbase_0.93-9          
##  [25] ggsignif_0.6.3              MatrixGenerics_1.6.0        labeling_0.4.2              optparse_1.7.1             
##  [29] mbzinb_0.2                  GenomeInfoDbData_1.2.7      lpsymphony_1.22.0           bit64_4.0.5                
##  [33] farver_2.1.0                pheatmap_1.0.12             rhdf5_2.38.1                vctrs_0.3.8                
##  [37] generics_0.1.2              TH.data_1.1-0               xfun_0.30                   Maaslin2_1.8.0             
##  [41] R6_2.5.1                    GenomeInfoDb_1.30.1         RVenn_1.1.0                 locfit_1.5-9.5             
##  [45] RcppZiggurat_0.1.6          bitops_1.0-7                rhdf5filters_1.6.0          cachem_1.0.6               
##  [49] DelayedArray_0.20.0         assertthat_0.2.1            scales_1.1.1                multcomp_1.4-18            
##  [53] nnet_7.3-17                 gtable_0.3.0                sandwich_3.0-1              rlang_1.0.2                
##  [57] genefilter_1.76.0           splines_4.1.2               protoclust_1.6.3            rstatix_0.7.0              
##  [61] broom_0.7.12                checkmate_2.0.0             yaml_2.3.5                  reshape2_1.4.4             
##  [65] abind_1.4-5                 crosstalk_1.2.0             backports_1.4.1             Rfast_2.0.6                
##  [69] qvalue_2.26.0               Hmisc_4.6-0                 DiagrammeR_1.0.9            tensorA_0.36.2             
##  [73] tools_4.1.2                 bookdown_0.24               zCompositions_1.4.0         ellipsis_0.3.2             
##  [77] gplots_3.1.1                jquerylib_0.1.4             biomformat_1.22.0           proxy_0.4-26               
##  [81] BiocGenerics_0.40.0         Rcpp_1.0.8.2                plyr_1.8.6                  visNetwork_2.1.0           
##  [85] base64enc_0.1-3             zlibbioc_1.40.0             classInt_0.4-3              purrr_0.3.4                
##  [89] RCurl_1.98-1.6              openssl_2.0.0               rpart_4.1.16                Wrench_1.12.0              
##  [93] cowplot_1.1.1               S4Vectors_0.32.3            zoo_1.8-9                   SummarizedExperiment_1.24.0
##  [97] ggrepel_0.9.1               cluster_2.1.2               magrittr_2.0.2              RSpectra_0.16-0            
## [101] data.table_1.14.2           truncnorm_1.0-8             mvtnorm_1.1-3               matrixStats_0.61.0         
## [105] ALDEx2_1.26.0               corncob_0.2.0               RAIDA_1.0                   evaluate_0.15              
## [109] xtable_1.8-4                XML_3.99-0.9                jpeg_0.1-9                  IRanges_2.28.0             
## [113] gridExtra_2.3               shape_1.4.6                 compiler_4.1.2              KernSmooth_2.23-20         
## [117] crayon_1.5.0                htmltools_0.5.2             mgcv_1.8-39                 pcaPP_1.9-74               
## [121] ggVennDiagram_1.2.1         Formula_1.2-4               tidyr_1.2.0                 geneplotter_1.72.0         
## [125] libcoin_1.0-9               DBI_1.1.2                   corrplot_0.92               biglm_0.9-2.1              
## [129] MASS_7.3-55                 sf_1.0-7                    compositions_2.0-4          Matrix_1.4-0               
## [133] ade4_1.7-18                 getopt_1.20.3               car_3.0-12                  permute_0.9-7              
## [137] cli_3.2.0                   parallel_4.1.2              igraph_1.2.11               GenomicRanges_1.46.1       
## [141] pkgconfig_2.0.3             coin_1.4-2                  foreign_0.8-82              foreach_1.5.2              
## [145] annotate_1.72.0             bslib_0.3.1                 multtest_2.50.0             XVector_0.34.0             
## [149] NADA_1.6-1.1                stringr_1.4.0               digest_0.6.29               vegan_2.5-7                
## [153] Biostrings_2.62.0           rmarkdown_2.13              htmlTable_2.4.0             edgeR_3.36.0               
## [157] gtools_3.9.2                modeltools_0.2-23           lifecycle_1.0.1             nlme_3.1-155               
## [161] jsonlite_1.8.0              Rhdf5lib_1.16.0             carData_3.0-5               askpass_1.1                
## [165] limma_3.50.1                fansi_1.0.2                 pillar_1.7.0                lattice_0.20-45            
## [169] KEGGREST_1.34.0             fastmap_1.1.0               httr_1.4.2                  DEoptimR_1.0-10            
## [173] survival_3.3-1              glue_1.6.2                  png_0.1-7                   iterators_1.0.14           
## [177] glmnet_4.1-3                bit_4.0.4                   class_7.3-20                sass_0.4.0                 
## [181] stringi_1.7.6               metagenomeSeq_1.36.0        blob_1.2.2                  DESeq2_1.34.0              
## [185] latticeExtra_0.6-29         caTools_1.18.2              memoise_2.0.1               e1071_1.7-9                
## [189] ape_5.6-2</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chen2018omnibus" class="csl-entry">
Chen, Jun, Emily King, Rebecca Deek, Zhi Wei, Yue Yu, Diane Grill, and Karla Ballman. 2018. <span>“An Omnibus Test for Differential Distribution Analysis of Microbiome Sequencing Data.”</span> <em>Bioinformatics</em> 34 (4): 643–51.
</div>
<div id="ref-chen2018gmpr" class="csl-entry">
Chen, Li, James Reeve, Lujun Zhang, Shengbing Huang, Xuefeng Wang, and Jun Chen. 2018. <span>“GMPR: A Robust Normalization Method for Zero-Inflated Count Data with Application to Microbiome Sequencing Data.”</span> <em>PeerJ</em> 6: e4600.
</div>
<div id="ref-fernandes2014unifying" class="csl-entry">
Fernandes, Andrew D, Jennifer Ns Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>“Unifying the Analysis of High-Throughput Sequencing Datasets: Characterizing RNA-Seq, 16s rRNA Gene Sequencing and Selective Growth Experiments by Compositional Data Analysis.”</span> <em>Microbiome</em> 2 (1): 1–13.
</div>
<div id="ref-law2014voom" class="csl-entry">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. <span>“Voom: Precision Weights Unlock Linear Model Analysis Tools for RNA-Seq Read Counts.”</span> <em>Genome Biology</em> 15 (2): 1–17.
</div>
<div id="ref-love2014moderated" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.”</span> <em>Genome Biology</em> 15 (12): 1–21.
</div>
<div id="ref-mallick2021multivariable" class="csl-entry">
Mallick, Himel, Ali Rahnavard, Lauren J McIver, Siyuan Ma, Yancong Zhang, Long H Nguyen, Timothy L Tickle, et al. 2021. <span>“Multivariable Association Discovery in Population-Scale Meta-Omics Studies.”</span> <em>PLoS Computational Biology</em> 17 (11): e1009442.
</div>
<div id="ref-mandal2015analysis" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A White, Merete Eggesbø, Rob Knight, and Shyamal D Peddada. 2015. <span>“Analysis of Composition of Microbiomes: A Novel Method for Studying Microbial Composition.”</span> <em>Microbial Ecology in Health and Disease</em> 26 (1): 27663.
</div>
<div id="ref-martin2020modeling" class="csl-entry">
Martin, Bryan D, Daniela Witten, and Amy D Willis. 2020. <span>“Modeling Microbial Abundances and Dysbiosis with Beta-Binomial Regression.”</span> <em>The Annals of Applied Statistics</em> 14 (1): 94.
</div>
<div id="ref-nearing2022microbiome" class="csl-entry">
Nearing, Jacob T, Gavin M Douglas, Molly G Hayes, Jocelyn MacDonald, Dhwani K Desai, Nicole Allward, Casey Jones, et al. 2022. <span>“Microbiome Differential Abundance Methods Produce Different Results Across 38 Datasets.”</span> <em>Nature Communications</em> 13 (1): 1–16.
</div>
<div id="ref-paulson2013differential" class="csl-entry">
Paulson, Joseph N, O Colin Stine, Héctor Corrada Bravo, and Mihai Pop. 2013. <span>“Differential Abundance Analysis for Microbial Marker-Gene Surveys.”</span> <em>Nature Methods</em> 10 (12): 1200–1202.
</div>
<div id="ref-robinson2010scaling" class="csl-entry">
Robinson, Mark D, and Alicia Oshlack. 2010. <span>“A Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data.”</span> <em>Genome Biology</em> 11 (3): 1–9.
</div>
<div id="ref-segata2011metagenomic" class="csl-entry">
Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. <span>“Metagenomic Biomarker Discovery and Explanation.”</span> <em>Genome Biology</em> 12 (6): 1–18.
</div>
<div id="ref-sohn2015robust" class="csl-entry">
Sohn, Michael B, Ruofei Du, and Lingling An. 2015. <span>“A Robust Approach for Identifying Differentially Abundant Features in Metagenomic Samples.”</span> <em>Bioinformatics</em> 31 (14): 2269–75.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="core-microbiota.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualization-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/10-differentail-analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
