<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Differential analysis | Data analysis based on XMAS 2.0 package in microbiota data" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-06-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="core-microbiota.html"/>
<link rel="next" href="visualization-5.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation<span></span></a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette<span></span></a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors<span></span></a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2<span></span></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>4.2.1</b> Importing results from dada2 pipeline<span></span></a></li>
<li class="chapter" data-level="4.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>4.2.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>4.2.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.2.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>4.2.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.2.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>4.2.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3<span></span></a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>4.3.1</b> Importing results<span></span></a></li>
<li class="chapter" data-level="4.3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table-1"><i class="fa fa-check"></i><b>4.3.2</b> taxa table<span></span></a></li>
<li class="chapter" data-level="4.3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table-1"><i class="fa fa-check"></i><b>4.3.3</b> otu table<span></span></a></li>
<li class="chapter" data-level="4.3.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table-1"><i class="fa fa-check"></i><b>4.3.4</b> metadata table<span></span></a></li>
<li class="chapter" data-level="4.3.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object-1"><i class="fa fa-check"></i><b>4.3.5</b> phyloseq object<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Reads’ track<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics<span></span></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.6</b> Analysis of Similarity (ANOSIM)<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.7</b> Ordination<span></span></a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.7.1</b> Principal Component Analysis (PCA)<span></span></a></li>
<li class="chapter" data-level="8.7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.7.2</b> Principal Coordinate Analysis (PCoA)<span></span></a></li>
<li class="chapter" data-level="8.7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.7.3</b> t-distributed stochastic neighbor embedding (t-SNE)<span></span></a></li>
<li class="chapter" data-level="8.7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.7.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="composition.html"><a href="composition.html"><i class="fa fa-check"></i><b>9</b> Composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="composition.html"><a href="composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="composition.html"><a href="composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.2</b> XVIZ <code>plot_stacked_bar</code><span></span></a></li>
<li class="chapter" data-level="9.3" data-path="composition.html"><a href="composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.3</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="composition.html"><a href="composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="composition.html"><a href="composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus<span></span></a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq<span></span></a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR<span></span></a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM<span></span></a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob<span></span></a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)<span></span></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom-1"><i class="fa fa-check"></i><b>11.2.1</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.2</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.3</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.4</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.2.5" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.5</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot<span></span></a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot<span></span></a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot<span></span></a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density<span></span></a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination<span></span></a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot<span></span></a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot<span></span></a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn<span></span></a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.10</b> plot_stacked_bar_XIVZ<span></span></a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.11</b> plot_StackBarPlot<span></span></a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.12</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>13</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>13.1</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>13.2</b> Amplicon sequencing (16s)<span></span></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>13.2.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="13.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>13.2.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="13.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>13.2.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="13.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>13.2.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="13.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>13.2.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="13.2.6" data-path="examples.html"><a href="examples.html#microbial-composition"><i class="fa fa-check"></i><b>13.2.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="13.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>13.2.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="examples.html"><a href="examples.html#metagenomics"><i class="fa fa-check"></i><b>13.3</b> Metagenomics<span></span></a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>13.3.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="13.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>13.3.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="13.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>13.3.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="13.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>13.3.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="13.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>13.3.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="13.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>13.3.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="13.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>13.3.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>13.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data analysis based on XMAS 2.0 package in microbiota data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-analysis" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Differential analysis<a href="differential-analysis.html#differential-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Identifying the significant taxa between case and control group is necessary. There are too many differential analysis (DA) approaches to choose <span class="citation">(<a href="#ref-nearing2022microbiome" role="doc-biblioref">Nearing et al. 2022</a>)</span>, we provide some of them which focusing on microbial data, including:</p>
<table style="width:100%;">
<colgroup>
<col width="14%" />
<col width="15%" />
<col width="24%" />
<col width="15%" />
<col width="19%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Tool(version)</strong></th>
<th><strong>Input</strong></th>
<th><strong>Normalization</strong></th>
<th><strong>Transformation</strong></th>
<th><strong>Distribution</strong></th>
<th><strong>MicrobialData</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>ALDEx2 (1.26.0)</td>
<td>Counts</td>
<td>None</td>
<td>CLR</td>
<td>Dirichlet-multinormial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>limma voom (3.50.1)</td>
<td>Counts/Relative</td>
<td>None/TMM</td>
<td>Log; Precision weighting</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>mbzinb (0.2)</td>
<td>Counts</td>
<td>RLE</td>
<td>None</td>
<td>zero-inflated negative binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>omnibus (0.2)</td>
<td>Counts</td>
<td>GMPR(Geometric Mean of Pairwise Ratios)</td>
<td>None</td>
<td>zero-inflated negative binomial</td>
<td>16s</td>
</tr>
<tr class="odd">
<td>RAIDA (1.0)</td>
<td>Counts</td>
<td>None</td>
<td>Log</td>
<td>modified t-test</td>
<td>16s</td>
</tr>
<tr class="even">
<td>Wilcox(rare/CLR)</td>
<td>Counts/Relative</td>
<td>None</td>
<td>None/CLR</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>LEfSe</td>
<td>Rarefied Counts//Relative</td>
<td>TSS</td>
<td>None</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="even">
<td>t-test (rare)</td>
<td>Counts//Relative</td>
<td>None</td>
<td>None</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>metagenomeSeq (1.36.0)</td>
<td>Counts</td>
<td>CSS</td>
<td>Log</td>
<td>Zero-inflated (log) Normal</td>
<td>16s</td>
</tr>
<tr class="even">
<td>DESeq2 (1.34.0)</td>
<td>Counts</td>
<td>RLE</td>
<td>None</td>
<td>Negative binomial</td>
<td>16s</td>
</tr>
<tr class="odd">
<td>edgeR (3.36.0)</td>
<td>Counts</td>
<td>RLE/TMM</td>
<td>None</td>
<td>Negative binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>ANCOM-II (2.1)</td>
<td>Counts/Relative</td>
<td>None</td>
<td>ALR</td>
<td>Non-parametric</td>
<td>16s/MGS</td>
</tr>
<tr class="odd">
<td>Corncob (0.2.0)</td>
<td>Counts</td>
<td>None</td>
<td>None</td>
<td>Beta-binomial</td>
<td>16s</td>
</tr>
<tr class="even">
<td>MaAslin2 (1.8.0)</td>
<td>Counts/Relative abundance</td>
<td>None/TSS</td>
<td>AST</td>
<td>Normal</td>
<td>16s/MGS</td>
</tr>
</tbody>
</table>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="differential-analysis.html#cb193-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb193-2"><a href="differential-analysis.html#cb193-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb193-3"><a href="differential-analysis.html#cb193-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb193-4"><a href="differential-analysis.html#cb193-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb193-5"><a href="differential-analysis.html#cb193-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb193-6"><a href="differential-analysis.html#cb193-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb193-7"><a href="differential-analysis.html#cb193-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
<div id="amplicon-sequencing-microbial-data-16s" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Amplicon sequencing microbial data (16s)<a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same strategy to filter phyloseq object.</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="differential-analysis.html#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;dada2_ps&quot;</span>)</span>
<span id="cb194-2"><a href="differential-analysis.html#cb194-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-3"><a href="differential-analysis.html#cb194-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb194-4"><a href="differential-analysis.html#cb194-4" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb194-5"><a href="differential-analysis.html#cb194-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb194-6"><a href="differential-analysis.html#cb194-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb194-7"><a href="differential-analysis.html#cb194-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb194-8"><a href="differential-analysis.html#cb194-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-9"><a href="differential-analysis.html#cb194-9" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Rarefying counts in phyloseq-class object</span></span>
<span id="cb194-10"><a href="differential-analysis.html#cb194-10" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(dada2_ps_remove_BRS, <span class="at">size =</span> <span class="dv">51181</span>, <span class="at">rng_seed =</span> <span class="dv">123</span>)</span>
<span id="cb194-11"><a href="differential-analysis.html#cb194-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-12"><a href="differential-analysis.html#cb194-12" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb194-13"><a href="differential-analysis.html#cb194-13" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(dada2_ps_rarefy, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, <span class="at">absolute =</span> <span class="cn">TRUE</span>)</span>
<span id="cb194-14"><a href="differential-analysis.html#cb194-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-15"><a href="differential-analysis.html#cb194-15" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb194-16"><a href="differential-analysis.html#cb194-16" aria-hidden="true" tabindex="-1"></a><span class="co">#dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(dada2_ps_rare_genus, cutoff = 10, unclass = TRUE)</span></span>
<span id="cb194-17"><a href="differential-analysis.html#cb194-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-18"><a href="differential-analysis.html#cb194-18" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb194-19"><a href="differential-analysis.html#cb194-19" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(dada2_ps_rare_genus, <span class="at">cutoff =</span> <span class="dv">10</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb194-20"><a href="differential-analysis.html#cb194-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-21"><a href="differential-analysis.html#cb194-21" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb194-22"><a href="differential-analysis.html#cb194-22" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(dada2_ps_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb194-23"><a href="differential-analysis.html#cb194-23" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 99 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 99 taxa by 6 taxonomic ranks ]</code></pre>
<div id="aldex2" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> ALDEx2<a href="differential-analysis.html#aldex2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ALDEx2</strong> package is from <em>Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis</em> <span class="citation">(<a href="#ref-fernandes2014unifying" role="doc-biblioref">Fernandes et al. 2014</a>)</span>, and its principle is using log-ratio transformation and statistical testing to find the significant Taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<p><code>run_aldex</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_aldex)</code>.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="differential-analysis.html#cb196-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb196-2"><a href="differential-analysis.html#cb196-2" aria-hidden="true" tabindex="-1"></a>                     dada2_ps_genus_filter_trim,</span>
<span id="cb196-3"><a href="differential-analysis.html#cb196-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb196-4"><a href="differential-analysis.html#cb196-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb196-5"><a href="differential-analysis.html#cb196-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="differential-analysis.html#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="differential-analysis.html#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment  EffectSize    Pvalue AdjustedPvalue Median CLR \n(All) Median CLR\nAA Median CLR\nBB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif  0.06559243 0.6440126      0.9019852         -3.6026491      -3.838012     -3.3645033
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif  0.09469081 0.6282127      0.9119830          2.5597286       1.817144      3.0977503
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif  0.32811284 0.2907930      0.7651024         -0.7363202      -2.199241      0.1264844
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif -0.08512778 0.7025276      0.9266443          5.2495812       5.725139      5.1430504
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif -0.08124686 0.5948883      0.8836179         -3.2182432      -2.983162     -3.3315364
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif -0.09356289 0.6809942      0.9204686          3.5419528       4.486732      3.3742086
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB
## 1                                NA                       0                    0                  0.0                       1.7273369
## 2                        -1.1699250                      30                   22                 49.5                      -1.3859026
## 3                                NA                       0                    0                  9.5                      -2.5541902
## 4                         0.1772545                     311                  333                294.5                       1.7486195
## 5                                NA                       0                    0                  0.0                      -2.2347229
## 6                        -0.4280937                      64                   55                 74.0                       0.2295656
##   Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1              61.34783         106.666667           32.21429                    21.74                 22.22                 21.43
## 2              50.65217          25.555556           66.78571                    82.61                 88.89                 78.57
## 3              19.39130           4.888889           28.71429                    43.48                 33.33                 50.00
## 4             997.26087        1742.111111          518.42857                    65.22                 55.56                 71.43
## 5              92.26087          28.333333          133.35714                    21.74                 22.22                 21.43
## 6             163.17391         179.222222          152.85714                    73.91                 77.78                 71.43
##   Odds Ratio (95% CI)
## 1    0.67 (-0.12;1.5)
## 2       4.3 (7.1;1.4)
## 3           13 (18;8)
## 4     0.39 (-1.4;2.2)
## 5       1.5 (2.3;0.7)
## 6     0.87 (0.59;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Median CLR (All)/(group AA)/(group BB)</strong>: Median CLR (normalization by <strong>ALDEx2</strong>) in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>ALDEx2</strong>.</p>
<details>
<summary>
<strong>run_da() pattern</strong>
</summary>
<p>We also provide another function <code>run_da</code> to run ALDEx2 differential analysis.</p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="differential-analysis.html#cb202-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb202-2"><a href="differential-analysis.html#cb202-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb202-3"><a href="differential-analysis.html#cb202-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb202-4"><a href="differential-analysis.html#cb202-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb202-5"><a href="differential-analysis.html#cb202-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;aldex&quot;</span>,</span>
<span id="cb202-6"><a href="differential-analysis.html#cb202-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="differential-analysis.html#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>otu_table or sample_table as inputdata</strong>
</summary>
<p>We also provide <em>data_otu</em> and <em>data_sam</em> as input data to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="differential-analysis.html#cb206-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb206-2"><a href="differential-analysis.html#cb206-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb206-3"><a href="differential-analysis.html#cb206-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb206-4"><a href="differential-analysis.html#cb206-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb206-5"><a href="differential-analysis.html#cb206-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb206-6"><a href="differential-analysis.html#cb206-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="differential-analysis.html#cb208-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>taxa_level option</strong>
</summary>
<p>We also provide <em>taxa_level</em> for choosing the specific taxonomic level to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="differential-analysis.html#cb210-1" aria-hidden="true" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb210-2"><a href="differential-analysis.html#cb210-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb210-3"><a href="differential-analysis.html#cb210-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb210-4"><a href="differential-analysis.html#cb210-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb210-5"><a href="differential-analysis.html#cb210-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb210-6"><a href="differential-analysis.html#cb210-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="differential-analysis.html#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
</div>
<div id="limma_voom" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> limma_voom<a href="differential-analysis.html#limma_voom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>limma</strong> package is from <em>voom: Precision weights unlock linear model analysis tools for RNA-seq read counts</em> <span class="citation">(<a href="#ref-law2014voom" role="doc-biblioref">Law et al. 2014</a>)</span>. Firstly, transforming count data to log2-counts per million (logCPM), estimate the mean-variance relationship and use this to compute appropriate observation-level weights. Secondly, fitting multiple linear models by weighted or generalized least squares. Finally, performing empirical bayes statistics for differential expression.</p>
<p><code>run_limma_voom</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_limma_voom)</code>.</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="differential-analysis.html#cb214-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb214-2"><a href="differential-analysis.html#cb214-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb214-3"><a href="differential-analysis.html#cb214-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb214-4"><a href="differential-analysis.html#cb214-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb214-5"><a href="differential-analysis.html#cb214-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-6"><a href="differential-analysis.html#cb214-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="differential-analysis.html#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment EffectSize     logFC       Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif   5.039869  5.039869 0.0008818997     0.08201667                                NA
## 2             g__Intestinibacter 9_AA vs 14_BB  Nonsignif   4.630693  4.630693 0.0019116736     0.08889282                                NA
## 3                  g__Romboutsia 9_AA vs 14_BB  Nonsignif   4.683036  4.683036 0.0040405228     0.12525621                         -6.499846
## 4     g__Ruminococcaceae_UCG_004 9_AA vs 14_BB  Nonsignif  -3.917121 -3.917121 0.0205760827     0.39783421                                NA
## 5               g__Lactobacillus 9_AA vs 14_BB  Nonsignif   3.814631  3.814631 0.0245721281     0.39783421                         -2.870821
## 6                 g__Veillonella 9_AA vs 14_BB  Nonsignif   3.431369  3.431369 0.0256667235     0.39783421                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1                       8                    0                 38.0                      -4.9962795             257.65217           13.00000
## 2                      12                    0                 63.0                      -2.1133078              82.34783           27.22222
## 3                     105                    3                271.5                      -3.1976766             295.47826           49.44444
## 4                      11                  107                  0.0                       2.2111652              66.91304          128.00000
## 5                      57                   27                197.5                      -5.0678928             799.95652           38.44444
## 6                      11                    0                 39.5                      -0.6689452              64.78261           47.66667
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1          414.92857                    60.87                 22.22                 85.71 18000 (18000;18000)
## 2          117.78571                    60.87                 22.22                 85.71       4.1 (6.8;1.3)
## 3          453.64286                    78.26                 55.56                 92.86          42 (49;34)
## 4           27.64286                    52.17                 77.78                 35.71     0.19 (-3.1;3.5)
## 5         1289.50000                    86.96                 77.78                 92.86 4e+08 (4e+08;4e+08)
## 6           75.78571                    56.52                 22.22                 78.57      1.3 (1.8;0.78)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>limma</strong>;</p></li>
<li><p><strong>logFC</strong>: LogFC from groups’ coefficient by <strong>limma</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>limma</strong>;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>limma</strong>.</p>
<details>
<summary>
<strong>other options for limma-voom</strong>
</summary>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="differential-analysis.html#cb218-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb218-2"><a href="differential-analysis.html#cb218-2" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb218-3"><a href="differential-analysis.html#cb218-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb218-4"><a href="differential-analysis.html#cb218-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb218-5"><a href="differential-analysis.html#cb218-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb218-6"><a href="differential-analysis.html#cb218-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb218-7"><a href="differential-analysis.html#cb218-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb218-8"><a href="differential-analysis.html#cb218-8" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb218-9"><a href="differential-analysis.html#cb218-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb218-10"><a href="differential-analysis.html#cb218-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb218-11"><a href="differential-analysis.html#cb218-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb218-12"><a href="differential-analysis.html#cb218-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb218-13"><a href="differential-analysis.html#cb218-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb218-14"><a href="differential-analysis.html#cb218-14" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb218-15"><a href="differential-analysis.html#cb218-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb218-16"><a href="differential-analysis.html#cb218-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb218-17"><a href="differential-analysis.html#cb218-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb218-18"><a href="differential-analysis.html#cb218-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb218-19"><a href="differential-analysis.html#cb218-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span>
<span id="cb218-20"><a href="differential-analysis.html#cb218-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="mbzinb" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> mbzinb<a href="differential-analysis.html#mbzinb" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>mbzinb</strong> package is from <em>An omnibus test for differential distribution analysis of microbiome sequencing data</em> <span class="citation">(<a href="#ref-chen2018omnibus" role="doc-biblioref">J. Chen et al. 2018</a>)</span>. It uses zeroinflated negative binomial model to investigate the significant taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="differential-analysis.html#cb219-1" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb219-2"><a href="differential-analysis.html#cb219-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb219-3"><a href="differential-analysis.html#cb219-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb219-4"><a href="differential-analysis.html#cb219-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb219-5"><a href="differential-analysis.html#cb219-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;base.mean&quot;                         &quot;mean.LFC&quot;                          &quot;base.abund&quot;                       
## [10] &quot;abund.LFC&quot;                         &quot;base.prev&quot;                         &quot;prev.change&quot;                      
## [13] &quot;base.disp&quot;                         &quot;disp.LFC&quot;                          &quot;statistic&quot;                        
## [16] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [19] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [22] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [25] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="differential-analysis.html#cb221-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment      Pvalue AdjustedPvalue EffectSize base.mean  mean.LFC base.abund  abund.LFC
## 1               g__Lactobacillus 9_AA vs 14_BB  Nonsignif 0.001876914      0.1734434  2.2347562 34.555899 4.5326842   44.15652  4.2457644
## 2                 g__Veillonella 9_AA vs 14_BB  Nonsignif 0.006397971      0.1734434  0.4281936 40.130920 0.6403763  180.53474 -1.1962972
## 3         g__Ruminiclostridium_5 9_AA vs 14_BB  Nonsignif 0.006833079      0.1734434  0.1425052 30.514058 0.8502009   30.51406  1.0720082
## 4 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif 0.007459931      0.1734434  1.6292120  9.268194 4.7644216   39.25916  2.6817529
## 5                  g__Romboutsia 9_AA vs 14_BB  Nonsignif 0.014857191      0.2166607  2.2429224 36.999013 3.3902509   63.81539  2.6937370
## 6             g__Intestinibacter 9_AA vs 14_BB  Nonsignif 0.016342731      0.2166607  1.3235433 19.454775 2.1732995   80.49455  0.3325049
##   base.prev prev.change  base.disp    disp.LFC statistic Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA
## 1 0.7825775   0.1721927 0.71177509  1.56437765  14.93061                        -2.8708213                      57                   27
## 2 0.2222892   0.5716955 0.06947861  3.77254666  12.30842                                NA                      11                    0
## 3 1.0000000  -0.1425095 0.11868487  1.91428035  12.16680                        -0.3504972                      46                   40
## 4 0.2360773   0.7639226 1.44644880  1.22973258  11.97770                                NA                       8                    0
## 5 0.5797819   0.3598035 1.40281769 -0.09845963  10.48585                        -6.4998459                     105                    3
## 6 0.2416906   0.6240627 2.05512513 -1.10172058  10.27830                                NA                      12                    0
##   Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                197.5                      -5.0678928             799.95652           38.44444         1289.50000                    86.96
## 2                 39.5                      -0.6689452              64.78261           47.66667           75.78571                    56.52
## 3                 51.0                      -0.7826083              51.78261           36.00000           61.92857                    91.30
## 4                 38.0                      -4.9962795             257.65217           13.00000          414.92857                    60.87
## 5                271.5                      -3.1976766             295.47826           49.44444          453.64286                    78.26
## 6                 63.0                      -2.1133078              82.34783           27.22222          117.78571                    60.87
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 77.78                 92.86 4e+08 (4e+08;4e+08)
## 2                 22.22                 78.57      1.3 (1.8;0.78)
## 3                100.00                 85.71        2.3 (4;0.67)
## 4                 22.22                 85.71 18000 (18000;18000)
## 5                 55.56                 92.86          42 (49;34)
## 6                 22.22                 85.71       4.1 (6.8;1.3)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>mbzinb</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>mbzinb</strong>;</p></li>
<li><p><strong>base.mean</strong>: fitted mean abundance parameter times the fitted prevalence in baseline group;</p></li>
<li><p><strong>mean.LFC</strong>: log2-fold change in fitted mean between other group and baseline;</p></li>
<li><p><strong>base.abund</strong>: fitted mean abundance parameter in baseline group;</p></li>
<li><p><strong>abund.LFC</strong>: log2-fold change in fitted mean abundance parameter between other group and baseline;</p></li>
<li><p><strong>base.prev</strong>: fitted prevalence in baseline group;</p></li>
<li><p><strong>prev.change</strong>: (linear) difference in prevalence between baseline group and other group (other-baseline);</p></li>
<li><p><strong>base.disp</strong>: fitted dispersion parameter in baseline group;</p></li>
<li><p><strong>disp.LFC</strong>: log2-fold change in fitted dispersion parameter between other group and baseline;</p></li>
<li><p><strong>statistic</strong>: value of likelihood ratio test statistic;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>mbzinb</strong>.</p>
<details>
<summary>
<strong>other options for mbzinb</strong>
</summary>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="differential-analysis.html#cb223-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb223-2"><a href="differential-analysis.html#cb223-2" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb223-3"><a href="differential-analysis.html#cb223-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb223-4"><a href="differential-analysis.html#cb223-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb223-5"><a href="differential-analysis.html#cb223-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb223-6"><a href="differential-analysis.html#cb223-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb223-7"><a href="differential-analysis.html#cb223-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb223-8"><a href="differential-analysis.html#cb223-8" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb223-9"><a href="differential-analysis.html#cb223-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb223-10"><a href="differential-analysis.html#cb223-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb223-11"><a href="differential-analysis.html#cb223-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb223-12"><a href="differential-analysis.html#cb223-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb223-13"><a href="differential-analysis.html#cb223-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb223-14"><a href="differential-analysis.html#cb223-14" aria-hidden="true" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb223-15"><a href="differential-analysis.html#cb223-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb223-16"><a href="differential-analysis.html#cb223-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb223-17"><a href="differential-analysis.html#cb223-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb223-18"><a href="differential-analysis.html#cb223-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb223-19"><a href="differential-analysis.html#cb223-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span>
<span id="cb223-20"><a href="differential-analysis.html#cb223-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="omnibus" class="section level3 hasAnchor" number="11.1.4">
<h3><span class="header-section-number">11.1.4</span> omnibus<a href="differential-analysis.html#omnibus" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This approach is also from <strong>mbzinb</strong> <span class="citation">(<a href="#ref-chen2018omnibus" role="doc-biblioref">J. Chen et al. 2018</a>)</span> package. it uses GMPR (Geometric Mean of Pairwise Ratios) <span class="citation">(<a href="#ref-chen2018gmpr" role="doc-biblioref">L. Chen et al. 2018</a>)</span> to get the size factors.</p>
<blockquote>
<p>where we specify models for count (abundance), zero (prevalence) and dispersion part. We also provide likelihood ratio test (zinb.lrt) for different models.</p>
</blockquote>
<p>(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="differential-analysis.html#cb224-1" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb224-2"><a href="differential-analysis.html#cb224-2" aria-hidden="true" tabindex="-1"></a>                     dada2_ps_genus_filter_trim,</span>
<span id="cb224-3"><a href="differential-analysis.html#cb224-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb224-4"><a href="differential-analysis.html#cb224-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb224-5"><a href="differential-analysis.html#cb224-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span></code></pre></div>
<pre><code>## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  86  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="differential-analysis.html#cb226-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;chi.stat&quot;                          &quot;df&quot;                                &quot;abund.baseline&quot;                   
## [10] &quot;prev.baseline&quot;                     &quot;dispersion.baseline&quot;               &quot;abund.LFC.CompvarBB.est&quot;          
## [13] &quot;abund.LFC.CompvarBB.se&quot;            &quot;prev.LOD.CompvarBB.est&quot;            &quot;prev.LOD.CompvarBB.se&quot;            
## [16] &quot;dispersion.LFC.CompvarBB.est&quot;      &quot;dispersion.LFC.CompvarBB.se&quot;       &quot;method&quot;                           
## [19] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [22] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [25] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [28] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="differential-analysis.html#cb228-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment     Pvalue AdjustedPvalue EffectSize  chi.stat df abund.baseline prev.baseline
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.22044477      0.5761264 0.02964464 4.4101643  3   0.0047263983     0.3333878
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.16457464      0.5443623 0.63796683 5.1006364  3   0.0006924418     0.8899455
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.03703889      0.2938543 0.58663534 8.4816439  3   0.0003466130     0.3333494
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.11546264      0.5332403 0.18791168 5.9220944  3   0.0554835223     0.5555649
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.86835069      0.9282789 0.07859332 0.7205897  3   0.0029468707     0.2364082
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.97678464      0.9840791 0.19518106 0.2049600  3   0.0049326257     0.7815153
##   dispersion.baseline abund.LFC.CompvarBB.est abund.LFC.CompvarBB.se prev.LOD.CompvarBB.est prev.LOD.CompvarBB.se
## 1           0.1508104             -0.27128370              1.5189155             -0.6063788             0.9613322
## 2           2.7108690              0.75537052              0.3280475             -0.7684155             1.2507437
## 3          20.6804819              1.33735990              0.3331999              0.6951556             0.8863890
## 4           1.6635277             -1.15702512              0.4707010              0.6963793             0.8947143
## 5           0.5130542             -0.81430941              1.7372127              2.3640357             1.0076124
## 6           1.0119195              0.02310437              0.4724806             -0.3545974             1.0006635
##   dispersion.LFC.CompvarBB.est dispersion.LFC.CompvarBB.se  method Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1                    3.1746543                   1.1289196 omnibus                                NA                       0
## 2                   -0.5133984                   0.6657207 omnibus                        -1.1699250                      30
## 3                   -2.3070151                   1.9880746 omnibus                                NA                       0
## 4                   -0.5207238                   0.7022562 omnibus                         0.1772545                     311
## 5                   -2.4466987                   1.0591662 omnibus                                NA                       0
## 6                    0.1907014                   0.6278565 omnibus                        -0.4280937                      64
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1                    0                  0.0                       1.7273369              61.34783         106.666667           32.21429
## 2                   22                 49.5                      -1.3859026              50.65217          25.555556           66.78571
## 3                    0                  9.5                      -2.5541902              19.39130           4.888889           28.71429
## 4                  333                294.5                       1.7486195             997.26087        1742.111111          518.42857
## 5                    0                  0.0                      -2.2347229              92.26087          28.333333          133.35714
## 6                   55                 74.0                       0.2295656             163.17391         179.222222          152.85714
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                    21.74                 22.22                 21.43    0.67 (-0.12;1.5)
## 2                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 3                    43.48                 33.33                 50.00           13 (18;8)
## 4                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                    21.74                 22.22                 21.43       1.5 (2.3;0.7)
## 6                    73.91                 77.78                 71.43     0.87 (0.59;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>omnibus</strong>;</p></li>
<li><p><strong>chi.stat</strong>: chisquare statistics;</p></li>
<li><p><strong>df</strong>: degree freedom;</p></li>
<li><p><strong>abund.baseline</strong>: mean abundance in baseline group;</p></li>
<li><p><strong>prev.baseline</strong>: prevalence in baseline group;</p></li>
<li><p><strong>dispersion.baseline</strong>: dispersion in baseline group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.est</strong>: log2-fold change in abundance between group BB and other group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.se</strong>: log2-fold change of standard errors in abundance between group BB and other group;</p></li>
<li><p><strong>prev.LOD.CompvarBB.est</strong>: prevalence of low of detect value in group BB;</p></li>
<li><p><strong>prev.LOD.CompvarBB.se</strong>: prevalence’s standard errors of low of detect value in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.est</strong>: log2-fold change in dispersion in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.se</strong>: log2-fold change in dispersion’s standard errors in group BB;</p></li>
<li><p><strong>method</strong>: methods of test;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>omnibus</strong>.</p>
<details>
<summary>
<strong>other options for omnibus</strong>
</summary>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="differential-analysis.html#cb230-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb230-2"><a href="differential-analysis.html#cb230-2" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb230-3"><a href="differential-analysis.html#cb230-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb230-4"><a href="differential-analysis.html#cb230-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb230-5"><a href="differential-analysis.html#cb230-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb230-6"><a href="differential-analysis.html#cb230-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb230-7"><a href="differential-analysis.html#cb230-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb230-8"><a href="differential-analysis.html#cb230-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb230-9"><a href="differential-analysis.html#cb230-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb230-10"><a href="differential-analysis.html#cb230-10" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb230-11"><a href="differential-analysis.html#cb230-11" aria-hidden="true" tabindex="-1"></a>                     dada2_ps,</span>
<span id="cb230-12"><a href="differential-analysis.html#cb230-12" aria-hidden="true" tabindex="-1"></a>                     <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb230-13"><a href="differential-analysis.html#cb230-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb230-14"><a href="differential-analysis.html#cb230-14" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb230-15"><a href="differential-analysis.html#cb230-15" aria-hidden="true" tabindex="-1"></a>                     <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb230-16"><a href="differential-analysis.html#cb230-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb230-17"><a href="differential-analysis.html#cb230-17" aria-hidden="true" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb230-18"><a href="differential-analysis.html#cb230-18" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb230-19"><a href="differential-analysis.html#cb230-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb230-20"><a href="differential-analysis.html#cb230-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb230-21"><a href="differential-analysis.html#cb230-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;omnibus&quot;</span>,</span>
<span id="cb230-22"><a href="differential-analysis.html#cb230-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb230-23"><a href="differential-analysis.html#cb230-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb230-24"><a href="differential-analysis.html#cb230-24" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span>
<span id="cb230-25"><a href="differential-analysis.html#cb230-25" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="raida" class="section level3 hasAnchor" number="11.1.5">
<h3><span class="header-section-number">11.1.5</span> RAIDA<a href="differential-analysis.html#raida" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>RAIDA</strong> package is from <em>A robust approach for identifying differentially abundant features in metagenomic samples</em> <span class="citation">(<a href="#ref-sohn2015robust" role="doc-biblioref">Sohn, Du, and An 2015</a>)</span>. It uses Ratio Approach for Identifying Differential Abundance (RAIDA). (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="differential-analysis.html#cb231-1" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb231-2"><a href="differential-analysis.html#cb231-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb231-3"><a href="differential-analysis.html#cb231-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb231-4"><a href="differential-analysis.html#cb231-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb231-5"><a href="differential-analysis.html#cb231-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb231-6"><a href="differential-analysis.html#cb231-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;eta_AA&quot;                            &quot;mean_AA&quot;                           &quot;sd_AA&quot;                            
## [10] &quot;eta_BB&quot;                            &quot;mean_BB&quot;                           &quot;sd_BB&quot;                            
## [13] &quot;mod.pool.var&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [16] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [19] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [22] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [25] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="differential-analysis.html#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##                             TaxaID         Block Enrichment EffectSize     Pvalue AdjustedPvalue    eta_AA   mean_AA     sd_AA    eta_BB
## 1                   g__Megasphaera 9_AA vs 14_BB  Nonsignif  0.7959956 0.00693536      0.5257476 0.6666667  1.055214 0.6712661 0.6428571
## 2              g__Faecalibacterium 9_AA vs 14_BB  Nonsignif  0.5748034 0.02246283      0.5257476 0.3333333  1.314901 0.5082491 0.2857143
## 3                   g__Bacteroides 9_AA vs 14_BB  Nonsignif  1.1075400 0.03732132      0.5257476 0.0000000  1.444747 1.2175042 0.0000000
## 4          g__Escherichia.Shigella 9_AA vs 14_BB  Nonsignif  1.1418407 0.03889020      0.5257476 0.1111111 -1.116098 1.7445986 0.1428571
## 5       g__Lachnospiraceae_UCG_010 9_AA vs 14_BB  Nonsignif  0.7300738 0.03912676      0.5257476 0.4444444 -3.202051 0.6342112 0.7142857
## 6 g__Christensenellaceae_R_7_group 9_AA vs 14_BB  Nonsignif  0.7496467 0.04486392      0.5257476 0.6666667 -2.187407 0.5655608 0.6428571
##       mean_BB     sd_BB mod.pool.var Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1 -1.77913964 1.3971812    1.4083996                                NA                       0                    0                  0.0
## 2 -0.03025548 1.1895440    1.1058278                          2.373851                    1377                 4512                870.5
## 3  0.00266215 1.8327408    2.3717931                          1.552035                    3806                 9118               3109.5
## 4 -2.80211637 1.8709056    2.8534784                          2.261867                      70                  247                 51.5
## 5 -4.59866370 0.5906816    0.8191908                                NA                       0                   34                  0.0
## 6 -4.47615788 1.8934855    1.9459993                                NA                       0                    0                  0.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA
## 1                       1.7193715             333.91304          579.55556          176.00000                    34.78                 33.33
## 2                       1.4344092            2678.78261         4345.00000         1607.64286                    69.57                 66.67
## 3                       0.9814940            5740.60870         8205.77778         4155.85714                   100.00                100.00
## 4                       0.4374399             574.26087          683.00000          504.35714                    86.96                 88.89
## 5                       2.2719126              18.73913           36.22222            7.50000                    39.13                 55.56
## 6                       1.7281282              35.65217           62.00000           18.71429                    34.78                 33.33
##   Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 35.71    0.55 (-0.64;1.7)
## 2                 71.43     0.37 (-1.6;2.3)
## 3                100.00       0.45 (-1.1;2)
## 4                 85.71     0.85 (0.53;1.2)
## 5                 28.57     0.33 (-1.8;2.5)
## 6                 35.71    0.56 (-0.59;1.7)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>RAIDA</strong>;</p></li>
<li><p><strong>eta_AA</strong>: vector containing estimated probabilities of the false zero state for group AA;</p></li>
<li><p><strong>mean_AA</strong>: vector containing estimated means of log ratios for group AA;</p></li>
<li><p><strong>sd_AA</strong>: vector containing estimated standard deviations of log ratios for group AA;</p></li>
<li><p><strong>eta_BB</strong>: vector containing estimated probabilities of the false zero state for group BB;</p></li>
<li><p><strong>mean_BB</strong>: vector containing estimated means of log ratios for group BB;</p></li>
<li><p><strong>sd_BB</strong>: vector containing estimated standard deviations of log ratios for group BB;</p></li>
<li><p><strong>mod.pool.var</strong>: vector containing estimated posterior variances of log ratios;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>RAIDA</strong>.</p>
<details>
<summary>
<strong>other options for RAIDA</strong>
</summary>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="differential-analysis.html#cb235-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb235-2"><a href="differential-analysis.html#cb235-2" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb235-3"><a href="differential-analysis.html#cb235-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb235-4"><a href="differential-analysis.html#cb235-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb235-5"><a href="differential-analysis.html#cb235-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb235-6"><a href="differential-analysis.html#cb235-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb235-7"><a href="differential-analysis.html#cb235-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-8"><a href="differential-analysis.html#cb235-8" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb235-9"><a href="differential-analysis.html#cb235-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb235-10"><a href="differential-analysis.html#cb235-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb235-11"><a href="differential-analysis.html#cb235-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb235-12"><a href="differential-analysis.html#cb235-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb235-13"><a href="differential-analysis.html#cb235-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-14"><a href="differential-analysis.html#cb235-14" aria-hidden="true" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb235-15"><a href="differential-analysis.html#cb235-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb235-16"><a href="differential-analysis.html#cb235-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb235-17"><a href="differential-analysis.html#cb235-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb235-18"><a href="differential-analysis.html#cb235-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;raida&quot;</span>)</span>
<span id="cb235-19"><a href="differential-analysis.html#cb235-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-20"><a href="differential-analysis.html#cb235-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span>
<span id="cb235-21"><a href="differential-analysis.html#cb235-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="wilcoxon-rank-sum-and-signed-rank-tests" class="section level3 hasAnchor" number="11.1.6">
<h3><span class="header-section-number">11.1.6</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparametric test methods, use the rank of taxa abundance to find the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="differential-analysis.html#cb236-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb236-2"><a href="differential-analysis.html#cb236-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb236-3"><a href="differential-analysis.html#cb236-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb236-4"><a href="differential-analysis.html#cb236-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb236-5"><a href="differential-analysis.html#cb236-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb236-6"><a href="differential-analysis.html#cb236-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="differential-analysis.html#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.02964464      63.5 1.0000000      1.0000000                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.63796683      45.0 0.2689844      0.7238145                        -1.1699250
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.58663534      44.0 0.1981520      0.7238145                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.18791168      71.0 0.6293971      0.9212826                         0.1772545
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.07859332      64.5 0.9304707      0.9833383                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.19518106      67.0 0.8239507      0.9487727                        -0.4280937
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1                       0                    0                  0.0                      0.01201252                   12.06
## 2                      30                   22                 49.5                     -0.41034110                   10.00
## 3                       0                    0                  9.5                     -0.43387758                    9.89
## 4                     311                  333                294.5                      0.17342686                   12.89
## 5                       0                    0                  0.0                      0.03358045                   12.17
## 6                      64                   55                 74.0                      0.08724541                   12.44
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   11.96                    21.74                 22.22                 21.43    0.67 (-0.12;1.5)
## 2                   13.29                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 3                   13.36                    43.48                 33.33                 50.00           13 (18;8)
## 4                   11.43                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                   11.89                    21.74                 22.22                 21.43       1.5 (2.3;0.7)
## 6                   11.71                    73.91                 77.78                 71.43     0.87 (0.59;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean Rank)_vs_BB**: Log2FoldChange (Mean Rank Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Rank Abundance (All)/(group AA)/(group BB)</strong>: Mean Rank Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for wilcox</strong>
</summary>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="differential-analysis.html#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb240-2"><a href="differential-analysis.html#cb240-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb240-3"><a href="differential-analysis.html#cb240-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb240-4"><a href="differential-analysis.html#cb240-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb240-5"><a href="differential-analysis.html#cb240-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb240-6"><a href="differential-analysis.html#cb240-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb240-7"><a href="differential-analysis.html#cb240-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb240-8"><a href="differential-analysis.html#cb240-8" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb240-9"><a href="differential-analysis.html#cb240-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb240-10"><a href="differential-analysis.html#cb240-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb240-11"><a href="differential-analysis.html#cb240-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb240-12"><a href="differential-analysis.html#cb240-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb240-13"><a href="differential-analysis.html#cb240-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb240-14"><a href="differential-analysis.html#cb240-14" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb240-15"><a href="differential-analysis.html#cb240-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb240-16"><a href="differential-analysis.html#cb240-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb240-17"><a href="differential-analysis.html#cb240-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb240-18"><a href="differential-analysis.html#cb240-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb240-19"><a href="differential-analysis.html#cb240-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb240-20"><a href="differential-analysis.html#cb240-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span>
<span id="cb240-21"><a href="differential-analysis.html#cb240-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="differential-analysis.html#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summary</span></span>
<span id="cb241-2"><a href="differential-analysis.html#cb241-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(dada2_ps_genus_filter_trim)</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 32373&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 49344&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 1007994&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 43825.8260869565&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 45587&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.49802371541502&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 0&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 1&quot;
## 
## [[11]]
## [1] &quot;Group&quot;</code></pre>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="differential-analysis.html#cb243-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run</span></span>
<span id="cb243-2"><a href="differential-analysis.html#cb243-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb243-3"><a href="differential-analysis.html#cb243-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb243-4"><a href="differential-analysis.html#cb243-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb243-5"><a href="differential-analysis.html#cb243-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb243-6"><a href="differential-analysis.html#cb243-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb243-7"><a href="differential-analysis.html#cb243-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_rarefy</strong>
</summary>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="differential-analysis.html#cb245-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb245-2"><a href="differential-analysis.html#cb245-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb245-3"><a href="differential-analysis.html#cb245-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb245-4"><a href="differential-analysis.html#cb245-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb245-5"><a href="differential-analysis.html#cb245-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb245-6"><a href="differential-analysis.html#cb245-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb245-7"><a href="differential-analysis.html#cb245-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb245-8"><a href="differential-analysis.html#cb245-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb245-9"><a href="differential-analysis.html#cb245-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_rarefy)</span>
<span id="cb245-10"><a href="differential-analysis.html#cb245-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_CLR</strong>: centered log-ratio normalization.</li>
</ul>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="differential-analysis.html#cb246-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb246-2"><a href="differential-analysis.html#cb246-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb246-3"><a href="differential-analysis.html#cb246-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb246-4"><a href="differential-analysis.html#cb246-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb246-5"><a href="differential-analysis.html#cb246-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb246-6"><a href="differential-analysis.html#cb246-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb246-7"><a href="differential-analysis.html#cb246-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_CLR)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_CLR</strong>
</summary>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="differential-analysis.html#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb248-2"><a href="differential-analysis.html#cb248-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb248-3"><a href="differential-analysis.html#cb248-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb248-4"><a href="differential-analysis.html#cb248-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb248-5"><a href="differential-analysis.html#cb248-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb248-6"><a href="differential-analysis.html#cb248-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb248-7"><a href="differential-analysis.html#cb248-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb248-8"><a href="differential-analysis.html#cb248-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-9"><a href="differential-analysis.html#cb248-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_CLR)</span>
<span id="cb248-10"><a href="differential-analysis.html#cb248-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse" class="section level3 hasAnchor" number="11.1.7">
<h3><span class="header-section-number">11.1.7</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="differential-analysis.html#cb249-1" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb249-2"><a href="differential-analysis.html#cb249-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb249-3"><a href="differential-analysis.html#cb249-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb249-4"><a href="differential-analysis.html#cb249-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb249-5"><a href="differential-analysis.html#cb249-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb249-6"><a href="differential-analysis.html#cb249-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;LDA_Score&quot;                         &quot;EffectSize&quot;                        &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
##  [7] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [10] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [13] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [16] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="differential-analysis.html#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1              g__Parasutterella 9_AA vs 14_BB         AA -3.688638   2.120078                          5.109407               203.81424
## 2     g__Ruminococcaceae_UCG_004 9_AA vs 14_BB         AA -2.840224   2.031956                                NA               241.29686
## 3                 g__Odoribacter 9_AA vs 14_BB         BB  1.672435   1.732109                                NA                65.80823
## 4             g__Intestinibacter 9_AA vs 14_BB         BB  3.033100   2.288011                                NA               328.68608
## 5 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB         BB  3.531594   2.652245                                NA               163.77669
## 6                  g__Romboutsia 9_AA vs 14_BB         BB  3.668216   2.901361                         -6.323595              2149.56906
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1           1059.27623             30.68489                        4.358465             3829.5975          9096.9057           443.4708
## 2           2522.09782              0.00000                        2.270464             1476.6749          2853.6733           591.4617
## 3            615.67509              0.00000                        2.106533              580.3768          1089.6104           253.0123
## 4              0.00000           1528.00388                       -1.923849             1856.3117           687.2929          2607.8238
## 5              0.00000           1130.36401                       -4.853394             5867.8516           326.2193          9430.3295
## 6             75.54011           6050.19693                       -3.401631             7079.2882          1037.4100         10963.3528
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                    65.22                 88.89                 50.00     0.0043 (-11;11)
## 2                    52.17                 77.78                 35.71     0.16 (-3.4;3.7)
## 3                    52.17                 77.78                 35.71       0.31 (-2;2.6)
## 4                    60.87                 22.22                 85.71      3.2 (5.6;0.94)
## 5                    60.87                 22.22                 85.71    8700 (8700;8700)
## 6                    78.26                 55.56                 92.86          90 (99;81)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>lefse</em>;</p></li>
<li><p><strong>LDA_Score</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for lefse</strong>
</summary>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="differential-analysis.html#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb253-2"><a href="differential-analysis.html#cb253-2" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb253-3"><a href="differential-analysis.html#cb253-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb253-4"><a href="differential-analysis.html#cb253-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb253-5"><a href="differential-analysis.html#cb253-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb253-6"><a href="differential-analysis.html#cb253-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb253-7"><a href="differential-analysis.html#cb253-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb253-8"><a href="differential-analysis.html#cb253-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-9"><a href="differential-analysis.html#cb253-9" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb253-10"><a href="differential-analysis.html#cb253-10" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb253-11"><a href="differential-analysis.html#cb253-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb253-12"><a href="differential-analysis.html#cb253-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb253-13"><a href="differential-analysis.html#cb253-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb253-14"><a href="differential-analysis.html#cb253-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb253-15"><a href="differential-analysis.html#cb253-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-16"><a href="differential-analysis.html#cb253-16" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb253-17"><a href="differential-analysis.html#cb253-17" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb253-18"><a href="differential-analysis.html#cb253-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb253-19"><a href="differential-analysis.html#cb253-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb253-20"><a href="differential-analysis.html#cb253-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb253-21"><a href="differential-analysis.html#cb253-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb253-22"><a href="differential-analysis.html#cb253-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb253-23"><a href="differential-analysis.html#cb253-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span>
<span id="cb253-24"><a href="differential-analysis.html#cb253-24" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test" class="section level3 hasAnchor" number="11.1.8">
<h3><span class="header-section-number">11.1.8</span> t-test<a href="differential-analysis.html#t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="differential-analysis.html#cb254-1" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb254-2"><a href="differential-analysis.html#cb254-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb254-3"><a href="differential-analysis.html#cb254-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb254-4"><a href="differential-analysis.html#cb254-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb254-5"><a href="differential-analysis.html#cb254-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-6"><a href="differential-analysis.html#cb254-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                    &quot;Enrichment&quot;                              
##  [4] &quot;EffectSize&quot;                               &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;        &quot;Median Abundance\n(All)&quot;                 
## [10] &quot;Median Abundance\nAA&quot;                     &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;              &quot;GeometricMean Abundance\nBB&quot;             
## [16] &quot;Occurrence (100%)\n(All)&quot;                 &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="differential-analysis.html#cb256-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.02964464  0.6885335 0.50934060      0.7584236                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.63796683 -1.9509301 0.06966436      0.6428197                        -1.1699250
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.58663534 -2.2520991 0.04004033      0.6428197                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.18791168  1.4025255 0.19444323      0.6428197                         0.1772545
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.07859332 -0.7746013 0.45141826      0.7490259                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.19518106  0.3094918 0.76107100      0.9176314                        -0.4280937
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (geometricmean)\nAA_vs_BB GeometricMean Abundance\n(All)
## 1                       0                    0                  0.0                                2.7295288                      1.2436945
## 2                      30                   22                 49.5                               -5.2268338                      0.5117863
## 3                       0                    0                  9.5                               -3.1311631                      0.5218980
## 4                     311                  333                294.5                                1.4704215                      0.7835288
## 5                       0                    0                  0.0                               -3.5525954                      1.3278400
## 6                      64                   55                 74.0                               -0.6655791                      0.7002247
##   GeometricMean Abundance\nAA GeometricMean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1                  4.39029547                   0.6619477                    21.74                 22.22                 21.43
## 2                  0.02149483                   0.8049481                    82.61                 88.89                 78.57
## 3                  0.08122039                   0.7116052                    43.48                 33.33                 50.00
## 4                  1.10053019                   0.3971559                    65.22                 55.56                 71.43
## 5                  0.38763845                   4.5484625                    21.74                 22.22                 21.43
## 6                  0.54190943                   0.8595794                    73.91                 77.78                 71.43
##   Odds Ratio (95% CI)
## 1    0.67 (-0.12;1.5)
## 2       4.3 (7.1;1.4)
## 3           13 (18;8)
## 4     0.39 (-1.4;2.2)
## 5       1.5 (2.3;0.7)
## 6     0.87 (0.59;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (GeometricMean)_vs_BB**: Log2FoldChange (GeometricMean Abundance) between group AA and group BB;</p></li>
<li><p><strong>GeometricMean Abundance (All)/(group AA)/(group BB)</strong>: GeometricMean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for t-test</strong>
</summary>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="differential-analysis.html#cb258-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb258-2"><a href="differential-analysis.html#cb258-2" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb258-3"><a href="differential-analysis.html#cb258-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb258-4"><a href="differential-analysis.html#cb258-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb258-5"><a href="differential-analysis.html#cb258-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb258-6"><a href="differential-analysis.html#cb258-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb258-7"><a href="differential-analysis.html#cb258-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-8"><a href="differential-analysis.html#cb258-8" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb258-9"><a href="differential-analysis.html#cb258-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb258-10"><a href="differential-analysis.html#cb258-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb258-11"><a href="differential-analysis.html#cb258-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb258-12"><a href="differential-analysis.html#cb258-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb258-13"><a href="differential-analysis.html#cb258-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-14"><a href="differential-analysis.html#cb258-14" aria-hidden="true" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb258-15"><a href="differential-analysis.html#cb258-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb258-16"><a href="differential-analysis.html#cb258-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb258-17"><a href="differential-analysis.html#cb258-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb258-18"><a href="differential-analysis.html#cb258-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb258-19"><a href="differential-analysis.html#cb258-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb258-20"><a href="differential-analysis.html#cb258-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span>
<span id="cb258-21"><a href="differential-analysis.html#cb258-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>t-test_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="differential-analysis.html#cb259-1" aria-hidden="true" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb259-2"><a href="differential-analysis.html#cb259-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb259-3"><a href="differential-analysis.html#cb259-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb259-4"><a href="differential-analysis.html#cb259-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb259-5"><a href="differential-analysis.html#cb259-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb259-6"><a href="differential-analysis.html#cb259-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                    &quot;Enrichment&quot;                              
##  [4] &quot;EffectSize&quot;                               &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;        &quot;Median Abundance\n(All)&quot;                 
## [10] &quot;Median Abundance\nAA&quot;                     &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;              &quot;GeometricMean Abundance\nBB&quot;             
## [16] &quot;Occurrence (100%)\n(All)&quot;                 &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for ttest_rarefy</strong>
</summary>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="differential-analysis.html#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb261-2"><a href="differential-analysis.html#cb261-2" aria-hidden="true" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb261-3"><a href="differential-analysis.html#cb261-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb261-4"><a href="differential-analysis.html#cb261-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb261-5"><a href="differential-analysis.html#cb261-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb261-6"><a href="differential-analysis.html#cb261-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>,</span>
<span id="cb261-7"><a href="differential-analysis.html#cb261-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb261-8"><a href="differential-analysis.html#cb261-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb261-9"><a href="differential-analysis.html#cb261-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_rarefy)</span>
<span id="cb261-10"><a href="differential-analysis.html#cb261-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="metagenomeseq" class="section level3 hasAnchor" number="11.1.9">
<h3><span class="header-section-number">11.1.9</span> MetagenomeSeq<a href="differential-analysis.html#metagenomeseq" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>MetagenomeSeq</strong> package is from <em>Differential abundance analysis for microbial marker-gene surveys</em> <span class="citation">(<a href="#ref-paulson2013differential" role="doc-biblioref">Paulson et al. 2013</a>)</span>. It uses zero-inflated Log-Normal mixture model or Zero-inflated Gaussian mixture model to identify the significant taxa between groups. (Caution: the <strong>otu_table</strong> should be integers).</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="differential-analysis.html#cb262-1" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb262-2"><a href="differential-analysis.html#cb262-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb262-3"><a href="differential-analysis.html#cb262-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb262-4"><a href="differential-analysis.html#cb262-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb262-5"><a href="differential-analysis.html#cb262-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb262-6"><a href="differential-analysis.html#cb262-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb262-7"><a href="differential-analysis.html#cb262-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb262-8"><a href="differential-analysis.html#cb262-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;logFC&quot;                             &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;se&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="differential-analysis.html#cb264-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment      logFC      Pvalue AdjustedPvalue        se Log2FoldChange (Median)\nAA_vs_BB
## 1                  g__Romboutsia 9_AA vs 14_BB  Nonsignif  1.5554520 0.002340957      0.2200500 0.5111279                         -5.213950
## 2                 g__Megasphaera 9_AA vs 14_BB  Nonsignif -1.7566352 0.008960092      0.4046775 0.6721202                                NA
## 3             g__Intestinibacter 9_AA vs 14_BB  Nonsignif  1.2174655 0.012915240      0.4046775 0.4897094                                NA
## 4 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif  1.4564056 0.018516204      0.4351308 0.6183970                                NA
## 5                 g__Haemophilus 9_AA vs 14_BB  Nonsignif -1.3341212 0.060764773      0.8279049 0.7114542                                NA
## 6     g__Ruminococcaceae_UCG_013 9_AA vs 14_BB  Nonsignif  0.7367506 0.086962790      0.8279049 0.4304352                         -2.792157
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1               96.153846             7.672634            284.77369                       -4.157644             423.30698          37.610729
## 2                0.000000             0.000000              0.00000                        3.234514             882.94339        1936.380922
## 3               19.834711             0.000000             65.21018                       -3.310524             101.85987          15.840523
## 4                7.326007             0.000000             41.02910                       -6.241005             361.15734           7.778412
## 5                0.000000             0.000000              0.00000                        1.729433              88.76226         154.404353
## 6               45.197740            10.662605             73.85609                       -1.702861             121.28674          51.113551
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1          671.25457                    78.26                 55.56                 92.86          5300 (5300;5300)
## 2          205.73355                    34.78                 33.33                 35.71           0.29 (-2.1;2.7)
## 3          157.15802                    60.87                 22.22                 85.71                44 (51;36)
## 4          588.32950                    60.87                 22.22                 85.71 3.5e+11 (3.5e+11;3.5e+11)
## 5           46.56378                    34.78                 22.22                 42.86          0.63 (-0.28;1.5)
## 6          166.39808                    60.87                 55.56                 64.29               3.4 (5.9;1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>se</strong>: standard error;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for metagenomeseq</strong>
</summary>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="differential-analysis.html#cb266-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb266-2"><a href="differential-analysis.html#cb266-2" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb266-3"><a href="differential-analysis.html#cb266-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb266-4"><a href="differential-analysis.html#cb266-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb266-5"><a href="differential-analysis.html#cb266-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb266-6"><a href="differential-analysis.html#cb266-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb266-7"><a href="differential-analysis.html#cb266-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb266-8"><a href="differential-analysis.html#cb266-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb266-9"><a href="differential-analysis.html#cb266-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-10"><a href="differential-analysis.html#cb266-10" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb266-11"><a href="differential-analysis.html#cb266-11" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb266-12"><a href="differential-analysis.html#cb266-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb266-13"><a href="differential-analysis.html#cb266-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb266-14"><a href="differential-analysis.html#cb266-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb266-15"><a href="differential-analysis.html#cb266-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb266-16"><a href="differential-analysis.html#cb266-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb266-17"><a href="differential-analysis.html#cb266-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-18"><a href="differential-analysis.html#cb266-18" aria-hidden="true" tabindex="-1"></a>DA_metagenomeseq<span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb266-19"><a href="differential-analysis.html#cb266-19" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb266-20"><a href="differential-analysis.html#cb266-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb266-21"><a href="differential-analysis.html#cb266-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb266-22"><a href="differential-analysis.html#cb266-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;metagenomeseq&quot;</span>,</span>
<span id="cb266-23"><a href="differential-analysis.html#cb266-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb266-24"><a href="differential-analysis.html#cb266-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb266-25"><a href="differential-analysis.html#cb266-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-26"><a href="differential-analysis.html#cb266-26" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span>
<span id="cb266-27"><a href="differential-analysis.html#cb266-27" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="deseq2" class="section level3 hasAnchor" number="11.1.10">
<h3><span class="header-section-number">11.1.10</span> DESeq2<a href="differential-analysis.html#deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>DESeq2</strong> package is from <em>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</em> <span class="citation">(<a href="#ref-love2014moderated" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution.(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="differential-analysis.html#cb267-1" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb267-2"><a href="differential-analysis.html#cb267-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb267-3"><a href="differential-analysis.html#cb267-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb267-4"><a href="differential-analysis.html#cb267-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb267-5"><a href="differential-analysis.html#cb267-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-6"><a href="differential-analysis.html#cb267-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;logFC&quot;                            
##  [7] &quot;Statistic&quot;                         &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="differential-analysis.html#cb269-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment      Pvalue AdjustedPvalue      logFC  Statistic Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.001257944     0.01169888  3.4914753  3.2254054                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.100115783     0.30034735  1.4934480  1.6442926                        -1.1699250
## 3   g__Adlercreutzia 9_AA vs 14_BB         BB 0.006191868     0.04115470  2.5088703  2.7374439                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.451586460     0.72409553 -0.8376048 -0.7527728                         0.1772545
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.534169949     0.76698059 -0.4163455 -0.6216532                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.702520607     0.84849891 -0.3532118 -0.3819201                        -0.4280937
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1                       0                    0                  0.0                       1.7273369              61.34783         106.666667
## 2                      30                   22                 49.5                      -1.3859026              50.65217          25.555556
## 3                       0                    0                  9.5                      -2.5541902              19.39130           4.888889
## 4                     311                  333                294.5                       1.7486195             997.26087        1742.111111
## 5                       0                    0                  0.0                      -2.2347229              92.26087          28.333333
## 6                      64                   55                 74.0                       0.2295656             163.17391         179.222222
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1           32.21429                    21.74                 22.22                 21.43    0.67 (-0.12;1.5)
## 2           66.78571                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 3           28.71429                    43.48                 33.33                 50.00           13 (18;8)
## 4          518.42857                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5          133.35714                    21.74                 22.22                 21.43       1.5 (2.3;0.7)
## 6          152.85714                    73.91                 77.78                 71.43     0.87 (0.59;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>Statistic</strong>: test statistic (negative binomial model);</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for deseq2</strong>
</summary>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="differential-analysis.html#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb271-2"><a href="differential-analysis.html#cb271-2" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb271-3"><a href="differential-analysis.html#cb271-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb271-4"><a href="differential-analysis.html#cb271-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb271-5"><a href="differential-analysis.html#cb271-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb271-6"><a href="differential-analysis.html#cb271-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb271-7"><a href="differential-analysis.html#cb271-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-8"><a href="differential-analysis.html#cb271-8" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb271-9"><a href="differential-analysis.html#cb271-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb271-10"><a href="differential-analysis.html#cb271-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb271-11"><a href="differential-analysis.html#cb271-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb271-12"><a href="differential-analysis.html#cb271-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb271-13"><a href="differential-analysis.html#cb271-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-14"><a href="differential-analysis.html#cb271-14" aria-hidden="true" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb271-15"><a href="differential-analysis.html#cb271-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb271-16"><a href="differential-analysis.html#cb271-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb271-17"><a href="differential-analysis.html#cb271-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb271-18"><a href="differential-analysis.html#cb271-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;deseq2&quot;</span>)</span>
<span id="cb271-19"><a href="differential-analysis.html#cb271-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb271-20"><a href="differential-analysis.html#cb271-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span>
<span id="cb271-21"><a href="differential-analysis.html#cb271-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="edger" class="section level3 hasAnchor" number="11.1.11">
<h3><span class="header-section-number">11.1.11</span> EdgeR<a href="differential-analysis.html#edger" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>EdgeR</strong> package is from <em>A scaling normalization method for differential expression analysis of RNA-seq data</em> <span class="citation">(<a href="#ref-robinson2010scaling" role="doc-biblioref">Robinson and Oshlack 2010</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="differential-analysis.html#cb272-1" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb272-2"><a href="differential-analysis.html#cb272-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb272-3"><a href="differential-analysis.html#cb272-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb272-4"><a href="differential-analysis.html#cb272-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-139"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-139-1.png" alt="EdgeR (BVC distance)" width="480" />
<p class="caption">
Figure 11.1: EdgeR (BVC distance)
</p>
</div>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="differential-analysis.html#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_edger)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjPval&quot;                           &quot;logFC&quot;                            
##  [7] &quot;logCPM&quot;                            &quot;LR&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
## [10] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [13] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [16] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [19] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="differential-analysis.html#cb275-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment       Pvalue     AdjPval     logFC    logCPM        LR
## 1 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB         BB 2.174311e-05 0.002022109  6.772483 13.328302 18.030175
## 2               g__Lactobacillus 9_AA vs 14_BB         BB 4.759473e-05 0.002213155  6.973542 16.257029 16.541585
## 3                 g__Eubacterium 9_AA vs 14_BB  Nonsignif 1.022250e-04 0.003168974 -4.200333  9.414079 15.095162
## 4                  g__Romboutsia 9_AA vs 14_BB         BB 1.517094e-04 0.003527243  5.072407 13.660834 14.350873
## 5              g__Parasutterella 9_AA vs 14_BB         AA 1.555264e-03 0.028927913 -3.265304 11.079025 10.011966
## 6                  g__Holdemania 9_AA vs 14_BB  Nonsignif 2.280315e-03 0.031400097 -2.355665  7.031037  9.308995
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB
## 1                                NA                       8                    0                 38.0                       -4.996279
## 2                         -2.870821                      57                   27                197.5                       -5.067893
## 3                                NA                       0                    0                  0.0                        2.508697
## 4                         -6.499846                     105                    3                271.5                       -3.197677
## 5                          5.029747                       7                   49                  1.5                        4.244838
## 6                                NA                       0                    0                  0.0                        3.052467
##   Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1            257.652174          13.000000        414.9285714                    60.87                 22.22                 85.71
## 2            799.956522          38.444444       1289.5000000                    86.96                 77.78                 92.86
## 3              8.304348          16.666667          2.9285714                    30.43                 44.44                 21.43
## 4            295.478261          49.444444        453.6428571                    78.26                 55.56                 92.86
## 5            164.565217         388.666667         20.5000000                    65.22                 88.89                 50.00
## 6              3.304348           7.111111          0.8571429                    26.09                 44.44                 14.29
##   Odds Ratio (95% CI)
## 1 18000 (18000;18000)
## 2 4e+08 (4e+08;4e+08)
## 3     0.33 (-1.8;2.5)
## 4          42 (49;34)
## 5         0.01 (-9;9)
## 6       0.14 (-3.7;4)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>logCPM</strong>: is the average expression of all samples for that particular gene across all samples on the log-scale expressed in counts per million (cpm, as calculated by edgeR after normalization);</p></li>
<li><p><strong>LR</strong>: the signed likelihood ratio test statistic;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for EdgeR</strong>
</summary>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="differential-analysis.html#cb277-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb277-2"><a href="differential-analysis.html#cb277-2" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb277-3"><a href="differential-analysis.html#cb277-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb277-4"><a href="differential-analysis.html#cb277-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb277-5"><a href="differential-analysis.html#cb277-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb277-6"><a href="differential-analysis.html#cb277-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb277-7"><a href="differential-analysis.html#cb277-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-8"><a href="differential-analysis.html#cb277-8" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb277-9"><a href="differential-analysis.html#cb277-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb277-10"><a href="differential-analysis.html#cb277-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb277-11"><a href="differential-analysis.html#cb277-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb277-12"><a href="differential-analysis.html#cb277-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb277-13"><a href="differential-analysis.html#cb277-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-14"><a href="differential-analysis.html#cb277-14" aria-hidden="true" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb277-15"><a href="differential-analysis.html#cb277-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb277-16"><a href="differential-analysis.html#cb277-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb277-17"><a href="differential-analysis.html#cb277-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb277-18"><a href="differential-analysis.html#cb277-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;edger&quot;</span>)</span>
<span id="cb277-19"><a href="differential-analysis.html#cb277-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-20"><a href="differential-analysis.html#cb277-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span>
<span id="cb277-21"><a href="differential-analysis.html#cb277-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="ancom" class="section level3 hasAnchor" number="11.1.12">
<h3><span class="header-section-number">11.1.12</span> ANCOM<a href="differential-analysis.html#ancom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ANCOM</strong> (Analysis of composition of microbiomes) is from <em>Analysis of composition of microbiomes: a novel method for studying microbial composition”, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-mandal2015analysis" role="doc-biblioref">Mandal et al. 2015</a>)</span>. ANCOM makes no distributional assumptions and can be implemented in a linear model framework. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="differential-analysis.html#cb278-1" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb278-2"><a href="differential-analysis.html#cb278-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb278-3"><a href="differential-analysis.html#cb278-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb278-4"><a href="differential-analysis.html#cb278-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-141"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-141-1.png" alt="ANCOM (Structure Zero)" width="480" />
<p class="caption">
Figure 11.2: ANCOM (Structure Zero)
</p>
</div>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="differential-analysis.html#cb279-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;(W)q-values &lt; alpha&quot;               &quot;W_ratio&quot;                          
##  [7] &quot;detected_0.7&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="differential-analysis.html#cb281-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment   EffectSize (W)q-values &lt; alpha W_ratio detected_0.7 Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB         BB  0.056059308                   0       0        FALSE                                NA
## 2     g__Actinomyces 9_AA vs 14_BB         BB  0.733680102                   0       0        FALSE                        -1.1699250
## 3   g__Adlercreutzia 9_AA vs 14_BB         BB  0.303739656                   0       0        FALSE                                NA
## 4    g__Agathobacter 9_AA vs 14_BB         AA -0.504122195                   0       0        FALSE                         0.1772545
## 5     g__Akkermansia 9_AA vs 14_BB         AA -0.070529304                   0       0        FALSE                                NA
## 6       g__Alistipes 9_AA vs 14_BB         BB  0.003048187                   0       0        FALSE                        -0.4280937
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1                       0                    0                  0.0                       1.7273369              61.34783         106.666667
## 2                      30                   22                 49.5                      -1.3859026              50.65217          25.555556
## 3                       0                    0                  9.5                      -2.5541902              19.39130           4.888889
## 4                     311                  333                294.5                       1.7486195             997.26087        1742.111111
## 5                       0                    0                  0.0                      -2.2347229              92.26087          28.333333
## 6                      64                   55                 74.0                       0.2295656             163.17391         179.222222
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1           32.21429                    21.74                 22.22                 21.43    0.67 (-0.12;1.5)
## 2           66.78571                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 3           28.71429                    43.48                 33.33                 50.00           13 (18;8)
## 4          518.42857                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5          133.35714                    21.74                 22.22                 21.43       1.5 (2.3;0.7)
## 6          152.85714                    73.91                 77.78                 71.43     0.87 (0.59;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size by <strong>ANCOM</strong>;</p></li>
<li><p><strong>(W)q-values &lt; alpha</strong>: q-values less than alpha;</p></li>
<li><p><strong>W_ratio</strong>: the ratio of W values;</p></li>
<li><p><strong>detected_0.7</strong>: W_ratio more than 0.7;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for ANCOM</strong>
</summary>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="differential-analysis.html#cb283-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb283-2"><a href="differential-analysis.html#cb283-2" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb283-3"><a href="differential-analysis.html#cb283-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb283-4"><a href="differential-analysis.html#cb283-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb283-5"><a href="differential-analysis.html#cb283-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-6"><a href="differential-analysis.html#cb283-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb283-7"><a href="differential-analysis.html#cb283-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-8"><a href="differential-analysis.html#cb283-8" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb283-9"><a href="differential-analysis.html#cb283-9" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb283-10"><a href="differential-analysis.html#cb283-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb283-11"><a href="differential-analysis.html#cb283-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-12"><a href="differential-analysis.html#cb283-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb283-13"><a href="differential-analysis.html#cb283-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-14"><a href="differential-analysis.html#cb283-14" aria-hidden="true" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb283-15"><a href="differential-analysis.html#cb283-15" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb283-16"><a href="differential-analysis.html#cb283-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-17"><a href="differential-analysis.html#cb283-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb283-18"><a href="differential-analysis.html#cb283-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ancom&quot;</span>)</span>
<span id="cb283-19"><a href="differential-analysis.html#cb283-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb283-20"><a href="differential-analysis.html#cb283-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb283-21"><a href="differential-analysis.html#cb283-21" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="corncob" class="section level3 hasAnchor" number="11.1.13">
<h3><span class="header-section-number">11.1.13</span> Corncob<a href="differential-analysis.html#corncob" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Corncob</strong> package is from <em>Modeling microbial abundances and dysbiosis with beta-binomial regression”, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-martin2020modeling" role="doc-biblioref">Martin, Witten, and Willis 2020</a>)</span>. Corncob is based on beta-binomial regression. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="differential-analysis.html#cb284-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb284-2"><a href="differential-analysis.html#cb284-2" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb284-3"><a href="differential-analysis.html#cb284-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb284-4"><a href="differential-analysis.html#cb284-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb284-5"><a href="differential-analysis.html#cb284-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb284-6"><a href="differential-analysis.html#cb284-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb284-7"><a href="differential-analysis.html#cb284-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-8"><a href="differential-analysis.html#cb284-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span>
<span id="cb284-9"><a href="differential-analysis.html#cb284-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb284-10"><a href="differential-analysis.html#cb284-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>other options for Corncob</strong>
</summary>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="differential-analysis.html#cb285-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb285-2"><a href="differential-analysis.html#cb285-2" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb285-3"><a href="differential-analysis.html#cb285-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb285-4"><a href="differential-analysis.html#cb285-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb285-5"><a href="differential-analysis.html#cb285-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb285-6"><a href="differential-analysis.html#cb285-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb285-7"><a href="differential-analysis.html#cb285-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb285-8"><a href="differential-analysis.html#cb285-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb285-9"><a href="differential-analysis.html#cb285-9" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb285-10"><a href="differential-analysis.html#cb285-10" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb285-11"><a href="differential-analysis.html#cb285-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb285-12"><a href="differential-analysis.html#cb285-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb285-13"><a href="differential-analysis.html#cb285-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb285-14"><a href="differential-analysis.html#cb285-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb285-15"><a href="differential-analysis.html#cb285-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb285-16"><a href="differential-analysis.html#cb285-16" aria-hidden="true" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb285-17"><a href="differential-analysis.html#cb285-17" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb285-18"><a href="differential-analysis.html#cb285-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb285-19"><a href="differential-analysis.html#cb285-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb285-20"><a href="differential-analysis.html#cb285-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;corncob&quot;</span>,</span>
<span id="cb285-21"><a href="differential-analysis.html#cb285-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb285-22"><a href="differential-analysis.html#cb285-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb285-23"><a href="differential-analysis.html#cb285-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_corncob)</span>
<span id="cb285-24"><a href="differential-analysis.html#cb285-24" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models" class="section level3 hasAnchor" number="11.1.14">
<h3><span class="header-section-number">11.1.14</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="differential-analysis.html#cb286-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb286-2"><a href="differential-analysis.html#cb286-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb286-3"><a href="differential-analysis.html#cb286-3" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb286-4"><a href="differential-analysis.html#cb286-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb286-5"><a href="differential-analysis.html#cb286-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb286-6"><a href="differential-analysis.html#cb286-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb286-7"><a href="differential-analysis.html#cb286-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb286-8"><a href="differential-analysis.html#cb286-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb286-9"><a href="differential-analysis.html#cb286-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb286-10"><a href="differential-analysis.html#cb286-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-11"><a href="differential-analysis.html#cb286-11" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb286-12"><a href="differential-analysis.html#cb286-12" aria-hidden="true" tabindex="-1"></a>                    dada2_ps,</span>
<span id="cb286-13"><a href="differential-analysis.html#cb286-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb286-14"><a href="differential-analysis.html#cb286-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb286-15"><a href="differential-analysis.html#cb286-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb286-16"><a href="differential-analysis.html#cb286-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb286-17"><a href="differential-analysis.html#cb286-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb286-18"><a href="differential-analysis.html#cb286-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb286-19"><a href="differential-analysis.html#cb286-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb286-20"><a href="differential-analysis.html#cb286-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-21"><a href="differential-analysis.html#cb286-21" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb286-22"><a href="differential-analysis.html#cb286-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb286-23"><a href="differential-analysis.html#cb286-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb286-24"><a href="differential-analysis.html#cb286-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb286-25"><a href="differential-analysis.html#cb286-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb286-26"><a href="differential-analysis.html#cb286-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb286-27"><a href="differential-analysis.html#cb286-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb286-28"><a href="differential-analysis.html#cb286-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb286-29"><a href="differential-analysis.html#cb286-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb286-30"><a href="differential-analysis.html#cb286-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-31"><a href="differential-analysis.html#cb286-31" aria-hidden="true" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb286-32"><a href="differential-analysis.html#cb286-32" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb286-33"><a href="differential-analysis.html#cb286-33" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb286-34"><a href="differential-analysis.html#cb286-34" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb286-35"><a href="differential-analysis.html#cb286-35" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb286-36"><a href="differential-analysis.html#cb286-36" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb286-37"><a href="differential-analysis.html#cb286-37" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb286-38"><a href="differential-analysis.html#cb286-38" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb286-39"><a href="differential-analysis.html#cb286-39" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb286-40"><a href="differential-analysis.html#cb286-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-41"><a href="differential-analysis.html#cb286-41" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2)</span>
<span id="cb286-42"><a href="differential-analysis.html#cb286-42" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
<div id="metagenomic-sequencing-microbial-data-metaphlan23" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Metagenomic sequencing microbial data (metaphlan2/3)<a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the same strategy to filter phyloseq object.</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="differential-analysis.html#cb287-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;metaphlan2_ps&quot;</span>)</span>
<span id="cb287-2"><a href="differential-analysis.html#cb287-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-3"><a href="differential-analysis.html#cb287-3" aria-hidden="true" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb287-4"><a href="differential-analysis.html#cb287-4" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb287-5"><a href="differential-analysis.html#cb287-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb287-6"><a href="differential-analysis.html#cb287-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb287-7"><a href="differential-analysis.html#cb287-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb287-8"><a href="differential-analysis.html#cb287-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-9"><a href="differential-analysis.html#cb287-9" aria-hidden="true" tabindex="-1"></a><span class="co"># step2: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb287-10"><a href="differential-analysis.html#cb287-10" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps_remove_BRS, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb287-11"><a href="differential-analysis.html#cb287-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-12"><a href="differential-analysis.html#cb287-12" aria-hidden="true" tabindex="-1"></a><span class="co"># step3: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb287-13"><a href="differential-analysis.html#cb287-13" aria-hidden="true" tabindex="-1"></a><span class="co">#dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(dada2_ps_genus, cutoff = 10, unclass = TRUE)</span></span>
<span id="cb287-14"><a href="differential-analysis.html#cb287-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-15"><a href="differential-analysis.html#cb287-15" aria-hidden="true" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb287-16"><a href="differential-analysis.html#cb287-16" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(metaphlan2_ps_genus, <span class="at">cutoff =</span> <span class="fl">1e-4</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb287-17"><a href="differential-analysis.html#cb287-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-18"><a href="differential-analysis.html#cb287-18" aria-hidden="true" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb287-19"><a href="differential-analysis.html#cb287-19" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(metaphlan2_ps_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb287-20"><a href="differential-analysis.html#cb287-20" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 54 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 54 taxa by 6 taxonomic ranks ]</code></pre>
<div id="limma_voom-1" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> limma_voom<a href="differential-analysis.html#limma_voom-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>limma</strong> package is from <em>voom: Precision weights unlock linear model analysis tools for RNA-seq read counts</em> <span class="citation">(<a href="#ref-law2014voom" role="doc-biblioref">Law et al. 2014</a>)</span>. Firstly, transforming count data to log2-counts per million (logCPM), estimate the mean-variance relationship and use this to compute appropriate observation-level weights. Secondly, fitting multiple linear models by weighted or generalized least squares. Finally, performing empirical bayes statistics for differential expression.</p>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="differential-analysis.html#cb289-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_mgs <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb289-2"><a href="differential-analysis.html#cb289-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb289-3"><a href="differential-analysis.html#cb289-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb289-4"><a href="differential-analysis.html#cb289-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb289-5"><a href="differential-analysis.html#cb289-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="differential-analysis.html#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom_mgs)</span></code></pre></div>
<pre><code>##                TaxaID         Block Enrichment EffectSize     logFC       Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1     g__Ruminococcus 7_AA vs 15_BB         BB  0.2233696 0.2233696 1.548665e-07   3.894463e-06                         -2.906486
## 2 g__Faecalibacterium 7_AA vs 15_BB         BB  0.3063850 0.3063850 1.854506e-07   3.894463e-06                         -4.011562
## 3          g__Blautia 7_AA vs 15_BB         BB  0.1619875 0.1619875 1.187496e-05   1.662495e-04                         -3.182422
## 4    g__Lactobacillus 7_AA vs 15_BB         BB  0.1399117 0.1399117 5.287895e-05   5.552289e-04                                NA
## 5  g__Bifidobacterium 7_AA vs 15_BB         BB  0.3779862 0.3779862 9.644339e-05   8.101244e-04                         -6.549698
## 6        g__Roseburia 7_AA vs 15_BB         BB  0.1275366 0.1275366 1.728128e-04   1.209690e-03                         -4.442582
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA
## 1              0.02141570            0.0044236            0.0331677                       -2.892349            0.03578485       0.0066508000
## 2              0.05346880            0.0047826            0.0771373                       -3.992131            0.05750352       0.0051489857
## 3              0.02000425            0.0036769            0.0333800                       -2.265551            0.02431008       0.0067591429
## 4              0.00155690            0.0000000            0.0024249                       -5.399318            0.02118822       0.0007282714
## 5              0.03731980            0.0009090            0.0851568                       -3.394708            0.10730063       0.0143274857
## 6              0.00752330            0.0008393            0.0182502                       -2.451452            0.02013462       0.0049745857
##   Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1         0.04938074                    81.82                 57.14                 93.33             270 (280;260)
## 2         0.08193563                    77.27                 57.14                 86.67                75 (83;66)
## 3         0.03250052                   100.00                100.00                100.00                90 (99;81)
## 4         0.03073619                    72.73                 42.86                 86.67 5.3e+10 (5.3e+10;5.3e+10)
## 5         0.15068810                   100.00                100.00                100.00             130 (140;120)
## 6         0.02720931                    81.82                 57.14                 93.33               12 (17;7.2)</code></pre>
<details>
<summary>
<strong>limma_voom Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="differential-analysis.html#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb293-2"><a href="differential-analysis.html#cb293-2" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb293-3"><a href="differential-analysis.html#cb293-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb293-4"><a href="differential-analysis.html#cb293-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb293-5"><a href="differential-analysis.html#cb293-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb293-6"><a href="differential-analysis.html#cb293-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;limma_voom&quot;</span>)</span>
<span id="cb293-7"><a href="differential-analysis.html#cb293-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="wilcoxon-rank-sum-and-signed-rank-tests-1" class="section level3 hasAnchor" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparameter test methods, use the rank of taxa abundance to find the significant taxa.</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="differential-analysis.html#cb294-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb294-2"><a href="differential-analysis.html#cb294-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb294-3"><a href="differential-analysis.html#cb294-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb294-4"><a href="differential-analysis.html#cb294-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb294-5"><a href="differential-analysis.html#cb294-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="differential-analysis.html#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic      Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1       g__Alistipes 7_AA vs 15_BB  Nonsignif  1.7499838      71.0 0.201934650     0.53007846                          2.061604
## 2    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.2384009      31.0 0.138345605     0.38736769                         -4.964322
## 3   g__Anaerotruncus 7_AA vs 15_BB  Nonsignif  0.9439864      47.5 0.712887079     0.83170159                                NA
## 4     g__Bacteroides 7_AA vs 15_BB  Nonsignif  1.1135266      85.0 0.021284830     0.07947442                          1.229057
## 5 g__Bifidobacterium 7_AA vs 15_BB         BB  4.2421190      11.0 0.002122619     0.04028636                         -6.549698
## 6       g__Bilophila 7_AA vs 15_BB  Nonsignif  0.2514869      60.0 0.616070968     0.77874024                          1.177523
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1              0.00711780            0.0236665            0.0056693                       0.4613459                   14.14
## 2              0.00031435            0.0000229            0.0007149                      -0.6171177                    8.43
## 3              0.00000000            0.0000000            0.0000000                      -0.1327552                   10.79
## 4              0.24894650            0.4568864            0.1949060                       0.7906916                   16.14
## 5              0.03731980            0.0009090            0.0851568                      -1.3572361                    5.57
## 6              0.00104585            0.0020513            0.0009069                       0.1924811                   12.57
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   10.27                    77.27                 71.43                 80.00     0.22 (-2.7;3.2)
## 2                   12.93                    86.36                 71.43                 93.33         6 (9.5;2.5)
## 3                   11.83                    36.36                 28.57                 40.00      1.3 (1.8;0.78)
## 4                    9.33                   100.00                100.00                100.00       0.31 (-2;2.6)
## 5                   14.27                   100.00                100.00                100.00       130 (140;120)
## 6                   11.00                    68.18                 71.43                 66.67      0.4 (-1.4;2.2)</code></pre>
<details>
<summary>
<strong>wilcox Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="differential-analysis.html#cb298-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb298-2"><a href="differential-analysis.html#cb298-2" aria-hidden="true" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb298-3"><a href="differential-analysis.html#cb298-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb298-4"><a href="differential-analysis.html#cb298-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb298-5"><a href="differential-analysis.html#cb298-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb298-6"><a href="differential-analysis.html#cb298-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb298-7"><a href="differential-analysis.html#cb298-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb298-8"><a href="differential-analysis.html#cb298-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-1" class="section level3 hasAnchor" number="11.2.3">
<h3><span class="header-section-number">11.2.3</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="differential-analysis.html#cb299-1" aria-hidden="true" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb299-2"><a href="differential-analysis.html#cb299-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb299-3"><a href="differential-analysis.html#cb299-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb299-4"><a href="differential-analysis.html#cb299-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb299-5"><a href="differential-analysis.html#cb299-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)  </span>
<span id="cb299-6"><a href="differential-analysis.html#cb299-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb299-7"><a href="differential-analysis.html#cb299-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;LDA_Score&quot;                         &quot;EffectSize&quot;                        &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
##  [7] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [10] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [13] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [16] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="differential-analysis.html#cb301-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##              TaxaID         Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1    g__Bacteroides 7_AA vs 15_BB         AA -5.043523  2.4235602                          1.190750            254167.50879
## 2  g__Fusobacterium 7_AA vs 15_BB         AA -4.177621  1.9028644                                NA                 0.00000
## 3 g__Granulicatella 7_AA vs 15_BB         BB  3.027420  0.4271128                                NA                25.66255
## 4    g__Eggerthella 7_AA vs 15_BB         BB  3.250342  1.2650206                                NA               591.34240
## 5          g__Dorea 7_AA vs 15_BB         BB  3.843600  2.0751421                         -5.725323              8554.29279
## 6    g__Collinsella 7_AA vs 15_BB         BB  3.897707  2.4170225                                NA              7735.89039
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1          464614.8499         203536.16836                        1.141368           291673.9273       464989.41139       210793.36812
## 2             123.8325              0.00000                        6.941519             9891.9128        30556.16386          248.59562
## 3               0.0000             52.35042                       -3.071984               64.1451           10.59936           89.13311
## 4               0.0000            771.47121                       -3.764280             2231.9872          232.91482         3164.88766
## 5             270.5150          14311.50707                       -3.454344            11339.4984         1455.31524        15952.11728
## 6               0.0000          11641.72307                       -2.928116            12495.9911         2268.85954        17268.65252
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   100.00                100.00                100.00     0.32 (-1.9;2.5)
## 2                    40.91                 85.71                 20.00     0.0032 (-11;11)
## 3                    72.73                 42.86                 86.67       220 (230;210)
## 4                    63.64                 42.86                 73.33       430 (450;420)
## 5                    90.91                 71.43                100.00          28 (34;21)
## 6                    68.18                 42.86                 80.00          20 (25;14)</code></pre>
<details>
<summary>
<strong>lefse Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="differential-analysis.html#cb303-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb303-2"><a href="differential-analysis.html#cb303-2" aria-hidden="true" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb303-3"><a href="differential-analysis.html#cb303-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb303-4"><a href="differential-analysis.html#cb303-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb303-5"><a href="differential-analysis.html#cb303-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb303-6"><a href="differential-analysis.html#cb303-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb303-7"><a href="differential-analysis.html#cb303-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb303-8"><a href="differential-analysis.html#cb303-8" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test-1" class="section level3 hasAnchor" number="11.2.4">
<h3><span class="header-section-number">11.2.4</span> t-test<a href="differential-analysis.html#t-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="differential-analysis.html#cb304-1" aria-hidden="true" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb304-2"><a href="differential-analysis.html#cb304-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb304-3"><a href="differential-analysis.html#cb304-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb304-4"><a href="differential-analysis.html#cb304-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb304-5"><a href="differential-analysis.html#cb304-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                                   &quot;Block&quot;                                    &quot;Enrichment&quot;                              
##  [4] &quot;EffectSize&quot;                               &quot;Statistic&quot;                                &quot;Pvalue&quot;                                  
##  [7] &quot;AdjustedPvalue&quot;                           &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;        &quot;Median Abundance\n(All)&quot;                 
## [10] &quot;Median Abundance\nAA&quot;                     &quot;Median Abundance\nBB&quot;                     &quot;Log2FoldChange (geometricmean)\nAA_vs_BB&quot;
## [13] &quot;GeometricMean Abundance\n(All)&quot;           &quot;GeometricMean Abundance\nAA&quot;              &quot;GeometricMean Abundance\nBB&quot;             
## [16] &quot;Occurrence (100%)\n(All)&quot;                 &quot;Occurrence (100%)\nAA&quot;                    &quot;Occurrence (100%)\nBB&quot;                   
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="differential-analysis.html#cb306-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1       g__Alistipes 7_AA vs 15_BB  Nonsignif  1.7499838  1.5963681 0.15951558      0.4674316                          2.061604
## 2    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.2384009 -1.6970563 0.10937430      0.3828101                         -4.964322
## 3   g__Anaerotruncus 7_AA vs 15_BB  Nonsignif  0.9439864 -0.5871152 0.56369794      0.6398733                                NA
## 4     g__Bacteroides 7_AA vs 15_BB  Nonsignif  1.1135266  2.2882319 0.04545820      0.1909244                          1.229057
## 5 g__Bifidobacterium 7_AA vs 15_BB  Nonsignif  4.2421190 -2.7252006 0.01523564      0.1089140                         -6.549698
## 6       g__Bilophila 7_AA vs 15_BB  Nonsignif  0.2514869  1.2030139 0.27277490      0.5565431                          1.177523
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (geometricmean)\nAA_vs_BB GeometricMean Abundance\n(All)
## 1              0.00711780            0.0236665            0.0056693                               2.05528762                      0.3989446
## 2              0.00031435            0.0000229            0.0007149                                       NA                      1.8882218
## 3              0.00000000            0.0000000            0.0000000                              -0.05926392                      1.0212861
## 4              0.24894650            0.4568864            0.1949060                               1.60721398                      0.3871167
## 5              0.03731980            0.0009090            0.0851568                                       NA                      0.1936930
## 6              0.00104585            0.0020513            0.0009069                               3.17465522                      0.6988408
##   GeometricMean Abundance\nAA GeometricMean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1                   0.7346341                   0.1767534                    77.27                 71.43                 80.00
## 2                         NaN                   1.8882218                    86.36                 71.43                 93.33
## 3                   0.9903011                   1.0318284                    36.36                 28.57                 40.00
## 4                   0.6044631                   0.1984039                   100.00                100.00                100.00
## 5                         NaN                   0.1936930                   100.00                100.00                100.00
## 6                   2.6168456                   0.2898089                    68.18                 71.43                 66.67
##   Odds Ratio (95% CI)
## 1     0.22 (-2.7;3.2)
## 2         6 (9.5;2.5)
## 3      1.3 (1.8;0.78)
## 4       0.31 (-2;2.6)
## 5       130 (140;120)
## 6      0.4 (-1.4;2.2)</code></pre>
<details>
<summary>
<strong>ttest Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="differential-analysis.html#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb308-2"><a href="differential-analysis.html#cb308-2" aria-hidden="true" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb308-3"><a href="differential-analysis.html#cb308-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb308-4"><a href="differential-analysis.html#cb308-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb308-5"><a href="differential-analysis.html#cb308-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb308-6"><a href="differential-analysis.html#cb308-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb308-7"><a href="differential-analysis.html#cb308-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models-1" class="section level3 hasAnchor" number="11.2.5">
<h3><span class="header-section-number">11.2.5</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="differential-analysis.html#cb309-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb309-2"><a href="differential-analysis.html#cb309-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb309-3"><a href="differential-analysis.html#cb309-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb309-4"><a href="differential-analysis.html#cb309-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb309-5"><a href="differential-analysis.html#cb309-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb309-6"><a href="differential-analysis.html#cb309-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb309-7"><a href="differential-analysis.html#cb309-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb309-8"><a href="differential-analysis.html#cb309-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb309-9"><a href="differential-analysis.html#cb309-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb309-10"><a href="differential-analysis.html#cb309-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-11"><a href="differential-analysis.html#cb309-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2_mgs)</span>
<span id="cb309-12"><a href="differential-analysis.html#cb309-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>maaslin2 Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="differential-analysis.html#cb310-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb310-2"><a href="differential-analysis.html#cb310-2" aria-hidden="true" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb310-3"><a href="differential-analysis.html#cb310-3" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_genus_filter_trim,</span>
<span id="cb310-4"><a href="differential-analysis.html#cb310-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb310-5"><a href="differential-analysis.html#cb310-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb310-6"><a href="differential-analysis.html#cb310-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb310-7"><a href="differential-analysis.html#cb310-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb310-8"><a href="differential-analysis.html#cb310-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb310-9"><a href="differential-analysis.html#cb310-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb310-10"><a href="differential-analysis.html#cb310-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb310-11"><a href="differential-analysis.html#cb310-11" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
</div>
<div id="visualization-4" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Visualization<a href="differential-analysis.html#visualization-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Volcano plot is used to display differential analysis. <code>plot_volcano</code> provides multiple parameters for plotting volcano. More details to see <code>help(plot_volcano)</code>.</p>
<p>The barplot is used to display the lefse results.</p>
<div id="volcano-plot" class="section level3 hasAnchor" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Volcano plot<a href="differential-analysis.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The X and Y coordinate axis are flexible to choose for Volcano. Here, we choose <em>logFC</em> and <em>AdjustedPvalue</em> to visualize the results</p>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="differential-analysis.html#cb311-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb311-2"><a href="differential-analysis.html#cb311-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_genus_filter_trim,</span>
<span id="cb311-3"><a href="differential-analysis.html#cb311-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb311-4"><a href="differential-analysis.html#cb311-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb311-5"><a href="differential-analysis.html#cb311-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-6"><a href="differential-analysis.html#cb311-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="differential-analysis.html#cb313-1" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_volcano <span class="ot">&lt;-</span> <span class="fu">plot_volcano</span>(</span>
<span id="cb313-2"><a href="differential-analysis.html#cb313-2" aria-hidden="true" tabindex="-1"></a>    DA_limma_voom,</span>
<span id="cb313-3"><a href="differential-analysis.html#cb313-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb313-4"><a href="differential-analysis.html#cb313-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;logFC&quot;</span>,</span>
<span id="cb313-5"><a href="differential-analysis.html#cb313-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb313-6"><a href="differential-analysis.html#cb313-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb313-7"><a href="differential-analysis.html#cb313-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb313-8"><a href="differential-analysis.html#cb313-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb313-9"><a href="differential-analysis.html#cb313-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span>
<span id="cb313-10"><a href="differential-analysis.html#cb313-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb313-11"><a href="differential-analysis.html#cb313-11" aria-hidden="true" tabindex="-1"></a>DA_limma_voom_volcano</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-157"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-157-1.png" alt="Volcano (limma-voom: logFC and AdjustedPvalue)" width="672" />
<p class="caption">
Figure 11.3: Volcano (limma-voom: logFC and AdjustedPvalue)
</p>
</div>
<details>
<summary>
<strong>DA_limma_voom_volcano effectsize</strong>
</summary>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="differential-analysis.html#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb314-2"><a href="differential-analysis.html#cb314-2" aria-hidden="true" tabindex="-1"></a>    DA_limma_voom,</span>
<span id="cb314-3"><a href="differential-analysis.html#cb314-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb314-4"><a href="differential-analysis.html#cb314-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;EffectSize&quot;</span>,</span>
<span id="cb314-5"><a href="differential-analysis.html#cb314-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb314-6"><a href="differential-analysis.html#cb314-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb314-7"><a href="differential-analysis.html#cb314-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb314-8"><a href="differential-analysis.html#cb314-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb314-9"><a href="differential-analysis.html#cb314-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">8</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-158"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-158-1.png" alt="Volcano (limma-voom: EffectSize and Pvalue)" width="672" />
<p class="caption">
Figure 11.4: Volcano (limma-voom: EffectSize and Pvalue)
</p>
</div>
<p>The logFC and EffectSize are the same values in limma-voom.</p>
</details>
</div>
<div id="barplot-in-lefse" class="section level3 hasAnchor" number="11.3.2">
<h3><span class="header-section-number">11.3.2</span> barplot in lefse<a href="differential-analysis.html#barplot-in-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="differential-analysis.html#cb315-1" aria-hidden="true" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb315-2"><a href="differential-analysis.html#cb315-2" aria-hidden="true" tabindex="-1"></a>    dada2_ps_genus_filter_trim,</span>
<span id="cb315-3"><a href="differential-analysis.html#cb315-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb315-4"><a href="differential-analysis.html#cb315-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb315-5"><a href="differential-analysis.html#cb315-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb315-6"><a href="differential-analysis.html#cb315-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb315-7"><a href="differential-analysis.html#cb315-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb315-8"><a href="differential-analysis.html#cb315-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">da_res =</span> DA_lefse,</span>
<span id="cb315-9"><a href="differential-analysis.html#cb315-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb315-10"><a href="differential-analysis.html#cb315-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb315-11"><a href="differential-analysis.html#cb315-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>)) </span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-159"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-159-1.png" alt="Barplot (Lefse)" width="576" />
<p class="caption">
Figure 11.5: Barplot (Lefse)
</p>
</div>
</div>
</div>
<div id="dominant-taxa" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Dominant taxa<a href="differential-analysis.html#dominant-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Display the significant taxa with selection using boxplot.</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="differential-analysis.html#cb316-1" aria-hidden="true" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb316-2"><a href="differential-analysis.html#cb316-2" aria-hidden="true" tabindex="-1"></a>   metaphlan2_ps_genus_filter_trim,</span>
<span id="cb316-3"><a href="differential-analysis.html#cb316-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb316-4"><a href="differential-analysis.html#cb316-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb316-5"><a href="differential-analysis.html#cb316-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb316-6"><a href="differential-analysis.html#cb316-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_topN_boxplot</span>(</span>
<span id="cb316-7"><a href="differential-analysis.html#cb316-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb316-8"><a href="differential-analysis.html#cb316-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">da_res =</span> DA_wilcox,</span>
<span id="cb316-9"><a href="differential-analysis.html#cb316-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Median)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb316-10"><a href="differential-analysis.html#cb316-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.2</span>,</span>
<span id="cb316-11"><a href="differential-analysis.html#cb316-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb316-12"><a href="differential-analysis.html#cb316-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.3</span>,</span>
<span id="cb316-13"><a href="differential-analysis.html#cb316-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">3</span>,</span>
<span id="cb316-14"><a href="differential-analysis.html#cb316-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">taxa_name =</span> <span class="st">&quot;s__Ruminococcus_torques&quot;</span>,</span>
<span id="cb316-15"><a href="differential-analysis.html#cb316-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-160"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-160-1.png" alt="Dominant Taxa" width="768" />
<p class="caption">
Figure 11.6: Dominant Taxa
</p>
</div>
</div>
<div id="multiple-differential-analysis-by-one-function" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> Multiple differential analysis by one function<a href="differential-analysis.html#multiple-differential-analysis-by-one-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>here, we provide the <code>run_multiple_da</code> for obtaining the results list from multiple differential analysis methods.</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="differential-analysis.html#cb317-1" aria-hidden="true" tabindex="-1"></a>multiple_res <span class="ot">&lt;-</span> <span class="fu">run_multiple_da</span>(</span>
<span id="cb317-2"><a href="differential-analysis.html#cb317-2" aria-hidden="true" tabindex="-1"></a>       dada2_ps_genus_filter_trim,</span>
<span id="cb317-3"><a href="differential-analysis.html#cb317-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb317-4"><a href="differential-analysis.html#cb317-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb317-5"><a href="differential-analysis.html#cb317-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">da_method =</span> <span class="fu">c</span>(<span class="st">&quot;aldex&quot;</span>, <span class="st">&quot;limma_voom&quot;</span>, <span class="st">&quot;mbzinb&quot;</span>, <span class="st">&quot;omnibus&quot;</span>),</span>
<span id="cb317-6"><a href="differential-analysis.html#cb317-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">p_adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|
## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  86  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="differential-analysis.html#cb319-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(multiple_res)</span></code></pre></div>
<pre><code>## [1] &quot;aldex&quot;      &quot;limma_voom&quot; &quot;mbzinb&quot;     &quot;omnibus&quot;</code></pre>
<ul>
<li>plot results</li>
</ul>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="differential-analysis.html#cb321-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_multiple_DA</span>(</span>
<span id="cb321-2"><a href="differential-analysis.html#cb321-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Multip_DA_res =</span> multiple_res,</span>
<span id="cb321-3"><a href="differential-analysis.html#cb321-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_list =</span> <span class="fu">c</span>(<span class="st">&quot;EffectSize&quot;</span>, <span class="st">&quot;logFC&quot;</span>, <span class="st">&quot;mean.LFC&quot;</span>, <span class="st">&quot;abund.LFC.CompvarBB.est&quot;</span>),</span>
<span id="cb321-4"><a href="differential-analysis.html#cb321-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb321-5"><a href="differential-analysis.html#cb321-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_index =</span> <span class="st">&quot;AdjustedPvalue&quot;</span>,</span>
<span id="cb321-6"><a href="differential-analysis.html#cb321-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">y_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb321-7"><a href="differential-analysis.html#cb321-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellwidth =</span> <span class="dv">50</span>,</span>
<span id="cb321-8"><a href="differential-analysis.html#cb321-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellheight =</span> <span class="dv">10</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-162"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-162-1.png" alt="Multiple DA results" width="768" />
<p class="caption">
Figure 11.7: Multiple DA results
</p>
</div>
</div>
<div id="systematic-information-8" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> Systematic Information<a href="differential-analysis.html#systematic-information-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="differential-analysis.html#cb322-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] RColorBrewer_1.1-2          viridis_0.6.2               viridisLite_0.4.0           conflicted_1.1.0           
##  [5] SummarizedExperiment_1.24.0 Biobase_2.54.0              GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
##  [9] IRanges_2.28.0              S4Vectors_0.32.3            BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
## [13] matrixStats_0.61.0          ggpubr_0.4.0                ggplot2_3.3.5               phyloseq_1.38.0            
## [17] tibble_3.1.6                dplyr_1.0.8                 XMAS2_2.1.4                
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2             reticulate_1.24        tidyselect_1.1.2       RSQLite_2.2.10         AnnotationDbi_1.56.2  
##   [6] htmlwidgets_1.5.4      grid_4.1.2             BiocParallel_1.28.3    Rtsne_0.15             munsell_0.5.0         
##  [11] codetools_0.2-18       umap_0.2.8.0           DT_0.21                withr_2.5.0            colorspace_2.0-3      
##  [16] highr_0.9              knitr_1.37             rstudioapi_0.13        pscl_1.5.5             robustbase_0.93-9     
##  [21] bayesm_3.1-4           ggsignif_0.6.3         labeling_0.4.2         mbzinb_0.2             GenomeInfoDbData_1.2.7
##  [26] pheatmap_1.0.12        bit64_4.0.5            farver_2.1.0           rhdf5_2.38.1           vctrs_0.3.8           
##  [31] generics_0.1.2         TH.data_1.1-0          xfun_0.30              R6_2.5.1               locfit_1.5-9.5        
##  [36] RcppZiggurat_0.1.6     bitops_1.0-7           rhdf5filters_1.6.0     cachem_1.0.6           DelayedArray_0.20.0   
##  [41] assertthat_0.2.1       scales_1.1.1           multcomp_1.4-18        nnet_7.3-17            gtable_0.3.0          
##  [46] sandwich_3.0-1         rlang_1.0.2            genefilter_1.76.0      splines_4.1.2          protoclust_1.6.3      
##  [51] rstatix_0.7.0          broom_0.7.12           checkmate_2.0.0        yaml_2.3.5             reshape2_1.4.4        
##  [56] abind_1.4-5            crosstalk_1.2.0        backports_1.4.1        Rfast_2.0.6            qvalue_2.26.0         
##  [61] Hmisc_4.6-0            DiagrammeR_1.0.9       tensorA_0.36.2         tools_4.1.2            bookdown_0.24         
##  [66] zCompositions_1.4.0    ellipsis_0.3.2         gplots_3.1.1           jquerylib_0.1.4        biomformat_1.22.0     
##  [71] Rcpp_1.0.8.2           plyr_1.8.6             base64enc_0.1-3        visNetwork_2.1.0       zlibbioc_1.40.0       
##  [76] purrr_0.3.4            RCurl_1.98-1.6         rpart_4.1.16           openssl_2.0.0          Wrench_1.12.0         
##  [81] cowplot_1.1.1          zoo_1.8-9              ggrepel_0.9.1          cluster_2.1.2          magrittr_2.0.2        
##  [86] data.table_1.14.2      RSpectra_0.16-0        truncnorm_1.0-8        mvtnorm_1.1-3          ALDEx2_1.26.0         
##  [91] RAIDA_1.0              evaluate_0.15          xtable_1.8-4           XML_3.99-0.9           jpeg_0.1-9            
##  [96] gridExtra_2.3          shape_1.4.6            compiler_4.1.2         KernSmooth_2.23-20     crayon_1.5.0          
## [101] htmltools_0.5.2        mgcv_1.8-39            Formula_1.2-4          tidyr_1.2.0            geneplotter_1.72.0    
## [106] libcoin_1.0-9          DBI_1.1.2              corrplot_0.92          MASS_7.3-55            compositions_2.0-4    
## [111] Matrix_1.4-0           ade4_1.7-18            car_3.0-12             permute_0.9-7          cli_3.2.0             
## [116] parallel_4.1.2         igraph_1.2.11          pkgconfig_2.0.3        coin_1.4-2             foreign_0.8-82        
## [121] foreach_1.5.2          annotate_1.72.0        bslib_0.3.1            multtest_2.50.0        XVector_0.34.0        
## [126] NADA_1.6-1.1           stringr_1.4.0          digest_0.6.29          vegan_2.5-7            Biostrings_2.62.0     
## [131] rmarkdown_2.13         htmlTable_2.4.0        edgeR_3.36.0           gtools_3.9.2           modeltools_0.2-23     
## [136] lifecycle_1.0.1        nlme_3.1-155           jsonlite_1.8.0         Rhdf5lib_1.16.0        carData_3.0-5         
## [141] askpass_1.1            limma_3.50.1           fansi_1.0.2            pillar_1.7.0           lattice_0.20-45       
## [146] DEoptimR_1.0-10        KEGGREST_1.34.0        fastmap_1.1.0          httr_1.4.2             survival_3.3-1        
## [151] glue_1.6.2             png_0.1-7              iterators_1.0.14       glmnet_4.1-3           bit_4.0.4             
## [156] stringi_1.7.6          sass_0.4.0             metagenomeSeq_1.36.0   blob_1.2.2             DESeq2_1.34.0         
## [161] latticeExtra_0.6-29    caTools_1.18.2         memoise_2.0.1          ape_5.6-2</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chen2018omnibus" class="csl-entry">
Chen, Jun, Emily King, Rebecca Deek, Zhi Wei, Yue Yu, Diane Grill, and Karla Ballman. 2018. <span>“An Omnibus Test for Differential Distribution Analysis of Microbiome Sequencing Data.”</span> <em>Bioinformatics</em> 34 (4): 643–51.
</div>
<div id="ref-chen2018gmpr" class="csl-entry">
Chen, Li, James Reeve, Lujun Zhang, Shengbing Huang, Xuefeng Wang, and Jun Chen. 2018. <span>“GMPR: A Robust Normalization Method for Zero-Inflated Count Data with Application to Microbiome Sequencing Data.”</span> <em>PeerJ</em> 6: e4600.
</div>
<div id="ref-fernandes2014unifying" class="csl-entry">
Fernandes, Andrew D, Jennifer Ns Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>“Unifying the Analysis of High-Throughput Sequencing Datasets: Characterizing RNA-Seq, 16s rRNA Gene Sequencing and Selective Growth Experiments by Compositional Data Analysis.”</span> <em>Microbiome</em> 2 (1): 1–13.
</div>
<div id="ref-law2014voom" class="csl-entry">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. <span>“Voom: Precision Weights Unlock Linear Model Analysis Tools for RNA-Seq Read Counts.”</span> <em>Genome Biology</em> 15 (2): 1–17.
</div>
<div id="ref-love2014moderated" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.”</span> <em>Genome Biology</em> 15 (12): 1–21.
</div>
<div id="ref-mallick2021multivariable" class="csl-entry">
Mallick, Himel, Ali Rahnavard, Lauren J McIver, Siyuan Ma, Yancong Zhang, Long H Nguyen, Timothy L Tickle, et al. 2021. <span>“Multivariable Association Discovery in Population-Scale Meta-Omics Studies.”</span> <em>PLoS Computational Biology</em> 17 (11): e1009442.
</div>
<div id="ref-mandal2015analysis" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A White, Merete Eggesbø, Rob Knight, and Shyamal D Peddada. 2015. <span>“Analysis of Composition of Microbiomes: A Novel Method for Studying Microbial Composition.”</span> <em>Microbial Ecology in Health and Disease</em> 26 (1): 27663.
</div>
<div id="ref-martin2020modeling" class="csl-entry">
Martin, Bryan D, Daniela Witten, and Amy D Willis. 2020. <span>“Modeling Microbial Abundances and Dysbiosis with Beta-Binomial Regression.”</span> <em>The Annals of Applied Statistics</em> 14 (1): 94.
</div>
<div id="ref-nearing2022microbiome" class="csl-entry">
Nearing, Jacob T, Gavin M Douglas, Molly G Hayes, Jocelyn MacDonald, Dhwani K Desai, Nicole Allward, Casey Jones, et al. 2022. <span>“Microbiome Differential Abundance Methods Produce Different Results Across 38 Datasets.”</span> <em>Nature Communications</em> 13 (1): 1–16.
</div>
<div id="ref-paulson2013differential" class="csl-entry">
Paulson, Joseph N, O Colin Stine, Héctor Corrada Bravo, and Mihai Pop. 2013. <span>“Differential Abundance Analysis for Microbial Marker-Gene Surveys.”</span> <em>Nature Methods</em> 10 (12): 1200–1202.
</div>
<div id="ref-robinson2010scaling" class="csl-entry">
Robinson, Mark D, and Alicia Oshlack. 2010. <span>“A Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data.”</span> <em>Genome Biology</em> 11 (3): 1–9.
</div>
<div id="ref-segata2011metagenomic" class="csl-entry">
Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. <span>“Metagenomic Biomarker Discovery and Explanation.”</span> <em>Genome Biology</em> 12 (6): 1–18.
</div>
<div id="ref-sohn2015robust" class="csl-entry">
Sohn, Michael B, Ruofei Du, and Lingling An. 2015. <span>“A Robust Approach for Identifying Differentially Abundant Features in Metagenomic Samples.”</span> <em>Bioinformatics</em> 31 (14): 2269–75.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="core-microbiota.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualization-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/10-differentail-analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
