<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Examples | Data analysis based on XMAS 2.0 package in microbiota data</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Examples | Data analysis based on XMAS 2.0 package in microbiota data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Examples | Data analysis based on XMAS 2.0 package in microbiota data" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-05-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visualization-5.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation<span></span></a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette<span></span></a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors<span></span></a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Reads’ track<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.3</b> Rarefaction Curves<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.4</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#removing-brs"><i class="fa fa-check"></i><b>6.1</b> Removing BRS<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.2</b> Transformation<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.3</b> Normalization<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.4</b> Filtering<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.5</b> Trimming<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.6</b> Imputation<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.7</b> Extracting specific taxa phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.8</b> Aggregating low relative abundance or unclassified taxa into others<span></span></a></li>
<li class="chapter" data-level="6.9" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.9</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics<span></span></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.6</b> Analysis of Similarity (ANOSIM)<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.7</b> Ordination<span></span></a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.7.1</b> Principal Component Analysis (PCA)<span></span></a></li>
<li class="chapter" data-level="8.7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.7.2</b> Principal Coordinate Analysis (PCoA)<span></span></a></li>
<li class="chapter" data-level="8.7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.7.3</b> t-distributed stochastic neighbor embedding (t-SNE)<span></span></a></li>
<li class="chapter" data-level="8.7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.7.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="composition.html"><a href="composition.html"><i class="fa fa-check"></i><b>9</b> Composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="composition.html"><a href="composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="composition.html"><a href="composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.2</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="composition.html"><a href="composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.3</b> XVIZ <code>plot_stacked_bar</code><span></span></a></li>
<li class="chapter" data-level="9.4" data-path="composition.html"><a href="composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="composition.html"><a href="composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus<span></span></a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq<span></span></a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR<span></span></a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM<span></span></a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob<span></span></a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)<span></span></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom-1"><i class="fa fa-check"></i><b>11.2.1</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.2</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.3</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.4</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.2.5" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.5</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot<span></span></a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot<span></span></a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot<span></span></a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density<span></span></a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination<span></span></a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot<span></span></a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>13</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>13.1</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>13.2</b> Amplicon sequencing (16s)<span></span></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>13.2.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="13.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>13.2.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="13.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>13.2.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="13.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>13.2.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="13.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>13.2.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="13.2.6" data-path="examples.html"><a href="examples.html#microbial-composition"><i class="fa fa-check"></i><b>13.2.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="13.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>13.2.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="examples.html"><a href="examples.html#metagenomics"><i class="fa fa-check"></i><b>13.3</b> Metagenomics<span></span></a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>13.3.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="13.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>13.3.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="13.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>13.3.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="13.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>13.3.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="13.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>13.3.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="13.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>13.3.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="13.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>13.3.7</b> Differential Analysis<span></span></a></li>
</ul></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data analysis based on XMAS 2.0 package in microbiota data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="examples" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">Chapter 13</span> Examples<a href="examples.html#examples" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here, we give users two examples to practice the data analysis workflow by <strong>XMAS 2.0</strong>. By the way, we also recommend users handling your own microbiota data in a reasonable manner when you utilize this package or workflow. Pay attention to whether your data fit the methods this package provided.</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="examples.html#cb309-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS)</span>
<span id="cb309-2"><a href="examples.html#cb309-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb309-3"><a href="examples.html#cb309-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb309-4"><a href="examples.html#cb309-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb309-5"><a href="examples.html#cb309-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb309-6"><a href="examples.html#cb309-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
<div id="workflow-description" class="section level2 hasAnchor" number="13.1">
<h2><span class="header-section-number">13.1</span> Workflow description<a href="examples.html#workflow-description" class="anchor-section" aria-label="Anchor link to header"></a></h2>
The standard data analysis of 16S and MGS data by <strong>XMAS 2.0</strong>.
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-162"></span>
<div id="htmlwidget-242d236746922dd22c2a" style="width:672px;height:768px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-242d236746922dd22c2a">{"x":{"diagram":"digraph flowchart { \n\n# initiate graph\ngraph [layout = dot, rankdir = TB, label = \"[dada2_res|metadata|tree]\n\n\", labelloc = t]\n\n\n# global node settings\nnode [shape = rectangle, style = filled, fillcolor = Linen, fontname = arial, width = 4]\n \ntab1 [label = \"Importing data: import_dada2()/summarize_phyloseq()\"]  \ntab2 [label = \"Quality Control(RefCheck/Remove): summarize_taxa()/run_RefCheck()/get_GroupPhyloseq()\"]  \ntab3 [label = \"Quality Control(Rarefaction curves): plot_RarefCure()\"]\ntab4 [label = \"Quality Control(Reads Track): plot_Dada2Track()\"]  \ntab5 [label = \"Quality Control(Variability): plot_taxa_cv()/plot_read_distribution()\"]  \ntab6 [label = \"Data processing(Rarefy): normalize()\"]\ntab7 [label = \"Data processing(Extract specific taxonomic level): summarize_taxa()\"]  \ntab8 [label = \"Data processing(Count2Relative): summarize_taxa(absolute = FALSE)\"]  \ntab9 [label = \"Data processing(Filter&trim): run_filter()/run_trim()\"]\ntab10 [label = \"GlobalView(Alpha diveristy): run_alpha_diversity()/plot_boxplot()\"]  \ntab11 [label = \"GlobalView(Beta diveristy): run_beta_diversity()/run_permanova()/run_ordination()/plot_Ordination()\"]  \ntab12 [label = \"GlobalView(microbial composition): plot_StackBarPlot()/plot_core_taxa()/core_members()\"]\ntab13 [label = \"DifferentialAnalysis: run_lefse()/plot_lefse()/run_wilcox()/plot_volcano()\"] \n \ntab1 -> tab2 -> tab3 -> tab4 -> tab5 -> tab6 -> tab7 -> tab8 -> tab9 -> tab10 -> tab11 -> tab12 -> tab13;  \n}  ","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 13.1: Functions of XMAS 2.0 in 16s
</p>
</div>
</div>
<div id="amplicon-sequencing-16s" class="section level2 hasAnchor" number="13.2">
<h2><span class="header-section-number">13.2</span> Amplicon sequencing (16s)<a href="examples.html#amplicon-sequencing-16s" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The upstream process is performed by in-house pipeline. and this example just shows how to perform downstream data analysis. In briefly, the example comprises the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Converting inputs into phyloseq object;</p></li>
<li><p>Quality Evaluation;</p></li>
<li><p>Pre-Processing Data;</p></li>
<li><p>Diversity analysis;</p></li>
<li><p>Ordination analysis;</p></li>
<li><p>Composition analysis;</p></li>
<li><p>Differential analysis.</p></li>
</ol>
<div id="converting-inputs-into-phyloseq-class-object" class="section level3 hasAnchor" number="13.2.1">
<h3><span class="header-section-number">13.2.1</span> Converting inputs into phyloseq-class object<a href="examples.html#converting-inputs-into-phyloseq-class-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>dada2</strong> result from standardized_analytics_workflow_R_function.</p>
<ol style="list-style-type: decimal">
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/xdada2/dada2_res.rds</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/fasta2tree/tree.nwk</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/metadata.txt</p></li>
</ol>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="examples.html#cb310-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dada2 results from in-house 16s pipeline</span></span>
<span id="cb310-2"><a href="examples.html#cb310-2" aria-hidden="true" tabindex="-1"></a>dada2_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb310-3"><a href="examples.html#cb310-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system.file</span>(</span>
<span id="cb310-4"><a href="examples.html#cb310-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dada2_res.rds&quot;</span>,</span>
<span id="cb310-5"><a href="examples.html#cb310-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span>    </span>
<span id="cb310-6"><a href="examples.html#cb310-6" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb310-7"><a href="examples.html#cb310-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb310-8"><a href="examples.html#cb310-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-9"><a href="examples.html#cb310-9" aria-hidden="true" tabindex="-1"></a><span class="co"># the metadata matches to dada2 result</span></span>
<span id="cb310-10"><a href="examples.html#cb310-10" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb310-11"><a href="examples.html#cb310-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb310-12"><a href="examples.html#cb310-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dada2_metadata.tsv&quot;</span>,</span>
<span id="cb310-13"><a href="examples.html#cb310-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span></span>
<span id="cb310-14"><a href="examples.html#cb310-14" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb310-15"><a href="examples.html#cb310-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb310-16"><a href="examples.html#cb310-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb310-17"><a href="examples.html#cb310-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb310-18"><a href="examples.html#cb310-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb310-19"><a href="examples.html#cb310-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-20"><a href="examples.html#cb310-20" aria-hidden="true" tabindex="-1"></a><span class="co"># tree file from dada2 reference data silva</span></span>
<span id="cb310-21"><a href="examples.html#cb310-21" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">read_tree</span>(</span>
<span id="cb310-22"><a href="examples.html#cb310-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system.file</span>(</span>
<span id="cb310-23"><a href="examples.html#cb310-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;tree.nwk&quot;</span>,</span>
<span id="cb310-24"><a href="examples.html#cb310-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span>    </span>
<span id="cb310-25"><a href="examples.html#cb310-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb310-26"><a href="examples.html#cb310-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb310-27"><a href="examples.html#cb310-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-28"><a href="examples.html#cb310-28" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> <span class="fu">import_dada2_taxa</span>(dada2_res<span class="sc">$</span>tax_tab)</span>
<span id="cb310-29"><a href="examples.html#cb310-29" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> dada2_res<span class="sc">$</span>seq_tab</span>
<span id="cb310-30"><a href="examples.html#cb310-30" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> sam_tab <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;seqID&quot;</span>)</span>
<span id="cb310-31"><a href="examples.html#cb310-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-32"><a href="examples.html#cb310-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Shouldn&#39;t use the Total Number as SampleID (wrong: 123456; right: X123456)</span></span>
<span id="cb310-33"><a href="examples.html#cb310-33" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(otu_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="fu">rownames</span>(otu_tab))</span>
<span id="cb310-34"><a href="examples.html#cb310-34" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="fu">rownames</span>(sam_tab))</span>
<span id="cb310-35"><a href="examples.html#cb310-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-36"><a href="examples.html#cb310-36" aria-hidden="true" tabindex="-1"></a>dada2_ps <span class="ot">&lt;-</span> <span class="fu">import_dada2</span>(<span class="at">seq_tab =</span> otu_tab, <span class="at">tax_tab =</span> tax_tab, <span class="at">sam_tab =</span> sam_tab, <span class="at">phy_tree =</span> tree)</span>
<span id="cb310-37"><a href="examples.html#cb310-37" aria-hidden="true" tabindex="-1"></a>dada2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 24 samples ]
## sample_data() Sample Data:       [ 24 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
<p>Here, the phyloseq object comprises five components (<em>OTU Table</em>, <em>Sample Data</em>, <em>Taxonomy Table</em>, <em>Phylogenetic Tree</em> and <em>DNAStringSet</em>).</p>
<div id="summarize-phyloseq-class-object-1" class="section level4 hasAnchor" number="13.2.1.1">
<h4><span class="header-section-number">13.2.1.1</span> Summarize phyloseq-class object<a href="examples.html#summarize-phyloseq-class-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="examples.html#cb312-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(dada2_ps)</span>
<span id="cb312-2"><a href="examples.html#cb312-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Compositional = NO2</span></span>
<span id="cb312-3"><a href="examples.html#cb312-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 1] Min. number of reads = 511812] Max. number of reads = 936223] Total number of reads = 15025374] Average number of reads = 62605.70833333335] Median number of reads = 619157] Sparsity = 0.8653738839285716] Any OTU sum to 1 or less? NO8] Number of singletons = 09] Percent of OTUs that are singletons</span></span>
<span id="cb312-4"><a href="examples.html#cb312-4" aria-hidden="true" tabindex="-1"></a><span class="do">##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 1Group2</span></span>
<span id="cb312-5"><a href="examples.html#cb312-5" aria-hidden="true" tabindex="-1"></a><span class="do">## [[1]]</span></span>
<span id="cb312-6"><a href="examples.html#cb312-6" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;1] Min. number of reads = 51181&quot;</span></span>
<span id="cb312-7"><a href="examples.html#cb312-7" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-8"><a href="examples.html#cb312-8" aria-hidden="true" tabindex="-1"></a><span class="do">## [[2]]</span></span>
<span id="cb312-9"><a href="examples.html#cb312-9" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;2] Max. number of reads = 93622&quot;</span></span>
<span id="cb312-10"><a href="examples.html#cb312-10" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-11"><a href="examples.html#cb312-11" aria-hidden="true" tabindex="-1"></a><span class="do">## [[3]]</span></span>
<span id="cb312-12"><a href="examples.html#cb312-12" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;3] Total number of reads = 1502537&quot;</span></span>
<span id="cb312-13"><a href="examples.html#cb312-13" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-14"><a href="examples.html#cb312-14" aria-hidden="true" tabindex="-1"></a><span class="do">## [[4]]</span></span>
<span id="cb312-15"><a href="examples.html#cb312-15" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;4] Average number of reads = 62605.7083333333&quot;</span></span>
<span id="cb312-16"><a href="examples.html#cb312-16" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-17"><a href="examples.html#cb312-17" aria-hidden="true" tabindex="-1"></a><span class="do">## [[5]]</span></span>
<span id="cb312-18"><a href="examples.html#cb312-18" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;5] Median number of reads = 61915&quot;</span></span>
<span id="cb312-19"><a href="examples.html#cb312-19" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-20"><a href="examples.html#cb312-20" aria-hidden="true" tabindex="-1"></a><span class="do">## [[6]]</span></span>
<span id="cb312-21"><a href="examples.html#cb312-21" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;7] Sparsity = 0.865373883928571&quot;</span></span>
<span id="cb312-22"><a href="examples.html#cb312-22" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-23"><a href="examples.html#cb312-23" aria-hidden="true" tabindex="-1"></a><span class="do">## [[7]]</span></span>
<span id="cb312-24"><a href="examples.html#cb312-24" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;</span></span>
<span id="cb312-25"><a href="examples.html#cb312-25" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-26"><a href="examples.html#cb312-26" aria-hidden="true" tabindex="-1"></a><span class="do">## [[8]]</span></span>
<span id="cb312-27"><a href="examples.html#cb312-27" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;8] Number of singletons = 0&quot;</span></span>
<span id="cb312-28"><a href="examples.html#cb312-28" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-29"><a href="examples.html#cb312-29" aria-hidden="true" tabindex="-1"></a><span class="do">## [[9]]</span></span>
<span id="cb312-30"><a href="examples.html#cb312-30" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;</span></span>
<span id="cb312-31"><a href="examples.html#cb312-31" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-32"><a href="examples.html#cb312-32" aria-hidden="true" tabindex="-1"></a><span class="do">## [[10]]</span></span>
<span id="cb312-33"><a href="examples.html#cb312-33" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;10] Number of sample variables are: 1&quot;</span></span>
<span id="cb312-34"><a href="examples.html#cb312-34" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb312-35"><a href="examples.html#cb312-35" aria-hidden="true" tabindex="-1"></a><span class="do">## [[11]]</span></span>
<span id="cb312-36"><a href="examples.html#cb312-36" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;Group&quot;</span></span></code></pre></div>
<p>The minus account of the OTU counts is <em>51181</em> in the phyloseq object, and we can use it as the threshold to rarefy.</p>
<p>Notice the <strong>Sparsity (0.865)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.</p>
</div>
</div>
<div id="quality-control" class="section level3 hasAnchor" number="13.2.2">
<h3><span class="header-section-number">13.2.2</span> Quality Control<a href="examples.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Quality control of DADA2 results will help us have more rational determinations on the further data analysis.</p>
<div id="reads-track-by-dada2" class="section level4 hasAnchor" number="13.2.2.1">
<h4><span class="header-section-number">13.2.2.1</span> Reads’ track by DADA2<a href="examples.html#reads-track-by-dada2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="examples.html#cb313-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Dada2Track</span>(</span>
<span id="cb313-2"><a href="examples.html#cb313-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data=</span>dada2_res<span class="sc">$</span>reads_track,</span>
<span id="cb313-3"><a href="examples.html#cb313-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mode=</span><span class="st">&quot;PE&quot;</span>,</span>
<span id="cb313-4"><a href="examples.html#cb313-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">relative_abundance=</span><span class="cn">TRUE</span>,</span>
<span id="cb313-5"><a href="examples.html#cb313-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position=</span><span class="st">&quot;top&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-165"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-165-1.png" alt="Reads' track by DADA2 (16s example)" width="576" />
<p class="caption">
Figure 13.2: Reads’ track by DADA2 (16s example)
</p>
</div>
<p>The percentage of the final remained read counts approximate <strong>70%</strong>, indicating that we should consider the sequence depth for analysis when we build the sequence library.</p>
</div>
<div id="spike-in-sample-brs-assessment" class="section level4 hasAnchor" number="13.2.2.2">
<h4><span class="header-section-number">13.2.2.2</span> Spike-in sample (BRS) assessment<a href="examples.html#spike-in-sample-brs-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="examples.html#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extracting the genus level phyloseq  </span></span>
<span id="cb314-2"><a href="examples.html#cb314-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb314-3"><a href="examples.html#cb314-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(dada2_ps_genus)</span>
<span id="cb314-4"><a href="examples.html#cb314-4" aria-hidden="true" tabindex="-1"></a><span class="do">##       Group</span></span>
<span id="cb314-5"><a href="examples.html#cb314-5" aria-hidden="true" tabindex="-1"></a><span class="do">## S6030    BB</span></span>
<span id="cb314-6"><a href="examples.html#cb314-6" aria-hidden="true" tabindex="-1"></a><span class="do">## S6032    BB</span></span>
<span id="cb314-7"><a href="examples.html#cb314-7" aria-hidden="true" tabindex="-1"></a><span class="do">## S6033    BB</span></span>
<span id="cb314-8"><a href="examples.html#cb314-8" aria-hidden="true" tabindex="-1"></a><span class="do">## S6035    AA</span></span>
<span id="cb314-9"><a href="examples.html#cb314-9" aria-hidden="true" tabindex="-1"></a><span class="do">## S6036    BB</span></span>
<span id="cb314-10"><a href="examples.html#cb314-10" aria-hidden="true" tabindex="-1"></a><span class="do">## S6037    AA</span></span>
<span id="cb314-11"><a href="examples.html#cb314-11" aria-hidden="true" tabindex="-1"></a><span class="do">## S6040    BB</span></span>
<span id="cb314-12"><a href="examples.html#cb314-12" aria-hidden="true" tabindex="-1"></a><span class="do">## S6043    AA</span></span>
<span id="cb314-13"><a href="examples.html#cb314-13" aria-hidden="true" tabindex="-1"></a><span class="do">## S6045    BB</span></span>
<span id="cb314-14"><a href="examples.html#cb314-14" aria-hidden="true" tabindex="-1"></a><span class="do">## S6046    BB</span></span>
<span id="cb314-15"><a href="examples.html#cb314-15" aria-hidden="true" tabindex="-1"></a><span class="do">## S6048    BB</span></span>
<span id="cb314-16"><a href="examples.html#cb314-16" aria-hidden="true" tabindex="-1"></a><span class="do">## S6049    AA</span></span>
<span id="cb314-17"><a href="examples.html#cb314-17" aria-hidden="true" tabindex="-1"></a><span class="do">## S6050    BB</span></span>
<span id="cb314-18"><a href="examples.html#cb314-18" aria-hidden="true" tabindex="-1"></a><span class="do">## S6054    BB</span></span>
<span id="cb314-19"><a href="examples.html#cb314-19" aria-hidden="true" tabindex="-1"></a><span class="do">## S6055    BB</span></span>
<span id="cb314-20"><a href="examples.html#cb314-20" aria-hidden="true" tabindex="-1"></a><span class="do">## S6058    BB</span></span>
<span id="cb314-21"><a href="examples.html#cb314-21" aria-hidden="true" tabindex="-1"></a><span class="do">## S6059    AA</span></span>
<span id="cb314-22"><a href="examples.html#cb314-22" aria-hidden="true" tabindex="-1"></a><span class="do">## S6060    AA</span></span>
<span id="cb314-23"><a href="examples.html#cb314-23" aria-hidden="true" tabindex="-1"></a><span class="do">## S6061    AA</span></span>
<span id="cb314-24"><a href="examples.html#cb314-24" aria-hidden="true" tabindex="-1"></a><span class="do">## S6063    BB</span></span>
<span id="cb314-25"><a href="examples.html#cb314-25" aria-hidden="true" tabindex="-1"></a><span class="do">## S6065    AA</span></span>
<span id="cb314-26"><a href="examples.html#cb314-26" aria-hidden="true" tabindex="-1"></a><span class="do">## S6066    AA</span></span>
<span id="cb314-27"><a href="examples.html#cb314-27" aria-hidden="true" tabindex="-1"></a><span class="do">## S6068    BB</span></span>
<span id="cb314-28"><a href="examples.html#cb314-28" aria-hidden="true" tabindex="-1"></a><span class="do">## S8005    QC</span></span>
<span id="cb314-29"><a href="examples.html#cb314-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb314-30"><a href="examples.html#cb314-30" aria-hidden="true" tabindex="-1"></a>RefCheck_16s <span class="ot">&lt;-</span> <span class="fu">run_RefCheck</span>(</span>
<span id="cb314-31"><a href="examples.html#cb314-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps=</span>dada2_ps_genus,</span>
<span id="cb314-32"><a href="examples.html#cb314-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">BRS_ID=</span><span class="st">&quot;S8005&quot;</span>,</span>
<span id="cb314-33"><a href="examples.html#cb314-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">Reference=</span><span class="cn">NULL</span>,</span>
<span id="cb314-34"><a href="examples.html#cb314-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">Ref_type=</span><span class="st">&quot;16s&quot;</span>,</span>
<span id="cb314-35"><a href="examples.html#cb314-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">Save=</span><span class="cn">NULL</span>)</span>
<span id="cb314-36"><a href="examples.html#cb314-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Noting: the Reference Matrix is for 16s</span></span>
<span id="cb314-37"><a href="examples.html#cb314-37" aria-hidden="true" tabindex="-1"></a><span class="do">## S8005 is in the Reference Matrix&#39;s samples and remove it to run</span></span>
<span id="cb314-38"><a href="examples.html#cb314-38" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb314-39"><a href="examples.html#cb314-39" aria-hidden="true" tabindex="-1"></a><span class="do">## #########Matched baterica of the Spike-in sample###########</span></span>
<span id="cb314-40"><a href="examples.html#cb314-40" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking out all the taxa of the Reference Matrix are in the sequencing dataset </span></span>
<span id="cb314-41"><a href="examples.html#cb314-41" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Bifidobacterium</span></span>
<span id="cb314-42"><a href="examples.html#cb314-42" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Bacteroides</span></span>
<span id="cb314-43"><a href="examples.html#cb314-43" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Faecalibacterium</span></span>
<span id="cb314-44"><a href="examples.html#cb314-44" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Lactobacillus</span></span>
<span id="cb314-45"><a href="examples.html#cb314-45" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Parabacteroides</span></span>
<span id="cb314-46"><a href="examples.html#cb314-46" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Collinsella</span></span>
<span id="cb314-47"><a href="examples.html#cb314-47" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Coprococcus_3</span></span>
<span id="cb314-48"><a href="examples.html#cb314-48" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Dorea</span></span>
<span id="cb314-49"><a href="examples.html#cb314-49" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Streptococcus</span></span>
<span id="cb314-50"><a href="examples.html#cb314-50" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Roseburia</span></span>
<span id="cb314-51"><a href="examples.html#cb314-51" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Anaerostipes</span></span>
<span id="cb314-52"><a href="examples.html#cb314-52" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Escherichia/Shigella</span></span>
<span id="cb314-53"><a href="examples.html#cb314-53" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Enterococcus</span></span>
<span id="cb314-54"><a href="examples.html#cb314-54" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Prevotella_9</span></span>
<span id="cb314-55"><a href="examples.html#cb314-55" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Eggerthella</span></span>
<span id="cb314-56"><a href="examples.html#cb314-56" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb314-57"><a href="examples.html#cb314-57" aria-hidden="true" tabindex="-1"></a><span class="do">## The number of the additional Taxa in the sequencing dataset is 1</span></span>
<span id="cb314-58"><a href="examples.html#cb314-58" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb314-59"><a href="examples.html#cb314-59" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb314-60"><a href="examples.html#cb314-60" aria-hidden="true" tabindex="-1"></a><span class="do">## ##############Value of the Reference sample###############</span></span>
<span id="cb314-61"><a href="examples.html#cb314-61" aria-hidden="true" tabindex="-1"></a><span class="do">## Correlation Coefficient of the Reference sample is: 0.9714</span></span>
<span id="cb314-62"><a href="examples.html#cb314-62" aria-hidden="true" tabindex="-1"></a><span class="do">## Bray Curtis of the Reference sample is: 0.07607</span></span>
<span id="cb314-63"><a href="examples.html#cb314-63" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity of the Reference sample is: 0.06409</span></span>
<span id="cb314-64"><a href="examples.html#cb314-64" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb314-65"><a href="examples.html#cb314-65" aria-hidden="true" tabindex="-1"></a><span class="do">## #####Final Evaluation Results of the Spike-in sample#######</span></span>
<span id="cb314-66"><a href="examples.html#cb314-66" aria-hidden="true" tabindex="-1"></a><span class="do">## The spike-in sample of sequencing dataset passed the cutoff of the Reference Matrix </span></span>
<span id="cb314-67"><a href="examples.html#cb314-67" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of Coefficient is 0.8946</span></span>
<span id="cb314-68"><a href="examples.html#cb314-68" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of BrayCurtis is 0.3878</span></span>
<span id="cb314-69"><a href="examples.html#cb314-69" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of Impurity is 0.1565</span></span>
<span id="cb314-70"><a href="examples.html#cb314-70" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span></code></pre></div>
</div>
<div id="spike-in-samples-remove" class="section level4 hasAnchor" number="13.2.2.3">
<h4><span class="header-section-number">13.2.2.3</span> Spike-in sample’s remove<a href="examples.html#spike-in-samples-remove" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="examples.html#cb315-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_NoQC <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb315-2"><a href="examples.html#cb315-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb315-3"><a href="examples.html#cb315-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb315-4"><a href="examples.html#cb315-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb315-5"><a href="examples.html#cb315-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">discard =</span> <span class="cn">TRUE</span></span>
<span id="cb315-6"><a href="examples.html#cb315-6" aria-hidden="true" tabindex="-1"></a>                     )</span>
<span id="cb315-7"><a href="examples.html#cb315-7" aria-hidden="true" tabindex="-1"></a>dada2_ps_NoQC</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
</div>
<div id="rarefaction-curves-1" class="section level4 hasAnchor" number="13.2.2.4">
<h4><span class="header-section-number">13.2.2.4</span> Rarefaction curves<a href="examples.html#rarefaction-curves-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="examples.html#cb317-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_RarefCure</span>(<span class="at">ps =</span> dada2_ps_NoQC,</span>
<span id="cb317-2"><a href="examples.html#cb317-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">taxa_level =</span> <span class="st">&quot;OTU&quot;</span>,</span>
<span id="cb317-3"><a href="examples.html#cb317-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">step =</span> <span class="dv">100</span>,</span>
<span id="cb317-4"><a href="examples.html#cb317-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">label =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb317-5"><a href="examples.html#cb317-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb317-6"><a href="examples.html#cb317-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">parallel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb317-7"><a href="examples.html#cb317-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">se =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-168"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-168-1.png" alt="Rarefaction curves (16s example)" width="576" />
<p class="caption">
Figure 13.3: Rarefaction curves (16s example)
</p>
</div>
<p>The result showed that all the samples had <strong>different sequence depth</strong> but had the full sample richness.</p>
</div>
</div>
<div id="data-processing" class="section level3 hasAnchor" number="13.2.3">
<h3><span class="header-section-number">13.2.3</span> Data processing<a href="examples.html#data-processing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This part has too may procedures and we only choose some of them. Please go to <strong>Chapter 6</strong> to see more approaches and details for being familiar with this part.</p>
<div id="rarefy-otu-counts" class="section level4 hasAnchor" number="13.2.3.1">
<h4><span class="header-section-number">13.2.3.1</span> Rarefy otu counts<a href="examples.html#rarefy-otu-counts" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>From previous results of quality evaluation, the sequence depth of samples are different which have effects on the downstream analysis. Here, choosing the <strong>rarefy</strong> (Normaliztion method: random subsampling counts to the smallest library size) to get the equal sample sums.</p>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="examples.html#cb318-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare <span class="ot">&lt;-</span> <span class="fu">normalize</span>(dada2_ps_NoQC, <span class="at">method =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb318-2"><a href="examples.html#cb318-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 881 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 881 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 881 tips and 878 internal nodes ]
## refseq()      DNAStringSet:      [ 881 reference sequences ]</code></pre>
<p>In addition, we could also perform some other normalization methods on the rarefied phyloseq object. By the way, we didn’t transform the data by using log algorithm because the count matrix is required by the following data analysis methods.</p>
</div>
<div id="extracting-specific-taxonomic-level" class="section level4 hasAnchor" number="13.2.3.2">
<h4><span class="header-section-number">13.2.3.2</span> Extracting specific taxonomic level<a href="examples.html#extracting-specific-taxonomic-level" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="examples.html#cb320-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(dada2_ps_rare, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb320-2"><a href="examples.html#cb320-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 196 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 196 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10" class="section level4 hasAnchor" number="13.2.3.3">
<h4><span class="header-section-number">13.2.3.3</span> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 10)<a href="examples.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="examples.html#cb322-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(dada2_ps_rare_genus, <span class="at">cutoff =</span> <span class="dv">10</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb322-2"><a href="examples.html#cb322-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter </span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 140 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 140 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="trimming-the-taxa-with-low-occurrence-less-than-threshold" class="section level4 hasAnchor" number="13.2.3.4">
<h4><span class="header-section-number">13.2.3.4</span> Trimming the taxa with low occurrence less than threshold<a href="examples.html#trimming-the-taxa-with-low-occurrence-less-than-threshold" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="examples.html#cb324-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(dada2_ps_rare_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb324-2"><a href="examples.html#cb324-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_final <span class="ot">&lt;-</span> dada2_ps_rare_genus_filter_trim</span>
<span id="cb324-3"><a href="examples.html#cb324-3" aria-hidden="true" tabindex="-1"></a>dada2_ps_final</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 96 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 96 taxa by 6 taxonomic ranks ]</code></pre>
<p>Finally, we obtained the final phyloseq-class object <em>dada2_ps_rare_genus_filter_trim</em> and changed its name.</p>
</div>
</div>
<div id="diversity-analysis" class="section level3 hasAnchor" number="13.2.4">
<h3><span class="header-section-number">13.2.4</span> Diversity analysis<a href="examples.html#diversity-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="alpha-diveristy" class="section level4 hasAnchor" number="13.2.4.1">
<h4><span class="header-section-number">13.2.4.1</span> Alpha diveristy<a href="examples.html#alpha-diveristy" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculate the alpha diversity</li>
</ul>
<p><strong>Notes:</strong> the otu table must be counts matrix (<em>rarefied but not trim counts matrix</em>) when you choose <em>Observed</em> etc measures.</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="examples.html#cb326-1" aria-hidden="true" tabindex="-1"></a>dada_ps_rare_genus_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps=</span>dada2_ps_rare_genus, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>))</span>
<span id="cb326-2"><a href="examples.html#cb326-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(dada_ps_rare_genus_alpha)</span></code></pre></div>
<pre><code>##    SampleID Group Observed    Chao1   se.chao1  Shannon
## 1     S6030    BB       73 74.66667 2.20420289 2.625015
## 2     S6032    BB       38 38.33333 0.92351054 1.706103
## 3     S6033    BB       78 81.75000 4.20661800 2.715398
## 4     S6035    AA       34 34.60000 1.18427266 1.950613
## 5     S6036    BB       90 90.50000 1.02680466 2.926823
## 6     S6037    AA       35 35.00000 0.24640269 1.940912
## 7     S6040    BB       58 58.00000 0.07081013 2.490386
## 8     S6043    AA       64 64.42857 0.91243464 2.100695
## 9     S6045    BB       59 59.50000 1.29539423 2.914391
## 10    S6046    BB       70 70.20000 0.62095047 2.261940
## 11    S6048    BB       66 66.33333 0.92468743 2.622039
## 12    S6049    AA       60 60.00000 0.16527194 2.505391
## 13    S6050    BB       52 53.50000 2.58996987 2.437949
## 14    S6054    BB       83 83.00000 0.16565961 2.765414
## 15    S6055    BB       74 74.00000 0.12415253 2.302557
## 16    S6058    BB       36 36.00000 0.12325166 2.266876
## 17    S6059    AA       71 71.25000 0.73593351 2.680326
## 18    S6060    AA       44 47.75000 4.20366887 1.873090
## 19    S6061    AA       80 80.00000 0.16562172 3.089127
## 20    S6063    BB       38 38.00000 0.09867544 1.541715
## 21    S6065    AA       74 74.00000 0.00000000 2.876134
## 22    S6066    AA       67 67.75000 1.42220728 2.645127
## 23    S6068    BB       40 40.50000 1.29368122 2.087037</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="examples.html#cb328-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data=</span>dada_ps_rare_genus_alpha,</span>
<span id="cb328-2"><a href="examples.html#cb328-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb328-3"><a href="examples.html#cb328-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb328-4"><a href="examples.html#cb328-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb328-5"><a href="examples.html#cb328-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb328-6"><a href="examples.html#cb328-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb328-7"><a href="examples.html#cb328-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-174"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-174-1.png" alt="Alpha diversity (16s example)" width="576" />
<p class="caption">
Figure 13.4: Alpha diversity (16s example)
</p>
</div>
</div>
<div id="beta-diversity-1" class="section level4 hasAnchor" number="13.2.4.2">
<h4><span class="header-section-number">13.2.4.2</span> Beta diversity<a href="examples.html#beta-diversity-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>beta dipersion</li>
</ul>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="examples.html#cb329-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>dada2_ps_final, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
## Groups     1 0.014401 0.0144006 1.7097    999  0.193
## Residuals 21 0.176885 0.0084231                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         AA    BB
## AA         0.209
## BB 0.20517</code></pre>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="examples.html#cb331-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_beta<span class="sc">$</span>BetaDispersion</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-175"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-175-1.png" alt="Beta diversity (16s example)" width="480" />
<p class="caption">
Figure 13.5: Beta diversity (16s example)
</p>
</div>
</div>
</div>
<div id="permanova-ordination" class="section level3 hasAnchor" number="13.2.5">
<h3><span class="header-section-number">13.2.5</span> PERMANOVA + Ordination<a href="examples.html#permanova-ordination" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="permanova" class="section level4 hasAnchor" number="13.2.5.1">
<h4><span class="header-section-number">13.2.5.1</span> PERMANOVA<a href="examples.html#permanova" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="examples.html#cb332-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_per <span class="ot">&lt;-</span> <span class="fu">run_permanova</span>(<span class="at">ps=</span>dada2_ps_final, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb332-2"><a href="examples.html#cb332-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(dada2_ps_per)</span></code></pre></div>
<pre><code>##          SumsOfSample Df SumsOfSqs   MeanSqs  F.Model         R2 Pr(&gt;F) AdjustedPvalue
## SampleID           23 22 3.8334625 0.1742483 0.000000 1.00000000  1.000          1.000
## Group              23  1 0.2303178 0.2303178 1.342348 0.06008087  0.217          0.434</code></pre>
<p>The PERMANOVA result of the Group (AdjustedPvalue &gt; 0.05) revealed that the two groups had not the distinct patterns of microbial community.</p>
</div>
<div id="ordination-1" class="section level4 hasAnchor" number="13.2.5.2">
<h4><span class="header-section-number">13.2.5.2</span> Ordination<a href="examples.html#ordination-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We performed ordination by using Principal Coordinate Analysis (PCoA). If you want to try other methods please go to see <strong>Chapter 8</strong> for more details.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="examples.html#cb334-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb334-2"><a href="examples.html#cb334-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ps =</span> dada2_ps_final,</span>
<span id="cb334-3"><a href="examples.html#cb334-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb334-4"><a href="examples.html#cb334-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb334-5"><a href="examples.html#cb334-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb334-6"><a href="examples.html#cb334-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> dada2_ps_ordination, </span>
<span id="cb334-7"><a href="examples.html#cb334-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb334-8"><a href="examples.html#cb334-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb334-9"><a href="examples.html#cb334-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-177"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-177-1.png" alt="PCoA (16s example)" width="768" />
<p class="caption">
Figure 13.6: PCoA (16s example)
</p>
</div>
</div>
</div>
<div id="microbial-composition" class="section level3 hasAnchor" number="13.2.6">
<h3><span class="header-section-number">13.2.6</span> Microbial composition<a href="examples.html#microbial-composition" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A whole picture of the microbial composition.</p>
<div id="stacked-barplot-1" class="section level4 hasAnchor" number="13.2.6.1">
<h4><span class="header-section-number">13.2.6.1</span> Stacked barplot<a href="examples.html#stacked-barplot-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>XMAS</strong> package</li>
</ul>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="examples.html#cb335-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb335-2"><a href="examples.html#cb335-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">ps=</span>dada2_ps_rare,</span>
<span id="cb335-3"><a href="examples.html#cb335-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">taxa_level=</span><span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb335-4"><a href="examples.html#cb335-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb335-5"><a href="examples.html#cb335-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-178"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-178-1.png" alt="Microbial composition (16s example)" width="768" />
<p class="caption">
Figure 13.7: Microbial composition (16s example)
</p>
</div>
<ul>
<li><strong>XVIZ</strong> package</li>
</ul>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="examples.html#cb336-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_stacked_bar</span>(</span>
<span id="cb336-2"><a href="examples.html#cb336-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">phyloseq =</span> dada2_ps_rare,</span>
<span id="cb336-3"><a href="examples.html#cb336-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb336-4"><a href="examples.html#cb336-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-179"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-179-1.png" alt="Microbial composition (16s example) XVIZ" width="768" />
<p class="caption">
Figure 13.8: Microbial composition (16s example) XVIZ
</p>
</div>
</div>
<div id="core-microbiota-1" class="section level4 hasAnchor" number="13.2.6.2">
<h4><span class="header-section-number">13.2.6.2</span> Core microbiota<a href="examples.html#core-microbiota-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>convert absolute abundance into relative abundance</li>
</ul>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="examples.html#cb337-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_final_rb <span class="ot">&lt;-</span> <span class="fu">normalize</span>(dada2_ps_final, <span class="at">method =</span> <span class="st">&quot;TSS&quot;</span>)</span>
<span id="cb337-2"><a href="examples.html#cb337-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_final_rb</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 96 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 96 taxa by 6 taxonomic ranks ]</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="examples.html#cb339-1" aria-hidden="true" tabindex="-1"></a>prevalences <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.05</span>, <span class="dv">1</span>, <span class="fl">0.05</span>)</span>
<span id="cb339-2"><a href="examples.html#cb339-2" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="fu">seq</span>(<span class="fu">log10</span>(<span class="fl">1e-3</span>), <span class="fu">log10</span>(<span class="fl">0.2</span>), <span class="at">length =</span> <span class="dv">10</span>)</span>
<span id="cb339-3"><a href="examples.html#cb339-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb339-4"><a href="examples.html#cb339-4" aria-hidden="true" tabindex="-1"></a>pl_core <span class="ot">&lt;-</span> <span class="fu">plot_core_taxa</span>(dada2_ps_final_rb, </span>
<span id="cb339-5"><a href="examples.html#cb339-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, </span>
<span id="cb339-6"><a href="examples.html#cb339-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colours =</span> <span class="fu">gray</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length=</span><span class="dv">5</span>)),</span>
<span id="cb339-7"><a href="examples.html#cb339-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prevalences =</span> prevalences, </span>
<span id="cb339-8"><a href="examples.html#cb339-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">detections =</span> detections, </span>
<span id="cb339-9"><a href="examples.html#cb339-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">min.prevalence =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb339-10"><a href="examples.html#cb339-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)</span>
<span id="cb339-11"><a href="examples.html#cb339-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb339-12"><a href="examples.html#cb339-12" aria-hidden="true" tabindex="-1"></a>pl_core</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-181"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-181-1.png" alt="Core taxa (16s example)" width="768" />
<p class="caption">
Figure 13.9: Core taxa (16s example)
</p>
</div>
<p>The degree of color indicates the size of abundance and prevalence.</p>
<ul>
<li>Use <code>core_members</code> to obtain the core taxa. detection for abundance and prevalence for occurrence.</li>
</ul>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="examples.html#cb340-1" aria-hidden="true" tabindex="-1"></a>core_taxa_name <span class="ot">&lt;-</span> <span class="fu">core_members</span>(dada2_ps_final_rb, <span class="at">detection =</span> <span class="fl">0.01</span>, <span class="at">prevalence =</span> <span class="fl">0.8</span>)</span>
<span id="cb340-2"><a href="examples.html#cb340-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(core_taxa_name)</span></code></pre></div>
<pre><code>## [1] &quot;g__Bifidobacterium&quot; &quot;g__Blautia&quot;</code></pre>
<p><strong>Result</strong>:</p>
<p>Only 2 genera (g__Bifidobacterium and g__Blautia) passed the threshold of detection and prevalence which we choose.</p>
</div>
</div>
<div id="differential-analysis-1" class="section level3 hasAnchor" number="13.2.7">
<h3><span class="header-section-number">13.2.7</span> Differential Analysis<a href="examples.html#differential-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to <strong>Chapter 10</strong> to see more detials.</p>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-2" class="section level4 hasAnchor" number="13.2.7.1">
<h4><span class="header-section-number">13.2.7.1</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="examples.html#liner-discriminant-analysis-lda-effect-size-lefse-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="examples.html#cb342-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb342-2"><a href="examples.html#cb342-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_final_rb,</span>
<span id="cb342-3"><a href="examples.html#cb342-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb342-4"><a href="examples.html#cb342-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb342-5"><a href="examples.html#cb342-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;identity&quot;</span>,</span>
<span id="cb342-6"><a href="examples.html#cb342-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb342-7"><a href="examples.html#cb342-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">wl.p =</span> <span class="fl">0.05</span>,</span>
<span id="cb342-8"><a href="examples.html#cb342-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb342-9"><a href="examples.html#cb342-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dada2_ps_lefse)</span></code></pre></div>
<pre><code>##                       TaxaID         Block Enrichment   LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Ruminococcaceae_UCG_004 9_AA vs 14_BB         BB 0.001566338 0.68422265                                NA
## 2             g__Odoribacter 9_AA vs 14_BB         BB 0.001627981 1.32750624                                NA
## 3              g__Romboutsia 9_AA vs 14_BB         BB 0.002714096 0.02228566                         -5.912727
## 4           g__Lactobacillus 9_AA vs 14_BB         BB 0.005610690 1.46885395                         -3.089192
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB
## 1            1.801477e-04         0.0025187697          0.000000000                        2.293068
## 2            4.503693e-05         0.0005461955          0.000000000                        2.227047
## 3            2.098501e-03         0.0001024748          0.006173409                       -3.513120
## 4            1.149985e-03         0.0005802535          0.004938069                       -5.183311
##   Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA
## 1          0.0014714807       0.0028544544       0.0005824261                    52.17                 77.78
## 2          0.0005956449       0.0011425726       0.0002440484                    52.17                 77.78
## 3          0.0072239557       0.0009840801       0.0112353043                    78.26                 55.56
## 4          0.0199029898       0.0008842387       0.0321293297                    86.96                 77.78
##   Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                 35.71           0.18 (-3.2;3.5)
## 2                 35.71           0.28 (-2.2;2.8)
## 3                 92.86             140 (150;130)
## 4                 92.86 1.2e+08 (1.2e+08;1.2e+08)</code></pre>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="examples.html#cb344-1" aria-hidden="true" tabindex="-1"></a><span class="co"># in order to show barplot (don&#39;t run it in reality)</span></span>
<span id="cb344-2"><a href="examples.html#cb344-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_lefse<span class="sc">$</span>LDA_Score <span class="ot">&lt;-</span> dada2_ps_lefse<span class="sc">$</span>LDA_Score <span class="sc">*</span> <span class="dv">1000</span></span>
<span id="cb344-3"><a href="examples.html#cb344-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb344-4"><a href="examples.html#cb344-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">da_res =</span> dada2_ps_lefse,</span>
<span id="cb344-5"><a href="examples.html#cb344-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb344-6"><a href="examples.html#cb344-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb344-7"><a href="examples.html#cb344-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-184"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-184-1.png" alt="Lefse analysis (16s example)" width="576" />
<p class="caption">
Figure 13.10: Lefse analysis (16s example)
</p>
</div>
</div>
<div id="wilcoxon-rank-sum-test" class="section level4 hasAnchor" number="13.2.7.2">
<h4><span class="header-section-number">13.2.7.2</span> Wilcoxon Rank-Sum test<a href="examples.html#wilcoxon-rank-sum-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="examples.html#cb345-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb345-2"><a href="examples.html#cb345-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_final_rb,</span>
<span id="cb345-3"><a href="examples.html#cb345-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb345-4"><a href="examples.html#cb345-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb345-5"><a href="examples.html#cb345-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;identity&quot;</span>,</span>
<span id="cb345-6"><a href="examples.html#cb345-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb345-7"><a href="examples.html#cb345-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">p_adjust =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb345-8"><a href="examples.html#cb345-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pvalue_cutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb345-9"><a href="examples.html#cb345-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-10"><a href="examples.html#cb345-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dada2_ps_wilcox)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue
## 1   g__Actinomyces 9_AA vs 14_BB  Nonsignif  0.9512599      40.5 0.1647397      0.7451345
## 2 g__Adlercreutzia 9_AA vs 14_BB  Nonsignif  0.1324555      44.0 0.1981520      0.7451345
## 3  g__Agathobacter 9_AA vs 14_BB  Nonsignif  1.1650423      70.0 0.6757773      0.9068010
## 4     g__Alistipes 9_AA vs 14_BB  Nonsignif  0.3384346      68.0 0.7749114      0.9068010
## 5  g__Anaerostipes 9_AA vs 14_BB  Nonsignif  0.6666619      59.0 0.8290752      0.9068010
## 6   g__Bacteroides 9_AA vs 14_BB  Nonsignif  1.2542011      86.0 0.1594464      0.7451345
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                      -1.572730054            0.0007544323         0.0004098991         0.0012193148
## 2                                NA            0.0000000000         0.0000000000         0.0001260398
## 3                       0.055501556            0.0066545385         0.0066886836         0.0064362517
## 4                       0.005734469            0.0016395962         0.0016395962         0.0016330921
## 5                       0.798769872            0.0123768737         0.0206372354         0.0118630900
## 6                       1.580884527            0.0842090514         0.2009908945         0.0671866088
##   Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA Mean Rank Abundance\nBB Occurrence (100%)\n(All)
## 1                     -0.51866765                    9.50                   13.61                    82.61
## 2                     -0.43387758                    9.89                   13.36                    43.48
## 3                      0.15225398                   12.78                   11.50                    65.22
## 4                      0.10974541                   12.56                   11.64                    73.91
## 5                     -0.08834352                   11.56                   12.29                    95.65
## 6                      0.49098635                   14.56                   10.36                   100.00
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 88.89                 78.57       4.6 (7.6;1.6)
## 2                 33.33                 50.00           12 (17;7)
## 3                 55.56                 71.43     0.39 (-1.5;2.3)
## 4                 77.78                 71.43     0.88 (0.61;1.1)
## 5                100.00                 92.86      1.1 (1.3;0.92)
## 6                100.00                100.00     0.49 (-0.9;1.9)</code></pre>
<ul>
<li>Volcano</li>
</ul>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="examples.html#cb347-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb347-2"><a href="examples.html#cb347-2" aria-hidden="true" tabindex="-1"></a>    dada2_ps_wilcox,</span>
<span id="cb347-3"><a href="examples.html#cb347-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb347-4"><a href="examples.html#cb347-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb347-5"><a href="examples.html#cb347-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb347-6"><a href="examples.html#cb347-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb347-7"><a href="examples.html#cb347-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb347-8"><a href="examples.html#cb347-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb347-9"><a href="examples.html#cb347-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-186"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-186-1.png" alt="Wilcoxon Rank-Sum test (16s example)" width="768" />
<p class="caption">
Figure 13.11: Wilcoxon Rank-Sum test (16s example)
</p>
</div>
</div>
</div>
</div>
<div id="metagenomics" class="section level2 hasAnchor" number="13.3">
<h2><span class="header-section-number">13.3</span> Metagenomics<a href="examples.html#metagenomics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The metagenomic data analysis pipeline is just the same as 16s. In briefly, the example comprises the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Converting inputs into phyloseq object;</p></li>
<li><p>Quality Evaluation;</p></li>
<li><p>Pre-Processing Data;</p></li>
<li><p>Diversity analysis;</p></li>
<li><p>Ordination analysis;</p></li>
<li><p>Composition analysis;</p></li>
</ol>
<div id="converting-inputs-into-phyloseq-class-object-1" class="section level3 hasAnchor" number="13.3.1">
<h3><span class="header-section-number">13.3.1</span> Converting inputs into phyloseq-class object<a href="examples.html#converting-inputs-into-phyloseq-class-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The result of the in-house <strong><a href="https://huttenhower.sph.harvard.edu/metaphlan2">Metaphlan2/3</a></strong> pipeline:</p>
<ol style="list-style-type: decimal">
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metaphlan2_merged.tsv</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metadata.txt</p></li>
</ol>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="examples.html#cb348-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_res <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb348-2"><a href="examples.html#cb348-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb348-3"><a href="examples.html#cb348-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;metaphlan2_merged.tsv&quot;</span>,</span>
<span id="cb348-4"><a href="examples.html#cb348-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span></span>
<span id="cb348-5"><a href="examples.html#cb348-5" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb348-6"><a href="examples.html#cb348-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb348-7"><a href="examples.html#cb348-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb348-8"><a href="examples.html#cb348-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb348-9"><a href="examples.html#cb348-9" aria-hidden="true" tabindex="-1"></a>metaphlan2_sam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb348-10"><a href="examples.html#cb348-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb348-11"><a href="examples.html#cb348-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;metaphlan2_metadata.tsv&quot;</span>,</span>
<span id="cb348-12"><a href="examples.html#cb348-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span></span>
<span id="cb348-13"><a href="examples.html#cb348-13" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb348-14"><a href="examples.html#cb348-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb348-15"><a href="examples.html#cb348-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb348-16"><a href="examples.html#cb348-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb348-17"><a href="examples.html#cb348-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb348-18"><a href="examples.html#cb348-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb348-19"><a href="examples.html#cb348-19" aria-hidden="true" tabindex="-1"></a>metaphlan2_res_list <span class="ot">&lt;-</span> <span class="fu">import_metaphlan_taxa</span>(metaphlan2_res, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb348-20"><a href="examples.html#cb348-20" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>abu_tab</span>
<span id="cb348-21"><a href="examples.html#cb348-21" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>tax_tab</span>
<span id="cb348-22"><a href="examples.html#cb348-22" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> metaphlan2_sam <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb348-23"><a href="examples.html#cb348-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb348-24"><a href="examples.html#cb348-24" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps <span class="ot">&lt;-</span> <span class="fu">import_metaphlan2</span>(<span class="at">otu_tab =</span> otu_tab, </span>
<span id="cb348-25"><a href="examples.html#cb348-25" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">sam_tab =</span> sam_tab,</span>
<span id="cb348-26"><a href="examples.html#cb348-26" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">tax_tab =</span> tax_tab)</span>
<span id="cb348-27"><a href="examples.html#cb348-27" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 326 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 326 taxa by 7 taxonomic ranks ]</code></pre>
<p>Here, the phyloseq object comprises three components (<em>OTU Table</em>, <em>Sample Data</em> and <em>Taxonomy Table</em>).</p>
<div id="summarize-phyloseq-class-object-2" class="section level4 hasAnchor" number="13.3.1.1">
<h4><span class="header-section-number">13.3.1.1</span> Summarize phyloseq-class object<a href="examples.html#summarize-phyloseq-class-object-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="examples.html#cb350-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(metaphlan2_ps)</span></code></pre></div>
<pre><code>## Compositional = NO2</code></pre>
<pre><code>## 1] Min. number of reads = 0.98218212] Max. number of reads = 1.00000053] Total number of reads = 22.94511414] Average number of reads = 0.9976136565217395] Median number of reads = 0.99980247] Sparsity = 0.7131234995998936] Any OTU sum to 1 or less? YES8] Number of singletons = 3239] Percent of OTUs that are singletons
##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 2Groupphynotype2</code></pre>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 0.9821821&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 1.0000005&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 22.9451141&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 0.997613656521739&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 0.9998024&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.713123499599893&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? YES&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 323&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 2&quot;
## 
## [[11]]
## [1] &quot;Group&quot;     &quot;phynotype&quot;</code></pre>
<p>Notice the <strong>Sparsity (0.713)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis.</p>
</div>
</div>
<div id="quality-control-1" class="section level3 hasAnchor" number="13.3.2">
<h3><span class="header-section-number">13.3.2</span> Quality Control<a href="examples.html#quality-control-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="spike-in-sample-brs-assessment-1" class="section level4 hasAnchor" number="13.3.2.1">
<h4><span class="header-section-number">13.3.2.1</span> Spike-in sample (BRS) assessment<a href="examples.html#spike-in-sample-brs-assessment-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="examples.html#cb354-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extracting the genus level phyloseq  </span></span>
<span id="cb354-2"><a href="examples.html#cb354-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb354-3"><a href="examples.html#cb354-3" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species<span class="sc">@</span>sam_data</span>
<span id="cb354-4"><a href="examples.html#cb354-4" aria-hidden="true" tabindex="-1"></a><span class="do">##      Group phynotype</span></span>
<span id="cb354-5"><a href="examples.html#cb354-5" aria-hidden="true" tabindex="-1"></a><span class="do">## s1      BB      0.00</span></span>
<span id="cb354-6"><a href="examples.html#cb354-6" aria-hidden="true" tabindex="-1"></a><span class="do">## s2      AA      2.50</span></span>
<span id="cb354-7"><a href="examples.html#cb354-7" aria-hidden="true" tabindex="-1"></a><span class="do">## s3      BB      0.00</span></span>
<span id="cb354-8"><a href="examples.html#cb354-8" aria-hidden="true" tabindex="-1"></a><span class="do">## s4      AA      1.25</span></span>
<span id="cb354-9"><a href="examples.html#cb354-9" aria-hidden="true" tabindex="-1"></a><span class="do">## s5      AA     30.00</span></span>
<span id="cb354-10"><a href="examples.html#cb354-10" aria-hidden="true" tabindex="-1"></a><span class="do">## s6      AA     15.00</span></span>
<span id="cb354-11"><a href="examples.html#cb354-11" aria-hidden="true" tabindex="-1"></a><span class="do">## s7      BB      8.75</span></span>
<span id="cb354-12"><a href="examples.html#cb354-12" aria-hidden="true" tabindex="-1"></a><span class="do">## s8      BB      0.00</span></span>
<span id="cb354-13"><a href="examples.html#cb354-13" aria-hidden="true" tabindex="-1"></a><span class="do">## s9      BB      3.75</span></span>
<span id="cb354-14"><a href="examples.html#cb354-14" aria-hidden="true" tabindex="-1"></a><span class="do">## s10     BB      2.50</span></span>
<span id="cb354-15"><a href="examples.html#cb354-15" aria-hidden="true" tabindex="-1"></a><span class="do">## s11     BB     15.00</span></span>
<span id="cb354-16"><a href="examples.html#cb354-16" aria-hidden="true" tabindex="-1"></a><span class="do">## s12     BB      2.50</span></span>
<span id="cb354-17"><a href="examples.html#cb354-17" aria-hidden="true" tabindex="-1"></a><span class="do">## s13     BB      2.50</span></span>
<span id="cb354-18"><a href="examples.html#cb354-18" aria-hidden="true" tabindex="-1"></a><span class="do">## s14     BB      0.00</span></span>
<span id="cb354-19"><a href="examples.html#cb354-19" aria-hidden="true" tabindex="-1"></a><span class="do">## s15     BB      1.07</span></span>
<span id="cb354-20"><a href="examples.html#cb354-20" aria-hidden="true" tabindex="-1"></a><span class="do">## s16     BB      2.50</span></span>
<span id="cb354-21"><a href="examples.html#cb354-21" aria-hidden="true" tabindex="-1"></a><span class="do">## s17     AA      5.00</span></span>
<span id="cb354-22"><a href="examples.html#cb354-22" aria-hidden="true" tabindex="-1"></a><span class="do">## s18     BB     35.00</span></span>
<span id="cb354-23"><a href="examples.html#cb354-23" aria-hidden="true" tabindex="-1"></a><span class="do">## s19     BB      7.50</span></span>
<span id="cb354-24"><a href="examples.html#cb354-24" aria-hidden="true" tabindex="-1"></a><span class="do">## s20     BB     15.00</span></span>
<span id="cb354-25"><a href="examples.html#cb354-25" aria-hidden="true" tabindex="-1"></a><span class="do">## s21     AA      3.75</span></span>
<span id="cb354-26"><a href="examples.html#cb354-26" aria-hidden="true" tabindex="-1"></a><span class="do">## s22     AA      3.75</span></span>
<span id="cb354-27"><a href="examples.html#cb354-27" aria-hidden="true" tabindex="-1"></a><span class="do">## refE    QC        NA</span></span>
<span id="cb354-28"><a href="examples.html#cb354-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb354-29"><a href="examples.html#cb354-29" aria-hidden="true" tabindex="-1"></a>RefCheck_MGS <span class="ot">&lt;-</span> <span class="fu">run_RefCheck</span>(</span>
<span id="cb354-30"><a href="examples.html#cb354-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps=</span>metaphlan2_ps_species,</span>
<span id="cb354-31"><a href="examples.html#cb354-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">BRS_ID=</span><span class="st">&quot;refE&quot;</span>,</span>
<span id="cb354-32"><a href="examples.html#cb354-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">Reference=</span><span class="cn">NULL</span>,</span>
<span id="cb354-33"><a href="examples.html#cb354-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">Ref_type=</span><span class="st">&quot;MGS&quot;</span>,</span>
<span id="cb354-34"><a href="examples.html#cb354-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">Save=</span><span class="cn">NULL</span></span>
<span id="cb354-35"><a href="examples.html#cb354-35" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb354-36"><a href="examples.html#cb354-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Noting: the Reference Matrix is for MGS</span></span>
<span id="cb354-37"><a href="examples.html#cb354-37" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb354-38"><a href="examples.html#cb354-38" aria-hidden="true" tabindex="-1"></a><span class="do">## #########Matched baterica of the Spike-in sample###########</span></span>
<span id="cb354-39"><a href="examples.html#cb354-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Identifying some taxa of the Reference Matrix are not in the dataset </span></span>
<span id="cb354-40"><a href="examples.html#cb354-40" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Lactobacillus_salivarius</span></span>
<span id="cb354-41"><a href="examples.html#cb354-41" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Bacteroides_fragilis</span></span>
<span id="cb354-42"><a href="examples.html#cb354-42" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Parabacteroides_goldsteinii</span></span>
<span id="cb354-43"><a href="examples.html#cb354-43" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Escherichia_coli</span></span>
<span id="cb354-44"><a href="examples.html#cb354-44" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Enterococcus_faecalis</span></span>
<span id="cb354-45"><a href="examples.html#cb354-45" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Bifidobacterium_bifidum</span></span>
<span id="cb354-46"><a href="examples.html#cb354-46" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Lactobacillus_pentosus</span></span>
<span id="cb354-47"><a href="examples.html#cb354-47" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Bacteroides_intestinalis</span></span>
<span id="cb354-48"><a href="examples.html#cb354-48" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Eggerthella_unclassified</span></span>
<span id="cb354-49"><a href="examples.html#cb354-49" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb354-50"><a href="examples.html#cb354-50" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking out the additional Taxa are in the sequencing dataset</span></span>
<span id="cb354-51"><a href="examples.html#cb354-51" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb354-52"><a href="examples.html#cb354-52" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb354-53"><a href="examples.html#cb354-53" aria-hidden="true" tabindex="-1"></a><span class="do">## ##############Value of the Reference sample###############</span></span>
<span id="cb354-54"><a href="examples.html#cb354-54" aria-hidden="true" tabindex="-1"></a><span class="do">## Correlation Coefficient of the Reference sample is: -0.06471</span></span>
<span id="cb354-55"><a href="examples.html#cb354-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Bray Curtis of the Reference sample is: 0.8752</span></span>
<span id="cb354-56"><a href="examples.html#cb354-56" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity of the Reference sample is: 32.69</span></span>
<span id="cb354-57"><a href="examples.html#cb354-57" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb354-58"><a href="examples.html#cb354-58" aria-hidden="true" tabindex="-1"></a><span class="do">## #####Final Evaluation Results of the Spike-in sample#######</span></span>
<span id="cb354-59"><a href="examples.html#cb354-59" aria-hidden="true" tabindex="-1"></a><span class="do">## The spike-in sample of sequencing dataset didn&#39;t pass the cutoff of the Reference Matrix</span></span>
<span id="cb354-60"><a href="examples.html#cb354-60" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span></code></pre></div>
<p>The spike-in samples didn’t pass the cutoff and failed to add the the Reference Matrix.</p>
</div>
<div id="spike-in-samples-remove-1" class="section level4 hasAnchor" number="13.3.2.2">
<h4><span class="header-section-number">13.3.2.2</span> Spike-in sample’s remove<a href="examples.html#spike-in-samples-remove-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="examples.html#cb355-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_NoQC <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb355-2"><a href="examples.html#cb355-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb355-3"><a href="examples.html#cb355-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb355-4"><a href="examples.html#cb355-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb355-5"><a href="examples.html#cb355-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">discard =</span> <span class="cn">TRUE</span></span>
<span id="cb355-6"><a href="examples.html#cb355-6" aria-hidden="true" tabindex="-1"></a>                     )</span>
<span id="cb355-7"><a href="examples.html#cb355-7" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_NoQC</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 326 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 326 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
<div id="data-processing-1" class="section level3 hasAnchor" number="13.3.3">
<h3><span class="header-section-number">13.3.3</span> Data processing<a href="examples.html#data-processing-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This part has too may procedures and we only choose some of them. Please go to <strong>Chapter 6</strong> to see more approaches and details for being familiar with this part.</p>
<div id="extracting-specific-taxonomic-level-1" class="section level4 hasAnchor" number="13.3.3.1">
<h4><span class="header-section-number">13.3.3.1</span> Extracting specific taxonomic level<a href="examples.html#extracting-specific-taxonomic-level-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="examples.html#cb357-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps_NoQC, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb357-2"><a href="examples.html#cb357-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 325 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 325 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4" class="section level4 hasAnchor" number="13.3.3.2">
<h4><span class="header-section-number">13.3.3.2</span> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)<a href="examples.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb359"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb359-1"><a href="examples.html#cb359-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(metaphlan2_ps_species, <span class="at">cutoff =</span> <span class="fl">1e-4</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb359-2"><a href="examples.html#cb359-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter </span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 233 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 233 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="trimming-the-taxa-with-low-occurrence-less-than-threshold-1" class="section level4 hasAnchor" number="13.3.3.3">
<h4><span class="header-section-number">13.3.3.3</span> Trimming the taxa with low occurrence less than threshold<a href="examples.html#trimming-the-taxa-with-low-occurrence-less-than-threshold-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="examples.html#cb361-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(metaphlan2_ps_species_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb361-2"><a href="examples.html#cb361-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_final <span class="ot">&lt;-</span> metaphlan2_ps_species_filter_trim</span>
<span id="cb361-3"><a href="examples.html#cb361-3" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_final</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 144 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 144 taxa by 7 taxonomic ranks ]</code></pre>
<p>Finally, we obtained the final phyloseq-class object <em>metaphlan2_ps_species_filter_trim</em> and changed its name.</p>
</div>
</div>
<div id="diversity-analysis-1" class="section level3 hasAnchor" number="13.3.4">
<h3><span class="header-section-number">13.3.4</span> Diversity analysis<a href="examples.html#diversity-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="alpha-diveristy-1" class="section level4 hasAnchor" number="13.3.4.1">
<h4><span class="header-section-number">13.3.4.1</span> Alpha diveristy<a href="examples.html#alpha-diveristy-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculate the alpha diversity</li>
</ul>
<p><strong>Notes:</strong> choosing the measures (<em>Shannon, Simpson and InvSimpson</em>) only for relative abundance.</p>
<div class="sourceCode" id="cb363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb363-1"><a href="examples.html#cb363-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps=</span>metaphlan2_ps_NoQC, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>))</span>
<span id="cb363-2"><a href="examples.html#cb363-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(metaphlan2_ps_genus_alpha)</span></code></pre></div>
<div id="htmlwidget-7e7eefe4f665d046a8e6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7e7eefe4f665d046a8e6">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"],["s1","s2","s3","s4","s5","s6","s7","s8","s9","s10","s11","s12","s13","s14","s15","s16","s17","s18","s19","s20","s21","s22"],["BB","AA","BB","AA","AA","AA","BB","BB","BB","BB","BB","BB","BB","BB","BB","BB","AA","BB","BB","BB","AA","AA"],[0,2.5,0,1.25,30,15,8.75,0,3.75,2.5,15,2.5,2.5,0,1.07,2.5,5,35,7.5,15,3.75,3.75],[2.87600158768499,2.04539233869885,3.44117611994772,2.7469172562596,1.45072169425927,2.6199509193286,2.89751908665775,2.97172006041627,3.02066180013859,3.18502591655083,3.06736907722589,3.04704459677349,2.7838992417821,2.96614439261485,3.34506490182498,2.91391100078963,2.81277952299149,1.97816224774278,1.91166100940612,3.15434082279347,2.22525500416406,3.38676763871053],[0.910885740721574,0.810574170417666,0.943937119774493,0.829076757317215,0.641217839699691,0.89503686117821,0.919240703875622,0.888601335158955,0.894822383475912,0.930550291751187,0.921001492615739,0.902804899430806,0.881418103127103,0.920236736855303,0.94171938037692,0.921691174156033,0.886562457342054,0.794122962312653,0.730419204261905,0.925996371376016,0.83760976311228,0.929564684333857],[11.2215486959907,5.27911110224463,17.837114254166,5.85057938466505,2.78720658564232,9.52715411548276,12.3824754299479,8.97676827120772,9.50772638749593,14.3989085802544,12.6584670155328,10.2885844465801,8.43299041734728,12.5370999201214,17.1583625305862,12.7699526742047,8.81542368222266,4.85726825698085,3.70946304710639,13.5128509046638,6.15800567303452,14.1974234166836]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SampleID<\/th>\n      <th>Group<\/th>\n      <th>phynotype<\/th>\n      <th>Shannon<\/th>\n      <th>Simpson<\/th>\n      <th>InvSimpson<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,6]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="examples.html#cb364-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data=</span>metaphlan2_ps_genus_alpha,</span>
<span id="cb364-2"><a href="examples.html#cb364-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>),</span>
<span id="cb364-3"><a href="examples.html#cb364-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb364-4"><a href="examples.html#cb364-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb364-5"><a href="examples.html#cb364-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb364-6"><a href="examples.html#cb364-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb364-7"><a href="examples.html#cb364-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-195"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-195-1.png" alt="Alpha diversity (MGS example)" width="576" />
<p class="caption">
Figure 13.12: Alpha diversity (MGS example)
</p>
</div>
</div>
<div id="beta-diversity-2" class="section level4 hasAnchor" number="13.3.4.2">
<h4><span class="header-section-number">13.3.4.2</span> Beta diversity<a href="examples.html#beta-diversity-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>beta dipersion</li>
</ul>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="examples.html#cb365-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>metaphlan2_ps_final, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##           Df  Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
## Groups     1 0.00725 0.0072500 0.8423    999  0.362
## Residuals 20 0.17215 0.0086076                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         AA   BB
## AA         0.37
## BB 0.36968</code></pre>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="examples.html#cb367-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_beta<span class="sc">$</span>BetaDispersion</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-196"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-196-1.png" alt="Beta diversity (MGS example)" width="480" />
<p class="caption">
Figure 13.13: Beta diversity (MGS example)
</p>
</div>
</div>
</div>
<div id="permanova-ordination-1" class="section level3 hasAnchor" number="13.3.5">
<h3><span class="header-section-number">13.3.5</span> PERMANOVA + Ordination<a href="examples.html#permanova-ordination-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="permanova-1" class="section level4 hasAnchor" number="13.3.5.1">
<h4><span class="header-section-number">13.3.5.1</span> PERMANOVA<a href="examples.html#permanova-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="examples.html#cb368-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_per <span class="ot">&lt;-</span> <span class="fu">run_permanova</span>(<span class="at">ps=</span>metaphlan2_ps_final, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb368-2"><a href="examples.html#cb368-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(metaphlan2_ps_per)</span></code></pre></div>
<div id="htmlwidget-4c589983702aa81569af" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4c589983702aa81569af">{"x":{"filter":"none","vertical":false,"data":[["SampleID","Group","phynotype"],[22,22,22],[21,1,10],[6.86882919889401,0.864659030474406,3.50343060254529],[0.327087104709239,0.864659030474406,0.350343060254529],[0,2.88019495190956,1.14511655973855],[1,0.125881573909805,0.51004770989347],[1,0.002,0.173],[1,0.006,0.2595]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SumsOfSample<\/th>\n      <th>Df<\/th>\n      <th>SumsOfSqs<\/th>\n      <th>MeanSqs<\/th>\n      <th>F.Model<\/th>\n      <th>R2<\/th>\n      <th>Pr(&gt;F)<\/th>\n      <th>AdjustedPvalue<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>The PERMANOVA result of the Group (AdjustedPvalue &lt; 0.05) revealed that the two groups had the distinct patterns of microbial community.</p>
</div>
<div id="ordination-2" class="section level4 hasAnchor" number="13.3.5.2">
<h4><span class="header-section-number">13.3.5.2</span> Ordination<a href="examples.html#ordination-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We performed ordination by using Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP). If you want to try other methods please go to see <strong>Chapter 8</strong> for more details.</p>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="examples.html#cb369-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb369-2"><a href="examples.html#cb369-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ps =</span> metaphlan2_ps_final,</span>
<span id="cb369-3"><a href="examples.html#cb369-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb369-4"><a href="examples.html#cb369-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;UMAP&quot;</span>)</span>
<span id="cb369-5"><a href="examples.html#cb369-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb369-6"><a href="examples.html#cb369-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> metaphlan2_ps_ordination, </span>
<span id="cb369-7"><a href="examples.html#cb369-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb369-8"><a href="examples.html#cb369-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb369-9"><a href="examples.html#cb369-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb369-10"><a href="examples.html#cb369-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_groups&quot;</span>,</span>
<span id="cb369-11"><a href="examples.html#cb369-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">sideboxplot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb369-12"><a href="examples.html#cb369-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-198"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-198-1.png" alt="PCoA (MGS example)" width="576" />
<p class="caption">
Figure 13.14: PCoA (MGS example)
</p>
</div>
</div>
</div>
<div id="microbial-composition-1" class="section level3 hasAnchor" number="13.3.6">
<h3><span class="header-section-number">13.3.6</span> Microbial composition<a href="examples.html#microbial-composition-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="stacked-barplot-2" class="section level4 hasAnchor" number="13.3.6.1">
<h4><span class="header-section-number">13.3.6.1</span> Stacked barplot<a href="examples.html#stacked-barplot-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>XMAS</strong> package</li>
</ul>
<div class="sourceCode" id="cb370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb370-1"><a href="examples.html#cb370-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb370-2"><a href="examples.html#cb370-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">ps=</span>metaphlan2_ps_NoQC,</span>
<span id="cb370-3"><a href="examples.html#cb370-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">taxa_level=</span><span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb370-4"><a href="examples.html#cb370-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb370-5"><a href="examples.html#cb370-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-199"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-199-1.png" alt="Microbial composition (MGS example)" width="768" />
<p class="caption">
Figure 13.15: Microbial composition (MGS example)
</p>
</div>
<ul>
<li><strong>XVIZ</strong> package</li>
</ul>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="examples.html#cb371-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_stacked_bar</span>(</span>
<span id="cb371-2"><a href="examples.html#cb371-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">phyloseq =</span> metaphlan2_ps_NoQC,</span>
<span id="cb371-3"><a href="examples.html#cb371-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb371-4"><a href="examples.html#cb371-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-200"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-200-1.png" alt="Microbial composition (MGS) XVIZ" width="768" />
<p class="caption">
Figure 13.16: Microbial composition (MGS) XVIZ
</p>
</div>
</div>
<div id="core-microbiota-2" class="section level4 hasAnchor" number="13.3.6.2">
<h4><span class="header-section-number">13.3.6.2</span> Core microbiota<a href="examples.html#core-microbiota-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb372-1"><a href="examples.html#cb372-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb372-2"><a href="examples.html#cb372-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-3"><a href="examples.html#cb372-3" aria-hidden="true" tabindex="-1"></a>prevalences <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.05</span>, <span class="dv">1</span>, <span class="fl">0.05</span>)</span>
<span id="cb372-4"><a href="examples.html#cb372-4" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="fu">seq</span>(<span class="fu">log10</span>(<span class="fl">1e-3</span>), <span class="fu">log10</span>(.<span class="dv">2</span>), <span class="at">length =</span> <span class="dv">10</span>)</span>
<span id="cb372-5"><a href="examples.html#cb372-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb372-6"><a href="examples.html#cb372-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_core_taxa</span>(metaphlan2_ps_final, </span>
<span id="cb372-7"><a href="examples.html#cb372-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, </span>
<span id="cb372-8"><a href="examples.html#cb372-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">colours =</span> <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">5</span>, <span class="st">&quot;Spectral&quot;</span>)),</span>
<span id="cb372-9"><a href="examples.html#cb372-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">prevalences =</span> prevalences, </span>
<span id="cb372-10"><a href="examples.html#cb372-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">detections =</span> detections, </span>
<span id="cb372-11"><a href="examples.html#cb372-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">min.prevalence =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb372-12"><a href="examples.html#cb372-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>) <span class="sc">+</span></span>
<span id="cb372-13"><a href="examples.html#cb372-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&quot;italic&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-201"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-201-1.png" alt="Core taxa (MGS example)" width="768" />
<p class="caption">
Figure 13.17: Core taxa (MGS example)
</p>
</div>
<p>The degree of color indicates the size of abundance and prevalence.</p>
<ul>
<li>Use <code>core_members</code> to obtain the core taxa. detection for abundance and prevalence for occurrence.</li>
</ul>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="examples.html#cb373-1" aria-hidden="true" tabindex="-1"></a>core_taxa_name <span class="ot">&lt;-</span> <span class="fu">core_members</span>(metaphlan2_ps_final, <span class="at">detection =</span> <span class="fl">0.001</span>, <span class="at">prevalence =</span> <span class="fl">0.5</span>)</span>
<span id="cb373-2"><a href="examples.html#cb373-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(core_taxa_name)</span></code></pre></div>
<pre><code>##  [1] &quot;s__Bacteroides_ovatus&quot;                &quot;s__Bacteroides_thetaiotaomicron&quot;     
##  [3] &quot;s__Bacteroides_uniformis&quot;             &quot;s__Bacteroides_vulgatus&quot;             
##  [5] &quot;s__Bifidobacterium_longum&quot;            &quot;s__Bifidobacterium_pseudocatenulatum&quot;
##  [7] &quot;s__Collinsella_aerofaciens&quot;           &quot;s__Escherichia_coli&quot;                 
##  [9] &quot;s__Eubacterium_eligens&quot;               &quot;s__Eubacterium_hallii&quot;               
## [11] &quot;s__Faecalibacterium_prausnitzii&quot;      &quot;s__Roseburia_inulinivorans&quot;          
## [13] &quot;s__Ruminococcus_gnavus&quot;               &quot;s__Ruminococcus_obeum&quot;               
## [15] &quot;s__Ruminococcus_sp_5_1_39BFAA&quot;        &quot;s__Ruminococcus_torques&quot;             
## [17] &quot;s__Streptococcus_salivarius&quot;</code></pre>
<p><strong>Result</strong>:</p>
<p>17 species passed the threshold of detection and prevalence which we choose.</p>
</div>
</div>
<div id="differential-analysis-2" class="section level3 hasAnchor" number="13.3.7">
<h3><span class="header-section-number">13.3.7</span> Differential Analysis<a href="examples.html#differential-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to <strong>Chapter 10</strong> to see more detials.</p>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-3" class="section level4 hasAnchor" number="13.3.7.1">
<h4><span class="header-section-number">13.3.7.1</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="examples.html#liner-discriminant-analysis-lda-effect-size-lefse-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="examples.html#cb375-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb375-2"><a href="examples.html#cb375-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_final,</span>
<span id="cb375-3"><a href="examples.html#cb375-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb375-4"><a href="examples.html#cb375-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb375-5"><a href="examples.html#cb375-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;identity&quot;</span>,</span>
<span id="cb375-6"><a href="examples.html#cb375-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb375-7"><a href="examples.html#cb375-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">wl.p =</span> <span class="fl">0.05</span>,</span>
<span id="cb375-8"><a href="examples.html#cb375-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb375-9"><a href="examples.html#cb375-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_lefse)</span></code></pre></div>
<pre><code>##                            TaxaID         Block Enrichment     LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB
## 1 s__Bacteroides_thetaiotaomicron 7_AA vs 15_BB         AA -0.0097587531  1.9418180                          5.505858
## 2            s__Dorea_longicatena 7_AA vs 15_BB         BB  0.0003492349  1.7435380                                NA
## 3       s__Clostridium_bartlettii 7_AA vs 15_BB         BB  0.0006298963  0.5784397                                NA
## 4           s__Eubacterium_hallii 7_AA vs 15_BB         BB  0.0021243839  0.7995960                                NA
## 5        s__Megamonas_rupellensis 7_AA vs 15_BB         BB  0.0024505506  0.5938727                                NA
## 6      s__Collinsella_aerofaciens 7_AA vs 15_BB         BB  0.0026197908  0.9034469                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB
## 1              0.00247415            0.0453799            0.0009987                        3.327766
## 2              0.00078795            0.0000000            0.0081496                       -7.307095
## 3              0.00025355            0.0000000            0.0004470                       -3.821895
## 4              0.00436885            0.0000000            0.0109294                       -3.781465
## 5              0.00000000            0.0005600            0.0000000                        5.020069
## 6              0.00738145            0.0000000            0.0112480                       -3.336571
##   Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA
## 1          0.0219183273       5.677024e-02       5.654100e-03                    90.91                100.00
## 2          0.0065455500       6.044286e-05       9.571933e-03                    63.64                 42.86
## 3          0.0012202818       1.225143e-04       1.732573e-03                    68.18                 42.86
## 4          0.0087429000       9.019000e-04       1.240203e-02                    63.64                 42.86
## 5          0.0008989182       2.650157e-03       8.167333e-05                    31.82                 57.14
## 6          0.0118099818       1.638929e-03       1.655647e-02                    63.64                 42.86
##   Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                 86.67          0.038 (-6.4;6.4)
## 2                 73.33 8.9e+12 (8.9e+12;8.9e+12)
## 3                 80.00             600 (610;580)
## 4                 73.33             110 (120;100)
## 5                 20.00           0.0055 (-10;10)
## 6                 73.33                41 (48;34)</code></pre>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="examples.html#cb377-1" aria-hidden="true" tabindex="-1"></a><span class="co"># in order to show barplot (don&#39;t run it in reality)</span></span>
<span id="cb377-2"><a href="examples.html#cb377-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_lefse<span class="sc">$</span>LDA_Score <span class="ot">&lt;-</span> metaphlan2_ps_lefse<span class="sc">$</span>LDA_Score <span class="sc">*</span> <span class="dv">1000</span></span>
<span id="cb377-3"><a href="examples.html#cb377-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb377-4"><a href="examples.html#cb377-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb377-5"><a href="examples.html#cb377-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">da_res =</span> metaphlan2_ps_lefse,</span>
<span id="cb377-6"><a href="examples.html#cb377-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb377-7"><a href="examples.html#cb377-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb377-8"><a href="examples.html#cb377-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-204"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-204-1.png" alt="Lefse analysis (MGS example)" width="768" />
<p class="caption">
Figure 13.18: Lefse analysis (MGS example)
</p>
</div>
</div>
<div id="wilcoxon-rank-sum-test-1" class="section level4 hasAnchor" number="13.3.7.2">
<h4><span class="header-section-number">13.3.7.2</span> Wilcoxon Rank-Sum test<a href="examples.html#wilcoxon-rank-sum-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="examples.html#cb378-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb378-2"><a href="examples.html#cb378-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_final,</span>
<span id="cb378-3"><a href="examples.html#cb378-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb378-4"><a href="examples.html#cb378-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb378-5"><a href="examples.html#cb378-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">transform =</span> <span class="st">&quot;identity&quot;</span>,</span>
<span id="cb378-6"><a href="examples.html#cb378-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb378-7"><a href="examples.html#cb378-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">p_adjust =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb378-8"><a href="examples.html#cb378-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">pvalue_cutoff =</span> <span class="fl">0.05</span>)</span>
<span id="cb378-9"><a href="examples.html#cb378-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb378-10"><a href="examples.html#cb378-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_wilcox)</span></code></pre></div>
<pre><code>##                      TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue
## 1   s__Alistipes_finegoldii 7_AA vs 15_BB  Nonsignif 1.23094855      73.5 0.1145448      0.4521504
## 2 s__Alistipes_indistinctus 7_AA vs 15_BB  Nonsignif 0.78954179      66.0 0.2677815      0.6616137
## 3  s__Alistipes_onderdonkii 7_AA vs 15_BB  Nonsignif 1.14756258      63.5 0.4370143      0.7578592
## 4   s__Alistipes_putredinis 7_AA vs 15_BB  Nonsignif 1.34298152      54.0 0.9409918      0.9830621
## 5       s__Alistipes_shahii 7_AA vs 15_BB  Nonsignif 1.04736939      71.0 0.1751506      0.5801989
## 6    s__Anaerostipes_hadrus 7_AA vs 15_BB  Nonsignif 0.02734989      30.5 0.1258306      0.4625882
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA              0.00000000            0.0027362            0.0000000
## 2                                NA              0.00000000            0.0000000            0.0000000
## 3                          5.819157              0.00000825            0.0002823            0.0000050
## 4                                NA              0.00016915            0.0000000            0.0002200
## 5                                NA              0.00012580            0.0062474            0.0000000
## 6                         -3.900867              0.00013665            0.0000144            0.0002151
##   Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA Mean Rank Abundance\nBB Occurrence (100%)\n(All)
## 1                      0.52169761                   14.50                   10.10                    45.45
## 2                      0.34139504                   13.43                   10.60                    31.82
## 3                      0.27924089                   13.07                   10.77                    54.55
## 4                      0.03870725                   11.71                   11.40                    54.55
## 5                      0.46134594                   14.14                   10.27                    50.00
## 6                     -0.63360363                    8.36                   12.97                    72.73
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 57.14                 40.00     0.0028 (-12;12)
## 2                 42.86                 26.67    0.067 (-5.2;5.4)
## 3                 57.14                 53.33     0.38 (-1.5;2.3)
## 4                 42.86                 60.00    0.47 (-0.99;1.9)
## 5                 57.14                 46.67     0.17 (-3.2;3.6)
## 6                 57.14                 80.00          31 (38;24)</code></pre>
<ul>
<li>Volcano</li>
</ul>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="examples.html#cb380-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb380-2"><a href="examples.html#cb380-2" aria-hidden="true" tabindex="-1"></a>    metaphlan2_ps_wilcox,</span>
<span id="cb380-3"><a href="examples.html#cb380-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb380-4"><a href="examples.html#cb380-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb380-5"><a href="examples.html#cb380-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb380-6"><a href="examples.html#cb380-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb380-7"><a href="examples.html#cb380-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb380-8"><a href="examples.html#cb380-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb380-9"><a href="examples.html#cb380-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>,</span>
<span id="cb380-10"><a href="examples.html#cb380-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">taxa_name =</span> <span class="cn">NULL</span></span>
<span id="cb380-11"><a href="examples.html#cb380-11" aria-hidden="true" tabindex="-1"></a>   )</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-206"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-206-1.png" alt="Wilcoxon Rank-Sum test (MGS example)" width="768" />
<p class="caption">
Figure 13.19: Wilcoxon Rank-Sum test (MGS example)
</p>
</div>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visualization-5.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/19-examples.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
