<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Examples | Data analysis based on XMAS 2.0 package in microbiota data</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Examples | Data analysis based on XMAS 2.0 package in microbiota data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Examples | Data analysis based on XMAS 2.0 package in microbiota data" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-05-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="visualization-5.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation<span></span></a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette<span></span></a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors<span></span></a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Reads’ track<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics<span></span></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.6</b> Analysis of Similarity (ANOSIM)<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.7</b> Ordination<span></span></a>
<ul>
<li class="chapter" data-level="8.7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.7.1</b> Principal Component Analysis (PCA)<span></span></a></li>
<li class="chapter" data-level="8.7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.7.2</b> Principal Coordinate Analysis (PCoA)<span></span></a></li>
<li class="chapter" data-level="8.7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.7.3</b> t-distributed stochastic neighbor embedding (t-SNE)<span></span></a></li>
<li class="chapter" data-level="8.7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.7.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.8</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="composition.html"><a href="composition.html"><i class="fa fa-check"></i><b>9</b> Composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="composition.html"><a href="composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="composition.html"><a href="composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.2</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="composition.html"><a href="composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.3</b> XVIZ <code>plot_stacked_bar</code><span></span></a></li>
<li class="chapter" data-level="9.4" data-path="composition.html"><a href="composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="composition.html"><a href="composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus<span></span></a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq<span></span></a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR<span></span></a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM<span></span></a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob<span></span></a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)<span></span></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom-1"><i class="fa fa-check"></i><b>11.2.1</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.2</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.3</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.4</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.2.5" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.5</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function<span></span></a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.6</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot<span></span></a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot<span></span></a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot<span></span></a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density<span></span></a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination<span></span></a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot<span></span></a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot<span></span></a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn<span></span></a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.10</b> plot_stacked_bar_XIVZ<span></span></a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.11</b> plot_StackBarPlot<span></span></a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.12</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>13</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>13.1</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>13.2</b> Amplicon sequencing (16s)<span></span></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>13.2.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="13.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>13.2.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="13.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>13.2.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="13.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>13.2.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="13.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>13.2.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="13.2.6" data-path="examples.html"><a href="examples.html#microbial-composition"><i class="fa fa-check"></i><b>13.2.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="13.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>13.2.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="examples.html"><a href="examples.html#metagenomics"><i class="fa fa-check"></i><b>13.3</b> Metagenomics<span></span></a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>13.3.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="13.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>13.3.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="13.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>13.3.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="13.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>13.3.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="13.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>13.3.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="13.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>13.3.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="13.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>13.3.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>13.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data analysis based on XMAS 2.0 package in microbiota data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="examples" class="section level1 hasAnchor" number="13">
<h1><span class="header-section-number">Chapter 13</span> Examples<a href="examples.html#examples" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here, we give users two examples to practice the data analysis workflow by <strong>XMAS 2.0</strong>. By the way, we also recommend users handling your own microbiota data in a reasonable manner when you utilize this package or workflow. Pay attention to whether your data fit the methods this package provided.</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="examples.html#cb352-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS)</span>
<span id="cb352-2"><a href="examples.html#cb352-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb352-3"><a href="examples.html#cb352-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb352-4"><a href="examples.html#cb352-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb352-5"><a href="examples.html#cb352-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb352-6"><a href="examples.html#cb352-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
<div id="workflow-description" class="section level2 hasAnchor" number="13.1">
<h2><span class="header-section-number">13.1</span> Workflow description<a href="examples.html#workflow-description" class="anchor-section" aria-label="Anchor link to header"></a></h2>
The standard data analysis of 16S and MGS data by <strong>XMAS 2.0</strong>.
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-190"></span>
<div id="htmlwidget-053f3400ccfb97821528" style="width:672px;height:768px;" class="grViz html-widget"></div>
<script type="application/json" data-for="htmlwidget-053f3400ccfb97821528">{"x":{"diagram":"digraph flowchart { \n\n# initiate graph\ngraph [layout = dot, rankdir = TB, label = \"[dada2_res|metadata|tree]\n\n\", labelloc = t]\n\n\n# global node settings\nnode [shape = rectangle, style = filled, fillcolor = Linen, fontname = arial, width = 4]\n \ntab1 [label = \"Importing data: import_dada2()/summarize_phyloseq()\"]  \ntab2 [label = \"Quality Control(RefCheck/Remove): summarize_taxa()/run_RefCheck()/get_GroupPhyloseq()\"]  \ntab3 [label = \"Quality Control(Rarefaction curves): plot_RarefCure()\"]\ntab4 [label = \"Quality Control(Reads Track): plot_Dada2Track()\"]  \ntab5 [label = \"Quality Control(Variability): plot_taxa_cv()/plot_read_distribution()\"]  \ntab6 [label = \"Data processing(Rarefy): normalize()\"]\ntab7 [label = \"Data processing(Extract specific taxonomic level): summarize_taxa()\"]  \ntab8 [label = \"Data processing(Count2Relative): summarize_taxa(absolute = FALSE)\"]  \ntab9 [label = \"Data processing(Filter&trim): run_filter()/run_trim()\"]\ntab10 [label = \"GlobalView(Alpha diveristy): run_alpha_diversity()/plot_boxplot()\"]  \ntab11 [label = \"GlobalView(Beta diveristy): run_beta_diversity()/run_permanova()/run_ordination()/plot_Ordination()\"]  \ntab12 [label = \"GlobalView(microbial composition): plot_StackBarPlot()/plot_core_taxa()/core_members()\"]\ntab13 [label = \"DifferentialAnalysis: run_lefse()/plot_lefse()/run_wilcox()/plot_volcano()\"] \n \ntab1 -> tab2 -> tab3 -> tab4 -> tab5 -> tab6 -> tab7 -> tab8 -> tab9 -> tab10 -> tab11 -> tab12 -> tab13;  \n}  ","config":{"engine":"dot","options":null}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 13.1: Functions of XMAS 2.0 in 16s and MGS
</p>
</div>
</div>
<div id="amplicon-sequencing-16s" class="section level2 hasAnchor" number="13.2">
<h2><span class="header-section-number">13.2</span> Amplicon sequencing (16s)<a href="examples.html#amplicon-sequencing-16s" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The upstream process is performed by in-house pipeline. and this example just shows how to perform downstream data analysis. In briefly, the example comprises the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Converting inputs into phyloseq object;</p></li>
<li><p>Quality Evaluation;</p></li>
<li><p>Pre-Processing Data;</p></li>
<li><p>Diversity analysis;</p></li>
<li><p>Ordination analysis;</p></li>
<li><p>Composition analysis;</p></li>
<li><p>Differential analysis.</p></li>
</ol>
<div id="converting-inputs-into-phyloseq-class-object" class="section level3 hasAnchor" number="13.2.1">
<h3><span class="header-section-number">13.2.1</span> Converting inputs into phyloseq-class object<a href="examples.html#converting-inputs-into-phyloseq-class-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>dada2</strong> result from standardized_analytics_workflow_R_function.</p>
<ol style="list-style-type: decimal">
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/xdada2/dada2_res.rds</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/fasta2tree/tree.nwk</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/metadata.txt</p></li>
</ol>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="examples.html#cb353-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dada2 results from in-house 16s pipeline</span></span>
<span id="cb353-2"><a href="examples.html#cb353-2" aria-hidden="true" tabindex="-1"></a>dada2_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb353-3"><a href="examples.html#cb353-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system.file</span>(</span>
<span id="cb353-4"><a href="examples.html#cb353-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dada2_res.rds&quot;</span>,</span>
<span id="cb353-5"><a href="examples.html#cb353-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span>    </span>
<span id="cb353-6"><a href="examples.html#cb353-6" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb353-7"><a href="examples.html#cb353-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb353-8"><a href="examples.html#cb353-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb353-9"><a href="examples.html#cb353-9" aria-hidden="true" tabindex="-1"></a><span class="co"># the metadata matches to dada2 result</span></span>
<span id="cb353-10"><a href="examples.html#cb353-10" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb353-11"><a href="examples.html#cb353-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb353-12"><a href="examples.html#cb353-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dada2_metadata.tsv&quot;</span>,</span>
<span id="cb353-13"><a href="examples.html#cb353-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span></span>
<span id="cb353-14"><a href="examples.html#cb353-14" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb353-15"><a href="examples.html#cb353-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb353-16"><a href="examples.html#cb353-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb353-17"><a href="examples.html#cb353-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb353-18"><a href="examples.html#cb353-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb353-19"><a href="examples.html#cb353-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb353-20"><a href="examples.html#cb353-20" aria-hidden="true" tabindex="-1"></a><span class="co"># tree file from dada2 reference data silva</span></span>
<span id="cb353-21"><a href="examples.html#cb353-21" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">read_tree</span>(</span>
<span id="cb353-22"><a href="examples.html#cb353-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system.file</span>(</span>
<span id="cb353-23"><a href="examples.html#cb353-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;tree.nwk&quot;</span>,</span>
<span id="cb353-24"><a href="examples.html#cb353-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span>    </span>
<span id="cb353-25"><a href="examples.html#cb353-25" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb353-26"><a href="examples.html#cb353-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb353-27"><a href="examples.html#cb353-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb353-28"><a href="examples.html#cb353-28" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> <span class="fu">import_dada2_taxa</span>(dada2_res<span class="sc">$</span>tax_tab)</span>
<span id="cb353-29"><a href="examples.html#cb353-29" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> dada2_res<span class="sc">$</span>seq_tab</span>
<span id="cb353-30"><a href="examples.html#cb353-30" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> sam_tab <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;seqID&quot;</span>)</span>
<span id="cb353-31"><a href="examples.html#cb353-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb353-32"><a href="examples.html#cb353-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Shouldn&#39;t use the Total Number as SampleID (wrong: 123456; right: X123456)</span></span>
<span id="cb353-33"><a href="examples.html#cb353-33" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(otu_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="fu">rownames</span>(otu_tab))</span>
<span id="cb353-34"><a href="examples.html#cb353-34" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="fu">rownames</span>(sam_tab))</span>
<span id="cb353-35"><a href="examples.html#cb353-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb353-36"><a href="examples.html#cb353-36" aria-hidden="true" tabindex="-1"></a>dada2_ps <span class="ot">&lt;-</span> <span class="fu">import_dada2</span>(<span class="at">seq_tab =</span> otu_tab, </span>
<span id="cb353-37"><a href="examples.html#cb353-37" aria-hidden="true" tabindex="-1"></a>                         <span class="at">tax_tab =</span> tax_tab, </span>
<span id="cb353-38"><a href="examples.html#cb353-38" aria-hidden="true" tabindex="-1"></a>                         <span class="at">sam_tab =</span> sam_tab, </span>
<span id="cb353-39"><a href="examples.html#cb353-39" aria-hidden="true" tabindex="-1"></a>                         <span class="at">phy_tree =</span> tree)</span>
<span id="cb353-40"><a href="examples.html#cb353-40" aria-hidden="true" tabindex="-1"></a>dada2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 24 samples ]
## sample_data() Sample Data:       [ 24 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
<p>Here, the phyloseq object comprises five components (<em>OTU Table</em>, <em>Sample Data</em>, <em>Taxonomy Table</em>, <em>Phylogenetic Tree</em> and <em>DNAStringSet</em>).</p>
<div id="summarize-phyloseq-class-object-1" class="section level4 hasAnchor" number="13.2.1.1">
<h4><span class="header-section-number">13.2.1.1</span> Summarize phyloseq-class object<a href="examples.html#summarize-phyloseq-class-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="examples.html#cb355-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(dada2_ps)</span>
<span id="cb355-2"><a href="examples.html#cb355-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Compositional = NO2</span></span>
<span id="cb355-3"><a href="examples.html#cb355-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 1] Min. number of reads = 511812] Max. number of reads = 936223] Total number of reads = 15025374] Average number of reads = 62605.70833333335] Median number of reads = 619157] Sparsity = 0.8653738839285716] Any OTU sum to 1 or less? NO8] Number of singletons = 09] Percent of OTUs that are singletons</span></span>
<span id="cb355-4"><a href="examples.html#cb355-4" aria-hidden="true" tabindex="-1"></a><span class="do">##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 1Group2</span></span>
<span id="cb355-5"><a href="examples.html#cb355-5" aria-hidden="true" tabindex="-1"></a><span class="do">## [[1]]</span></span>
<span id="cb355-6"><a href="examples.html#cb355-6" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;1] Min. number of reads = 51181&quot;</span></span>
<span id="cb355-7"><a href="examples.html#cb355-7" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-8"><a href="examples.html#cb355-8" aria-hidden="true" tabindex="-1"></a><span class="do">## [[2]]</span></span>
<span id="cb355-9"><a href="examples.html#cb355-9" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;2] Max. number of reads = 93622&quot;</span></span>
<span id="cb355-10"><a href="examples.html#cb355-10" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-11"><a href="examples.html#cb355-11" aria-hidden="true" tabindex="-1"></a><span class="do">## [[3]]</span></span>
<span id="cb355-12"><a href="examples.html#cb355-12" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;3] Total number of reads = 1502537&quot;</span></span>
<span id="cb355-13"><a href="examples.html#cb355-13" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-14"><a href="examples.html#cb355-14" aria-hidden="true" tabindex="-1"></a><span class="do">## [[4]]</span></span>
<span id="cb355-15"><a href="examples.html#cb355-15" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;4] Average number of reads = 62605.7083333333&quot;</span></span>
<span id="cb355-16"><a href="examples.html#cb355-16" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-17"><a href="examples.html#cb355-17" aria-hidden="true" tabindex="-1"></a><span class="do">## [[5]]</span></span>
<span id="cb355-18"><a href="examples.html#cb355-18" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;5] Median number of reads = 61915&quot;</span></span>
<span id="cb355-19"><a href="examples.html#cb355-19" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-20"><a href="examples.html#cb355-20" aria-hidden="true" tabindex="-1"></a><span class="do">## [[6]]</span></span>
<span id="cb355-21"><a href="examples.html#cb355-21" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;7] Sparsity = 0.865373883928571&quot;</span></span>
<span id="cb355-22"><a href="examples.html#cb355-22" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-23"><a href="examples.html#cb355-23" aria-hidden="true" tabindex="-1"></a><span class="do">## [[7]]</span></span>
<span id="cb355-24"><a href="examples.html#cb355-24" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;</span></span>
<span id="cb355-25"><a href="examples.html#cb355-25" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-26"><a href="examples.html#cb355-26" aria-hidden="true" tabindex="-1"></a><span class="do">## [[8]]</span></span>
<span id="cb355-27"><a href="examples.html#cb355-27" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;8] Number of singletons = 0&quot;</span></span>
<span id="cb355-28"><a href="examples.html#cb355-28" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-29"><a href="examples.html#cb355-29" aria-hidden="true" tabindex="-1"></a><span class="do">## [[9]]</span></span>
<span id="cb355-30"><a href="examples.html#cb355-30" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;</span></span>
<span id="cb355-31"><a href="examples.html#cb355-31" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-32"><a href="examples.html#cb355-32" aria-hidden="true" tabindex="-1"></a><span class="do">## [[10]]</span></span>
<span id="cb355-33"><a href="examples.html#cb355-33" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;10] Number of sample variables are: 1&quot;</span></span>
<span id="cb355-34"><a href="examples.html#cb355-34" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb355-35"><a href="examples.html#cb355-35" aria-hidden="true" tabindex="-1"></a><span class="do">## [[11]]</span></span>
<span id="cb355-36"><a href="examples.html#cb355-36" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;Group&quot;</span></span></code></pre></div>
<p>The minus account of the OTU counts is <em>51181</em> in the phyloseq object, and we can use it as the threshold to rarefy.</p>
<p>Notice the <strong>Sparsity (0.865)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.</p>
</div>
</div>
<div id="quality-control" class="section level3 hasAnchor" number="13.2.2">
<h3><span class="header-section-number">13.2.2</span> Quality Control<a href="examples.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Quality control of DADA2 results will help us have more rational determinations on the further data analysis.</p>
<div id="reads-track-by-dada2" class="section level4 hasAnchor" number="13.2.2.1">
<h4><span class="header-section-number">13.2.2.1</span> Reads’ track by DADA2<a href="examples.html#reads-track-by-dada2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="examples.html#cb356-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Dada2Track</span>(<span class="at">data=</span>dada2_res<span class="sc">$</span>reads_track)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-193"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-193-1.png" alt="Reads' track by DADA2 (16s example)" width="576" />
<p class="caption">
Figure 13.2: Reads’ track by DADA2 (16s example)
</p>
</div>
<p>The percentage of the final remained read counts approximate <strong>70%</strong>, indicating that we should consider the sequence depth for analysis when we build the sequence library.</p>
</div>
<div id="spike-in-sample-brs-assessment" class="section level4 hasAnchor" number="13.2.2.2">
<h4><span class="header-section-number">13.2.2.2</span> Spike-in sample (BRS) assessment<a href="examples.html#spike-in-sample-brs-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Extract the genus level phyloseq and getting the BRS_ID</li>
</ul>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="examples.html#cb357-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb357-2"><a href="examples.html#cb357-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_data</span>(dada2_ps_genus)</span></code></pre></div>
<pre><code>##       Group
## S6030    BB
## S6032    BB
## S6033    BB
## S6035    AA
## S6036    BB
## S6037    AA
## S6040    BB
## S6043    AA
## S6045    BB
## S6046    BB
## S6048    BB
## S6049    AA
## S6050    BB
## S6054    BB
## S6055    BB
## S6058    BB
## S6059    AA
## S6060    AA
## S6061    AA
## S6063    BB
## S6065    AA
## S6066    AA
## S6068    BB
## S8005    QC</code></pre>
<p><strong>The BRS_ID is <em>S8005</em> </strong>.</p>
<ul>
<li>Run <code>run_RefCheck</code></li>
</ul>
<div class="sourceCode" id="cb359"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb359-1"><a href="examples.html#cb359-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_RefCheck</span>(</span>
<span id="cb359-2"><a href="examples.html#cb359-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps=</span>dada2_ps_genus,</span>
<span id="cb359-3"><a href="examples.html#cb359-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">BRS_ID=</span><span class="st">&quot;S8005&quot;</span>,</span>
<span id="cb359-4"><a href="examples.html#cb359-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">Reference=</span><span class="cn">NULL</span>,</span>
<span id="cb359-5"><a href="examples.html#cb359-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Ref_type=</span><span class="st">&quot;16s&quot;</span>,</span>
<span id="cb359-6"><a href="examples.html#cb359-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">Save=</span><span class="cn">NULL</span>)</span>
<span id="cb359-7"><a href="examples.html#cb359-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Noting: the Reference Matrix is for 16s</span></span>
<span id="cb359-8"><a href="examples.html#cb359-8" aria-hidden="true" tabindex="-1"></a><span class="do">## S8005 is in the Reference Matrix&#39;s samples and remove it to run</span></span>
<span id="cb359-9"><a href="examples.html#cb359-9" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb359-10"><a href="examples.html#cb359-10" aria-hidden="true" tabindex="-1"></a><span class="do">## #########Matched baterica of the Spike-in sample###########</span></span>
<span id="cb359-11"><a href="examples.html#cb359-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking out all the taxa of the Reference Matrix are in the sequencing dataset </span></span>
<span id="cb359-12"><a href="examples.html#cb359-12" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Bifidobacterium</span></span>
<span id="cb359-13"><a href="examples.html#cb359-13" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Bacteroides</span></span>
<span id="cb359-14"><a href="examples.html#cb359-14" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Faecalibacterium</span></span>
<span id="cb359-15"><a href="examples.html#cb359-15" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Lactobacillus</span></span>
<span id="cb359-16"><a href="examples.html#cb359-16" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Parabacteroides</span></span>
<span id="cb359-17"><a href="examples.html#cb359-17" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Collinsella</span></span>
<span id="cb359-18"><a href="examples.html#cb359-18" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Coprococcus_3</span></span>
<span id="cb359-19"><a href="examples.html#cb359-19" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Dorea</span></span>
<span id="cb359-20"><a href="examples.html#cb359-20" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Streptococcus</span></span>
<span id="cb359-21"><a href="examples.html#cb359-21" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Roseburia</span></span>
<span id="cb359-22"><a href="examples.html#cb359-22" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Anaerostipes</span></span>
<span id="cb359-23"><a href="examples.html#cb359-23" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Escherichia/Shigella</span></span>
<span id="cb359-24"><a href="examples.html#cb359-24" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Enterococcus</span></span>
<span id="cb359-25"><a href="examples.html#cb359-25" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Prevotella_9</span></span>
<span id="cb359-26"><a href="examples.html#cb359-26" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Eggerthella</span></span>
<span id="cb359-27"><a href="examples.html#cb359-27" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb359-28"><a href="examples.html#cb359-28" aria-hidden="true" tabindex="-1"></a><span class="do">## The number of the additional Taxa in the sequencing dataset is 1</span></span>
<span id="cb359-29"><a href="examples.html#cb359-29" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb359-30"><a href="examples.html#cb359-30" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb359-31"><a href="examples.html#cb359-31" aria-hidden="true" tabindex="-1"></a><span class="do">## ##############Value of the Reference sample###############</span></span>
<span id="cb359-32"><a href="examples.html#cb359-32" aria-hidden="true" tabindex="-1"></a><span class="do">## Correlation Coefficient of the Reference sample is: 0.9714</span></span>
<span id="cb359-33"><a href="examples.html#cb359-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Bray Curtis of the Reference sample is: 0.07607</span></span>
<span id="cb359-34"><a href="examples.html#cb359-34" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity of the Reference sample is: 0.06409</span></span>
<span id="cb359-35"><a href="examples.html#cb359-35" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb359-36"><a href="examples.html#cb359-36" aria-hidden="true" tabindex="-1"></a><span class="do">## #####Final Evaluation Results of the Spike-in sample#######</span></span>
<span id="cb359-37"><a href="examples.html#cb359-37" aria-hidden="true" tabindex="-1"></a><span class="do">## The spike-in sample of sequencing dataset passed the cutoff of the Reference Matrix </span></span>
<span id="cb359-38"><a href="examples.html#cb359-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of Coefficient is 0.8946</span></span>
<span id="cb359-39"><a href="examples.html#cb359-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of BrayCurtis is 0.3878</span></span>
<span id="cb359-40"><a href="examples.html#cb359-40" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of Impurity is 0.1565</span></span>
<span id="cb359-41"><a href="examples.html#cb359-41" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb359-42"><a href="examples.html#cb359-42" aria-hidden="true" tabindex="-1"></a><span class="do">##                             8002        8003        8004       8006        8007        8008        8009        8005        mean</span></span>
<span id="cb359-43"><a href="examples.html#cb359-43" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium      31.11079015 30.88310969 32.31232692 18.4930259 20.20409870 17.96225391 18.03588291 27.22437034 24.52823232</span></span>
<span id="cb359-44"><a href="examples.html#cb359-44" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides          20.44753484 14.46581958 24.57151411 26.7370147 25.85863655 27.51353663 26.99272343 24.23896093 23.85321759</span></span>
<span id="cb359-45"><a href="examples.html#cb359-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Faecalibacterium      0.79850615  0.62937893  1.05531023  1.7487249  1.64282727  1.96346413  1.81219797  1.04035376  1.33634542</span></span>
<span id="cb359-46"><a href="examples.html#cb359-46" aria-hidden="true" tabindex="-1"></a><span class="do">## Lactobacillus         2.61732573  3.36856272  3.44379163  5.9292703  5.78000836  5.78189064  6.32672332  3.87088505  4.63980722</span></span>
<span id="cb359-47"><a href="examples.html#cb359-47" aria-hidden="true" tabindex="-1"></a><span class="do">## Parabacteroides       7.11124408  7.45952579  5.36075144  8.7149995  8.01840234  8.74899584  8.62634005  5.61833757  7.45732457</span></span>
<span id="cb359-48"><a href="examples.html#cb359-48" aria-hidden="true" tabindex="-1"></a><span class="do">## Collinsella           0.12792605  0.88271385  0.55665744  1.2764130  0.67921372  1.89356397  1.26367828  0.45502126  0.89189845</span></span>
<span id="cb359-49"><a href="examples.html#cb359-49" aria-hidden="true" tabindex="-1"></a><span class="do">## Coprococcus_3         1.00380683  0.97969362  0.80270938  1.6693557  1.56419908  1.67969035  1.71221463  0.87586251  1.28594151</span></span>
<span id="cb359-50"><a href="examples.html#cb359-50" aria-hidden="true" tabindex="-1"></a><span class="do">## Dorea                 2.80715148  3.45564660  2.46613277  3.9684605  3.99163530  3.81529666  3.69382881  2.34880690  3.31836988</span></span>
<span id="cb359-51"><a href="examples.html#cb359-51" aria-hidden="true" tabindex="-1"></a><span class="do">## Streptococcus         2.91960260  3.43387563  2.59149764  3.4818362  3.31409452  3.51378702  3.37721491  2.68740253  3.16491388</span></span>
<span id="cb359-52"><a href="examples.html#cb359-52" aria-hidden="true" tabindex="-1"></a><span class="do">## Roseburia             0.03404484  0.04750030  0.02806676  0.0338295  0.03178586  0.02503886  0.02499583  0.03311188  0.03229673</span></span>
<span id="cb359-53"><a href="examples.html#cb359-53" aria-hidden="true" tabindex="-1"></a><span class="do">## Anaerostipes          0.32291011  0.43245062  0.31528329  0.5386697  0.44500209  0.53416240  0.48047548  0.32471000  0.42420796</span></span>
<span id="cb359-54"><a href="examples.html#cb359-54" aria-hidden="true" tabindex="-1"></a><span class="do">## Escherichia/Shigella 15.27581475 16.00265210 12.36527954 10.8423545 13.43203680 10.15743185 11.59945565 14.03516268 12.96377349</span></span>
<span id="cb359-55"><a href="examples.html#cb359-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Enterococcus         14.51444842 14.66472707 11.04239952 13.1674820 12.07360937 12.88771113 12.84924735 11.61906390 12.85233610</span></span>
<span id="cb359-56"><a href="examples.html#cb359-56" aria-hidden="true" tabindex="-1"></a><span class="do">## Prevotella_9          0.77374627  3.07465463  2.75709154  2.9145415  2.52446675  3.04952478  2.84952508  5.40257632  2.91826586</span></span>
<span id="cb359-57"><a href="examples.html#cb359-57" aria-hidden="true" tabindex="-1"></a><span class="do">## Eggerthella           0.04951976  0.15437596  0.27131203  0.4840221  0.43998327  0.47365181  0.35549631  0.16128688  0.29870601</span></span>
<span id="cb359-58"><a href="examples.html#cb359-58" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity_level        0.08562792  0.06531291  0.05987576  0.0000000  0.00000000  0.00000000  0.00000000  0.06409000  0.03436332</span></span></code></pre></div>
</div>
<div id="spike-in-samples-remove" class="section level4 hasAnchor" number="13.2.2.3">
<h4><span class="header-section-number">13.2.2.3</span> Spike-in sample’s remove<a href="examples.html#spike-in-samples-remove" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="examples.html#cb360-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb360-2"><a href="examples.html#cb360-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb360-3"><a href="examples.html#cb360-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb360-4"><a href="examples.html#cb360-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb360-5"><a href="examples.html#cb360-5" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
</div>
<div id="rarefaction-curves-1" class="section level4 hasAnchor" number="13.2.2.4">
<h4><span class="header-section-number">13.2.2.4</span> Rarefaction curves<a href="examples.html#rarefaction-curves-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb362-1"><a href="examples.html#cb362-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_RarefCure</span>(<span class="at">ps =</span> dada2_ps_remove_BRS,</span>
<span id="cb362-2"><a href="examples.html#cb362-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">taxa_level =</span> <span class="st">&quot;OTU&quot;</span>,</span>
<span id="cb362-3"><a href="examples.html#cb362-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">step =</span> <span class="dv">100</span>,</span>
<span id="cb362-4"><a href="examples.html#cb362-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">label =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb362-5"><a href="examples.html#cb362-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## rarefying sample S6030
## rarefying sample S6032
## rarefying sample S6033
## rarefying sample S6035
## rarefying sample S6036
## rarefying sample S6037
## rarefying sample S6040
## rarefying sample S6043
## rarefying sample S6045
## rarefying sample S6046
## rarefying sample S6048
## rarefying sample S6049
## rarefying sample S6050
## rarefying sample S6054
## rarefying sample S6055
## rarefying sample S6058
## rarefying sample S6059
## rarefying sample S6060
## rarefying sample S6061
## rarefying sample S6063
## rarefying sample S6065
## rarefying sample S6066
## rarefying sample S6068</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-197"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-197-1.png" alt="Rarefaction curves (16s example)" width="576" />
<p class="caption">
Figure 13.3: Rarefaction curves (16s example)
</p>
</div>
<p>The result showed that all the samples had <strong>different sequence depth</strong> but had the full sample richness.</p>
</div>
</div>
<div id="data-processing" class="section level3 hasAnchor" number="13.2.3">
<h3><span class="header-section-number">13.2.3</span> Data processing<a href="examples.html#data-processing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This part has too may procedures and we only choose some of them. Please go to <strong>Chapter 6</strong> to see more approaches and details for being familiar with this part.</p>
<div id="rarefy-otu-counts" class="section level4 hasAnchor" number="13.2.3.1">
<h4><span class="header-section-number">13.2.3.1</span> Rarefy otu counts<a href="examples.html#rarefy-otu-counts" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>From previous results of quality evaluation, the sequence depth of samples are different which have effects on the downstream analysis. Here, choosing the <strong>rarefy</strong> (Normaliztion method: random subsampling counts to the smallest library size) to get the equal sample sums.</p>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="examples.html#cb364-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(dada2_ps_remove_BRS, <span class="at">size =</span> <span class="dv">51181</span>, <span class="at">rng_seed =</span> <span class="dv">123</span>)</span>
<span id="cb364-2"><a href="examples.html#cb364-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_rarefy</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 892 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 892 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 892 tips and 889 internal nodes ]
## refseq()      DNAStringSet:      [ 892 reference sequences ]</code></pre>
<p>In addition, we could also perform some other normalization methods on the rarefied phyloseq object. By the way, we didn’t transform the data by using log algorithm because the count matrix is required by the following data analysis methods.</p>
</div>
<div id="extracting-specific-taxonomic-level" class="section level4 hasAnchor" number="13.2.3.2">
<h4><span class="header-section-number">13.2.3.2</span> Extracting specific taxonomic level<a href="examples.html#extracting-specific-taxonomic-level" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="examples.html#cb366-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(dada2_ps_rarefy, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb366-2"><a href="examples.html#cb366-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 198 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 198 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10" class="section level4 hasAnchor" number="13.2.3.3">
<h4><span class="header-section-number">13.2.3.3</span> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 10)<a href="examples.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="examples.html#cb368-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(dada2_ps_rare_genus, <span class="at">cutoff =</span> <span class="dv">10</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb368-2"><a href="examples.html#cb368-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter </span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 149 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 149 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="trimming-the-taxa-with-low-occurrence-less-than-threshold" class="section level4 hasAnchor" number="13.2.3.4">
<h4><span class="header-section-number">13.2.3.4</span> Trimming the taxa with low occurrence less than threshold<a href="examples.html#trimming-the-taxa-with-low-occurrence-less-than-threshold" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb370-1"><a href="examples.html#cb370-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(dada2_ps_rare_genus_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb370-2"><a href="examples.html#cb370-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 99 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 99 taxa by 6 taxonomic ranks ]</code></pre>
<p>Finally, we obtained the final phyloseq-class object <em>dada2_ps_rare_genus_filter_trim</em> and changed its name.</p>
</div>
</div>
<div id="diversity-analysis" class="section level3 hasAnchor" number="13.2.4">
<h3><span class="header-section-number">13.2.4</span> Diversity analysis<a href="examples.html#diversity-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="alpha-diveristy" class="section level4 hasAnchor" number="13.2.4.1">
<h4><span class="header-section-number">13.2.4.1</span> Alpha diveristy<a href="examples.html#alpha-diveristy" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculate the alpha diversity</li>
</ul>
<p><strong>Notes:</strong> the otu table must be counts matrix (<em>rarefied but not trim counts matrix</em>) when you choose <em>Observed</em> etc measures.</p>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb372-1"><a href="examples.html#cb372-1" aria-hidden="true" tabindex="-1"></a>dada_ps_rare_genus_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps=</span>dada2_ps_rare_genus, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>))</span>
<span id="cb372-2"><a href="examples.html#cb372-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(dada_ps_rare_genus_alpha)</span></code></pre></div>
<pre><code>##    SampleID Group Observed Chao1  se.chao1  Shannon
## 1     S6030    BB       74    74 0.1241525 2.620338
## 2     S6032    BB       38    38 0.0000000 1.710464
## 3     S6033    BB       80    80 0.0000000 2.717826
## 4     S6035    AA       36    36 0.0000000 1.947824
## 5     S6036    BB       95    95 0.0000000 2.941189
## 6     S6037    AA       35    35 0.0000000 1.933868
## 7     S6040    BB       58    58 0.0000000 2.493444
## 8     S6043    AA       65    65 0.0000000 2.104350
## 9     S6045    BB       60    60 0.0000000 2.918472
## 10    S6046    BB       70    70 0.0000000 2.264455
## 11    S6048    BB       66    66 0.0000000 2.630083
## 12    S6049    AA       60    60 0.0000000 2.511144
## 13    S6050    BB       53    53 0.0000000 2.431334
## 14    S6054    BB       84    84 0.0000000 2.773183
## 15    S6055    BB       75    75 0.4966555 2.293124
## 16    S6058    BB       37    37 0.0000000 2.260300
## 17    S6059    AA       71    71 0.0000000 2.683334
## 18    S6060    AA       48    48 0.1236911 1.882097
## 19    S6061    AA       80    80 0.0000000 3.085755
## 20    S6063    BB       38    38 0.0000000 1.538945
## 21    S6065    AA       77    77 0.0000000 2.878531
## 22    S6066    AA       67    67 0.0000000 2.638335
## 23    S6068    BB       40    40 0.0000000 2.099203</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb374"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb374-1"><a href="examples.html#cb374-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data=</span>dada_ps_rare_genus_alpha,</span>
<span id="cb374-2"><a href="examples.html#cb374-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb374-3"><a href="examples.html#cb374-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb374-4"><a href="examples.html#cb374-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb374-5"><a href="examples.html#cb374-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb374-6"><a href="examples.html#cb374-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-203"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-203-1.png" alt="Alpha diversity (16s example)" width="576" />
<p class="caption">
Figure 13.4: Alpha diversity (16s example)
</p>
</div>
</div>
<div id="beta-diversity-1" class="section level4 hasAnchor" number="13.2.4.2">
<h4><span class="header-section-number">13.2.4.2</span> Beta diversity<a href="examples.html#beta-diversity-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>beta dipersion</li>
</ul>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="examples.html#cb375-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>dada2_ps_rare_genus_filter_trim, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
## Groups     1 0.014165 0.0141653 1.6825    999    0.2
## Residuals 21 0.176808 0.0084194                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         AA   BB
## AA         0.21
## BB 0.20867</code></pre>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="examples.html#cb377-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_beta<span class="sc">$</span>BetaDispersion</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-204"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-204-1.png" alt="Beta diversity (16s example)" width="480" />
<p class="caption">
Figure 13.5: Beta diversity (16s example)
</p>
</div>
</div>
</div>
<div id="permanova-ordination" class="section level3 hasAnchor" number="13.2.5">
<h3><span class="header-section-number">13.2.5</span> PERMANOVA + Ordination<a href="examples.html#permanova-ordination" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="permanova" class="section level4 hasAnchor" number="13.2.5.1">
<h4><span class="header-section-number">13.2.5.1</span> PERMANOVA<a href="examples.html#permanova" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="examples.html#cb378-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_per <span class="ot">&lt;-</span> <span class="fu">run_permanova</span>(<span class="at">ps=</span>dada2_ps_rare_genus_filter_trim, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">columns =</span> <span class="st">&quot;Group&quot;</span>)</span>
<span id="cb378-2"><a href="examples.html#cb378-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(dada2_ps_per)</span></code></pre></div>
<pre><code>##       SumsOfSample Df SumsOfSqs   MeanSqs  F.Model         R2 Pr(&gt;F) AdjustedPvalue
## Group           23  1 0.2300226 0.2300226 1.337627 0.05988222  0.219          0.219</code></pre>
<p>The PERMANOVA result of the Group (<code>Pr(&gt;F)</code> &gt; 0.05) revealed that the two groups had not the distinct patterns of microbial community.</p>
</div>
<div id="ordination-1" class="section level4 hasAnchor" number="13.2.5.2">
<h4><span class="header-section-number">13.2.5.2</span> Ordination<a href="examples.html#ordination-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We performed ordination by using Principal Coordinate Analysis (PCoA). If you want to try other methods please go to see <strong>Chapter 8</strong> for more details.</p>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="examples.html#cb380-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb380-2"><a href="examples.html#cb380-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ps =</span> dada2_ps_rare_genus_filter_trim,</span>
<span id="cb380-3"><a href="examples.html#cb380-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb380-4"><a href="examples.html#cb380-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb380-5"><a href="examples.html#cb380-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb380-6"><a href="examples.html#cb380-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> dada2_ps_ordination, </span>
<span id="cb380-7"><a href="examples.html#cb380-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb380-8"><a href="examples.html#cb380-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb380-9"><a href="examples.html#cb380-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-206"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-206-1.png" alt="PCoA (16s example)" width="768" />
<p class="caption">
Figure 13.6: PCoA (16s example)
</p>
</div>
</div>
</div>
<div id="microbial-composition" class="section level3 hasAnchor" number="13.2.6">
<h3><span class="header-section-number">13.2.6</span> Microbial composition<a href="examples.html#microbial-composition" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A whole picture of the microbial composition.</p>
<div id="stacked-barplot-1" class="section level4 hasAnchor" number="13.2.6.1">
<h4><span class="header-section-number">13.2.6.1</span> Stacked barplot<a href="examples.html#stacked-barplot-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>XMAS</strong> package</li>
</ul>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="examples.html#cb381-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb381-2"><a href="examples.html#cb381-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">ps=</span>dada2_ps_rarefy,</span>
<span id="cb381-3"><a href="examples.html#cb381-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">taxa_level=</span><span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb381-4"><a href="examples.html#cb381-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb381-5"><a href="examples.html#cb381-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-207"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-207-1.png" alt="Microbial composition (16s example)" width="768" />
<p class="caption">
Figure 13.7: Microbial composition (16s example)
</p>
</div>
<ul>
<li><strong>XVIZ</strong> package</li>
</ul>
<div class="sourceCode" id="cb382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb382-1"><a href="examples.html#cb382-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(</span>
<span id="cb382-2"><a href="examples.html#cb382-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">phyloseq=</span>dada2_ps_rarefy,</span>
<span id="cb382-3"><a href="examples.html#cb382-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">level=</span><span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb382-4"><a href="examples.html#cb382-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-208"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-208-1.png" alt="Microbial composition (16s example) XVIZ" width="768" />
<p class="caption">
Figure 13.8: Microbial composition (16s example) XVIZ
</p>
</div>
</div>
<div id="core-microbiota-1" class="section level4 hasAnchor" number="13.2.6.2">
<h4><span class="header-section-number">13.2.6.2</span> Core microbiota<a href="examples.html#core-microbiota-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>convert absolute abundance into relative abundance</li>
</ul>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="examples.html#cb383-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim_rb <span class="ot">&lt;-</span> <span class="fu">normalize</span>(dada2_ps_rare_genus_filter_trim, <span class="at">method =</span> <span class="st">&quot;TSS&quot;</span>)</span>
<span id="cb383-2"><a href="examples.html#cb383-2" aria-hidden="true" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim_rb</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 99 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 99 taxa by 6 taxonomic ranks ]</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="examples.html#cb385-1" aria-hidden="true" tabindex="-1"></a>prevalences <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.05</span>, <span class="dv">1</span>, <span class="fl">0.05</span>)</span>
<span id="cb385-2"><a href="examples.html#cb385-2" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="fu">seq</span>(<span class="fu">log10</span>(<span class="fl">1e-3</span>), <span class="fu">log10</span>(<span class="fl">0.2</span>), <span class="at">length =</span> <span class="dv">10</span>)</span>
<span id="cb385-3"><a href="examples.html#cb385-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb385-4"><a href="examples.html#cb385-4" aria-hidden="true" tabindex="-1"></a>pl_core <span class="ot">&lt;-</span> <span class="fu">plot_core_taxa</span>(dada2_ps_rare_genus_filter_trim_rb, </span>
<span id="cb385-5"><a href="examples.html#cb385-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, </span>
<span id="cb385-6"><a href="examples.html#cb385-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colours =</span> <span class="fu">gray</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length=</span><span class="dv">5</span>)),</span>
<span id="cb385-7"><a href="examples.html#cb385-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prevalences =</span> prevalences, </span>
<span id="cb385-8"><a href="examples.html#cb385-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">detections =</span> detections, </span>
<span id="cb385-9"><a href="examples.html#cb385-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">min.prevalence =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb385-10"><a href="examples.html#cb385-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)</span>
<span id="cb385-11"><a href="examples.html#cb385-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb385-12"><a href="examples.html#cb385-12" aria-hidden="true" tabindex="-1"></a>pl_core</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-210"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-210-1.png" alt="Core taxa (16s example)" width="768" />
<p class="caption">
Figure 13.9: Core taxa (16s example)
</p>
</div>
<p>The degree of color indicates the size of abundance and prevalence.</p>
<ul>
<li>Use <code>core_members</code> to obtain the core taxa. detection for abundance and prevalence for occurrence.</li>
</ul>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb386-1"><a href="examples.html#cb386-1" aria-hidden="true" tabindex="-1"></a>core_taxa_name <span class="ot">&lt;-</span> <span class="fu">core_members</span>(dada2_ps_rare_genus_filter_trim_rb, <span class="at">detection =</span> <span class="fl">0.01</span>, <span class="at">prevalence =</span> <span class="fl">0.8</span>)</span>
<span id="cb386-2"><a href="examples.html#cb386-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(core_taxa_name)</span></code></pre></div>
<pre><code>## [1] &quot;g__Bifidobacterium&quot; &quot;g__Blautia&quot;</code></pre>
<p><strong>Result</strong>:</p>
<p>Only 2 genera (g__Bifidobacterium and g__Blautia) passed the threshold of detection and prevalence which we choose.</p>
</div>
</div>
<div id="differential-analysis-1" class="section level3 hasAnchor" number="13.2.7">
<h3><span class="header-section-number">13.2.7</span> Differential Analysis<a href="examples.html#differential-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to <strong>Chapter 10</strong> to see more detials.</p>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-2" class="section level4 hasAnchor" number="13.2.7.1">
<h4><span class="header-section-number">13.2.7.1</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="examples.html#liner-discriminant-analysis-lda-effect-size-lefse-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="examples.html#cb388-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb388-2"><a href="examples.html#cb388-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_rare_genus_filter_trim,</span>
<span id="cb388-3"><a href="examples.html#cb388-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb388-4"><a href="examples.html#cb388-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb388-5"><a href="examples.html#cb388-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb388-6"><a href="examples.html#cb388-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dada2_ps_lefse)</span></code></pre></div>
<pre><code>##                       TaxaID         Block Enrichment LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB
## 1          g__Parasutterella 9_AA vs 14_BB         AA -2.278934  1.1118869                          5.029747
## 2 g__Ruminococcaceae_UCG_004 9_AA vs 14_BB         AA -1.091599  1.0652610                                NA
## 3             g__Odoribacter 9_AA vs 14_BB         BB  1.442442  0.6595894                                NA
## 4              g__Romboutsia 9_AA vs 14_BB         BB  2.375824  2.2507136                         -6.499846
## 5           g__Lactobacillus 9_AA vs 14_BB         BB  2.690182  2.2952990                         -2.870821
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                       7                   49                  1.5                        4.244838             164.56522
## 2                      11                  107                  0.0                        2.211165              66.91304
## 3                       3                   30                  0.0                        2.050293              26.43478
## 4                     105                    3                271.5                       -3.197677             295.47826
## 5                      57                   27                197.5                       -5.067893             799.95652
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1          388.66667           20.50000                    65.22                 88.89                 50.00         0.01 (-9;9)
## 2          128.00000           27.64286                    52.17                 77.78                 35.71     0.19 (-3.1;3.5)
## 3           49.11111           11.85714                    52.17                 77.78                 35.71     0.33 (-1.9;2.5)
## 4           49.44444          453.64286                    78.26                 55.56                 92.86          42 (49;34)
## 5           38.44444         1289.50000                    86.96                 77.78                 92.86 4e+08 (4e+08;4e+08)</code></pre>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb390"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb390-1"><a href="examples.html#cb390-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb390-2"><a href="examples.html#cb390-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">da_res =</span> dada2_ps_lefse,</span>
<span id="cb390-3"><a href="examples.html#cb390-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb390-4"><a href="examples.html#cb390-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb390-5"><a href="examples.html#cb390-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-213"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-213-1.png" alt="Lefse analysis (16s example)" width="576" />
<p class="caption">
Figure 13.10: Lefse analysis (16s example)
</p>
</div>
</div>
<div id="wilcoxon-rank-sum-test" class="section level4 hasAnchor" number="13.2.7.2">
<h4><span class="header-section-number">13.2.7.2</span> Wilcoxon Rank-Sum test<a href="examples.html#wilcoxon-rank-sum-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="examples.html#cb391-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb391-2"><a href="examples.html#cb391-2" aria-hidden="true" tabindex="-1"></a>                    dada2_ps_rare_genus_filter_trim,</span>
<span id="cb391-3"><a href="examples.html#cb391-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb391-4"><a href="examples.html#cb391-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb391-5"><a href="examples.html#cb391-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb391-6"><a href="examples.html#cb391-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dada2_ps_wilcox)</span></code></pre></div>
<pre><code>##             TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1   g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.87195760        45 0.2689844      0.6737725                        -1.1699250
## 2 g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.57627740        44 0.1981520      0.6737725                                NA
## 3  g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.10803167        71 0.6293971      0.9129068                         0.1772545
## 4     g__Alistipes 9_AA vs 14_BB  Nonsignif 0.07097212        67 0.8239507      0.9129068                        -0.4280937
## 5  g__Anaerostipes 9_AA vs 14_BB  Nonsignif 0.40991461        58 0.7767603      0.9129068                         1.0204641
## 6   g__Bacteroides 9_AA vs 14_BB  Nonsignif 0.93028412        89 0.1093014      0.6737725                         1.5520348
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1                      30                   22                 49.5                     -0.41034110                   10.00
## 2                       0                    0                  9.5                     -0.43387758                    9.89
## 3                     311                  333                294.5                      0.17342686                   12.89
## 4                      64                   55                 74.0                      0.08724541                   12.44
## 5                     515                  923                455.0                     -0.11159169                   11.44
## 6                    3806                 9118               3109.5                      0.55428610                   14.89
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   13.29                    82.61                 88.89                 78.57       4.3 (7.1;1.4)
## 2                   13.36                    43.48                 33.33                 50.00           13 (18;8)
## 3                   11.43                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 4                   11.71                    73.91                 77.78                 71.43     0.87 (0.59;1.1)
## 5                   12.36                    95.65                100.00                 92.86      1.1 (1.2;0.93)
## 6                   10.14                   100.00                100.00                100.00       0.45 (-1.1;2)</code></pre>
<ul>
<li>Volcano</li>
</ul>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="examples.html#cb393-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb393-2"><a href="examples.html#cb393-2" aria-hidden="true" tabindex="-1"></a>    dada2_ps_wilcox,</span>
<span id="cb393-3"><a href="examples.html#cb393-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb393-4"><a href="examples.html#cb393-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb393-5"><a href="examples.html#cb393-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb393-6"><a href="examples.html#cb393-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb393-7"><a href="examples.html#cb393-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb393-8"><a href="examples.html#cb393-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb393-9"><a href="examples.html#cb393-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-215"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-215-1.png" alt="Wilcoxon Rank-Sum test (16s example)" width="768" />
<p class="caption">
Figure 13.11: Wilcoxon Rank-Sum test (16s example)
</p>
</div>
</div>
</div>
</div>
<div id="metagenomics" class="section level2 hasAnchor" number="13.3">
<h2><span class="header-section-number">13.3</span> Metagenomics<a href="examples.html#metagenomics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The metagenomic data analysis pipeline is just the same as 16s. In briefly, the example comprises the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Converting inputs into phyloseq object;</p></li>
<li><p>Quality Evaluation;</p></li>
<li><p>Pre-Processing Data;</p></li>
<li><p>Diversity analysis;</p></li>
<li><p>Ordination analysis;</p></li>
<li><p>Composition analysis;</p></li>
</ol>
<div id="converting-inputs-into-phyloseq-class-object-1" class="section level3 hasAnchor" number="13.3.1">
<h3><span class="header-section-number">13.3.1</span> Converting inputs into phyloseq-class object<a href="examples.html#converting-inputs-into-phyloseq-class-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The result of the in-house <strong><a href="https://huttenhower.sph.harvard.edu/metaphlan2">Metaphlan2/3</a></strong> pipeline:</p>
<ol style="list-style-type: decimal">
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metaphlan2_merged.tsv</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metadata.txt</p></li>
</ol>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="examples.html#cb394-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_res <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb394-2"><a href="examples.html#cb394-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb394-3"><a href="examples.html#cb394-3" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;metaphlan2_merged.tsv&quot;</span>,</span>
<span id="cb394-4"><a href="examples.html#cb394-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span></span>
<span id="cb394-5"><a href="examples.html#cb394-5" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb394-6"><a href="examples.html#cb394-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb394-7"><a href="examples.html#cb394-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb394-8"><a href="examples.html#cb394-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb394-9"><a href="examples.html#cb394-9" aria-hidden="true" tabindex="-1"></a>metaphlan2_sam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb394-10"><a href="examples.html#cb394-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb394-11"><a href="examples.html#cb394-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;metaphlan2_metadata.tsv&quot;</span>,</span>
<span id="cb394-12"><a href="examples.html#cb394-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS&quot;</span></span>
<span id="cb394-13"><a href="examples.html#cb394-13" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb394-14"><a href="examples.html#cb394-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb394-15"><a href="examples.html#cb394-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb394-16"><a href="examples.html#cb394-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb394-17"><a href="examples.html#cb394-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb394-18"><a href="examples.html#cb394-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb394-19"><a href="examples.html#cb394-19" aria-hidden="true" tabindex="-1"></a>metaphlan2_res_list <span class="ot">&lt;-</span> <span class="fu">import_metaphlan_taxa</span>(metaphlan2_res, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb394-20"><a href="examples.html#cb394-20" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>abu_tab</span>
<span id="cb394-21"><a href="examples.html#cb394-21" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>tax_tab</span>
<span id="cb394-22"><a href="examples.html#cb394-22" aria-hidden="true" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> metaphlan2_sam <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb394-23"><a href="examples.html#cb394-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb394-24"><a href="examples.html#cb394-24" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps <span class="ot">&lt;-</span> <span class="fu">import_metaphlan2</span>(<span class="at">otu_tab =</span> otu_tab, </span>
<span id="cb394-25"><a href="examples.html#cb394-25" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">sam_tab =</span> sam_tab,</span>
<span id="cb394-26"><a href="examples.html#cb394-26" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">tax_tab =</span> tax_tab)</span>
<span id="cb394-27"><a href="examples.html#cb394-27" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 326 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 326 taxa by 7 taxonomic ranks ]</code></pre>
<p>Here, the phyloseq object comprises three components (<em>OTU Table</em>, <em>Sample Data</em> and <em>Taxonomy Table</em>).</p>
<div id="summarize-phyloseq-class-object-2" class="section level4 hasAnchor" number="13.3.1.1">
<h4><span class="header-section-number">13.3.1.1</span> Summarize phyloseq-class object<a href="examples.html#summarize-phyloseq-class-object-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="examples.html#cb396-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(metaphlan2_ps)</span></code></pre></div>
<pre><code>## Compositional = NO2</code></pre>
<pre><code>## 1] Min. number of reads = 0.98218212] Max. number of reads = 1.00000053] Total number of reads = 22.94511414] Average number of reads = 0.9976136565217395] Median number of reads = 0.99980247] Sparsity = 0.7131234995998936] Any OTU sum to 1 or less? YES8] Number of singletons = 3239] Percent of OTUs that are singletons
##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 2Groupphynotype2</code></pre>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 0.9821821&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 1.0000005&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 22.9451141&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 0.997613656521739&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 0.9998024&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.713123499599893&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? YES&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 323&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 2&quot;
## 
## [[11]]
## [1] &quot;Group&quot;     &quot;phynotype&quot;</code></pre>
<p>Notice the <strong>Sparsity (0.713)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis.</p>
</div>
</div>
<div id="quality-control-1" class="section level3 hasAnchor" number="13.3.2">
<h3><span class="header-section-number">13.3.2</span> Quality Control<a href="examples.html#quality-control-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="spike-in-sample-brs-assessment-1" class="section level4 hasAnchor" number="13.3.2.1">
<h4><span class="header-section-number">13.3.2.1</span> Spike-in sample (BRS) assessment<a href="examples.html#spike-in-sample-brs-assessment-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Extract the species level phyloseq and obtain the BRS_ID</li>
</ul>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="examples.html#cb400-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb400-2"><a href="examples.html#cb400-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species<span class="sc">@</span>sam_data</span></code></pre></div>
<pre><code>##      Group phynotype
## s1      BB      0.00
## s2      AA      2.50
## s3      BB      0.00
## s4      AA      1.25
## s5      AA     30.00
## s6      AA     15.00
## s7      BB      8.75
## s8      BB      0.00
## s9      BB      3.75
## s10     BB      2.50
## s11     BB     15.00
## s12     BB      2.50
## s13     BB      2.50
## s14     BB      0.00
## s15     BB      1.07
## s16     BB      2.50
## s17     AA      5.00
## s18     BB     35.00
## s19     BB      7.50
## s20     BB     15.00
## s21     AA      3.75
## s22     AA      3.75
## refE    QC        NA</code></pre>
<ul>
<li>Run <code>run_RefCheck</code></li>
</ul>
<div class="sourceCode" id="cb402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb402-1"><a href="examples.html#cb402-1" aria-hidden="true" tabindex="-1"></a><span class="fu">run_RefCheck</span>(</span>
<span id="cb402-2"><a href="examples.html#cb402-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps=</span>metaphlan2_ps_species,</span>
<span id="cb402-3"><a href="examples.html#cb402-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">BRS_ID=</span><span class="st">&quot;refE&quot;</span>,</span>
<span id="cb402-4"><a href="examples.html#cb402-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">Reference=</span><span class="cn">NULL</span>,</span>
<span id="cb402-5"><a href="examples.html#cb402-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">Ref_type=</span><span class="st">&quot;MGS&quot;</span>,</span>
<span id="cb402-6"><a href="examples.html#cb402-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">Save=</span><span class="cn">NULL</span>)</span>
<span id="cb402-7"><a href="examples.html#cb402-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Noting: the Reference Matrix is for MGS</span></span>
<span id="cb402-8"><a href="examples.html#cb402-8" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb402-9"><a href="examples.html#cb402-9" aria-hidden="true" tabindex="-1"></a><span class="do">## #########Matched baterica of the Spike-in sample###########</span></span>
<span id="cb402-10"><a href="examples.html#cb402-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Identifying some taxa of the Reference Matrix are not in the dataset </span></span>
<span id="cb402-11"><a href="examples.html#cb402-11" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Lactobacillus_salivarius</span></span>
<span id="cb402-12"><a href="examples.html#cb402-12" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Bacteroides_fragilis</span></span>
<span id="cb402-13"><a href="examples.html#cb402-13" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Parabacteroides_goldsteinii</span></span>
<span id="cb402-14"><a href="examples.html#cb402-14" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Escherichia_coli</span></span>
<span id="cb402-15"><a href="examples.html#cb402-15" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Enterococcus_faecalis</span></span>
<span id="cb402-16"><a href="examples.html#cb402-16" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Bifidobacterium_bifidum</span></span>
<span id="cb402-17"><a href="examples.html#cb402-17" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Lactobacillus_pentosus</span></span>
<span id="cb402-18"><a href="examples.html#cb402-18" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Bacteroides_intestinalis</span></span>
<span id="cb402-19"><a href="examples.html#cb402-19" aria-hidden="true" tabindex="-1"></a><span class="do">## s__Eggerthella_unclassified</span></span>
<span id="cb402-20"><a href="examples.html#cb402-20" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb402-21"><a href="examples.html#cb402-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking out the additional Taxa are in the sequencing dataset</span></span>
<span id="cb402-22"><a href="examples.html#cb402-22" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb402-23"><a href="examples.html#cb402-23" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb402-24"><a href="examples.html#cb402-24" aria-hidden="true" tabindex="-1"></a><span class="do">## ##############Value of the Reference sample###############</span></span>
<span id="cb402-25"><a href="examples.html#cb402-25" aria-hidden="true" tabindex="-1"></a><span class="do">## Correlation Coefficient of the Reference sample is: -0.06471</span></span>
<span id="cb402-26"><a href="examples.html#cb402-26" aria-hidden="true" tabindex="-1"></a><span class="do">## Bray Curtis of the Reference sample is: 0.8752</span></span>
<span id="cb402-27"><a href="examples.html#cb402-27" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity of the Reference sample is: 32.69</span></span>
<span id="cb402-28"><a href="examples.html#cb402-28" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb402-29"><a href="examples.html#cb402-29" aria-hidden="true" tabindex="-1"></a><span class="do">## #####Final Evaluation Results of the Spike-in sample#######</span></span>
<span id="cb402-30"><a href="examples.html#cb402-30" aria-hidden="true" tabindex="-1"></a><span class="do">## The spike-in sample of sequencing dataset didn&#39;t pass the cutoff of the Reference Matrix</span></span>
<span id="cb402-31"><a href="examples.html#cb402-31" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb402-32"><a href="examples.html#cb402-32" aria-hidden="true" tabindex="-1"></a><span class="do">##                                         7682    7683    7684    7685     7842     7843     7844     7845     refE        mean</span></span>
<span id="cb402-33"><a href="examples.html#cb402-33" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_longum              10.31563 9.25812 9.69184 7.76031 11.03311 11.61484 12.29030 11.69019  0.01646  9.29675556</span></span>
<span id="cb402-34"><a href="examples.html#cb402-34" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_uniformis                2.24195 2.24035 1.92015 2.18435  2.43230  2.38180  2.13830  2.41437  0.21061  2.01824222</span></span>
<span id="cb402-35"><a href="examples.html#cb402-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Faecalibacterium_prausnitzii         0.65615 0.60153 0.60112 0.62079  0.54147  0.55383  0.58806  0.54655  1.61939  0.70321000</span></span>
<span id="cb402-36"><a href="examples.html#cb402-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_adolescentis         7.05426 6.28460 6.57297 6.25448  4.69357  4.80628  4.94943  4.84278  0.04649  5.05609556</span></span>
<span id="cb402-37"><a href="examples.html#cb402-37" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_thetaiotaomicron         3.25076 3.31897 3.22418 3.43809  3.35611  3.38323  3.29098  3.30355  1.47422  3.11556556</span></span>
<span id="cb402-38"><a href="examples.html#cb402-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Collinsella_aerofaciens              0.56220 0.53249 0.60476 0.47934  0.65513  0.68833  0.76063  0.66896  0.07251  0.55826111</span></span>
<span id="cb402-39"><a href="examples.html#cb402-39" aria-hidden="true" tabindex="-1"></a><span class="do">## Coprococcus_comes                    2.28581 2.42978 2.25227 2.82904  1.21527  1.16485  1.07160  1.07834  0.01444  1.59348889</span></span>
<span id="cb402-40"><a href="examples.html#cb402-40" aria-hidden="true" tabindex="-1"></a><span class="do">## Dorea_formicigenerans                4.83509 5.02149 5.18268 5.56891  3.34720  3.09877  3.08677  2.70732  0.02775  3.65288667</span></span>
<span id="cb402-41"><a href="examples.html#cb402-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Streptococcus_salivarius             3.54266 3.74119 3.62036 4.01546  2.90216  2.70193  2.61760  2.57348  0.02846  2.86036667</span></span>
<span id="cb402-42"><a href="examples.html#cb402-42" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_xylanisolvens            1.55648 1.84824 1.91166 1.85273  1.75220  1.74002  1.69811  1.67676  0.32466  1.59565111</span></span>
<span id="cb402-43"><a href="examples.html#cb402-43" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_ovatus                   3.08489 3.27226 3.10904 3.24565  3.51376  3.50063  3.37872  3.42300  0.25782  2.97619667</span></span>
<span id="cb402-44"><a href="examples.html#cb402-44" aria-hidden="true" tabindex="-1"></a><span class="do">## Roseburia_hominis                    0.04383 0.04183 0.04107 0.02304  0.03853  0.03464  0.03532  0.03597  0.01307  0.03414444</span></span>
<span id="cb402-45"><a href="examples.html#cb402-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_vulgatus                 3.06713 3.20369 3.14979 3.15352  3.24822  3.09280  3.13038  3.06113  2.14684  3.02816667</span></span>
<span id="cb402-46"><a href="examples.html#cb402-46" aria-hidden="true" tabindex="-1"></a><span class="do">## Prevotella_copri                     2.03128 1.99619 1.92504 2.15422  1.57638  1.60584  1.57913  1.59224 60.84109  8.36682333</span></span>
<span id="cb402-47"><a href="examples.html#cb402-47" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_pseudocatenulatum    6.89310 6.36177 7.39605 5.76804  6.19464  6.47094  7.41615  6.27837  0.15023  5.88103222</span></span>
<span id="cb402-48"><a href="examples.html#cb402-48" aria-hidden="true" tabindex="-1"></a><span class="do">## Lachnospiraceae_bacterium_5_1_63FAA  0.22002 0.26062 0.28058 0.33329  0.07093  0.06778  0.05229  0.04781  0.06825  0.15573000</span></span>
<span id="cb402-49"><a href="examples.html#cb402-49" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity_level                       7.33876 7.34799 7.14891 6.47250  8.88190  9.72198  9.72162 10.57600 32.69000 11.09996222</span></span>
<span id="cb402-50"><a href="examples.html#cb402-50" aria-hidden="true" tabindex="-1"></a><span class="do">##                                                                                Evaluation</span></span>
<span id="cb402-51"><a href="examples.html#cb402-51" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_longum              refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-52"><a href="examples.html#cb402-52" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_uniformis               refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-53"><a href="examples.html#cb402-53" aria-hidden="true" tabindex="-1"></a><span class="do">## Faecalibacterium_prausnitzii        refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-54"><a href="examples.html#cb402-54" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_adolescentis        refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-55"><a href="examples.html#cb402-55" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_thetaiotaomicron        refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-56"><a href="examples.html#cb402-56" aria-hidden="true" tabindex="-1"></a><span class="do">## Collinsella_aerofaciens             refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-57"><a href="examples.html#cb402-57" aria-hidden="true" tabindex="-1"></a><span class="do">## Coprococcus_comes                   refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-58"><a href="examples.html#cb402-58" aria-hidden="true" tabindex="-1"></a><span class="do">## Dorea_formicigenerans               refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-59"><a href="examples.html#cb402-59" aria-hidden="true" tabindex="-1"></a><span class="do">## Streptococcus_salivarius            refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-60"><a href="examples.html#cb402-60" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_xylanisolvens           refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-61"><a href="examples.html#cb402-61" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_ovatus                  refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-62"><a href="examples.html#cb402-62" aria-hidden="true" tabindex="-1"></a><span class="do">## Roseburia_hominis                   refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-63"><a href="examples.html#cb402-63" aria-hidden="true" tabindex="-1"></a><span class="do">## Bacteroides_vulgatus                refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-64"><a href="examples.html#cb402-64" aria-hidden="true" tabindex="-1"></a><span class="do">## Prevotella_copri                    refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-65"><a href="examples.html#cb402-65" aria-hidden="true" tabindex="-1"></a><span class="do">## Bifidobacterium_pseudocatenulatum   refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-66"><a href="examples.html#cb402-66" aria-hidden="true" tabindex="-1"></a><span class="do">## Lachnospiraceae_bacterium_5_1_63FAA refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span>
<span id="cb402-67"><a href="examples.html#cb402-67" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity_level                      refE didn&#39;t pass the threshold (2022-05-24 11:09:18).</span></span></code></pre></div>
<p>The spike-in samples didn’t pass the cutoff and failed to add the the Reference Matrix.</p>
</div>
<div id="spike-in-samples-remove-1" class="section level4 hasAnchor" number="13.3.2.2">
<h4><span class="header-section-number">13.3.2.2</span> Spike-in sample’s remove<a href="examples.html#spike-in-samples-remove-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="examples.html#cb403-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb403-2"><a href="examples.html#cb403-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb403-3"><a href="examples.html#cb403-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb403-4"><a href="examples.html#cb403-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb403-5"><a href="examples.html#cb403-5" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 326 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 326 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
<div id="data-processing-1" class="section level3 hasAnchor" number="13.3.3">
<h3><span class="header-section-number">13.3.3</span> Data processing<a href="examples.html#data-processing-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This part has too may procedures and we only choose some of them. Please go to <strong>Chapter 6</strong> to see more approaches and details for being familiar with this part.</p>
<div id="extracting-specific-taxonomic-level-1" class="section level4 hasAnchor" number="13.3.3.1">
<h4><span class="header-section-number">13.3.3.1</span> Extracting specific taxonomic level<a href="examples.html#extracting-specific-taxonomic-level-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="examples.html#cb405-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps_remove_BRS, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb405-2"><a href="examples.html#cb405-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 325 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 325 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4" class="section level4 hasAnchor" number="13.3.3.2">
<h4><span class="header-section-number">13.3.3.2</span> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)<a href="examples.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb407"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb407-1"><a href="examples.html#cb407-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(metaphlan2_ps_species, <span class="at">cutoff =</span> <span class="fl">1e-4</span>, <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb407-2"><a href="examples.html#cb407-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter </span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 233 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 233 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="trimming-the-taxa-with-low-occurrence-less-than-threshold-1" class="section level4 hasAnchor" number="13.3.3.3">
<h4><span class="header-section-number">13.3.3.3</span> Trimming the taxa with low occurrence less than threshold<a href="examples.html#trimming-the-taxa-with-low-occurrence-less-than-threshold-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb409"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb409-1"><a href="examples.html#cb409-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(metaphlan2_ps_species_filter, <span class="at">cutoff =</span> <span class="fl">0.2</span>, <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb409-2"><a href="examples.html#cb409-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 144 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 144 taxa by 7 taxonomic ranks ]</code></pre>
<p>Finally, we obtained the final phyloseq-class object <em>metaphlan2_ps_species_filter_trim</em> and changed its name.</p>
</div>
</div>
<div id="diversity-analysis-1" class="section level3 hasAnchor" number="13.3.4">
<h3><span class="header-section-number">13.3.4</span> Diversity analysis<a href="examples.html#diversity-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="alpha-diveristy-1" class="section level4 hasAnchor" number="13.3.4.1">
<h4><span class="header-section-number">13.3.4.1</span> Alpha diveristy<a href="examples.html#alpha-diveristy-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculate the alpha diversity</li>
</ul>
<p><strong>Notes:</strong> choosing the measures (<em>Shannon, Simpson and InvSimpson</em>) only for relative abundance.</p>
<div class="sourceCode" id="cb411"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb411-1"><a href="examples.html#cb411-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_genus_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps=</span>metaphlan2_ps_remove_BRS, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>))</span>
<span id="cb411-2"><a href="examples.html#cb411-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_genus_alpha)</span></code></pre></div>
<pre><code>##   SampleID Group phynotype  Shannon   Simpson InvSimpson
## 1       s1    BB      0.00 2.876002 0.9108857  11.221549
## 2       s2    AA      2.50 2.045392 0.8105742   5.279111
## 3       s3    BB      0.00 3.441176 0.9439371  17.837114
## 4       s4    AA      1.25 2.746917 0.8290768   5.850579
## 5       s5    AA     30.00 1.450722 0.6412178   2.787207
## 6       s6    AA     15.00 2.619951 0.8950369   9.527154</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb413"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb413-1"><a href="examples.html#cb413-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data=</span>metaphlan2_ps_genus_alpha,</span>
<span id="cb413-2"><a href="examples.html#cb413-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>),</span>
<span id="cb413-3"><a href="examples.html#cb413-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb413-4"><a href="examples.html#cb413-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb413-5"><a href="examples.html#cb413-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb413-6"><a href="examples.html#cb413-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-225"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-225-1.png" alt="Alpha diversity (MGS example)" width="576" />
<p class="caption">
Figure 13.12: Alpha diversity (MGS example)
</p>
</div>
</div>
<div id="beta-diversity-2" class="section level4 hasAnchor" number="13.3.4.2">
<h4><span class="header-section-number">13.3.4.2</span> Beta diversity<a href="examples.html#beta-diversity-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>beta dipersion</li>
</ul>
<div class="sourceCode" id="cb414"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb414-1"><a href="examples.html#cb414-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps=</span>metaphlan2_ps_species_filter_trim, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##           Df  Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
## Groups     1 0.00725 0.0072500 0.8423    999  0.362
## Residuals 20 0.17215 0.0086076                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         AA   BB
## AA         0.37
## BB 0.36968</code></pre>
<div class="sourceCode" id="cb416"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb416-1"><a href="examples.html#cb416-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_beta<span class="sc">$</span>BetaDispersion</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-226"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-226-1.png" alt="Beta diversity (MGS example)" width="480" />
<p class="caption">
Figure 13.13: Beta diversity (MGS example)
</p>
</div>
</div>
</div>
<div id="permanova-ordination-1" class="section level3 hasAnchor" number="13.3.5">
<h3><span class="header-section-number">13.3.5</span> PERMANOVA + Ordination<a href="examples.html#permanova-ordination-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="permanova-1" class="section level4 hasAnchor" number="13.3.5.1">
<h4><span class="header-section-number">13.3.5.1</span> PERMANOVA<a href="examples.html#permanova-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb417-1"><a href="examples.html#cb417-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_per <span class="ot">&lt;-</span> <span class="fu">run_permanova</span>(<span class="at">ps=</span>metaphlan2_ps_species_filter_trim, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, <span class="at">columns =</span> <span class="st">&quot;Group&quot;</span>)</span>
<span id="cb417-2"><a href="examples.html#cb417-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_per)</span></code></pre></div>
<pre><code>##       SumsOfSample Df SumsOfSqs  MeanSqs  F.Model        R2 Pr(&gt;F) AdjustedPvalue
## Group           22  1  0.864659 0.864659 2.880195 0.1258816  0.002          0.002</code></pre>
<p>The PERMANOVA result of the Group (AdjustedPvalue &lt; 0.05) revealed that the two groups had the distinct patterns of microbial community.</p>
</div>
<div id="ordination-2" class="section level4 hasAnchor" number="13.3.5.2">
<h4><span class="header-section-number">13.3.5.2</span> Ordination<a href="examples.html#ordination-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We performed ordination by using Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP). If you want to try other methods please go to see <strong>Chapter 8</strong> for more details.</p>
<div class="sourceCode" id="cb419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb419-1"><a href="examples.html#cb419-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb419-2"><a href="examples.html#cb419-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ps =</span> metaphlan2_ps_species_filter_trim,</span>
<span id="cb419-3"><a href="examples.html#cb419-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb419-4"><a href="examples.html#cb419-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;UMAP&quot;</span>)</span>
<span id="cb419-5"><a href="examples.html#cb419-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb419-6"><a href="examples.html#cb419-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> metaphlan2_ps_ordination, </span>
<span id="cb419-7"><a href="examples.html#cb419-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb419-8"><a href="examples.html#cb419-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb419-9"><a href="examples.html#cb419-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb419-10"><a href="examples.html#cb419-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-228"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-228-1.png" alt="PCoA (MGS example)" width="768" />
<p class="caption">
Figure 13.14: PCoA (MGS example)
</p>
</div>
</div>
</div>
<div id="microbial-composition-1" class="section level3 hasAnchor" number="13.3.6">
<h3><span class="header-section-number">13.3.6</span> Microbial composition<a href="examples.html#microbial-composition-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="stacked-barplot-2" class="section level4 hasAnchor" number="13.3.6.1">
<h4><span class="header-section-number">13.3.6.1</span> Stacked barplot<a href="examples.html#stacked-barplot-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>XMAS</strong> package</li>
</ul>
<div class="sourceCode" id="cb420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb420-1"><a href="examples.html#cb420-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb420-2"><a href="examples.html#cb420-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">ps=</span>metaphlan2_ps_species,</span>
<span id="cb420-3"><a href="examples.html#cb420-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">taxa_level=</span><span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb420-4"><a href="examples.html#cb420-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb420-5"><a href="examples.html#cb420-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-229"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-229-1.png" alt="Microbial composition (MGS example)" width="768" />
<p class="caption">
Figure 13.15: Microbial composition (MGS example)
</p>
</div>
<ul>
<li><strong>XVIZ</strong> package</li>
</ul>
<div class="sourceCode" id="cb421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb421-1"><a href="examples.html#cb421-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(</span>
<span id="cb421-2"><a href="examples.html#cb421-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">phyloseq =</span> metaphlan2_ps_species,</span>
<span id="cb421-3"><a href="examples.html#cb421-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb421-4"><a href="examples.html#cb421-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-230"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-230-1.png" alt="Microbial composition (MGS) XVIZ" width="768" />
<p class="caption">
Figure 13.16: Microbial composition (MGS) XVIZ
</p>
</div>
</div>
<div id="core-microbiota-2" class="section level4 hasAnchor" number="13.3.6.2">
<h4><span class="header-section-number">13.3.6.2</span> Core microbiota<a href="examples.html#core-microbiota-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="examples.html#cb422-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb422-2"><a href="examples.html#cb422-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-3"><a href="examples.html#cb422-3" aria-hidden="true" tabindex="-1"></a>prevalences <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.05</span>, <span class="dv">1</span>, <span class="fl">0.05</span>)</span>
<span id="cb422-4"><a href="examples.html#cb422-4" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="fu">seq</span>(<span class="fu">log10</span>(<span class="fl">1e-3</span>), <span class="fu">log10</span>(.<span class="dv">2</span>), <span class="at">length =</span> <span class="dv">10</span>)</span>
<span id="cb422-5"><a href="examples.html#cb422-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-6"><a href="examples.html#cb422-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_core_taxa</span>(metaphlan2_ps_species_filter_trim, </span>
<span id="cb422-7"><a href="examples.html#cb422-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, </span>
<span id="cb422-8"><a href="examples.html#cb422-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">colours =</span> <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">5</span>, <span class="st">&quot;Spectral&quot;</span>)),</span>
<span id="cb422-9"><a href="examples.html#cb422-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">prevalences =</span> prevalences, </span>
<span id="cb422-10"><a href="examples.html#cb422-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">detections =</span> detections, </span>
<span id="cb422-11"><a href="examples.html#cb422-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">min.prevalence =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb422-12"><a href="examples.html#cb422-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>) <span class="sc">+</span></span>
<span id="cb422-13"><a href="examples.html#cb422-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&quot;italic&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-231"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-231-1.png" alt="Core taxa (MGS example)" width="768" />
<p class="caption">
Figure 13.17: Core taxa (MGS example)
</p>
</div>
<p>The degree of color indicates the size of abundance and prevalence.</p>
<ul>
<li>Use <code>core_members</code> to obtain the core taxa. detection for abundance and prevalence for occurrence.</li>
</ul>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="examples.html#cb423-1" aria-hidden="true" tabindex="-1"></a>core_taxa_name <span class="ot">&lt;-</span> <span class="fu">core_members</span>(metaphlan2_ps_species_filter_trim, <span class="at">detection =</span> <span class="fl">0.001</span>, <span class="at">prevalence =</span> <span class="fl">0.5</span>)</span>
<span id="cb423-2"><a href="examples.html#cb423-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(core_taxa_name)</span></code></pre></div>
<pre><code>##  [1] &quot;s__Bacteroides_ovatus&quot;                &quot;s__Bacteroides_thetaiotaomicron&quot;      &quot;s__Bacteroides_uniformis&quot;            
##  [4] &quot;s__Bacteroides_vulgatus&quot;              &quot;s__Bifidobacterium_longum&quot;            &quot;s__Bifidobacterium_pseudocatenulatum&quot;
##  [7] &quot;s__Collinsella_aerofaciens&quot;           &quot;s__Escherichia_coli&quot;                  &quot;s__Eubacterium_eligens&quot;              
## [10] &quot;s__Eubacterium_hallii&quot;                &quot;s__Faecalibacterium_prausnitzii&quot;      &quot;s__Roseburia_inulinivorans&quot;          
## [13] &quot;s__Ruminococcus_gnavus&quot;               &quot;s__Ruminococcus_obeum&quot;                &quot;s__Ruminococcus_sp_5_1_39BFAA&quot;       
## [16] &quot;s__Ruminococcus_torques&quot;              &quot;s__Streptococcus_salivarius&quot;</code></pre>
<p><strong>Result</strong>:</p>
<p>17 species passed the threshold of detection and prevalence which we choose.</p>
</div>
</div>
<div id="differential-analysis-2" class="section level3 hasAnchor" number="13.3.7">
<h3><span class="header-section-number">13.3.7</span> Differential Analysis<a href="examples.html#differential-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to <strong>Chapter 10</strong> to see more detials.</p>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-3" class="section level4 hasAnchor" number="13.3.7.1">
<h4><span class="header-section-number">13.3.7.1</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="examples.html#liner-discriminant-analysis-lda-effect-size-lefse-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb425-1"><a href="examples.html#cb425-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse</span>(</span>
<span id="cb425-2"><a href="examples.html#cb425-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_species_filter_trim,</span>
<span id="cb425-3"><a href="examples.html#cb425-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb425-4"><a href="examples.html#cb425-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb425-5"><a href="examples.html#cb425-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb425-6"><a href="examples.html#cb425-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_lefse)</span></code></pre></div>
<pre><code>##                            TaxaID         Block Enrichment     LDA_Score EffectSize Log2FoldChange (Median)\nAA_vs_BB
## 1 s__Bacteroides_thetaiotaomicron 7_AA vs 15_BB         AA -0.0097587531  1.9418180                          5.505858
## 2            s__Dorea_longicatena 7_AA vs 15_BB         BB  0.0003492349  1.7435380                                NA
## 3       s__Clostridium_bartlettii 7_AA vs 15_BB         BB  0.0006298963  0.5784397                                NA
## 4           s__Eubacterium_hallii 7_AA vs 15_BB         BB  0.0021243839  0.7995960                                NA
## 5        s__Megamonas_rupellensis 7_AA vs 15_BB         BB  0.0024505506  0.5938727                                NA
## 6      s__Collinsella_aerofaciens 7_AA vs 15_BB         BB  0.0026197908  0.9034469                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1              0.00247415            0.0453799            0.0009987                        3.327766          0.0219183273
## 2              0.00078795            0.0000000            0.0081496                       -7.307095          0.0065455500
## 3              0.00025355            0.0000000            0.0004470                       -3.821895          0.0012202818
## 4              0.00436885            0.0000000            0.0109294                       -3.781465          0.0087429000
## 5              0.00000000            0.0005600            0.0000000                        5.020069          0.0008989182
## 6              0.00738145            0.0000000            0.0112480                       -3.336571          0.0118099818
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1       5.677024e-02       5.654100e-03                    90.91                100.00                 86.67
## 2       6.044286e-05       9.571933e-03                    63.64                 42.86                 73.33
## 3       1.225143e-04       1.732573e-03                    68.18                 42.86                 80.00
## 4       9.019000e-04       1.240203e-02                    63.64                 42.86                 73.33
## 5       2.650157e-03       8.167333e-05                    31.82                 57.14                 20.00
## 6       1.638929e-03       1.655647e-02                    63.64                 42.86                 73.33
##         Odds Ratio (95% CI)
## 1          0.038 (-6.4;6.4)
## 2 8.9e+12 (8.9e+12;8.9e+12)
## 3             600 (610;580)
## 4             110 (120;100)
## 5           0.0055 (-10;10)
## 6                41 (48;34)</code></pre>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="examples.html#cb427-1" aria-hidden="true" tabindex="-1"></a><span class="co"># in order to show barplot (don&#39;t run it in reality)</span></span>
<span id="cb427-2"><a href="examples.html#cb427-2" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_lefse<span class="sc">$</span>LDA_Score <span class="ot">&lt;-</span> metaphlan2_ps_lefse<span class="sc">$</span>LDA_Score <span class="sc">*</span> <span class="dv">1000</span></span>
<span id="cb427-3"><a href="examples.html#cb427-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb427-4"><a href="examples.html#cb427-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb427-5"><a href="examples.html#cb427-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">da_res =</span> metaphlan2_ps_lefse,</span>
<span id="cb427-6"><a href="examples.html#cb427-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb427-7"><a href="examples.html#cb427-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb427-8"><a href="examples.html#cb427-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-234"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-234-1.png" alt="Lefse analysis (MGS example)" width="768" />
<p class="caption">
Figure 13.18: Lefse analysis (MGS example)
</p>
</div>
</div>
<div id="wilcoxon-rank-sum-test-1" class="section level4 hasAnchor" number="13.3.7.2">
<h4><span class="header-section-number">13.3.7.2</span> Wilcoxon Rank-Sum test<a href="examples.html#wilcoxon-rank-sum-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="examples.html#cb428-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb428-2"><a href="examples.html#cb428-2" aria-hidden="true" tabindex="-1"></a>                    metaphlan2_ps_species_filter_trim,</span>
<span id="cb428-3"><a href="examples.html#cb428-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb428-4"><a href="examples.html#cb428-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb428-5"><a href="examples.html#cb428-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb428-6"><a href="examples.html#cb428-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_wilcox)</span></code></pre></div>
<pre><code>##                      TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue
## 1   s__Alistipes_finegoldii 7_AA vs 15_BB  Nonsignif 1.23094855      73.5 0.1145448      0.4521504
## 2 s__Alistipes_indistinctus 7_AA vs 15_BB  Nonsignif 0.78954179      66.0 0.2677815      0.6616137
## 3  s__Alistipes_onderdonkii 7_AA vs 15_BB  Nonsignif 1.14756258      63.5 0.4370143      0.7578592
## 4   s__Alistipes_putredinis 7_AA vs 15_BB  Nonsignif 1.34298152      54.0 0.9409918      0.9830621
## 5       s__Alistipes_shahii 7_AA vs 15_BB  Nonsignif 1.04736939      71.0 0.1751506      0.5801989
## 6    s__Anaerostipes_hadrus 7_AA vs 15_BB  Nonsignif 0.02734989      30.5 0.1258306      0.4625882
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1                                NA              0.00000000            0.0027362            0.0000000
## 2                                NA              0.00000000            0.0000000            0.0000000
## 3                          5.819157              0.00000825            0.0002823            0.0000050
## 4                                NA              0.00016915            0.0000000            0.0002200
## 5                                NA              0.00012580            0.0062474            0.0000000
## 6                         -3.900867              0.00013665            0.0000144            0.0002151
##   Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA
## 1                      0.52169761                   14.50                   10.10                    45.45                 57.14
## 2                      0.34139504                   13.43                   10.60                    31.82                 42.86
## 3                      0.27924089                   13.07                   10.77                    54.55                 57.14
## 4                      0.03870725                   11.71                   11.40                    54.55                 42.86
## 5                      0.46134594                   14.14                   10.27                    50.00                 57.14
## 6                     -0.63360363                    8.36                   12.97                    72.73                 57.14
##   Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 40.00     0.0028 (-12;12)
## 2                 26.67    0.067 (-5.2;5.4)
## 3                 53.33     0.38 (-1.5;2.3)
## 4                 60.00    0.47 (-0.99;1.9)
## 5                 46.67     0.17 (-3.2;3.6)
## 6                 80.00          31 (38;24)</code></pre>
<ul>
<li>Volcano</li>
</ul>
<div class="sourceCode" id="cb430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb430-1"><a href="examples.html#cb430-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb430-2"><a href="examples.html#cb430-2" aria-hidden="true" tabindex="-1"></a>    metaphlan2_ps_wilcox,</span>
<span id="cb430-3"><a href="examples.html#cb430-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb430-4"><a href="examples.html#cb430-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb430-5"><a href="examples.html#cb430-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb430-6"><a href="examples.html#cb430-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb430-7"><a href="examples.html#cb430-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb430-8"><a href="examples.html#cb430-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb430-9"><a href="examples.html#cb430-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-236"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-236-1.png" alt="Wilcoxon Rank-Sum test (MGS example)" width="768" />
<p class="caption">
Figure 13.19: Wilcoxon Rank-Sum test (MGS example)
</p>
</div>
</div>
</div>
</div>
<div id="systematic-information-10" class="section level2 hasAnchor" number="13.4">
<h2><span class="header-section-number">13.4</span> Systematic Information<a href="examples.html#systematic-information-10" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb431-1"><a href="examples.html#cb431-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Monterey 12.2.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] RColorBrewer_1.1-2 viridis_0.6.2      viridisLite_0.4.0  ggpubr_0.4.0       ggplot2_3.3.5      phyloseq_1.38.0   
## [7] tibble_3.1.6       dplyr_1.0.8        XMAS_2.1.1        
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.2                  reticulate_1.24             tidyselect_1.1.2            RSQLite_2.2.10             
##   [5] AnnotationDbi_1.56.2        htmlwidgets_1.5.4           grid_4.1.2                  BiocParallel_1.28.3        
##   [9] Rtsne_0.15                  munsell_0.5.0               units_0.8-0                 codetools_0.2-18           
##  [13] DT_0.21                     umap_0.2.8.0                withr_2.5.0                 colorspace_2.0-3           
##  [17] Biobase_2.54.0              highr_0.9                   knitr_1.37                  rstudioapi_0.13            
##  [21] stats4_4.1.2                pscl_1.5.5                  bayesm_3.1-4                robustbase_0.93-9          
##  [25] ggsignif_0.6.3              MatrixGenerics_1.6.0        labeling_0.4.2              optparse_1.7.1             
##  [29] mbzinb_0.2                  GenomeInfoDbData_1.2.7      lpsymphony_1.22.0           bit64_4.0.5                
##  [33] farver_2.1.0                pheatmap_1.0.12             rhdf5_2.38.1                vctrs_0.3.8                
##  [37] generics_0.1.2              TH.data_1.1-0               xfun_0.30                   Maaslin2_1.8.0             
##  [41] R6_2.5.1                    GenomeInfoDb_1.30.1         RVenn_1.1.0                 locfit_1.5-9.5             
##  [45] RcppZiggurat_0.1.6          bitops_1.0-7                rhdf5filters_1.6.0          cachem_1.0.6               
##  [49] DelayedArray_0.20.0         assertthat_0.2.1            scales_1.1.1                multcomp_1.4-18            
##  [53] nnet_7.3-17                 gtable_0.3.0                sandwich_3.0-1              rlang_1.0.2                
##  [57] genefilter_1.76.0           splines_4.1.2               protoclust_1.6.3            rstatix_0.7.0              
##  [61] broom_0.7.12                checkmate_2.0.0             yaml_2.3.5                  reshape2_1.4.4             
##  [65] abind_1.4-5                 crosstalk_1.2.0             backports_1.4.1             Rfast_2.0.6                
##  [69] qvalue_2.26.0               Hmisc_4.6-0                 DiagrammeR_1.0.9            tensorA_0.36.2             
##  [73] tools_4.1.2                 bookdown_0.24               zCompositions_1.4.0         ellipsis_0.3.2             
##  [77] gplots_3.1.1                jquerylib_0.1.4             biomformat_1.22.0           proxy_0.4-26               
##  [81] BiocGenerics_0.40.0         Rcpp_1.0.8.2                plyr_1.8.6                  visNetwork_2.1.0           
##  [85] base64enc_0.1-3             zlibbioc_1.40.0             classInt_0.4-3              purrr_0.3.4                
##  [89] RCurl_1.98-1.6              openssl_2.0.0               rpart_4.1.16                Wrench_1.12.0              
##  [93] cowplot_1.1.1               S4Vectors_0.32.3            zoo_1.8-9                   SummarizedExperiment_1.24.0
##  [97] ggrepel_0.9.1               cluster_2.1.2               magrittr_2.0.2              RSpectra_0.16-0            
## [101] data.table_1.14.2           truncnorm_1.0-8             mvtnorm_1.1-3               matrixStats_0.61.0         
## [105] ALDEx2_1.26.0               corncob_0.2.0               RAIDA_1.0                   evaluate_0.15              
## [109] xtable_1.8-4                XML_3.99-0.9                jpeg_0.1-9                  IRanges_2.28.0             
## [113] gridExtra_2.3               shape_1.4.6                 compiler_4.1.2              KernSmooth_2.23-20         
## [117] crayon_1.5.0                htmltools_0.5.2             mgcv_1.8-39                 pcaPP_1.9-74               
## [121] ggVennDiagram_1.2.1         Formula_1.2-4               tidyr_1.2.0                 geneplotter_1.72.0         
## [125] libcoin_1.0-9               DBI_1.1.2                   corrplot_0.92               biglm_0.9-2.1              
## [129] MASS_7.3-55                 sf_1.0-7                    compositions_2.0-4          Matrix_1.4-0               
## [133] ade4_1.7-18                 getopt_1.20.3               car_3.0-12                  permute_0.9-7              
## [137] cli_3.2.0                   parallel_4.1.2              igraph_1.2.11               GenomicRanges_1.46.1       
## [141] pkgconfig_2.0.3             coin_1.4-2                  foreign_0.8-82              foreach_1.5.2              
## [145] annotate_1.72.0             bslib_0.3.1                 multtest_2.50.0             XVector_0.34.0             
## [149] NADA_1.6-1.1                stringr_1.4.0               digest_0.6.29               vegan_2.5-7                
## [153] Biostrings_2.62.0           rmarkdown_2.13              htmlTable_2.4.0             edgeR_3.36.0               
## [157] gtools_3.9.2                modeltools_0.2-23           lifecycle_1.0.1             nlme_3.1-155               
## [161] jsonlite_1.8.0              Rhdf5lib_1.16.0             carData_3.0-5               askpass_1.1                
## [165] limma_3.50.1                fansi_1.0.2                 pillar_1.7.0                lattice_0.20-45            
## [169] KEGGREST_1.34.0             fastmap_1.1.0               httr_1.4.2                  DEoptimR_1.0-10            
## [173] survival_3.3-1              glue_1.6.2                  png_0.1-7                   iterators_1.0.14           
## [177] glmnet_4.1-3                bit_4.0.4                   class_7.3-20                sass_0.4.0                 
## [181] stringi_1.7.6               metagenomeSeq_1.36.0        blob_1.2.2                  DESeq2_1.34.0              
## [185] latticeExtra_0.6-29         caTools_1.18.2              memoise_2.0.1               e1071_1.7-9                
## [189] ape_5.6-2</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visualization-5.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/19-examples.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
