<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Principals of Differential Analysis methods | XMAS 2.0 Tutorial</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Principals of Differential Analysis methods | XMAS 2.0 Tutorial" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Principals of Differential Analysis methods | XMAS 2.0 Tutorial" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-08-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="association-analysis.html"/>
<link rel="next" href="examples.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing</a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers</a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book</a></li>
<li class="chapter" data-level="2.6" data-path="introduction.html"><a href="introduction.html#input-data"><i class="fa fa-check"></i><b>2.6</b> Input data</a></li>
<li class="chapter" data-level="2.7" data-path="introduction.html"><a href="introduction.html#key-publications"><i class="fa fa-check"></i><b>2.7</b> Key publications</a></li>
<li class="chapter" data-level="2.8" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>2.8</b> Citation</a></li>
<li class="chapter" data-level="2.9" data-path="introduction.html"><a href="introduction.html#change-log"><i class="fa fa-check"></i><b>2.9</b> Change log</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="introduction.html"><a href="introduction.html#xmas-2.1.7"><i class="fa fa-check"></i><b>2.9.1</b> XMAS 2.1.7</a></li>
<li class="chapter" data-level="2.9.2" data-path="introduction.html"><a href="introduction.html#xmas-2.1.6"><i class="fa fa-check"></i><b>2.9.2</b> XMAS 2.1.6</a></li>
<li class="chapter" data-level="2.9.3" data-path="introduction.html"><a href="introduction.html#xmas-2.1.5"><i class="fa fa-check"></i><b>2.9.3</b> XMAS 2.1.5</a></li>
<li class="chapter" data-level="2.9.4" data-path="introduction.html"><a href="introduction.html#xmas-2.1.4"><i class="fa fa-check"></i><b>2.9.4</b> XMAS 2.1.4</a></li>
<li class="chapter" data-level="2.9.5" data-path="introduction.html"><a href="introduction.html#xmas-2.1.3"><i class="fa fa-check"></i><b>2.9.5</b> XMAS 2.1.3</a></li>
<li class="chapter" data-level="2.9.6" data-path="introduction.html"><a href="introduction.html#xmas-2.1.2"><i class="fa fa-check"></i><b>2.9.6</b> XMAS 2.1.2</a></li>
<li class="chapter" data-level="2.9.7" data-path="introduction.html"><a href="introduction.html#xmas-2.1.1"><i class="fa fa-check"></i><b>2.9.7</b> XMAS 2.1.1</a></li>
<li class="chapter" data-level="2.9.8" data-path="introduction.html"><a href="introduction.html#xmas-2.1.0"><i class="fa fa-check"></i><b>2.9.8</b> XMAS 2.1.0</a></li>
<li class="chapter" data-level="2.9.9" data-path="introduction.html"><a href="introduction.html#xmas-2.0.13"><i class="fa fa-check"></i><b>2.9.9</b> XMAS 2.0.13</a></li>
<li class="chapter" data-level="2.9.10" data-path="introduction.html"><a href="introduction.html#xmas-2.0.12"><i class="fa fa-check"></i><b>2.9.10</b> XMAS 2.0.12</a></li>
<li class="chapter" data-level="2.9.11" data-path="introduction.html"><a href="introduction.html#xmas-2.0.7-2022-04-12"><i class="fa fa-check"></i><b>2.9.11</b> XMAS 2.0.7 (2022-04-12)</a></li>
<li class="chapter" data-level="2.9.12" data-path="introduction.html"><a href="introduction.html#xmas-2.0.2-2022-03-17"><i class="fa fa-check"></i><b>2.9.12</b> XMAS 2.0.2 (2022-03-17)</a></li>
<li class="chapter" data-level="2.9.13" data-path="introduction.html"><a href="introduction.html#xmas-2.0.1-2022-03-14"><i class="fa fa-check"></i><b>2.9.13</b> XMAS 2.0.1 (2022-03-14)</a></li>
<li class="chapter" data-level="2.9.14" data-path="introduction.html"><a href="introduction.html#xmas-1.0.0-2021-12-13"><i class="fa fa-check"></i><b>2.9.14</b> XMAS 1.0.0 (2021-12-13)</a></li>
<li class="chapter" data-level="2.9.15" data-path="introduction.html"><a href="introduction.html#initial-repository-0.0.1-2021-07-25"><i class="fa fa-check"></i><b>2.9.15</b> Initial repository 0.0.1 (2021-07-25)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial</a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2</a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description</a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages</a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation</a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette</a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors</a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan</a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>4.2.1</b> Importing results from dada2 pipeline</a></li>
<li class="chapter" data-level="4.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>4.2.2</b> taxa table</a></li>
<li class="chapter" data-level="4.2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>4.2.3</b> otu table</a></li>
<li class="chapter" data-level="4.2.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>4.2.4</b> metadata table</a></li>
<li class="chapter" data-level="4.2.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>4.2.5</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>4.3.1</b> Importing results</a></li>
<li class="chapter" data-level="4.3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table-1"><i class="fa fa-check"></i><b>4.3.2</b> taxa table</a></li>
<li class="chapter" data-level="4.3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table-1"><i class="fa fa-check"></i><b>4.3.3</b> otu table</a></li>
<li class="chapter" data-level="4.3.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table-1"><i class="fa fa-check"></i><b>4.3.4</b> metadata table</a></li>
<li class="chapter" data-level="4.3.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object-1"><i class="fa fa-check"></i><b>4.3.5</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Readsâ€™ track</a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS</a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves</a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object</a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>5.6</b> Variability</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>5.6.1</b> Coefficient of variation</a></li>
<li class="chapter" data-level="5.6.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>5.6.2</b> Distribution of number of sequences</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation</a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization</a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering</a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming</a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation</a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object</a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others</a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>6.8</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)</a></li>
<li class="chapter" data-level="6.9" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>7.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="7.2.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance</a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#evenness"><i class="fa fa-check"></i><b>7.4</b> Evenness</a></li>
<li class="chapter" data-level="7.5" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS</a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac</a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition</a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)</a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>8.6</b> Mantel Test</a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.7</b> Analysis of Similarity (ANOSIM)</a></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>8.8</b> Multi-response permutation procedures (MRPP)</a></li>
<li class="chapter" data-level="8.9" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.9</b> Ordination</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.9.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="8.9.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.9.2</b> Principal Coordinate Analysis (PCoA)</a></li>
<li class="chapter" data-level="8.9.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.9.3</b> t-distributed stochastic neighbor embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.9.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.9.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>9</b> Microbial Composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbial-composition.html"><a href="microbial-composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object</a></li>
<li class="chapter" data-level="9.2" data-path="microbial-composition.html"><a href="microbial-composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.2</b> XVIZ <code>plot_stacked_bar</code></a></li>
<li class="chapter" data-level="9.3" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.3</b> Stacked barplot</a></li>
<li class="chapter" data-level="9.4" data-path="microbial-composition.html"><a href="microbial-composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps</a></li>
<li class="chapter" data-level="9.5" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota</a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap</a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test</a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq</a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2</a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR</a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM</a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob</a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
<li class="chapter" data-level="11.1.15" data-path="differential-analysis.html"><a href="differential-analysis.html#locom"><i class="fa fa-check"></i><b>11.1.15</b> LOCOM</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.1</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.3</b> t-test</a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.4</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot</a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa</a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#comparing-outputs-from-xmas2-lefse-conda-and-lefse-galaxy-using-the-same-in-house-datasets-amplicon_pszeybel_gut"><i class="fa fa-check"></i><b>11.6</b> Comparing outputs from XMAS2, lefse-conda, and lefse-galaxy using the same in-house datasets (amplicon_ps|Zeybel_Gut)</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="differential-analysis.html"><a href="differential-analysis.html#introduction-1"><i class="fa fa-check"></i><b>11.6.1</b> Introduction</a></li>
<li class="chapter" data-level="11.6.2" data-path="differential-analysis.html"><a href="differential-analysis.html#dataset"><i class="fa fa-check"></i><b>11.6.2</b> Dataset</a></li>
<li class="chapter" data-level="11.6.3" data-path="differential-analysis.html"><a href="differential-analysis.html#run-lefse-independently-with-the-three-applications-r-conda-galaxy"><i class="fa fa-check"></i><b>11.6.3</b> Run lefse independently with the three applications (R, conda, galaxy)</a></li>
<li class="chapter" data-level="11.6.4" data-path="differential-analysis.html"><a href="differential-analysis.html#comparison-of-lefse-conda-with-xmas2"><i class="fa fa-check"></i><b>11.6.4</b> Comparison of lefse-conda with XMAS2</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization</a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot</a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot</a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot</a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot</a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density</a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination</a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot</a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot</a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn</a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot"><i class="fa fa-check"></i><b>12.10</b> plot the DA results from the significant taxa by double barplot</a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.11</b> plot_stacked_bar_XIVZ</a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.12</b> plot_StackBarPlot</a></li>
<li class="chapter" data-level="12.13" data-path="visualization-5.html"><a href="visualization-5.html#color-palettes"><i class="fa fa-check"></i><b>12.13</b> Color Palettes</a>
<ul>
<li class="chapter" data-level="12.13.1" data-path="visualization-5.html"><a href="visualization-5.html#wes-anderson-palettes"><i class="fa fa-check"></i><b>12.13.1</b> Wes Anderson Palettes</a></li>
</ul></li>
<li class="chapter" data-level="12.14" data-path="visualization-5.html"><a href="visualization-5.html#ordination-plots-with-ggplot2"><i class="fa fa-check"></i><b>12.14</b> Ordination plots with ggplot2</a>
<ul>
<li class="chapter" data-level="12.14.1" data-path="visualization-5.html"><a href="visualization-5.html#principal-components-analysis-with-the-iris-data-set"><i class="fa fa-check"></i><b>12.14.1</b> principal components analysis with the iris data set</a></li>
<li class="chapter" data-level="12.14.2" data-path="visualization-5.html"><a href="visualization-5.html#multiple-correspondence-analysis-with-the-tea-dataset"><i class="fa fa-check"></i><b>12.14.2</b> multiple correspondence analysis with the tea dataset</a></li>
<li class="chapter" data-level="12.14.3" data-path="visualization-5.html"><a href="visualization-5.html#linear-discriminant-analysis"><i class="fa fa-check"></i><b>12.14.3</b> linear discriminant analysis</a></li>
</ul></li>
<li class="chapter" data-level="12.15" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.15</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>13</b> Association Analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="association-analysis.html"><a href="association-analysis.html#loading-and-showing-phyloseq-object"><i class="fa fa-check"></i><b>13.1</b> Loading and showing phyloseq object</a></li>
<li class="chapter" data-level="13.2" data-path="association-analysis.html"><a href="association-analysis.html#spearman-correlation-analysis"><i class="fa fa-check"></i><b>13.2</b> Spearman Correlation Analysis</a></li>
<li class="chapter" data-level="13.3" data-path="association-analysis.html"><a href="association-analysis.html#partial-correlation-analysis"><i class="fa fa-check"></i><b>13.3</b> Partial Correlation Analysis</a></li>
<li class="chapter" data-level="13.4" data-path="association-analysis.html"><a href="association-analysis.html#systematic-information-10"><i class="fa fa-check"></i><b>13.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>14</b> Principals of Differential Analysis methods</a>
<ul>
<li class="chapter" data-level="14.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>14.1</b> ALDEx2</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-1"><i class="fa fa-check"></i><b>14.1.1</b> Vignette</a></li>
<li class="chapter" data-level="14.1.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type"><i class="fa fa-check"></i><b>14.1.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.1.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments"><i class="fa fa-check"></i><b>14.1.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.1.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures"><i class="fa fa-check"></i><b>14.1.4</b> Procedures</a></li>
<li class="chapter" data-level="14.1.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output"><i class="fa fa-check"></i><b>14.1.5</b> Key output</a></li>
<li class="chapter" data-level="14.1.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax"><i class="fa fa-check"></i><b>14.1.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.1.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax"><i class="fa fa-check"></i><b>14.1.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.1.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes"><i class="fa fa-check"></i><b>14.1.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>14.2</b> ANCOM-II</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-1"><i class="fa fa-check"></i><b>14.2.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.2.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-1"><i class="fa fa-check"></i><b>14.2.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.2.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-1"><i class="fa fa-check"></i><b>14.2.3</b> Procedures</a></li>
<li class="chapter" data-level="14.2.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-1"><i class="fa fa-check"></i><b>14.2.4</b> Key output</a></li>
<li class="chapter" data-level="14.2.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-1"><i class="fa fa-check"></i><b>14.2.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.2.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-1"><i class="fa fa-check"></i><b>14.2.6</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.2.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-1"><i class="fa fa-check"></i><b>14.2.7</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>14.3</b> corncob</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-2"><i class="fa fa-check"></i><b>14.3.1</b> Vignette</a></li>
<li class="chapter" data-level="14.3.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-2"><i class="fa fa-check"></i><b>14.3.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.3.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-2"><i class="fa fa-check"></i><b>14.3.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.3.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-2"><i class="fa fa-check"></i><b>14.3.4</b> Procedures</a></li>
<li class="chapter" data-level="14.3.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-2"><i class="fa fa-check"></i><b>14.3.5</b> Key output</a></li>
<li class="chapter" data-level="14.3.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-2"><i class="fa fa-check"></i><b>14.3.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.3.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-2"><i class="fa fa-check"></i><b>14.3.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.3.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-2"><i class="fa fa-check"></i><b>14.3.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>14.4</b> DESeq2</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-3"><i class="fa fa-check"></i><b>14.4.1</b> Vignette</a></li>
<li class="chapter" data-level="14.4.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-3"><i class="fa fa-check"></i><b>14.4.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.4.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-3"><i class="fa fa-check"></i><b>14.4.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.4.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-3"><i class="fa fa-check"></i><b>14.4.4</b> Procedures</a></li>
<li class="chapter" data-level="14.4.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-3"><i class="fa fa-check"></i><b>14.4.5</b> Key output</a></li>
<li class="chapter" data-level="14.4.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-3"><i class="fa fa-check"></i><b>14.4.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.4.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-3"><i class="fa fa-check"></i><b>14.4.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>14.5</b> edgeR</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-4"><i class="fa fa-check"></i><b>14.5.1</b> Vignette</a></li>
<li class="chapter" data-level="14.5.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-4"><i class="fa fa-check"></i><b>14.5.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.5.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-4"><i class="fa fa-check"></i><b>14.5.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.5.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-4"><i class="fa fa-check"></i><b>14.5.4</b> Procedures</a></li>
<li class="chapter" data-level="14.5.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-4"><i class="fa fa-check"></i><b>14.5.5</b> Key output</a></li>
<li class="chapter" data-level="14.5.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-4"><i class="fa fa-check"></i><b>14.5.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.5.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-4"><i class="fa fa-check"></i><b>14.5.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.5.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-3"><i class="fa fa-check"></i><b>14.5.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>14.6</b> LEfSe</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-5"><i class="fa fa-check"></i><b>14.6.1</b> Vignette</a></li>
<li class="chapter" data-level="14.6.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-5"><i class="fa fa-check"></i><b>14.6.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.6.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-5"><i class="fa fa-check"></i><b>14.6.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.6.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-5"><i class="fa fa-check"></i><b>14.6.4</b> Procedures</a></li>
<li class="chapter" data-level="14.6.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-5"><i class="fa fa-check"></i><b>14.6.5</b> Key output</a></li>
<li class="chapter" data-level="14.6.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-5"><i class="fa fa-check"></i><b>14.6.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.6.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-5"><i class="fa fa-check"></i><b>14.6.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.6.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-4"><i class="fa fa-check"></i><b>14.6.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>14.7</b> limma voom</a>
<ul>
<li class="chapter" data-level="14.7.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-6"><i class="fa fa-check"></i><b>14.7.1</b> Vignette</a></li>
<li class="chapter" data-level="14.7.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-6"><i class="fa fa-check"></i><b>14.7.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.7.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-6"><i class="fa fa-check"></i><b>14.7.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.7.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-6"><i class="fa fa-check"></i><b>14.7.4</b> Procedures</a></li>
<li class="chapter" data-level="14.7.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-6"><i class="fa fa-check"></i><b>14.7.5</b> Key output</a></li>
<li class="chapter" data-level="14.7.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-6"><i class="fa fa-check"></i><b>14.7.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.7.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-6"><i class="fa fa-check"></i><b>14.7.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.7.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-5"><i class="fa fa-check"></i><b>14.7.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>14.8</b> MaAsLin2</a>
<ul>
<li class="chapter" data-level="14.8.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-7"><i class="fa fa-check"></i><b>14.8.1</b> Vignette</a></li>
<li class="chapter" data-level="14.8.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-7"><i class="fa fa-check"></i><b>14.8.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.8.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-7"><i class="fa fa-check"></i><b>14.8.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.8.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-7"><i class="fa fa-check"></i><b>14.8.4</b> Procedures</a></li>
<li class="chapter" data-level="14.8.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-7"><i class="fa fa-check"></i><b>14.8.5</b> Key output</a></li>
<li class="chapter" data-level="14.8.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-7"><i class="fa fa-check"></i><b>14.8.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.8.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-7"><i class="fa fa-check"></i><b>14.8.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>14.9</b> metagenomeSeq</a>
<ul>
<li class="chapter" data-level="14.9.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-8"><i class="fa fa-check"></i><b>14.9.1</b> Vignette</a></li>
<li class="chapter" data-level="14.9.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-8"><i class="fa fa-check"></i><b>14.9.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.9.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-8"><i class="fa fa-check"></i><b>14.9.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.9.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-8"><i class="fa fa-check"></i><b>14.9.4</b> Procedures</a></li>
<li class="chapter" data-level="14.9.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-8"><i class="fa fa-check"></i><b>14.9.5</b> Key output</a></li>
<li class="chapter" data-level="14.9.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-8"><i class="fa fa-check"></i><b>14.9.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>14.10</b> t-test</a>
<ul>
<li class="chapter" data-level="14.10.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-9"><i class="fa fa-check"></i><b>14.10.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.10.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-9"><i class="fa fa-check"></i><b>14.10.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.10.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-9"><i class="fa fa-check"></i><b>14.10.3</b> Procedures</a></li>
<li class="chapter" data-level="14.10.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-9"><i class="fa fa-check"></i><b>14.10.4</b> Key output</a></li>
<li class="chapter" data-level="14.10.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-8"><i class="fa fa-check"></i><b>14.10.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.10.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-9"><i class="fa fa-check"></i><b>14.10.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>14.11</b> Wilcoxon test</a>
<ul>
<li class="chapter" data-level="14.11.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-10"><i class="fa fa-check"></i><b>14.11.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.11.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-10"><i class="fa fa-check"></i><b>14.11.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.11.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-10"><i class="fa fa-check"></i><b>14.11.3</b> Procedures</a></li>
<li class="chapter" data-level="14.11.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-10"><i class="fa fa-check"></i><b>14.11.4</b> Key output</a></li>
<li class="chapter" data-level="14.11.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-9"><i class="fa fa-check"></i><b>14.11.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.11.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-10"><i class="fa fa-check"></i><b>14.11.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>14.12</b> RAIDA</a>
<ul>
<li class="chapter" data-level="14.12.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-9"><i class="fa fa-check"></i><b>14.12.1</b> Vignette</a></li>
<li class="chapter" data-level="14.12.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-11"><i class="fa fa-check"></i><b>14.12.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.12.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-11"><i class="fa fa-check"></i><b>14.12.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.12.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-11"><i class="fa fa-check"></i><b>14.12.4</b> Procedures</a></li>
<li class="chapter" data-level="14.12.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-11"><i class="fa fa-check"></i><b>14.12.5</b> Key output</a></li>
<li class="chapter" data-level="14.12.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-10"><i class="fa fa-check"></i><b>14.12.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.12.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-11"><i class="fa fa-check"></i><b>14.12.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>14.13</b> Summary</a></li>
<li class="chapter" data-level="14.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>14.14</b> Advantages and disadvantages</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>15</b> Examples</a>
<ul>
<li class="chapter" data-level="15.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>15.1</b> Workflow description</a></li>
<li class="chapter" data-level="15.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>15.2</b> Amplicon sequencing (16s)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>15.2.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="15.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>15.2.2</b> Quality Control</a></li>
<li class="chapter" data-level="15.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>15.2.3</b> Data processing</a></li>
<li class="chapter" data-level="15.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>15.2.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="15.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>15.2.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="15.2.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>15.2.6</b> Microbial composition</a></li>
<li class="chapter" data-level="15.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>15.2.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="examples.html"><a href="examples.html#metagenomics-mgs"><i class="fa fa-check"></i><b>15.3</b> Metagenomics (MGS)</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>15.3.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="15.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>15.3.2</b> Quality Control</a></li>
<li class="chapter" data-level="15.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>15.3.3</b> Data processing</a></li>
<li class="chapter" data-level="15.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>15.3.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="15.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>15.3.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="15.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-2"><i class="fa fa-check"></i><b>15.3.6</b> Microbial composition</a></li>
<li class="chapter" data-level="15.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>15.3.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="examples.html"><a href="examples.html#systematic-information-11"><i class="fa fa-check"></i><b>15.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">XMAS 2.0 Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="principals-of-differential-analysis-methods" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">Chapter 14</span> Principals of Differential Analysis methods<a href="principals-of-differential-analysis-methods.html#principals-of-differential-analysis-methods" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><span class="citation">(<a href="#ref-nearing2022microbiome" role="doc-biblioref">Nearing et al. 2022</a>)</span> has showed that the multiple DA methods have different performances on microbiota (16s). Here, we integrated the its methodsâ€™ part and our own understandings to give a draft whole picture for users who using DA of XMAS2.</p>
<p>Microbiota data have the following characteristics:</p>
<ol style="list-style-type: decimal">
<li><strong>Compositional</strong>: Mathematically, a data is defined as compositional, if it contains D multiple parts of nonnegative numbers whose sum is 1 or any constant-sum constraint. It can be formally stated as:</li>
</ol>
<p><span class="math display">\[S^{D} = \left \{ X = [x_{1}, x_{2}, ..., x_{D}] | x_{i} &gt; 0, i = 1, 2, ..., D; \sum^{D}_{i=1} x_{i} = K \right \}\]</span></p>
<ol start="2" style="list-style-type: decimal">
<li><p><strong>Over-Dispersed</strong>: The variance are generally greater than the mean. There are several possible causes of overdispersion: correlated taxa counts, clustering of subjects, and within group heterogeneity. Negative Binomial Model addresses overdispersion by by explicitly modeling the correlated and sparse events via a latent variable.</p></li>
<li><p><strong>Sparsity</strong>: Microbiome data typically is overdispersed and sparse with many zeros. When the number of zeros is excess than the standard distributions (e.g., normal, Poisson, binomial, NB, beta or gamma) can be readily fit, the data set is considered as â€™zero inflate.</p></li>
</ol>
<div id="aldex2-1" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">14.1</span> ALDEx2<a href="principals-of-differential-analysis-methods.html#aldex2-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>ALDEx2 is from <span class="citation">(<a href="#ref-fernandes2014unifying" role="doc-biblioref">Fernandes et al. 2014</a>)</span></p>
<div id="vignette-1" class="section level3 hasAnchor" number="14.1.1">
<h3><span class="header-section-number">14.1.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/ALDEx2/inst/doc/ALDEx2_vignette.html">ANOVA-Like Differential Expression tool for high throughput sequencing data: ALDEx2</a></p>
</div>
<div id="input-data-type" class="section level3 hasAnchor" number="14.1.2">
<h3><span class="header-section-number">14.1.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type should be <strong>Raw Counts</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments" class="section level3 hasAnchor" number="14.1.3">
<h3><span class="header-section-number">14.1.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>reads</strong>: Raw counts matrix (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>conditions</strong>: A vector of group labels for testing.</p></li>
<li><p><strong>mc.samples</strong>: The number of Monte Carlo samples, default is 128.</p></li>
<li><p><strong>denom</strong>: The methods for Geometric Mean calculation, default is â€œallâ€.</p></li>
<li><p><strong>test</strong>: The test to perform, default is â€œt.testâ€.</p></li>
</ol>
</div>
<div id="procedures" class="section level3 hasAnchor" number="14.1.4">
<h3><span class="header-section-number">14.1.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Monte Carlo samples of Dirichlet distributions for each sample, using a uniform prior; (Sparsity)</p></li>
<li><p>Performing CLR (log-ratio) transformation of each realization; (Compositional &amp; Overdispersion)</p></li>
<li><p>Performing Wilcoxon tests on the transformed realizations.</p></li>
</ol>
<p>Finally, the function returned the expected Benjamini-Hochberg (BH) FDR-corrected p-value for each feature based on the results the different across Monte Carlo samples.</p>
</div>
<div id="key-output" class="section level3 hasAnchor" number="14.1.5">
<h3><span class="header-section-number">14.1.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>EffectSize</strong>: Effect Size generated by <code>Aldex2</code>, which equals to Log2FoldChange.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Median CLR (All/Groups)</strong>: median value of CLR abundance normalized by <code>Aldex2</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax" class="section level3 hasAnchor" number="14.1.6">
<h3><span class="header-section-number">14.1.6</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb490-1"><a href="principals-of-differential-analysis-methods.html#cb490-1" aria-hidden="true" tabindex="-1"></a><span class="fu">aldex</span>(</span>
<span id="cb490-2"><a href="principals-of-differential-analysis-methods.html#cb490-2" aria-hidden="true" tabindex="-1"></a>  reads,</span>
<span id="cb490-3"><a href="principals-of-differential-analysis-methods.html#cb490-3" aria-hidden="true" tabindex="-1"></a>  conditions,</span>
<span id="cb490-4"><a href="principals-of-differential-analysis-methods.html#cb490-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">mc.samples =</span> <span class="dv">128</span>,</span>
<span id="cb490-5"><a href="principals-of-differential-analysis-methods.html#cb490-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">test =</span> <span class="st">&quot;t&quot;</span>,</span>
<span id="cb490-6"><a href="principals-of-differential-analysis-methods.html#cb490-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">effect =</span> <span class="cn">TRUE</span>,</span>
<span id="cb490-7"><a href="principals-of-differential-analysis-methods.html#cb490-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">include.sample.summary =</span> <span class="cn">FALSE</span>,</span>
<span id="cb490-8"><a href="principals-of-differential-analysis-methods.html#cb490-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span>,</span>
<span id="cb490-9"><a href="principals-of-differential-analysis-methods.html#cb490-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">denom =</span> <span class="st">&quot;all&quot;</span>,</span>
<span id="cb490-10"><a href="principals-of-differential-analysis-methods.html#cb490-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">iterate =</span> <span class="cn">FALSE</span>,</span>
<span id="cb490-11"><a href="principals-of-differential-analysis-methods.html#cb490-11" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb490-12"><a href="principals-of-differential-analysis-methods.html#cb490-12" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="xmas2-syntax" class="section level3 hasAnchor" number="14.1.7">
<h3><span class="header-section-number">14.1.7</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb491"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb491-1"><a href="principals-of-differential-analysis-methods.html#cb491-1" aria-hidden="true" tabindex="-1"></a>da_res <span class="ot">&lt;-</span> ALDEx2<span class="sc">::</span><span class="fu">aldex</span>(<span class="at">reads =</span> otu_tab,</span>
<span id="cb491-2"><a href="principals-of-differential-analysis-methods.html#cb491-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">conditions =</span> sam_tab<span class="sc">$</span>Compvar,</span>
<span id="cb491-3"><a href="principals-of-differential-analysis-methods.html#cb491-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">mc.samples =</span> <span class="dv">128</span>,</span>
<span id="cb491-4"><a href="principals-of-differential-analysis-methods.html#cb491-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">test =</span> test,</span>
<span id="cb491-5"><a href="principals-of-differential-analysis-methods.html#cb491-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">effect =</span> <span class="cn">TRUE</span>,</span>
<span id="cb491-6"><a href="principals-of-differential-analysis-methods.html#cb491-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">include.sample.summary =</span> <span class="cn">FALSE</span>,</span>
<span id="cb491-7"><a href="principals-of-differential-analysis-methods.html#cb491-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">denom =</span> <span class="st">&quot;all&quot;</span>,</span>
<span id="cb491-8"><a href="principals-of-differential-analysis-methods.html#cb491-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="notes" class="section level3 hasAnchor" number="14.1.8">
<h3><span class="header-section-number">14.1.8</span> Notes<a href="principals-of-differential-analysis-methods.html#notes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Effect Size could be regarded as Log2FoldChange;</li>
</ol>
</div>
</div>
<div id="ancom-ii" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">14.2</span> ANCOM-II<a href="principals-of-differential-analysis-methods.html#ancom-ii" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>ANCOM-II is from <a href="https://github.com/FrederickHuangLin/Microbiome-Review-Code-Archive/tree/master/scripts">ANCOM-II (version 2.1)</a> and it derived from <span class="citation">(<a href="#ref-mandal2015analysis" role="doc-biblioref">Mandal et al. 2015</a>)</span>.</p>
<div id="input-data-type-1" class="section level3 hasAnchor" number="14.2.1">
<h3><span class="header-section-number">14.2.1</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type should be <strong>Raw Counts</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-1" class="section level3 hasAnchor" number="14.2.2">
<h3><span class="header-section-number">14.2.2</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>feature_table</strong>: Raw counts matrix (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>meta_data</strong>: data.frame with group information.</p></li>
<li><p><strong>struc_zero</strong>: data.frame with structure zero taxa.</p></li>
<li><p><strong>group_var</strong>: group variable for comparison.</p></li>
<li><p><strong>alpha</strong>: the significant level cutoff for testing.</p></li>
</ol>
</div>
<div id="procedures-1" class="section level3 hasAnchor" number="14.2.3">
<h3><span class="header-section-number">14.2.3</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><code>feature_table_pre_process</code> firstly identify outlier zeros and structural zeros;</li>
</ol>
<ul>
<li><p>Outlier zeros, identified by finding outliers in the distribution of taxon counts within each sample grouping, were ignored during differential abundance analysis, and replaced with NA.</p></li>
<li><p>Structural zeros, taxa that were absent in one grouping but present in the other, were ignored during data analysis and automatically called as differential abundant.</p></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li><p>additive log-ratios for transformation with a pseudo count of 1;</p></li>
<li><p>Wilcoxon rank-sum tests for significance, and p-values were FDR-corrected using the BH method;</p></li>
<li><p>Detection threshold for Ajusted-pvalue per taxa, if the number (ratio) of Ajusted-pvalue reaches nominal significance would be called as DA taxa.</p></li>
</ol>
</div>
<div id="key-output-1" class="section level3 hasAnchor" number="14.2.4">
<h3><span class="header-section-number">14.2.4</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>EffectSize</strong>: Effect Size to evaluate the quality of <em>(W)q-values &lt; alpha</em>.</p></li>
<li><p><strong>W_ratio</strong>: the ratio of significant taxa compared to totoal number of taxa.</p></li>
<li><p><strong>detected</strong>: the taxa is significant passed the W_ratio cutoff.</p></li>
</ol>
</div>
<div id="original-syntax-1" class="section level3 hasAnchor" number="14.2.5">
<h3><span class="header-section-number">14.2.5</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb492-1"><a href="principals-of-differential-analysis-methods.html#cb492-1" aria-hidden="true" tabindex="-1"></a>preprocess_ANCOM <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb492-2"><a href="principals-of-differential-analysis-methods.html#cb492-2" aria-hidden="true" tabindex="-1"></a>          feature_table,</span>
<span id="cb492-3"><a href="principals-of-differential-analysis-methods.html#cb492-3" aria-hidden="true" tabindex="-1"></a>          meta_data,</span>
<span id="cb492-4"><a href="principals-of-differential-analysis-methods.html#cb492-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">sample_var =</span> <span class="cn">NULL</span>,</span>
<span id="cb492-5"><a href="principals-of-differential-analysis-methods.html#cb492-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">group_var =</span> <span class="cn">NULL</span>,</span>
<span id="cb492-6"><a href="principals-of-differential-analysis-methods.html#cb492-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">out_cut =</span> <span class="fl">0.05</span>,</span>
<span id="cb492-7"><a href="principals-of-differential-analysis-methods.html#cb492-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">zero_cut =</span> <span class="fl">0.90</span>,</span>
<span id="cb492-8"><a href="principals-of-differential-analysis-methods.html#cb492-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">lib_cut =</span> <span class="dv">1000</span>,</span>
<span id="cb492-9"><a href="principals-of-differential-analysis-methods.html#cb492-9" aria-hidden="true" tabindex="-1"></a>          <span class="at">neg_lb =</span> <span class="cn">FALSE</span>)</span>
<span id="cb492-10"><a href="principals-of-differential-analysis-methods.html#cb492-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb492-11"><a href="principals-of-differential-analysis-methods.html#cb492-11" aria-hidden="true" tabindex="-1"></a>main_ANCOM <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb492-12"><a href="principals-of-differential-analysis-methods.html#cb492-12" aria-hidden="true" tabindex="-1"></a>                  feature_table,</span>
<span id="cb492-13"><a href="principals-of-differential-analysis-methods.html#cb492-13" aria-hidden="true" tabindex="-1"></a>                  meta_data,</span>
<span id="cb492-14"><a href="principals-of-differential-analysis-methods.html#cb492-14" aria-hidden="true" tabindex="-1"></a>                  <span class="at">struc_zero =</span> <span class="cn">NULL</span>,</span>
<span id="cb492-15"><a href="principals-of-differential-analysis-methods.html#cb492-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">group_var =</span> <span class="cn">NULL</span>,</span>
<span id="cb492-16"><a href="principals-of-differential-analysis-methods.html#cb492-16" aria-hidden="true" tabindex="-1"></a>                  <span class="at">p_adj_method =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb492-17"><a href="principals-of-differential-analysis-methods.html#cb492-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">alpha =</span> <span class="fl">0.05</span>,</span>
<span id="cb492-18"><a href="principals-of-differential-analysis-methods.html#cb492-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">adj_formula =</span> <span class="cn">NULL</span>,</span>
<span id="cb492-19"><a href="principals-of-differential-analysis-methods.html#cb492-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">rand_formula =</span> <span class="cn">NULL</span>,</span>
<span id="cb492-20"><a href="principals-of-differential-analysis-methods.html#cb492-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">w_cutoff =</span> <span class="fl">0.7</span>,</span>
<span id="cb492-21"><a href="principals-of-differential-analysis-methods.html#cb492-21" aria-hidden="true" tabindex="-1"></a>                  ...)</span></code></pre></div>
</div>
<div id="xmas2-syntax-1" class="section level3 hasAnchor" number="14.2.6">
<h3><span class="header-section-number">14.2.6</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb493-1"><a href="principals-of-differential-analysis-methods.html#cb493-1" aria-hidden="true" tabindex="-1"></a>prepro_res <span class="ot">&lt;-</span> <span class="fu">preprocess_ANCOM</span>(</span>
<span id="cb493-2"><a href="principals-of-differential-analysis-methods.html#cb493-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">feature_table =</span> otu_tab,</span>
<span id="cb493-3"><a href="principals-of-differential-analysis-methods.html#cb493-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">meta_data =</span> sam_tab <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;TempRowNames&quot;</span>),</span>
<span id="cb493-4"><a href="principals-of-differential-analysis-methods.html#cb493-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">sample_var =</span> <span class="st">&quot;TempRowNames&quot;</span>,</span>
<span id="cb493-5"><a href="principals-of-differential-analysis-methods.html#cb493-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">group_var =</span> <span class="st">&quot;Compvar&quot;</span>,</span>
<span id="cb493-6"><a href="principals-of-differential-analysis-methods.html#cb493-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">out_cut =</span> <span class="fl">0.05</span>,</span>
<span id="cb493-7"><a href="principals-of-differential-analysis-methods.html#cb493-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">zero_cut =</span> <span class="fl">0.90</span>,</span>
<span id="cb493-8"><a href="principals-of-differential-analysis-methods.html#cb493-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">lib_cut =</span> <span class="dv">1000</span>,</span>
<span id="cb493-9"><a href="principals-of-differential-analysis-methods.html#cb493-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">neg_lb =</span> <span class="cn">FALSE</span>)</span>
<span id="cb493-10"><a href="principals-of-differential-analysis-methods.html#cb493-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-11"><a href="principals-of-differential-analysis-methods.html#cb493-11" aria-hidden="true" tabindex="-1"></a>da_res <span class="ot">&lt;-</span> <span class="fu">main_ANCOM</span>(</span>
<span id="cb493-12"><a href="principals-of-differential-analysis-methods.html#cb493-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">feature_table =</span> prepro_res<span class="sc">$</span>feature_table,</span>
<span id="cb493-13"><a href="principals-of-differential-analysis-methods.html#cb493-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">meta_data =</span> prepro_res<span class="sc">$</span>meta_data,</span>
<span id="cb493-14"><a href="principals-of-differential-analysis-methods.html#cb493-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">struc_zero =</span> prepro_res<span class="sc">$</span>structure_zeros,</span>
<span id="cb493-15"><a href="principals-of-differential-analysis-methods.html#cb493-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">group_var =</span> <span class="st">&quot;Compvar&quot;</span>,</span>
<span id="cb493-16"><a href="principals-of-differential-analysis-methods.html#cb493-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">p_adj_method =</span> p_adjust,</span>
<span id="cb493-17"><a href="principals-of-differential-analysis-methods.html#cb493-17" aria-hidden="true" tabindex="-1"></a>      <span class="at">alpha =</span> pvalue_cutoff,</span>
<span id="cb493-18"><a href="principals-of-differential-analysis-methods.html#cb493-18" aria-hidden="true" tabindex="-1"></a>      <span class="at">rand_formula =</span> <span class="cn">NULL</span>,</span>
<span id="cb493-19"><a href="principals-of-differential-analysis-methods.html#cb493-19" aria-hidden="true" tabindex="-1"></a>      <span class="at">w_cutoff =</span> <span class="fl">0.7</span>)</span></code></pre></div>
</div>
<div id="notes-1" class="section level3 hasAnchor" number="14.2.7">
<h3><span class="header-section-number">14.2.7</span> Notes<a href="principals-of-differential-analysis-methods.html#notes-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>the column <strong>detected_</strong> with TRUE shows the significant taxa by `run_ancom``;</li>
</ol>
</div>
</div>
<div id="corncob-1" class="section level2 hasAnchor" number="14.3">
<h2><span class="header-section-number">14.3</span> corncob<a href="principals-of-differential-analysis-methods.html#corncob-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>corncob is from <span class="citation">(<a href="#ref-martin2020modeling" role="doc-biblioref">Martin, Witten, and Willis 2020</a>)</span></p>
<div id="vignette-2" class="section level3 hasAnchor" number="14.3.1">
<h3><span class="header-section-number">14.3.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://cran.r-project.org/web/packages/corncob/vignettes/corncob-intro.pdf">Count Regression for Correlated Observations with the Beta-Binomial: corncob</a></p>
</div>
<div id="input-data-type-2" class="section level3 hasAnchor" number="14.3.2">
<h3><span class="header-section-number">14.3.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type should be <strong>Raw Counts</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-2" class="section level3 hasAnchor" number="14.3.3">
<h3><span class="header-section-number">14.3.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>data</strong>: a data frame containing the OTU table, or
phyloseq object containing the variables in the models (counts matrix).</p></li>
<li><p><strong>formula</strong>: an object of class formula without the response.</p></li>
<li><p><strong>test</strong>: test method, options include: â€œWaldâ€ and â€œLRTâ€ for two groups comparison (default: â€œWaldâ€).</p></li>
<li><p><strong>fdr_cutoff</strong>: cutoff of FDR.</p></li>
</ol>
</div>
<div id="procedures-2" class="section level3 hasAnchor" number="14.3.4">
<h3><span class="header-section-number">14.3.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Taxon count abundance fit to a <strong>beta-binomial model</strong>, using
logit link functions for both the mean and overdispersion.</p></li>
<li><p>Wald tests for significance testing, and BH for FDR-corrected p-values.</p></li>
</ol>
</div>
<div id="key-output-2" class="section level3 hasAnchor" number="14.3.5">
<h3><span class="header-section-number">14.3.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>EffectSize</strong>: Effect Size to evaluate the power of pvalue.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax-2" class="section level3 hasAnchor" number="14.3.6">
<h3><span class="header-section-number">14.3.6</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb494"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb494-1"><a href="principals-of-differential-analysis-methods.html#cb494-1" aria-hidden="true" tabindex="-1"></a><span class="fu">differentialTest</span>(</span>
<span id="cb494-2"><a href="principals-of-differential-analysis-methods.html#cb494-2" aria-hidden="true" tabindex="-1"></a>  formula,</span>
<span id="cb494-3"><a href="principals-of-differential-analysis-methods.html#cb494-3" aria-hidden="true" tabindex="-1"></a>  phi.formula,</span>
<span id="cb494-4"><a href="principals-of-differential-analysis-methods.html#cb494-4" aria-hidden="true" tabindex="-1"></a>  formula_null,</span>
<span id="cb494-5"><a href="principals-of-differential-analysis-methods.html#cb494-5" aria-hidden="true" tabindex="-1"></a>  phi.formula_null,</span>
<span id="cb494-6"><a href="principals-of-differential-analysis-methods.html#cb494-6" aria-hidden="true" tabindex="-1"></a>  data,</span>
<span id="cb494-7"><a href="principals-of-differential-analysis-methods.html#cb494-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">link =</span> <span class="st">&quot;logit&quot;</span>,</span>
<span id="cb494-8"><a href="principals-of-differential-analysis-methods.html#cb494-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi.link =</span> <span class="st">&quot;logit&quot;</span>,</span>
<span id="cb494-9"><a href="principals-of-differential-analysis-methods.html#cb494-9" aria-hidden="true" tabindex="-1"></a>  test,</span>
<span id="cb494-10"><a href="principals-of-differential-analysis-methods.html#cb494-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">boot =</span> <span class="cn">FALSE</span>,</span>
<span id="cb494-11"><a href="principals-of-differential-analysis-methods.html#cb494-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">B =</span> <span class="dv">1000</span>,</span>
<span id="cb494-12"><a href="principals-of-differential-analysis-methods.html#cb494-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample_data =</span> <span class="cn">NULL</span>,</span>
<span id="cb494-13"><a href="principals-of-differential-analysis-methods.html#cb494-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>,</span>
<span id="cb494-14"><a href="principals-of-differential-analysis-methods.html#cb494-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">filter_discriminant =</span> <span class="cn">TRUE</span>,</span>
<span id="cb494-15"><a href="principals-of-differential-analysis-methods.html#cb494-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">fdr_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb494-16"><a href="principals-of-differential-analysis-methods.html#cb494-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">fdr =</span> <span class="st">&quot;fdr&quot;</span>,</span>
<span id="cb494-17"><a href="principals-of-differential-analysis-methods.html#cb494-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">full_output =</span> <span class="cn">FALSE</span>,</span>
<span id="cb494-18"><a href="principals-of-differential-analysis-methods.html#cb494-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">inits =</span> <span class="cn">NULL</span>,</span>
<span id="cb494-19"><a href="principals-of-differential-analysis-methods.html#cb494-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">inits_null =</span> <span class="cn">NULL</span>,</span>
<span id="cb494-20"><a href="principals-of-differential-analysis-methods.html#cb494-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">try_only =</span> <span class="cn">NULL</span>,</span>
<span id="cb494-21"><a href="principals-of-differential-analysis-methods.html#cb494-21" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb494-22"><a href="principals-of-differential-analysis-methods.html#cb494-22" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="xmas2-syntax-2" class="section level3 hasAnchor" number="14.3.7">
<h3><span class="header-section-number">14.3.7</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb495"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb495-1"><a href="principals-of-differential-analysis-methods.html#cb495-1" aria-hidden="true" tabindex="-1"></a>da_res <span class="ot">&lt;-</span> corncob<span class="sc">::</span><span class="fu">differentialTest</span>(</span>
<span id="cb495-2"><a href="principals-of-differential-analysis-methods.html#cb495-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">data =</span> ps_normed_new,</span>
<span id="cb495-3"><a href="principals-of-differential-analysis-methods.html#cb495-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">formula =</span> my_formula,</span>
<span id="cb495-4"><a href="principals-of-differential-analysis-methods.html#cb495-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">phi.formula =</span> my_formula,</span>
<span id="cb495-5"><a href="principals-of-differential-analysis-methods.html#cb495-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">phi.formula_null =</span> my_formula,</span>
<span id="cb495-6"><a href="principals-of-differential-analysis-methods.html#cb495-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">formula_null =</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb495-7"><a href="principals-of-differential-analysis-methods.html#cb495-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">test =</span> method,</span>
<span id="cb495-8"><a href="principals-of-differential-analysis-methods.html#cb495-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">boot =</span> F,</span>
<span id="cb495-9"><a href="principals-of-differential-analysis-methods.html#cb495-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fdr_cutoff =</span> pvalue_cutoff,</span>
<span id="cb495-10"><a href="principals-of-differential-analysis-methods.html#cb495-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">fdr =</span> p_adjust)</span></code></pre></div>
</div>
<div id="notes-2" class="section level3 hasAnchor" number="14.3.8">
<h3><span class="header-section-number">14.3.8</span> Notes<a href="principals-of-differential-analysis-methods.html#notes-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Columns with Log2FoldChange are based on <strong>raw counts</strong>;</li>
</ol>
</div>
</div>
<div id="deseq2-1" class="section level2 hasAnchor" number="14.4">
<h2><span class="header-section-number">14.4</span> DESeq2<a href="principals-of-differential-analysis-methods.html#deseq2-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>DESeq2 is from <span class="citation">(<a href="#ref-love2014moderated" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span></p>
<div id="vignette-3" class="section level3 hasAnchor" number="14.4.1">
<h3><span class="header-section-number">14.4.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2</a></p>
</div>
<div id="input-data-type-3" class="section level3 hasAnchor" number="14.4.2">
<h3><span class="header-section-number">14.4.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type should be <strong>un-normalized raw counts</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-3" class="section level3 hasAnchor" number="14.4.3">
<h3><span class="header-section-number">14.4.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>countData</strong>: Raw counts matrix without normalization (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>colData</strong>: a DataFrame or data.frame with at least a single column..</p></li>
<li><p><strong>design</strong>: a formula or matrix. the formula expresses how the counts for each gene depend on the variables in colData.</p></li>
</ol>
</div>
<div id="procedures-3" class="section level3 hasAnchor" number="14.4.4">
<h3><span class="header-section-number">14.4.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>estimation of size factors, which are used to normalize library sizes in a model-based fashion;</p></li>
<li><p>estimation of dispersions from the negative binomial likelihood for each feature, and subsequent shrinkage of each dispersion estimate towards the parametric (default) trendline by empirical Bayes;</p></li>
<li><p>fitting each feature to the specified class groupings with negative binomial generalized linear models and performing hypothesis testing, for which we chose the default Wald test;</p></li>
<li><p>Finally, using the <code>results</code> function to obtain the resulting BH FDR-corrected p-values.</p></li>
</ol>
</div>
<div id="key-output-3" class="section level3 hasAnchor" number="14.4.5">
<h3><span class="header-section-number">14.4.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>logFC</strong>: Log2FoldChange by <code>DESeq2::DESeq</code>.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax-3" class="section level3 hasAnchor" number="14.4.6">
<h3><span class="header-section-number">14.4.6</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb496"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb496-1"><a href="principals-of-differential-analysis-methods.html#cb496-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb496-2"><a href="principals-of-differential-analysis-methods.html#cb496-2" aria-hidden="true" tabindex="-1"></a>  countData,</span>
<span id="cb496-3"><a href="principals-of-differential-analysis-methods.html#cb496-3" aria-hidden="true" tabindex="-1"></a>  colData,</span>
<span id="cb496-4"><a href="principals-of-differential-analysis-methods.html#cb496-4" aria-hidden="true" tabindex="-1"></a>  design,</span>
<span id="cb496-5"><a href="principals-of-differential-analysis-methods.html#cb496-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">tidy =</span> <span class="cn">FALSE</span>,</span>
<span id="cb496-6"><a href="principals-of-differential-analysis-methods.html#cb496-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">ignoreRank =</span> <span class="cn">FALSE</span>,</span>
<span id="cb496-7"><a href="principals-of-differential-analysis-methods.html#cb496-7" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb496-8"><a href="principals-of-differential-analysis-methods.html#cb496-8" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="xmas2-syntax-3" class="section level3 hasAnchor" number="14.4.7">
<h3><span class="header-section-number">14.4.7</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb497"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb497-1"><a href="principals-of-differential-analysis-methods.html#cb497-1" aria-hidden="true" tabindex="-1"></a><span class="do">## build Matrix</span></span>
<span id="cb497-2"><a href="principals-of-differential-analysis-methods.html#cb497-2" aria-hidden="true" tabindex="-1"></a>ddsm <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb497-3"><a href="principals-of-differential-analysis-methods.html#cb497-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">countData =</span> otu_tab,</span>
<span id="cb497-4"><a href="principals-of-differential-analysis-methods.html#cb497-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">colData =</span>  sam_tab,</span>
<span id="cb497-5"><a href="principals-of-differential-analysis-methods.html#cb497-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">design =</span><span class="sc">~</span> Compvar)</span>
<span id="cb497-6"><a href="principals-of-differential-analysis-methods.html#cb497-6" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ddsm)</span>
<span id="cb497-7"><a href="principals-of-differential-analysis-methods.html#cb497-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb497-8"><a href="principals-of-differential-analysis-methods.html#cb497-8" aria-hidden="true" tabindex="-1"></a><span class="co"># extract the results</span></span>
<span id="cb497-9"><a href="principals-of-differential-analysis-methods.html#cb497-9" aria-hidden="true" tabindex="-1"></a>DESeq_res <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;Compvar&quot;</span>, <span class="fu">rev</span>(group_names)))</span></code></pre></div>
</div>
</div>
<div id="edger-1" class="section level2 hasAnchor" number="14.5">
<h2><span class="header-section-number">14.5</span> edgeR<a href="principals-of-differential-analysis-methods.html#edger-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>edgeR is from <span class="citation">(<a href="#ref-robinson2010scaling" role="doc-biblioref">Robinson and Oshlack 2010</a>)</span></p>
<div id="vignette-4" class="section level3 hasAnchor" number="14.5.1">
<h3><span class="header-section-number">14.5.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf">edgeR</a></p>
</div>
<div id="input-data-type-4" class="section level3 hasAnchor" number="14.5.2">
<h3><span class="header-section-number">14.5.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type should be <strong>Raw Counts</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-4" class="section level3 hasAnchor" number="14.5.3">
<h3><span class="header-section-number">14.5.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>counts</strong>: Raw counts matrix (Rows-&gt;Features; columns-&gt;Samples) for DEGlist.</p></li>
<li><p><strong>design</strong>: A vector of group labels for testing.</p></li>
<li><p><strong>contrast</strong>: A string for comparison groups.</p></li>
</ol>
</div>
<div id="procedures-4" class="section level3 hasAnchor" number="14.5.4">
<h3><span class="header-section-number">14.5.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Add a pseudocount of 1 to the non-rarefied feature table and used the
function <code>calcNormFactors</code> from the edgeR package to compute relative
log expression normalization factors;</p></li>
<li><p>Negative binomial dispersion parameters were then estimated using the
functions <code>estimateCommonDisp</code> followed by <code>estimateTagwiseDisp</code> to shrink
feature-wise dispersion estimates through an empirical Bayes approach;</p></li>
<li><p><code>exactTest</code> for negative binomial data to identify features that differ
between the specified groups;</p></li>
<li><p>The resulting p-values were then corrected for multiple testing with
the BH method with the function <code>topTags</code>.</p></li>
</ol>
</div>
<div id="key-output-4" class="section level3 hasAnchor" number="14.5.5">
<h3><span class="header-section-number">14.5.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>logFC</strong>: Log2FoldChange by <code>edgeR::topTags</code>.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax-4" class="section level3 hasAnchor" number="14.5.6">
<h3><span class="header-section-number">14.5.6</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb498"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb498-1"><a href="principals-of-differential-analysis-methods.html#cb498-1" aria-hidden="true" tabindex="-1"></a><span class="co"># DGEList object &amp; normalized factor</span></span>
<span id="cb498-2"><a href="principals-of-differential-analysis-methods.html#cb498-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb498-3"><a href="principals-of-differential-analysis-methods.html#cb498-3" aria-hidden="true" tabindex="-1"></a><span class="co"># GLM estimates of dispersion</span></span>
<span id="cb498-4"><a href="principals-of-differential-analysis-methods.html#cb498-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb498-5"><a href="principals-of-differential-analysis-methods.html#cb498-5" aria-hidden="true" tabindex="-1"></a><span class="co"># GLM testing for differential expression</span></span></code></pre></div>
</div>
<div id="xmas2-syntax-4" class="section level3 hasAnchor" number="14.5.7">
<h3><span class="header-section-number">14.5.7</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb499-1"><a href="principals-of-differential-analysis-methods.html#cb499-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter data</span></span>
<span id="cb499-2"><a href="principals-of-differential-analysis-methods.html#cb499-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(edgeR<span class="sc">::</span><span class="fu">cpm</span>(otu_tab) <span class="sc">&gt;</span> <span class="dv">100</span>) <span class="sc">&gt;=</span> <span class="dv">2</span></span>
<span id="cb499-3"><a href="principals-of-differential-analysis-methods.html#cb499-3" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> otu_tab[keep, ]</span>
<span id="cb499-4"><a href="principals-of-differential-analysis-methods.html#cb499-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb499-5"><a href="principals-of-differential-analysis-methods.html#cb499-5" aria-hidden="true" tabindex="-1"></a><span class="co"># DGEList object: otu_tab -&gt; colnames:Samples; rownames:Taxa</span></span>
<span id="cb499-6"><a href="principals-of-differential-analysis-methods.html#cb499-6" aria-hidden="true" tabindex="-1"></a>dge_obj <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">DGEList</span>(<span class="at">counts =</span> otu_tab)</span>
<span id="cb499-7"><a href="principals-of-differential-analysis-methods.html#cb499-7" aria-hidden="true" tabindex="-1"></a>dge_obj_factors <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">calcNormFactors</span>(dge_obj)</span>
<span id="cb499-8"><a href="principals-of-differential-analysis-methods.html#cb499-8" aria-hidden="true" tabindex="-1"></a>limma<span class="sc">::</span><span class="fu">plotMDS</span>(dge_obj_factors, <span class="at">method =</span> <span class="st">&quot;bcv&quot;</span>, <span class="at">col =</span> <span class="fu">as.numeric</span>(sam_tab<span class="sc">$</span>Compvar))</span>
<span id="cb499-9"><a href="principals-of-differential-analysis-methods.html#cb499-9" aria-hidden="true" tabindex="-1"></a>graphics<span class="sc">::</span><span class="fu">legend</span>(<span class="st">&quot;bottomleft&quot;</span>, group_names, <span class="at">col =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">pch =</span> <span class="dv">20</span>)</span>
<span id="cb499-10"><a href="principals-of-differential-analysis-methods.html#cb499-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb499-11"><a href="principals-of-differential-analysis-methods.html#cb499-11" aria-hidden="true" tabindex="-1"></a><span class="co"># GLM estimates of dispersion</span></span>
<span id="cb499-12"><a href="principals-of-differential-analysis-methods.html#cb499-12" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">estimateGLMCommonDisp</span>(dge_obj_factors, design)</span>
<span id="cb499-13"><a href="principals-of-differential-analysis-methods.html#cb499-13" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">estimateGLMTrendedDisp</span>(dge, design, <span class="at">method =</span> <span class="st">&quot;power&quot;</span>)</span>
<span id="cb499-14"><a href="principals-of-differential-analysis-methods.html#cb499-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb499-15"><a href="principals-of-differential-analysis-methods.html#cb499-15" aria-hidden="true" tabindex="-1"></a><span class="co"># You can change method to &quot;auto&quot;, &quot;bin.spline&quot;, &quot;power&quot;, &quot;spline&quot;, &quot;bin.loess&quot;.</span></span>
<span id="cb499-16"><a href="principals-of-differential-analysis-methods.html#cb499-16" aria-hidden="true" tabindex="-1"></a><span class="co"># The default is &quot;auto&quot; which chooses &quot;bin.spline&quot; when &gt; 200 tags and &quot;power&quot; otherwise.</span></span>
<span id="cb499-17"><a href="principals-of-differential-analysis-methods.html#cb499-17" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">estimateGLMTagwiseDisp</span>(dge, design)</span>
<span id="cb499-18"><a href="principals-of-differential-analysis-methods.html#cb499-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb499-19"><a href="principals-of-differential-analysis-methods.html#cb499-19" aria-hidden="true" tabindex="-1"></a><span class="co"># GLM testing for differential expression:</span></span>
<span id="cb499-20"><a href="principals-of-differential-analysis-methods.html#cb499-20" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmFit</span>(dge, design)</span>
<span id="cb499-21"><a href="principals-of-differential-analysis-methods.html#cb499-21" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">glmLRT</span>(fit, <span class="at">contrast =</span> contrast) <span class="co"># Normal:-1; Tumor:1</span></span>
<span id="cb499-22"><a href="principals-of-differential-analysis-methods.html#cb499-22" aria-hidden="true" tabindex="-1"></a>edgeR_res <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">topTags</span>(fit2, <span class="at">n =</span> <span class="fu">nrow</span>(otu_tab)) <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>()</span></code></pre></div>
</div>
<div id="notes-3" class="section level3 hasAnchor" number="14.5.8">
<h3><span class="header-section-number">14.5.8</span> Notes<a href="principals-of-differential-analysis-methods.html#notes-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Effect Size could be regarded as Log2FoldChange;</li>
</ol>
</div>
</div>
<div id="lefse" class="section level2 hasAnchor" number="14.6">
<h2><span class="header-section-number">14.6</span> LEfSe<a href="principals-of-differential-analysis-methods.html#lefse" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>LEfSe is from <span class="citation">(<a href="#ref-segata2011metagenomic" role="doc-biblioref">Segata et al. 2011</a>)</span></p>
<div id="vignette-5" class="section level3 hasAnchor" number="14.6.1">
<h3><span class="header-section-number">14.6.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://bioconductor.org/packages/devel/bioc/vignettes/lefser/inst/doc/lefser.html">Liner discriminant analysis (LDA) effect size: LEfSe</a></p>
</div>
<div id="input-data-type-5" class="section level3 hasAnchor" number="14.6.2">
<h3><span class="header-section-number">14.6.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type could be <strong>Raw Counts</strong> or <strong>Relative abundance</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA.</p></li>
</ol>
</div>
<div id="input-key-arguments-5" class="section level3 hasAnchor" number="14.6.3">
<h3><span class="header-section-number">14.6.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>assays</strong>: data.frame or matrix of features (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>colData</strong>: data.frame or matrix with group information.</p></li>
<li><p><strong>kruskal.threshold </strong>: cutoff of KW test pvalue.</p></li>
<li><p><strong>wilcox.threshold</strong>: cutoff of wilcox test pvalue.</p></li>
<li><p><strong>lda.threshold</strong>: cutoff of LDA scores.</p></li>
</ol>
</div>
<div id="procedures-5" class="section level3 hasAnchor" number="14.6.4">
<h3><span class="header-section-number">14.6.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Normalizing the matrix data using <em>CPM: pre-sample normalization of the sum of the values to 1e+06.</em>;</p></li>
<li><p>Performing a Kruskal-Wallis (which in our two-group case reduces to
the Wilcoxon rank-sum) hypothesis test to identify potential differentially abundant features;</p></li>
<li><p>Using differentially abundant features to perform linear discriminant analysis (LDA) of
class labels on abundances to estimate the effect sizes for sig- nificant features.</p></li>
<li><p>Only features with scaled LDA analysis scores above the threshold (default: 2.0)
were called as DA.</p></li>
</ol>
</div>
<div id="key-output-5" class="section level3 hasAnchor" number="14.6.5">
<h3><span class="header-section-number">14.6.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>LDA score</strong>: LDA score to determine the significant taxa.</p></li>
</ol>
</div>
<div id="original-syntax-5" class="section level3 hasAnchor" number="14.6.6">
<h3><span class="header-section-number">14.6.6</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb500"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb500-1"><a href="principals-of-differential-analysis-methods.html#cb500-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lefser</span>(expr,</span>
<span id="cb500-2"><a href="principals-of-differential-analysis-methods.html#cb500-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">kruskal.threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb500-3"><a href="principals-of-differential-analysis-methods.html#cb500-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">wilcox.threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb500-4"><a href="principals-of-differential-analysis-methods.html#cb500-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">lda.threshold =</span> <span class="fl">2.0</span>,</span>
<span id="cb500-5"><a href="principals-of-differential-analysis-methods.html#cb500-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">groupCol =</span> <span class="st">&quot;GROUP&quot;</span>,</span>
<span id="cb500-6"><a href="principals-of-differential-analysis-methods.html#cb500-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">blockCol =</span> <span class="cn">NULL</span>,</span>
<span id="cb500-7"><a href="principals-of-differential-analysis-methods.html#cb500-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">assay =</span> 1L,</span>
<span id="cb500-8"><a href="principals-of-differential-analysis-methods.html#cb500-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">trim.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="xmas2-syntax-5" class="section level3 hasAnchor" number="14.6.7">
<h3><span class="header-section-number">14.6.7</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb501"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb501-1"><a href="principals-of-differential-analysis-methods.html#cb501-1" aria-hidden="true" tabindex="-1"></a>se_object <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">SummarizedExperiment</span>(</span>
<span id="cb501-2"><a href="principals-of-differential-analysis-methods.html#cb501-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">assays =</span> <span class="fu">list</span>(<span class="at">counts =</span> otu_tab),</span>
<span id="cb501-3"><a href="principals-of-differential-analysis-methods.html#cb501-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">colData =</span> sam_tab,</span>
<span id="cb501-4"><a href="principals-of-differential-analysis-methods.html#cb501-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">metadata =</span> <span class="st">&quot;Profile&quot;</span>)</span>
<span id="cb501-5"><a href="principals-of-differential-analysis-methods.html#cb501-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb501-6"><a href="principals-of-differential-analysis-methods.html#cb501-6" aria-hidden="true" tabindex="-1"></a>lefse_res <span class="ot">&lt;-</span> <span class="fu">lefser</span>(se_object,</span>
<span id="cb501-7"><a href="principals-of-differential-analysis-methods.html#cb501-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">kruskal.threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb501-8"><a href="principals-of-differential-analysis-methods.html#cb501-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">wilcox.threshold  =</span> wl.p,</span>
<span id="cb501-9"><a href="principals-of-differential-analysis-methods.html#cb501-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">lda.threshold     =</span> Lda,</span>
<span id="cb501-10"><a href="principals-of-differential-analysis-methods.html#cb501-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">groupCol =</span> <span class="st">&quot;Compvar&quot;</span>,</span>
<span id="cb501-11"><a href="principals-of-differential-analysis-methods.html#cb501-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">blockCol =</span> <span class="cn">NULL</span>,</span>
<span id="cb501-12"><a href="principals-of-differential-analysis-methods.html#cb501-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">assay    =</span> 1L,</span>
<span id="cb501-13"><a href="principals-of-differential-analysis-methods.html#cb501-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">trim.names =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="notes-4" class="section level3 hasAnchor" number="14.6.8">
<h3><span class="header-section-number">14.6.8</span> Notes<a href="principals-of-differential-analysis-methods.html#notes-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Effect Size could be regarded as Log2FoldChange;</li>
</ol>
</div>
</div>
<div id="limma-voom" class="section level2 hasAnchor" number="14.7">
<h2><span class="header-section-number">14.7</span> limma voom<a href="principals-of-differential-analysis-methods.html#limma-voom" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>limma-voom is from <span class="citation">(<a href="#ref-law2014voom" role="doc-biblioref">Law et al. 2014</a>)</span></p>
<div id="vignette-6" class="section level3 hasAnchor" number="14.7.1">
<h3><span class="header-section-number">14.7.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf">limma</a></p>
</div>
<div id="input-data-type-6" class="section level3 hasAnchor" number="14.7.2">
<h3><span class="header-section-number">14.7.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type should be <strong>Raw Counts</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-6" class="section level3 hasAnchor" number="14.7.3">
<h3><span class="header-section-number">14.7.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>counts</strong>: a numeric matrix containing raw counts (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>design</strong>: design matrix with rows corresponding to samples and columns to coefficients to be estimated.</p></li>
<li><p><strong>lib.size</strong>: numeric vector containing total library sizes for each sample.</p></li>
<li><p><strong>span</strong>: width of the smoothing window used for the lowess mean-variance trend.</p></li>
</ol>
</div>
<div id="procedures-6" class="section level3 hasAnchor" number="14.7.4">
<h3><span class="header-section-number">14.7.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Normalizing table by the trimmed mean of M-values (TMM) or TMM with
singleton pairing (TMMwsp) option via <code>calcNormFactors function</code>
(highly sparse data);</p></li>
<li><p>After normalization, using the limma R package function <code>voom</code> to
convert normalized counts to log2-counts-per-million and assign precision
weights to each observation based on the mean-variance trend;</p></li>
<li><p>Using the functions <code>lmFit</code>, <code>eBayes</code>, and <code>topTable</code> in the limma R package to fit weighted linear regression models, perform tests based on an empirical Bayes moderated t-statistic76 and obtain BH FDR-corrected p-values.</p></li>
</ol>
</div>
<div id="key-output-6" class="section level3 hasAnchor" number="14.7.5">
<h3><span class="header-section-number">14.7.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>EffectSize</strong> and <strong>logFC</strong>: both results equals to Log2FoldChange.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax-6" class="section level3 hasAnchor" number="14.7.6">
<h3><span class="header-section-number">14.7.6</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="xmas2-syntax-6" class="section level3 hasAnchor" number="14.7.7">
<h3><span class="header-section-number">14.7.7</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb502"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb502-1"><a href="principals-of-differential-analysis-methods.html#cb502-1" aria-hidden="true" tabindex="-1"></a><span class="co"># voom fitting</span></span>
<span id="cb502-2"><a href="principals-of-differential-analysis-methods.html#cb502-2" aria-hidden="true" tabindex="-1"></a>voom_out <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">voom</span>(</span>
<span id="cb502-3"><a href="principals-of-differential-analysis-methods.html#cb502-3" aria-hidden="true" tabindex="-1"></a>    otu_tab,</span>
<span id="cb502-4"><a href="principals-of-differential-analysis-methods.html#cb502-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">design =</span> design,</span>
<span id="cb502-5"><a href="principals-of-differential-analysis-methods.html#cb502-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">lib.size =</span> lib_size,</span>
<span id="cb502-6"><a href="principals-of-differential-analysis-methods.html#cb502-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">span =</span> voom_span)</span>
<span id="cb502-7"><a href="principals-of-differential-analysis-methods.html#cb502-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb502-8"><a href="principals-of-differential-analysis-methods.html#cb502-8" aria-hidden="true" tabindex="-1"></a><span class="co"># linear model</span></span>
<span id="cb502-9"><a href="principals-of-differential-analysis-methods.html#cb502-9" aria-hidden="true" tabindex="-1"></a>fit_out <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">lmFit</span>(voom_out, <span class="at">design =</span> design)</span>
<span id="cb502-10"><a href="principals-of-differential-analysis-methods.html#cb502-10" aria-hidden="true" tabindex="-1"></a>test_out <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">eBayes</span>(fit_out)</span>
<span id="cb502-11"><a href="principals-of-differential-analysis-methods.html#cb502-11" aria-hidden="true" tabindex="-1"></a>test_df <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">topTable</span>(</span>
<span id="cb502-12"><a href="principals-of-differential-analysis-methods.html#cb502-12" aria-hidden="true" tabindex="-1"></a>    test_out,</span>
<span id="cb502-13"><a href="principals-of-differential-analysis-methods.html#cb502-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">number =</span> <span class="fu">nrow</span>(otu_tab),</span>
<span id="cb502-14"><a href="principals-of-differential-analysis-methods.html#cb502-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">adjust.method =</span> p_adjust)</span></code></pre></div>
</div>
<div id="notes-5" class="section level3 hasAnchor" number="14.7.8">
<h3><span class="header-section-number">14.7.8</span> Notes<a href="principals-of-differential-analysis-methods.html#notes-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Effect Size could be regarded as Log2FoldChange;</li>
</ol>
</div>
</div>
<div id="maaslin2" class="section level2 hasAnchor" number="14.8">
<h2><span class="header-section-number">14.8</span> MaAsLin2<a href="principals-of-differential-analysis-methods.html#maaslin2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>MaAsLin2 is from <span class="citation">(<a href="#ref-mallick2021multivariable" role="doc-biblioref">Mallick et al. 2021</a>)</span></p>
<div id="vignette-7" class="section level3 hasAnchor" number="14.8.1">
<h3><span class="header-section-number">14.8.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://rdrr.io/bioc/Maaslin2/f/vignettes/maaslin2.Rmd">Multivariable association discovery in population-scale meta-omics studies: MaAsLin2</a></p>
</div>
<div id="input-data-type-7" class="section level3 hasAnchor" number="14.8.2">
<h3><span class="header-section-number">14.8.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type could be <strong>Raw Counts</strong> or <strong>Relative abundance</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-7" class="section level3 hasAnchor" number="14.8.3">
<h3><span class="header-section-number">14.8.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>input_data</strong>: The tab-delimited input file of features (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>input_metadata</strong>: The tab-delimited input file of metadata.</p></li>
<li><p><strong>normalization</strong>: The normalization method to apply.</p></li>
<li><p><strong>transform</strong>: The transform to apply.</p></li>
<li><p><strong>analysis_method</strong>: The analysis method for linear regression to apply.</p></li>
<li><p><strong>random_effects</strong>: The random effects for the model.</p></li>
<li><p><strong>fixed_effects</strong>: The fixed effects for the model, such as the groups to compare.</p></li>
</ol>
</div>
<div id="procedures-7" class="section level3 hasAnchor" number="14.8.4">
<h3><span class="header-section-number">14.8.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Using <em>arcsine square-root transformation (AST)</em> for transformation
and <em>total sum scaling normalization (TSS)</em> for normalization;</p></li>
<li><p>The function fit a linear model (without random effects chosen) to
each featureâ€™s transformed abundance on the specified sample grouping,
tested significance using a Wald test, and output BH FDR-corrected p-values;</p></li>
</ol>
</div>
<div id="key-output-7" class="section level3 hasAnchor" number="14.8.5">
<h3><span class="header-section-number">14.8.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax-7" class="section level3 hasAnchor" number="14.8.6">
<h3><span class="header-section-number">14.8.6</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb503"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb503-1"><a href="principals-of-differential-analysis-methods.html#cb503-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Maaslin2</span>(</span>
<span id="cb503-2"><a href="principals-of-differential-analysis-methods.html#cb503-2" aria-hidden="true" tabindex="-1"></a>    input_data,</span>
<span id="cb503-3"><a href="principals-of-differential-analysis-methods.html#cb503-3" aria-hidden="true" tabindex="-1"></a>    input_metadata,</span>
<span id="cb503-4"><a href="principals-of-differential-analysis-methods.html#cb503-4" aria-hidden="true" tabindex="-1"></a>    output,</span>
<span id="cb503-5"><a href="principals-of-differential-analysis-methods.html#cb503-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">min_abundance =</span> <span class="fl">0.0</span>,</span>
<span id="cb503-6"><a href="principals-of-differential-analysis-methods.html#cb503-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">min_prevalence =</span> <span class="fl">0.1</span>,</span>
<span id="cb503-7"><a href="principals-of-differential-analysis-methods.html#cb503-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">min_variance =</span> <span class="fl">0.0</span>,</span>
<span id="cb503-8"><a href="principals-of-differential-analysis-methods.html#cb503-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">normalization =</span> <span class="st">&quot;TSS&quot;</span>,</span>
<span id="cb503-9"><a href="principals-of-differential-analysis-methods.html#cb503-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb503-10"><a href="principals-of-differential-analysis-methods.html#cb503-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">analysis_method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb503-11"><a href="principals-of-differential-analysis-methods.html#cb503-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">max_significance =</span> <span class="fl">0.25</span>,</span>
<span id="cb503-12"><a href="principals-of-differential-analysis-methods.html#cb503-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">random_effects =</span> <span class="cn">NULL</span>,</span>
<span id="cb503-13"><a href="principals-of-differential-analysis-methods.html#cb503-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_effects =</span> <span class="cn">NULL</span>,</span>
<span id="cb503-14"><a href="principals-of-differential-analysis-methods.html#cb503-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">correction =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb503-15"><a href="principals-of-differential-analysis-methods.html#cb503-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">standardize =</span> <span class="cn">TRUE</span>,</span>
<span id="cb503-16"><a href="principals-of-differential-analysis-methods.html#cb503-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">cores =</span> <span class="dv">1</span>,</span>
<span id="cb503-17"><a href="principals-of-differential-analysis-methods.html#cb503-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot_heatmap =</span> <span class="cn">TRUE</span>,</span>
<span id="cb503-18"><a href="principals-of-differential-analysis-methods.html#cb503-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot_scatter =</span> <span class="cn">TRUE</span>,</span>
<span id="cb503-19"><a href="principals-of-differential-analysis-methods.html#cb503-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">heatmap_first_n =</span> <span class="dv">50</span>,</span>
<span id="cb503-20"><a href="principals-of-differential-analysis-methods.html#cb503-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">reference =</span> <span class="cn">NULL</span></span>
<span id="cb503-21"><a href="principals-of-differential-analysis-methods.html#cb503-21" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="xmas2-syntax-7" class="section level3 hasAnchor" number="14.8.7">
<h3><span class="header-section-number">14.8.7</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb504"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb504-1"><a href="principals-of-differential-analysis-methods.html#cb504-1" aria-hidden="true" tabindex="-1"></a>da_res <span class="ot">&lt;-</span> Maaslin2<span class="sc">::</span><span class="fu">Maaslin2</span>(</span>
<span id="cb504-2"><a href="principals-of-differential-analysis-methods.html#cb504-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">input_data =</span> otu_tab <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>(), <span class="co"># row-&gt;smapleID; col-&gt;Taxa</span></span>
<span id="cb504-3"><a href="principals-of-differential-analysis-methods.html#cb504-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">input_metadata =</span> sam_tab,</span>
<span id="cb504-4"><a href="principals-of-differential-analysis-methods.html#cb504-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">output =</span> outdir,</span>
<span id="cb504-5"><a href="principals-of-differential-analysis-methods.html#cb504-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">normalization =</span> norm,</span>
<span id="cb504-6"><a href="principals-of-differential-analysis-methods.html#cb504-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">transform =</span> transform,</span>
<span id="cb504-7"><a href="principals-of-differential-analysis-methods.html#cb504-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">analysis_method =</span> method,</span>
<span id="cb504-8"><a href="principals-of-differential-analysis-methods.html#cb504-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">max_significance =</span> pvalue_cutoff,</span>
<span id="cb504-9"><a href="principals-of-differential-analysis-methods.html#cb504-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">fixed_effects =</span> <span class="st">&quot;Compvar&quot;</span>,</span>
<span id="cb504-10"><a href="principals-of-differential-analysis-methods.html#cb504-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">correction =</span> p_adjust,</span>
<span id="cb504-11"><a href="principals-of-differential-analysis-methods.html#cb504-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">plot_heatmap =</span> <span class="cn">FALSE</span>,</span>
<span id="cb504-12"><a href="principals-of-differential-analysis-methods.html#cb504-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">plot_scatter =</span> <span class="cn">FALSE</span>,</span>
<span id="cb504-13"><a href="principals-of-differential-analysis-methods.html#cb504-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">cores =</span> <span class="dv">2</span>)</span></code></pre></div>
</div>
</div>
<div id="metagenomeseq-1" class="section level2 hasAnchor" number="14.9">
<h2><span class="header-section-number">14.9</span> metagenomeSeq<a href="principals-of-differential-analysis-methods.html#metagenomeseq-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>metagenomeSeq is from <span class="citation">(<a href="#ref-paulson2013differential" role="doc-biblioref">Paulson et al. 2013</a>)</span></p>
<div id="vignette-8" class="section level3 hasAnchor" number="14.9.1">
<h3><span class="header-section-number">14.9.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/metagenomeSeq/inst/doc/metagenomeSeq.pdf">metagenomeSeq</a></p>
</div>
<div id="input-data-type-8" class="section level3 hasAnchor" number="14.9.2">
<h3><span class="header-section-number">14.9.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type should be <strong>Raw Counts</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-8" class="section level3 hasAnchor" number="14.9.3">
<h3><span class="header-section-number">14.9.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>reads</strong>: Raw counts matrix (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>conditions</strong>: A vector of group labels for testing.</p></li>
<li><p><strong>mc.samples</strong>: The number of Monte Carlo samples, default is 128.</p></li>
<li><p><strong>denom</strong>: The methods for Geometric Mean calculation, default is â€œallâ€.</p></li>
<li><p><strong>test</strong>: The test to perform, default is â€œt.testâ€.</p></li>
</ol>
</div>
<div id="procedures-8" class="section level3 hasAnchor" number="14.9.4">
<h3><span class="header-section-number">14.9.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Converting the counts and sample information into newMRexperiment
object by the function <code>newMRexperiment</code> from the metagenomeSeq R package;</p></li>
<li><p>Using <code>cumNormStat</code> and <code>cumNorm</code> to apply <strong>cumulative sum-scaling normalization (CSS)</strong>,
which attempts to normalize sequence counts based on the lower-quartile abundance of features;</p></li>
<li><p>Using <code>fitFeatureModel</code> to fit normalized feature counts with
zero-inflated log-normal models (with pseudo-counts of 1 added
prior to log2 transformation) and perform empirical Bayes moderated t-tests,
and <code>MRfulltable</code> to obtain BH FDR-corrected p-values.</p></li>
</ol>
</div>
<div id="key-output-8" class="section level3 hasAnchor" number="14.9.5">
<h3><span class="header-section-number">14.9.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
</ol>
</div>
<div id="xmas2-syntax-8" class="section level3 hasAnchor" number="14.9.6">
<h3><span class="header-section-number">14.9.6</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="principals-of-differential-analysis-methods.html#cb505-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run DA</span></span>
<span id="cb505-2"><a href="principals-of-differential-analysis-methods.html#cb505-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;ZILN&quot;</span>) {</span>
<span id="cb505-3"><a href="principals-of-differential-analysis-methods.html#cb505-3" aria-hidden="true" tabindex="-1"></a>  fit <span class="ot">&lt;-</span> metagenomeSeq<span class="sc">::</span><span class="fu">fitFeatureModel</span>(mgs_summarized, mod)</span>
<span id="cb505-4"><a href="principals-of-differential-analysis-methods.html#cb505-4" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb505-5"><a href="principals-of-differential-analysis-methods.html#cb505-5" aria-hidden="true" tabindex="-1"></a>  fit <span class="ot">&lt;-</span> metagenomeSeq<span class="sc">::</span><span class="fu">fitZig</span>(mgs_summarized, mod)</span>
<span id="cb505-6"><a href="principals-of-differential-analysis-methods.html#cb505-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb505-7"><a href="principals-of-differential-analysis-methods.html#cb505-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb505-8"><a href="principals-of-differential-analysis-methods.html#cb505-8" aria-hidden="true" tabindex="-1"></a><span class="co"># metagenomeSeq vignette: We recommend the user remove features based on</span></span>
<span id="cb505-9"><a href="principals-of-differential-analysis-methods.html#cb505-9" aria-hidden="true" tabindex="-1"></a><span class="co"># the number of estimated effective samples, please see</span></span>
<span id="cb505-10"><a href="principals-of-differential-analysis-methods.html#cb505-10" aria-hidden="true" tabindex="-1"></a><span class="co"># calculateEffectiveSamples. We recommend removing features with less</span></span>
<span id="cb505-11"><a href="principals-of-differential-analysis-methods.html#cb505-11" aria-hidden="true" tabindex="-1"></a><span class="co"># than the average number of effective samples in all features. In</span></span>
<span id="cb505-12"><a href="principals-of-differential-analysis-methods.html#cb505-12" aria-hidden="true" tabindex="-1"></a><span class="co"># essence, setting eff = .5 when using MRcoefs, MRfulltable, or MRtable.</span></span>
<span id="cb505-13"><a href="principals-of-differential-analysis-methods.html#cb505-13" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> metagenomeSeq<span class="sc">::</span><span class="fu">MRcoefs</span>(</span>
<span id="cb505-14"><a href="principals-of-differential-analysis-methods.html#cb505-14" aria-hidden="true" tabindex="-1"></a>    fit,</span>
<span id="cb505-15"><a href="principals-of-differential-analysis-methods.html#cb505-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">number =</span> <span class="fu">ntaxa</span>(ps_normed_new),</span>
<span id="cb505-16"><a href="principals-of-differential-analysis-methods.html#cb505-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">adjustMethod =</span> p_adjust,</span>
<span id="cb505-17"><a href="principals-of-differential-analysis-methods.html#cb505-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="dv">3</span>,</span>
<span id="cb505-18"><a href="principals-of-differential-analysis-methods.html#cb505-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">eff =</span> <span class="fl">0.5</span>)</span>
<span id="cb505-19"><a href="principals-of-differential-analysis-methods.html#cb505-19" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(</span>
<span id="cb505-20"><a href="principals-of-differential-analysis-methods.html#cb505-20" aria-hidden="true" tabindex="-1"></a>    res,</span>
<span id="cb505-21"><a href="principals-of-differential-analysis-methods.html#cb505-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">Pvalue =</span> .data<span class="sc">$</span>pvalues,</span>
<span id="cb505-22"><a href="principals-of-differential-analysis-methods.html#cb505-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">AdjustedPvalue =</span> .data<span class="sc">$</span>adjPvalues)</span></code></pre></div>
</div>
</div>
<div id="t-test-2" class="section level2 hasAnchor" number="14.10">
<h2><span class="header-section-number">14.10</span> t-test<a href="principals-of-differential-analysis-methods.html#t-test-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="input-data-type-9" class="section level3 hasAnchor" number="14.10.1">
<h3><span class="header-section-number">14.10.1</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type could be <strong>Raw Counts</strong> or <strong>relative abundance</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-9" class="section level3 hasAnchor" number="14.10.2">
<h3><span class="header-section-number">14.10.2</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>x</strong>: Raw counts matrix (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>y</strong>: A vector of group labels for testing.</p></li>
</ol>
</div>
<div id="procedures-9" class="section level3 hasAnchor" number="14.10.3">
<h3><span class="header-section-number">14.10.3</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Applying <em>total sum scaling normalization (TSS)</em> or not to the feature table;</p></li>
<li><p>Performing an unpaired Welchâ€™s t-test for each feature to compare the specified groups;</p></li>
<li><p>Correcting the resulting p-values with the BH method.</p></li>
</ol>
</div>
<div id="key-output-9" class="section level3 hasAnchor" number="14.10.4">
<h3><span class="header-section-number">14.10.4</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax-8" class="section level3 hasAnchor" number="14.10.5">
<h3><span class="header-section-number">14.10.5</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb506-1"><a href="principals-of-differential-analysis-methods.html#cb506-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(x, <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb506-2"><a href="principals-of-differential-analysis-methods.html#cb506-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">alternative =</span> <span class="fu">c</span>(<span class="st">&quot;two.sided&quot;</span>, <span class="st">&quot;less&quot;</span>, <span class="st">&quot;greater&quot;</span>),</span>
<span id="cb506-3"><a href="principals-of-differential-analysis-methods.html#cb506-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">mu =</span> <span class="dv">0</span>, <span class="at">paired =</span> <span class="cn">FALSE</span>, <span class="at">var.equal =</span> <span class="cn">FALSE</span>,</span>
<span id="cb506-4"><a href="principals-of-differential-analysis-methods.html#cb506-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">conf.level =</span> <span class="fl">0.95</span>, ...)</span></code></pre></div>
</div>
<div id="xmas2-syntax-9" class="section level3 hasAnchor" number="14.10.6">
<h3><span class="header-section-number">14.10.6</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb507"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb507-1"><a href="principals-of-differential-analysis-methods.html#cb507-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run wilcox rank sum test</span></span>
<span id="cb507-2"><a href="principals-of-differential-analysis-methods.html#cb507-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (paired) {</span>
<span id="cb507-3"><a href="principals-of-differential-analysis-methods.html#cb507-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(paired_column)) {</span>
<span id="cb507-4"><a href="principals-of-differential-analysis-methods.html#cb507-4" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">paired_t</span>(<span class="at">x =</span> otu_tab,</span>
<span id="cb507-5"><a href="principals-of-differential-analysis-methods.html#cb507-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">y =</span> sam_tab,</span>
<span id="cb507-6"><a href="principals-of-differential-analysis-methods.html#cb507-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Compvar&quot;</span>,</span>
<span id="cb507-7"><a href="principals-of-differential-analysis-methods.html#cb507-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">gnames =</span> group_names,</span>
<span id="cb507-8"><a href="principals-of-differential-analysis-methods.html#cb507-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">paired =</span> paired,</span>
<span id="cb507-9"><a href="principals-of-differential-analysis-methods.html#cb507-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">PID =</span> paired_column,</span>
<span id="cb507-10"><a href="principals-of-differential-analysis-methods.html#cb507-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">p_ad =</span> p_adjust)</span>
<span id="cb507-11"><a href="principals-of-differential-analysis-methods.html#cb507-11" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb507-12"><a href="principals-of-differential-analysis-methods.html#cb507-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">&quot;Please provide the paired column for paired test&quot;</span>)</span>
<span id="cb507-13"><a href="principals-of-differential-analysis-methods.html#cb507-13" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb507-14"><a href="principals-of-differential-analysis-methods.html#cb507-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb507-15"><a href="principals-of-differential-analysis-methods.html#cb507-15" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb507-16"><a href="principals-of-differential-analysis-methods.html#cb507-16" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">apply</span>(otu_tab, <span class="dv">1</span>, <span class="cf">function</span>(x, y) {</span>
<span id="cb507-17"><a href="principals-of-differential-analysis-methods.html#cb507-17" aria-hidden="true" tabindex="-1"></a>      dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(x), <span class="at">group =</span> y)</span>
<span id="cb507-18"><a href="principals-of-differential-analysis-methods.html#cb507-18" aria-hidden="true" tabindex="-1"></a>      rest <span class="ot">&lt;-</span> <span class="fu">t.test</span>(<span class="at">data =</span> dat, value <span class="sc">~</span> group)</span>
<span id="cb507-19"><a href="principals-of-differential-analysis-methods.html#cb507-19" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">c</span>(rest<span class="sc">$</span>statistic, rest<span class="sc">$</span>p.value)</span>
<span id="cb507-20"><a href="principals-of-differential-analysis-methods.html#cb507-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(res)</span>
<span id="cb507-21"><a href="principals-of-differential-analysis-methods.html#cb507-21" aria-hidden="true" tabindex="-1"></a>    }, sam_tab<span class="sc">$</span>Compvar) <span class="sc">%&gt;%</span></span>
<span id="cb507-22"><a href="principals-of-differential-analysis-methods.html#cb507-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb507-23"><a href="principals-of-differential-analysis-methods.html#cb507-23" aria-hidden="true" tabindex="-1"></a>      stats<span class="sc">::</span><span class="fu">setNames</span>(<span class="fu">c</span>(<span class="st">&quot;Statistic&quot;</span>, <span class="st">&quot;Pvalue&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb507-24"><a href="principals-of-differential-analysis-methods.html#cb507-24" aria-hidden="true" tabindex="-1"></a>      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;TaxaID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb507-25"><a href="principals-of-differential-analysis-methods.html#cb507-25" aria-hidden="true" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">AdjustedPvalue =</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(Pvalue), <span class="at">method =</span> p_adjust))</span>
<span id="cb507-26"><a href="principals-of-differential-analysis-methods.html#cb507-26" aria-hidden="true" tabindex="-1"></a>  }</span></code></pre></div>
</div>
</div>
<div id="wilcoxon-test" class="section level2 hasAnchor" number="14.11">
<h2><span class="header-section-number">14.11</span> Wilcoxon test<a href="principals-of-differential-analysis-methods.html#wilcoxon-test" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="input-data-type-10" class="section level3 hasAnchor" number="14.11.1">
<h3><span class="header-section-number">14.11.1</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type could be <strong>Raw Counts</strong> or <strong>relative abundance</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-10" class="section level3 hasAnchor" number="14.11.2">
<h3><span class="header-section-number">14.11.2</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>x</strong>: Raw counts matrix (Rows-&gt;Features; columns-&gt;Samples).</p></li>
<li><p><strong>y</strong>: A vector of group labels for testing.</p></li>
</ol>
</div>
<div id="procedures-10" class="section level3 hasAnchor" number="14.11.3">
<h3><span class="header-section-number">14.11.3</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>CLR (after applying a pseudocount of 1) transforms or not to abundances;</p></li>
<li><p>Performing Wilcoxon rank-sum tests for each feature to compare the specified sample groupings;</p></li>
<li><p>Correcting the resulting p-values with the BH method.</p></li>
</ol>
</div>
<div id="key-output-10" class="section level3 hasAnchor" number="14.11.4">
<h3><span class="header-section-number">14.11.4</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax-9" class="section level3 hasAnchor" number="14.11.5">
<h3><span class="header-section-number">14.11.5</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb508"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb508-1"><a href="principals-of-differential-analysis-methods.html#cb508-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(x, <span class="at">y =</span> <span class="cn">NULL</span>,</span>
<span id="cb508-2"><a href="principals-of-differential-analysis-methods.html#cb508-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">alternative =</span> <span class="fu">c</span>(<span class="st">&quot;two.sided&quot;</span>, <span class="st">&quot;less&quot;</span>, <span class="st">&quot;greater&quot;</span>),</span>
<span id="cb508-3"><a href="principals-of-differential-analysis-methods.html#cb508-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">mu =</span> <span class="dv">0</span>, <span class="at">paired =</span> <span class="cn">FALSE</span>, <span class="at">exact =</span> <span class="cn">NULL</span>, <span class="at">correct =</span> <span class="cn">TRUE</span>,</span>
<span id="cb508-4"><a href="principals-of-differential-analysis-methods.html#cb508-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">conf.int =</span> <span class="cn">FALSE</span>, <span class="at">conf.level =</span> <span class="fl">0.95</span>,</span>
<span id="cb508-5"><a href="principals-of-differential-analysis-methods.html#cb508-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">tol.root =</span> <span class="fl">1e-4</span>, <span class="at">digits.rank =</span> <span class="cn">Inf</span>, ...)</span></code></pre></div>
</div>
<div id="xmas2-syntax-10" class="section level3 hasAnchor" number="14.11.6">
<h3><span class="header-section-number">14.11.6</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb509-1"><a href="principals-of-differential-analysis-methods.html#cb509-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run wilcox rank sum test</span></span>
<span id="cb509-2"><a href="principals-of-differential-analysis-methods.html#cb509-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (paired) {</span>
<span id="cb509-3"><a href="principals-of-differential-analysis-methods.html#cb509-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(paired_column)) {</span>
<span id="cb509-4"><a href="principals-of-differential-analysis-methods.html#cb509-4" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">paired_wilcox</span>(<span class="at">x =</span> otu_tab,</span>
<span id="cb509-5"><a href="principals-of-differential-analysis-methods.html#cb509-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">y =</span> sam_tab,</span>
<span id="cb509-6"><a href="principals-of-differential-analysis-methods.html#cb509-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Compvar&quot;</span>,</span>
<span id="cb509-7"><a href="principals-of-differential-analysis-methods.html#cb509-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">gnames =</span> group_names,</span>
<span id="cb509-8"><a href="principals-of-differential-analysis-methods.html#cb509-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">paired =</span> paired,</span>
<span id="cb509-9"><a href="principals-of-differential-analysis-methods.html#cb509-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">PID =</span> paired_column,</span>
<span id="cb509-10"><a href="principals-of-differential-analysis-methods.html#cb509-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">p_ad =</span> p_adjust)</span>
<span id="cb509-11"><a href="principals-of-differential-analysis-methods.html#cb509-11" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb509-12"><a href="principals-of-differential-analysis-methods.html#cb509-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">&quot;Please provide the paired column for paired test&quot;</span>)</span>
<span id="cb509-13"><a href="principals-of-differential-analysis-methods.html#cb509-13" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb509-14"><a href="principals-of-differential-analysis-methods.html#cb509-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb509-15"><a href="principals-of-differential-analysis-methods.html#cb509-15" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb509-16"><a href="principals-of-differential-analysis-methods.html#cb509-16" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">apply</span>(otu_tab, <span class="dv">1</span>, <span class="cf">function</span>(x, y) {</span>
<span id="cb509-17"><a href="principals-of-differential-analysis-methods.html#cb509-17" aria-hidden="true" tabindex="-1"></a>      dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(x), <span class="at">group =</span> y)</span>
<span id="cb509-18"><a href="principals-of-differential-analysis-methods.html#cb509-18" aria-hidden="true" tabindex="-1"></a>      rest <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(<span class="at">data =</span> dat, value <span class="sc">~</span> group)</span>
<span id="cb509-19"><a href="principals-of-differential-analysis-methods.html#cb509-19" aria-hidden="true" tabindex="-1"></a>      res <span class="ot">&lt;-</span> <span class="fu">c</span>(rest<span class="sc">$</span>statistic, rest<span class="sc">$</span>p.value)</span>
<span id="cb509-20"><a href="principals-of-differential-analysis-methods.html#cb509-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(res)</span>
<span id="cb509-21"><a href="principals-of-differential-analysis-methods.html#cb509-21" aria-hidden="true" tabindex="-1"></a>    }, sam_tab<span class="sc">$</span>Compvar) <span class="sc">%&gt;%</span></span>
<span id="cb509-22"><a href="principals-of-differential-analysis-methods.html#cb509-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb509-23"><a href="principals-of-differential-analysis-methods.html#cb509-23" aria-hidden="true" tabindex="-1"></a>    stats<span class="sc">::</span><span class="fu">setNames</span>(<span class="fu">c</span>(<span class="st">&quot;Statistic&quot;</span>, <span class="st">&quot;Pvalue&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb509-24"><a href="principals-of-differential-analysis-methods.html#cb509-24" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;TaxaID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb509-25"><a href="principals-of-differential-analysis-methods.html#cb509-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">AdjustedPvalue =</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(Pvalue), <span class="at">method =</span> p_adjust))</span>
<span id="cb509-26"><a href="principals-of-differential-analysis-methods.html#cb509-26" aria-hidden="true" tabindex="-1"></a>  }</span></code></pre></div>
</div>
</div>
<div id="raida-1" class="section level2 hasAnchor" number="14.12">
<h2><span class="header-section-number">14.12</span> RAIDA<a href="principals-of-differential-analysis-methods.html#raida-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>RAIDA is from <span class="citation">(<a href="#ref-sohn2015robust" role="doc-biblioref">Sohn, Du, and An 2015</a>)</span></p>
<div id="vignette-9" class="section level3 hasAnchor" number="14.12.1">
<h3><span class="header-section-number">14.12.1</span> Vignette<a href="principals-of-differential-analysis-methods.html#vignette-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495302/">A robust approach for identifying differentially abundant features in metagenomic samples: RAIDA</a></p>
</div>
<div id="input-data-type-11" class="section level3 hasAnchor" number="14.12.2">
<h3><span class="header-section-number">14.12.2</span> Input Data Type<a href="principals-of-differential-analysis-methods.html#input-data-type-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>All the input data type should be <strong>Raw Counts</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>16s rRNA sequencing (amplicon sequencing);</p></li>
<li><p>Metagenomic sequencing (absolute abundance);</p></li>
<li><p>RNA-seq.</p></li>
</ol>
</div>
<div id="input-key-arguments-11" class="section level3 hasAnchor" number="14.12.3">
<h3><span class="header-section-number">14.12.3</span> Input Key Arguments<a href="principals-of-differential-analysis-methods.html#input-key-arguments-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>c.data</strong>: a data frame containing counts for both conditions.</p></li>
<li><p><strong>n.lib</strong>: a vector containing the numbers of samples for both conditions.</p></li>
</ol>
</div>
<div id="procedures-11" class="section level3 hasAnchor" number="14.12.4">
<h3><span class="header-section-number">14.12.4</span> Procedures<a href="principals-of-differential-analysis-methods.html#procedures-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Approach for Identifying Differential Abundance (RAIDA)</strong> - is a robust approach for identifying differentially abundant features in metagenomic samples across different conditions. It utilizes the ratio between features in a modified zero-inflated log-normal model.</p>
<ol style="list-style-type: decimal">
<li><p>Utilizing the ratios between the counts of features in each sample, eliminating possible problems associated with counts on different scales within and between conditions;</p></li>
<li><p>To fit ratios with zeros by EM algorithm, using a modified ZIL (zero-inflated log normal: sparsity);</p></li>
<li><p>Constructing a moderated t-statistics for the log ratio of each feature <span class="math inline">\(y_{ij}\)</span> using the estimated mean <span class="math inline">\(\mu\)</span> and variance <span class="math inline">\(r^2\)</span> and obtain <span class="math inline">\(P\)</span> values for the null hypotheses for all features;</p></li>
<li><p>Adjust <span class="math inline">\(P\)</span> values using a multiple testing correction method</p></li>
</ol>
</div>
<div id="key-output-11" class="section level3 hasAnchor" number="14.12.5">
<h3><span class="header-section-number">14.12.5</span> Key output<a href="principals-of-differential-analysis-methods.html#key-output-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa ID.</p></li>
<li><p><strong>AdjustedPvalue</strong>: Adjusted Pvalue by <code>stats::p.adjust</code>.</p></li>
<li><p><strong>Occurrence (All/Groups)</strong>: prevalence of taxa.</p></li>
</ol>
</div>
<div id="original-syntax-10" class="section level3 hasAnchor" number="14.12.6">
<h3><span class="header-section-number">14.12.6</span> Original Syntax<a href="principals-of-differential-analysis-methods.html#original-syntax-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb510-1"><a href="principals-of-differential-analysis-methods.html#cb510-1" aria-hidden="true" tabindex="-1"></a><span class="fu">raida</span>(c.data, </span>
<span id="cb510-2"><a href="principals-of-differential-analysis-methods.html#cb510-2" aria-hidden="true" tabindex="-1"></a>      n.lib, </span>
<span id="cb510-3"><a href="principals-of-differential-analysis-methods.html#cb510-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">show.ref.features =</span> <span class="cn">FALSE</span>, </span>
<span id="cb510-4"><a href="principals-of-differential-analysis-methods.html#cb510-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">show.all.features =</span> <span class="cn">FALSE</span>, </span>
<span id="cb510-5"><a href="principals-of-differential-analysis-methods.html#cb510-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">mtcm =</span> <span class="st">&quot;BH&quot;</span>, </span>
<span id="cb510-6"><a href="principals-of-differential-analysis-methods.html#cb510-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">zsc =</span> <span class="dv">0</span>)</span></code></pre></div>
</div>
<div id="xmas2-syntax-11" class="section level3 hasAnchor" number="14.12.7">
<h3><span class="header-section-number">14.12.7</span> XMAS2 Syntax<a href="principals-of-differential-analysis-methods.html#xmas2-syntax-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb511"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb511-1"><a href="principals-of-differential-analysis-methods.html#cb511-1" aria-hidden="true" tabindex="-1"></a>raida_res <span class="ot">&lt;-</span> RAIDA<span class="sc">::</span><span class="fu">raida</span>(<span class="at">c.data =</span> otu_tab_reorder,</span>
<span id="cb511-2"><a href="principals-of-differential-analysis-methods.html#cb511-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">n.lib =</span> n.samples)</span></code></pre></div>
</div>
</div>
<div id="summary" class="section level2 hasAnchor" number="14.13">
<h2><span class="header-section-number">14.13</span> Summary<a href="principals-of-differential-analysis-methods.html#summary" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>The most commonly assumed data distribution was the negative binomial distribution (DESeq2, edgeR, omnibus, mbzinb).</li>
</ol>
<blockquote>
<p>No data transformations were performed for negative binomial methods, or metagenomeSeq methods, to try and bring the data to normality as non-normality of data is taken into account in their statistical models.</p>
</blockquote>
<ol start="2" style="list-style-type: decimal">
<li><p>The remaining parametric methods (ALDEx2 t-test, t-test, limma-voom) all used statistical tests that assumed a Gaussian distribution of the data, therefore, transformations were needed before analysis, which here included a log transform of some kind.</p></li>
<li><p>Three methods (ALDEx2 Wilcoxon, ANCOM-II, LEfSe) were considered non-parametric (assumes no underlying distribution of data) as they used statistical tests that transformed data to ranks.</p></li>
<li><p>Two of the four negative binomial methods (DESeq2, edgeR) calculated scaling factors for each sample to account for uneven sequence count.</p></li>
<li><p>Cumulative sum scaling (CSS) was used for both metagenomeSeq and MaAsLin2. Total sum scaling (TSS; also referred to as relative abundance) was performed for LEfSe, and for method that did not have a built-in normalization function (t-test) as this strategy is commonly used in the literature. Log-ratio based transformations were used for ALDEx2 methods, ANCOM, and, in addition to TSS, were applied to methods without a built-in normalization function.</p></li>
</ol>
</div>
<div id="advantages-and-disadvantages" class="section level2 hasAnchor" number="14.14">
<h2><span class="header-section-number">14.14</span> Advantages and disadvantages<a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><p>limma-voom, edgeR, Wilcoxon (CLR), and LEfSe output a high number of significant ASVs on average, while ALDEx2 and ANCOM-II tended to identify only a relatively small number of ASVs as significant;</p></li>
<li><p>limma-voom, edgeR, Wilcoxon (CLR), and LEfSe methods have been previously found to exhibit a high FDR, in contrast, ANCOM appropriately controls the FDR;</p></li>
<li><p>We can clearly recommend that users avoid using edgeR (a tool primarily intended for RNA-seq data) as well as LEfSe (without p-value correction) for conducting DA testing with 16S rRNA gene data.</p></li>
<li><p>Users should also be aware that limma voom and the Wilcoxon (CLR) approaches may perform poorly on unfiltered data that is highly sparse.</p></li>
<li><p>More generally, we recommend that users employ several methods and focus on significant features identified by most tools.</p></li>
</ol>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-fernandes2014unifying" class="csl-entry">
Fernandes, Andrew D, Jennifer Ns Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>â€œUnifying the Analysis of High-Throughput Sequencing Datasets: Characterizing RNA-Seq, 16s rRNA Gene Sequencing and Selective Growth Experiments by Compositional Data Analysis.â€</span> <em>Microbiome</em> 2 (1): 1â€“13.
</div>
<div id="ref-law2014voom" class="csl-entry">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. <span>â€œVoom: Precision Weights Unlock Linear Model Analysis Tools for RNA-Seq Read Counts.â€</span> <em>Genome Biology</em> 15 (2): 1â€“17.
</div>
<div id="ref-love2014moderated" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>â€œModerated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.â€</span> <em>Genome Biology</em> 15 (12): 1â€“21.
</div>
<div id="ref-mallick2021multivariable" class="csl-entry">
Mallick, Himel, Ali Rahnavard, Lauren J McIver, Siyuan Ma, Yancong Zhang, Long H Nguyen, Timothy L Tickle, et al. 2021. <span>â€œMultivariable Association Discovery in Population-Scale Meta-Omics Studies.â€</span> <em>PLoS Computational Biology</em> 17 (11): e1009442.
</div>
<div id="ref-mandal2015analysis" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A White, Merete EggesbÃ¸, Rob Knight, and Shyamal D Peddada. 2015. <span>â€œAnalysis of Composition of Microbiomes: A Novel Method for Studying Microbial Composition.â€</span> <em>Microbial Ecology in Health and Disease</em> 26 (1): 27663.
</div>
<div id="ref-martin2020modeling" class="csl-entry">
Martin, Bryan D, Daniela Witten, and Amy D Willis. 2020. <span>â€œModeling Microbial Abundances and Dysbiosis with Beta-Binomial Regression.â€</span> <em>The Annals of Applied Statistics</em> 14 (1): 94.
</div>
<div id="ref-nearing2022microbiome" class="csl-entry">
Nearing, Jacob T, Gavin M Douglas, Molly G Hayes, Jocelyn MacDonald, Dhwani K Desai, Nicole Allward, Casey Jones, et al. 2022. <span>â€œMicrobiome Differential Abundance Methods Produce Different Results Across 38 Datasets.â€</span> <em>Nature Communications</em> 13 (1): 1â€“16.
</div>
<div id="ref-paulson2013differential" class="csl-entry">
Paulson, Joseph N, O Colin Stine, HÃ©ctor Corrada Bravo, and Mihai Pop. 2013. <span>â€œDifferential Abundance Analysis for Microbial Marker-Gene Surveys.â€</span> <em>Nature Methods</em> 10 (12): 1200â€“1202.
</div>
<div id="ref-robinson2010scaling" class="csl-entry">
Robinson, Mark D, and Alicia Oshlack. 2010. <span>â€œA Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data.â€</span> <em>Genome Biology</em> 11 (3): 1â€“9.
</div>
<div id="ref-segata2011metagenomic" class="csl-entry">
Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. <span>â€œMetagenomic Biomarker Discovery and Explanation.â€</span> <em>Genome Biology</em> 12 (6): 1â€“18.
</div>
<div id="ref-sohn2015robust" class="csl-entry">
Sohn, Michael B, Ruofei Du, and Lingling An. 2015. <span>â€œA Robust Approach for Identifying Differentially Abundant Features in Metagenomic Samples.â€</span> <em>Bioinformatics</em> 31 (14): 2269â€“75.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="association-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="examples.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/18-Method-Principal.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
