<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Quality Evaluation | Data analysis based on XMAS 2.0 package in microbiota data</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Quality Evaluation | Data analysis based on XMAS 2.0 package in microbiota data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Quality Evaluation | Data analysis based on XMAS 2.0 package in microbiota data" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-05-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="convert-inputdata-into-phyloseq-object.html"/>
<link rel="next" href="pre-processing.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/viz-1.8.2/viz.js"></script>
<link href="libs/DiagrammeR-styles-0.2/styles.css" rel="stylesheet" />
<script src="libs/grViz-binding-1.0.9/grViz.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology<span></span></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing<span></span></a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources<span></span></a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2<span></span></a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package<span></span></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description<span></span></a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages<span></span></a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation<span></span></a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette<span></span></a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors<span></span></a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Reads’ track<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment<span></span></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data<span></span></a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.3</b> Rarefaction Curves<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.4</b> Summarize phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#removing-samples-of-specific-group-in-phyloseq-class-object"><i class="fa fa-check"></i><b>6.8</b> Removing samples of specific group in phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="6.9" data-path="pre-processing.html"><a href="pre-processing.html#checking-samples-name-whether-start-with-character"><i class="fa fa-check"></i><b>6.9</b> Checking samples’ name whether start with character<span></span></a></li>
<li class="chapter" data-level="6.10" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.10</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.1</b> Phylogenetic beta-diversity metrics<span></span></a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.1.1</b> Unweighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.1.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.1.2</b> Weighted Unifrac<span></span></a></li>
<li class="chapter" data-level="8.1.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.1.3</b> Generalized UniFrac Distance Metrics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.2</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.3</b> Checking the homogeneity condition<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.4</b> Permutational Multivariate Analysis of Variance (PERMANOVA)<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.5</b> Analysis of Similarity (ANOSIM)<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.6</b> Ordination<span></span></a>
<ul>
<li class="chapter" data-level="8.6.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.6.1</b> Principal Component Analysis (PCA)<span></span></a></li>
<li class="chapter" data-level="8.6.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.6.2</b> Principal Coordinate Analysis (PCoA)<span></span></a></li>
<li class="chapter" data-level="8.6.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.6.3</b> t-distributed stochastic neighbor embedding (t-SNE)<span></span></a></li>
<li class="chapter" data-level="8.6.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.6.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.7</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="composition.html"><a href="composition.html"><i class="fa fa-check"></i><b>9</b> Composition<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="composition.html"><a href="composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="composition.html"><a href="composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.2</b> Stacked barplot<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="composition.html"><a href="composition.html#heatmaps"><i class="fa fa-check"></i><b>9.3</b> Heatmaps<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="composition.html"><a href="composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential analysis<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)<span></span></a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2<span></span></a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb<span></span></a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus<span></span></a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA<span></span></a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq<span></span></a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2<span></span></a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR<span></span></a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM<span></span></a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob<span></span></a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)<span></span></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom-1"><i class="fa fa-check"></i><b>11.2.1</b> limma_voom<span></span></a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.2</b> Wilcoxon Rank Sum and Signed Rank Tests<span></span></a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.3</b> Liner discriminant analysis (LDA) effect size (LEfSe)<span></span></a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.4</b> t-test<span></span></a></li>
<li class="chapter" data-level="11.2.5" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.5</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization<span></span></a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot<span></span></a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.5</b> Systematic Information<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>12</b> Examples<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>12.1</b> Workflow description<span></span></a></li>
<li class="chapter" data-level="12.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>12.2</b> Amplicon sequencing (16s)<span></span></a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>12.2.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="12.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>12.2.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="12.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>12.2.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="12.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>12.2.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="12.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>12.2.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="12.2.6" data-path="examples.html"><a href="examples.html#microbial-composition"><i class="fa fa-check"></i><b>12.2.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="12.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>12.2.7</b> Differential Analysis<span></span></a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="examples.html"><a href="examples.html#metagenomics"><i class="fa fa-check"></i><b>12.3</b> Metagenomics<span></span></a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>12.3.1</b> Converting inputs into phyloseq-class object<span></span></a></li>
<li class="chapter" data-level="12.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>12.3.2</b> Quality Control<span></span></a></li>
<li class="chapter" data-level="12.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>12.3.3</b> Data processing<span></span></a></li>
<li class="chapter" data-level="12.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>12.3.4</b> Diversity analysis<span></span></a></li>
<li class="chapter" data-level="12.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>12.3.5</b> PERMANOVA + Ordination<span></span></a></li>
<li class="chapter" data-level="12.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>12.3.6</b> Microbial composition<span></span></a></li>
<li class="chapter" data-level="12.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>12.3.7</b> Differential Analysis<span></span></a></li>
</ul></li>
</ul></li>
<li><a href="references.html#references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data analysis based on XMAS 2.0 package in microbiota data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-evaluation" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Quality Evaluation<a href="quality-evaluation.html#quality-evaluation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Quality control of DADA2 results will help us have more rational determinations on the further data analysis. Firstly, the reads’ track of DADA2 could show us the Changed Ratio of reads through the in-house standard amplicon sequencing data upstream pipeline. Then, the Evaluation of the spike-in samples from the Reference Matrix will reflect the quality of sequencing data. Finally, we recommend you that utilizing the rarefaction curves to assess the sequence depth per sample and choose the rational cutoff of OTU Number to do rarefy.</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="quality-evaluation.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS)</span>
<span id="cb21-2"><a href="quality-evaluation.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb21-3"><a href="quality-evaluation.html#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb21-4"><a href="quality-evaluation.html#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb21-5"><a href="quality-evaluation.html#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<div id="reads-track" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Reads’ track<a href="quality-evaluation.html#reads-track" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>this procedure only perform in 16s data</li>
</ul>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="quality-evaluation.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Dada2Track</span>(</span>
<span id="cb22-2"><a href="quality-evaluation.html#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data=</span>dada2_res<span class="sc">$</span>reads_track,</span>
<span id="cb22-3"><a href="quality-evaluation.html#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mode=</span><span class="st">&quot;PE&quot;</span>,</span>
<span id="cb22-4"><a href="quality-evaluation.html#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">relative_abundance=</span><span class="cn">TRUE</span>,</span>
<span id="cb22-5"><a href="quality-evaluation.html#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend_position=</span><span class="st">&quot;top&quot;</span></span>
<span id="cb22-6"><a href="quality-evaluation.html#cb22-6" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-7-1.png" alt="Reads' track of DADA2" width="672" />
<p class="caption">
Figure 5.1: Reads’ track of DADA2
</p>
</div>
</div>
<div id="spike-in-brs-sample-assessment" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Spike-in (BRS) sample assessment<a href="quality-evaluation.html#spike-in-brs-sample-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The spike-in sample is use to evaluate the consistent quality on bacteria when we have multiple sequence batches data. We devised an evaluation system containing the <em>Correlation Coefficient</em>, <em>Bray Curtis Distance</em> and <em>Impurity Level</em> to assess the sequencing data quality.</p>
<p><strong>Please use the default Reference and Save directory to obtain and save the spike-in sample matrix when you run <code>run_RefCheck</code>.</strong></p>
<div id="s-data" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> 16s data<a href="quality-evaluation.html#s-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The taxonomic levels of spike-in sample’s bacteria is genus. Firstly, using the <code>summarize_taxa</code> to get the genus level phyloseq object and then do <code>run_RefCheck</code> under the optimal parameters.</p>
<ul>
<li><p><em>group</em>: the column contains the spike-in sample;</p></li>
<li><p><em>group_names</em>: the group names of the spike-in sample;</p></li>
<li><p><em>Reference</em>: the directory of the latest spike-in sample matrix (default: /share/projects/Analytics/analytics/XMAS/RefCheck/);</p></li>
<li><p><em>Save</em>: the directory to save the latest spike-in sample matrix (default: /share/projects/Analytics/analytics/XMAS/RefCheck/).</p></li>
</ul>
<p>To see more details to use <code>?run_RefCheck</code>.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="quality-evaluation.html#cb23-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(dada2_ps, <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb23-2"><a href="quality-evaluation.html#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="quality-evaluation.html#cb23-3" aria-hidden="true" tabindex="-1"></a>RefCheck_16s <span class="ot">&lt;-</span> <span class="fu">run_RefCheck</span>(</span>
<span id="cb23-4"><a href="quality-evaluation.html#cb23-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps=</span>dada2_ps_genus,</span>
<span id="cb23-5"><a href="quality-evaluation.html#cb23-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">group=</span><span class="st">&quot;Group&quot;</span>,</span>
<span id="cb23-6"><a href="quality-evaluation.html#cb23-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names=</span><span class="st">&quot;QC&quot;</span>,</span>
<span id="cb23-7"><a href="quality-evaluation.html#cb23-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">Reference=</span><span class="cn">NULL</span>,</span>
<span id="cb23-8"><a href="quality-evaluation.html#cb23-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">Ref_type=</span><span class="st">&quot;16s&quot;</span>,</span>
<span id="cb23-9"><a href="quality-evaluation.html#cb23-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">Save=</span><span class="cn">NULL</span></span>
<span id="cb23-10"><a href="quality-evaluation.html#cb23-10" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb23-11"><a href="quality-evaluation.html#cb23-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Noting: the Reference Matrix is for 16s</span></span>
<span id="cb23-12"><a href="quality-evaluation.html#cb23-12" aria-hidden="true" tabindex="-1"></a><span class="do">## S8005 is in the Reference Matrix&#39;s samples and remove it to run</span></span>
<span id="cb23-13"><a href="quality-evaluation.html#cb23-13" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb23-14"><a href="quality-evaluation.html#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="do">## #########Matched baterica of the Spike-in sample###########</span></span>
<span id="cb23-15"><a href="quality-evaluation.html#cb23-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking out all the taxa of the Reference Matrix are in the sequencing dataset </span></span>
<span id="cb23-16"><a href="quality-evaluation.html#cb23-16" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Bifidobacterium</span></span>
<span id="cb23-17"><a href="quality-evaluation.html#cb23-17" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Bacteroides</span></span>
<span id="cb23-18"><a href="quality-evaluation.html#cb23-18" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Faecalibacterium</span></span>
<span id="cb23-19"><a href="quality-evaluation.html#cb23-19" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Lactobacillus</span></span>
<span id="cb23-20"><a href="quality-evaluation.html#cb23-20" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Parabacteroides</span></span>
<span id="cb23-21"><a href="quality-evaluation.html#cb23-21" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Collinsella</span></span>
<span id="cb23-22"><a href="quality-evaluation.html#cb23-22" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Coprococcus_3</span></span>
<span id="cb23-23"><a href="quality-evaluation.html#cb23-23" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Dorea</span></span>
<span id="cb23-24"><a href="quality-evaluation.html#cb23-24" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Streptococcus</span></span>
<span id="cb23-25"><a href="quality-evaluation.html#cb23-25" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Roseburia</span></span>
<span id="cb23-26"><a href="quality-evaluation.html#cb23-26" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Anaerostipes</span></span>
<span id="cb23-27"><a href="quality-evaluation.html#cb23-27" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Escherichia/Shigella</span></span>
<span id="cb23-28"><a href="quality-evaluation.html#cb23-28" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Enterococcus</span></span>
<span id="cb23-29"><a href="quality-evaluation.html#cb23-29" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Prevotella_9</span></span>
<span id="cb23-30"><a href="quality-evaluation.html#cb23-30" aria-hidden="true" tabindex="-1"></a><span class="do">## g__Eggerthella</span></span>
<span id="cb23-31"><a href="quality-evaluation.html#cb23-31" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb23-32"><a href="quality-evaluation.html#cb23-32" aria-hidden="true" tabindex="-1"></a><span class="do">## The number of the additional Taxa in the sequencing dataset is 1</span></span>
<span id="cb23-33"><a href="quality-evaluation.html#cb23-33" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb23-34"><a href="quality-evaluation.html#cb23-34" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb23-35"><a href="quality-evaluation.html#cb23-35" aria-hidden="true" tabindex="-1"></a><span class="do">## ##############Value of the Reference sample###############</span></span>
<span id="cb23-36"><a href="quality-evaluation.html#cb23-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Correlation Coefficient of the Reference sample is: 0.9714</span></span>
<span id="cb23-37"><a href="quality-evaluation.html#cb23-37" aria-hidden="true" tabindex="-1"></a><span class="do">## Bray Curtis of the Reference sample is: 0.07607</span></span>
<span id="cb23-38"><a href="quality-evaluation.html#cb23-38" aria-hidden="true" tabindex="-1"></a><span class="do">## Impurity of the Reference sample is: 0.06409</span></span>
<span id="cb23-39"><a href="quality-evaluation.html#cb23-39" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span>
<span id="cb23-40"><a href="quality-evaluation.html#cb23-40" aria-hidden="true" tabindex="-1"></a><span class="do">## #####Final Evaluation Results of the Spike-in sample#######</span></span>
<span id="cb23-41"><a href="quality-evaluation.html#cb23-41" aria-hidden="true" tabindex="-1"></a><span class="do">## The spike-in sample of sequencing dataset passed the cutoff of the Reference Matrix </span></span>
<span id="cb23-42"><a href="quality-evaluation.html#cb23-42" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of Coefficient is 0.8946</span></span>
<span id="cb23-43"><a href="quality-evaluation.html#cb23-43" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of BrayCurtis is 0.3878</span></span>
<span id="cb23-44"><a href="quality-evaluation.html#cb23-44" aria-hidden="true" tabindex="-1"></a><span class="do">## Cutoff of Impurity is 0.1565</span></span>
<span id="cb23-45"><a href="quality-evaluation.html#cb23-45" aria-hidden="true" tabindex="-1"></a><span class="do">## ###########################################################</span></span></code></pre></div>
<p>We could see that the messages are comprised of four parts.</p>
<ol style="list-style-type: decimal">
<li><p>the 1st part showed the type of <em>reference matrix</em> and whether the spike-in sample had been added to reference matrix;</p></li>
<li><p>the 2nd part revealed that <em>what</em> and <em>how many</em> the <em>matched bacterica</em> of the spike-in sample are;</p></li>
<li><p>the 3nd part showed that the <em>value</em> of evaluation system in the spike-in sample;</p></li>
<li><p>the 4nd part showed that whether the spike-in sample <em>passes</em> the cutoff of evaluation system.</p></li>
</ol>
</div>
<div id="metagenomic-data" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Metagenomic data<a href="quality-evaluation.html#metagenomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The taxonomic levels of spike-in sample’s bacteria is species. Firstly, using the <code>summarize_taxa</code> to get the genus level phyloseq object and then do <code>run_RefCheck</code> under the optimal parameters.</p>
<details>
<summary>
<strong>Quality Control by spike-in sample in metagenomic</strong>
</summary>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="quality-evaluation.html#cb24-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(metaphlan2_ps, <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb24-2"><a href="quality-evaluation.html#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="quality-evaluation.html#cb24-3" aria-hidden="true" tabindex="-1"></a>RefCheck_MGS <span class="ot">&lt;-</span> <span class="fu">run_RefCheck</span>(</span>
<span id="cb24-4"><a href="quality-evaluation.html#cb24-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">ps=</span>metaphlan2_ps_species,</span>
<span id="cb24-5"><a href="quality-evaluation.html#cb24-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">group=</span><span class="st">&quot;Group&quot;</span>,</span>
<span id="cb24-6"><a href="quality-evaluation.html#cb24-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">group_names=</span><span class="st">&quot;QC&quot;</span>,</span>
<span id="cb24-7"><a href="quality-evaluation.html#cb24-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">Reference=</span><span class="cn">NULL</span>,</span>
<span id="cb24-8"><a href="quality-evaluation.html#cb24-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">Ref_type=</span><span class="st">&quot;MGS&quot;</span>,</span>
<span id="cb24-9"><a href="quality-evaluation.html#cb24-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">Save=</span><span class="cn">NULL</span></span>
<span id="cb24-10"><a href="quality-evaluation.html#cb24-10" aria-hidden="true" tabindex="-1"></a>    )</span></code></pre></div>
</details>
</div>
</div>
<div id="rarefaction-curves" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Rarefaction Curves<a href="quality-evaluation.html#rarefaction-curves" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Rarefaction curves are often used when calculating alpha diversity indices, because increasing numbers of sequenced taxa allow increasingly accurate estimates of total population diversity. Rarefaction curves can therefore be used to estimate the full sample richness, as compared to the observed sample richness.</p>
</blockquote>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="quality-evaluation.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_RarefCure</span>(dada2_ps,</span>
<span id="cb25-2"><a href="quality-evaluation.html#cb25-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">taxa_level =</span> <span class="st">&quot;OTU&quot;</span>,</span>
<span id="cb25-3"><a href="quality-evaluation.html#cb25-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">step =</span> <span class="dv">100</span>,</span>
<span id="cb25-4"><a href="quality-evaluation.html#cb25-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">label =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb25-5"><a href="quality-evaluation.html#cb25-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb25-6"><a href="quality-evaluation.html#cb25-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">parallel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb25-7"><a href="quality-evaluation.html#cb25-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">se =</span> <span class="cn">TRUE</span></span>
<span id="cb25-8"><a href="quality-evaluation.html#cb25-8" aria-hidden="true" tabindex="-1"></a>               )</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-10"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-10-1.png" alt="Rarefaction Curves" width="672" />
<p class="caption">
Figure 5.2: Rarefaction Curves
</p>
</div>
<p>The result showed that all the samples had different sequencing depth but had the full sample richness.</p>
</div>
<div id="summarize-phyloseq-class-object" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Summarize phyloseq-class object<a href="quality-evaluation.html#summarize-phyloseq-class-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Summarizing the phyloseq-class object by using <code>summarize_phyloseq</code>. It displayed that briefly introduction of the object.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="quality-evaluation.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(dada2_ps)</span>
<span id="cb26-2"><a href="quality-evaluation.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Compositional = NO2</span></span>
<span id="cb26-3"><a href="quality-evaluation.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 1] Min. number of reads = 511812] Max. number of reads = 936223] Total number of reads = 15025374] Average number of reads = 62605.70833333335] Median number of reads = 619157] Sparsity = 0.8653738839285716] Any OTU sum to 1 or less? NO8] Number of singletons = 09] Percent of OTUs that are singletons</span></span>
<span id="cb26-4"><a href="quality-evaluation.html#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="do">##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 1Group2</span></span>
<span id="cb26-5"><a href="quality-evaluation.html#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="do">## [[1]]</span></span>
<span id="cb26-6"><a href="quality-evaluation.html#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;1] Min. number of reads = 51181&quot;</span></span>
<span id="cb26-7"><a href="quality-evaluation.html#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-8"><a href="quality-evaluation.html#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="do">## [[2]]</span></span>
<span id="cb26-9"><a href="quality-evaluation.html#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;2] Max. number of reads = 93622&quot;</span></span>
<span id="cb26-10"><a href="quality-evaluation.html#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-11"><a href="quality-evaluation.html#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="do">## [[3]]</span></span>
<span id="cb26-12"><a href="quality-evaluation.html#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;3] Total number of reads = 1502537&quot;</span></span>
<span id="cb26-13"><a href="quality-evaluation.html#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-14"><a href="quality-evaluation.html#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="do">## [[4]]</span></span>
<span id="cb26-15"><a href="quality-evaluation.html#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;4] Average number of reads = 62605.7083333333&quot;</span></span>
<span id="cb26-16"><a href="quality-evaluation.html#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-17"><a href="quality-evaluation.html#cb26-17" aria-hidden="true" tabindex="-1"></a><span class="do">## [[5]]</span></span>
<span id="cb26-18"><a href="quality-evaluation.html#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;5] Median number of reads = 61915&quot;</span></span>
<span id="cb26-19"><a href="quality-evaluation.html#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-20"><a href="quality-evaluation.html#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="do">## [[6]]</span></span>
<span id="cb26-21"><a href="quality-evaluation.html#cb26-21" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;7] Sparsity = 0.865373883928571&quot;</span></span>
<span id="cb26-22"><a href="quality-evaluation.html#cb26-22" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-23"><a href="quality-evaluation.html#cb26-23" aria-hidden="true" tabindex="-1"></a><span class="do">## [[7]]</span></span>
<span id="cb26-24"><a href="quality-evaluation.html#cb26-24" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;</span></span>
<span id="cb26-25"><a href="quality-evaluation.html#cb26-25" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-26"><a href="quality-evaluation.html#cb26-26" aria-hidden="true" tabindex="-1"></a><span class="do">## [[8]]</span></span>
<span id="cb26-27"><a href="quality-evaluation.html#cb26-27" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;8] Number of singletons = 0&quot;</span></span>
<span id="cb26-28"><a href="quality-evaluation.html#cb26-28" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-29"><a href="quality-evaluation.html#cb26-29" aria-hidden="true" tabindex="-1"></a><span class="do">## [[9]]</span></span>
<span id="cb26-30"><a href="quality-evaluation.html#cb26-30" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;</span></span>
<span id="cb26-31"><a href="quality-evaluation.html#cb26-31" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-32"><a href="quality-evaluation.html#cb26-32" aria-hidden="true" tabindex="-1"></a><span class="do">## [[10]]</span></span>
<span id="cb26-33"><a href="quality-evaluation.html#cb26-33" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;10] Number of sample variables are: 1&quot;</span></span>
<span id="cb26-34"><a href="quality-evaluation.html#cb26-34" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb26-35"><a href="quality-evaluation.html#cb26-35" aria-hidden="true" tabindex="-1"></a><span class="do">## [[11]]</span></span>
<span id="cb26-36"><a href="quality-evaluation.html#cb26-36" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;Group&quot;</span></span></code></pre></div>
<p>The minus account of the OTU counts is <em>51181</em> in the phyloseq object, and we can use it as the threshold to rarefy.</p>
<p>Notice the <strong>Sparsity (0.865)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.</p>
</div>
<div id="systematic-information-2" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Systematic Information<a href="quality-evaluation.html#systematic-information-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="quality-evaluation.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.2 (2021-11-01)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] C/UTF-8/C/C/C/C
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] picante_1.8.2   nlme_3.1-155    vegan_2.5-7     lattice_0.20-45
##  [5] permute_0.9-7   ape_5.6-2       ggpubr_0.4.0    ggplot2_3.3.5  
##  [9] phyloseq_1.38.0 tibble_3.1.6    dplyr_1.0.8     XMAS_2.0.12    
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.4.1             Hmisc_4.6-0                
##   [3] plyr_1.8.6                  igraph_1.2.11              
##   [5] splines_4.1.2               BiocParallel_1.28.3        
##   [7] GenomeInfoDb_1.30.1         TH.data_1.1-0              
##   [9] lpsymphony_1.22.0           digest_0.6.29              
##  [11] foreach_1.5.2               htmltools_0.5.2            
##  [13] fansi_1.0.2                 checkmate_2.0.0            
##  [15] magrittr_2.0.2              memoise_2.0.1              
##  [17] cluster_2.1.2               limma_3.50.1               
##  [19] Biostrings_2.62.0           annotate_1.72.0            
##  [21] matrixStats_0.61.0          sandwich_3.0-1             
##  [23] metagenomeSeq_1.36.0        jpeg_0.1-9                 
##  [25] colorspace_2.0-3            ggrepel_0.9.1              
##  [27] blob_1.2.2                  corncob_0.2.0              
##  [29] xfun_0.30                   crayon_1.5.0               
##  [31] RCurl_1.98-1.6              jsonlite_1.8.0             
##  [33] libcoin_1.0-9               genefilter_1.76.0          
##  [35] biglm_0.9-2.1               survival_3.3-1             
##  [37] zoo_1.8-9                   iterators_1.0.14           
##  [39] glue_1.6.2                  gtable_0.3.0               
##  [41] zlibbioc_1.40.0             XVector_0.34.0             
##  [43] DelayedArray_0.20.0         car_3.0-12                 
##  [45] Rhdf5lib_1.16.0             shape_1.4.6                
##  [47] BiocGenerics_0.40.0         DEoptimR_1.0-10            
##  [49] abind_1.4-5                 scales_1.1.1               
##  [51] mvtnorm_1.1-3               DBI_1.1.2                  
##  [53] edgeR_3.36.0                rstatix_0.7.0              
##  [55] Rcpp_1.0.8.2                htmlTable_2.4.0            
##  [57] xtable_1.8-4                foreign_0.8-82             
##  [59] bit_4.0.4                   Formula_1.2-4              
##  [61] stats4_4.1.2                glmnet_4.1-3               
##  [63] htmlwidgets_1.5.4           httr_1.4.2                 
##  [65] getopt_1.20.3               gplots_3.1.1               
##  [67] RColorBrewer_1.1-2          modeltools_0.2-23          
##  [69] ellipsis_0.3.2              farver_2.1.0               
##  [71] pkgconfig_2.0.3             XML_3.99-0.9               
##  [73] nnet_7.3-17                 sass_0.4.0                 
##  [75] locfit_1.5-9.5              utf8_1.2.2                 
##  [77] labeling_0.4.2              tidyselect_1.1.2           
##  [79] rlang_1.0.2                 reshape2_1.4.4             
##  [81] AnnotationDbi_1.56.2        munsell_0.5.0              
##  [83] tools_4.1.2                 cachem_1.0.6               
##  [85] cli_3.2.0                   generics_0.1.2             
##  [87] RSQLite_2.2.10              ade4_1.7-18                
##  [89] broom_0.7.12                evaluate_0.15              
##  [91] biomformat_1.22.0           stringr_1.4.0              
##  [93] fastmap_1.1.0               yaml_2.3.5                 
##  [95] knitr_1.37                  bit64_4.0.5                
##  [97] robustbase_0.93-9           caTools_1.18.2             
##  [99] purrr_0.3.4                 KEGGREST_1.34.0            
## [101] coin_1.4-2                  protoclust_1.6.3           
## [103] RAIDA_1.0                   compiler_4.1.2             
## [105] rstudioapi_0.13             png_0.1-7                  
## [107] ggsignif_0.6.3              geneplotter_1.72.0         
## [109] bslib_0.3.1                 pcaPP_1.9-74               
## [111] stringi_1.7.6               highr_0.9                  
## [113] Maaslin2_1.8.0              Matrix_1.4-0               
## [115] multtest_2.50.0             vctrs_0.3.8                
## [117] pillar_1.7.0                lifecycle_1.0.1            
## [119] rhdf5filters_1.6.0          optparse_1.7.1             
## [121] jquerylib_0.1.4             data.table_1.14.2          
## [123] cowplot_1.1.1               bitops_1.0-7               
## [125] qvalue_2.26.0               GenomicRanges_1.46.1       
## [127] latticeExtra_0.6-29         R6_2.5.1                   
## [129] bookdown_0.24               gridExtra_2.3              
## [131] KernSmooth_2.23-20          IRanges_2.28.0             
## [133] codetools_0.2-18            gtools_3.9.2               
## [135] MASS_7.3-55                 assertthat_0.2.1           
## [137] Wrench_1.12.0               rhdf5_2.38.1               
## [139] SummarizedExperiment_1.24.0 DESeq2_1.34.0              
## [141] withr_2.5.0                 multcomp_1.4-18            
## [143] S4Vectors_0.32.3            GenomeInfoDbData_1.2.7     
## [145] mgcv_1.8-39                 parallel_4.1.2             
## [147] rpart_4.1.16                grid_4.1.2                 
## [149] tidyr_1.2.0                 rmarkdown_2.13             
## [151] MatrixGenerics_1.6.0        carData_3.0-5              
## [153] base64enc_0.1-3             Biobase_2.54.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="convert-inputdata-into-phyloseq-object.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pre-processing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/04-quality-control.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
