<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Quality Evaluation | XMAS 2.0 Tutorial</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Quality Evaluation | XMAS 2.0 Tutorial" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Quality Evaluation | XMAS 2.0 Tutorial" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-08-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="convert-inputdata-into-phyloseq-object.html"/>
<link rel="next" href="pre-processing.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/bsTable-3.3.7/bootstrapTable.min.css" rel="stylesheet" />
<script src="libs/bsTable-3.3.7/bootstrapTable.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.21/datatables.js"></script>
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data analysis based on XMAS 2.0 package</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#preparing-for-the-tutorial"><i class="fa fa-check"></i><b>1.1</b> Preparing for the tutorial</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="introduction.html"><a href="introduction.html#microbiome"><i class="fa fa-check"></i><b>2.1</b> Microbiome</a></li>
<li class="chapter" data-level="2.2" data-path="introduction.html"><a href="introduction.html#sequencing-technology"><i class="fa fa-check"></i><b>2.2</b> Sequencing technology</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="introduction.html"><a href="introduction.html#amplicon-sequencing"><i class="fa fa-check"></i><b>2.2.1</b> Amplicon sequencing</a></li>
<li class="chapter" data-level="2.2.2" data-path="introduction.html"><a href="introduction.html#whole-genome-shotgun-sequencing"><i class="fa fa-check"></i><b>2.2.2</b> Whole genome shotgun sequencing</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="introduction.html"><a href="introduction.html#the-gut-microbiome-relationships-with-disease-and-opportunities-for-therapy"><i class="fa fa-check"></i><b>2.3</b> The gut microbiome: Relationships with disease and opportunities for therapy</a></li>
<li class="chapter" data-level="2.4" data-path="introduction.html"><a href="introduction.html#data-analysis-apply-for-discovering-the-potential-microbial-biomarkers"><i class="fa fa-check"></i><b>2.4</b> Data Analysis apply for discovering the potential microbial biomarkers</a></li>
<li class="chapter" data-level="2.5" data-path="introduction.html"><a href="introduction.html#book"><i class="fa fa-check"></i><b>2.5</b> Book</a></li>
<li class="chapter" data-level="2.6" data-path="introduction.html"><a href="introduction.html#input-data"><i class="fa fa-check"></i><b>2.6</b> Input data</a></li>
<li class="chapter" data-level="2.7" data-path="introduction.html"><a href="introduction.html#key-publications"><i class="fa fa-check"></i><b>2.7</b> Key publications</a></li>
<li class="chapter" data-level="2.8" data-path="introduction.html"><a href="introduction.html#citation"><i class="fa fa-check"></i><b>2.8</b> Citation</a></li>
<li class="chapter" data-level="2.9" data-path="introduction.html"><a href="introduction.html#change-log"><i class="fa fa-check"></i><b>2.9</b> Change log</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="introduction.html"><a href="introduction.html#xmas-2.1.7"><i class="fa fa-check"></i><b>2.9.1</b> XMAS 2.1.7</a></li>
<li class="chapter" data-level="2.9.2" data-path="introduction.html"><a href="introduction.html#xmas-2.1.6"><i class="fa fa-check"></i><b>2.9.2</b> XMAS 2.1.6</a></li>
<li class="chapter" data-level="2.9.3" data-path="introduction.html"><a href="introduction.html#xmas-2.1.5"><i class="fa fa-check"></i><b>2.9.3</b> XMAS 2.1.5</a></li>
<li class="chapter" data-level="2.9.4" data-path="introduction.html"><a href="introduction.html#xmas-2.1.4"><i class="fa fa-check"></i><b>2.9.4</b> XMAS 2.1.4</a></li>
<li class="chapter" data-level="2.9.5" data-path="introduction.html"><a href="introduction.html#xmas-2.1.3"><i class="fa fa-check"></i><b>2.9.5</b> XMAS 2.1.3</a></li>
<li class="chapter" data-level="2.9.6" data-path="introduction.html"><a href="introduction.html#xmas-2.1.2"><i class="fa fa-check"></i><b>2.9.6</b> XMAS 2.1.2</a></li>
<li class="chapter" data-level="2.9.7" data-path="introduction.html"><a href="introduction.html#xmas-2.1.1"><i class="fa fa-check"></i><b>2.9.7</b> XMAS 2.1.1</a></li>
<li class="chapter" data-level="2.9.8" data-path="introduction.html"><a href="introduction.html#xmas-2.1.0"><i class="fa fa-check"></i><b>2.9.8</b> XMAS 2.1.0</a></li>
<li class="chapter" data-level="2.9.9" data-path="introduction.html"><a href="introduction.html#xmas-2.0.13"><i class="fa fa-check"></i><b>2.9.9</b> XMAS 2.0.13</a></li>
<li class="chapter" data-level="2.9.10" data-path="introduction.html"><a href="introduction.html#xmas-2.0.12"><i class="fa fa-check"></i><b>2.9.10</b> XMAS 2.0.12</a></li>
<li class="chapter" data-level="2.9.11" data-path="introduction.html"><a href="introduction.html#xmas-2.0.7-2022-04-12"><i class="fa fa-check"></i><b>2.9.11</b> XMAS 2.0.7 (2022-04-12)</a></li>
<li class="chapter" data-level="2.9.12" data-path="introduction.html"><a href="introduction.html#xmas-2.0.2-2022-03-17"><i class="fa fa-check"></i><b>2.9.12</b> XMAS 2.0.2 (2022-03-17)</a></li>
<li class="chapter" data-level="2.9.13" data-path="introduction.html"><a href="introduction.html#xmas-2.0.1-2022-03-14"><i class="fa fa-check"></i><b>2.9.13</b> XMAS 2.0.1 (2022-03-14)</a></li>
<li class="chapter" data-level="2.9.14" data-path="introduction.html"><a href="introduction.html#xmas-1.0.0-2021-12-13"><i class="fa fa-check"></i><b>2.9.14</b> XMAS 1.0.0 (2021-12-13)</a></li>
<li class="chapter" data-level="2.9.15" data-path="introduction.html"><a href="introduction.html#initial-repository-0.0.1-2021-07-25"><i class="fa fa-check"></i><b>2.9.15</b> Initial repository 0.0.1 (2021-07-25)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html"><i class="fa fa-check"></i><b>3</b> Material in this tutorial</a>
<ul>
<li class="chapter" data-level="3.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#resources"><i class="fa fa-check"></i><b>3.1</b> Resources</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#dada2"><i class="fa fa-check"></i><b>3.1.1</b> DADA2</a></li>
<li class="chapter" data-level="3.1.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#metaphlan2"><i class="fa fa-check"></i><b>3.1.2</b> Metaphlan2</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#overview-of-the-design-of-xmas-package"><i class="fa fa-check"></i><b>3.2</b> Overview of the design of <strong>XMAS</strong> package</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#description"><i class="fa fa-check"></i><b>3.2.1</b> Description</a></li>
<li class="chapter" data-level="3.2.2" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation-of-dependent-packages"><i class="fa fa-check"></i><b>3.2.2</b> Installation of dependent packages</a></li>
<li class="chapter" data-level="3.2.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#installation"><i class="fa fa-check"></i><b>3.2.3</b> Installation</a></li>
<li class="chapter" data-level="3.2.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#vignette"><i class="fa fa-check"></i><b>3.2.4</b> Vignette</a></li>
<li class="chapter" data-level="3.2.5" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#authors"><i class="fa fa-check"></i><b>3.2.5</b> Authors</a></li>
<li class="chapter" data-level="3.2.6" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#library-package"><i class="fa fa-check"></i><b>3.2.6</b> library package</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#future-plan"><i class="fa fa-check"></i><b>3.3</b> Future plan</a></li>
<li class="chapter" data-level="3.4" data-path="material-in-this-tutorial.html"><a href="material-in-this-tutorial.html#systematic-information"><i class="fa fa-check"></i><b>3.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>4</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="4.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>4.1</b> loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#dada2-1"><i class="fa fa-check"></i><b>4.2</b> DADA2</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results-from-dada2-pipeline"><i class="fa fa-check"></i><b>4.2.1</b> Importing results from dada2 pipeline</a></li>
<li class="chapter" data-level="4.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table"><i class="fa fa-check"></i><b>4.2.2</b> taxa table</a></li>
<li class="chapter" data-level="4.2.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table"><i class="fa fa-check"></i><b>4.2.3</b> otu table</a></li>
<li class="chapter" data-level="4.2.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table"><i class="fa fa-check"></i><b>4.2.4</b> metadata table</a></li>
<li class="chapter" data-level="4.2.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object"><i class="fa fa-check"></i><b>4.2.5</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metaphlan23"><i class="fa fa-check"></i><b>4.3</b> Metaphlan2/3</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-results"><i class="fa fa-check"></i><b>4.3.1</b> Importing results</a></li>
<li class="chapter" data-level="4.3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#taxa-table-1"><i class="fa fa-check"></i><b>4.3.2</b> taxa table</a></li>
<li class="chapter" data-level="4.3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#otu-table-1"><i class="fa fa-check"></i><b>4.3.3</b> otu table</a></li>
<li class="chapter" data-level="4.3.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#metadata-table-1"><i class="fa fa-check"></i><b>4.3.4</b> metadata table</a></li>
<li class="chapter" data-level="4.3.5" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#phyloseq-object-1"><i class="fa fa-check"></i><b>4.3.5</b> phyloseq object</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information-1"><i class="fa fa-check"></i><b>4.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>5</b> Quality Evaluation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>5.1</b> Reads’ track</a></li>
<li class="chapter" data-level="5.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>5.2</b> Spike-in (BRS) sample assessment</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>5.2.1</b> 16s data</a></li>
<li class="chapter" data-level="5.2.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>5.2.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#remove-brs"><i class="fa fa-check"></i><b>5.3</b> Remove BRS</a></li>
<li class="chapter" data-level="5.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>5.4</b> Rarefaction Curves</a></li>
<li class="chapter" data-level="5.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>5.5</b> Summarize phyloseq-class object</a></li>
<li class="chapter" data-level="5.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>5.6</b> Variability</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>5.6.1</b> Coefficient of variation</a></li>
<li class="chapter" data-level="5.6.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>5.6.2</b> Distribution of number of sequences</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-2"><i class="fa fa-check"></i><b>5.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>6</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="6.1" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>6.1</b> Transformation</a></li>
<li class="chapter" data-level="6.2" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>6.2</b> Normalization</a></li>
<li class="chapter" data-level="6.3" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering</a></li>
<li class="chapter" data-level="6.4" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>6.4</b> Trimming</a></li>
<li class="chapter" data-level="6.5" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>6.5</b> Imputation</a></li>
<li class="chapter" data-level="6.6" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>6.6</b> Extracting specific taxa phyloseq-class object</a></li>
<li class="chapter" data-level="6.7" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>6.7</b> Aggregating low relative abundance or unclassified taxa into others</a></li>
<li class="chapter" data-level="6.8" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>6.8</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)</a></li>
<li class="chapter" data-level="6.9" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-3"><i class="fa fa-check"></i><b>6.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>7</b> Aplha diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>7.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="7.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#diversities"><i class="fa fa-check"></i><b>7.2</b> Diversities</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>7.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="7.2.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>7.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>7.3</b> Rank Abundance</a></li>
<li class="chapter" data-level="7.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#evenness"><i class="fa fa-check"></i><b>7.4</b> Evenness</a></li>
<li class="chapter" data-level="7.5" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-data-and-removing-brs"><i class="fa fa-check"></i><b>8.1</b> Loading data and Removing BRS</a></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>8.2</b> Phylogenetic beta-diversity metrics</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>8.2.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>8.2.2</b> Weighted Unifrac</a></li>
<li class="chapter" data-level="8.2.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>8.2.3</b> Generalized UniFrac Distance Metrics</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>8.3</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>8.4</b> Checking the homogeneity condition</a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>8.5</b> Permutational Multivariate Analysis of Variance (PERMANOVA)</a></li>
<li class="chapter" data-level="8.6" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>8.6</b> Mantel Test</a></li>
<li class="chapter" data-level="8.7" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>8.7</b> Analysis of Similarity (ANOSIM)</a></li>
<li class="chapter" data-level="8.8" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>8.8</b> Multi-response permutation procedures (MRPP)</a></li>
<li class="chapter" data-level="8.9" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>8.9</b> Ordination</a>
<ul>
<li class="chapter" data-level="8.9.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.9.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="8.9.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>8.9.2</b> Principal Coordinate Analysis (PCoA)</a></li>
<li class="chapter" data-level="8.9.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>8.9.3</b> t-distributed stochastic neighbor embedding (t-SNE)</a></li>
<li class="chapter" data-level="8.9.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>8.9.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-5"><i class="fa fa-check"></i><b>8.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>9</b> Microbial Composition</a>
<ul>
<li class="chapter" data-level="9.1" data-path="microbial-composition.html"><a href="microbial-composition.html#loading-precessing-phyloseq-object"><i class="fa fa-check"></i><b>9.1</b> Loading &amp; precessing phyloseq object</a></li>
<li class="chapter" data-level="9.2" data-path="microbial-composition.html"><a href="microbial-composition.html#xviz-plot_stacked_bar"><i class="fa fa-check"></i><b>9.2</b> XVIZ <code>plot_stacked_bar</code></a></li>
<li class="chapter" data-level="9.3" data-path="microbial-composition.html"><a href="microbial-composition.html#stacked-barplot"><i class="fa fa-check"></i><b>9.3</b> Stacked barplot</a></li>
<li class="chapter" data-level="9.4" data-path="microbial-composition.html"><a href="microbial-composition.html#heatmaps"><i class="fa fa-check"></i><b>9.4</b> Heatmaps</a></li>
<li class="chapter" data-level="9.5" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-6"><i class="fa fa-check"></i><b>9.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>10</b> Core microbiota</a>
<ul>
<li class="chapter" data-level="10.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>10.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="10.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>10.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="10.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>10.3</b> Core visualization</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>10.3.1</b> heatmap</a></li>
<li class="chapter" data-level="10.3.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>10.3.2</b> linechart</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-7"><i class="fa fa-check"></i><b>10.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>11</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>11.1</b> Amplicon sequencing microbial data (16s)</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>11.1.1</b> ALDEx2</a></li>
<li class="chapter" data-level="11.1.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>11.1.2</b> limma_voom</a></li>
<li class="chapter" data-level="11.1.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>11.1.3</b> mbzinb</a></li>
<li class="chapter" data-level="11.1.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>11.1.4</b> omnibus</a></li>
<li class="chapter" data-level="11.1.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>11.1.5</b> RAIDA</a></li>
<li class="chapter" data-level="11.1.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>11.1.6</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="11.1.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>11.1.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="11.1.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>11.1.8</b> t-test</a></li>
<li class="chapter" data-level="11.1.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>11.1.9</b> MetagenomeSeq</a></li>
<li class="chapter" data-level="11.1.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>11.1.10</b> DESeq2</a></li>
<li class="chapter" data-level="11.1.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>11.1.11</b> EdgeR</a></li>
<li class="chapter" data-level="11.1.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>11.1.12</b> ANCOM</a></li>
<li class="chapter" data-level="11.1.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>11.1.13</b> Corncob</a></li>
<li class="chapter" data-level="11.1.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>11.1.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
<li class="chapter" data-level="11.1.15" data-path="differential-analysis.html"><a href="differential-analysis.html#locom"><i class="fa fa-check"></i><b>11.1.15</b> LOCOM</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>11.2</b> Metagenomic sequencing microbial data (metaphlan2/3)</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>11.2.1</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="11.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>11.2.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="11.2.3" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>11.2.3</b> t-test</a></li>
<li class="chapter" data-level="11.2.4" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>11.2.4</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>11.3</b> Visualization</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>11.3.1</b> Volcano plot</a></li>
<li class="chapter" data-level="11.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>11.3.2</b> barplot in lefse</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>11.4</b> Dominant taxa</a></li>
<li class="chapter" data-level="11.5" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>11.5</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="11.6" data-path="differential-analysis.html"><a href="differential-analysis.html#comparing-outputs-from-xmas2-lefse-conda-and-lefse-galaxy-using-the-same-in-house-datasets-amplicon_pszeybel_gut"><i class="fa fa-check"></i><b>11.6</b> Comparing outputs from XMAS2, lefse-conda, and lefse-galaxy using the same in-house datasets (amplicon_ps|Zeybel_Gut)</a>
<ul>
<li class="chapter" data-level="11.6.1" data-path="differential-analysis.html"><a href="differential-analysis.html#introduction-1"><i class="fa fa-check"></i><b>11.6.1</b> Introduction</a></li>
<li class="chapter" data-level="11.6.2" data-path="differential-analysis.html"><a href="differential-analysis.html#dataset"><i class="fa fa-check"></i><b>11.6.2</b> Dataset</a></li>
<li class="chapter" data-level="11.6.3" data-path="differential-analysis.html"><a href="differential-analysis.html#run-lefse-independently-with-the-three-applications-r-conda-galaxy"><i class="fa fa-check"></i><b>11.6.3</b> Run lefse independently with the three applications (R, conda, galaxy)</a></li>
<li class="chapter" data-level="11.6.4" data-path="differential-analysis.html"><a href="differential-analysis.html#comparison-of-lefse-conda-with-xmas2"><i class="fa fa-check"></i><b>11.6.4</b> Comparison of lefse-conda with XMAS2</a></li>
</ul></li>
<li class="chapter" data-level="11.7" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-8"><i class="fa fa-check"></i><b>11.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>12</b> Visualization</a>
<ul>
<li class="chapter" data-level="12.1" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>12.1</b> plot_boxplot</a></li>
<li class="chapter" data-level="12.2" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>12.2</b> plot_barplot</a></li>
<li class="chapter" data-level="12.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>12.3</b> plot_dotplot</a></li>
<li class="chapter" data-level="12.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>12.4</b> plot_correlation_boxplot</a></li>
<li class="chapter" data-level="12.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>12.5</b> plot_correlation_density</a></li>
<li class="chapter" data-level="12.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>12.6</b> plot_Ordination</a></li>
<li class="chapter" data-level="12.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>12.7</b> plot_ggbiplot</a></li>
<li class="chapter" data-level="12.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>12.8</b> plot_corrplot</a></li>
<li class="chapter" data-level="12.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>12.9</b> plot_2DA_venn</a></li>
<li class="chapter" data-level="12.10" data-path="visualization-5.html"><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot"><i class="fa fa-check"></i><b>12.10</b> plot the DA results from the significant taxa by double barplot</a></li>
<li class="chapter" data-level="12.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>12.11</b> plot_stacked_bar_XIVZ</a></li>
<li class="chapter" data-level="12.12" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>12.12</b> plot_StackBarPlot</a></li>
<li class="chapter" data-level="12.13" data-path="visualization-5.html"><a href="visualization-5.html#color-palettes"><i class="fa fa-check"></i><b>12.13</b> Color Palettes</a>
<ul>
<li class="chapter" data-level="12.13.1" data-path="visualization-5.html"><a href="visualization-5.html#wes-anderson-palettes"><i class="fa fa-check"></i><b>12.13.1</b> Wes Anderson Palettes</a></li>
<li class="chapter" data-level="12.13.2" data-path="visualization-5.html"><a href="visualization-5.html#usemycol"><i class="fa fa-check"></i><b>12.13.2</b> useMyCol</a></li>
</ul></li>
<li class="chapter" data-level="12.14" data-path="visualization-5.html"><a href="visualization-5.html#ordination-plots-with-ggplot2"><i class="fa fa-check"></i><b>12.14</b> Ordination plots with ggplot2</a>
<ul>
<li class="chapter" data-level="12.14.1" data-path="visualization-5.html"><a href="visualization-5.html#principal-components-analysis-with-the-iris-data-set"><i class="fa fa-check"></i><b>12.14.1</b> principal components analysis with the iris data set</a></li>
<li class="chapter" data-level="12.14.2" data-path="visualization-5.html"><a href="visualization-5.html#multiple-correspondence-analysis-with-the-tea-dataset"><i class="fa fa-check"></i><b>12.14.2</b> multiple correspondence analysis with the tea dataset</a></li>
<li class="chapter" data-level="12.14.3" data-path="visualization-5.html"><a href="visualization-5.html#linear-discriminant-analysis"><i class="fa fa-check"></i><b>12.14.3</b> linear discriminant analysis</a></li>
</ul></li>
<li class="chapter" data-level="12.15" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-9"><i class="fa fa-check"></i><b>12.15</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>13</b> Association Analysis</a>
<ul>
<li class="chapter" data-level="13.1" data-path="association-analysis.html"><a href="association-analysis.html#loading-and-showing-phyloseq-object"><i class="fa fa-check"></i><b>13.1</b> Loading and showing phyloseq object</a></li>
<li class="chapter" data-level="13.2" data-path="association-analysis.html"><a href="association-analysis.html#spearman-correlation-analysis"><i class="fa fa-check"></i><b>13.2</b> Spearman Correlation Analysis</a></li>
<li class="chapter" data-level="13.3" data-path="association-analysis.html"><a href="association-analysis.html#partial-correlation-analysis"><i class="fa fa-check"></i><b>13.3</b> Partial Correlation Analysis</a></li>
<li class="chapter" data-level="13.4" data-path="association-analysis.html"><a href="association-analysis.html#systematic-information-10"><i class="fa fa-check"></i><b>13.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>14</b> Principals of Differential Analysis methods</a>
<ul>
<li class="chapter" data-level="14.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>14.1</b> ALDEx2</a>
<ul>
<li class="chapter" data-level="14.1.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-1"><i class="fa fa-check"></i><b>14.1.1</b> Vignette</a></li>
<li class="chapter" data-level="14.1.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type"><i class="fa fa-check"></i><b>14.1.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.1.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments"><i class="fa fa-check"></i><b>14.1.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.1.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures"><i class="fa fa-check"></i><b>14.1.4</b> Procedures</a></li>
<li class="chapter" data-level="14.1.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output"><i class="fa fa-check"></i><b>14.1.5</b> Key output</a></li>
<li class="chapter" data-level="14.1.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax"><i class="fa fa-check"></i><b>14.1.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.1.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax"><i class="fa fa-check"></i><b>14.1.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.1.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes"><i class="fa fa-check"></i><b>14.1.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>14.2</b> ANCOM-II</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-1"><i class="fa fa-check"></i><b>14.2.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.2.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-1"><i class="fa fa-check"></i><b>14.2.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.2.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-1"><i class="fa fa-check"></i><b>14.2.3</b> Procedures</a></li>
<li class="chapter" data-level="14.2.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-1"><i class="fa fa-check"></i><b>14.2.4</b> Key output</a></li>
<li class="chapter" data-level="14.2.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-1"><i class="fa fa-check"></i><b>14.2.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.2.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-1"><i class="fa fa-check"></i><b>14.2.6</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.2.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-1"><i class="fa fa-check"></i><b>14.2.7</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>14.3</b> corncob</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-2"><i class="fa fa-check"></i><b>14.3.1</b> Vignette</a></li>
<li class="chapter" data-level="14.3.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-2"><i class="fa fa-check"></i><b>14.3.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.3.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-2"><i class="fa fa-check"></i><b>14.3.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.3.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-2"><i class="fa fa-check"></i><b>14.3.4</b> Procedures</a></li>
<li class="chapter" data-level="14.3.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-2"><i class="fa fa-check"></i><b>14.3.5</b> Key output</a></li>
<li class="chapter" data-level="14.3.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-2"><i class="fa fa-check"></i><b>14.3.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.3.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-2"><i class="fa fa-check"></i><b>14.3.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.3.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-2"><i class="fa fa-check"></i><b>14.3.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>14.4</b> DESeq2</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-3"><i class="fa fa-check"></i><b>14.4.1</b> Vignette</a></li>
<li class="chapter" data-level="14.4.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-3"><i class="fa fa-check"></i><b>14.4.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.4.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-3"><i class="fa fa-check"></i><b>14.4.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.4.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-3"><i class="fa fa-check"></i><b>14.4.4</b> Procedures</a></li>
<li class="chapter" data-level="14.4.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-3"><i class="fa fa-check"></i><b>14.4.5</b> Key output</a></li>
<li class="chapter" data-level="14.4.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-3"><i class="fa fa-check"></i><b>14.4.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.4.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-3"><i class="fa fa-check"></i><b>14.4.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>14.5</b> edgeR</a>
<ul>
<li class="chapter" data-level="14.5.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-4"><i class="fa fa-check"></i><b>14.5.1</b> Vignette</a></li>
<li class="chapter" data-level="14.5.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-4"><i class="fa fa-check"></i><b>14.5.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.5.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-4"><i class="fa fa-check"></i><b>14.5.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.5.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-4"><i class="fa fa-check"></i><b>14.5.4</b> Procedures</a></li>
<li class="chapter" data-level="14.5.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-4"><i class="fa fa-check"></i><b>14.5.5</b> Key output</a></li>
<li class="chapter" data-level="14.5.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-4"><i class="fa fa-check"></i><b>14.5.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.5.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-4"><i class="fa fa-check"></i><b>14.5.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.5.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-3"><i class="fa fa-check"></i><b>14.5.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>14.6</b> LEfSe</a>
<ul>
<li class="chapter" data-level="14.6.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-5"><i class="fa fa-check"></i><b>14.6.1</b> Vignette</a></li>
<li class="chapter" data-level="14.6.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-5"><i class="fa fa-check"></i><b>14.6.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.6.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-5"><i class="fa fa-check"></i><b>14.6.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.6.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-5"><i class="fa fa-check"></i><b>14.6.4</b> Procedures</a></li>
<li class="chapter" data-level="14.6.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-5"><i class="fa fa-check"></i><b>14.6.5</b> Key output</a></li>
<li class="chapter" data-level="14.6.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-5"><i class="fa fa-check"></i><b>14.6.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.6.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-5"><i class="fa fa-check"></i><b>14.6.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.6.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-4"><i class="fa fa-check"></i><b>14.6.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>14.7</b> limma voom</a>
<ul>
<li class="chapter" data-level="14.7.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-6"><i class="fa fa-check"></i><b>14.7.1</b> Vignette</a></li>
<li class="chapter" data-level="14.7.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-6"><i class="fa fa-check"></i><b>14.7.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.7.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-6"><i class="fa fa-check"></i><b>14.7.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.7.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-6"><i class="fa fa-check"></i><b>14.7.4</b> Procedures</a></li>
<li class="chapter" data-level="14.7.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-6"><i class="fa fa-check"></i><b>14.7.5</b> Key output</a></li>
<li class="chapter" data-level="14.7.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-6"><i class="fa fa-check"></i><b>14.7.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.7.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-6"><i class="fa fa-check"></i><b>14.7.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="14.7.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-5"><i class="fa fa-check"></i><b>14.7.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>14.8</b> MaAsLin2</a>
<ul>
<li class="chapter" data-level="14.8.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-7"><i class="fa fa-check"></i><b>14.8.1</b> Vignette</a></li>
<li class="chapter" data-level="14.8.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-7"><i class="fa fa-check"></i><b>14.8.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.8.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-7"><i class="fa fa-check"></i><b>14.8.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.8.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-7"><i class="fa fa-check"></i><b>14.8.4</b> Procedures</a></li>
<li class="chapter" data-level="14.8.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-7"><i class="fa fa-check"></i><b>14.8.5</b> Key output</a></li>
<li class="chapter" data-level="14.8.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-7"><i class="fa fa-check"></i><b>14.8.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.8.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-7"><i class="fa fa-check"></i><b>14.8.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>14.9</b> metagenomeSeq</a>
<ul>
<li class="chapter" data-level="14.9.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-8"><i class="fa fa-check"></i><b>14.9.1</b> Vignette</a></li>
<li class="chapter" data-level="14.9.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-8"><i class="fa fa-check"></i><b>14.9.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.9.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-8"><i class="fa fa-check"></i><b>14.9.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.9.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-8"><i class="fa fa-check"></i><b>14.9.4</b> Procedures</a></li>
<li class="chapter" data-level="14.9.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-8"><i class="fa fa-check"></i><b>14.9.5</b> Key output</a></li>
<li class="chapter" data-level="14.9.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-8"><i class="fa fa-check"></i><b>14.9.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>14.10</b> t-test</a>
<ul>
<li class="chapter" data-level="14.10.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-9"><i class="fa fa-check"></i><b>14.10.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.10.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-9"><i class="fa fa-check"></i><b>14.10.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.10.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-9"><i class="fa fa-check"></i><b>14.10.3</b> Procedures</a></li>
<li class="chapter" data-level="14.10.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-9"><i class="fa fa-check"></i><b>14.10.4</b> Key output</a></li>
<li class="chapter" data-level="14.10.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-8"><i class="fa fa-check"></i><b>14.10.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.10.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-9"><i class="fa fa-check"></i><b>14.10.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>14.11</b> Wilcoxon test</a>
<ul>
<li class="chapter" data-level="14.11.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-10"><i class="fa fa-check"></i><b>14.11.1</b> Input Data Type</a></li>
<li class="chapter" data-level="14.11.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-10"><i class="fa fa-check"></i><b>14.11.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.11.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-10"><i class="fa fa-check"></i><b>14.11.3</b> Procedures</a></li>
<li class="chapter" data-level="14.11.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-10"><i class="fa fa-check"></i><b>14.11.4</b> Key output</a></li>
<li class="chapter" data-level="14.11.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-9"><i class="fa fa-check"></i><b>14.11.5</b> Original Syntax</a></li>
<li class="chapter" data-level="14.11.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-10"><i class="fa fa-check"></i><b>14.11.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>14.12</b> RAIDA</a>
<ul>
<li class="chapter" data-level="14.12.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-9"><i class="fa fa-check"></i><b>14.12.1</b> Vignette</a></li>
<li class="chapter" data-level="14.12.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-11"><i class="fa fa-check"></i><b>14.12.2</b> Input Data Type</a></li>
<li class="chapter" data-level="14.12.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-11"><i class="fa fa-check"></i><b>14.12.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="14.12.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-11"><i class="fa fa-check"></i><b>14.12.4</b> Procedures</a></li>
<li class="chapter" data-level="14.12.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-11"><i class="fa fa-check"></i><b>14.12.5</b> Key output</a></li>
<li class="chapter" data-level="14.12.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-10"><i class="fa fa-check"></i><b>14.12.6</b> Original Syntax</a></li>
<li class="chapter" data-level="14.12.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-11"><i class="fa fa-check"></i><b>14.12.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="14.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>14.13</b> Summary</a></li>
<li class="chapter" data-level="14.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>14.14</b> Advantages and disadvantages</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>15</b> Examples</a>
<ul>
<li class="chapter" data-level="15.1" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>15.1</b> Workflow description</a></li>
<li class="chapter" data-level="15.2" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>15.2</b> Amplicon sequencing (16s)</a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>15.2.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="15.2.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>15.2.2</b> Quality Control</a></li>
<li class="chapter" data-level="15.2.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>15.2.3</b> Data processing</a></li>
<li class="chapter" data-level="15.2.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>15.2.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="15.2.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>15.2.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="15.2.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>15.2.6</b> Microbial composition</a></li>
<li class="chapter" data-level="15.2.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>15.2.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="examples.html"><a href="examples.html#metagenomics-mgs"><i class="fa fa-check"></i><b>15.3</b> Metagenomics (MGS)</a>
<ul>
<li class="chapter" data-level="15.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>15.3.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="15.3.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>15.3.2</b> Quality Control</a></li>
<li class="chapter" data-level="15.3.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>15.3.3</b> Data processing</a></li>
<li class="chapter" data-level="15.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>15.3.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="15.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>15.3.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="15.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-2"><i class="fa fa-check"></i><b>15.3.6</b> Microbial composition</a></li>
<li class="chapter" data-level="15.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>15.3.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="examples.html"><a href="examples.html#systematic-information-11"><i class="fa fa-check"></i><b>15.4</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">XMAS 2.0 Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quality-evaluation" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Quality Evaluation<a href="quality-evaluation.html#quality-evaluation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Quality control of DADA2 results will help us have more rational determinations on the further data analysis. Firstly, the reads’ track of DADA2 could show us the Changed Ratio of reads through the in-house standard amplicon sequencing data upstream pipeline. Then, the Evaluation of the spike-in samples from the Reference Matrix will reflect the quality of sequencing data. Finally, we recommend you that utilizing the rarefaction curves to assess the sequence depth per sample and choose the rational cutoff of OTU Number to do rarefy.</p>
<p><strong>Loading packages</strong></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="quality-evaluation.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb41-2"><a href="quality-evaluation.html#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb41-3"><a href="quality-evaluation.html#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb41-4"><a href="quality-evaluation.html#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb41-5"><a href="quality-evaluation.html#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb41-6"><a href="quality-evaluation.html#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb41-7"><a href="quality-evaluation.html#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
<div id="reads-track" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Reads’ track<a href="quality-evaluation.html#reads-track" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>this procedure only perform in 16s data</li>
</ul>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="quality-evaluation.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_Dada2Track</span>(<span class="at">data =</span> dada2_res<span class="sc">$</span>reads_track)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-47"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-47-1.png" alt="Reads' track of DADA2" width="672" />
<p class="caption">
Figure 5.1: Reads’ track of DADA2
</p>
</div>
</div>
<div id="spike-in-brs-sample-assessment" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Spike-in (BRS) sample assessment<a href="quality-evaluation.html#spike-in-brs-sample-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The spike-in sample is use to evaluate the consistent quality on bacteria when we have multiple sequence batches data. We devised an evaluation system containing the <em>Correlation Coefficient</em>, <em>Bray Curtis Distance</em> and <em>Impurity Level</em> to assess the sequencing data quality.</p>
<p><strong>Please use the default Reference and Save directory to obtain and save the spike-in sample matrix when you run <code>run_RefCheck</code>.</strong></p>
<div id="s-data" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> 16s data<a href="quality-evaluation.html#s-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The taxonomic levels of spike-in sample’s bacteria is genus. Firstly, using the <code>summarize_taxa</code> to get the genus level phyloseq object and get the BRS_ID.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="quality-evaluation.html#cb43-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps, </span>
<span id="cb43-2"><a href="quality-evaluation.html#cb43-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb43-3"><a href="quality-evaluation.html#cb43-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-4"><a href="quality-evaluation.html#cb43-4" aria-hidden="true" tabindex="-1"></a>dada2_ps_genus<span class="sc">@</span>sam_data</span></code></pre></div>
<pre><code>##       Group
## S6030    BB
## S6032    BB
## S6033    BB
## S6035    AA
## S6036    BB
## S6037    AA
## S6040    BB
## S6043    AA
## S6045    BB
## S6046    BB
## S6048    BB
## S6049    AA
## S6050    BB
## S6054    BB
## S6055    BB
## S6058    BB
## S6059    AA
## S6060    AA
## S6061    AA
## S6063    BB
## S6065    AA
## S6066    AA
## S6068    BB
## S8005    QC</code></pre>
<p>do <code>run_RefCheck</code> under the optimal parameters.</p>
<ul>
<li><p><em>BRS_ID</em>: the ID of BRS sample;</p></li>
<li><p><em>Reference</em>: the directory of the latest spike-in sample matrix (default: /share/projects/Analytics/analytics/XMAS/RefCheck/);</p></li>
<li><p><em>Save</em>: the directory to save the latest spike-in sample matrix (default: /share/projects/Analytics/analytics/XMAS/RefCheck/).</p></li>
</ul>
<p>To see more details to use <code>?run_RefCheck</code>.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb45-1"><a href="quality-evaluation.html#cb45-1" aria-hidden="true" tabindex="-1"></a>run_RefCheck(</span>
<span id="cb45-2"><a href="quality-evaluation.html#cb45-2" aria-hidden="true" tabindex="-1"></a>    ps = dada2_ps_genus,</span>
<span id="cb45-3"><a href="quality-evaluation.html#cb45-3" aria-hidden="true" tabindex="-1"></a>    BRS_ID = &quot;S8005&quot;,</span>
<span id="cb45-4"><a href="quality-evaluation.html#cb45-4" aria-hidden="true" tabindex="-1"></a>    Ref_type = &quot;16s&quot;)</span></code></pre></div>
<pre><code>## Noting: the Reference Matrix is for 16s
## S8005 is in the Reference Matrix&#39;s samples and remove it to run
## 
## ############Matched baterica of the BRS sample#############
## The number of BRS&#39; bacteria matched the Reference Matrix is [15]
## g__Bifidobacterium
## g__Bacteroides
## g__Faecalibacterium
## g__Lactobacillus
## g__Parabacteroides
## g__Collinsella
## g__Coprococcus_3
## g__Dorea
## g__Streptococcus
## g__Roseburia
## g__Anaerostipes
## g__Escherichia_Shigella
## g__Enterococcus
## g__Prevotella_9
## g__Eggerthella
## 
## The number of the additional bacteria compared to Reference Matrix is [1]
## ###########################################################
## 
## ##################Status of the BRS sample##################
## Whether the BRS has the all bateria of Reference Matrix: TRUE
## Correlation Coefficient of the BRS is: 0.9714
## Bray Curtis of the BRS is: 0.07607
## Impurity of the BRS is: 0.06409
## ###########################################################
## #####Final Evaluation Results of the BRS #######
## The BRS of sequencing dataset passed the cutoff of the Reference Matrix 
## Cutoff of Coefficient is 0.8946
## Cutoff of BrayCurtis is 0.3878
## Cutoff of Impurity is 0.1565
## ###########################################################
##                             8002        8003        8004       8006        8007        8008        8009        8005        mean
## Bifidobacterium      31.11079015 30.88310969 32.31232692 18.4930259 20.20409870 17.96225391 18.03588291 27.22437034 24.52823232
## Bacteroides          20.44753484 14.46581958 24.57151411 26.7370147 25.85863655 27.51353663 26.99272343 24.23896093 23.85321759
## Faecalibacterium      0.79850615  0.62937893  1.05531023  1.7487249  1.64282727  1.96346413  1.81219797  1.04035376  1.33634542
## Lactobacillus         2.61732573  3.36856272  3.44379163  5.9292703  5.78000836  5.78189064  6.32672332  3.87088505  4.63980722
## Parabacteroides       7.11124408  7.45952579  5.36075144  8.7149995  8.01840234  8.74899584  8.62634005  5.61833757  7.45732457
## Collinsella           0.12792605  0.88271385  0.55665744  1.2764130  0.67921372  1.89356397  1.26367828  0.45502126  0.89189845
## Coprococcus_3         1.00380683  0.97969362  0.80270938  1.6693557  1.56419908  1.67969035  1.71221463  0.87586251  1.28594151
## Dorea                 2.80715148  3.45564660  2.46613277  3.9684605  3.99163530  3.81529666  3.69382881  2.34880690  3.31836988
## Streptococcus         2.91960260  3.43387563  2.59149764  3.4818362  3.31409452  3.51378702  3.37721491  2.68740253  3.16491388
## Roseburia             0.03404484  0.04750030  0.02806676  0.0338295  0.03178586  0.02503886  0.02499583  0.03311188  0.03229673
## Anaerostipes          0.32291011  0.43245062  0.31528329  0.5386697  0.44500209  0.53416240  0.48047548  0.32471000  0.42420796
## Escherichia_Shigella 15.27581475 16.00265210 12.36527954 10.8423545 13.43203680 10.15743185 11.59945565 14.03516268 12.96377349
## Enterococcus         14.51444842 14.66472707 11.04239952 13.1674820 12.07360937 12.88771113 12.84924735 11.61906390 12.85233610
## Prevotella_9          0.77374627  3.07465463  2.75709154  2.9145415  2.52446675  3.04952478  2.84952508  5.40257632  2.91826586
## Eggerthella           0.04951976  0.15437596  0.27131203  0.4840221  0.43998327  0.47365181  0.35549631  0.16128688  0.29870601
## Impurity_level        0.08562792  0.06531291  0.05987576  0.0000000  0.00000000  0.00000000  0.00000000  0.06409000  0.03436332</code></pre>
<p>We could see that the messages are comprised of four parts.</p>
<ol style="list-style-type: decimal">
<li><p>the 1st part showed the type of <em>reference matrix</em> and whether the spike-in sample had been added to reference matrix;</p></li>
<li><p>the 2nd part revealed that <em>what</em> and <em>how many</em> the <em>matched bacterica</em> of the spike-in sample are;</p></li>
<li><p>the 3nd part showed that the <em>value</em> of evaluation system in the spike-in sample;</p></li>
<li><p>the 4nd part showed that whether the spike-in sample <em>passes</em> the cutoff of evaluation system.</p></li>
</ol>
</div>
<div id="metagenomic-data" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Metagenomic data<a href="quality-evaluation.html#metagenomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The taxonomic levels of spike-in sample’s bacteria is species. Firstly, using the <code>summarize_taxa</code> to get the species level phyloseq object and then do <code>run_RefCheck</code> under the optimal parameters.</p>
<details>
<summary>
<strong>Quality Control by spike-in sample in metagenomic</strong>
</summary>
<ul>
<li>get the BRS_ID</li>
</ul>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="quality-evaluation.html#cb47-1" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> metaphlan2_ps, </span>
<span id="cb47-2"><a href="quality-evaluation.html#cb47-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb47-3"><a href="quality-evaluation.html#cb47-3" aria-hidden="true" tabindex="-1"></a>metaphlan2_ps_species<span class="sc">@</span>sam_data</span></code></pre></div>
<pre><code>##      Group phynotype
## s1      BB      0.00
## s2      AA      2.50
## s3      BB      0.00
## s4      AA      1.25
## s5      AA     30.00
## s6      AA     15.00
## s7      BB      8.75
## s8      BB      0.00
## s9      BB      3.75
## s10     BB      2.50
## s11     BB     15.00
## s12     BB      2.50
## s13     BB      2.50
## s14     BB      0.00
## s15     BB      1.07
## s16     BB      2.50
## s17     AA      5.00
## s18     BB     35.00
## s19     BB      7.50
## s20     BB     15.00
## s21     AA      3.75
## s22     AA      3.75
## refE    QC        NA</code></pre>
<ul>
<li>run <code>run_RefCheck</code></li>
</ul>
<div class="sourceCode" id="cb49"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb49-1"><a href="quality-evaluation.html#cb49-1" aria-hidden="true" tabindex="-1"></a>run_RefCheck(</span>
<span id="cb49-2"><a href="quality-evaluation.html#cb49-2" aria-hidden="true" tabindex="-1"></a>    ps = metaphlan2_ps_species,</span>
<span id="cb49-3"><a href="quality-evaluation.html#cb49-3" aria-hidden="true" tabindex="-1"></a>    BRS_ID = &quot;refE&quot;,</span>
<span id="cb49-4"><a href="quality-evaluation.html#cb49-4" aria-hidden="true" tabindex="-1"></a>    Ref_type = &quot;MGS&quot;)</span></code></pre></div>
<pre><code>## Noting: the Reference Matrix is for MGS
## 
## ############Matched baterica of the BRS sample#############
## The number of BRS&#39; bacteria matched the Reference Matrix is [16]
## s__Bifidobacterium_longum
## s__Bacteroides_uniformis
## s__Faecalibacterium_prausnitzii
## s__Bifidobacterium_adolescentis
## s__Bacteroides_thetaiotaomicron
## s__Collinsella_aerofaciens
## s__Coprococcus_comes
## s__Dorea_formicigenerans
## s__Streptococcus_salivarius
## s__Bacteroides_xylanisolvens
## s__Bacteroides_ovatus
## s__Roseburia_hominis
## s__Bacteroides_vulgatus
## s__Prevotella_copri
## s__Bifidobacterium_pseudocatenulatum
## s__Lachnospiraceae_bacterium_5_1_63FAA
## The number of bacteria unmatched the Reference Matrix is [9]
## s__Lactobacillus_salivarius
## s__Bacteroides_fragilis
## s__Parabacteroides_goldsteinii
## s__Escherichia_coli
## s__Enterococcus_faecalis
## s__Bifidobacterium_bifidum
## s__Lactobacillus_pentosus
## s__Bacteroides_intestinalis
## s__Eggerthella_unclassified
## The number of the additional bacteria compared to the Reference Matrix is [56]
## ###########################################################
## 
## ##################Status of the BRS sample##################
## Whether the BRS has the all bateria of Reference Matrix: FALSE
## Correlation Coefficient of the BRS is: -0.06471
## Bray Curtis of the BRS is: 0.8752
## Impurity of the BRS is: 32.69
## ###########################################################
## #####Final Evaluation Results of the BRS #######
## The BRS of sequencing dataset didn&#39;t pass the cutoff of the Reference Matrix
## ###########################################################
##                                         7682    7683    7684    7685     7842     7843     7844     7845     refE        mean
## Bifidobacterium_longum              10.31563 9.25812 9.69184 7.76031 11.03311 11.61484 12.29030 11.69019  0.01646  9.29675556
## Bacteroides_uniformis                2.24195 2.24035 1.92015 2.18435  2.43230  2.38180  2.13830  2.41437  0.21061  2.01824222
## Faecalibacterium_prausnitzii         0.65615 0.60153 0.60112 0.62079  0.54147  0.55383  0.58806  0.54655  1.61939  0.70321000
## Bifidobacterium_adolescentis         7.05426 6.28460 6.57297 6.25448  4.69357  4.80628  4.94943  4.84278  0.04649  5.05609556
## Bacteroides_thetaiotaomicron         3.25076 3.31897 3.22418 3.43809  3.35611  3.38323  3.29098  3.30355  1.47422  3.11556556
## Collinsella_aerofaciens              0.56220 0.53249 0.60476 0.47934  0.65513  0.68833  0.76063  0.66896  0.07251  0.55826111
## Coprococcus_comes                    2.28581 2.42978 2.25227 2.82904  1.21527  1.16485  1.07160  1.07834  0.01444  1.59348889
## Dorea_formicigenerans                4.83509 5.02149 5.18268 5.56891  3.34720  3.09877  3.08677  2.70732  0.02775  3.65288667
## Streptococcus_salivarius             3.54266 3.74119 3.62036 4.01546  2.90216  2.70193  2.61760  2.57348  0.02846  2.86036667
## Bacteroides_xylanisolvens            1.55648 1.84824 1.91166 1.85273  1.75220  1.74002  1.69811  1.67676  0.32466  1.59565111
## Bacteroides_ovatus                   3.08489 3.27226 3.10904 3.24565  3.51376  3.50063  3.37872  3.42300  0.25782  2.97619667
## Roseburia_hominis                    0.04383 0.04183 0.04107 0.02304  0.03853  0.03464  0.03532  0.03597  0.01307  0.03414444
## Bacteroides_vulgatus                 3.06713 3.20369 3.14979 3.15352  3.24822  3.09280  3.13038  3.06113  2.14684  3.02816667
## Prevotella_copri                     2.03128 1.99619 1.92504 2.15422  1.57638  1.60584  1.57913  1.59224 60.84109  8.36682333
## Bifidobacterium_pseudocatenulatum    6.89310 6.36177 7.39605 5.76804  6.19464  6.47094  7.41615  6.27837  0.15023  5.88103222
## Lachnospiraceae_bacterium_5_1_63FAA  0.22002 0.26062 0.28058 0.33329  0.07093  0.06778  0.05229  0.04781  0.06825  0.15573000
## Impurity_level                       7.33876 7.34799 7.14891 6.47250  8.88190  9.72198  9.72162 10.57600 32.69000 11.09996222
##                                                                                Evaluation
## Bifidobacterium_longum              refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Bacteroides_uniformis               refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Faecalibacterium_prausnitzii        refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Bifidobacterium_adolescentis        refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Bacteroides_thetaiotaomicron        refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Collinsella_aerofaciens             refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Coprococcus_comes                   refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Dorea_formicigenerans               refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Streptococcus_salivarius            refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Bacteroides_xylanisolvens           refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Bacteroides_ovatus                  refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Roseburia_hominis                   refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Bacteroides_vulgatus                refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Prevotella_copri                    refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Bifidobacterium_pseudocatenulatum   refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Lachnospiraceae_bacterium_5_1_63FAA refE didn&#39;t pass the threshold (2022-08-16 10:29:55).
## Impurity_level                      refE didn&#39;t pass the threshold (2022-08-16 10:29:55).</code></pre>
</details>
</div>
</div>
<div id="remove-brs" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Remove BRS<a href="quality-evaluation.html#remove-brs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After evaluating the sequencing quality, we remove the BRS.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="quality-evaluation.html#cb51-1" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb51-2"><a href="quality-evaluation.html#cb51-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">ps =</span> dada2_ps,</span>
<span id="cb51-3"><a href="quality-evaluation.html#cb51-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb51-4"><a href="quality-evaluation.html#cb51-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb51-5"><a href="quality-evaluation.html#cb51-5" aria-hidden="true" tabindex="-1"></a>dada2_ps_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
</div>
<div id="rarefaction-curves" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> Rarefaction Curves<a href="quality-evaluation.html#rarefaction-curves" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Rarefaction curves are often used when calculating alpha diversity indices, because increasing numbers of sequenced taxa allow increasingly accurate estimates of total population diversity. Rarefaction curves can therefore be used to estimate the full sample richness, as compared to the observed sample richness.</p>
</blockquote>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="quality-evaluation.html#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_RarefCurve</span>(<span class="at">ps =</span> dada2_ps_remove_BRS,</span>
<span id="cb53-2"><a href="quality-evaluation.html#cb53-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">taxa_level =</span> <span class="st">&quot;OTU&quot;</span>,</span>
<span id="cb53-3"><a href="quality-evaluation.html#cb53-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">step =</span> <span class="dv">400</span>,</span>
<span id="cb53-4"><a href="quality-evaluation.html#cb53-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">label =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb53-5"><a href="quality-evaluation.html#cb53-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## rarefying sample S6030
## rarefying sample S6032
## rarefying sample S6033
## rarefying sample S6035
## rarefying sample S6036
## rarefying sample S6037
## rarefying sample S6040
## rarefying sample S6043
## rarefying sample S6045
## rarefying sample S6046
## rarefying sample S6048
## rarefying sample S6049
## rarefying sample S6050
## rarefying sample S6054
## rarefying sample S6055
## rarefying sample S6058
## rarefying sample S6059
## rarefying sample S6060
## rarefying sample S6061
## rarefying sample S6063
## rarefying sample S6065
## rarefying sample S6066
## rarefying sample S6068</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-53"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-53-1.png" alt="Rarefaction Curves" width="672" />
<p class="caption">
Figure 5.2: Rarefaction Curves
</p>
</div>
<p>The result showed that all the samples had different sequencing depth but had the full sample richness.</p>
</div>
<div id="summarize-phyloseq-class-object" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Summarize phyloseq-class object<a href="quality-evaluation.html#summarize-phyloseq-class-object" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Summarizing the phyloseq-class object by using <code>summarize_phyloseq</code>. It displayed that briefly introduction of the object.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="quality-evaluation.html#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(<span class="at">ps =</span> dada2_ps_remove_BRS)</span>
<span id="cb55-2"><a href="quality-evaluation.html#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Compositional = NO2</span></span>
<span id="cb55-3"><a href="quality-evaluation.html#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="do">## 1] Min. number of reads = 511812] Max. number of reads = 716673] Total number of reads = 14089154] Average number of reads = 61257.17391304355] Median number of reads = 614357] Sparsity = 0.8610248447204976] Any OTU sum to 1 or less? YES8] Number of singletons = 49] Percent of OTUs that are singletons</span></span>
<span id="cb55-4"><a href="quality-evaluation.html#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="do">##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 1Group2</span></span>
<span id="cb55-5"><a href="quality-evaluation.html#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="do">## [[1]]</span></span>
<span id="cb55-6"><a href="quality-evaluation.html#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;1] Min. number of reads = 51181&quot;</span></span>
<span id="cb55-7"><a href="quality-evaluation.html#cb55-7" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-8"><a href="quality-evaluation.html#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="do">## [[2]]</span></span>
<span id="cb55-9"><a href="quality-evaluation.html#cb55-9" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;2] Max. number of reads = 71667&quot;</span></span>
<span id="cb55-10"><a href="quality-evaluation.html#cb55-10" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-11"><a href="quality-evaluation.html#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="do">## [[3]]</span></span>
<span id="cb55-12"><a href="quality-evaluation.html#cb55-12" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;3] Total number of reads = 1408915&quot;</span></span>
<span id="cb55-13"><a href="quality-evaluation.html#cb55-13" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-14"><a href="quality-evaluation.html#cb55-14" aria-hidden="true" tabindex="-1"></a><span class="do">## [[4]]</span></span>
<span id="cb55-15"><a href="quality-evaluation.html#cb55-15" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;4] Average number of reads = 61257.1739130435&quot;</span></span>
<span id="cb55-16"><a href="quality-evaluation.html#cb55-16" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-17"><a href="quality-evaluation.html#cb55-17" aria-hidden="true" tabindex="-1"></a><span class="do">## [[5]]</span></span>
<span id="cb55-18"><a href="quality-evaluation.html#cb55-18" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;5] Median number of reads = 61435&quot;</span></span>
<span id="cb55-19"><a href="quality-evaluation.html#cb55-19" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-20"><a href="quality-evaluation.html#cb55-20" aria-hidden="true" tabindex="-1"></a><span class="do">## [[6]]</span></span>
<span id="cb55-21"><a href="quality-evaluation.html#cb55-21" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;7] Sparsity = 0.861024844720497&quot;</span></span>
<span id="cb55-22"><a href="quality-evaluation.html#cb55-22" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-23"><a href="quality-evaluation.html#cb55-23" aria-hidden="true" tabindex="-1"></a><span class="do">## [[7]]</span></span>
<span id="cb55-24"><a href="quality-evaluation.html#cb55-24" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;6] Any OTU sum to 1 or less? YES&quot;</span></span>
<span id="cb55-25"><a href="quality-evaluation.html#cb55-25" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-26"><a href="quality-evaluation.html#cb55-26" aria-hidden="true" tabindex="-1"></a><span class="do">## [[8]]</span></span>
<span id="cb55-27"><a href="quality-evaluation.html#cb55-27" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;8] Number of singletons = 4&quot;</span></span>
<span id="cb55-28"><a href="quality-evaluation.html#cb55-28" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-29"><a href="quality-evaluation.html#cb55-29" aria-hidden="true" tabindex="-1"></a><span class="do">## [[9]]</span></span>
<span id="cb55-30"><a href="quality-evaluation.html#cb55-30" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;</span></span>
<span id="cb55-31"><a href="quality-evaluation.html#cb55-31" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-32"><a href="quality-evaluation.html#cb55-32" aria-hidden="true" tabindex="-1"></a><span class="do">## [[10]]</span></span>
<span id="cb55-33"><a href="quality-evaluation.html#cb55-33" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;10] Number of sample variables are: 1&quot;</span></span>
<span id="cb55-34"><a href="quality-evaluation.html#cb55-34" aria-hidden="true" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb55-35"><a href="quality-evaluation.html#cb55-35" aria-hidden="true" tabindex="-1"></a><span class="do">## [[11]]</span></span>
<span id="cb55-36"><a href="quality-evaluation.html#cb55-36" aria-hidden="true" tabindex="-1"></a><span class="do">## [1] &quot;Group&quot;</span></span></code></pre></div>
<p>The minus account of the OTU counts is <em>51181</em> in the phyloseq object, and we can use it as the threshold to rarefy.</p>
<p>Notice the <strong>Sparsity (0.865)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.</p>
</div>
<div id="variability" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> Variability<a href="quality-evaluation.html#variability" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We use the variability to measure measure heterogeneity in OTU/ASV abundance data.</p>
<p><span class="math display">\[Variability_{X} = \frac{sd(X)}{mean(X)}\]</span></p>
<div id="coefficient-of-variation" class="section level3 hasAnchor" number="5.6.1">
<h3><span class="header-section-number">5.6.1</span> Coefficient of variation<a href="quality-evaluation.html#coefficient-of-variation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Coefficient of variation (C.V)</strong>, i.e. sd(x)/mean(x) is a widely used approach to measure heterogeneity in OTU/ASV abundance data. The plot below shows CV-mean(relaive mean abundance) relationship in the scatter plot, where variation is calculated for each OTU/ASV across samples versus mean relative abundance.
Now plot C.V.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="quality-evaluation.html#cb56-1" aria-hidden="true" tabindex="-1"></a>pl <span class="ot">&lt;-</span> <span class="fu">plot_taxa_cv</span>(<span class="at">ps =</span> dada2_ps_remove_BRS, </span>
<span id="cb56-2"><a href="quality-evaluation.html#cb56-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">plot.type =</span> <span class="st">&quot;scatter&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-55"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-55-1.png" alt="Density of the Taxa Mean abundance" width="672" />
<p class="caption">
Figure 5.3: Density of the Taxa Mean abundance
</p>
</div>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="quality-evaluation.html#cb57-1" aria-hidden="true" tabindex="-1"></a>pl <span class="sc">+</span> <span class="fu">scale_x_log10</span>()</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-56"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-56-1.png" alt="Coefficient of variation" width="672" />
<p class="caption">
Figure 5.4: Coefficient of variation
</p>
</div>
<p>From the above two plots, we see that there are several OTUs which have <strong>high C.V.</strong> and <strong>low mean</strong>.</p>
</div>
<div id="distribution-of-number-of-sequences" class="section level3 hasAnchor" number="5.6.2">
<h3><span class="header-section-number">5.6.2</span> Distribution of number of sequences<a href="quality-evaluation.html#distribution-of-number-of-sequences" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Checking for distribution of number of sequences</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="quality-evaluation.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_read_distribution</span>(<span class="at">ps =</span> dada2_ps_remove_BRS, </span>
<span id="cb58-2"><a href="quality-evaluation.html#cb58-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">group=</span><span class="st">&quot;Group&quot;</span>, </span>
<span id="cb58-3"><a href="quality-evaluation.html#cb58-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">plot.type =</span> <span class="st">&quot;density&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-57"></span>
<img src="Data-analysis-based-on-XMAS-package_files/figure-html/unnamed-chunk-57-1.png" alt="Distribution of number of sequences" width="672" />
<p class="caption">
Figure 5.5: Distribution of number of sequences
</p>
</div>
<p>Different distribution between group <em>AA</em> and <em>BB</em>.</p>
</div>
</div>
<div id="systematic-information-2" class="section level2 hasAnchor" number="5.7">
<h2><span class="header-section-number">5.7</span> Systematic Information<a href="quality-evaluation.html#systematic-information-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="quality-evaluation.html#cb59-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-08-16
##  rstudio  2022.07.1+554 Spotted Wakerobin (desktop)
##  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version  date (UTC) lib source
##  abind                  1.4-5    2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18   2021-09-16 [1] CRAN (R 4.1.0)
##  annotate               1.72.0   2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2   2021-11-09 [1] Bioconductor
##  ape                    5.6-2    2022-03-02 [1] CRAN (R 4.1.2)
##  assertthat             0.2.1    2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1    2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3    2015-07-28 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0   2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0   2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3   2021-12-09 [1] Bioconductor
##  biomformat             1.22.0   2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0   2021-10-26 [1] Bioconductor
##  bit                    4.0.4    2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5    2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7    2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2    2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.27     2022-06-14 [1] CRAN (R 4.1.2)
##  brio                   1.1.3    2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12   2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1    2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6    2021-08-19 [1] CRAN (R 4.1.0)
##  callr                  3.7.0    2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12   2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5    2022-01-06 [1] CRAN (R 4.1.2)
##  caTools                1.18.2   2021-03-28 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0    2020-02-06 [1] CRAN (R 4.1.0)
##  cli                    3.3.0    2022-04-25 [1] CRAN (R 4.1.2)
##  cluster                2.1.2    2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18   2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2    2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3    2022-02-21 [1] CRAN (R 4.1.2)
##  cowplot                1.1.1    2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0    2022-02-14 [1] CRAN (R 4.1.2)
##  data.table             1.14.2   2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2    2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0   2021-10-26 [1] Bioconductor
##  desc                   1.4.1    2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0   2021-10-26 [1] Bioconductor
##  devtools               2.4.3    2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29   2021-12-01 [1] CRAN (R 4.1.0)
##  dplyr                * 1.0.8    2022-02-08 [1] CRAN (R 4.1.2)
##  DT                     0.21     2022-02-26 [1] CRAN (R 4.1.2)
##  edgeR                  3.36.0   2021-10-26 [1] Bioconductor
##  ellipsis               0.3.2    2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15     2022-02-18 [1] CRAN (R 4.1.2)
##  FactoMineR             2.4      2020-12-11 [1] CRAN (R 4.1.0)
##  fansi                  1.0.2    2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0    2021-02-28 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0    2021-01-25 [1] CRAN (R 4.1.0)
##  flashClust             1.01-2   2012-08-21 [1] CRAN (R 4.1.0)
##  foreach                1.5.2    2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82   2022-01-13 [1] CRAN (R 4.1.2)
##  Formula                1.2-4    2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2    2021-12-08 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0   2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0   2021-10-26 [1] Bioconductor
##  generics               0.1.2    2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1   2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7    2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1   2021-11-18 [1] Bioconductor
##  ggiraph                0.8.2    2022-02-22 [1] CRAN (R 4.1.2)
##  ggiraphExtra           0.3.0    2020-10-06 [1] CRAN (R 4.1.2)
##  ggplot2              * 3.3.5    2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr               * 0.4.0    2020-06-27 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1    2021-01-15 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3    2021-09-09 [1] CRAN (R 4.1.0)
##  glmnet                 4.1-3    2021-11-02 [1] CRAN (R 4.1.0)
##  glue                   1.6.2    2022-02-24 [1] CRAN (R 4.1.2)
##  gplots                 3.1.1    2020-11-28 [1] CRAN (R 4.1.0)
##  gridExtra              2.3      2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0    2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2    2021-06-06 [1] CRAN (R 4.1.0)
##  highr                  0.9      2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0    2021-10-07 [1] CRAN (R 4.1.0)
##  htmlTable              2.4.0    2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2    2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4    2021-09-08 [1] CRAN (R 4.1.0)
##  httr                   1.4.2    2020-07-20 [1] CRAN (R 4.1.0)
##  igraph                 1.2.11   2022-01-04 [1] CRAN (R 4.1.2)
##  insight                0.17.0   2022-03-29 [1] CRAN (R 4.1.2)
##  IRanges              * 2.28.0   2021-10-26 [1] Bioconductor
##  iterators              1.0.14   2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9    2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4    2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0    2022-02-22 [1] CRAN (R 4.1.2)
##  kableExtra             1.3.4    2021-02-20 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0   2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20  2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.39     2022-04-26 [1] CRAN (R 4.1.2)
##  labeling               0.4.2    2020-10-20 [1] CRAN (R 4.1.0)
##  lattice                0.20-45  2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29   2019-12-19 [1] CRAN (R 4.1.0)
##  leaps                  3.1      2020-01-16 [1] CRAN (R 4.1.0)
##  libcoin                1.0-9    2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1    2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1   2022-02-17 [1] Bioconductor
##  locfit                 1.5-9.5  2022-03-03 [1] CRAN (R 4.1.2)
##  magrittr               2.0.2    2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55   2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix                 1.4-0    2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0    2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0   2021-09-17 [1] CRAN (R 4.1.0)
##  memoise                2.0.1    2021-11-26 [1] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0   2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39   2022-02-24 [1] CRAN (R 4.1.2)
##  modeltools             0.2-23   2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18   2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0   2021-10-26 [1] Bioconductor
##  munsell                0.5.0    2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3    2021-10-08 [1] CRAN (R 4.1.0)
##  mycor                  0.1.1    2018-04-10 [1] CRAN (R 4.1.0)
##  nlme                   3.1-155  2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17   2022-01-13 [1] CRAN (R 4.1.2)
##  permute                0.9-7    2022-01-27 [1] CRAN (R 4.1.2)
##  phyloseq             * 1.38.0   2021-10-26 [1] Bioconductor
##  pillar                 1.7.0    2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1    2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3    2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4    2021-11-30 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6    2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7    2013-12-03 [1] CRAN (R 4.1.0)
##  ppcor                  1.1      2015-12-03 [1] CRAN (R 4.1.0)
##  prettyunits            1.1.1    2020-01-24 [1] CRAN (R 4.1.0)
##  processx               3.5.2    2021-04-30 [1] CRAN (R 4.1.0)
##  ps                     1.6.0    2021-02-28 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4    2020-04-17 [1] CRAN (R 4.1.0)
##  R6                     2.5.1    2021-08-19 [1] CRAN (R 4.1.0)
##  RColorBrewer           1.1-2    2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                   1.0.8.2  2022-03-11 [1] CRAN (R 4.1.2)
##  RCurl                  1.98-1.6 2022-02-08 [1] CRAN (R 4.1.2)
##  remotes                2.4.2    2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4    2020-04-09 [1] CRAN (R 4.1.0)
##  rhdf5                  2.38.1   2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0    2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0   2021-10-26 [1] Bioconductor
##  rlang                  1.0.2    2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.14     2022-04-25 [1] CRAN (R 4.1.2)
##  rpart                  4.1.16   2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2    2020-11-15 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10   2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0    2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13     2020-11-12 [1] CRAN (R 4.1.0)
##  rvest                  1.0.2    2021-10-16 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3   2021-11-21 [1] Bioconductor
##  sandwich               3.0-1    2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0    2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1    2020-05-11 [1] CRAN (R 4.1.0)
##  scatterplot3d          0.3-41   2018-03-14 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2    2021-12-06 [1] CRAN (R 4.1.0)
##  shape                  1.4.6    2021-05-19 [1] CRAN (R 4.1.0)
##  sjlabelled             1.2.0    2022-04-10 [1] CRAN (R 4.1.2)
##  sjmisc                 2.8.9    2021-12-03 [1] CRAN (R 4.1.0)
##  stringi                1.7.6    2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0    2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0   2021-10-26 [1] Bioconductor
##  survival               3.3-1    2022-03-03 [1] CRAN (R 4.1.2)
##  svglite                2.1.0    2022-02-03 [1] CRAN (R 4.1.2)
##  systemfonts            1.0.4    2022-02-11 [1] CRAN (R 4.1.2)
##  testthat               3.1.2    2022-01-20 [1] CRAN (R 4.1.2)
##  TH.data                1.1-0    2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6    2021-11-07 [1] CRAN (R 4.1.0)
##  tidyr                  1.2.0    2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2    2022-02-21 [1] CRAN (R 4.1.2)
##  usethis                2.1.5    2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2    2021-07-24 [1] CRAN (R 4.1.0)
##  uuid                   1.0-3    2021-11-01 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8    2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                  2.5-7    2020-11-28 [1] CRAN (R 4.1.0)
##  viridisLite            0.4.0    2021-04-13 [1] CRAN (R 4.1.0)
##  webshot                0.5.3    2022-04-14 [1] CRAN (R 4.1.2)
##  withr                  2.5.0    2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0   2021-10-26 [1] Bioconductor
##  xfun                   0.30     2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                * 2.1.7.5  2022-08-16 [1] local
##  XML                    3.99-0.9 2022-02-24 [1] CRAN (R 4.1.2)
##  xml2                   1.3.3    2021-11-30 [1] CRAN (R 4.1.0)
##  xtable                 1.8-4    2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0   2021-10-26 [1] Bioconductor
##  yaml                   2.3.5    2022-02-21 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0   2021-10-26 [1] Bioconductor
##  zoo                    1.8-9    2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="convert-inputdata-into-phyloseq-object.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pre-processing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/04-Quality-Control.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Data analysis based on XMAS package.pdf", "Data analysis based on XMAS package.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
